{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "6ecd8ee0",
   "metadata": {},
   "source": [
    "# NER - extraction and visualisation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2918251a",
   "metadata": {},
   "source": [
    "## Pre-requisite"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d694a710",
   "metadata": {},
   "source": [
    "### gnfinder"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5306e3fa",
   "metadata": {},
   "source": [
    "    1. Download the latest version at: https://github.com/gnames/gnfinder/releases\n",
    "    2. Dezip the file and put the gnfinder.exe in a folder (e.g. \"C/bin/\"\") \n",
    "\n",
    "More info (installation, package): https://github.com/gnames/gnfinder"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "091edeb5",
   "metadata": {},
   "source": [
    "### spacy\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "021d7bd4",
   "metadata": {},
   "source": [
    "    1. Install the package in the Anaconda Prompt command line: pip install spacy\n",
    "    2. Install the model en_core_web_lg through the Anacaond Prompt command line: python -m spacy download en_core_web_lg    \n",
    "  \n",
    " More info: https://spacy.io/models/en"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a506c6d6",
   "metadata": {},
   "source": [
    "## Import"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eb5b2368",
   "metadata": {},
   "source": [
    "### Packages and functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "526e8f3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import requests\n",
    "import regex as re\n",
    "import pandas as pd\n",
    "import matplotlib\n",
    "import itertools\n",
    "import inflect\n",
    "import spacy\n",
    "from itertools import chain\n",
    "from pygbif import name_lookup\n",
    "import webbrowser\n",
    "\n",
    "nlp = spacy.load(\"en_core_web_lg\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "feb02151",
   "metadata": {},
   "outputs": [],
   "source": [
    "def nercolor(out):\n",
    "   \n",
    "    # list of colours to highlight the entities\n",
    "    d = {\"location\": \"#9FC9C9\", \"date\" :\"#C7E5C3\", \"time_range\": \"#eec9db\", \"dataset_type\": \"#DB97C7\", \n",
    "         \"scientific_name\" : \"#FFCCA1\", \"vernacular_name\":\"#F5878B\" }\n",
    "    \n",
    "    strh = \"<html><div style='width:70%;display:inline-block'>\"\n",
    "    for i,j in out[\"tagged_text\"]:\n",
    "        if j == 'other':\n",
    "            strh += i\n",
    "            strh += \" \"\n",
    "        if j != 'other':\n",
    "            strh += \"<strong><span style='background-color:%s'>\" % d[j]\n",
    "            strh += i\n",
    "            strh += \"</span></strong>\"\n",
    "            strh += \" \"\n",
    "    strh += \"</div></html>\"\n",
    "\n",
    "    #Legends\n",
    "    stri = '<div style=\"width:20%;display:inline-block\">'\n",
    "    stri += \"<table>\"\n",
    "    for j, i in d.items():\n",
    "        stri += '<tr>'\n",
    "        stri += \"<td><span style='background-color: %s'>__________</span>&nbsp;</td>\" %i\n",
    "        stri += \"<td>%s</td>\" %j\n",
    "        stri += '</tr>'\n",
    "    stri += \"</table></div>\"\n",
    "    \n",
    "    #taxo\n",
    "    taxo = '<div style=\"width:40%;display:inline-block\">'\n",
    "    taxo += \"<p>\"\n",
    "    for k in out[\"taxonomy\"].keys():\n",
    "        for j, i in out[\"taxonomy\"][k].items():\n",
    "            j2 = j.replace(\"_\", \" \").capitalize()\n",
    "            taxo += \"<strong>%s: </strong>\" %j2\n",
    "            taxo += \"%s <br>\" %i\n",
    "    taxo += \"</p>\"\n",
    "    new = stri+strh + taxo\n",
    "    log_file = open('article_ner.html', 'wb+')\n",
    "    log_file.write(new.encode('utf-8'))\n",
    "    return webbrowser.open('article_ner.html')\n",
    "\n",
    "def tag_entities(text, vocab_dataset = [], vocab_scientific_names = [], vocab_vernacular_names = [], taxonomy = []):\n",
    "    locations = detect_locations(text, type_ent = True, start = True)\n",
    "    date_range = detect_time_range(text, type_ent = True, start = True, norm=False)\n",
    "    date_punc = detect_dates(text)\n",
    "    dates = []\n",
    "    if len(date_range) >0:\n",
    "        for i in date_punc:\n",
    "            inc = True\n",
    "            for j in date_range:\n",
    "                if (i[2]==j[2]):\n",
    "                    inc = False\n",
    "            if inc & (i not in dates):\n",
    "                dates.append(i)\n",
    "\n",
    "    else:\n",
    "        dates = date_punc\n",
    "    dataset_type = detect_dataset_type(text, vocab_dataset)\n",
    "    scientific_name = detect_keywords(text, kw = vocab_scientific_names, type_ent= True, ent_value = \"scientific_name\", start = True)\n",
    "    vernacular_name = detect_keywords(text, kw = vocab_vernacular_names, type_ent= True, ent_value = \"vernacular_name\", start = True)\n",
    "    \n",
    "    # extract taxonomy\n",
    "    set_vernacular_name = list(set([i[0] for i in vernacular_name]))\n",
    "    set_vernacular_name = list(set(set_vernacular_name).intersection(list(taxonomy.keys())))\n",
    "    taxo = dict((k, taxonomy[k]) for k in set_vernacular_name)\n",
    "    \n",
    "    not_loc = list(set([x[0].lower() for x in scientific_name] + [x[0].lower() for x in vernacular_name]))\n",
    "    locations = [x for x in locations if x[0].lower() not in not_loc]\n",
    "    entities = locations + date_range + dates + dataset_type + scientific_name + vernacular_name\n",
    "    elm = 2\n",
    "    entities.sort(key=lambda x: x[elm])\n",
    "    entities =  list(k for k,_ in itertools.groupby(entities))\n",
    "    cut_id = vec_entities(entities)\n",
    "    cut1 = []\n",
    "    for i in entities:\n",
    "            n = len(i[0])\n",
    "            cut1.append(i[2])\n",
    "            cut1.append(i[2] + n)\n",
    "    cut1.append(len(text))\n",
    "    cut1 = list(set(cut1))\n",
    "    cut1.sort()\n",
    "    cut2 = [0] + cut1[:-1]\n",
    "    n = [a1 - a2 for a1, a2 in zip(cut1, cut2)]\n",
    "    text_sep = [text[sum(n[:i]):sum(n[:i+1])] for i in range(len(n))]\n",
    "    tagged_entities = {\"tagged_text\": [(a1, a2) for a1, a2 in zip(text_sep, cut_id)], \"taxonomy\": taxo}\n",
    "    return(tagged_entities)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ecfeadb5",
   "metadata": {},
   "outputs": [],
   "source": [
    "def vec_entities(l_entities):\n",
    "    for i, val in enumerate(l_entities):\n",
    "        if (i == 0):\n",
    "            if val[2] != 0:\n",
    "                vec = [\"other\", val[1]]\n",
    "            else:\n",
    "                vec = val[2]\n",
    "        else:\n",
    "            end_prev = l_entities[i-1][2] + len(l_entities[i-1][0])\n",
    "            if ((val[2] >= end_prev + 1) & (vec[-1] != \"other\")):\n",
    "                vec.append(\"other\")\n",
    "            vec.append(val[1])\n",
    "    return(vec)\n",
    "        \n",
    "def detect_locations(text, type_ent = True, start = True):\n",
    "    locations = []\n",
    "    nlp = spacy.load(\"en_core_web_lg\")\n",
    "    doc = nlp(text)\n",
    "    for chunk in doc.noun_chunks:\n",
    "        if chunk.root.text.lower() in [\"canada\", \"québec\", \"quebec\"]:\n",
    "            loc = [chunk.text]\n",
    "            if type_ent:\n",
    "                loc.append(\"location\")\n",
    "            if start:\n",
    "                loc.append(chunk.start_char)  \n",
    "            locations.append((loc))\n",
    "    for e in doc.ents:\n",
    "        if e.label_ in [\"LOC\", \"GPE\"]:\n",
    "            loc = [e.text]\n",
    "            if type_ent:\n",
    "                loc.append(\"location\")\n",
    "            if start:\n",
    "                loc.append(e.start_char)  \n",
    "            locations.append((loc))\n",
    "    #locations = list(set(locations))\n",
    "    locations = list(k for k,_ in itertools.groupby(locations))\n",
    "    return(locations)\n",
    "\n",
    "def detect_dates(text, type_ent = True, start = True, min_y = 1900, max_y = 2023):\n",
    "    nlp = spacy.load(\"en_core_web_sm\")\n",
    "    doc = nlp(text)\n",
    "    dates = []\n",
    "    for e in doc.ents:\n",
    "        if \"DATE\" in e.label_:\n",
    "            if e.text.isdigit():\n",
    "                if ((int(e.text) >= min_y) & (int(e.text) <= max_y)):\n",
    "                    date = [e.text]\n",
    "                    if type_ent:\n",
    "                        date.append(\"date\")\n",
    "                    if start:\n",
    "                        date.append(e.start_char) \n",
    "                    dates.append((date))\n",
    "            elif (\"orcid\" not in text[(max(0, (e.start_char - 10) )):e.start_char].lower()):\n",
    "                date = [e.text]\n",
    "                if type_ent:\n",
    "                    date.append(\"date\")\n",
    "                if start:\n",
    "                    date.append(e.start_char) \n",
    "                dates.append((date))\n",
    "    return(dates)\n",
    "\n",
    "def detect_time_range(text, type_ent = True, start = True, norm = True, min_y = 1900, max_y = 2023):\n",
    "    nlp = spacy.load(\"en_core_web_sm\")\n",
    "    doc = nlp(text)\n",
    "    time_range = []\n",
    "    \n",
    "    # use Spacy duration entities\n",
    "    for e in doc.ents:\n",
    "        if \"DURATION\" in e.kb_id_:\n",
    "            r = r\"(?<=value=).*\"\n",
    "            val =  re.search(r, e.kb_id_).group(0).replace('\"', '')\n",
    "            time = [e.text]\n",
    "            if type_ent:\n",
    "                time.append(\"time_range\")\n",
    "            if start:\n",
    "                time.append(e.start_char) \n",
    "            if norm:\n",
    "                time.append(val) \n",
    "            time_range.append((time))\n",
    "\n",
    "    # use custom built regex\n",
    "    regex_intervals = [r\"(\\d{4}) ?(to|-) ?(\\d{4})\", #  \"2004-2005\", \"2019 to 2021\"\n",
    "                    r\"between (\\d{4}) (and) (\\d{4})\"]  #  \"betwwen 2004 and 2021\"\n",
    "    \n",
    "    regex_durations = [r\"over (\\d{1,4}) ([[:alpha:]]*)( *)year[s]?\", #\"over 7 years\", \"over 5 consecutive years\"\n",
    "      r\"(\\d{1,4})( |-)(year)[s]?\"]   #  \"7-years study\"\n",
    "    \n",
    "    for reg in regex_intervals:\n",
    "        result = re.finditer(reg, text.lower())\n",
    "        for res in result:\n",
    "            if ((int(res.group(3)) >= min_y) & (int(res.group(3)) <= max_y)):\n",
    "                val = \"P{}Y\".format(int(res.group(3)) - int(res.group(1)))\n",
    "\n",
    "                time = [res.group(0)]            \n",
    "                if type_ent:\n",
    "                    time.append(\"time_range\")\n",
    "                if start:\n",
    "                    time.append(res.start(0)) \n",
    "                if norm:\n",
    "                    time.append(val) \n",
    "                time_range.append((time))\n",
    "\n",
    "    for reg in regex_durations:\n",
    "        result = re.finditer(reg, text.lower())\n",
    "        for res in result:\n",
    "            val = \"P{}Y\".format(int(res.group(1)))\n",
    "            \n",
    "            time = [res.group(0)]            \n",
    "            if type_ent:\n",
    "                time.append(\"time_range\")\n",
    "            if start:\n",
    "                time.append(res.start(0)) \n",
    "            if norm:\n",
    "                time.append(val) \n",
    "            time_range.append((time))\n",
    "    return(time_range)\n",
    "\n",
    "def detect_dataset_type(text, kw, type_ent= True, start = True):\n",
    "    #text = re.sub('[^0-9a-zA-Z]', ' ', text.lower())\n",
    "    text = text.lower()\n",
    "    kw = [k.lower() for k in kw]\n",
    "    dataset_types = []\n",
    "    for k in kw:\n",
    "        #regs = [r\" ({}) \".format(k), r\" \" +  r\" ({}) \".format(get_plural(k))]\n",
    "        regs = [r\"({}[a-z]?)( )\".format(k), r\"({})( )\".format(get_plural(k))]\n",
    "        for reg in regs:\n",
    "            result = re.finditer(reg, text)\n",
    "            for res in result:\n",
    "                kws = [res.group(0).strip()]           \n",
    "                if type_ent:\n",
    "                    kws.append(\"dataset_type\")\n",
    "                if start:\n",
    "                    kws.append(res.start(0)) \n",
    "                dataset_types.append((kws))\n",
    "    return(dataset_types)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8c297c9d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def detect_keywords(text, kw, type_ent= True, ent_value = \"\", start = True):\n",
    "    text = text.lower()\n",
    "    kw = [k.lower() for k in kw]\n",
    "    keywords = []\n",
    "    for k in kw:\n",
    "        reg = r\"({}[a-z]?)\".format(k)\n",
    "        result = re.finditer(reg, text)\n",
    "        for res in result:\n",
    "            kws = [res.group(0)]           \n",
    "            if type_ent:\n",
    "                kws.append(ent_value)\n",
    "            if start:\n",
    "                kws.append(res.start(0)) \n",
    "            keywords.append((kws))\n",
    "    return(keywords)\n",
    "\n",
    "def get_vernacular_names(scientific_name, limit = 20, language = \"eng\"):\n",
    "    out = name_lookup(q=scientific_name, limit = limit)\n",
    "    out = out['results']\n",
    "    res = [z['vernacularNames'][0] for z in out if len(z['vernacularNames'])>0]\n",
    "    \n",
    "    kingdom, classx, order, family, genus = \"\", \"\", \"\", \"\", \"\"\n",
    "    for i in out:\n",
    "        if (kingdom == \"\" and \"kingdom\" in i.keys()):\n",
    "            kingdom = i[\"kingdom\"]\n",
    "        if (classx == \"\" and \"class\" in i.keys()):\n",
    "            classx = i[\"class\"]\n",
    "        if (order == \"\" and \"order\" in i.keys()):\n",
    "            order = i[\"order\"]\n",
    "        if (family == \"\" and \"family\" in i.keys()):\n",
    "            family = i[\"family\"]\n",
    "        if (genus == \"\" and \"genus\" in i.keys()):\n",
    "            genus = i[\"genus\"]\n",
    "    res = {\"scientific_name\": scientific_name, \"vernacular_names\" :list(set([z['vernacularName'].lower() for z in res if z['language'] == language])),\n",
    "          \"kingdom\": kingdom, \"class\": classx, 'order': order,\n",
    "  'family': family, 'genus':genus}\n",
    "    return(res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a668babf",
   "metadata": {},
   "outputs": [],
   "source": [
    "def kw_lists(x, lkw):\n",
    "    v = 0\n",
    "    for kw in lkw:\n",
    "        cnt = kw_list(x, kw)\n",
    "        if cnt == 1:\n",
    "            v = 1\n",
    "    return(v)\n",
    "\n",
    "def kw_list(x, kw):\n",
    "    cnt = 0\n",
    "    for k in kw:\n",
    "        k = k.replace('\"', \"\")\n",
    "        if (x.count(k) > 0 or x.count(get_plural(k)) > 0):\n",
    "            cnt = cnt + 1\n",
    "    if cnt == len(kw):\n",
    "        v = 1\n",
    "    else:\n",
    "        v= 0\n",
    "    return(v)\n",
    "\n",
    "def conv_query(x):\n",
    "    l = [y.strip() for y in x.split(\"AND\")]\n",
    "    if 'Québec' in l:\n",
    "        l = [l] + [[w.replace('Québec', 'Quebec') for w in l]]\n",
    "    else:\n",
    "        l = [l]\n",
    "    return(l)\n",
    "\n",
    "def get_plural(sg):\n",
    "    if sg[0].isupper():\n",
    "        pl = sg\n",
    "    elif sg[0] == '\"\"':\n",
    "        pl =sg\n",
    "    else:\n",
    "        engine = inflect.engine()\n",
    "        pl = engine.plural(sg)\n",
    "    return(pl)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b6ccb5ff",
   "metadata": {},
   "source": [
    "### Data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "abc41336",
   "metadata": {},
   "source": [
    "#### Vocabulary used to detect dataset-type related keywords"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "16ae672c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['presence-absence',\n",
       " 'occupancy',\n",
       " 'density',\n",
       " 'camera-trap',\n",
       " 'niche',\n",
       " 'percent cover',\n",
       " 'capture-mark-recapture',\n",
       " 'richness',\n",
       " 'specimen',\n",
       " 'counts',\n",
       " 'transect',\n",
       " 'habitat suitability',\n",
       " 'specimen records',\n",
       " 'relative proportion',\n",
       " 'quadrat',\n",
       " 'collection',\n",
       " 'occurrence',\n",
       " 'distribution',\n",
       " 'presence only',\n",
       " 'species distribution model',\n",
       " 'inventory',\n",
       " 'range',\n",
       " 'abundance',\n",
       " 'sampling plot']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vocabulary = pd.read_csv(\"C://GitHub//biodiv_retrieval//BiodivDatasetRetrieval//data//dataset_types.csv\", sep = \";\")\n",
    "vocabulary_dataset = list(set(vocabulary[\"keywords\"].tolist()))\n",
    "vocabulary_dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ae6d457",
   "metadata": {},
   "source": [
    "#### Dataset to analyse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "10b947f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_datasets = pd.read_excel(\"C://Users//vals3103//Post-doc//Text_mining//df_work.xlsx\", engine = \"openpyxl\")\n",
    "df_datasets[\"full_text\"] = df_datasets['title'] + \" \"+ df_datasets['description'].fillna('') + \" \"+ df_datasets['method'].fillna('')+ \" \"+ df_datasets['notes'].fillna('') +\" \"+  df_datasets['keywords'].fillna('') +\" \"+  df_datasets['locations'].fillna('')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a211e4f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>url</th>\n",
       "      <th>title</th>\n",
       "      <th>description</th>\n",
       "      <th>method</th>\n",
       "      <th>notes</th>\n",
       "      <th>keywords</th>\n",
       "      <th>locations</th>\n",
       "      <th>publication_date</th>\n",
       "      <th>cited_articles</th>\n",
       "      <th>id_query</th>\n",
       "      <th>...</th>\n",
       "      <th>temporal_duration</th>\n",
       "      <th>temporal_duration_position</th>\n",
       "      <th>spatial range</th>\n",
       "      <th>spatial range units</th>\n",
       "      <th>spatial range position</th>\n",
       "      <th>data_type</th>\n",
       "      <th>relevant_data_type</th>\n",
       "      <th>time_series</th>\n",
       "      <th>non_valid</th>\n",
       "      <th>Comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://doi.org/10.5061/dryad.8w9ghx3pf</td>\n",
       "      <td>A mechanistic model of functional response pro...</td>\n",
       "      <td>Prey handling processes are considered a domin...</td>\n",
       "      <td>&lt;p&gt;The datasets (fox_daily_activity and LPS_De...</td>\n",
       "      <td>&lt;div class=\"t-landing__text-wall \"&gt;\\n&lt;p&gt;The RE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2022-03-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.1</td>\n",
       "      <td>...</td>\n",
       "      <td>15 years</td>\n",
       "      <td>main</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not provided</td>\n",
       "      <td>density</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>valid</td>\n",
       "      <td>locatiy mentioned in the README</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>https://doi.org/10.5061/dryad.866t1g1q1</td>\n",
       "      <td>Biodiversity facets, canopy structure and surf...</td>\n",
       "      <td>&lt;ol&gt;\\n\\t&lt;li&gt;\\n&lt;a name=\"_Hlk61288143\"&gt; Canopy s...</td>\n",
       "      <td>&lt;p&gt;&lt;span&gt;&lt;span class=\"objectBox objectBox-stri...</td>\n",
       "      <td>&lt;p&gt;There are no missing values. It follows a R...</td>\n",
       "      <td>surface temperature; terrestrial laser scannin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2021-03-02</td>\n",
       "      <td>https://doi.org/10.5281/zenodo.4548262</td>\n",
       "      <td>0,7</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>other</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>location</td>\n",
       "      <td>Germany. \"Quebec\" comes only from author affil...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>https://doi.org/10.5061/dryad.s1rn8pk7d</td>\n",
       "      <td>Boreal aspen understory diversity along a cont...</td>\n",
       "      <td>&lt;p&gt;This dataset contains vascular plant specie...</td>\n",
       "      <td>&lt;p&gt;Sampling took place in 33 trembling aspen (...</td>\n",
       "      <td>&lt;p&gt;Associated paper: Crispo, Jean, Fenton, Led...</td>\n",
       "      <td>understory vegetation; trembling aspen; plant ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2021-06-09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4,5,7,8,0,9</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Canada</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>abundance, richness, presence-absence</td>\n",
       "      <td>1.0</td>\n",
       "      <td>no</td>\n",
       "      <td>valid</td>\n",
       "      <td>percent cover</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>https://doi.org/10.5061/dryad.vt4b8gtnv</td>\n",
       "      <td>Copy number variants outperform SNPs to reveal...</td>\n",
       "      <td>&lt;p&gt;Copy number variants (CNVs) are a major com...</td>\n",
       "      <td>&lt;p align=\"justify\" class=\"western\" style=\"text...</td>\n",
       "      <td>&lt;p&gt;&lt;strong&gt;Raw VCF consisted of 44374 unfilter...</td>\n",
       "      <td>CNVs; copy number variants; SNPs; American lob...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020-08-19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6,7,8</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not provided</td>\n",
       "      <td>EBV genetic analysis</td>\n",
       "      <td>0.0</td>\n",
       "      <td>no</td>\n",
       "      <td>valid</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>https://doi.org/10.5061/dryad.8cz8w9gr1</td>\n",
       "      <td>Data for the detection of the boreal chorus fr...</td>\n",
       "      <td>&lt;p&gt;The boreal chorus frog (&lt;em&gt;Pseudacris macu...</td>\n",
       "      <td>&lt;p&gt;The data presented in the files include the...</td>\n",
       "      <td>&lt;p&gt;The values presented in the data set are th...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2021-09-13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0,7,5,8</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2250</td>\n",
       "      <td>km2</td>\n",
       "      <td>source_publication</td>\n",
       "      <td>presence-absence</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>valid</td>\n",
       "      <td>sampling location in the main publication. In ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       url  \\\n",
       "0  https://doi.org/10.5061/dryad.8w9ghx3pf   \n",
       "1  https://doi.org/10.5061/dryad.866t1g1q1   \n",
       "2  https://doi.org/10.5061/dryad.s1rn8pk7d   \n",
       "3  https://doi.org/10.5061/dryad.vt4b8gtnv   \n",
       "4  https://doi.org/10.5061/dryad.8cz8w9gr1   \n",
       "\n",
       "                                               title  \\\n",
       "0  A mechanistic model of functional response pro...   \n",
       "1  Biodiversity facets, canopy structure and surf...   \n",
       "2  Boreal aspen understory diversity along a cont...   \n",
       "3  Copy number variants outperform SNPs to reveal...   \n",
       "4  Data for the detection of the boreal chorus fr...   \n",
       "\n",
       "                                         description  \\\n",
       "0  Prey handling processes are considered a domin...   \n",
       "1  <ol>\\n\\t<li>\\n<a name=\"_Hlk61288143\"> Canopy s...   \n",
       "2  <p>This dataset contains vascular plant specie...   \n",
       "3  <p>Copy number variants (CNVs) are a major com...   \n",
       "4  <p>The boreal chorus frog (<em>Pseudacris macu...   \n",
       "\n",
       "                                              method  \\\n",
       "0  <p>The datasets (fox_daily_activity and LPS_De...   \n",
       "1  <p><span><span class=\"objectBox objectBox-stri...   \n",
       "2  <p>Sampling took place in 33 trembling aspen (...   \n",
       "3  <p align=\"justify\" class=\"western\" style=\"text...   \n",
       "4  <p>The data presented in the files include the...   \n",
       "\n",
       "                                               notes  \\\n",
       "0  <div class=\"t-landing__text-wall \">\\n<p>The RE...   \n",
       "1  <p>There are no missing values. It follows a R...   \n",
       "2  <p>Associated paper: Crispo, Jean, Fenton, Led...   \n",
       "3  <p><strong>Raw VCF consisted of 44374 unfilter...   \n",
       "4  <p>The values presented in the data set are th...   \n",
       "\n",
       "                                            keywords locations  \\\n",
       "0                                                NaN       NaN   \n",
       "1  surface temperature; terrestrial laser scannin...       NaN   \n",
       "2  understory vegetation; trembling aspen; plant ...       NaN   \n",
       "3  CNVs; copy number variants; SNPs; American lob...       NaN   \n",
       "4                                                NaN       NaN   \n",
       "\n",
       "  publication_date                          cited_articles     id_query  ...  \\\n",
       "0       2022-03-16                                     NaN         10.1  ...   \n",
       "1       2021-03-02  https://doi.org/10.5281/zenodo.4548262          0,7  ...   \n",
       "2       2021-06-09                                     NaN  4,5,7,8,0,9  ...   \n",
       "3       2020-08-19                                     NaN        6,7,8  ...   \n",
       "4       2021-09-13                                     NaN      0,7,5,8  ...   \n",
       "\n",
       "  temporal_duration temporal_duration_position  spatial range  \\\n",
       "0          15 years                       main            NaN   \n",
       "1               NaN                        NaN            NaN   \n",
       "2               NaN                        NaN         Canada   \n",
       "3               NaN                        NaN            NaN   \n",
       "4               NaN                        NaN           2250   \n",
       "\n",
       "   spatial range units spatial range position  \\\n",
       "0                  NaN           not provided   \n",
       "1                  NaN                    NaN   \n",
       "2                  NaN                    NaN   \n",
       "3                  NaN           not provided   \n",
       "4                  km2     source_publication   \n",
       "\n",
       "                               data_type relevant_data_type time_series  \\\n",
       "0                                density                1.0         yes   \n",
       "1                                  other                0.0         NaN   \n",
       "2  abundance, richness, presence-absence                1.0          no   \n",
       "3                   EBV genetic analysis                0.0          no   \n",
       "4                       presence-absence                1.0         yes   \n",
       "\n",
       "  non_valid                                           Comments  \n",
       "0     valid                    locatiy mentioned in the README  \n",
       "1  location  Germany. \"Quebec\" comes only from author affil...  \n",
       "2     valid                                      percent cover  \n",
       "3     valid                                                NaN  \n",
       "4     valid  sampling location in the main publication. In ...  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_datasets.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d23408e",
   "metadata": {},
   "source": [
    "# Named Entities Recognition "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb0440b0",
   "metadata": {},
   "source": [
    "## Scientific names"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e63d939d",
   "metadata": {},
   "source": [
    "We detect the scientific names with the package gnfinder, which combines machine learning and dictionnary approaches.\n",
    "We first retrieve all the full texts from our dataset into a list called texts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a8fd3b65",
   "metadata": {},
   "outputs": [],
   "source": [
    "texts = df_datasets[\"full_text\"].tolist()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd665d72",
   "metadata": {},
   "source": [
    "We join all the texts in one piece of string and export them as a .txt file, to be used an an input for gnfinder. We export them into the folder containing the gnfinder.exe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a1409941",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\bin\n"
     ]
    }
   ],
   "source": [
    "all_texts = \".\".join(texts)\n",
    "with open('C://bin//text.txt', 'w', encoding=\"utf-8\") as f:\n",
    "    f.write(all_texts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4417abef",
   "metadata": {},
   "source": [
    "We set the working directory to the folder containing gnfinder.exe and the text.txt file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "47104f17",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\bin\n"
     ]
    }
   ],
   "source": [
    "%cd \"C:\\bin\\\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4248d251",
   "metadata": {},
   "source": [
    "We run gnfinder on the .txt file. The detected scientific names are exported to a json file from which we can retrieve the list of names."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "195119ac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Microsoft Windows [version 10.0.19044.2130]\r\n",
      "(c) Microsoft Corporation. Tous droits r‚serv‚s.\r\n",
      "\r\n",
      "(base) C:\\bin>for /r %i in (*.txt) do gnfinder %i -U > scientific_names.json\n",
      "\r\n",
      "(base) C:\\bin>gnfinder C:\\bin\\text.txt -U  1>scientific_names.json \r\n",
      "\r\n",
      "(base) C:\\bin>"
     ]
    }
   ],
   "source": [
    "%%cmd\n",
    "for /r %i in (*.txt) do gnfinder %i -U > scientific_names.json"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1d7c055b",
   "metadata": {},
   "source": [
    "We can now open the scientific names stored into a json file and store them into a list. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e0feac0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "scientific_names = pd.read_csv(\"C://bin//scientific_names.json\")[\"Name\"].tolist()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a4a2eb54",
   "metadata": {},
   "source": [
    "The function Counter is a convenient way to count the number of occurrences of each element in a list. The output is a dictionary in which the keys are each unique element from the list (here, the scientific names) and the value is the number of occurence. If we add the method \"most_common()\", the dictionnary is sorted by decreasing value of the frequency."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "dccf98c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('Protomognathus americanus', 1),\n",
       " ('Tcherskidium tenuicostatum', 1),\n",
       " ('Onnia tomentosa', 1),\n",
       " ('Canis', 1),\n",
       " ('Pinus uncinata', 1),\n",
       " ('Vaccinium myrtilloides', 1),\n",
       " ('Cassiope', 1),\n",
       " ('Ichneumonidae', 1),\n",
       " ('Blenniidae', 1),\n",
       " ('Rallus limicola', 1),\n",
       " ('Polyphemidae', 1),\n",
       " ('Carex oligosperma', 1),\n",
       " ('Bombus bimaculatus', 1),\n",
       " ('Canis lycaon', 1),\n",
       " ('Andrena bradleyi', 1),\n",
       " ('Viridiplantae', 1),\n",
       " ('Agrilus planipennis', 1),\n",
       " ('Augochlorella aurata', 1),\n",
       " ('Calanus hyperboreus', 1),\n",
       " ('Andrena regularis', 1),\n",
       " ('Betula papyrifera', 1),\n",
       " ('Procyon lotor', 1),\n",
       " ('Andrena erythronii', 1),\n",
       " ('Castor canadensis', 1),\n",
       " ('Ceratina dupla', 1),\n",
       " ('Eptesicus fuscus', 1),\n",
       " ('Lasioglossum sagax', 1),\n",
       " ('Odocoileus virginianus', 1),\n",
       " ('Lasionycteris noctivagans', 1),\n",
       " ('Andrena duningii', 1),\n",
       " ('Coleoptera', 1),\n",
       " ('Andrena barbilabris', 1),\n",
       " ('Andrena nivalis', 1),\n",
       " ('Plectrophenax nivalis', 1),\n",
       " ('Nothofagus solandri', 1),\n",
       " ('Bombus vagans', 1),\n",
       " ('Staphylinidae', 1),\n",
       " ('V. rossicum', 1),\n",
       " ('Lasioglossum nigroviride', 1),\n",
       " ('Globodera rostochiensis', 1),\n",
       " ('Concordia', 1),\n",
       " ('P. deltoides', 1),\n",
       " ('Ceratina calcarata', 1),\n",
       " ('Daphniidae', 1),\n",
       " ('Kalmia angustifolia', 1),\n",
       " ('Parada', 1),\n",
       " ('Betula alleghaniensis', 1),\n",
       " ('Carex gynocrates', 1),\n",
       " ('Apanteles fumiferanae', 1),\n",
       " ('Mya arenaria', 1),\n",
       " ('Lasioglossum coriaceum', 1),\n",
       " ('Augochlora pura', 1),\n",
       " ('Lasioglossum quebecense', 1),\n",
       " ('Araneae', 1),\n",
       " ('Exoglossum maxillingua', 1),\n",
       " ('Texella', 1),\n",
       " ('Eptesicus', 1),\n",
       " ('Ophiuroids', 1),\n",
       " ('Catostomus commersoni', 1),\n",
       " ('Andrena bisalicis', 1),\n",
       " ('Picea glauca', 1),\n",
       " ('Salix', 1),\n",
       " ('Burkholderiaceae', 1),\n",
       " ('Malascosoma disstria', 1),\n",
       " ('Acer pseudoplatanus', 1),\n",
       " ('Vario', 1),\n",
       " ('Laminaria digitata', 1),\n",
       " ('Vitales', 1),\n",
       " ('Dms stigmas', 1),\n",
       " ('Rhinichthys atratulus', 1),\n",
       " ('Amyot', 1),\n",
       " ('Clangula hyemalis', 1),\n",
       " ('Galapagos', 1),\n",
       " ('P. balsamifera', 1),\n",
       " ('Abies amabilis', 1),\n",
       " ('Prumnopitys ferruginea', 1),\n",
       " ('P. maximowiczii', 1),\n",
       " ('Acer pensylvanicum', 1),\n",
       " ('Lepidoptera', 1),\n",
       " ('Calidris fuscicollis', 1),\n",
       " ('Rangifer tarandus', 1),\n",
       " ('Chydoridae', 1),\n",
       " ('Camponotus', 1),\n",
       " ('Chen caerulescens atlantica', 1),\n",
       " ('Hoeck', 1),\n",
       " ('Populus nigra', 1),\n",
       " ('Somateria mollissima dresseri', 1),\n",
       " ('Anser caerulescens atlantica', 1),\n",
       " ('Andrena forbesii', 1),\n",
       " ('Enallagma', 1),\n",
       " ('Hesperorthis pyramidalis', 1),\n",
       " ('C. hyperboreus', 1),\n",
       " ('Fraxinus excelsior', 1),\n",
       " ('Eriophorum viridicarinatum', 1),\n",
       " ('Calidris bairdii', 1),\n",
       " ('Oncorhynchus kisutch', 1),\n",
       " ('Cucurbitaceae', 1),\n",
       " ('Pinus strobus', 1),\n",
       " ('Ambloplites rupestris', 1),\n",
       " ('Vibrio cholerae', 1),\n",
       " ('Tsuga heterophylla', 1),\n",
       " ('Protomagnatus americanus', 1),\n",
       " ('Poecile', 1),\n",
       " ('Sinocalycanthus chinensis', 1),\n",
       " ('Agapostemon virescens', 1),\n",
       " ('Andrena vicina', 1),\n",
       " ('Magnoliids', 1),\n",
       " ('Megasporaceae', 1),\n",
       " ('Mallotus villosus', 1),\n",
       " ('Carex echinata', 1),\n",
       " ('Melanitta fusca', 1),\n",
       " ('Brevilamnulella minuta', 1),\n",
       " ('Pheidole', 1),\n",
       " ('Quercus rubra', 1),\n",
       " ('Pseudotsuga menziesii var. menziesii', 1),\n",
       " ('Alexa', 1),\n",
       " ('Andrena cressonii', 1),\n",
       " ('Agapostemon texanus', 1),\n",
       " ('Carex pauciflora', 1),\n",
       " ('Enallagma ebrium', 1),\n",
       " ('Larvae', 1),\n",
       " ('Osmia atriventris', 1),\n",
       " ('Calanus', 1),\n",
       " ('Aleocharinae', 1),\n",
       " ('Weinmannia racemosa', 1),\n",
       " ('Oncorhynchus tshawytscha', 1),\n",
       " ('Caribou rangifer tarandus', 1),\n",
       " ('Sina', 1),\n",
       " ('Bacteroidetes', 1),\n",
       " ('Lasioglossum laevissimum', 1),\n",
       " ('Coregonus clupeaformis', 1),\n",
       " ('Eoplectodonta', 1),\n",
       " ('Umbra limi', 1),\n",
       " ('S. namaycush', 1),\n",
       " ('Alitta virens', 1),\n",
       " ('Carex tsiangii', 1),\n",
       " ('Osmia pumila', 1),\n",
       " ('Betula glandulosa', 1),\n",
       " ('Agrilus', 1),\n",
       " ('Salvelinus fontinalis', 1),\n",
       " ('Macoma balthica', 1),\n",
       " ('Carex', 1),\n",
       " ('Carex tenuiflora', 1),\n",
       " ('Fraxinus americana', 1),\n",
       " ('Abies balsamea', 1),\n",
       " ('Eubalaena', 1),\n",
       " ('Plaesiomys occidentalis', 1),\n",
       " ('Osmia proxima', 1),\n",
       " ('Quercus', 1),\n",
       " ('Picea rubens', 1),\n",
       " ('Semotilus atromaculatus', 1),\n",
       " ('Heriades carinata', 1),\n",
       " ('Trichophorum alpinum', 1),\n",
       " ('Proteobacteria', 1),\n",
       " ('Myotis lucifugus', 1),\n",
       " ('Andrena wilkella', 1),\n",
       " ('Alces americanus', 1),\n",
       " ('Morone saxatilis', 1),\n",
       " ('Cottus bairdi', 1),\n",
       " ('Salmo salar', 1),\n",
       " ('Pinus banksiana', 1),\n",
       " ('Andrena rufosignata', 1),\n",
       " ('Microtus chrotorrhinus', 1),\n",
       " ('Laminaria rodriguezii', 1),\n",
       " ('Nothofagus menziesii', 1),\n",
       " ('Caryophyllales', 1),\n",
       " ('Carex chordorrhiza', 1),\n",
       " ('Acer rubrum', 1),\n",
       " ('C. finmarchicus', 1),\n",
       " ('Whitfieldella', 1),\n",
       " ('Phalaropus fulicarius', 1),\n",
       " ('Dryas', 1),\n",
       " ('Carex canescens', 1),\n",
       " ('Calidris canutus', 1),\n",
       " ('Asterids', 1),\n",
       " ('Dendragapus canadensis', 1),\n",
       " ('Fraxinus nigra', 1),\n",
       " ('Glaucomys sabrinus', 1),\n",
       " ('M. septentrionalis', 1),\n",
       " ('P. tremuloides', 1),\n",
       " ('Andrena carolina', 1),\n",
       " ('Acer spicatum', 1),\n",
       " ('Andrena milwaukeensis', 1),\n",
       " ('Pinus contorta ssp. latifolia', 1),\n",
       " ('Cyperaceae', 1),\n",
       " ('Holcopasites calliopsidis', 1),\n",
       " ('Mephitis mephitis', 1),\n",
       " ('C. canadensis', 1),\n",
       " ('Prunus serotina', 1),\n",
       " ('Bombus borealis', 1),\n",
       " ('Betula ermanii', 1),\n",
       " ('Rosids', 1),\n",
       " ('Tachycineta bicolor', 1),\n",
       " ('Vincetoxicum rossicum', 1),\n",
       " ('Alosa sapidissima', 1),\n",
       " ('Andrena nasonii', 1),\n",
       " ('Aspicilia', 1),\n",
       " ('Quercus robur', 1),\n",
       " ('Santalales', 1),\n",
       " ('Fagus sylvatica', 1),\n",
       " ('Andrena thaspii', 1),\n",
       " ('Vulpes lagopus', 1),\n",
       " ('Meiofauna', 1),\n",
       " ('Hylaeus modestus', 1),\n",
       " ('Somateria mollissima', 1),\n",
       " ('Chiroptera', 1),\n",
       " ('Stercorarius parasiticus', 1),\n",
       " ('Hylaeus mesillae', 1),\n",
       " ('Phalangodidae', 1),\n",
       " ('Populus deltoides', 1),\n",
       " ('Opiliones', 1),\n",
       " ('Populus trichocarpa', 1),\n",
       " ('Pseudacris maculata', 1),\n",
       " ('Carex aquatilis', 1),\n",
       " ('Arenaria interpres', 1),\n",
       " ('Alissa', 1),\n",
       " ('Andrena commoda', 1),\n",
       " ('Carex saxatilis', 1),\n",
       " ('Rhynchotrema iowense', 1),\n",
       " ('Merra', 1),\n",
       " ('Escherichia coli', 1),\n",
       " ('Chamaedaphne calyculata', 1),\n",
       " ('Ceratina mikmaqi', 1),\n",
       " ('Myotis', 1),\n",
       " ('Lasioglossum zonulum', 1),\n",
       " ('Larus hyperboreus', 1),\n",
       " ('Pluvialis dominica', 1),\n",
       " ('Gowardia', 1),\n",
       " ('Andrena w-scripta', 1),\n",
       " ('Andrena miranda', 1),\n",
       " ('Bombus', 1),\n",
       " ('Juglans cinerea', 1),\n",
       " ('Ogasawara', 1),\n",
       " ('Bombus griseocollis', 1),\n",
       " ('Histrionicus histrionicus', 1),\n",
       " ('Calidris melanotos', 1),\n",
       " ('Cynara', 1),\n",
       " ('Melampsora larici-populina', 1),\n",
       " ('Oxalis montana', 1),\n",
       " ('Populus balsamifera', 1),\n",
       " ('Thuja occidentalis', 1),\n",
       " ('Acer campestre', 1),\n",
       " ('O. tomentosa', 1),\n",
       " ('E. americanum', 1),\n",
       " ('Anolis', 1),\n",
       " ('Lasioglossum foxii', 1),\n",
       " ('Perca flavescens', 1),\n",
       " ('Temnothorax longispinosus', 1),\n",
       " ('Crocodylus acutus', 1),\n",
       " ('Tamias striatus', 1),\n",
       " ('Hymenoptera', 1),\n",
       " ('Trillium', 1),\n",
       " ('Lasioglossum cressonii', 1),\n",
       " ('Glypta fumiferanae', 1),\n",
       " ('Choristoneura fumiferana', 1),\n",
       " ('Anthophila', 1),\n",
       " ('Vaccinium', 1),\n",
       " ('Poecile hudsonicus', 1),\n",
       " ('Carex leptalea', 1),\n",
       " ('Clintonia borealis', 1),\n",
       " ('Trichophorum', 1),\n",
       " ('Loiseleuria procumbens', 1),\n",
       " ('Prisma', 1),\n",
       " ('Carex exilis', 1),\n",
       " ('Maianthemum canadense', 1),\n",
       " ('Fraxinus', 1),\n",
       " ('Bouteloua gracilis', 1),\n",
       " ('Thuringia', 1),\n",
       " ('Lepidoptera versus', 1),\n",
       " ('Hoplitis producta', 1),\n",
       " ('Etheostoma nigrum', 1),\n",
       " ('Poaceae', 1),\n",
       " ('Homarus americanus', 1),\n",
       " ('Cervus', 1),\n",
       " ('Liliaceae', 1),\n",
       " ('Apidae', 1),\n",
       " ('Calycanthaceae', 1),\n",
       " ('Pluvialis squatarola', 1),\n",
       " ('Protomagnathus americanus', 1),\n",
       " ('Podocarpus hallii', 1),\n",
       " ('M. medusae', 1),\n",
       " ('Thuja plicata', 1),\n",
       " ('Brown�s juglans', 1),\n",
       " ('Ericaceae', 1),\n",
       " ('Carex pachygyna', 1),\n",
       " ('Leptaena', 1),\n",
       " ('Crataegus monogyna', 1),\n",
       " ('Andrena perplexa', 1),\n",
       " ('Actinobacteria', 1),\n",
       " ('Ursus americanus', 1),\n",
       " ('Bouteloua dactyloides', 1),\n",
       " ('Carex siderosticta', 1),\n",
       " ('Rubus', 1),\n",
       " ('Saxifragales', 1),\n",
       " ('Carex disperma', 1),\n",
       " ('Trillium erectum', 1),\n",
       " ('Chamaenerion angustifolium', 1),\n",
       " ('Carex gibba', 1),\n",
       " ('Leptodoridae', 1),\n",
       " ('Collembola', 1),\n",
       " ('Bosminidae', 1),\n",
       " ('Fagus grandifolia', 1),\n",
       " ('T. longispinosus', 1),\n",
       " ('Fugu', 1),\n",
       " ('Anguilla rostrata', 1),\n",
       " ('Lasioglossum pilosum', 1),\n",
       " ('Carex heleonastes', 1),\n",
       " ('Rangifer', 1),\n",
       " ('Cladocera', 1),\n",
       " ('Eriophorum angustifolium', 1),\n",
       " ('Sergio', 1),\n",
       " ('Cornus canadensis', 1),\n",
       " ('Betula pendula', 1),\n",
       " ('Lynx rufus', 1),\n",
       " ('Andrena miserabilis', 1),\n",
       " ('Peromyscus maniculatus', 1),\n",
       " ('Tringa flavipes', 1),\n",
       " ('Fagaceae', 1),\n",
       " ('Bombus perplexus', 1),\n",
       " ('Pinus sylvestris', 1),\n",
       " ('Lasioglossum cinctipes', 1),\n",
       " ('Acer mono', 1),\n",
       " ('Temnothorax curvispinosus', 1),\n",
       " ('Phragmites australis', 1),\n",
       " ('Dryas data', 1),\n",
       " ('Lasioglossum ephialtum', 1),\n",
       " ('Alces alces', 1),\n",
       " ('Anacardiaceae', 1),\n",
       " ('Ixodes scapularis', 1),\n",
       " ('Canis lupus familiaris', 1),\n",
       " ('Valle', 1),\n",
       " ('Paridae', 1),\n",
       " ('Abies lasiocarpa', 1),\n",
       " ('Basa', 1),\n",
       " ('Arachnida', 1),\n",
       " ('Hylaeus affinis', 1),\n",
       " ('Alitta', 1),\n",
       " ('Lasioglossum lineatulum', 1),\n",
       " ('Synaptomys', 1),\n",
       " ('Melampsora', 1),\n",
       " ('M. perspicillata', 1),\n",
       " ('Salvelinus', 1),\n",
       " ('Corylus avellana', 1),\n",
       " ('Otus', 1),\n",
       " ('Nereis', 1),\n",
       " ('Salvelinus namaycush', 1),\n",
       " ('Centaurea diffusa', 1),\n",
       " ('Glyptemys insculpta', 1),\n",
       " ('Tsuga canadensis', 1),\n",
       " ('Andrena imitatrix', 1),\n",
       " ('Dryas heath', 1),\n",
       " ('Carex rostrata', 1),\n",
       " ('Halictus confusus', 1),\n",
       " ('Hoplitis spoliata', 1),\n",
       " ('Macrothricidae', 1),\n",
       " ('Juglandaceae', 1),\n",
       " ('Coiba', 1),\n",
       " ('Lepomis gibbosus', 1),\n",
       " ('Tryngites subruficollis', 1),\n",
       " ('T. curvispinosus', 1),\n",
       " ('Carex colwellensis', 1),\n",
       " ('Sididae', 1),\n",
       " ('Rangifer tarandus caribou', 1),\n",
       " ('Mallotus villossus', 1),\n",
       " ('Bombus ternarius', 1),\n",
       " ('Erythronium americanum', 1),\n",
       " ('Taimyra', 1),\n",
       " ('Carex livida', 1),\n",
       " ('Carex rariflora', 1),\n",
       " ('Myotis leibii', 1),\n",
       " ('Apocynaceae', 1),\n",
       " ('Vaccinium angustifolium', 1),\n",
       " ('Trientalis borealis', 1),\n",
       " ('Eriophorum', 1),\n",
       " ('Hoplitis pilosifrons', 1),\n",
       " ('M. lucifugus', 1),\n",
       " ('Fundulus heteroclitus', 1),\n",
       " ('Hylaeus annulatus', 1),\n",
       " ('A. virens', 1),\n",
       " ('E. australis', 1),\n",
       " ('Aralia nudicaulis', 1),\n",
       " ('Diptera', 1),\n",
       " ('Lasioglossum smilacinae', 1),\n",
       " ('Andrena crataegi', 1),\n",
       " ('Abies alba', 1),\n",
       " ('Lobothallia', 1),\n",
       " ('Andrena wheeleri', 1),\n",
       " ('Carex vaginata', 1),\n",
       " ('Lasioglossum coeruleum', 1),\n",
       " ('M. balthica', 1),\n",
       " ('Seiurus aurocapilla', 1),\n",
       " ('Lasioglossum versans', 1),\n",
       " ('Tilia japonica', 1),\n",
       " ('Bombus rufocinctus', 1),\n",
       " ('Porzana carolina', 1),\n",
       " ('Andrena', 1),\n",
       " ('Sphecodes davisii', 1),\n",
       " ('Corvus corax', 1),\n",
       " ('Firmicutes', 1),\n",
       " ('Dacrydium cupressinum', 1),\n",
       " ('Quercus crispula', 1),\n",
       " ('Carex trisperma', 1),\n",
       " ('T. erectum', 1),\n",
       " ('Coturnix japonica', 1),\n",
       " ('Charadrius hiaticula', 1),\n",
       " ('Carduoideae', 1),\n",
       " ('Canis latrans', 1),\n",
       " ('Brachiopoda', 1),\n",
       " ('Acer saccharum', 1),\n",
       " ('Trichophorum cespitosum', 1),\n",
       " ('Quercus aquifolioides', 1),\n",
       " ('Micropterus dolomieu', 1),\n",
       " ('Phaeocryptopus gaeumannii', 1),\n",
       " ('Archaea', 1),\n",
       " ('Carex utriculata', 1),\n",
       " ('Tcherskidium tenuicostata', 1),\n",
       " ('Calanus finmarchicus', 1),\n",
       " ('Enallagma hageni', 1),\n",
       " ('Rhododendron groenlandicum', 1),\n",
       " ('Halictus rubicundus', 1),\n",
       " ('Asteraceae', 1),\n",
       " ('Eriophorum chamissonis', 1),\n",
       " ('Carex magellanica', 1),\n",
       " ('Bryoria', 1),\n",
       " ('Carex limosa', 1),\n",
       " ('M. occidentalis', 1),\n",
       " ('Myodes gapperi', 1),\n",
       " ('Melanthiaceae', 1),\n",
       " ('Populus maximowiczii', 1),\n",
       " ('Aspicilia bicensis', 1),\n",
       " ('Larix laricina', 1),\n",
       " ('Populus tremuloides', 1),\n",
       " ('Andrena rugosa', 1),\n",
       " ('E. ebrium', 1),\n",
       " ('Coturnicops noveboracensis', 1),\n",
       " ('Canis lupus', 1),\n",
       " ('Eriophorum vaginatum', 1),\n",
       " ('Braconidae', 1),\n",
       " ('Andrena hippotes', 1),\n",
       " ('Colletes inaequalis', 1),\n",
       " ('Semotilus corporalis', 1),\n",
       " ('Chen caerulescens', 1),\n",
       " ('Abies sachalinensis', 1),\n",
       " ('Picea engelmanii', 1),\n",
       " ('Picea mariana', 1),\n",
       " ('Laurentia', 1),\n",
       " ('Salaria', 1),\n",
       " ('Metrosideros umbellata', 1),\n",
       " ('Populus', 1),\n",
       " ('Andrena carlini', 1)]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from collections import Counter\n",
    "Counter(scientific_names).most_common()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "973b8871",
   "metadata": {},
   "source": [
    "The list(set()) applied to a list gets the unique element of the list."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3da22ec5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Protomognathus americanus',\n",
       " 'Tcherskidium tenuicostatum',\n",
       " 'Onnia tomentosa',\n",
       " 'Canis',\n",
       " 'Pinus uncinata']"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scientific_names = list(set(scientific_names))\n",
    "scientific_names[0:5] # we print the 5 first element on the list scientific_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2780ce9b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "we have detected 450 unique scientific names in our corpus\n"
     ]
    }
   ],
   "source": [
    "print(\"we have detected {} unique scientific names in our corpus\".format(len(scientific_names)))\n",
    "# the function format allows integrating a value into a string."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "299956a7",
   "metadata": {},
   "source": [
    "## Vernacular names"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1171e155",
   "metadata": {},
   "source": [
    "To standardize scientific name into vernacular ones (in english) and retrieve their taxonomy, we use the custom-built function get_vernacular_name, which relies on the function name_lookup from pygbif package (itself using the GBIF taxonomic backbone: https://www.gbif.org/dataset/d7dddbf4-2cf0-4f39-9b2a-bb099caae36c). This step is long because the texonomy is huge."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "e88e994e",
   "metadata": {},
   "outputs": [],
   "source": [
    "taxo_names = [get_vernacular_names(scientific_name, limit = 20, language = \"eng\") for scientific_name in scientific_names]\n",
    "taxo_names = [x for x in taxo_names if len(x[\"vernacular_names\"])>0] # we keep only the hits for which the vernacular name in english are not empty"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "4ad0e22c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'scientific_name': 'Hylaeus annulatus',\n",
       " 'vernacular_names': ['annulate masked bee'],\n",
       " 'kingdom': 'Animalia',\n",
       " 'class': 'Insecta',\n",
       " 'order': 'Hymenoptera',\n",
       " 'family': 'Colletidae',\n",
       " 'genus': 'Hylaeus'}"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scientific_name = \"Hylaeus annulatus\"\n",
    "get_vernacular_names(scientific_name, limit = 20, language = \"eng\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "2c8e42a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'scientific_name': 'Hylaeus annulatus', 'vernacular_names': ['annulate masked bee'], 'kingdom': 'Metazoa', 'class': 'Insecta', 'order': 'Hymenoptera', 'family': 'Colletidae', 'genus': 'Hylaeus'}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'scientific_name': 'Hylaeus annulatus',\n",
       " 'kingdom': 'Metazoa',\n",
       " 'class': 'Insecta',\n",
       " 'order': 'Hymenoptera',\n",
       " 'family': 'Colletidae',\n",
       " 'genus': 'Hylaeus'}"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for i in taxo_names:\n",
    "    if i[\"scientific_name\"] == \"Hylaeus annulatus\":\n",
    "        print(i)\n",
    "vernacular_names_taxo['annulate masked bee']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef76915e",
   "metadata": {},
   "source": [
    "The output is a list of dictionaries. Each dictionary contains the keys: scientific_name,  vernacular_names, kingdom, class, order, family and genus. Let's look at the first element (1st dictionnary):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "2f2802e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'scientific_name': 'Onnia tomentosa',\n",
       " 'vernacular_names': ['woolly velvet polypore', 'velvet rosette'],\n",
       " 'kingdom': 'Fungi',\n",
       " 'class': 'Agaricomycetes',\n",
       " 'order': 'Hymenochaetales',\n",
       " 'family': 'Hymenochaetaceae',\n",
       " 'genus': 'Onnia'}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "taxo_names[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "885950f0",
   "metadata": {},
   "source": [
    "Let's print the keys of the dictionnary:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "dbf05bcf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['scientific_name', 'vernacular_names', 'kingdom', 'class', 'order', 'family', 'genus'])"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "taxo_names[0].keys()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b188866a",
   "metadata": {},
   "source": [
    "How many dictionaries do we have?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "aa9d30e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "291"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(taxo_names)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "709f6d63",
   "metadata": {},
   "source": [
    "We want to retrieve all the distinct vernacular names and their taxonomy, to detect them in the texts. We first create a second dictionary which the keys are all the distinct vernacular names."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "1b7f75c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "vernacular_names_taxo = {}\n",
    "for elmt in taxo_names:\n",
    "    for vn in elmt[\"vernacular_names\"]:\n",
    "        vernacular_names_taxo[vn] =   {'scientific_name':elmt['scientific_name'],\n",
    "        'kingdom': elmt['kingdom'], \n",
    "        'class': elmt['class'], 'order': elmt['order'], 'family': elmt['family'], 'genus': elmt['genus']}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "77d63ca9",
   "metadata": {},
   "outputs": [],
   "source": [
    "vernacular_names = list(set(vernacular_names_taxo.keys())) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "3d018fa8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "503"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(vernacular_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "f7f745d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "vernacular_name = detect_keywords(texts[4], kw = vernacular_names, type_ent= True, ent_value = \"vernacular_name\", start = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "c8a98977",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['boreal chorus frog', 'vernacular_name', 30],\n",
       " ['boreal chorus frog', 'vernacular_name', 184],\n",
       " ['boreal chorus frog', 'vernacular_name', 1403],\n",
       " ['boreal chorus frog', 'vernacular_name', 1638]]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vernacular_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "08c1d22f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'boreal chorus frog': {'scientific_name': 'Pseudacris maculata',\n",
       "  'kingdom': 'Animalia',\n",
       "  'class': 'Amphibia',\n",
       "  'order': 'Anura',\n",
       "  'family': 'Hylidae',\n",
       "  'genus': 'Pseudacris'}}"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "b7a6a8f5",
   "metadata": {},
   "source": [
    "## NER extraction"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0facd09a",
   "metadata": {},
   "source": [
    "The function tag_entities detect a list of entities based on machine-learning and rule-based algorithms (location, date, time_range) and based on dictionary-mactchning (dataset_type,\n",
    "scientific_name, vernacular_name). The output is a list of set where eahc chunk of the text is labelled wiht its entity type, or \"other\","
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "b96bdd34",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'tagged_text': [('Data from: Complementary crops and landscape features sustain wild bee communities Wild bees, which are important for commercial pollination, depend on floral and nesting resources both at farms and in the surrounding landscape. Mass-flowering crops are only in bloom for ',\n",
       "   'other'),\n",
       "  ('a few weeks', 'date'),\n",
       "  (' and unable to support bee populations that persist throughout ', 'other'),\n",
       "  ('the year', 'date'),\n",
       "  ('. Farm fields and orchards that flower in succession potentially can extend the availability of floral resources for pollinators. However, it is unclear whether the same bee species or genera will forage from one crop to the next, which bees specialize on particular crops, and to what degree inter-crop visitation patterns will be mediated by landscape context. We therefore studied local- and landscape-level drivers of bee diversity and species turnover in apple orchards, ',\n",
       "   'other'),\n",
       "  ('blueberry', 'vernacular_name'),\n",
       "  (' fields and raspberry fields that bloom sequentially in ', 'other'),\n",
       "  ('southern ', 'location'),\n",
       "  ('Quebec', 'location'),\n",
       "  (', ', 'other'),\n",
       "  ('Canada', 'location'),\n",
       "  ('. Despite the presence of high bee species turnover, orchards and small fruit fields complemented each other phenologically by supporting two bee genera essential to their pollination: mining bees (',\n",
       "   'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' spp.) and bumble bees (', 'other'),\n",
       "  ('Bombus', 'scientific_name'),\n",
       "  (' spp.). A number of bee species specialized on apple, ', 'other'),\n",
       "  ('blueberry', 'vernacular_name'),\n",
       "  (' or raspberry blossoms, suggesting that all three crops could be used to promote regional bee diversity. Bee diversity (rarefied richness, wild bee abundance) was highest across crops in landscapes containing hedgerows, meadows and suburban areas that provide ancillary nesting and floral resources throughout ',\n",
       "   'other'),\n",
       "  ('the spring and summer', 'date'),\n",
       "  ('. Promoting phenological complementarity in floral resources at the farmstead and landscape scales is essential to sustaining diverse wild bee populations.  <div class=\"o-metadata__file-usage-entry\">Bee community, environmental and landscape variables<div class=\"o-metadata__file-description\">Please refer to associated READ_ME file.</div><div class=\"o-metadata__file-name\">Martins_et_al_2018_data_files.zip<br></div><div class=\"o-metadata__file-name\"></div></div> ',\n",
       "   'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' barbilabris', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Agapostemon texanus', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Hylaeus annulatus', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Hylaeus affinis', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Bombus', 'scientific_name'),\n",
       "  (' rufocinctus', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Apidae', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Colletes inaequalis', 'scientific_name'),\n",
       "  ('; landscape ecology; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' cressonii', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' commoda', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' wilkella', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Halictus rubicundus', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' milwaukeensis', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Osmia pumila', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Ceratina', 'location'),\n",
       "  (' mikmaqi', 'scientific_name'),\n",
       "  ('; complementarity; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' hippotes', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum versans', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' nivalis', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' regularis', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' imitatrix', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' miserabilis', 'scientific_name'),\n",
       "  ('; raspberry; ', 'other'),\n",
       "  ('blueberry', 'vernacular_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum zonulum', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Bombus', 'scientific_name'),\n",
       "  (' bimaculatus', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' miranda', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Hoplitis producta', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum cinctipes', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' bisalicis', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Hylaeus modestus', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Hylaeus mesillae', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Augochlorella aurata', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Heriades carinata', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' carlini', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' nasonii', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Halictus confusus', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' perplexa', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Hoplitis spoliata', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' w-scripta', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum laevissimum', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum smilacinae', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Ceratina dupla', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' vicina', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum pilosum', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum foxii', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum cressonii', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Osmia atriventris', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum quebecense', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Bombus', 'scientific_name'),\n",
       "  (' perplexus', 'scientific_name'),\n",
       "  ('; wild pollinators; ', 'other'),\n",
       "  ('Lasioglossum sagax', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' forbesii', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Ceratina calcarata', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' wheeleri', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum nigroviride', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Holcopasites calliopsidis', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' carolina', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' duningii', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' rugosa', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' rufosignata', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' bradleyi', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum coriaceum', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' erythronii', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' thaspii', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Hoplitis pilosifrons', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Agapostemon virescens', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Bombus', 'scientific_name'),\n",
       "  (' vagans', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Bombus', 'scientific_name'),\n",
       "  (' griseocollis', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Bombus', 'scientific_name'),\n",
       "  (' borealis', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum ephialtum', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum lineatulum', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Andrena', 'scientific_name'),\n",
       "  (' crataegi', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Lasioglossum coeruleum', 'scientific_name'),\n",
       "  ('; apple; ', 'other'),\n",
       "  ('Osmia proxima', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Bombus', 'scientific_name'),\n",
       "  (' ternarius', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Augochlora pura', 'scientific_name'),\n",
       "  ('; ', 'other'),\n",
       "  ('Sphecodes davisii', 'scientific_name'),\n",
       "  (' Montérégie; ', 'other'),\n",
       "  ('Québec', 'location')],\n",
       " 'taxonomy': {'blueberry': {'scientific_name': 'Vaccinium angustifolium',\n",
       "   'kingdom': 'Plantae',\n",
       "   'class': 'Magnoliopsida',\n",
       "   'order': 'Ericales',\n",
       "   'family': 'Ericaceae',\n",
       "   'genus': 'Vaccinium'}}}"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tagged_text = tag_entities(texts[15], vocabulary_dataset, scientific_names, vernacular_names, vernacular_names_taxo )\n",
    "tagged_text"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "549c68f5",
   "metadata": {},
   "source": [
    "We can then display the tagged text as an html document."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "0d9720a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nercolor(tagged_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "4240f497",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'scientific_name': 'Onnia tomentosa',\n",
       " 'kingdom': 'Fungi',\n",
       " 'class': 'Agaricomycetes',\n",
       " 'order': 'Hymenochaetales',\n",
       " 'family': 'Hymenochaetaceae',\n",
       " 'genus': 'Onnia'}"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vernacular_names_taxo['woolly velvet polypore']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "8477ecae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Scientific name'"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'scientific_name'.replace(\"_\", \" \").capitalize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "6bd4383d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "\n",
    "log_file = open('article_ner.html', 'wb+')\n",
    "log_file.write(stri.encode('utf-8'))\n",
    "webbrowser.open('article_ner.html')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "756b7d4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "row=[]\n",
    "for i in range(0, max(df_analyse.Sum)+1):\n",
    "    idx = np.where((df_analyse.Sum.isin([i])) & (df_analyse.relevance_dataset.isin([\"H\", \"M\"])))\n",
    "    nb_rel = df_analyse.loc[idx].shape[0]\n",
    "    idx = np.where((df_analyse.Sum.isin([i])) & (df_analyse.relevance_dataset.isin([\"L\", \"X\"])))\n",
    "    nb_irrel = df_analyse.loc[idx].shape[0]\n",
    "    recall = nb_rel/TP\n",
    "    precision = nb_rel/(nb_irrel+nb_rel)\n",
    "    row.append(set([precision, recall]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "4bc0e5b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "row=[]\n",
    "for i in range(0, max(df_analyse.Sum)+1):\n",
    "    idx = np.where((df_analyse.Sum >= i) & (df_analyse.relevance_dataset.isin([\"H\", \"M\"])))\n",
    "    nb_rel = df_analyse.loc[idx].shape[0]\n",
    "    idx = np.where((df_analyse.Sum >= i) & (df_analyse.relevance_dataset.isin([\"L\", \"X\"])))\n",
    "    nb_irrel = df_analyse.loc[idx].shape[0]\n",
    "    recall = nb_rel/TP\n",
    "    precision = nb_rel/(nb_irrel+nb_rel)\n",
    "    row.append(set([precision, recall]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "86b43f69",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.246575</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.320755</td>\n",
       "      <td>0.944444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.340426</td>\n",
       "      <td>0.888889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.481481</td>\n",
       "      <td>0.722222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.700000</td>\n",
       "      <td>0.388889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.666667</td>\n",
       "      <td>0.111111</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   precision    recall\n",
       "0   0.246575  1.000000\n",
       "1   0.320755  0.944444\n",
       "2   0.340426  0.888889\n",
       "3   0.481481  0.722222\n",
       "4   0.700000  0.388889\n",
       "5   0.666667  0.111111"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame(row, columns= [\"precision\", \"recall\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "10b76aca",
   "metadata": {},
   "outputs": [],
   "source": [
    "idx = np.where((df_analyse.relevance_dataset.isin([\"H\", \"M\"])))\n",
    "df_rel = df_analyse.loc[idx]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "d8ec47ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "text = re.sub('[^0-9a-zA-Z]', ' ', text.lower())\n",
    "k = \"inventory\"\n",
    "reg = r\" (\" + re.escape(k) + r\") \"\n",
    "result = re.finditer(reg, text)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "633dd581",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "' (inventory) '"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reg =\n",
    "reg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "bdf73eac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "inventory\n"
     ]
    }
   ],
   "source": [
    "result = re.finditer(reg, text)\n",
    "for res in result:\n",
    "    print(res.group(1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "id": "5fce3a5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "text = df_datasets.full_text.tolist()[19]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "id": "53deb341",
   "metadata": {},
   "outputs": [],
   "source": [
    "base = \"http://gnrd.globalnames.org/name_finder.json\"\n",
    "payload = {\n",
    "    \"url\": url,\n",
    "    \"text\": text,\n",
    "    \"engine\": engine,\n",
    "    \"unique\": unique,\n",
    "    \"verbatim\": verbatim,\n",
    "    \"detect_language\": detect_language,\n",
    "    \"all_data_sources\": all_data_sources,\n",
    "    \"data_source_ids\": data_source_ids,\n",
    "}\n",
    "payload = {key: value for key, value in payload.items() if value != None}\n",
    "out = requests.get(base, params=payload)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "id": "821b2602",
   "metadata": {},
   "outputs": [],
   "source": [
    "allnames = []\n",
    "for i in range(90):\n",
    "    text = df_analyse.full_text.tolist()[i]\n",
    "    scientific_names = pd.read_csv(\"C://bin//{}.txt.json\".format(i))[\"Name\"].tolist()\n",
    "    if len(scientific_names)>0:\n",
    "        l5 = detect_scientific_name(text, scientific_names, type_ent= False, start = False)\n",
    "        allnames = allnames + l5\n",
    "        if len(l5) == 0:\n",
    "            print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 379,
   "id": "29aab6d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "allnames = [i[0] for i in allnames]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 381,
   "id": "8db74d38",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('Carex', 124),\n",
       " ('Temnothorax longispinosus', 121),\n",
       " ('Temnothorax curvispinosus', 100),\n",
       " ('Protomagnathus americanus', 81),\n",
       " ('Globodera rostochiensis', 64),\n",
       " ('Enallagma hageni', 49),\n",
       " ('Dryas', 36),\n",
       " ('Cyperaceae', 36),\n",
       " ('Laurentia', 36),\n",
       " ('Andrena', 28),\n",
       " ('Myotis lucifugus', 25),\n",
       " ('Trillium erectum', 25),\n",
       " ('Erythronium americanum', 25),\n",
       " ('Tachycineta bicolor', 16),\n",
       " ('Oxalis montana', 16),\n",
       " ('Salvelinus namaycush', 16),\n",
       " ('Ericaceae', 16),\n",
       " ('Onnia tomentosa', 16),\n",
       " ('Phragmites australis', 16),\n",
       " ('Acer saccharum', 14),\n",
       " ('Canis', 14),\n",
       " ('Enallagma', 11),\n",
       " ('Poecile hudsonicus', 10),\n",
       " ('P. balsamifera', 10),\n",
       " ('Morone saxatilis', 10),\n",
       " ('Glyptemys insculpta', 9),\n",
       " ('Anguilla rostrata', 9),\n",
       " ('Enallagma ebrium', 9),\n",
       " ('Diptera', 8),\n",
       " ('Aspicilia', 8),\n",
       " ('Rangifer tarandus caribou', 8),\n",
       " ('Bombus', 8),\n",
       " ('P. deltoides', 8),\n",
       " ('Salvelinus fontinalis', 8),\n",
       " ('Trillium', 8),\n",
       " ('E. ebrium', 8),\n",
       " ('Rallus limicola', 8),\n",
       " ('Coturnicops noveboracensis', 8),\n",
       " ('Rangifer tarandus', 7),\n",
       " ('Canis lupus', 7),\n",
       " ('Melampsora', 6),\n",
       " ('Picea mariana', 5),\n",
       " ('Eriophorum', 5),\n",
       " ('Populus balsamifera', 5),\n",
       " ('Salmo salar', 5),\n",
       " ('Eubalaena', 4),\n",
       " ('Aspicilia bicensis', 4),\n",
       " ('Megasporaceae', 4),\n",
       " ('Coregonus clupeaformis', 4),\n",
       " ('Rhododendron groenlandicum', 4),\n",
       " ('Kalmia angustifolia', 4),\n",
       " ('Chamaedaphne calyculata', 4),\n",
       " ('P. tremuloides', 4),\n",
       " ('Populus', 4),\n",
       " ('Populus nigra', 4),\n",
       " ('Alosa sapidissima', 4),\n",
       " ('Oncorhynchus tshawytscha', 4),\n",
       " ('T. erectum', 4),\n",
       " ('S. namaycush', 4),\n",
       " ('Alces alces', 4),\n",
       " ('Canis lycaon', 4),\n",
       " ('Anthophila', 4),\n",
       " ('Asteraceae', 4),\n",
       " ('Poecile', 4),\n",
       " ('Plaesiomys occidentalis', 4),\n",
       " ('Pseudacris maculata', 3),\n",
       " ('Trichophorum', 3),\n",
       " ('Vaccinium', 3),\n",
       " ('Laminaria rodriguezii', 3),\n",
       " ('Laminaria digitata', 3),\n",
       " ('Homarus americanus', 2),\n",
       " ('Lepidoptera', 2),\n",
       " ('Populus deltoides', 2),\n",
       " ('Thuja occidentalis', 2),\n",
       " ('Porzana carolina', 2),\n",
       " ('Pinus banksiana', 2),\n",
       " ('Tringa flavipes', 2),\n",
       " ('Myotis', 2),\n",
       " ('Eptesicus', 2),\n",
       " ('Thuringia', 1),\n",
       " ('Lepidoptera versus', 1),\n",
       " ('Dryas heath', 1),\n",
       " ('Dryas data', 1),\n",
       " ('Loiseleuria procumbens', 1),\n",
       " ('Cervus', 1),\n",
       " ('E. australis', 1),\n",
       " ('Lobothallia', 1),\n",
       " ('Betula glandulosa', 1),\n",
       " ('Andrena barbilabris', 1),\n",
       " ('Agapostemon texanus', 1),\n",
       " ('Hylaeus annulatus', 1),\n",
       " ('Hylaeus affinis', 1),\n",
       " ('Bombus rufocinctus', 1),\n",
       " ('Apidae', 1),\n",
       " ('Colletes inaequalis', 1),\n",
       " ('Andrena cressonii', 1),\n",
       " ('Andrena commoda', 1),\n",
       " ('Andrena wilkella', 1),\n",
       " ('Halictus rubicundus', 1),\n",
       " ('Andrena milwaukeensis', 1),\n",
       " ('Osmia pumila', 1),\n",
       " ('Ceratina mikmaqi', 1),\n",
       " ('Andrena hippotes', 1),\n",
       " ('Lasioglossum versans', 1),\n",
       " ('Andrena nivalis', 1),\n",
       " ('Andrena regularis', 1),\n",
       " ('Andrena imitatrix', 1),\n",
       " ('Andrena miserabilis', 1),\n",
       " ('Lasioglossum zonulum', 1),\n",
       " ('Bombus bimaculatus', 1),\n",
       " ('Andrena miranda', 1),\n",
       " ('Hoplitis producta', 1),\n",
       " ('Lasioglossum cinctipes', 1),\n",
       " ('Andrena bisalicis', 1),\n",
       " ('Hylaeus modestus', 1),\n",
       " ('Hylaeus mesillae', 1),\n",
       " ('Augochlorella aurata', 1),\n",
       " ('Heriades carinata', 1),\n",
       " ('Andrena carlini', 1),\n",
       " ('Andrena nasonii', 1),\n",
       " ('Halictus confusus', 1),\n",
       " ('Andrena perplexa', 1),\n",
       " ('Hoplitis spoliata', 1),\n",
       " ('Andrena w-scripta', 1),\n",
       " ('Lasioglossum laevissimum', 1),\n",
       " ('Lasioglossum smilacinae', 1),\n",
       " ('Ceratina dupla', 1),\n",
       " ('Andrena vicina', 1),\n",
       " ('Lasioglossum pilosum', 1),\n",
       " ('Lasioglossum foxii', 1),\n",
       " ('Lasioglossum cressonii', 1),\n",
       " ('Osmia atriventris', 1),\n",
       " ('Lasioglossum quebecense', 1),\n",
       " ('Bombus perplexus', 1),\n",
       " ('Lasioglossum sagax', 1),\n",
       " ('Andrena forbesii', 1),\n",
       " ('Ceratina calcarata', 1),\n",
       " ('Andrena wheeleri', 1),\n",
       " ('Lasioglossum nigroviride', 1),\n",
       " ('Holcopasites calliopsidis', 1),\n",
       " ('Andrena carolina', 1),\n",
       " ('Andrena duningii', 1),\n",
       " ('Andrena rugosa', 1),\n",
       " ('Andrena rufosignata', 1),\n",
       " ('Andrena bradleyi', 1),\n",
       " ('Lasioglossum coriaceum', 1),\n",
       " ('Andrena erythronii', 1),\n",
       " ('Andrena thaspii', 1),\n",
       " ('Hoplitis pilosifrons', 1),\n",
       " ('Agapostemon virescens', 1),\n",
       " ('Bombus vagans', 1),\n",
       " ('Bombus griseocollis', 1),\n",
       " ('Bombus borealis', 1),\n",
       " ('Lasioglossum ephialtum', 1),\n",
       " ('Lasioglossum lineatulum', 1),\n",
       " ('Andrena crataegi', 1),\n",
       " ('Lasioglossum coeruleum', 1),\n",
       " ('Osmia proxima', 1),\n",
       " ('Bombus ternarius', 1),\n",
       " ('Augochlora pura', 1),\n",
       " ('Sphecodes davisii', 1),\n",
       " ('Dendragapus canadensis', 1),\n",
       " ('Alces americanus', 1),\n",
       " ('Melanitta fusca', 1),\n",
       " ('M. perspicillata', 1),\n",
       " ('Clangula hyemalis', 1),\n",
       " ('Histrionicus histrionicus', 1),\n",
       " ('Somateria mollissima', 1),\n",
       " ('Carex colwellensis', 1),\n",
       " ('Carex vaginata', 1),\n",
       " ('Carex chordorrhiza', 1),\n",
       " ('Eriophorum viridicarinatum', 1),\n",
       " ('Carex aquatilis', 1),\n",
       " ('Carex rostrata', 1),\n",
       " ('Carex gynocrates', 1),\n",
       " ('Carex tsiangii', 1),\n",
       " ('Carex pachygyna', 1),\n",
       " ('Carex limosa', 1),\n",
       " ('Carex livida', 1),\n",
       " ('Carex heleonastes', 1),\n",
       " ('Carex tenuiflora', 1),\n",
       " ('Eriophorum chamissonis', 1),\n",
       " ('Carex disperma', 1),\n",
       " ('Carex trisperma', 1),\n",
       " ('Carex pauciflora', 1),\n",
       " ('Carex siderosticta', 1),\n",
       " ('Eriophorum vaginatum', 1),\n",
       " ('Carex echinata', 1),\n",
       " ('Carex exilis', 1),\n",
       " ('Carex magellanica', 1),\n",
       " ('Carex leptalea', 1),\n",
       " ('Eriophorum angustifolium', 1),\n",
       " ('Carex gibba', 1),\n",
       " ('Trichophorum cespitosum', 1),\n",
       " ('Carex saxatilis', 1),\n",
       " ('Carex rariflora', 1),\n",
       " ('Carex oligosperma', 1),\n",
       " ('Carex canescens', 1),\n",
       " ('Trichophorum alpinum', 1),\n",
       " ('Carex utriculata', 1),\n",
       " ('Vaccinium angustifolium', 1),\n",
       " ('Vaccinium myrtilloides', 1),\n",
       " ('Abies balsamea', 1),\n",
       " ('Populus tremuloides', 1),\n",
       " ('Melampsora larici-populina', 1),\n",
       " ('M. medusae', 1),\n",
       " ('M. occidentalis', 1),\n",
       " ('P. maximowiczii', 1),\n",
       " ('Populus maximowiczii', 1),\n",
       " ('Chiroptera', 1),\n",
       " ('Catostomus commersoni', 1),\n",
       " ('Exoglossum maxillingua', 1),\n",
       " ('Lepomis gibbosus', 1),\n",
       " ('Semotilus atromaculatus', 1),\n",
       " ('Ambloplites rupestris', 1),\n",
       " ('Semotilus corporalis', 1),\n",
       " ('Micropterus dolomieu', 1),\n",
       " ('Umbra limi', 1),\n",
       " ('Perca flavescens', 1),\n",
       " ('Melanthiaceae', 1),\n",
       " ('Liliaceae', 1),\n",
       " ('E. americanum', 1),\n",
       " ('Lynx rufus', 1),\n",
       " ('Escherichia coli', 1),\n",
       " ('Vibrio cholerae', 1),\n",
       " ('Caribou rangifer tarandus', 1),\n",
       " ('Ophiuroids', 1),\n",
       " ('Seiurus aurocapilla', 1),\n",
       " ('Canis lupus familiaris', 1),\n",
       " ('Canis latrans', 1),\n",
       " ('O. tomentosa', 1),\n",
       " ('Procyon lotor', 1),\n",
       " ('Mephitis mephitis', 1),\n",
       " ('Cucurbitaceae', 1),\n",
       " ('Protomognathus americanus', 1),\n",
       " ('T. curvispinosus', 1),\n",
       " ('T. longispinosus', 1),\n",
       " ('Protomagnatus americanus', 1),\n",
       " ('Microtus chrotorrhinus', 1),\n",
       " ('Synaptomys', 1),\n",
       " ('Ixodes scapularis', 1),\n",
       " ('Peromyscus maniculatus', 1),\n",
       " ('Myodes gapperi', 1),\n",
       " ('Glaucomys sabrinus', 1),\n",
       " ('Paridae', 1),\n",
       " ('Vario', 1),\n",
       " ('Pseudotsuga menziesii var. menziesii', 1),\n",
       " ('Phaeocryptopus gaeumannii', 1),\n",
       " ('Ogasawara', 1),\n",
       " ('Galapagos', 1),\n",
       " ('Hesperorthis pyramidalis', 1),\n",
       " ('Eoplectodonta', 1),\n",
       " ('Leptaena', 1),\n",
       " ('Brevilamnulella minuta', 1),\n",
       " ('Tcherskidium tenuicostatum', 1),\n",
       " ('Rhynchotrema iowense', 1),\n",
       " ('Whitfieldella', 1),\n",
       " ('Tcherskidium tenuicostata', 1),\n",
       " ('Brachiopoda', 1),\n",
       " ('Taimyra', 1),\n",
       " ('Myotis leibii', 1),\n",
       " ('M. lucifugus', 1),\n",
       " ('M. septentrionalis', 1),\n",
       " ('Eptesicus fuscus', 1),\n",
       " ('Lasionycteris noctivagans', 1)]"
      ]
     },
     "execution_count": 381,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from collections import Counter\n",
    "Counter(allnames).most_common()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 371,
   "id": "6172d40f",
   "metadata": {},
   "outputs": [],
   "source": [
    "species_convert = pd.read_csv(\"C://Users//vals3103//Web scraping//df_taxas.csv\", sep = \";\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 372,
   "id": "68d54bdc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>n</th>\n",
       "      <th>observed_scientific_name</th>\n",
       "      <th>valid_scientific_name</th>\n",
       "      <th>rank</th>\n",
       "      <th>vernacular_en</th>\n",
       "      <th>vernacular_fr</th>\n",
       "      <th>vernacular_fr_pl</th>\n",
       "      <th>group_en</th>\n",
       "      <th>group_fr</th>\n",
       "      <th>qc_status_fr</th>\n",
       "      <th>qc_status_en</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Rangifer tarandus</td>\n",
       "      <td>88.0</td>\n",
       "      <td>Rangifer tarandus</td>\n",
       "      <td>Rangifer tarandus</td>\n",
       "      <td>species</td>\n",
       "      <td>reindeer</td>\n",
       "      <td>caribou</td>\n",
       "      <td>caribous</td>\n",
       "      <td>Mammals</td>\n",
       "      <td>Mammifères</td>\n",
       "      <td>Espèce vulnérable</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Canis lupus</td>\n",
       "      <td>46.0</td>\n",
       "      <td>Canis lupus</td>\n",
       "      <td>Canis lupus</td>\n",
       "      <td>species</td>\n",
       "      <td>grey wolf</td>\n",
       "      <td>loup</td>\n",
       "      <td>loups</td>\n",
       "      <td>Mammals</td>\n",
       "      <td>Mammifères</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Odocoileus virginianus</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Odocoileus virginianus</td>\n",
       "      <td>Odocoileus virginianus</td>\n",
       "      <td>species</td>\n",
       "      <td>white-tailed deer</td>\n",
       "      <td>cerf</td>\n",
       "      <td>cerfs</td>\n",
       "      <td>Mammals</td>\n",
       "      <td>Mammifères</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Odocoileus virginianus</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Odocoileus virginianus</td>\n",
       "      <td>Odocoileus virginianus</td>\n",
       "      <td>species</td>\n",
       "      <td>white-tailed deer</td>\n",
       "      <td>chevreuil</td>\n",
       "      <td>chevreuils</td>\n",
       "      <td>Mammals</td>\n",
       "      <td>Mammifères</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Odocoileus virginianus</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Odocoileus virginianus</td>\n",
       "      <td>Odocoileus virginianus</td>\n",
       "      <td>species</td>\n",
       "      <td>white-tailed deer</td>\n",
       "      <td>cerf de virginie</td>\n",
       "      <td>cerfs de virginie</td>\n",
       "      <td>Mammals</td>\n",
       "      <td>Mammifères</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2009</th>\n",
       "      <td>Carex subspathacea</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Carex subspathacea</td>\n",
       "      <td>Carex subspathacea</td>\n",
       "      <td>species</td>\n",
       "      <td>hoppner's sedge</td>\n",
       "      <td>carex subspathacé</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Angiosperms</td>\n",
       "      <td>Angiospermes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2010</th>\n",
       "      <td>Scapania paludosa</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Scapania paludosa</td>\n",
       "      <td>Scapania paludosa</td>\n",
       "      <td>species</td>\n",
       "      <td>floppy earwort</td>\n",
       "      <td>scapanie à feuilles pendantes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Other plants</td>\n",
       "      <td>Autres plantes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2011</th>\n",
       "      <td>Lycodes paamiuti</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Lycodes paamiuti</td>\n",
       "      <td>Lycodes paamiuti</td>\n",
       "      <td>species</td>\n",
       "      <td>paamiut eelpout</td>\n",
       "      <td>lycode de paamiut</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fish</td>\n",
       "      <td>Poissons</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2012</th>\n",
       "      <td>Platanthera aquilonis</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Platanthera aquilonis</td>\n",
       "      <td>Platanthera aquilonis</td>\n",
       "      <td>species</td>\n",
       "      <td>northern green orchid</td>\n",
       "      <td>platanthère du nord</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Angiosperms</td>\n",
       "      <td>Angiospermes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2013</th>\n",
       "      <td>Kalmia angustifolia</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Kalmia angustifolia</td>\n",
       "      <td>Kalmia angustifolia</td>\n",
       "      <td>species</td>\n",
       "      <td>sheep-laurel</td>\n",
       "      <td>kalmia à feuilles étroites</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Angiosperms</td>\n",
       "      <td>Angiospermes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2014 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       index     n observed_scientific_name  \\\n",
       "0          Rangifer tarandus  88.0        Rangifer tarandus   \n",
       "1                Canis lupus  46.0              Canis lupus   \n",
       "2     Odocoileus virginianus  24.0   Odocoileus virginianus   \n",
       "3     Odocoileus virginianus  24.0   Odocoileus virginianus   \n",
       "4     Odocoileus virginianus  24.0   Odocoileus virginianus   \n",
       "...                      ...   ...                      ...   \n",
       "2009      Carex subspathacea   0.0       Carex subspathacea   \n",
       "2010       Scapania paludosa   0.0        Scapania paludosa   \n",
       "2011        Lycodes paamiuti   0.0         Lycodes paamiuti   \n",
       "2012   Platanthera aquilonis   0.0    Platanthera aquilonis   \n",
       "2013     Kalmia angustifolia   0.0      Kalmia angustifolia   \n",
       "\n",
       "       valid_scientific_name     rank          vernacular_en  \\\n",
       "0          Rangifer tarandus  species               reindeer   \n",
       "1                Canis lupus  species              grey wolf   \n",
       "2     Odocoileus virginianus  species      white-tailed deer   \n",
       "3     Odocoileus virginianus  species      white-tailed deer   \n",
       "4     Odocoileus virginianus  species      white-tailed deer   \n",
       "...                      ...      ...                    ...   \n",
       "2009      Carex subspathacea  species        hoppner's sedge   \n",
       "2010       Scapania paludosa  species         floppy earwort   \n",
       "2011        Lycodes paamiuti  species        paamiut eelpout   \n",
       "2012   Platanthera aquilonis  species  northern green orchid   \n",
       "2013     Kalmia angustifolia  species           sheep-laurel   \n",
       "\n",
       "                      vernacular_fr   vernacular_fr_pl      group_en  \\\n",
       "0                           caribou           caribous       Mammals   \n",
       "1                              loup              loups       Mammals   \n",
       "2                              cerf              cerfs       Mammals   \n",
       "3                         chevreuil         chevreuils       Mammals   \n",
       "4                  cerf de virginie  cerfs de virginie       Mammals   \n",
       "...                             ...                ...           ...   \n",
       "2009              carex subspathacé                NaN   Angiosperms   \n",
       "2010  scapanie à feuilles pendantes                NaN  Other plants   \n",
       "2011              lycode de paamiut                NaN          Fish   \n",
       "2012            platanthère du nord                NaN   Angiosperms   \n",
       "2013     kalmia à feuilles étroites                NaN   Angiosperms   \n",
       "\n",
       "            group_fr       qc_status_fr  qc_status_en  \n",
       "0         Mammifères  Espèce vulnérable           NaN  \n",
       "1         Mammifères                NaN           NaN  \n",
       "2         Mammifères                NaN           NaN  \n",
       "3         Mammifères                NaN           NaN  \n",
       "4         Mammifères                NaN           NaN  \n",
       "...              ...                ...           ...  \n",
       "2009    Angiospermes                NaN           NaN  \n",
       "2010  Autres plantes                NaN           NaN  \n",
       "2011        Poissons                NaN           NaN  \n",
       "2012    Angiospermes                NaN           NaN  \n",
       "2013    Angiospermes                NaN           NaN  \n",
       "\n",
       "[2014 rows x 12 columns]"
      ]
     },
     "execution_count": 372,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "species_convert"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "c6b219c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "text = \"We surveyed Yellow Rails (Coturnicops noveboracensis) in three marshes located in southeastern James Bay, Canada, in order to document the status and distribution of this species at risk. Night-surveys of calling males were made along 75 km of line transect on 21-25 July 2002, and 186 individual Yellow Rails were recorded: 80 in Cabbage Willows Bay, 77 in Boatswain Bay, and 29 in Hall Cove. Daytime surveys yielded 19 additional calling males, giving a total of 205 male rails. Transects were distributed in plant communities dominated by Slimstem Reedgrass (Calamagrostis stricta), Chaffy Sedge (Carex paleacea), Buckbean (Menyanthes trifoliata), or by Red Fescue (Festuca rubra). There was an overall density of 0.05 calling rails per ha with maximum densities in Boatswain Bay (0.08 male/ha) and Hall Cove (0.06 males/ha). Densities calculated for specific plant communities permitted an estimate of 397 male Yellow Rail inhabiting the marshes: 216 in Boatswain Bay, 132 in Cabbage Willows Bay, and 49 in Hall Cove. The highest densities were in Chaffy Sedge stands, Slimstem Reedgrass stands, and fens dominated by Buckbean. This study indicates that the Yellow Rail is a common bird in coastal high-marshes of the southeastern James Bay, and revealed the highest densities ever reported for the species. The extensive estuarine tidal marshes found in southern James Bay may be home of a thousand or more adult male Yellow Rails and thus represents a key area for this species.\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "aa3a0427",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>type_dataset</th>\n",
       "      <th>keywords</th>\n",
       "      <th>keywords_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>presence only</td>\n",
       "      <td>presence only</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>presence only</td>\n",
       "      <td>occurrence</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>presence only</td>\n",
       "      <td>specimen records</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>presence only</td>\n",
       "      <td>specimen</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>presence only</td>\n",
       "      <td>collection</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>presence-absence</td>\n",
       "      <td>presence-absence</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>presence-absence</td>\n",
       "      <td>sampling plot</td>\n",
       "      <td>methods</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>presence-absence</td>\n",
       "      <td>relative proportion</td>\n",
       "      <td>methods</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>presence-absence</td>\n",
       "      <td>inventory</td>\n",
       "      <td>methods</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>presence-absence</td>\n",
       "      <td>occupancy</td>\n",
       "      <td>methods</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>richness</td>\n",
       "      <td>richness</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>abundance</td>\n",
       "      <td>abundance</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>abundance</td>\n",
       "      <td>capture-mark-recapture</td>\n",
       "      <td>methods</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>abundance</td>\n",
       "      <td>relative proportion</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>abundance</td>\n",
       "      <td>percent cover</td>\n",
       "      <td>methods</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>density</td>\n",
       "      <td>density</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>distribution</td>\n",
       "      <td>distribution</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>distribution</td>\n",
       "      <td>range</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>distribution</td>\n",
       "      <td>habitat suitability</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>distribution</td>\n",
       "      <td>niche</td>\n",
       "      <td>synonym</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>distribution</td>\n",
       "      <td>species distribution model</td>\n",
       "      <td>methods</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>abundance</td>\n",
       "      <td>camera-trap</td>\n",
       "      <td>methods</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>density</td>\n",
       "      <td>quadrat</td>\n",
       "      <td>methods</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>abundance</td>\n",
       "      <td>quadrat</td>\n",
       "      <td>methods</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PA</td>\n",
       "      <td>quadrat</td>\n",
       "      <td>methods</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>abundance</td>\n",
       "      <td>counts</td>\n",
       "      <td>methods</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        type_dataset                    keywords keywords_type\n",
       "0      presence only               presence only       synonym\n",
       "1      presence only                  occurrence       synonym\n",
       "2      presence only            specimen records       synonym\n",
       "3      presence only                    specimen       synonym\n",
       "4      presence only                  collection       synonym\n",
       "5   presence-absence            presence-absence       synonym\n",
       "6   presence-absence               sampling plot       methods\n",
       "7   presence-absence         relative proportion       methods\n",
       "8   presence-absence                   inventory       methods\n",
       "9   presence-absence                   occupancy       methods\n",
       "10          richness                    richness       synonym\n",
       "11         abundance                   abundance       synonym\n",
       "12         abundance      capture-mark-recapture       methods\n",
       "13         abundance         relative proportion       synonym\n",
       "14         abundance               percent cover       methods\n",
       "15           density                     density       synonym\n",
       "16      distribution                distribution       synonym\n",
       "17      distribution                       range       synonym\n",
       "18      distribution         habitat suitability       synonym\n",
       "19      distribution                       niche       synonym\n",
       "20      distribution  species distribution model       methods\n",
       "21         abundance                 camera-trap       methods\n",
       "22           density                     quadrat       methods\n",
       "23         abundance                     quadrat       methods\n",
       "24                PA                     quadrat       methods\n",
       "25         abundance                      counts       methods"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vocabulary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "07d3c11d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "75c47f52",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 221,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "e5df9d44",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('holocene', 10),\n",
       " ('anthropocene', 3),\n",
       " ('rangifer tarandus', 3),\n",
       " ('salmo salar', 3),\n",
       " ('tachycineta bicolor', 3),\n",
       " ('procyon lotor', 2),\n",
       " ('population structure and phylogeography', 2),\n",
       " ('landscape ecology', 2),\n",
       " ('captive populations', 2),\n",
       " ('agricultural intensification', 2),\n",
       " ('host parasite interactions', 2),\n",
       " ('disease biology', 2),\n",
       " ('subarctic', 2),\n",
       " ('acer saccharum', 2),\n",
       " ('wildlife management', 2),\n",
       " ('spatial synchrony', 2),\n",
       " ('coexistence theory', 2),\n",
       " ('para-pseudocalanus spp.', 2),\n",
       " ('north american fish', 1),\n",
       " ('mock community', 1),\n",
       " ('acipenser fulvescens', 1),\n",
       " ('alosa aestivalis', 1),\n",
       " ('wildlife survey', 1),\n",
       " ('mephitis mephitis', 1),\n",
       " ('striped skunk', 1),\n",
       " ('raccoon', 1),\n",
       " ('capture', 1),\n",
       " ('brachycrus wilsoni', 1),\n",
       " ('otomys auretus', 1),\n",
       " ('experimental selection', 1),\n",
       " ('paleocene', 1),\n",
       " ('longitudinal time series', 1),\n",
       " ('merycochoerus proprius', 1),\n",
       " ('ectocion parvus', 1),\n",
       " ('passer montanus', 1),\n",
       " ('petrochelidon pyrrhonota', 1),\n",
       " ('metrarabdotos volkesorum', 1),\n",
       " ('metrarabdotos cubaguaense', 1),\n",
       " ('anolis carolinensis', 1),\n",
       " ('anakosmoceras sp.', 1),\n",
       " ('globoratalia tumida', 1),\n",
       " ('oligocene', 1),\n",
       " ('cervus elaphus', 1),\n",
       " ('rattus rattus', 1),\n",
       " ('ursavus elmensis', 1),\n",
       " ('merychippus paniensis', 1),\n",
       " ('longitudinal field study', 1),\n",
       " ('cross-sectional field study', 1),\n",
       " ('merychyus elegans', 1),\n",
       " ('mesohippus bairdi', 1),\n",
       " ('hyracotherium sandrae', 1),\n",
       " ('hyopsodus latidens', 1),\n",
       " ('cantius trigonodus', 1),\n",
       " ('cantius ralstoni', 1),\n",
       " ('kosmoceras sp.', 1),\n",
       " ('passer domesticus', 1),\n",
       " ('merycochoerus matthewi', 1),\n",
       " ('stephanodiscus yellowstonensis', 1),\n",
       " ('eocoelia sp.', 1),\n",
       " ('hyopsodus pauxillus', 1),\n",
       " ('jurassic', 1),\n",
       " ('merychyus crabilli', 1),\n",
       " ('neotoma albigula', 1),\n",
       " ('branta leucopsis', 1),\n",
       " ('mus sp.', 1),\n",
       " ('metrarabdotos colligatum', 1),\n",
       " ('peromyscus maniculatus', 1),\n",
       " ('loxodonta sp.', 1),\n",
       " ('mesomyla sclateri', 1),\n",
       " ('homo erectus', 1),\n",
       " ('brassica rapa', 1),\n",
       " ('mammuthus sp.', 1),\n",
       " ('artificial selection', 1),\n",
       " ('haplomylus simpsoni', 1),\n",
       " ('cormohipparion occidentale', 1),\n",
       " ('peromyscus leucopus', 1),\n",
       " ('cosomys primus', 1),\n",
       " ('hyracotherium grangeri', 1),\n",
       " ('parapelomys sp.', 1),\n",
       " ('globorotalia conoidea', 1),\n",
       " ('eohippus resartus', 1),\n",
       " ('hyopsodus miticulus', 1),\n",
       " ('zea mays', 1),\n",
       " ('dipodomys merriami', 1),\n",
       " ('urocitellus beldingi', 1),\n",
       " ('amara quenseli', 1),\n",
       " ('eocene', 1),\n",
       " ('herpestes auropunctatus', 1),\n",
       " ('otomys angoniensis', 1),\n",
       " ('hyopsodus powellianus', 1),\n",
       " ('hirundo rustica', 1),\n",
       " ('elephas sp.', 1),\n",
       " ('poseidonamicus spp.', 1),\n",
       " ('primelephas sp.', 1),\n",
       " ('euryderus grossus', 1),\n",
       " ('globoratalia plesiotumida', 1),\n",
       " ('merychyus relictus', 1),\n",
       " ('ursus etruscus', 1),\n",
       " ('cantius mckennai', 1),\n",
       " ('gyretes sinuatus', 1),\n",
       " ('globorotalia inflata', 1),\n",
       " ('gasterosteus aculeatus', 1),\n",
       " ('drepanis coccinea', 1),\n",
       " ('gymnodactylus amarali', 1),\n",
       " ('carabus nemoralis', 1),\n",
       " ('chlorocebus sabaeus', 1),\n",
       " ('experimental field study', 1),\n",
       " ('ursus spelaeus', 1),\n",
       " ('myzomela pammelaena', 1),\n",
       " ('cymindis planipennis', 1),\n",
       " ('thymallus thymallus', 1),\n",
       " ('ectocion ralstonensis', 1),\n",
       " ('haplomylus speirianus', 1),\n",
       " ('pseudocubus vema', 1),\n",
       " ('pterostichus algidus', 1),\n",
       " ('dinosauria', 1),\n",
       " ('pterostichus melanarius', 1),\n",
       " ('equus germanicus', 1),\n",
       " ('globorotalia puncticulata', 1),\n",
       " ('gallus gallus', 1),\n",
       " ('merychyus minimus', 1),\n",
       " ('cryptopecten vesicululosus', 1),\n",
       " ('equidae', 1),\n",
       " ('hyopsodus minor', 1),\n",
       " ('hyracotherium borealis', 1),\n",
       " ('oncorhyncus nerka', 1),\n",
       " ('ovis aries', 1),\n",
       " ('fringilla coelebs', 1),\n",
       " ('fossil record', 1),\n",
       " ('anser caerulescens', 1),\n",
       " ('haplomylus scottianus', 1),\n",
       " ('brachycrus siouense', 1),\n",
       " ('cantius torresi', 1),\n",
       " ('hyopsodus lysitensis', 1),\n",
       " ('ficedula albicollis', 1),\n",
       " ('ectocion osbornianus', 1),\n",
       " ('mus musculus', 1),\n",
       " ('bison bison', 1),\n",
       " ('metrarabdotos saundersi', 1),\n",
       " ('zugokosmoceras sp.', 1),\n",
       " ('bison priscus', 1),\n",
       " ('metrarabdotos auriculatum', 1),\n",
       " ('bison latifrons', 1),\n",
       " ('mandarina chichijimana', 1),\n",
       " ('gasterosteus doryssus', 1),\n",
       " ('geospiza scandens', 1),\n",
       " ('scaphinotus angusticollis', 1),\n",
       " ('telespyza canans', 1),\n",
       " ('recent', 1),\n",
       " ('oryctolagus cuniculus', 1),\n",
       " ('jadera haematoloma', 1),\n",
       " ('antemus sp.', 1),\n",
       " ('hyopsodus loomisi', 1),\n",
       " ('progonomys sp.', 1),\n",
       " ('drosophila subobscura', 1),\n",
       " ('harpalus fraternus', 1),\n",
       " ('introduced species', 1),\n",
       " ('laboratory selection', 1),\n",
       " ('haplomylus palustris', 1),\n",
       " ('haplomylus zalmouti', 1),\n",
       " ('cyclostephanos andinus', 1),\n",
       " ('loxodonta africana', 1),\n",
       " ('globorotalia conomiozea', 1),\n",
       " ('karnimata sp.', 1),\n",
       " ('plethodon spp.', 1),\n",
       " ('geospiza fortis', 1),\n",
       " ('ursus arctos', 1),\n",
       " ('bison antiquus', 1),\n",
       " ('metrarabdotos boldi', 1),\n",
       " ('metrarabdotos coatesi', 1),\n",
       " ('drosophila pseudoobscura', 1),\n",
       " ('littorina obtusata', 1),\n",
       " ('parus caeruleus', 1),\n",
       " ('poecilia reticulata', 1),\n",
       " ('miocene', 1),\n",
       " ('aspicilia', 1),\n",
       " ('sterile crustose lichens', 1),\n",
       " ('soralia', 1),\n",
       " ('lobothallia', 1),\n",
       " ('oceanic', 1),\n",
       " ('pustules', 1),\n",
       " ('maritime', 1),\n",
       " ('ascomycetes', 1),\n",
       " ('megasporaceae', 1),\n",
       " ('litter', 1),\n",
       " ('taxonomic diversity', 1),\n",
       " ('functional biodiversity', 1),\n",
       " ('tree', 1),\n",
       " ('genetic', 1),\n",
       " ('oncorhynchus tshawytscha', 1),\n",
       " ('salvelinus fontinalis', 1),\n",
       " ('genetic turnover', 1),\n",
       " ('zoonotic disease emergence', 1),\n",
       " ('ixodes scapularis', 1),\n",
       " ('canis lupus familiaris', 1),\n",
       " ('canis lycaon', 1),\n",
       " ('eastern wolf', 1),\n",
       " ('predator distribution', 1),\n",
       " ('canis lycaon sp. cf', 1),\n",
       " ('canis latrans', 1),\n",
       " ('canis lupus', 1),\n",
       " ('dynamic soaring', 1),\n",
       " ('laysan albatross', 1),\n",
       " ('streaked shearwater', 1),\n",
       " ('wandering albatross', 1),\n",
       " ('bio-logging', 1),\n",
       " ('gps', 1),\n",
       " ('satellite scatterometer', 1),\n",
       " ('land management', 1),\n",
       " ('expert-based knowledge', 1),\n",
       " ('wildlife-habitat model', 1),\n",
       " ('rangifer tarandus caribou', 1),\n",
       " ('anthropogenic disturbances', 1),\n",
       " ('kalmia angustifolia', 1),\n",
       " ('redundancy analysis', 1),\n",
       " ('rhododendron groenlandicum', 1),\n",
       " ('vaccinium angustifolium', 1),\n",
       " ('chamaedaphne calyculata', 1),\n",
       " ('vaccinium myrtilloides', 1),\n",
       " ('ecological drivers', 1),\n",
       " ('migratory caribou', 1),\n",
       " ('space use', 1),\n",
       " ('movements', 1),\n",
       " ('long-distance migration', 1),\n",
       " ('scales', 1),\n",
       " ('step selection function', 1),\n",
       " ('16s ssu rrna gene', 1),\n",
       " ('endangered species management', 1),\n",
       " ('2013', 1),\n",
       " ('metabarcoding sequencing', 1),\n",
       " ('aquaculture', 1),\n",
       " ('stocking', 1),\n",
       " ('community trait variation', 1),\n",
       " ('intraspecific trait variation', 1),\n",
       " ('transplant experiment', 1),\n",
       " ('oxalis montana l.', 1),\n",
       " ('populus deltoides', 1),\n",
       " ('snps', 1),\n",
       " ('melampsora sp.', 1),\n",
       " ('reproductive fitness', 1),\n",
       " ('postzygotic fitness', 1),\n",
       " ('populus balsamifera', 1),\n",
       " ('tree swallow', 1),\n",
       " ('aerial insectivores', 1),\n",
       " ('breeding success', 1),\n",
       " ('liver', 1),\n",
       " ('aerobic energy metabolism', 1),\n",
       " ('coregonus clupeaformis', 1),\n",
       " ('glycolysis', 1),\n",
       " ('red muscle', 1),\n",
       " ('white muscle', 1),\n",
       " ('heart', 1),\n",
       " ('aspen', 1),\n",
       " ('red queen', 1),\n",
       " ('coevolution', 1),\n",
       " ('caullerya', 1),\n",
       " ('networks', 1),\n",
       " ('matching-centrality formalism', 1),\n",
       " ('diplopoda', 1),\n",
       " ('isopoda', 1),\n",
       " ('carabidae', 1),\n",
       " ('pseudoscorpiones', 1),\n",
       " ('araneae', 1),\n",
       " ('trophic interactions', 1),\n",
       " ('host-parasite interaction', 1),\n",
       " ('temnothorax longispinosus', 1),\n",
       " ('temnothorax curvispinosus', 1),\n",
       " ('protomagnatus americanus', 1),\n",
       " ('buller phenomenon', 1),\n",
       " ('onnia tomentosa', 1),\n",
       " ('wahlund effect', 1),\n",
       " ('root rot', 1),\n",
       " ('pleistocene glaciations', 1),\n",
       " ('paridae', 1),\n",
       " ('boreal chickadee', 1),\n",
       " ('poecile hudsonicus', 1),\n",
       " ('poecile', 1),\n",
       " ('andrena barbilabris', 1),\n",
       " ('agapostemon texanus', 1),\n",
       " ('hylaeus annulatus', 1),\n",
       " ('hylaeus affinis', 1),\n",
       " ('bombus rufocinctus', 1),\n",
       " ('apidae', 1),\n",
       " ('colletes inaequalis', 1),\n",
       " ('andrena cressonii', 1),\n",
       " ('andrena commoda', 1),\n",
       " ('andrena wilkella', 1),\n",
       " ('halictus rubicundus', 1),\n",
       " ('andrena milwaukeensis', 1),\n",
       " ('osmia pumila', 1),\n",
       " ('ceratina mikmaqi', 1),\n",
       " ('complementarity', 1),\n",
       " ('andrena hippotes', 1),\n",
       " ('lasioglossum versans', 1),\n",
       " ('andrena nivalis', 1),\n",
       " ('andrena regularis', 1),\n",
       " ('andrena imitatrix', 1),\n",
       " ('andrena miserabilis', 1),\n",
       " ('raspberry', 1),\n",
       " ('blueberry', 1),\n",
       " ('lasioglossum zonulum', 1),\n",
       " ('bombus bimaculatus', 1),\n",
       " ('andrena miranda', 1),\n",
       " ('hoplitis producta', 1),\n",
       " ('lasioglossum cinctipes', 1),\n",
       " ('andrena bisalicis', 1),\n",
       " ('hylaeus modestus', 1),\n",
       " ('hylaeus mesillae', 1),\n",
       " ('augochlorella aurata', 1),\n",
       " ('heriades carinata', 1),\n",
       " ('andrena carlini', 1),\n",
       " ('andrena nasonii', 1),\n",
       " ('halictus confusus', 1),\n",
       " ('andrena perplexa', 1),\n",
       " ('hoplitis spoliata', 1),\n",
       " ('andrena w-scripta', 1),\n",
       " ('lasioglossum laevissimum', 1),\n",
       " ('lasioglossum smilacinae', 1),\n",
       " ('ceratina dupla', 1),\n",
       " ('andrena vicina', 1),\n",
       " ('lasioglossum pilosum', 1),\n",
       " ('lasioglossum foxii', 1),\n",
       " ('lasioglossum cressonii', 1),\n",
       " ('osmia atriventris', 1),\n",
       " ('lasioglossum quebecense', 1),\n",
       " ('bombus perplexus', 1),\n",
       " ('wild pollinators', 1),\n",
       " ('lasioglossum sagax', 1),\n",
       " ('andrena forbesii', 1),\n",
       " ('ceratina calcarata', 1),\n",
       " ('andrena wheeleri', 1),\n",
       " ('lasioglossum nigroviride', 1),\n",
       " ('holcopasites calliopsidis', 1),\n",
       " ('andrena carolina', 1),\n",
       " ('andrena duningii', 1),\n",
       " ('andrena rugosa', 1),\n",
       " ('andrena rufosignata', 1),\n",
       " ('andrena bradleyi', 1),\n",
       " ('lasioglossum coriaceum', 1),\n",
       " ('andrena erythronii', 1),\n",
       " ('andrena thaspii', 1),\n",
       " ('hoplitis pilosifrons', 1),\n",
       " ('agapostemon virescens', 1),\n",
       " ('bombus vagans', 1),\n",
       " ('bombus griseocollis', 1),\n",
       " ('bombus borealis', 1),\n",
       " ('lasioglossum ephialtum', 1),\n",
       " ('lasioglossum lineatulum', 1),\n",
       " ('andrena crataegi', 1),\n",
       " ('lasioglossum coeruleum', 1),\n",
       " ('apple', 1),\n",
       " ('osmia proxima', 1),\n",
       " ('bombus ternarius', 1),\n",
       " ('augochlora pura', 1),\n",
       " ('sphecodes davisii', 1),\n",
       " ('ecology, herpetology, landscape ecology', 1),\n",
       " ('latitudinal gradient', 1),\n",
       " ('nunavik', 1),\n",
       " ('lichen species richness', 1),\n",
       " ('lichen cover', 1),\n",
       " ('macrolichens', 1),\n",
       " ('crop yield', 1),\n",
       " ('reproductive strategies', 1),\n",
       " ('clinal variation', 1),\n",
       " ('american shad', 1),\n",
       " ('alosa sapidissima', 1),\n",
       " ('substrate', 1),\n",
       " ('seed predation', 1),\n",
       " ('parasitology', 1),\n",
       " ('life history evolution', 1),\n",
       " ('coregonus cupleaformis', 1),\n",
       " ('fat content', 1),\n",
       " ('protandry', 1),\n",
       " ('rearing', 1),\n",
       " ('body condition', 1),\n",
       " ('water content', 1),\n",
       " ('propagule pressure', 1),\n",
       " ('agrilus planipennis', 1),\n",
       " ('fraxinus', 1),\n",
       " ('myotis lucifugus', 1),\n",
       " ('population expansion', 1),\n",
       " ('chiroptera', 1),\n",
       " ('provenance', 1),\n",
       " ('temperate-boreal forest transition', 1),\n",
       " ('range shift', 1),\n",
       " ('2013-2015', 1),\n",
       " ('genomics/proteomics', 1),\n",
       " ('oribatids', 1),\n",
       " ('habitat modelling', 1),\n",
       " ('spatial model', 1),\n",
       " ('scale-dependent correlation', 1),\n",
       " ('selective pressure', 1),\n",
       " ('î²-defensin', 1),\n",
       " ('conservation genetics and biodiversity', 1),\n",
       " ('animal models', 1),\n",
       " ('pedigree simulation', 1),\n",
       " ('extra-pair paternities', 1),\n",
       " ('statistical power', 1),\n",
       " ('pedigree errors', 1),\n",
       " ('euplotes patella', 1),\n",
       " ('euplotes', 1),\n",
       " ('tetrahymena', 1),\n",
       " ('dispersal kernel', 1),\n",
       " ('tetrahymena pyriformis', 1),\n",
       " ('protists', 1),\n",
       " ('microcosms', 1),\n",
       " ('rallus limicola', 1),\n",
       " ('rails', 1),\n",
       " ('coturnicops noveboracensis', 1),\n",
       " ('porzana carolina', 1),\n",
       " ('migratory connectivity', 1),\n",
       " ('autumn migration', 1),\n",
       " ('density-dependence', 1),\n",
       " ('avian diversity', 1),\n",
       " ('canada', 1),\n",
       " ('farming practices', 1),\n",
       " ('organic farming', 1),\n",
       " ('pesticide', 1),\n",
       " ('farmland bird', 1),\n",
       " ('odocoileus virginianus', 1),\n",
       " ('nuclear ribosomal spacers', 1),\n",
       " ('tertiary and quaternary', 1),\n",
       " ('carex vaginata', 1),\n",
       " ('limiting similarity', 1),\n",
       " ('carex chordorrhiza', 1),\n",
       " ('community phylogenetics', 1),\n",
       " ('eriophorum viridicarinatum', 1),\n",
       " ('carex aquatilis', 1),\n",
       " ('carex rostrata', 1),\n",
       " ('carex gynocrates', 1),\n",
       " ('carex tsiangii', 1),\n",
       " ('carex pachygyna', 1),\n",
       " ('carex limosa', 1),\n",
       " ('pairwise co-occurrence', 1),\n",
       " ('ecophylogenetics', 1),\n",
       " ('present', 1),\n",
       " ('carex livida', 1),\n",
       " ('carex heleonastes', 1),\n",
       " ('carex tenuiflora', 1),\n",
       " ('eriophorum chamissonis', 1),\n",
       " ('carex disperma', 1),\n",
       " ('carex trisperma', 1),\n",
       " ('carex pauciflora', 1),\n",
       " ('niche differentiation', 1),\n",
       " ('', 1),\n",
       " ('carex siderosticta', 1),\n",
       " ('eriophorum vaginatum', 1),\n",
       " ('carex echinata', 1),\n",
       " ('carex exilis', 1),\n",
       " ('carex magellanica', 1),\n",
       " ('plastid dna', 1),\n",
       " ('carex leptalea', 1),\n",
       " ('eriophorum angustifolium', 1),\n",
       " ('carex gibba', 1),\n",
       " ('divergence times', 1),\n",
       " ('trichophorum cespitosum', 1),\n",
       " ('carex saxatilis', 1),\n",
       " ('carex rariflora', 1),\n",
       " ('subarctic peatlands', 1),\n",
       " ('quadrat sampling', 1),\n",
       " ('carex oligosperma', 1),\n",
       " ('carex canescens', 1),\n",
       " ('cyperaceae', 1),\n",
       " ('trichophorum alpinum', 1),\n",
       " ('dna alignment', 1),\n",
       " ('carex utriculata', 1),\n",
       " ('circuit theory', 1),\n",
       " ('causal modelling', 1),\n",
       " ('habitat suitability models', 1),\n",
       " ('least-cost path', 1),\n",
       " ('species pool', 1),\n",
       " ('null models', 1),\n",
       " ('phylogenetic beta-diversity', 1),\n",
       " ('net-relatedness index', 1),\n",
       " ('fossil time series', 1),\n",
       " ('paleontology', 1),\n",
       " ('imo-d2', 1),\n",
       " ('coliform', 1),\n",
       " ('vibrio cholerae', 1),\n",
       " ('synergistic', 1),\n",
       " ('ballast water treatment', 1),\n",
       " ('additive', 1),\n",
       " ('chlorine', 1),\n",
       " ('alien species', 1),\n",
       " ('nonindigenous', 1),\n",
       " ('enterococci', 1),\n",
       " ('genotyping-by-sequencing', 1),\n",
       " ('phragmites australis', 1),\n",
       " ('road ecology', 1),\n",
       " ('sexual reproduction', 1),\n",
       " ('common reed', 1),\n",
       " ('invasion ecology', 1),\n",
       " ('vegetative propagation', 1),\n",
       " ('freshwater wetlands', 1),\n",
       " ('temora longicornis', 1),\n",
       " ('euphausiacea', 1),\n",
       " ('nitzschia delicatissima', 1),\n",
       " ('centropages typicus', 1),\n",
       " ('ceratium tripos', 1),\n",
       " ('oithona spp.', 1),\n",
       " ('ceratium furca', 1),\n",
       " ('pseudocalanus elongatus', 1),\n",
       " ('metridia lucens', 1),\n",
       " ('calanus finmarchicus', 1),\n",
       " ('calanus i-iv', 1),\n",
       " ('rhizosolenia styliformis', 1),\n",
       " ('thalassiosira spp.', 1),\n",
       " ('ceratium macroceros', 1),\n",
       " ('calanus helgolandicus', 1),\n",
       " ('nitzschia seriata', 1),\n",
       " ('echinoderm larvae', 1),\n",
       " ('rhizosolenia alata alata', 1),\n",
       " ('acartia spp.', 1),\n",
       " ('wavelets', 1),\n",
       " ('decapoda larvae', 1),\n",
       " ('ceratium fusus', 1),\n",
       " ('patch', 1),\n",
       " ('local divergence', 1),\n",
       " ('enallagma ebrium', 1),\n",
       " ('aflp', 1),\n",
       " ('enallagma hageni', 1),\n",
       " ('distribution', 1),\n",
       " ('tringa flavipes', 1),\n",
       " ('random walk model', 1),\n",
       " ('harvest', 1),\n",
       " ('migratory shorebirds', 1),\n",
       " ('lesser yellowlegs', 1),\n",
       " ('binomial generalized linear model', 1),\n",
       " ('continuous-time functional movement models', 1),\n",
       " ('forest-tundra', 1),\n",
       " ('plant-climate interactions', 1),\n",
       " ('1950-2010', 1),\n",
       " ('treeline', 1),\n",
       " ('tree-ring', 1),\n",
       " ('betula glandulosa', 1),\n",
       " ('picea mariana', 1),\n",
       " ('snowpatch', 1),\n",
       " ('land use change', 1),\n",
       " ('temporal genetics', 1),\n",
       " ('network theory', 1),\n",
       " ('connectivity conservation', 1),\n",
       " ('circuitscape', 1),\n",
       " ('seiurus aurocapilla', 1),\n",
       " ('behavioural strategies', 1),\n",
       " ('conservation measures', 1),\n",
       " ('human activities', 1),\n",
       " ('otolith', 1),\n",
       " ('stable isotope', 1),\n",
       " ('secondary-ion mass spectrometry', 1),\n",
       " ('supplementation stocking', 1),\n",
       " ('fish management', 1),\n",
       " ('thermal habitat', 1),\n",
       " ('web archives_parquet_dataframes', 1),\n",
       " (\"milieu humide_québec_milieux humides forestiers_boréal_cartographie_milieux hydriques_habitats fauniques_classification_littoral d'eau douce_données écoforestières\",\n",
       "  1),\n",
       " ('fault reactivation potential, slip tendency, western quebec seismic zone, faults, eastern canada, intraplate faults',\n",
       "  1),\n",
       " ('[]', 1),\n",
       " ('life-cycle assessment_recycling_regionalization_parametrization_wastepaper',\n",
       "  1),\n",
       " ('dynamic link prediction, dynamic graphs representation learning, evaluation',\n",
       "  1)]"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from collections import Counter\n",
    "keywords = [x.split(\";\") for x in df_analyse.keywords.tolist() if isinstance(x, str)]\n",
    "keywords =  [x.strip().lower() for sublist in keywords for x in sublist]\n",
    "keyword = Counter(keywords)\n",
    "keyword.most_common()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "id": "614cc4f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "102"
      ]
     },
     "execution_count": 254,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_datasets = pd.read_excel(\"C://Users//vals3103//Post-doc//Text_mining//df_work.xlsx\", engine = \"openpyxl\")\n",
    "df_datasets[\"full_text\"] = df_datasets['title'] + \" \"+ df_datasets['description'].fillna('') + \" \"+ df_datasets['method'].fillna('')+ \" \"+ df_datasets['notes'].fillna('') +\" \"+  df_datasets['keywords'].fillna('') +\" \"+  df_datasets['locations'].fillna('')\n",
    "df_analyse = df_datasets[(df_datasets.done == 1)]\n",
    "len(df_analyse.index)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1fe6db51",
   "metadata": {},
   "source": [
    "Datasets non valid :"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "id": "dc432443",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "valid            77\n",
       "location         17\n",
       "Experimental      4\n",
       "eDNA              2\n",
       "No biological     1\n",
       "Fossils           1\n",
       "Name: non_valid, dtype: int64"
      ]
     },
     "execution_count": 238,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_datasets.non_valid.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "id": "2595dd7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "85"
      ]
     },
     "execution_count": 227,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_analyse = df_analyse[~(df_analyse.non_valid == \"location\")]\n",
    "nb_analyse = len(df_analyse.index)\n",
    "nb_analyse"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2898a182",
   "metadata": {},
   "source": [
    "Proportion of temporal information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "cfb6096c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "source_publication    0.423529\n",
       "not_provided          0.329412\n",
       "main                  0.247059\n",
       "Name: temporal_range_position, dtype: float64"
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_analyse.temporal_range_position.value_counts(normalize = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ba29731",
   "metadata": {},
   "source": [
    "Proportion of duration information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "id": "72d030ec",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-229-d4d6fe393f85>:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  df_analyse[~df_datasets.temporal_range.isna()].temporal_range_position.value_counts(normalize = True)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "source_publication    0.631579\n",
       "main                  0.368421\n",
       "Name: temporal_range_position, dtype: float64"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_analyse[~df_datasets.temporal_range.isna()].temporal_range_position.value_counts(normalize = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4423dd00",
   "metadata": {},
   "source": [
    "Proportion of temporal duration information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "f67d54d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "no                               0.452055\n",
       "site IDs                         0.191781\n",
       "sample coordinates               0.178082\n",
       "site coordinates                 0.054795\n",
       "administrative unit              0.027397\n",
       "geological feature name          0.027397\n",
       "site IDs, site coordinates       0.013699\n",
       "site coordinates, IDs            0.013699\n",
       "distribution models              0.013699\n",
       "site IDs, administrative unit    0.013699\n",
       "Administrative unit              0.013699\n",
       "Name: Geospatial_info_dataset, dtype: float64"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_analyse.Geospatial_info_dataset.value_counts(normalize = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "id": "ed13fd26",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "no                         0.788732\n",
       "administrative unit        0.084507\n",
       "geological feature name    0.056338\n",
       "Administrative unit        0.042254\n",
       "region names               0.014085\n",
       "site coordinates           0.014085\n",
       "Name: Geospatial_info_repo_page_text, dtype: float64"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_analyse.Geospatial_info_repo_page_text.value_counts(normalize = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "id": "4bf3d13e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Map                             0.235294\n",
       "site coordinates                0.205882\n",
       "no                              0.176471\n",
       "administrative unit             0.132353\n",
       "Map, geological feature name    0.073529\n",
       "Map, site coordinates           0.058824\n",
       "Map, range coordinates          0.044118\n",
       "Map, administrative unit        0.044118\n",
       "geological feature name         0.029412\n",
       "Name: Geospatial_info_article_text, dtype: float64"
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_analyse.Geospatial_info_article_text.value_counts(normalize = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "id": "a76e8b66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "L    0.376471\n",
       "M    0.282353\n",
       "H    0.200000\n",
       "X    0.141176\n",
       "Name: relevance_dataset, dtype: float64"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_analyse.relevance_dataset.value_counts(normalize = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "cf04b227",
   "metadata": {},
   "outputs": [],
   "source": [
    "def kw_lists(x, lkw):\n",
    "    v = 0\n",
    "    for kw in lkw:\n",
    "        cnt = kw_list(x, kw)\n",
    "        if cnt == 1:\n",
    "            v = 1\n",
    "    return(v)\n",
    "\n",
    "def kw_list(x, kw):\n",
    "    cnt = 0\n",
    "    for k in kw:\n",
    "        k = k.replace('\"', \"\")\n",
    "        if (x.count(k) > 0 or x.count(get_plural(k)) > 0):\n",
    "            cnt = 1\n",
    "    return(cnt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "id": "714b587f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['presence-absence',\n",
       " 'sampling plot',\n",
       " 'relative proportion',\n",
       " 'inventory',\n",
       " 'occupancy']"
      ]
     },
     "execution_count": 250,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dtype=\"presence-absence\"\n",
    "vocabulary[vocabulary.type_dataset == dtype].keywords.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e8af08b2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\bin\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "id": "08c475c6",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-321-bc5060d1fd6b>, line 6)",
     "output_type": "error",
     "traceback": [
      "\u001b[1;36m  File \u001b[1;32m\"<ipython-input-321-bc5060d1fd6b>\"\u001b[1;36m, line \u001b[1;32m6\u001b[0m\n\u001b[1;33m    gnfinder text.txt -U > bli.json\u001b[0m\n\u001b[1;37m             ^\u001b[0m\n\u001b[1;31mSyntaxError\u001b[0m\u001b[1;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "for index, row in df_analyse.iterrows():\n",
    "    text = row[\"full_text\"]\n",
    "    with open('C://bin//text.txt', 'w', encoding=\"utf-8\") as f:\n",
    "        f.write(text)\n",
    "        %%cmd\n",
    "        gnfinder text.txt -U > bli.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "3096f9b7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 334,
   "id": "9f7b36f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Microsoft Windows [version 10.0.19044.2130]\r\n",
      "(c) Microsoft Corporation. Tous droits r‚serv‚s.\r\n",
      "\r\n",
      "(base) C:\\bin>dir *.txt 'gnfinder {} -U > {.}.json'\n",
      "\r\n",
      "(base) C:\\bin>"
     ]
    }
   ],
   "source": [
    "%%cmd\n",
    "dir *.txt 'gnfinder {} -U > {.}.json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "id": "bba43284",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Microsoft Windows [version 10.0.19044.2130]\r\n",
      "(c) Microsoft Corporation. Tous droits r‚serv‚s.\r\n",
      "\r\n",
      "(base) C:\\bin>\"gnfinder {}.txt -U > bli.json\".format(idx)\n",
      "\r\n",
      "(base) C:\\bin>"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'\"gnfinder {}.txt -U > bli.json\".format' n'est pas reconnu en tant que commande interne\r\n",
      "ou externe, un programme ex‚cutable ou un fichier de commandes.\r\n"
     ]
    }
   ],
   "source": [
    "%%cmd\n",
    "\"gnfinder {}.txt -U > bli2.json\".format(idx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 337,
   "id": "1ab96a21",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Microsoft Windows [version 10.0.19044.2130]\r\n",
      "(c) Microsoft Corporation. Tous droits r‚serv‚s.\r\n",
      "\r\n",
      "(base) C:\\bin>for /r %i in (*.txt) do echo %i\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\0.txt \r\n",
      "C:\\bin\\0.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\1.txt \r\n",
      "C:\\bin\\1.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\10.txt \r\n",
      "C:\\bin\\10.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\100.txt \r\n",
      "C:\\bin\\100.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\101.txt \r\n",
      "C:\\bin\\101.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\11.txt \r\n",
      "C:\\bin\\11.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\12.txt \r\n",
      "C:\\bin\\12.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\13.txt \r\n",
      "C:\\bin\\13.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\14.txt \r\n",
      "C:\\bin\\14.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\15.txt \r\n",
      "C:\\bin\\15.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\16.txt \r\n",
      "C:\\bin\\16.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\17.txt \r\n",
      "C:\\bin\\17.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\18.txt \r\n",
      "C:\\bin\\18.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\19.txt \r\n",
      "C:\\bin\\19.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\2.txt \r\n",
      "C:\\bin\\2.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\20.txt \r\n",
      "C:\\bin\\20.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\21.txt \r\n",
      "C:\\bin\\21.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\22.txt \r\n",
      "C:\\bin\\22.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\23.txt \r\n",
      "C:\\bin\\23.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\24.txt \r\n",
      "C:\\bin\\24.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\25.txt \r\n",
      "C:\\bin\\25.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\26.txt \r\n",
      "C:\\bin\\26.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\27.txt \r\n",
      "C:\\bin\\27.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\28.txt \r\n",
      "C:\\bin\\28.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\29.txt \r\n",
      "C:\\bin\\29.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\3.txt \r\n",
      "C:\\bin\\3.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\30.txt \r\n",
      "C:\\bin\\30.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\31.txt \r\n",
      "C:\\bin\\31.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\32.txt \r\n",
      "C:\\bin\\32.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\33.txt \r\n",
      "C:\\bin\\33.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\34.txt \r\n",
      "C:\\bin\\34.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\35.txt \r\n",
      "C:\\bin\\35.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\36.txt \r\n",
      "C:\\bin\\36.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\37.txt \r\n",
      "C:\\bin\\37.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\38.txt \r\n",
      "C:\\bin\\38.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\39.txt \r\n",
      "C:\\bin\\39.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\4.txt \r\n",
      "C:\\bin\\4.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\40.txt \r\n",
      "C:\\bin\\40.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\41.txt \r\n",
      "C:\\bin\\41.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\42.txt \r\n",
      "C:\\bin\\42.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\43.txt \r\n",
      "C:\\bin\\43.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\44.txt \r\n",
      "C:\\bin\\44.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\45.txt \r\n",
      "C:\\bin\\45.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\46.txt \r\n",
      "C:\\bin\\46.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\47.txt \r\n",
      "C:\\bin\\47.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\48.txt \r\n",
      "C:\\bin\\48.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\49.txt \r\n",
      "C:\\bin\\49.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\5.txt \r\n",
      "C:\\bin\\5.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\50.txt \r\n",
      "C:\\bin\\50.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\51.txt \r\n",
      "C:\\bin\\51.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\52.txt \r\n",
      "C:\\bin\\52.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\53.txt \r\n",
      "C:\\bin\\53.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\54.txt \r\n",
      "C:\\bin\\54.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\55.txt \r\n",
      "C:\\bin\\55.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\56.txt \r\n",
      "C:\\bin\\56.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\57.txt \r\n",
      "C:\\bin\\57.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\58.txt \r\n",
      "C:\\bin\\58.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\59.txt \r\n",
      "C:\\bin\\59.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\6.txt \r\n",
      "C:\\bin\\6.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\60.txt \r\n",
      "C:\\bin\\60.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\61.txt \r\n",
      "C:\\bin\\61.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\62.txt \r\n",
      "C:\\bin\\62.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\63.txt \r\n",
      "C:\\bin\\63.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\64.txt \r\n",
      "C:\\bin\\64.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\65.txt \r\n",
      "C:\\bin\\65.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\66.txt \r\n",
      "C:\\bin\\66.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\67.txt \r\n",
      "C:\\bin\\67.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\68.txt \r\n",
      "C:\\bin\\68.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\69.txt \r\n",
      "C:\\bin\\69.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\7.txt \r\n",
      "C:\\bin\\7.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\70.txt \r\n",
      "C:\\bin\\70.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\71.txt \r\n",
      "C:\\bin\\71.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\72.txt \r\n",
      "C:\\bin\\72.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\73.txt \r\n",
      "C:\\bin\\73.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\74.txt \r\n",
      "C:\\bin\\74.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\75.txt \r\n",
      "C:\\bin\\75.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\76.txt \r\n",
      "C:\\bin\\76.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\77.txt \r\n",
      "C:\\bin\\77.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\78.txt \r\n",
      "C:\\bin\\78.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\79.txt \r\n",
      "C:\\bin\\79.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\8.txt \r\n",
      "C:\\bin\\8.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\80.txt \r\n",
      "C:\\bin\\80.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\81.txt \r\n",
      "C:\\bin\\81.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\82.txt \r\n",
      "C:\\bin\\82.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\83.txt \r\n",
      "C:\\bin\\83.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\84.txt \r\n",
      "C:\\bin\\84.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\85.txt \r\n",
      "C:\\bin\\85.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\86.txt \r\n",
      "C:\\bin\\86.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\87.txt \r\n",
      "C:\\bin\\87.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\88.txt \r\n",
      "C:\\bin\\88.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\89.txt \r\n",
      "C:\\bin\\89.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\9.txt \r\n",
      "C:\\bin\\9.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\90.txt \r\n",
      "C:\\bin\\90.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\91.txt \r\n",
      "C:\\bin\\91.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\92.txt \r\n",
      "C:\\bin\\92.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\93.txt \r\n",
      "C:\\bin\\93.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\94.txt \r\n",
      "C:\\bin\\94.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\95.txt \r\n",
      "C:\\bin\\95.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\96.txt \r\n",
      "C:\\bin\\96.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\97.txt \r\n",
      "C:\\bin\\97.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\98.txt \r\n",
      "C:\\bin\\98.txt\r\n",
      "\r\n",
      "(base) C:\\bin>echo C:\\bin\\99.txt \r\n",
      "C:\\bin\\99.txt\r\n",
      "\r\n",
      "(base) C:\\bin>"
     ]
    }
   ],
   "source": [
    "%%cmd\n",
    "for /r %i in (*.txt) do echo %i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 328,
   "id": "f813a300",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Microsoft Windows [version 10.0.19044.2130]\r\n",
      "(c) Microsoft Corporation. Tous droits r‚serv‚s.\r\n",
      "\r\n",
      "(base) C:\\bin>gnfinder 5.txt -U > bli.json\n",
      "\r\n",
      "(base) C:\\bin>"
     ]
    }
   ],
   "source": [
    "%%cmd\n",
    "gnfinder 5.txt -U > bli.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "id": "dd29f766",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Microsoft Windows [version 10.0.19044.2130]\r\n",
      "(c) Microsoft Corporation. Tous droits r‚serv‚s.\r\n",
      "\r\n",
      "(base) C:\\bin>gnfinder testab.txt -U -f tsv > bli2.json \n",
      "\r\n",
      "(base) C:\\bin>"
     ]
    }
   ],
   "source": [
    "%%cmd\n",
    "gnfinder testab.txt -U -f tsv > bli2.json "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "id": "f1d07316",
   "metadata": {},
   "outputs": [],
   "source": [
    "filename = 'C://bin//{}.json'.format(5)\n",
    "inputname = 'C://bin//{}.txt'.format(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "id": "bb38914d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Microsoft Windows [version 10.0.19044.2130]\r\n",
      "(c) Microsoft Corporation. Tous droits r‚serv‚s.\r\n",
      "\r\n",
      "(base) C:\\bin>file=\"thisfile.txt\"\n",
      "\r\n",
      "(base) C:\\bin>echo \"filename: ${file%.*}\"\n",
      "\"filename: ${file%.*}\"\r\n",
      "\r\n",
      "(base) C:\\bin>"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'file' n'est pas reconnu en tant que commande interne\r\n",
      "ou externe, un programme ex‚cutable ou un fichier de commandes.\r\n"
     ]
    }
   ],
   "source": [
    "%%cmd\n",
    "file=\"thisfile.txt\"\n",
    "echo \"filename: ${file%.*}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "9001aef5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Microsoft Windows [version 10.0.19044.2130]\r\n",
      "(c) Microsoft Corporation. Tous droits r‚serv‚s.\r\n",
      "\r\n",
      "(base) C:\\bin>for /r %i in (*.txt) do gnfinder %i -U > text.json \n",
      "\r\n",
      "(base) C:\\bin>gnfinder C:\\bin\\text.txt -U   1>text.json \r\n",
      "\r\n",
      "(base) C:\\bin>"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "id": "0cd429f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "dtype_dic = dict()\n",
    "rel_datatype = []\n",
    "for index, row in df_analyse.iterrows():\n",
    "    data_type = [x.strip() for x in row.data_type.split(\",\")]\n",
    "    found = 0\n",
    "    for dtype in data_type:\n",
    "        lkw = vocabulary[vocabulary.type_dataset == dtype].keywords.tolist()\n",
    "        found = kw_list(row.full_text, lkw)\n",
    "\n",
    "    if row.relevant_data_type == 1:\n",
    "        rel_datatype.append(found)\n",
    "        if dtype in dtype_dic.keys():\n",
    "            dtype_dic[dtype].append(found)\n",
    "        else:\n",
    "            dtype_dic[dtype] = [found]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "id": "0c2857c9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'density': [1, 0, 0, 1, 0, 0, 0],\n",
       " 'presence-absence': [0, 1, 0, 0, 0, 1, 1, 0, 0],\n",
       " 'abundance': [1, 1, 1, 0, 1, 1, 1],\n",
       " 'presence only': [0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 0],\n",
       " 'distribution': [1, 1, 1, 0],\n",
       " 'other': [0],\n",
       " 'EBV genetic analysis': [0, 0, 0, 0, 0, 0, 0, 0, 0],\n",
       " 'home range': [0],\n",
       " 'presence-only': [0]}"
      ]
     },
     "execution_count": 270,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dtype_dic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "id": "73fa7ff5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.34"
      ]
     },
     "execution_count": 265,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len([x for x in rel_datatype if x ==1])/len(rel_datatype)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "id": "8a09eda9",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "module 'urllib' has no attribute 'urlopen'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-234-fbefc0c9dc0d>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[0mlink\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\"https://doi.org/10.1038/hdy.2016.92\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 4\u001b[1;33m \u001b[0mf\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0murllib\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0murlopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlink\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      5\u001b[0m \u001b[0mmyfile\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mAttributeError\u001b[0m: module 'urllib' has no attribute 'urlopen'"
     ]
    }
   ],
   "source": [
    "import urllib\n",
    "\n",
    "link = \n",
    "f = urllib.urlopen(link)\n",
    "myfile = f.read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "id": "7e77b4e9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'<!DOCTYPE html>\\n<html lang=\"en\" class=\"grade-c\">\\n<head>\\n    <title>An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment | Heredity</title>\\n    \\n        \\n    \\n\\n<link rel=\"preconnect\" href=\"https://push-content.springernature.io\" crossorigin>\\n\\n<meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\">\\n<meta name=\"applicable-device\" content=\"pc,mobile\">\\n<meta name=\"viewport\" content=\"width=device-width,initial-scale=1.0,maximum-scale=5,user-scalable=yes\">\\n\\n<script data-test=\"dataLayer\">\\n    window.dataLayer = [{\"content\":{\"category\":{\"contentType\":\"original article\",\"legacy\":{\"webtrendsPrimaryArticleType\":\"research\",\"webtrendsSubjectTerms\":\"evolutionary-genetics\",\"webtrendsContentCategory\":null,\"webtrendsContentCollection\":null,\"webtrendsContentGroup\":\"Heredity\",\"webtrendsContentGroupType\":null,\"webtrendsContentSubGroup\":\"Original Article\"}},\"article\":{\"doi\":\"10.1038/hdy.2016.92\"},\"attributes\":{\"cms\":null,\"deliveryPlatform\":\"oscar\",\"copyright\":{\"open\":false,\"legacy\":{\"webtrendsLicenceType\":null}}},\"contentInfo\":{\"authors\":[\"A Bourret\",\"D Garant\"],\"publishedAt\":1477440000,\"publishedAtString\":\"2016-10-26\",\"title\":\"An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment\",\"legacy\":null,\"publishedAtTime\":null,\"documentType\":\"aplusplus\"},\"journal\":{\"pcode\":\"hdy\",\"title\":\"heredity\",\"volume\":\"118\",\"issue\":\"3\"},\"authorization\":{\"status\":true},\"features\":[{\"name\":\"furtherReadingSection\",\"present\":true}],\"collection\":null},\"page\":{\"category\":{\"pageType\":\"article\"},\"attributes\":{\"template\":\"mosaic\",\"featureFlags\":[{\"name\":\"nature-onwards-journey\",\"active\":false},{\"name\":\"recommendationAB\",\"active\":false}],\"testGroup\":null},\"search\":null},\"privacy\":{},\"version\":\"1.0.0\",\"product\":null,\"session\":null,\"user\":null,\"backHalfContent\":true,\"country\":\"CA\",\"hasBody\":true,\"uneditedManuscript\":false,\"twitterId\":[\"o3xnx\",\"o43y9\",\"o3ef7\"],\"japan\":false}];\\n    window.dataLayer.push({\\n        ga4MeasurementId: \\'G-ERRNTNZ807\\',\\n        ga360TrackingId: \\'UA-71668177-1\\',\\n        twitterId: [\\'3xnx\\', \\'o43y9\\', \\'o3ef7\\'],\\n        ga4ServerUrl: \\'https://collect.nature.com\\',\\n        imprint: \\'nature\\'\\n    });\\n</script>\\n\\n<script>\\n    (function(w, d) {\\n        w.config = w.config || {};\\n        w.config.mustardcut = false;\\n\\n        \\n        if (w.matchMedia && w.matchMedia(\\'only print, only all and (prefers-color-scheme: no-preference), only all and (prefers-color-scheme: light), only all and (prefers-color-scheme: dark)\\').matches) {\\n            w.config.mustardcut = true;\\n            d.classList.add(\\'js\\');\\n            d.classList.remove(\\'grade-c\\');\\n        }\\n    })(window, document.documentElement);\\n</script>\\n \\n\\n\\n\\n     \\n    \\n    \\n        \\n    \\n    <style>@media only print, only all and (prefers-color-scheme: no-preference), only all and (prefers-color-scheme: light), only all and (prefers-color-scheme: dark) { html{text-size-adjust:100%;box-sizing:border-box;font-family:sans-serif;font-size:100%;height:100%;line-height:1.15;overflow-y:scroll}body{background:#eee;color:#222;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;line-height:1.76;min-height:100%}article,aside,header,main,nav,section{display:block}h1{font-size:2em}a,sup{vertical-align:baseline}a{background-color:transparent;color:#069;overflow-wrap:break-word;text-decoration:underline;word-break:break-word}b{font-weight:bolder}sup{font-size:75%;line-height:0;position:relative;top:-.5em}img{border:0;height:auto;max-width:100%;vertical-align:middle}svg:not(:root){overflow:hidden}button,input,select{font-family:sans-serif;font-size:100%;line-height:1.15}select{margin:0}button,input{overflow:visible}button,select{text-transform:none}[type=submit],button{-webkit-appearance:button}[hidden]{display:none}button{border-radius:0;cursor:pointer}.c-card--major .c-card__title,.u-h1,.u-h2,button,h1{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif}.c-card--major .c-card__title,.u-h1,.u-h2,h1{font-weight:700}h1{font-size:2rem;letter-spacing:-.0390625rem;line-height:2.25rem}.c-card--major .c-card__title,.u-h2{font-size:1.5rem;letter-spacing:-.0117156rem;line-height:1.6rem;margin-bottom:8px}.u-h3,h2{letter-spacing:-.0117156rem}h2{font-size:1.5rem;line-height:1.6rem}.u-h3{font-size:1.25rem;margin-bottom:8px}.c-card__title,.u-h3,h2{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-weight:700}.c-card__title,.u-h3{line-height:1.4rem}.c-article-editorial-summary__container .c-article-editorial-summary__article-title{font-size:1.25rem;margin-bottom:8px}.c-card__title,h3{font-size:1.25rem}.u-h4{margin-bottom:8px}.c-article-editorial-summary__container .c-article-editorial-summary__article-title,.c-reading-companion__figure-title,.u-h4,h3,h4,h5,h6{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-weight:700;letter-spacing:-.0117156rem;line-height:1.4rem}.c-reading-companion__figure-title,.u-h4{font-size:1.125rem}body,button,div,form,h1,h2,h3,input{margin:0;padding:0}p{padding:0}nav ol,nav ul{list-style:none none}body{font-size:1.125rem}p{margin:0 0 28px}ol,ul{margin-bottom:28px;margin-top:0}p:empty{display:none}.sans-serif{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif}.article-page{background:#fff}.c-article-header{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;margin-bottom:40px}.c-article-identifiers{color:#6f6f6f;display:flex;flex-wrap:wrap;font-size:1rem;line-height:1.3;list-style:none;margin:0 0 8px;padding:0}.c-article-identifiers__item{border-right:1px solid #6f6f6f;list-style:none;margin-right:8px;padding-right:8px}.c-article-identifiers__item:last-child{border-right:0;margin-right:0;padding-right:0}.c-article-identifiers__open{color:#b74616}.c-article-title{font-size:1.5rem;line-height:1.25;margin-bottom:16px}@media only screen and (min-width:768px){.c-article-title{font-size:1.875rem;line-height:1.2}}.c-article-author-list{display:inline;font-size:1rem;list-style:none;margin:0 8px 0 0;padding:0;width:100%}.c-article-author-list__item{display:inline;padding-right:0}.c-article-author-list svg{margin-left:4px}.c-article-author-list__show-more{display:none;margin-right:4px}.c-article-author-list__button,.js .c-article-author-list__item--hide,.js .c-article-author-list__show-more{display:none}.js .c-article-author-list--long .c-article-author-list__show-more,.js .c-article-author-list--long+.c-article-author-list__button{display:inline}@media only screen and (max-width:539px){.js .c-article-author-list__item--hide-small-screen{display:none}.js .c-article-author-list--short .c-article-author-list__show-more,.js .c-article-author-list--short+.c-article-author-list__button{display:inline}}#uptodate-client,.js .c-article-author-list--expanded .c-article-author-list__show-more{display:none!important}.js .c-article-author-list--expanded .c-article-author-list__item--hide-small-screen{display:inline!important}.c-article-author-list__button{background:#ebf1f5;border:4px solid #ebf1f5;border-radius:20px;color:#666;font-size:.875rem;line-height:1.4;padding:2px 11px 2px 8px;text-decoration:none}.c-article-author-list__button svg{margin:1px 4px 0 0}.c-article-author-list__button:hover{background:#069;border-color:transparent;color:#fff}.c-article-info-details{font-size:1rem;margin-bottom:8px;margin-top:16px}.c-article-info-details__cite-as{border-left:1px solid #6f6f6f;margin-left:8px;padding-left:8px}.c-article-metrics-bar{display:flex;flex-wrap:wrap;font-size:1rem;line-height:1.3}.c-article-metrics-bar__wrapper{margin:0 0 16px}.c-article-metrics-bar__item{align-items:baseline;border-right:1px solid #6f6f6f;margin-right:8px}.c-article-metrics-bar__item:last-child{border-right:0}.c-article-metrics-bar__count{font-weight:700;margin:0}.c-article-metrics-bar__label{color:#626262;font-style:normal;font-weight:400;margin:0 10px 0 5px}.c-article-metrics-bar__details{margin:0}.c-article-main-column{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;margin-right:8.6%;width:60.2%}@media only screen and (max-width:1023px){.c-article-main-column{margin-right:0;width:100%}}.c-article-extras{float:left;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;width:31.2%}@media only screen and (max-width:1023px){.c-article-extras{display:none}}.c-article-associated-content__container .c-article-associated-content__title,.c-article-section__title{border-bottom:2px solid #d5d5d5;font-size:1.25rem;margin:0;padding-bottom:8px}@media only screen and (min-width:768px){.c-article-associated-content__container .c-article-associated-content__title,.c-article-section__title{font-size:1.5rem;line-height:1.24}}.c-article-section{clear:both}.c-article-section__content{margin-bottom:40px;padding-top:8px}@media only screen and (max-width:1023px){.c-article-section__content{padding-left:0}}.c-article-authors-search{margin-bottom:24px;margin-top:0}.c-article-authors-search__item,.c-article-authors-search__title{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif}.c-article-authors-search__title{color:#626262;font-size:1.05rem;font-weight:700;margin:0;padding:0}.c-article-authors-search__item{font-size:1rem}.c-article-authors-search__text{margin:0}.c-code-block{border:1px solid #eee;font-family:monospace;margin:0 0 24px;padding:20px}.c-code-block__heading{font-weight:400;margin-bottom:16px}.c-code-block__line{display:block;overflow-wrap:break-word;white-space:pre-wrap}.c-article-share-box__no-sharelink-info{font-size:.813rem;font-weight:700;margin-bottom:24px;padding-top:4px}.c-article-share-box__only-read-input{border:1px solid #d5d5d5;display:inline-block;font-size:.875rem;font-weight:700;margin-bottom:8px;padding:8px 10px}.c-article-share-box__button--link-like{background-color:transparent;border:0;color:#069;cursor:pointer;font-size:.875rem;margin-bottom:8px;margin-left:10px}.c-article-editorial-summary__container{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:1rem}.c-article-editorial-summary__container .c-article-editorial-summary__content p:last-child{margin-bottom:0}.c-article-editorial-summary__container .c-article-editorial-summary__content--less{max-height:9.5rem;overflow:hidden}.c-article-editorial-summary__container .c-article-editorial-summary__button{background-color:#fff;border:0;color:#069;font-size:.875rem;margin-bottom:16px;text-decoration:none}.c-article-editorial-summary__container .c-article-editorial-summary__button.active,.c-article-editorial-summary__container .c-article-editorial-summary__button.hover,.c-article-editorial-summary__container .c-article-editorial-summary__button:active,.c-article-editorial-summary__container .c-article-editorial-summary__button:hover{text-decoration:underline;text-decoration-skip-ink:auto}.c-article-editorial-summary__container .c-article-editorial-summary__button:focus{outline:3px solid #fece3e;will-change:transform}.c-article-associated-content__container .c-article-associated-content__collection-label{font-size:.875rem;line-height:1.4}.c-article-associated-content__container .c-article-associated-content__collection-title{line-height:1.3}.c-context-bar{box-shadow:0 0 10px 0 rgba(51,51,51,.2);position:relative;width:100%}.c-context-bar__title{display:none}.c-reading-companion{clear:both;min-height:389px}.c-reading-companion__sticky{max-width:389px}.c-reading-companion__scroll-pane{margin:0;min-height:200px;overflow:hidden auto}.c-reading-companion__tabs{display:flex;flex-flow:row nowrap;font-size:1rem;list-style:none;margin:0 0 8px;padding:0}.c-reading-companion__tabs>li{flex-grow:1}.c-reading-companion__tab{background-color:#eee;border:1px solid #d5d5d5;border-image:initial;border-left-width:0;color:#069;font-size:1rem;padding:8px 8px 8px 15px;text-align:left;width:100%}.c-reading-companion__tabs li:first-child .c-reading-companion__tab{border-left-width:1px}.c-reading-companion__tab--active{background-color:#fff;border-bottom:1px solid #fff;color:#222;font-weight:700}.c-reading-companion__sections-list{list-style:none;padding:0}.c-reading-companion__figures-list,.c-reading-companion__references-list{list-style:none;min-height:389px;padding:0}.c-reading-companion__sections-list{margin:0 0 8px;min-height:50px}.c-reading-companion__section-item{font-size:1rem;padding:0}.c-reading-companion__section-item a{display:block;line-height:1.5;overflow:hidden;padding:8px 0 8px 16px;text-overflow:ellipsis;white-space:nowrap}.c-reading-companion__figure-item{border-top:1px solid #d5d5d5;font-size:1rem;padding:16px 8px 16px 0}.c-reading-companion__figure-item:first-child{border-top:none;padding-top:8px}.c-reading-companion__reference-item{border-top:1px solid #d5d5d5;font-size:1rem;padding:8px 8px 8px 16px}.c-reading-companion__reference-item:first-child{border-top:none}.c-reading-companion__reference-item a{word-break:break-word}.c-reading-companion__reference-citation{display:inline}.c-reading-companion__reference-links{font-size:.813rem;font-weight:700;list-style:none;margin:8px 0 0;padding:0;text-align:right}.c-reading-companion__reference-links>a{display:inline-block;padding-left:8px}.c-reading-companion__reference-links>a:first-child{display:inline-block;padding-left:0}.c-reading-companion__figure-title{display:block;margin:0 0 8px}.c-reading-companion__figure-links{display:flex;justify-content:space-between;margin:8px 0 0}.c-reading-companion__figure-links>a{align-items:center;display:flex}.c-reading-companion__figure-full-link svg{height:.8em;margin-left:2px}.c-reading-companion__panel{border-top:none;display:none;margin-top:0;padding-top:0}.c-cod,.c-reading-companion__panel--active{display:block}.c-cod{font-size:1rem;width:100%}.c-cod__form{background:#ebf0f3}.c-cod__prompt{font-size:1.125rem;line-height:1.3;margin:0 0 24px}.c-cod__label{display:block;margin:0 0 4px}.c-cod__row{display:flex;margin:0 0 16px}.c-cod__row:last-child{margin:0}.c-cod__input{border:1px solid #d5d5d5;border-radius:2px;flex-basis:75%;flex-shrink:0;margin:0;padding:13px}.c-cod__input--submit{background-color:#069;border:1px solid #069;color:#fff;flex-shrink:1;margin-left:8px;transition:background-color .2s ease-out 0s,color .2s ease-out 0s}.c-cod__input--submit-single{flex-basis:100%;flex-shrink:0;margin:0}.c-cod__input--submit:focus,.c-cod__input--submit:hover{background-color:#fff;color:#069}.c-pdf-download__link .u-icon{padding-top:2px}.c-pdf-download{display:flex;margin-bottom:16px;max-height:48px}@media only screen and (min-width:540px){.c-pdf-download{max-height:none}}@media only screen and (min-width:1024px){.c-pdf-download{max-height:48px}}.c-pdf-download__link{display:flex;flex:1 1 0%;padding:13px 24px!important}.c-pdf-download__link:hover{text-decoration:none}.c-pdf-download__text{padding-right:4px}@media only screen and (max-width:539px){.c-pdf-download__text{text-transform:capitalize}}@media only screen and (min-width:540px){.c-pdf-download__text{padding-right:8px}}.c-context-bar--sticky .c-pdf-download{display:block;margin-bottom:0;white-space:nowrap}@media only screen and (max-width:539px){.c-pdf-download .u-sticky-visually-hidden{clip:rect(0,0,0,0);border:0;height:1px;margin:-100%;overflow:hidden;padding:0;position:absolute!important;width:1px}}.c-pdf-container{display:flex;justify-content:flex-end}@media only screen and (max-width:539px){.c-pdf-container .c-pdf-download{display:flex;flex-basis:100%}}.c-pdf-container .c-pdf-download+.c-pdf-download{margin-left:16px}.c-article-extras .c-pdf-container .c-pdf-download{width:100%}.c-article-extras .c-pdf-container .c-pdf-download+.c-pdf-download{margin-left:0}@media only screen and (min-width:540px){.c-context-bar--sticky .c-pdf-download__link{align-items:center;flex:1 1 183px}}@media only screen and (max-width:320px){.c-context-bar--sticky .c-pdf-download__link{padding:16px}}.article-page--commercial .c-article-main-column .c-pdf-button__container .c-pdf-download{display:none}@media only screen and (max-width:1023px){.article-page--commercial .c-article-main-column .c-pdf-button__container .c-pdf-download{display:block}}.c-article-metrics__heading a,.c-article-metrics__posts .c-card__title a{color:inherit}.c-article-metrics__posts .c-card__title{font-size:1.05rem}.c-article-metrics__posts .c-card__title+span{color:#6f6f6f;font-size:1rem}p{overflow-wrap:break-word;word-break:break-word}.c-ad{text-align:center}@media only screen and (min-width:320px){.c-ad{padding:8px}}.c-ad--728x90{background-color:#ccc;display:none}.c-ad--728x90 .c-ad__inner{min-height:calc(1.5em + 94px)}@media only screen and (min-width:768px){.js .c-ad--728x90{display:none}}.c-ad__label{color:#333;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:.875rem;font-weight:400;line-height:1.5;margin-bottom:4px}.c-author-list{color:#6f6f6f;font-family:inherit;font-size:1rem;line-height:inherit;list-style:none;margin:0;padding:0}.c-author-list>li,.c-breadcrumbs>li,.c-footer__list--inline>li,.js .c-author-list,.u-list-comma-separated>li,.u-list-inline>li{display:inline}.c-author-list>li:not(:first-child):not(:last-child):before{content:\", \"}.c-author-list>li:not(:only-child):last-child:before{content:\" & \"}.c-author-list--compact{font-size:.875rem;line-height:1.4}.c-author-list--truncated>li:not(:only-child):last-child:before{content:\" ... \"}.c-header__menu--global svg,.js .c-author-list__hide{display:none;visibility:hidden}.c-meta{color:inherit;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:.875rem;line-height:1.4;list-style:none;margin:0;padding:0}.c-meta--large{font-size:1rem}.c-meta--large .c-meta__item{margin-bottom:8px}.c-meta__item{display:inline-block;margin-bottom:4px}.c-meta__item:not(:last-child){border-right:1px solid #d5d5d5;margin-right:4px;padding-right:4px}@media only screen and (max-width:539px){.c-meta__item--block-sm-max{display:block}.c-meta__item--block-sm-max:not(:last-child){border-right:none;margin-right:0;padding-right:0}}@media only screen and (min-width:1024px){.c-meta__item--block-at-lg{display:block}.c-meta__item--block-at-lg:not(:last-child){border-right:none;margin-right:0;padding-right:0}}.c-meta__type{font-weight:700;text-transform:none}.c-skip-link{background:#069;bottom:auto;color:#fff;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:.875rem;padding:8px;position:absolute;text-align:center;transform:translateY(-100%);z-index:9999}@media (prefers-reduced-motion:reduce){.c-skip-link{transition:top .3s ease-in-out 0s}}@media print{.c-skip-link{display:none}}.c-skip-link:link{color:#fff}.c-status-message{align-items:center;box-sizing:border-box;display:flex;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:1rem;position:relative;width:100%}.c-card__summary>p:last-child,.c-status-message :last-child{margin-bottom:0}.c-status-message--boxed{background-color:#fff;border:1px solid #eee;border-radius:2px;line-height:1.4;padding:16px}.c-status-message__heading{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:1rem;font-weight:700}.c-status-message__icon{fill:currentcolor;display:inline-block;flex:0 0 auto;height:1.5em;margin-right:8px;transform:translate(0);vertical-align:text-top;width:1.5em}.c-status-message__icon--top{align-self:flex-start}.c-status-message--info .c-status-message__icon{color:#003f8d}.c-status-message--boxed.c-status-message--info{border-bottom:4px solid #003f8d}.c-status-message--error .c-status-message__icon{color:#c40606}.c-status-message--boxed.c-status-message--error{border-bottom:4px solid #c40606}.c-status-message--success .c-status-message__icon{color:#00b8b0}.c-status-message--boxed.c-status-message--success{border-bottom:4px solid #00b8b0}.c-status-message--warning .c-status-message__icon{color:#edbc53}.c-status-message--boxed.c-status-message--warning{border-bottom:4px solid #edbc53}.c-breadcrumbs{color:#000;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:1rem;list-style:none;margin:0;padding:0}.c-breadcrumbs__link{color:#666}.c-breadcrumbs__chevron{fill:#888;margin:4px 4px 0}.c-card{background-color:transparent;border:0;box-shadow:none;display:flex;flex-direction:column;font-size:14px;min-width:0;overflow:hidden;padding:0;position:relative}.c-card--no-shape{background:0 0;border:0;box-shadow:none}.c-card__image{display:flex;justify-content:center;overflow:hidden;padding-bottom:56.25%;position:relative}@supports (aspect-ratio:1/1){.c-card__image{padding-bottom:0}}.c-card__image img{left:0;min-height:100%;min-width:100%;position:absolute}@supports ((-o-object-fit:cover) or (object-fit:cover)){.c-card__image img{height:100%;object-fit:cover;width:100%}}.c-card__body{flex:1 1 auto;padding:16px}.c-card--no-shape .c-card__body{padding:0}.c-card--no-shape .c-card__body:not(:first-child){padding-top:16px}.c-card__title{letter-spacing:-.01875rem;margin-bottom:8px;margin-top:0}[lang=de] .c-card__title{hyphens:auto}.c-card__summary{line-height:1.4}.c-card__summary>p{margin-bottom:5px}.c-card__summary a{text-decoration:underline}.c-card__link:not(.c-card__link--no-block-link):before{bottom:0;content:\"\";left:0;position:absolute;right:0;top:0}.c-card--flush .c-card__body{padding:0}.c-card--major{font-size:1rem}.c-card--dark{background-color:#29303c;border-width:0;color:#e3e4e5}.c-card--dark .c-card__title{color:#fff}.c-card--dark .c-card__link,.c-card--dark .c-card__summary a{color:inherit}.c-header{background-color:#fff;border-bottom:5px solid #000;font-size:1rem;line-height:1.4;margin-bottom:16px}.c-header__row{padding:0;position:relative}.c-header__row:not(:last-child){border-bottom:1px solid #eee}.c-header__split{align-items:center;display:flex;justify-content:space-between}.c-header__logo-container{flex:1 1 0px;line-height:0;margin:8px 24px 8px 0}.c-header__logo{transform:translateZ(0)}.c-header__logo img{max-height:32px}.c-header__container{margin:0 auto;max-width:1280px;padding:0 16px}.c-header__menu{align-items:center;display:flex;flex:0 1 auto;flex-wrap:wrap;font-weight:700;gap:8px 8px;line-height:1.4;list-style:none;margin:0 -8px;padding:0}@media print{.c-header__menu{display:none}}@media only screen and (max-width:1023px){.c-header__menu--hide-lg-max{display:none;visibility:hidden}}.c-header__menu--global{font-weight:400;justify-content:flex-end}@media only screen and (min-width:540px){.c-header__menu--global svg{display:block;visibility:visible}}.c-header__menu--journal{font-size:.875rem;margin:8px 0 8px -8px}@media only screen and (min-width:540px){.c-header__menu--journal{flex-wrap:nowrap;font-size:1rem}}.c-header__item{padding-bottom:0;padding-top:0;position:static}.c-header__item--pipe{border-left:2px solid #eee;padding-left:8px}.c-header__item--padding{padding-bottom:8px;padding-top:8px}@media only screen and (min-width:540px){.c-header__item--dropdown-menu{position:relative}}@media only screen and (min-width:1024px){.c-header__item--hide-lg{display:none;visibility:hidden}}@media only screen and (max-width:767px){.c-header__item--hide-md-max{display:none;visibility:hidden}}.c-header__link{align-items:center;color:inherit;display:inline-flex;gap:4px 4px;padding:8px;white-space:nowrap}.c-header__link svg{transition-duration:.2s}.c-header__show-text{display:none;visibility:hidden}@media only screen and (min-width:540px){.c-header__show-text{display:inline;visibility:visible}}.c-header__dropdown{background-color:#000;border-bottom:1px solid #2f2f2f;color:#eee;font-size:.875rem;line-height:1.2;padding:16px 0}@media print{.c-header__dropdown{display:none}}.c-header__heading{display:inline-block;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:1.25rem;font-weight:400;line-height:1.4;margin-bottom:8px}.c-header__heading--keyline{border-top:1px solid;border-color:#2f2f2f;margin-top:16px;padding-top:16px;width:100%}.c-header__list{display:flex;flex-wrap:wrap;gap:0 16px;list-style:none;margin:0 -8px;padding:0}.c-header__flush{margin:0 -8px}.c-header__visually-hidden{clip:rect(0,0,0,0);border:0;height:1px;margin:-100%;overflow:hidden;padding:0;position:absolute!important;width:1px}.c-header__search-form{margin-bottom:8px}.c-header__search-layout{display:flex;flex-wrap:wrap;gap:16px 16px}.c-header__search-layout>:first-child{flex:999 1 auto}.c-header__search-layout>*{flex:1 1 auto}.c-header__search-layout--max-width{max-width:720px}.c-header__search-button{align-items:center;background-color:transparent;background-image:none;border:1px solid #fff;border-radius:2px;color:#fff;cursor:pointer;display:flex;font-family:sans-serif;font-size:1rem;justify-content:center;line-height:1.15;margin:0;padding:8px 16px;position:relative;text-decoration:none;transition:all .25s ease 0s,color .25s ease 0s,border-color .25s ease 0s;width:100%}.u-button svg,.u-button--primary svg{fill:currentcolor}.c-header__input,.c-header__select{border:1px solid;border-radius:3px;box-sizing:border-box;font-size:1rem;padding:8px 16px;width:100%}.c-header__select{-webkit-appearance:none;background-image:url(\"data:image/svg+xml,%3Csvg height=\\'16\\' viewBox=\\'0 0 16 16\\' width=\\'16\\' xmlns=\\'http://www.w3.org/2000/svg\\'%3E%3Cpath d=\\'m5.58578644 3-3.29289322-3.29289322c-.39052429-.39052429-.39052429-1.02368927 0-1.41421356s1.02368927-.39052429 1.41421356 0l4 4c.39052429.39052429.39052429 1.02368927 0 1.41421356l-4 4c-.39052429.39052429-1.02368927.39052429-1.41421356 0s-.39052429-1.02368927 0-1.41421356z\\' fill=\\'%23333\\' fill-rule=\\'evenodd\\' transform=\\'matrix(0 1 -1 0 11 3)\\'/%3E%3C/svg%3E\");background-position:right .7em top 50%;background-repeat:no-repeat;background-size:1em;box-shadow:0 1px 0 1px rgba(0,0,0,.04);display:block;margin:0;max-width:100%;min-width:150px}@media only screen and (min-width:540px){.c-header__menu--journal .c-header__item--dropdown-menu:last-child .c-header__dropdown.has-tethered{left:auto;right:0}}@media only screen and (min-width:768px){.c-header__menu--journal .c-header__item--dropdown-menu:last-child .c-header__dropdown.has-tethered{left:0;right:auto}}.c-header__dropdown.has-tethered{border-bottom:0;border-radius:0 0 2px 2px;left:0;position:absolute;top:100%;transform:translateY(5px);width:100%;z-index:1}@media only screen and (min-width:540px){.c-header__dropdown.has-tethered{transform:translateY(8px);width:auto}}@media only screen and (min-width:768px){.c-header__dropdown.has-tethered{min-width:225px}}.c-header__dropdown--full-width.has-tethered{padding:32px 0 24px;transform:none;width:100%}.has-tethered .c-header__heading--js-hide{display:none;visibility:hidden}.has-tethered .c-header__list--js-stack{flex-direction:column}.has-tethered .c-header__item--keyline,.has-tethered .c-header__list~.c-header__list .c-header__item:first-child{border-top:1px solid #d5d5d5;margin-top:8px;padding-top:8px}.u-button{align-items:center;background-color:transparent;background-image:none;border:1px solid #069;border-radius:2px;color:#069;cursor:pointer;display:inline-flex;font-family:sans-serif;font-size:1rem;justify-content:center;line-height:1.3;margin:0;padding:8px;position:relative;text-decoration:none;transition:all .25s ease 0s,color .25s ease 0s,border-color .25s ease 0s;width:auto}.u-button--primary{background-color:#069;background-image:none;border:1px solid #069;color:#fff}.u-button--full-width{display:flex;width:100%}.u-display-none{display:none}.js .u-js-hide,.u-hide{display:none;visibility:hidden}.u-visually-hidden{clip:rect(0,0,0,0);border:0;height:1px;margin:-100%;overflow:hidden;padding:0;position:absolute!important;width:1px}@media print{.u-hide-print{display:none}}@media only screen and (min-width:768px){.u-show-at-md{display:block;visibility:visible}}@media only screen and (min-width:1024px){.u-hide-at-lg{display:none;visibility:hidden}}.u-clearfix:after,.u-clearfix:before{content:\"\";display:table}.u-clearfix:after{clear:both}.u-float-left{float:left}.u-icon{fill:currentcolor;display:inline-block;height:1em;transform:translate(0);vertical-align:text-top;width:1em}.u-full-height{height:100%}.u-link-inherit{color:inherit}.u-list-reset{list-style:none;margin:0;padding:0}.u-sans-serif{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif}.u-text-bold{font-weight:700}.u-container{margin:0 auto;max-width:1280px;padding:0 16px}.u-justify-content-space-between{justify-content:space-between}.u-mt-32{margin-top:32px}.u-mb-8{margin-bottom:8px}.u-mb-16{margin-bottom:16px}.u-mb-24{margin-bottom:24px}.u-mb-32{margin-bottom:32px}.c-nature-box svg+.c-article__button-text,.u-ml-8{margin-left:8px}.u-pa-16{padding:16px}html *,html :after,html :before{box-sizing:inherit}.c-article-section__title,.c-article-title{font-weight:700}.c-card__title{line-height:1.4em}.c-article__button{background-color:#069;border:1px solid #069;border-radius:2px;color:#fff;display:flex;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:.875rem;line-height:1.4;margin-bottom:16px;padding:13px;transition:background-color .2s ease-out 0s,color .2s ease-out 0s}.c-article__button,.c-article__button:hover{text-decoration:none}.c-article__button--inverted,.c-article__button:hover{background-color:#fff;color:#069}.c-article__button--inverted:hover{background-color:#069;color:#fff}.c-header__link{text-decoration:inherit}.grade-c-hide{display:block}.u-lazy-ad-wrapper{background-color:#ccc;display:none;min-height:137px}@media only screen and (min-width:768px){.u-lazy-ad-wrapper{display:block}}.c-nature-box{background-color:#fff;border:1px solid #d5d5d5;border-radius:2px;box-shadow:0 0 5px 0 rgba(51,51,51,.1);line-height:1.3;margin-bottom:24px;padding:16px 16px 3px}.c-nature-box__text{font-size:1rem;margin-bottom:16px}.c-nature-box .c-pdf-download{margin-bottom:16px!important}.c-nature-box--version{background-color:#eee}.c-nature-box__wrapper{transform:translateZ(0)}.c-nature-box__wrapper--placeholder{min-height:165px} }</style>\\n\\n\\n\\n\\n    \\n        <link data-test=\"critical-css-handler\" data-inline-css-source=\"critical-css\" rel=\"stylesheet\" href=\"/static/css/enhanced-article-423f305a02.css\" media=\"print\" onload=\"this.media=\\'only print, only all and (prefers-color-scheme: no-preference), only all and (prefers-color-scheme: light), only all and (prefers-color-scheme: dark)\\';this.onload=null\">\\n    \\n    <noscript>\\n        <link rel=\"stylesheet\" type=\"text/css\" href=\"/static/css/enhanced-article-423f305a02.css\" media=\"only print, only all and (prefers-color-scheme: no-preference), only all and (prefers-color-scheme: light), only all and (prefers-color-scheme: dark)\">\\n    </noscript>\\n\\n<link rel=\"stylesheet\" type=\"text/css\" href=\"/static/css/article-print-f85fef52e3.css\" media=\"print\">\\n    \\n\\n\\n\\n<link rel=\"apple-touch-icon\" sizes=\"180x180\" href=/static/images/favicons/nature/apple-touch-icon-f39cb19454.png>\\n<link rel=\"icon\" type=\"image/png\" sizes=\"32x32\" href=/static/images/favicons/nature/favicon-32x32-3fe59ece92.png>\\n<link rel=\"icon\" type=\"image/png\" sizes=\"16x16\" href=/static/images/favicons/nature/favicon-16x16-951651ab72.png>\\n<link rel=\"manifest\" href=/static/manifest.json crossorigin=\"use-credentials\">\\n<link rel=\"mask-icon\" href=/static/images/favicons/nature/safari-pinned-tab-69bff48fe6.svg color=\"#000000\">\\n<link rel=\"shortcut icon\" href=/static/images/favicons/nature/favicon.ico>\\n<meta name=\"msapplication-TileColor\" content=\"#000000\">\\n<meta name=\"msapplication-config\" content=/static/browserconfig.xml>\\n<meta name=\"theme-color\" content=\"#000000\">\\n<meta name=\"application-name\" content=\"Nature\">\\n\\n\\n<script>\\n    (function () {\\n        if ( typeof window.CustomEvent === \"function\" ) return false;\\n        function CustomEvent ( event, params ) {\\n            params = params || { bubbles: false, cancelable: false, detail: null };\\n            var evt = document.createEvent( \\'CustomEvent\\' );\\n            evt.initCustomEvent( event, params.bubbles, params.cancelable, params.detail );\\n            return evt;\\n        }\\n\\n        CustomEvent.prototype = window.Event.prototype;\\n\\n        window.CustomEvent = CustomEvent;\\n    })();\\n</script>\\n\\n\\n\\n<!-- Google Tag Manager -->\\n<script data-test=\"gtm-head\">\\n    window.initGTM = function() {\\n        if (window.config.mustardcut) {\\n            (function (w, d, s, l, i) {\\n                w[l] = w[l] || [];\\n                w[l].push({\\'gtm.start\\': new Date().getTime(), event: \\'gtm.js\\'});\\n                var f = d.getElementsByTagName(s)[0],\\n                        j = d.createElement(s),\\n                        dl = l != \\'dataLayer\\' ? \\'&l=\\' + l : \\'\\';\\n                j.async = true;\\n                j.src = \\'https://collect.nature.com/gtm.js?id=\\' + i + dl;\\n                f.parentNode.insertBefore(j, f);\\n            })(window, document, \\'script\\', \\'dataLayer\\', \\'GTM-MRVXSHQ\\');\\n        }\\n    }\\n</script>\\n<!-- End Google Tag Manager -->\\n\\n    <script>\\n    (function(w,d,t) {\\n        function cc() {\\n            var h = w.location.hostname;\\n            if (h.indexOf(\\'preview-www.nature.com\\') > -1) return;\\n\\n            var e = d.createElement(t),\\n                    s = d.getElementsByTagName(t)[0];\\n\\n            if (h.indexOf(\\'nature.com\\') > -1) {\\n                e.src = \\'https://push-content.springernature.io/pcf_sb_5_1617714720898560639/production_live/consent-bundle-8-12.js\\';\\n                e.setAttribute(\\'onload\\', \"initGTM(window,document,\\'script\\',\\'dataLayer\\',\\'GTM-MRVXSHQ\\')\");\\n            } else {\\n                e.src = \\'/static/js/cookie-consent-es5-bundle-2b0f06c1e4.js\\';\\n                e.setAttribute(\\'data-consent\\', h);\\n            }\\n            s.insertAdjacentElement(\\'afterend\\', e);\\n        }\\n\\n        cc();\\n    })(window,document,\\'script\\');\\n</script>\\n\\n<script id=\"js-position0\">\\n    (function(w, d) {\\n        w.idpVerifyPrefix = \\'https://verify.nature.com\\';\\n        w.ra21Host = \\'https://wayf.springernature.com\\';\\n        var moduleSupport = (function() {\\n            return \\'noModule\\' in d.createElement(\\'script\\');\\n        })();\\n\\n        var polyfillsUrl = function() {\\n            var features = {\\n                \\'IntersectionObserver\\': window.IntersectionObserver,\\n                \\'Promise\\': window.Promise,\\n                \\'URLSearchParams\\': window.URLSearchParams,\\n                \\'Symbol.iterator\\': window.Symbol && Symbol.iterator,\\n                \\'Array.from\\': Array.from,\\n                \\'Array.prototype.includes\\': Array.prototype.includes,\\n                \\'Array.prototype.find\\': Array.prototype.find,\\n                \\'Array.prototype.forEach\\': Array.prototype.forEach,\\n                \\'NodeList.prototype.forEach\\': NodeList.prototype.forEach,\\n                \\'Element.prototype.closest\\': Element.prototype.closest,\\n                \\'Element.prototype.prepend\\': Element.prototype.prepend,\\n                \\'Element.prototype.remove\\': Element.prototype.remove,\\n                \\'Object.assign\\': Object.assign\\n            };\\n            var req = [];\\n            for (var feature in features) {\\n                if (Object.prototype.hasOwnProperty.call(features, feature) && !features[feature]) {\\n                    req.push(feature);\\n                }\\n            }\\n            if (req.length) {\\n                return \\'https://polyfill.io/v3/polyfill.min.js?features=\\' + req.join(\\'%2C\\') + \\'&flags=always\\';\\n            }\\n            return null;\\n        };\\n\\n        if (w.config.mustardcut === true) {\\n            w.loader = {\\n                index: 0,\\n                registered: [],\\n                scripts: [\\n                    {src: polyfillsUrl(), test: \\'polyfills-js\\', noinit: true},\\n                    \\n                        {src: \\'/static/js/global-article-es6-bundle-4d143076a3.js\\', test: \\'global-article-js\\', module: true},\\n                        {src: \\'/static/js/global-article-es5-bundle-e14facac25.js\\', test: \\'global-article-js\\', nomodule: true},\\n                        {src: \\'/static/js/shared-es6-bundle-f2b2e65421.js\\', test: \\'shared-js\\', module: true},\\n                        {src: \\'/static/js/shared-es5-bundle-1210870038.js\\', test: \\'shared-js\\', nomodule: true},\\n                        {src: \\'/static/js/header-150-es6-bundle-5bb959eaa1.js\\', test: \\'header-150-js\\', module: true},\\n                        {src: \\'/static/js/header-150-es5-bundle-c634a291c7.js\\', test: \\'header-150-js\\', nomodule: true}\\n                    \\n                ].filter(function (s) {\\n                    if (s.src === null) return false;\\n                    if (moduleSupport && s.nomodule) return false;\\n                    return !(!moduleSupport && s.module);\\n                }),\\n\\n                register: function (value) {\\n                    this.registered.push(value);\\n                },\\n\\n                ready: function () {\\n                    if (this.registered.length === this.scripts.length) {\\n                        this.registered.forEach(function (fn) {\\n                            if (typeof fn === \\'function\\') {\\n                                setTimeout(fn, 0); \\n                            }\\n                        });\\n                        this.ready = function () {};\\n                    }\\n                },\\n\\n                insert: function (s) {\\n                    var t = d.getElementById(\\'js-position\\' + this.index);\\n                    if (t && t.insertAdjacentElement) {\\n                        t.insertAdjacentElement(\\'afterend\\', s);\\n                    } else {\\n                        d.head.appendChild(s);\\n                    }\\n                    ++this.index;\\n                },\\n\\n                createScript: function (script, beforeLoad) {\\n                    var s = d.createElement(\\'script\\');\\n                    s.id = \\'js-position\\' + (this.index + 1);\\n                    s.setAttribute(\\'data-test\\', script.test);\\n                    if (beforeLoad) {\\n                        s.defer = \\'defer\\';\\n                        s.onload = function () {\\n                            if (script.noinit) {\\n                                loader.register(true);\\n                            }\\n                            if (d.readyState === \\'interactive\\' || d.readyState === \\'complete\\') {\\n                                loader.ready();\\n                            }\\n                        };\\n                    } else {\\n                        s.async = \\'async\\';\\n                    }\\n                    s.src = script.src;\\n                    return s;\\n                },\\n\\n                init: function () {\\n                    this.scripts.forEach(function (s) {\\n                        loader.insert(loader.createScript(s, true));\\n                    });\\n\\n                    d.addEventListener(\\'DOMContentLoaded\\', function () {\\n                        loader.ready();\\n                        \\n                            var conditionalScripts = [\\n                                {match: \\'div[data-pan-container]\\', src: \\'/static/js/pan-zoom-es6-bundle-885b4d4cd7.js\\', test: \\'pan-zoom-js\\',  module: true },\\n                                {match: \\'div[data-pan-container]\\', src: \\'/static/js/pan-zoom-es5-bundle-6a76e92cf3.js\\', test: \\'pan-zoom-js\\',  nomodule: true },\\n                                {match: \\'math,span.mathjax-tex\\', src: \\'/static/js/math-es6-bundle-8756d483f5.js\\', test: \\'math-js\\', module: true},\\n                                {match: \\'math,span.mathjax-tex\\', src: \\'/static/js/math-es5-bundle-25e28a6fc8.js\\', test: \\'math-js\\', nomodule: true}\\n                            ];\\n                        \\n\\n                        if (conditionalScripts) {\\n                            conditionalScripts.filter(function (script) {\\n                                return !!document.querySelector(script.match) && !((moduleSupport && script.nomodule) || (!moduleSupport && script.module));\\n                            }).forEach(function (script) {\\n                                loader.insert(loader.createScript(script));\\n                            });\\n                        }\\n                    }, false);\\n                }\\n            };\\n            loader.init();\\n        }\\n    })(window, document);\\n</script>\\n\\n\\n\\n<meta name=\"robots\" content=\"noarchive\">\\n<meta name=\"access\" content=\"Yes\">\\n\\n\\n<link rel=\"search\" href=\"https://www.nature.com/search\">\\n<link rel=\"search\" href=\"https://www.nature.com/opensearch/opensearch.xml\" type=\"application/opensearchdescription+xml\" title=\"nature.com\">\\n<link rel=\"search\" href=\"https://www.nature.com/opensearch/request\" type=\"application/sru+xml\" title=\"nature.com\">\\n\\n\\n\\n\\n\\n    \\n    <script type=\"application/ld+json\">{\"mainEntity\":{\"headline\":\"An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment\",\"description\":\"Quantitative genetics approaches, and particularly animal models, are widely used to assess the genetic (co)variance of key fitness related traits and infer adaptive potential of wild populations. Despite the importance of precision and accuracy of genetic variance estimates and their potential sensitivity to various ecological and population specific factors, their reliability is rarely tested explicitly. Here, we used simulations and empirical data collected from an 11-year study on tree swallow (Tachycineta bicolor), a species showing a high rate of extra-pair paternity and a low recruitment rate, to assess the importance of identity errors, structure and size of the pedigree on quantitative genetic estimates in our dataset. Our simulations revealed an important lack of precision in heritability and genetic-correlation estimates for most traits, a low power to detect significant effects and important identifiability problems. We also observed a large bias in heritability estimates when using the social pedigree instead of the genetic one (deflated heritabilities) or when not accounting for an important cause of resemblance among individuals (for example, permanent environment or brood effect) in model parameterizations for some traits (inflated heritabilities). We discuss the causes underlying the low reliability observed here and why they are also likely to occur in other study systems. Altogether, our results re-emphasize the difficulties of generalizing quantitative genetic estimates reliably from one study system to another and the importance of reporting simulation analyses to evaluate these important issues.\",\"datePublished\":\"2016-10-26\",\"dateModified\":\"2016-10-26\",\"pageStart\":\"229\",\"pageEnd\":\"238\",\"sameAs\":\"https://doi.org/10.1038/hdy.2016.92\",\"keywords\":\"Evolutionary genetics,Biomedicine,general,Human Genetics,Evolutionary Biology,Ecology,Cytogenetics,Plant Genetics and Genomics\",\"image\":\"https://static-content.springer.com/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig1_HTML.jpg\",\"isPartOf\":{\"name\":\"Heredity\",\"issn\":[\"1365-2540\",\"0018-067X\"],\"volumeNumber\":\"118\",\"@type\":[\"Periodical\",\"PublicationVolume\"]},\"publisher\":{\"name\":\"Springer International Publishing\",\"logo\":{\"url\":\"https://www.springernature.com/app-sn/public/images/logo-springernature.png\",\"@type\":\"ImageObject\"},\"@type\":\"Organization\"},\"author\":[{\"name\":\"Bourret, A\",\"url\":\"http://orcid.org/0000-0002-5387-3011\",\"affiliation\":[{\"name\":\"Universit\\xc3\\xa9 de Sherbrooke\",\"address\":{\"name\":\"D\\xc3\\xa9partement de Biologie, Universit\\xc3\\xa9 de Sherbrooke, Sherbrooke, Canada\",\"@type\":\"PostalAddress\"},\"@type\":\"Organization\"}],\"email\":\"Audrey.Bourret@USherbrooke.ca\",\"@type\":\"Person\"},{\"name\":\"Garant, D\",\"affiliation\":[{\"name\":\"Universit\\xc3\\xa9 de Sherbrooke\",\"address\":{\"name\":\"D\\xc3\\xa9partement de Biologie, Universit\\xc3\\xa9 de Sherbrooke, Sherbrooke, Canada\",\"@type\":\"PostalAddress\"},\"@type\":\"Organization\"}],\"@type\":\"Person\"}],\"isAccessibleForFree\":true,\"@type\":\"ScholarlyArticle\"},\"@context\":\"https://schema.org\",\"@type\":\"WebPage\"}</script>\\n\\n\\n\\n\\n\\n\\n\\n    \\n    <link rel=\"canonical\" href=\"https://www.nature.com/articles/hdy201692\">\\n    \\n    \\n    <meta name=\"journal_id\" content=\"41437\"/>\\n    <meta name=\"dc.title\" content=\"An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment\"/>\\n    <meta name=\"dc.source\" content=\"Heredity 2017 118:3\"/>\\n    <meta name=\"dc.format\" content=\"text/html\"/>\\n    <meta name=\"dc.publisher\" content=\"Nature Publishing Group\"/>\\n    <meta name=\"dc.date\" content=\"2016-10-26\"/>\\n    <meta name=\"dc.type\" content=\"OriginalPaper\"/>\\n    <meta name=\"dc.language\" content=\"En\"/>\\n    <meta name=\"dc.copyright\" content=\"2017 The Genetics Society Macmillan Publishers Limited, part of Springer Nature.\"/>\\n    <meta name=\"dc.rights\" content=\"2017 The Genetics Society Macmillan Publishers Limited, part of Springer Nature.\"/>\\n    <meta name=\"dc.rightsAgent\" content=\"journalpermissions@springernature.com\"/>\\n    <meta name=\"dc.description\" content=\"Quantitative genetics approaches, and particularly animal models, are widely used to assess the genetic (co)variance of key fitness related traits and infer adaptive potential of wild populations. Despite the importance of precision and accuracy of genetic variance estimates and their potential sensitivity to various ecological and population specific factors, their reliability is rarely tested explicitly. Here, we used simulations and empirical data collected from an 11-year study on tree swallow (Tachycineta bicolor), a species showing a high rate of extra-pair paternity and a low recruitment rate, to assess the importance of identity errors, structure and size of the pedigree on quantitative genetic estimates in our dataset. Our simulations revealed an important lack of precision in heritability and genetic-correlation estimates for most traits, a low power to detect significant effects and important identifiability problems. We also observed a large bias in heritability estimates when using the social pedigree instead of the genetic one (deflated heritabilities) or when not accounting for an important cause of resemblance among individuals (for example, permanent environment or brood effect) in model parameterizations for some traits (inflated heritabilities). We discuss the causes underlying the low reliability observed here and why they are also likely to occur in other study systems. Altogether, our results re-emphasize the difficulties of generalizing quantitative genetic estimates reliably from one study system to another and the importance of reporting simulation analyses to evaluate these important issues.\"/>\\n    <meta name=\"prism.issn\" content=\"1365-2540\"/>\\n    <meta name=\"prism.publicationName\" content=\"Heredity\"/>\\n    <meta name=\"prism.publicationDate\" content=\"2016-10-26\"/>\\n    <meta name=\"prism.volume\" content=\"118\"/>\\n    <meta name=\"prism.number\" content=\"3\"/>\\n    <meta name=\"prism.section\" content=\"OriginalPaper\"/>\\n    <meta name=\"prism.startingPage\" content=\"229\"/>\\n    <meta name=\"prism.endingPage\" content=\"238\"/>\\n    <meta name=\"prism.copyright\" content=\"2017 The Genetics Society Macmillan Publishers Limited, part of Springer Nature.\"/>\\n    <meta name=\"prism.rightsAgent\" content=\"journalpermissions@springernature.com\"/>\\n    <meta name=\"prism.url\" content=\"https://www.nature.com/articles/hdy201692\"/>\\n    <meta name=\"prism.doi\" content=\"doi:10.1038/hdy.2016.92\"/>\\n    <meta name=\"citation_pdf_url\" content=\"https://www.nature.com/articles/hdy201692.pdf\"/>\\n    <meta name=\"citation_fulltext_html_url\" content=\"https://www.nature.com/articles/hdy201692\"/>\\n    <meta name=\"citation_journal_title\" content=\"Heredity\"/>\\n    <meta name=\"citation_journal_abbrev\" content=\"Heredity\"/>\\n    <meta name=\"citation_publisher\" content=\"Nature Publishing Group\"/>\\n    <meta name=\"citation_issn\" content=\"1365-2540\"/>\\n    <meta name=\"citation_title\" content=\"An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment\"/>\\n    <meta name=\"citation_volume\" content=\"118\"/>\\n    <meta name=\"citation_issue\" content=\"3\"/>\\n    <meta name=\"citation_publication_date\" content=\"2017/03\"/>\\n    <meta name=\"citation_online_date\" content=\"2016/10/26\"/>\\n    <meta name=\"citation_firstpage\" content=\"229\"/>\\n    <meta name=\"citation_lastpage\" content=\"238\"/>\\n    <meta name=\"citation_article_type\" content=\"Article\"/>\\n    <meta name=\"citation_fulltext_world_readable\" content=\"\"/>\\n    <meta name=\"citation_language\" content=\"en\"/>\\n    <meta name=\"dc.identifier\" content=\"doi:10.1038/hdy.2016.92\"/>\\n    <meta name=\"DOI\" content=\"10.1038/hdy.2016.92\"/>\\n    <meta name=\"size\" content=\"199098\"/>\\n    <meta name=\"citation_doi\" content=\"10.1038/hdy.2016.92\"/>\\n    <meta name=\"citation_springer_api_url\" content=\"http://api.springer.com/xmldata/jats?q=doi:10.1038/hdy.2016.92&amp;api_key=\"/>\\n    <meta name=\"description\" content=\"Quantitative genetics approaches, and particularly animal models, are widely used to assess the genetic (co)variance of key fitness related traits and infer adaptive potential of wild populations. Despite the importance of precision and accuracy of genetic variance estimates and their potential sensitivity to various ecological and population specific factors, their reliability is rarely tested explicitly. Here, we used simulations and empirical data collected from an 11-year study on tree swallow (Tachycineta bicolor), a species showing a high rate of extra-pair paternity and a low recruitment rate, to assess the importance of identity errors, structure and size of the pedigree on quantitative genetic estimates in our dataset. Our simulations revealed an important lack of precision in heritability and genetic-correlation estimates for most traits, a low power to detect significant effects and important identifiability problems. We also observed a large bias in heritability estimates when using the social pedigree instead of the genetic one (deflated heritabilities) or when not accounting for an important cause of resemblance among individuals (for example, permanent environment or brood effect) in model parameterizations for some traits (inflated heritabilities). We discuss the causes underlying the low reliability observed here and why they are also likely to occur in other study systems. Altogether, our results re-emphasize the difficulties of generalizing quantitative genetic estimates reliably from one study system to another and the importance of reporting simulation analyses to evaluate these important issues.\"/>\\n    <meta name=\"dc.creator\" content=\"Bourret, A\"/>\\n    <meta name=\"dc.creator\" content=\"Garant, D\"/>\\n    <meta name=\"dc.subject\" content=\"Evolutionary genetics\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=PLoS One; citation_title=Estimating heritabilities and genetic correlations: comparing the &#8216;animal model&#8217; with parent&#8211;offspring regression using data from a natural population; citation_author=M &#197;kesson, S Bensch, D Hasselquist, M Tarka, B Hansson, M Akesson; citation_volume=3; citation_publication_date=2008; citation_pages=e1739; citation_doi=10.1371/journal.pone.0001739; citation_id=CR1\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Mol Ecol; citation_title=Estimating quantitative genetic parameters in wild populations: a comparison of pedigree and genomic approaches; citation_author=C B&#233;r&#233;nos, PA Ellis, JG Pilkington, JM Pemberton; citation_volume=23; citation_publication_date=2014; citation_pages=3434-3451; citation_doi=10.1111/mec.12827; citation_id=CR2\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Ecology; citation_title=A reassessment of genetic limits to evoutionary change; citation_author=MW Blows, AA Hoffmann; citation_volume=86; citation_publication_date=2005; citation_pages=1371-1384; citation_doi=10.1890/04-1209; citation_id=CR3\"/>\\n    <meta name=\"citation_reference\" content=\"citation_title=Ecological Models and Data in R; citation_publication_date=2008; citation_id=CR4; citation_author=B Bolker; citation_publisher=Princeton, University Princeton, NJ, USA\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Mol Ecol; citation_title=How do misassigned paternities affect the estimation of heritability in the wild?; citation_author=A Charmantier, D R&#233;ale; citation_volume=14; citation_publication_date=2005; citation_pages=2839-2850; citation_doi=10.1111/j.1365-294X.2005.02619.x; citation_id=CR5\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Trends Ecol Evol; citation_title=Behavioural reaction norms: animal personality meets individual plasticity; citation_author=NJ Dingemanse, AJN Kazem, D R&#233;ale, J Wright; citation_volume=25; citation_publication_date=2009; citation_pages=81-89; citation_doi=10.1016/j.tree.2009.07.013; citation_id=CR6\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Evolution; citation_title=Mating systems, sperm competition, and the evolution of sexual dimorphism in birds; citation_author=PO Dunn, LA Whittingham, TE Pitcher; citation_volume=55; citation_publication_date=2001; citation_pages=161-175; citation_doi=10.1111/j.0014-3820.2001.tb01281.x; citation_id=CR7\"/>\\n    <meta name=\"citation_reference\" content=\"citation_title=Introduction to Quantitative Genetics; citation_publication_date=1996; citation_id=CR8; citation_author=DS Falconer; citation_author=TFC Mackay; citation_publisher=Pearson: Essex, UK\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Evolution; citation_title=The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees; citation_author=JA Firth, JD Hadfield, AW Santure, J Slate, BC Sheldon; citation_volume=69; citation_publication_date=2015; citation_pages=1336-1344; citation_doi=10.1111/evo.12649; citation_id=CR9\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Evolution; citation_title=Comparing evolvabilities: common errors surrounding the calculation and use of coefficients of additive genetic variation; citation_author=F Garcia-Gonzalez, LW Simmons, JL Tomkins, JS Kotiaho, JP Evans; citation_volume=66; citation_publication_date=2012; citation_pages=2341-2349; citation_doi=10.1111/j.1558-5646.2011.01565.x; citation_id=CR10\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Ecol Appl; citation_title=Breeding success of tree swallows along a gradient of agricultural intensification; citation_author=A Ghilain, M B&#233;lisle; citation_volume=18; citation_publication_date=2008; citation_pages=1140-1154; citation_doi=10.1890/07-1107.1; citation_id=CR11\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Science; citation_title=Environmental determination of a sexually selected trait; citation_author=SC Griffith, IPF Owens, T Burke; citation_volume=400; citation_publication_date=1999; citation_pages=358-360; citation_id=CR12\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Mol Ecol; citation_title=Extra pair paternity in birds: a review of interspecific variation and adaptive function; citation_author=SC Griffith, IPF Owens, KA Thuman; citation_volume=11; citation_publication_date=2002; citation_pages=2195-2212; citation_doi=10.1046/j.1365-294X.2002.01613.x; citation_id=CR13\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=J Stat Softw; citation_title=MCMC methods for multi-response generalized linear mixed models: the MCMCglmm R package; citation_author=JD Hadfield; citation_volume=33; citation_publication_date=2010; citation_pages=1-22; citation_doi=10.18637/jss.v033.i02; citation_id=CR14\"/>\\n    <meta name=\"citation_reference\" content=\"citation_title=Applications of Linear Models in Animal Breeding; citation_publication_date=1984; citation_id=CR15; citation_author=CR Henderson; citation_publisher=University of Guelph: Guelph, ON, Canada\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Evol Appl; citation_title=Evolutionary principles and their practical application; citation_author=AP Hendry, MT Kinnison, M Heino, T Day, TB Smith, G Fitt; citation_volume=4; citation_publication_date=2011; citation_pages=159-183; citation_doi=10.1111/j.1752-4571.2010.00165.x; citation_id=CR16\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Trends Ecol Evol; citation_title=Heritable variation and evolution under favourable and unfavourable conditions; citation_author=AA Hoffmann, J Meril&#228;; citation_volume=14; citation_publication_date=1999; citation_pages=96-101; citation_doi=10.1016/S0169-5347(99)01595-5; citation_id=CR17\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=J Field Ornithol; citation_title=Dispersal distances of tree swallows estimated from continent-wide and limited-area data; citation_author=PA Hosner, DW Winkler; citation_volume=78; citation_publication_date=2007; citation_pages=290-297; citation_doi=10.1111/j.1557-9263.2007.00117.x; citation_id=CR18\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Genetics; citation_title=Comparing evolvability and variability of quantitative traits; citation_author=D Houle; citation_volume=130; citation_publication_date=1992; citation_pages=195-204; citation_id=CR19\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=J Evol Biol; citation_title=Estimating sampling error of evolutionary statistics based on genetic covariance matrices using maximum likelihood; citation_author=D Houle, K Meyer; citation_volume=28; citation_publication_date=2015; citation_pages=1542-1549; citation_doi=10.1111/jeb.12674; citation_id=CR20\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Proc R Soc B; citation_title=Genome-wide association mapping in a wild avian population identifies a link between genetic and phenotypic variation in a life-history trait; citation_author=A Husby, T Kawakami, L R&#246;nneg&#229;rd, L Smeds, H Ellegren, A Qvarnstr&#246;m; citation_volume=282; citation_publication_date=2015; citation_pages=20150156; citation_doi=10.1098/rspb.2015.0156; citation_id=CR21\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=J Field Ornithol; citation_title=Age and plumage color in female tree swallows; citation_author=DJT Hussell; citation_volume=54; citation_publication_date=1983; citation_pages=312-318; citation_id=CR22\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Mol Ecol; citation_title=Revising how the computer program cervus accommodates genotyping error increases success in paternity assignment; citation_author=ST Kalinowski, ML Taper, TC Marshall; citation_volume=16; citation_publication_date=2007; citation_pages=1099-1106; citation_doi=10.1111/j.1365-294X.2007.03089.x; citation_id=CR23\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Philos Trans R Soc B; citation_title=Estimating genetic parameters in natural populations using the &#8216;animal model&#8217;; citation_author=LEB Kruuk; citation_volume=359; citation_publication_date=2004; citation_pages=873-890; citation_doi=10.1098/rstb.2003.1437; citation_id=CR24\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=J Evol Biol; citation_title=How to separate genetic and environmental causes of similarity between relatives; citation_author=LEB Kruuk, JD Hadfield; citation_volume=20; citation_publication_date=2007; citation_pages=1890-1903; citation_doi=10.1111/j.1420-9101.2007.01377.x; citation_id=CR25\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Annu Rev Ecol Evol Syst; citation_title=New answers for old questions: the evolutionary quantitative genetics of wild animal populations; citation_author=LEB Kruuk, J Slate, AJ Wilson; citation_volume=39; citation_publication_date=2008; citation_pages=525-548; citation_doi=10.1146/annurev.ecolsys.39.110707.173542; citation_id=CR26\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Nat Methods; citation_title=Points of significance: error bars; citation_author=M Krzywinski, N Altman; citation_volume=10; citation_publication_date=2013; citation_pages=921-922; citation_doi=10.1038/nmeth.2659; citation_id=CR27\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Genetics; citation_title=The minimum number of genes contributing to quantitative variation between and within populations; citation_author=R Lande; citation_volume=99; citation_publication_date=1981; citation_pages=541-553; citation_id=CR28\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Evolution; citation_title=The measurement of selection on correlated characters; citation_author=R Lande, SJ Arnold; citation_volume=37; citation_publication_date=1983; citation_pages=1210-1226; citation_doi=10.1111/j.1558-5646.1983.tb00236.x; citation_id=CR29\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=J Anim Sci; citation_title=Influence of sire misidentification on sire x year interaction variance and direct-maternal genetic covariance for weaning weight in beef cattle; citation_author=C Lee, EJ Pollak; citation_volume=75; citation_publication_date=1997; citation_pages=2858-2863; citation_doi=10.2527/1997.75112858x; citation_id=CR30\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Mol Ecol Resour; citation_title=A likelihood-based approach for assessment of extra-pair paternity and conspecific brood parasitism in natural populations; citation_author=PR Lemons, TC Marshall, SE McCloskey, SA Sethi, JA Schmutz, JS Sedinger; citation_volume=7; citation_publication_date=2015; citation_pages=107-116; citation_doi=10.1111/1755-0998.12287; citation_id=CR31\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Behav Ecol Sociobiol; citation_title=Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor; citation_author=A Lessard, A Bourret, M B&#233;lisle, F Pelletier, D Garant; citation_volume=68; citation_publication_date=2014; citation_pages=733-742; citation_doi=10.1007/s00265-014-1686-y; citation_id=CR32\"/>\\n    <meta name=\"citation_reference\" content=\"citation_title=Genetics and Analysis of Quantitative Traits; citation_publication_date=1998; citation_id=CR33; citation_author=M Lynch; citation_author=B Walsh; citation_publisher=Sinauer Associates: Sunderland, MA, USA\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Heredity; citation_title=Likelihood calculations to evaluate experimental designs to estimate genetic variances; citation_author=K Meyer; citation_volume=101; citation_publication_date=2008; citation_pages=212-221; citation_doi=10.1038/hdy.2008.46; citation_id=CR34\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=J Evol Biol; citation_title=The danger of applying the breeder&#8217;s equation in observational studies of natural populations; citation_author=MB Morrissey, LEB Kruuk, AJ Wilson; citation_volume=23; citation_publication_date=2010; citation_pages=2277-2288; citation_doi=10.1111/j.1420-9101.2010.02084.x; citation_id=CR35\"/>\\n    <meta name=\"citation_reference\" content=\"citation_title=Bayesian approaches to the quantitative genetic analysis of natural populations; citation_inbook_title=Quantitative Genetics in the Wild; citation_publication_date=2014; citation_pages=228-253; citation_id=CR36; citation_author=Michael B. Morrissey; citation_author=Pierre de Villemereuil; citation_author=Blandine Doligez; citation_author=Olivier Gimenez\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Mol Ecol Resour; citation_title=PEDANTICS: an R package for pedigree-based genetic simulation and pedigree manipulation, characterization and viewing; citation_author=MB Morrissey, AJ Wilson; citation_volume=10; citation_publication_date=2010; citation_pages=711-719; citation_doi=10.1111/j.1755-0998.2009.02817.x; citation_id=CR37\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=J Evol Biol; citation_title=A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries; citation_author=MB Morrissey, AJ Wilson, JM Pemberton, MM Ferguson; citation_volume=20; citation_publication_date=2007; citation_pages=2309-2321; citation_doi=10.1111/j.1420-9101.2007.01412.x; citation_id=CR38\"/>\\n    <meta name=\"citation_reference\" content=\"citation_title=Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates?; citation_inbook_title=Quantitative Genetics in the Wild; citation_publication_date=2014; citation_pages=16-33; citation_id=CR39; citation_author=Erik Postma\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=J Evol Biol; citation_title=Data depth, data completeness, and their influence on quantitative genetic estimation in two contrasting bird populations; citation_author=JL Quinn, A Charmantier, D Garant, BC Sheldon; citation_volume=19; citation_publication_date=2006; citation_pages=994-1002; citation_doi=10.1111/j.1420-9101.2006.01081.x; citation_id=CR40\"/>\\n    <meta name=\"citation_reference\" content=\"Core Team R . (2015). R: a language and environment for statistical computing \\n                  http://www.R-project.org/\\n                  \\n                .\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Evolution; citation_title=Genotype-environment interaction and the evolution of phenotypic plasticity; citation_author=S Via, R Lande; citation_volume=39; citation_publication_date=1985; citation_pages=505-522; citation_doi=10.1111/j.1558-5646.1985.tb00391.x; citation_id=CR42\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Methods Ecol Evol; citation_title=Comparing parent&#8211;offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits; citation_author=P de Villemereuil, O Gimenez, B Doligez; citation_volume=4; citation_publication_date=2013; citation_pages=260-275; citation_doi=10.1111/2041-210X.12011; citation_id=CR43\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Annu Rev Genet; citation_title=Estimation and partition of heritability in human populations using whole-genome analysis methods; citation_author=AAE Vinkhuyzen, NR Wray, J Yang, ME Goddard, PM Visscher; citation_volume=47; citation_publication_date=2013; citation_pages=75-95; citation_doi=10.1146/annurev-genet-111212-133258; citation_id=CR44\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Ecography; citation_title=The concepts of bias, precision and accuracy, and their use in testing the performance of species richness estimators, with a literature review of estimator performance; citation_author=BA Walther, JL Moore; citation_volume=28; citation_publication_date=2005; citation_pages=815-829; citation_doi=10.1111/j.2005.0906-7590.04112.x; citation_id=CR45\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=J Evol Biol; citation_title=Why h2 does not always equal VA/VP?; citation_author=AJ Wilson; citation_volume=21; citation_publication_date=2008; citation_pages=647-650; citation_doi=10.1111/j.1420-9101.2008.01500.x; citation_id=CR46\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=J Evol Biol; citation_title=Indirect genetics effects and evolutionary constraint: an analysis of social dominance in red deer, Cervus elaphus; citation_author=AJ Wilson, MB Morrissey, MJ Adams, CA Walling, FE Guinness, JM Pemberton; citation_volume=24; citation_publication_date=2011; citation_pages=772-783; citation_doi=10.1111/j.1420-9101.2010.02212.x; citation_id=CR47\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=J Anim Ecol; citation_title=An ecologist&#8217;s guide to the animal model; citation_author=AJ Wilson, D R&#233;ale, MN Clements, MM Morrissey, E Postma, CA Walling; citation_volume=79; citation_publication_date=2010; citation_pages=13-26; citation_doi=10.1111/j.1365-2656.2009.01639.x; citation_id=CR48\"/>\\n    <meta name=\"citation_reference\" content=\"Winkler D, Hallinger K, Ardia D, Robertson R, Stutchbury B, Cohen R . (2011). Tree swallow (Tachycineta bicolor. Birds North Am Online. \\n                  http://bna.birds.cornelledu/bna/\\n                  \\n                . Accessed on 1 Oct 2013.\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Methods Ecol Evol; citation_title=nadiv : an R package to create relatedness matrices for estimating non-additive genetic variances in animal models; citation_author=ME Wolak; citation_volume=3; citation_publication_date=2012; citation_pages=792-796; citation_doi=10.1111/j.2041-210X.2012.00213.x; citation_id=CR50\"/>\\n    <meta name=\"citation_reference\" content=\"citation_journal_title=Ecol Evol; citation_title=Are we underestimating the genetic variances of dimorphic traits?; citation_author=ME Wolak, DA Roff, DJ Fairbairn; citation_volume=5; citation_publication_date=2015; citation_pages=590-597; citation_doi=10.1002/ece3.1361; citation_id=CR51\"/>\\n    <meta name=\"citation_author\" content=\"Bourret, A\"/>\\n    <meta name=\"citation_author_institution\" content=\"D&#233;partement de Biologie, Universit&#233; de Sherbrooke, Sherbrooke, Canada\"/>\\n    <meta name=\"citation_author\" content=\"Garant, D\"/>\\n    <meta name=\"citation_author_institution\" content=\"D&#233;partement de Biologie, Universit&#233; de Sherbrooke, Sherbrooke, Canada\"/>\\n    <meta name=\"access_endpoint\" content=\"https://www.nature.com/platform/readcube-access\"/>\\n    <meta name=\"twitter:site\" content=\"@HeredityJournal\"/>\\n    <meta name=\"twitter:card\" content=\"summary_large_image\"/>\\n    <meta name=\"twitter:image:alt\" content=\"Content cover image\"/>\\n    <meta name=\"twitter:title\" content=\"An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment\"/>\\n    <meta name=\"twitter:description\" content=\"Heredity - An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment\"/>\\n    <meta name=\"twitter:image\" content=\"https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig1_HTML.jpg\"/>\\n    \\n\\n    \\n    \\n    <meta property=\"og:url\" content=\"https://www.nature.com/articles/hdy201692\"/>\\n    <meta property=\"og:type\" content=\"article\"/>\\n    <meta property=\"og:site_name\" content=\"Nature\"/>\\n    <meta property=\"og:title\" content=\"An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment - Heredity\"/>\\n    <meta property=\"og:image\" content=\"https://media.springernature.com/m685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig1_HTML.jpg\"/>\\n    \\n\\n    <script>\\n        window.eligibleForRa21 = \\'false\\'; \\n    </script>\\n</head>\\n<body class=\"article-page\">\\n\\n<noscript><iframe src=\"https://collect.nature.com/ns.html?id=GTM-MRVXSHQ\"\\n                  height=\"0\" width=\"0\" style=\"display:none;visibility:hidden\"></iframe></noscript>\\n\\n\\n\\n<div class=\"position-relative cleared z-index-50 background-white\" data-test=\"top-containers\">\\n    <a class=\"c-skip-link\" href=\"#content\">Skip to main content</a>\\n\\n\\n\\n<div class=\"c-grade-c-banner u-hide\">\\n    <div class=\"c-grade-c-banner__container\">\\n        \\n        <p>Thank you for visiting nature.com. You are using a browser version with limited support for CSS. To obtain\\n            the best experience, we recommend you use a more up to date browser (or turn off compatibility mode in\\n            Internet Explorer). In the meantime, to ensure continued support, we are displaying the site without styles\\n            and JavaScript.</p>\\n\\n    </div>\\n</div>\\n\\n    \\n\\n    <div class=\"u-lazy-ad-wrapper u-mbs-0\">\\n            <div class=\"deferred-placeholder\" data-replace=\"true\"\\n                 data-placeholder=\"/placeholder/v1/institutionalBanner?bpids=[bpids] #institutional-banner-container\"></div>\\n            <aside class=\"c-ad c-ad--728x90\">\\n                <div class=\"c-ad__inner\" data-container-type=\"banner-advert\">\\n                    <p class=\"c-ad__label\">Advertisement</p>\\n                    \\n        \\n            \\n    <div id=\"div-gpt-ad-top-1\"\\n         class=\"div-gpt-ad advert leaderboard js-ad text-center hide-print grade-c-hide\"\\n         data-ad-type=\"top\"\\n         data-test=\"top-ad\"\\n         data-pa11y-ignore\\n         data-gpt\\n         data-gpt-unitpath=\"/285/hdy.nature.com/article\"\\n         data-gpt-sizes=\"728x90\"\\n         data-gpt-targeting=\"type=article;pos=top;artid=hdy201692;doi=10.1038/hdy.2016.92;techmeta=23,29,45,77;subjmeta=181,2474,631;kwrd=Evolutionary+genetics\">\\n        <noscript>\\n            <a href=\"//pubads.g.doubleclick.net/gampad/jump?iu=/285/hdy.nature.com/article&amp;sz=728x90&amp;c=-1826710347&amp;t=pos%3Dtop%26type%3Darticle%26artid%3Dhdy201692%26doi%3D10.1038/hdy.2016.92%26techmeta%3D23,29,45,77%26subjmeta%3D181,2474,631%26kwrd%3DEvolutionary+genetics\">\\n                <img data-test=\"gpt-advert-fallback-img\"\\n                     src=\"//pubads.g.doubleclick.net/gampad/ad?iu=/285/hdy.nature.com/article&amp;sz=728x90&amp;c=-1826710347&amp;t=pos%3Dtop%26type%3Darticle%26artid%3Dhdy201692%26doi%3D10.1038/hdy.2016.92%26techmeta%3D23,29,45,77%26subjmeta%3D181,2474,631%26kwrd%3DEvolutionary+genetics\"\\n                     alt=\"Advertisement\"\\n                     width=\"728\"\\n                     height=\"90\"></a>\\n        </noscript>\\n    </div>\\n\\n        \\n    \\n                </div>\\n            </aside>\\n        </div>\\n    <header class=\"c-header\" id=\"header\" data-header data-track-component=\"nature-150-split-header\" style=\"border-color:#252258\">\\n        <div class=\"c-header__row\">\\n            <div class=\"c-header__container\">\\n                <div class=\"c-header__split\">\\n                    \\n                    \\n                    <div class=\"c-header__logo-container\">\\n                        \\n                        <a href=\"/hdy\"\\n                           data-track=\"click\" data-track-action=\"home\" data-track-label=\"image\">\\n                            <picture class=\"c-header__logo\">\\n                                <source srcset=\"//media.springernature.com/full/nature-cms/uploads/product/hdy/header-3c785d60c600344908a3062a4389be3b.svg\" media=\"(min-width: 875px)\">\\n                                <img src=\"//media.springernature.com/full/nature-cms/uploads/product/hdy/header-3c785d60c600344908a3062a4389be3b.svg\" height=\"32\" alt=\"Heredity\">\\n                            </picture>\\n                        </a>\\n                    \\n                    </div>\\n                    \\n                    <ul class=\"c-header__menu c-header__menu--global\">\\n                        <li class=\"c-header__item c-header__item--padding c-header__item--hide-md-max\">\\n                            <a class=\"c-header__link\" href=\"https://www.nature.com/siteindex\" data-test=\"siteindex-link\"\\n                               data-track=\"click\" data-track-action=\"open nature research index\" data-track-label=\"link\">\\n                                <span>View all journals</span>\\n                            </a>\\n                        </li>\\n                        <li class=\"c-header__item c-header__item--padding c-header__item--pipe\">\\n                            <a class=\"c-header__link\"\\n                               href=\"#search-menu\"\\n                               data-header-expander\\n                               data-test=\"search-link\" data-track=\"click\" data-track-action=\"open search tray\" data-track-label=\"button\">\\n                                <span>Search</span><svg role=\"img\" aria-hidden=\"true\" focusable=\"false\" height=\"22\" width=\"22\" viewBox=\"0 0 18 18\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"M16.48 15.455c.283.282.29.749.007 1.032a.738.738 0 01-1.032-.007l-3.045-3.044a7 7 0 111.026-1.026zM8 14A6 6 0 108 2a6 6 0 000 12z\"/></svg>\\n                            </a>\\n                        </li>\\n                        <li class=\"c-header__item c-header__item--padding\">\\n                            \\n                                <a href=\"/nams/svc/myaccount\"\\n    id=\"my-account\"\\n    class=\"c-header__link placeholder\"\\n    data-test=\"login-link\" data-track=\"click\" data-track-action=\"my account\" data-track-category=\"nature-150-split-header\" data-track-label=\"link\">\\n    <span>My Account</span><svg role=\"img\" aria-hidden=\"true\" focusable=\"false\" height=\"22\" width=\"22\" viewBox=\"0 0 18 18\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"M10.238 16.905a7.96 7.96 0 003.53-1.48c-.874-2.514-2.065-3.936-3.768-4.319V9.83a3.001 3.001 0 10-2 0v1.277c-1.703.383-2.894 1.805-3.767 4.319A7.96 7.96 0 009 17c.419 0 .832-.032 1.238-.095zm4.342-2.172a8 8 0 10-11.16 0c.757-2.017 1.84-3.608 3.49-4.322a4 4 0 114.182 0c1.649.714 2.731 2.305 3.488 4.322zM9 18A9 9 0 119 0a9 9 0 010 18z\" fill=\"#333\" fill-rule=\"evenodd\"/></svg>\\n</a>\\n<a href=\"https://idp.nature.com/authorize/natureuser?client_id&#x3D;grover&amp;redirect_uri&#x3D;https%3A%2F%2Fwww.nature.com%2Farticles%2Fhdy201692\"\\n    id=\"login-button\"\\n    style=\"display: none;\"\\n    class=\"c-header__link placeholder\"\\n    data-test=\"login-link\" data-track=\"click\" data-track-action=\"login\" data-track-category=\"nature-150-split-header\" data-track-label=\"link\">\\n    <span>Login</span><svg role=\"img\" aria-hidden=\"true\" focusable=\"false\" height=\"22\" width=\"22\" viewBox=\"0 0 18 18\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"M10.238 16.905a7.96 7.96 0 003.53-1.48c-.874-2.514-2.065-3.936-3.768-4.319V9.83a3.001 3.001 0 10-2 0v1.277c-1.703.383-2.894 1.805-3.767 4.319A7.96 7.96 0 009 17c.419 0 .832-.032 1.238-.095zm4.342-2.172a8 8 0 10-11.16 0c.757-2.017 1.84-3.608 3.49-4.322a4 4 0 114.182 0c1.649.714 2.731 2.305 3.488 4.322zM9 18A9 9 0 119 0a9 9 0 010 18z\" fill=\"#333\" fill-rule=\"evenodd\"/></svg>\\n</a>\\n\\n                            \\n                        </li>\\n                    </ul>\\n                </div>\\n            </div>\\n        </div>\\n        \\n            <div class=\"c-header__row\">\\n                <div class=\"c-header__container\" data-test=\"navigation-row\">\\n                    <div class=\"c-header__split\">\\n                        <ul class=\"c-header__menu c-header__menu--journal\">\\n                            \\n                                <li class=\"c-header__item c-header__item--dropdown-menu\" data-test=\"explore-content-button\">\\n                                    <a href=\"#explore\"\\n                                       class=\"c-header__link\"\\n                                       data-header-expander\\n                                       data-test=\"menu-button--explore\"\\n                                       data-track=\"click\" data-track-action=\"open explore expander\" data-track-label=\"button\">\\n                                        <span><span class=\"c-header__show-text\">Explore</span> content</span><svg role=\"img\" aria-hidden=\"true\" focusable=\"false\" height=\"16\" viewBox=\"0 0 16 16\" width=\"16\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"m5.58578644 3-3.29289322-3.29289322c-.39052429-.39052429-.39052429-1.02368927 0-1.41421356s1.02368927-.39052429 1.41421356 0l4 4c.39052429.39052429.39052429 1.02368927 0 1.41421356l-4 4c-.39052429.39052429-1.02368927.39052429-1.41421356 0s-.39052429-1.02368927 0-1.41421356z\" transform=\"matrix(0 1 -1 0 11 3)\"/></svg>\\n                                    </a>\\n                                </li>\\n                            \\n                            \\n                                <li class=\"c-header__item c-header__item--dropdown-menu\">\\n                                    <a href=\"#about-the-journal\"\\n                                       class=\"c-header__link\"\\n                                       data-header-expander\\n                                       data-test=\"menu-button--about-the-journal\"\\n                                       data-track=\"click\" data-track-action=\"open about the journal expander\" data-track-label=\"button\">\\n                                        <span>About <span class=\"c-header__show-text\">the journal</span></span><svg role=\"img\" aria-hidden=\"true\" focusable=\"false\" height=\"16\" viewBox=\"0 0 16 16\" width=\"16\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"m5.58578644 3-3.29289322-3.29289322c-.39052429-.39052429-.39052429-1.02368927 0-1.41421356s1.02368927-.39052429 1.41421356 0l4 4c.39052429.39052429.39052429 1.02368927 0 1.41421356l-4 4c-.39052429.39052429-1.02368927.39052429-1.41421356 0s-.39052429-1.02368927 0-1.41421356z\" transform=\"matrix(0 1 -1 0 11 3)\"/></svg>\\n                                    </a>\\n                                </li>\\n                                \\n                                    <li class=\"c-header__item c-header__item--dropdown-menu\" data-test=\"publish-with-us-button\">\\n                                        <a href=\"#publish-with-us\"\\n                                           class=\"c-header__link c-header__link--dropdown-menu\"\\n                                           data-header-expander\\n                                           data-test=\"menu-button--publish\"\\n                                           data-track=\"click\" data-track-action=\"open publish with us expander\" data-track-label=\"button\">\\n                                            <span>Publish <span class=\"c-header__show-text\">with us</span></span><svg role=\"img\" aria-hidden=\"true\" focusable=\"false\" height=\"16\" viewBox=\"0 0 16 16\" width=\"16\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"m5.58578644 3-3.29289322-3.29289322c-.39052429-.39052429-.39052429-1.02368927 0-1.41421356s1.02368927-.39052429 1.41421356 0l4 4c.39052429.39052429.39052429 1.02368927 0 1.41421356l-4 4c-.39052429.39052429-1.02368927.39052429-1.41421356 0s-.39052429-1.02368927 0-1.41421356z\" transform=\"matrix(0 1 -1 0 11 3)\"/></svg>\\n                                        </a>\\n                                    </li>\\n                                \\n                            \\n                            \\n                        </ul>\\n                        <ul class=\"c-header__menu c-header__menu--hide-lg-max\">\\n                            \\n                                <li class=\"c-header__item\">\\n                                    <a class=\"c-header__link\"\\n                                       href=\"https://www.nature.com/my-account/alerts/subscribe-journal?list-id&#x3D;40\"\\n                                       rel=\"nofollow\"\\n                                       data-track=\"click\"\\n                                       data-track-action=\"Sign up for alerts\"\\n                                       data-track-label=\"link (desktop site header)\"\\n                                       data-track-external>\\n                                        <span>Sign up for alerts</span><svg role=\"img\" aria-hidden=\"true\" focusable=\"false\" height=\"18\" viewBox=\"0 0 18 18\" width=\"18\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"m4 10h2.5c.27614237 0 .5.2238576.5.5s-.22385763.5-.5.5h-3.08578644l-1.12132034 1.1213203c-.18753638.1875364-.29289322.4418903-.29289322.7071068v.1715729h14v-.1715729c0-.2652165-.1053568-.5195704-.2928932-.7071068l-1.7071068-1.7071067v-3.4142136c0-2.76142375-2.2385763-5-5-5-2.76142375 0-5 2.23857625-5 5zm3 4c0 1.1045695.8954305 2 2 2s2-.8954305 2-2zm-5 0c-.55228475 0-1-.4477153-1-1v-.1715729c0-.530433.21071368-1.0391408.58578644-1.4142135l1.41421356-1.4142136v-3c0-3.3137085 2.6862915-6 6-6s6 2.6862915 6 6v3l1.4142136 1.4142136c.3750727.3750727.5857864.8837805.5857864 1.4142135v.1715729c0 .5522847-.4477153 1-1 1h-4c0 1.6568542-1.3431458 3-3 3-1.65685425 0-3-1.3431458-3-3z\" fill=\"#222\"/></svg>\\n                                    </a>\\n                                </li>\\n                            \\n                            \\n                                <li class=\"c-header__item c-header__item--pipe\">\\n                                    <a class=\"c-header__link\"\\n                                       href=\"https://www.nature.com/hdy.rss\"\\n                                       data-track=\"click\"\\n                                       data-track-action=\"rss feed\"\\n                                       data-track-label=\"link\">\\n                                            <span>RSS feed</span>\\n                                    </a>\\n                                </li>\\n                            \\n                        </ul>\\n                    </div>\\n                </div>\\n            </div>\\n        \\n    </header>\\n\\n\\n    \\n    \\n        <nav class=\"u-mb-16\" aria-label=\"breadcrumbs\">\\n            <div class=\"u-container\">\\n                <ol class=\"c-breadcrumbs\" itemscope itemtype=\"https://schema.org/BreadcrumbList\">\\n                    <li class=\"c-breadcrumbs__item\" id=\"breadcrumb0\" itemprop=\"itemListElement\" itemscope itemtype=\"https://schema.org/ListItem\"><a class=\"c-breadcrumbs__link\"\\n                               href=\"/\" itemprop=\"item\"\\n                               data-track=\"click\" data-track-action=\"breadcrumb\" data-track-category=\"header\" data-track-label=\"link:nature\"><span itemprop=\"name\">nature</span></a><meta itemprop=\"position\" content=\"1\">\\n                                    <svg class=\"c-breadcrumbs__chevron\" role=\"img\" aria-hidden=\"true\" focusable=\"false\" height=\"10\" viewBox=\"0 0 10 10\" width=\"10\"\\n                                         xmlns=\"http://www.w3.org/2000/svg\">\\n                                        <path d=\"m5.96738168 4.70639573 2.39518594-2.41447274c.37913917-.38219212.98637524-.38972225 1.35419292-.01894278.37750606.38054586.37784436.99719163-.00013556 1.37821513l-4.03074001 4.06319683c-.37758093.38062133-.98937525.38100976-1.367372-.00003075l-4.03091981-4.06337806c-.37759778-.38063832-.38381821-.99150444-.01600053-1.3622839.37750607-.38054587.98772445-.38240057 1.37006824.00302197l2.39538588 2.4146743.96295325.98624457z\"\\n                                              fill=\"#666\" fill-rule=\"evenodd\" transform=\"matrix(0 -1 1 0 0 10)\"/>\\n                                    </svg>\\n                                </li><li class=\"c-breadcrumbs__item\" id=\"breadcrumb1\" itemprop=\"itemListElement\" itemscope itemtype=\"https://schema.org/ListItem\"><a class=\"c-breadcrumbs__link\"\\n                               href=\"/hdy\" itemprop=\"item\"\\n                               data-track=\"click\" data-track-action=\"breadcrumb\" data-track-category=\"header\" data-track-label=\"link:heredity\"><span itemprop=\"name\">heredity</span></a><meta itemprop=\"position\" content=\"2\">\\n                                    <svg class=\"c-breadcrumbs__chevron\" role=\"img\" aria-hidden=\"true\" focusable=\"false\" height=\"10\" viewBox=\"0 0 10 10\" width=\"10\"\\n                                         xmlns=\"http://www.w3.org/2000/svg\">\\n                                        <path d=\"m5.96738168 4.70639573 2.39518594-2.41447274c.37913917-.38219212.98637524-.38972225 1.35419292-.01894278.37750606.38054586.37784436.99719163-.00013556 1.37821513l-4.03074001 4.06319683c-.37758093.38062133-.98937525.38100976-1.367372-.00003075l-4.03091981-4.06337806c-.37759778-.38063832-.38381821-.99150444-.01600053-1.3622839.37750607-.38054587.98772445-.38240057 1.37006824.00302197l2.39538588 2.4146743.96295325.98624457z\"\\n                                              fill=\"#666\" fill-rule=\"evenodd\" transform=\"matrix(0 -1 1 0 0 10)\"/>\\n                                    </svg>\\n                                </li><li class=\"c-breadcrumbs__item\" id=\"breadcrumb2\" itemprop=\"itemListElement\" itemscope itemtype=\"https://schema.org/ListItem\"><a class=\"c-breadcrumbs__link\"\\n                               href=\"/hdy/articles?type&#x3D;original-article\" itemprop=\"item\"\\n                               data-track=\"click\" data-track-action=\"breadcrumb\" data-track-category=\"header\" data-track-label=\"link:original article\"><span itemprop=\"name\">original article</span></a><meta itemprop=\"position\" content=\"3\">\\n                                    <svg class=\"c-breadcrumbs__chevron\" role=\"img\" aria-hidden=\"true\" focusable=\"false\" height=\"10\" viewBox=\"0 0 10 10\" width=\"10\"\\n                                         xmlns=\"http://www.w3.org/2000/svg\">\\n                                        <path d=\"m5.96738168 4.70639573 2.39518594-2.41447274c.37913917-.38219212.98637524-.38972225 1.35419292-.01894278.37750606.38054586.37784436.99719163-.00013556 1.37821513l-4.03074001 4.06319683c-.37758093.38062133-.98937525.38100976-1.367372-.00003075l-4.03091981-4.06337806c-.37759778-.38063832-.38381821-.99150444-.01600053-1.3622839.37750607-.38054587.98772445-.38240057 1.37006824.00302197l2.39538588 2.4146743.96295325.98624457z\"\\n                                              fill=\"#666\" fill-rule=\"evenodd\" transform=\"matrix(0 -1 1 0 0 10)\"/>\\n                                    </svg>\\n                                </li><li class=\"c-breadcrumbs__item\" id=\"breadcrumb3\" itemprop=\"itemListElement\" itemscope itemtype=\"https://schema.org/ListItem\">\\n                                    <span itemprop=\"name\">article</span><meta itemprop=\"position\" content=\"4\"></li>\\n                </ol>\\n            </div>\\n        </nav>\\n    \\n\\n\\n\\n    \\n\\n</div>\\n\\n\\n<div class=\"u-container u-mt-32 u-mb-32 u-clearfix\" id=\"content\" data-component=\"article-container\"  data-container-type=\"article\">\\n    <main class=\"c-article-main-column u-float-left js-main-column\" data-track-component=\"article body\">\\n        \\n            <div class=\"c-context-bar u-hide\"\\n                 data-test=\"context-bar\"\\n                 data-context-bar\\n                 aria-hidden=\"true\">\\n                <div class=\"c-context-bar__container u-container\">\\n                    <div class=\"c-context-bar__title\">\\n                        An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment\\n                    </div>\\n                    \\n    \\n        <div class=\"c-pdf-download u-clear-both js-pdf-download\">\\n            <a href=\"/articles/hdy201692.pdf\" class=\"u-button u-button--full-width u-button--primary u-justify-content-space-between c-pdf-download__link\" data-article-pdf=\"true\" data-readcube-pdf-url=\"true\" data-test=\"download-pdf\" data-draft-ignore=\"true\" data-track=\"click\" data-track-action=\"download pdf\" data-track-label=\"link\" data-track-external download>\\n                <span class=\"c-pdf-download__text\">Download PDF</span>\\n                <svg aria-hidden=\"true\" focusable=\"false\" width=\"16\" height=\"16\" class=\"u-icon\"><use xlink:href=\"#global-icon-download\"/></svg>\\n            </a>\\n        </div>\\n    \\n\\n                </div>\\n                \\n    <div class=\"c-recommendations__container u-container u-display-none\" data-component-recommendations>\\n        <aside class=\"c-status-message c-status-message--success u-display-none\" data-component-status-msg>\\n            <svg class=\"c-status-message__icon\" width=\"24\" height=\"24\" role=\"img\" aria-label=\"success:\" focusable=\"false\">\\n                <use xlink:href=\"#icon-success\"></use>\\n            </svg>\\n            <div class=\"c-status-message__message\" tabindex=\"-1\" id=\"success-message\">\\n                Your article has downloaded\\n            </div>\\n        </aside>\\n\\n        <div class=\"c-recommendations-header u-display-flex u-justify-content-space-between\">\\n            <h2 class=\"c-recommendations-title\" id=\"recommendation-heading\">Similar articles being viewed by others</h2>\\n            <button class=\"c-recommendations-close u-flex-static\" type=\"button\" aria-label=\"Close\" data-track=\"click\" data-track-action=\"close recommendations\">\\n                <svg class=\"u-icon\" width=\"16\" height=\"16\" aria-hidden=\"true\" focusable=\"false\"><use xlink:href=\"#icon-close\"></use></svg>\\n            </button>\\n        </div>\\n\\n        <section aria-roledescription=\"carousel\" aria-labelledby=\"recommendation-heading\">\\n            <p class=\"u-visually-hidden\">Carousel with three slides shown at a time. Use the Previous and Next buttons to navigate three slides at a time, or the slide dot buttons at the end to jump three slides at a time.</p>\\n            <div class=\"c-recommendations-list-container\">\\n                <div class=\"c-recommendations-list\">\\n                    \\n                    <div class=\"c-recommendations-list__item\" role=\"group\" aria-roledescription=\"slide\" aria-label=\"Recommendation 1 of 9\">\\n                        <article class=\"u-full-height c-card c-card--flush\">\\n                            <div class=\"c-card__layout u-full-height \">\\n                                \\n                                    <div class=\"c-card__image\"><img src=\"//media.springernature.com/w136h75/springer-static/image/art%3A10.1038%2Fs41598-021-02468-9/MediaObjects/41598_2021_2468_Fig1_HTML.png\" alt=\"\"></div>\\n                                \\n                                <div class=\"c-card__body u-display-flex u-flex-direction-column\">\\n                                    <div class=\"c-recommendations-column-switch\">\\n                                        <h3 class=\"c-card__title-recommendation u-h4\"><a class=\"c-card__link u-link-inherit\" href=\"https://doi.org/10.1038/s41598-021-02468-9\" data-track=\"click\" data-track-action=\"click recommendations\" data-track-label=\"10.1038/s41598-021-02468-9\">Quantifying the effect of genetic, environmental and individual demographic stochastic variability for population dynamics in Plantago lanceolata</a></h3>\\n                                        <p class=\"u-sans-serif u-mb-0\">30 November 2021</p>\\n                                    </div>\\n                                    <p class=\"c-recommendations-authors u-hide-at-sm u-sans-serif\">Ulrich K. Steiner, Shripad Tuljapurkar &amp; Deborah A. Roach</p>\\n                                </div>\\n                            </div>\\n                        </article>\\n                    </div>\\n                    \\n                    <div class=\"c-recommendations-list__item\" role=\"group\" aria-roledescription=\"slide\" aria-label=\"Recommendation 2 of 9\">\\n                        <article class=\"u-full-height c-card c-card--flush\">\\n                            <div class=\"c-card__layout u-full-height \">\\n                                \\n                                    <div class=\"c-card__image\"><img src=\"//media.springernature.com/w136h75/springer-static/image/art%3A10.1038%2Fs41437-017-0045-y/MediaObjects/41437_2017_45_Fig1_HTML.jpg\" alt=\"\"></div>\\n                                \\n                                <div class=\"c-card__body u-display-flex u-flex-direction-column\">\\n                                    <div class=\"c-recommendations-column-switch\">\\n                                        <h3 class=\"c-card__title-recommendation u-h4\"><a class=\"c-card__link u-link-inherit\" href=\"https://doi.org/10.1038/s41437-017-0045-y\" data-track=\"click\" data-track-action=\"click recommendations\" data-track-label=\"10.1038/s41437-017-0045-y\">Detection of genetic purging and predictive value of purging parameters estimated in pedigreed populations</a></h3>\\n                                        <p class=\"u-sans-serif u-mb-0\">13 February 2018</p>\\n                                    </div>\\n                                    <p class=\"c-recommendations-authors u-hide-at-sm u-sans-serif\">Eugenio L\\xc3\\xb3pez-Cortegano, Diego Bersab\\xc3\\xa9, \\xe2\\x80\\xa6 Aurora Garc\\xc3\\xada-Dorado</p>\\n                                </div>\\n                            </div>\\n                        </article>\\n                    </div>\\n                    \\n                    <div class=\"c-recommendations-list__item\" role=\"group\" aria-roledescription=\"slide\" aria-label=\"Recommendation 3 of 9\">\\n                        <article class=\"u-full-height c-card c-card--flush\">\\n                            <div class=\"c-card__layout u-full-height \">\\n                                \\n                                    <div class=\"c-card__image\"><img src=\"//media.springernature.com/w136h75/springer-static/image/art%3A10.1038%2Fs41598-019-42562-7/MediaObjects/41598_2019_42562_Fig1_HTML.png\" alt=\"\"></div>\\n                                \\n                                <div class=\"c-card__body u-display-flex u-flex-direction-column\">\\n                                    <div class=\"c-recommendations-column-switch\">\\n                                        <h3 class=\"c-card__title-recommendation u-h4\"><a class=\"c-card__link u-link-inherit\" href=\"https://doi.org/10.1038/s41598-019-42562-7\" data-track=\"click\" data-track-action=\"click recommendations\" data-track-label=\"10.1038/s41598-019-42562-7\">Extending r/K selection with a maternal risk-management model that classifies animal species into divergent natural selection categories</a></h3>\\n                                        <p class=\"u-sans-serif u-mb-0\">16 April 2019</p>\\n                                    </div>\\n                                    <p class=\"c-recommendations-authors u-hide-at-sm u-sans-serif\">Deby L. Cassill</p>\\n                                </div>\\n                            </div>\\n                        </article>\\n                    </div>\\n                    \\n                    <div class=\"c-recommendations-list__item\" role=\"group\" aria-roledescription=\"slide\" aria-label=\"Recommendation 4 of 9\">\\n                        <article class=\"u-full-height c-card c-card--flush\">\\n                            <div class=\"c-card__layout u-full-height \">\\n                                \\n                                    <div class=\"c-card__image\"><img src=\"//media.springernature.com/w136h75/springer-static/image/art%3A10.1038%2Fs41437-022-00558-6/MediaObjects/41437_2022_558_Fig1_HTML.png\" alt=\"\"></div>\\n                                \\n                                <div class=\"c-card__body u-display-flex u-flex-direction-column\">\\n                                    <div class=\"c-recommendations-column-switch\">\\n                                        <h3 class=\"c-card__title-recommendation u-h4\"><a class=\"c-card__link u-link-inherit\" href=\"https://doi.org/10.1038/s41437-022-00558-6\" data-track=\"click\" data-track-action=\"click recommendations\" data-track-label=\"10.1038/s41437-022-00558-6\">Genetic, maternal, and environmental influences on sociality in a pedigreed primate population</a></h3>\\n                                        <p class=\"u-sans-serif u-mb-0\">02 September 2022</p>\\n                                    </div>\\n                                    <p class=\"c-recommendations-authors u-hide-at-sm u-sans-serif\">Irene Godoy, Peter Korsten &amp; Susan E. Perry</p>\\n                                </div>\\n                            </div>\\n                        </article>\\n                    </div>\\n                    \\n                    <div class=\"c-recommendations-list__item\" role=\"group\" aria-roledescription=\"slide\" aria-label=\"Recommendation 5 of 9\">\\n                        <article class=\"u-full-height c-card c-card--flush\">\\n                            <div class=\"c-card__layout u-full-height \">\\n                                \\n                                    <div class=\"c-card__image\"><img src=\"//media.springernature.com/w136h75/springer-static/image/art%3A10.1038%2Fs41467-021-23804-7/MediaObjects/41467_2021_23804_Fig1_HTML.png\" alt=\"\"></div>\\n                                \\n                                <div class=\"c-card__body u-display-flex u-flex-direction-column\">\\n                                    <div class=\"c-recommendations-column-switch\">\\n                                        <h3 class=\"c-card__title-recommendation u-h4\"><a class=\"c-card__link u-link-inherit\" href=\"https://doi.org/10.1038/s41467-021-23804-7\" data-track=\"click\" data-track-action=\"click recommendations\" data-track-label=\"10.1038/s41467-021-23804-7\">Natural selection increases female fitness by reversing the exaggeration of a male sexually selected trait</a></h3>\\n                                        <p class=\"u-sans-serif u-mb-0\">08 June 2021</p>\\n                                    </div>\\n                                    <p class=\"c-recommendations-authors u-hide-at-sm u-sans-serif\">Kensuke Okada, Masako Katsuki, \\xe2\\x80\\xa6 David J. Hosken</p>\\n                                </div>\\n                            </div>\\n                        </article>\\n                    </div>\\n                    \\n                    <div class=\"c-recommendations-list__item\" role=\"group\" aria-roledescription=\"slide\" aria-label=\"Recommendation 6 of 9\">\\n                        <article class=\"u-full-height c-card c-card--flush\">\\n                            <div class=\"c-card__layout u-full-height \">\\n                                \\n                                    <div class=\"c-card__image\"><img src=\"//media.springernature.com/w136h75/springer-static/image/art%3A10.1038%2Fs41598-020-65578-w/MediaObjects/41598_2020_65578_Fig1_HTML.png\" alt=\"\"></div>\\n                                \\n                                <div class=\"c-card__body u-display-flex u-flex-direction-column\">\\n                                    <div class=\"c-recommendations-column-switch\">\\n                                        <h3 class=\"c-card__title-recommendation u-h4\"><a class=\"c-card__link u-link-inherit\" href=\"https://doi.org/10.1038/s41598-020-65578-w\" data-track=\"click\" data-track-action=\"click recommendations\" data-track-label=\"10.1038/s41598-020-65578-w\">Social environment modulates investment in sex trait versus lifespan: red deer produce bigger antlers when facing more rivalry</a></h3>\\n                                        <p class=\"u-sans-serif u-mb-0\">08 June 2020</p>\\n                                    </div>\\n                                    <p class=\"c-recommendations-authors u-hide-at-sm u-sans-serif\">Juan Carranza, Javier P\\xc3\\xa9rez-Barber\\xc3\\xada, \\xe2\\x80\\xa6 Alberto Membrillo</p>\\n                                </div>\\n                            </div>\\n                        </article>\\n                    </div>\\n                    \\n                    <div class=\"c-recommendations-list__item\" role=\"group\" aria-roledescription=\"slide\" aria-label=\"Recommendation 7 of 9\">\\n                        <article class=\"u-full-height c-card c-card--flush\">\\n                            <div class=\"c-card__layout u-full-height \">\\n                                \\n                                    <div class=\"c-card__image\"><img src=\"//media.springernature.com/w136h75/springer-static/image/art%3A10.1038%2Fs41559-017-0397-y/MediaObjects/41559_2017_397_Fig1_HTML.jpg\" alt=\"\"></div>\\n                                \\n                                <div class=\"c-card__body u-display-flex u-flex-direction-column\">\\n                                    <div class=\"c-recommendations-column-switch\">\\n                                        <h3 class=\"c-card__title-recommendation u-h4\"><a class=\"c-card__link u-link-inherit\" href=\"https://doi.org/10.1038/s41559-017-0397-y\" data-track=\"click\" data-track-action=\"click recommendations\" data-track-label=\"10.1038/s41559-017-0397-y\">Social polymorphism is favoured by the co-evolution of dispersal with social behaviour</a></h3>\\n                                        <p class=\"u-sans-serif u-mb-0\">04 December 2017</p>\\n                                    </div>\\n                                    <p class=\"c-recommendations-authors u-hide-at-sm u-sans-serif\">Charles Mullon, Laurent Keller &amp; Laurent Lehmann</p>\\n                                </div>\\n                            </div>\\n                        </article>\\n                    </div>\\n                    \\n                    <div class=\"c-recommendations-list__item\" role=\"group\" aria-roledescription=\"slide\" aria-label=\"Recommendation 8 of 9\">\\n                        <article class=\"u-full-height c-card c-card--flush\">\\n                            <div class=\"c-card__layout u-full-height \">\\n                                \\n                                    <div class=\"c-card__image\"><img src=\"//media.springernature.com/w136h75/springer-static/image/art%3A10.1038%2Fs41559-018-0658-4/MediaObjects/41559_2018_658_Fig1_HTML.png\" alt=\"\"></div>\\n                                \\n                                <div class=\"c-card__body u-display-flex u-flex-direction-column\">\\n                                    <div class=\"c-recommendations-column-switch\">\\n                                        <h3 class=\"c-card__title-recommendation u-h4\"><a class=\"c-card__link u-link-inherit\" href=\"https://doi.org/10.1038/s41559-018-0658-4\" data-track=\"click\" data-track-action=\"click recommendations\" data-track-label=\"10.1038/s41559-018-0658-4\">Individual- and population-level drivers of consistent foraging success across environments</a></h3>\\n                                        <p class=\"u-sans-serif u-mb-0\">03 September 2018</p>\\n                                    </div>\\n                                    <p class=\"c-recommendations-authors u-hide-at-sm u-sans-serif\">Lysanne Snijders, Ralf H. J. M. Kurvers, \\xe2\\x80\\xa6 Jens Krause</p>\\n                                </div>\\n                            </div>\\n                        </article>\\n                    </div>\\n                    \\n                    <div class=\"c-recommendations-list__item\" role=\"group\" aria-roledescription=\"slide\" aria-label=\"Recommendation 9 of 9\">\\n                        <article class=\"u-full-height c-card c-card--flush\">\\n                            <div class=\"c-card__layout u-full-height \">\\n                                \\n                                    <div class=\"c-card__image\"><img src=\"//media.springernature.com/w136h75/springer-static/image/art%3A10.1038%2Fs42003-020-01597-7/MediaObjects/42003_2020_1597_Fig1_HTML.png\" alt=\"\"></div>\\n                                \\n                                <div class=\"c-card__body u-display-flex u-flex-direction-column\">\\n                                    <div class=\"c-recommendations-column-switch\">\\n                                        <h3 class=\"c-card__title-recommendation u-h4\"><a class=\"c-card__link u-link-inherit\" href=\"https://doi.org/10.1038/s42003-020-01597-7\" data-track=\"click\" data-track-action=\"click recommendations\" data-track-label=\"10.1038/s42003-020-01597-7\">Causal evidence for the adaptive benefits of social foraging in the wild</a></h3>\\n                                        <p class=\"u-sans-serif u-mb-0\">20 January 2021</p>\\n                                    </div>\\n                                    <p class=\"c-recommendations-authors u-hide-at-sm u-sans-serif\">Lysanne Snijders, Stefan Krause, \\xe2\\x80\\xa6 Ralf H.J.M. Kurvers</p>\\n                                </div>\\n                            </div>\\n                        </article>\\n                    </div>\\n                    \\n                </div>\\n            </div>\\n        </section>\\n    </div>\\n    <div class=\"js-greyout-page-background\" style=\"display:none\" data-component-grey-background></div>\\n        <script>\\n            window.dataLayer = window.dataLayer || [];\\n            window.dataLayer.push({\\n                recommendations: {\\n                    recommender: \\'semantic\\',\\n                    model: \\'specter\\',\\n                    policy_id: \\'NA\\',\\n                    timestamp: 1669935874\\n                }\\n            });\\n        </script>\\n    \\n\\n            </div>\\n        \\n        <article lang=\"en\">\\n            \\n                <div class=\"c-pdf-button__container u-mb-16 u-hide-at-lg js-context-bar-sticky-point-mobile\">\\n                    <div class=\"c-pdf-container\">\\n                        \\n                            \\n    \\n        <div class=\"c-pdf-download u-clear-both js-pdf-download\">\\n            <a href=\"/articles/hdy201692.pdf\" class=\"u-button u-button--full-width u-button--primary u-justify-content-space-between c-pdf-download__link\" data-article-pdf=\"true\" data-readcube-pdf-url=\"true\" data-test=\"download-pdf\" data-draft-ignore=\"true\" data-track=\"click\" data-track-action=\"download pdf\" data-track-label=\"link\" data-track-external download>\\n                <span class=\"c-pdf-download__text\">Download PDF</span>\\n                <svg aria-hidden=\"true\" focusable=\"false\" width=\"16\" height=\"16\" class=\"u-icon\"><use xlink:href=\"#global-icon-download\"/></svg>\\n            </a>\\n        </div>\\n    \\n\\n                        \\n                    </div>\\n                </div>\\n            \\n            <div class=\"c-article-header\">\\n                <header>\\n                    <ul class=\"c-article-identifiers\" data-test=\"article-identifier\">\\n                        \\n    \\n    \\n    \\n\\n                        <li class=\"c-article-identifiers__item\"><a href=\"#article-info\" data-track=\"click\" data-track-action=\"publication date\" data-track-label=\"link\">Published: <time datetime=\"2016-10-26\">26 October 2016</time></a></li>\\n                    </ul>\\n\\n                    <h1 class=\"c-article-title\" data-test=\"article-title\" data-article-title=\"\">An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment</h1>\\n                    <ul class=\"c-article-author-list c-article-author-list--short\" data-test=\"authors-list\" data-component-authors-activator=\"authors-list\"><li class=\"c-article-author-list__item\"><a data-test=\"author-name\" data-track=\"click\" data-track-action=\"open author\" data-track-label=\"link\" href=\"#auth-A-Bourret\" data-author-popup=\"auth-A-Bourret\" data-corresp-id=\"c1\">A Bourret<svg width=\"16\" height=\"16\" focusable=\"false\" role=\"img\" aria-hidden=\"true\" class=\"u-icon\"><use xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"#global-icon-email\"></use></svg></a><span class=\"u-js-hide\">\\xc2\\xa0\\n            <a class=\"js-orcid\" href=\"http://orcid.org/0000-0002-5387-3011\"><span class=\"u-visually-hidden\">ORCID: </span>orcid.org/0000-0002-5387-3011</a></span><sup class=\"u-js-hide\"><a href=\"#Aff1\">1</a></sup> &amp; </li><li class=\"c-article-author-list__item\"><a data-test=\"author-name\" data-track=\"click\" data-track-action=\"open author\" data-track-label=\"link\" href=\"#auth-D-Garant\" data-author-popup=\"auth-D-Garant\">D Garant</a><sup class=\"u-js-hide\"><a href=\"#Aff1\">1</a></sup>\\xc2\\xa0</li></ul>\\n\\n                    \\n\\n                    <p class=\"c-article-info-details\" data-container-section=\"info\">\\n                        \\n    <a data-test=\"journal-link\" href=\"/hdy\"><i data-test=\"journal-title\">Heredity</i></a>\\n\\n                        <b data-test=\"journal-volume\"><span class=\"u-visually-hidden\">volume</span>\\xc2\\xa0118</b>,\\xc2\\xa0<span class=\"u-visually-hidden\">pages </span>229\\xe2\\x80\\x93238 (<span data-test=\"article-publication-year\">2017</span>)<a href=\"#citeas\" class=\"c-article-info-details__cite-as u-hide-print\" data-track=\"click\" data-track-action=\"cite this article\" data-track-label=\"link\">Cite this article</a>\\n                    </p>\\n                    \\n    <div class=\"c-article-metrics-bar__wrapper u-clear-both\">\\n        <ul class=\"c-article-metrics-bar u-list-reset\">\\n            \\n                <li class=\" c-article-metrics-bar__item\">\\n                    <p class=\"c-article-metrics-bar__count\">944 <span class=\"c-article-metrics-bar__label\">Accesses</span></p>\\n                </li>\\n            \\n            \\n                <li class=\"c-article-metrics-bar__item\">\\n                    <p class=\"c-article-metrics-bar__count\">8 <span class=\"c-article-metrics-bar__label\">Citations</span></p>\\n                </li>\\n            \\n            \\n                \\n                    <li class=\"c-article-metrics-bar__item\">\\n                        <p class=\"c-article-metrics-bar__count\">2 <span class=\"c-article-metrics-bar__label\">Altmetric</span></p>\\n                    </li>\\n                \\n            \\n            <li class=\"c-article-metrics-bar__item\">\\n                <p class=\"c-article-metrics-bar__details\"><a href=\"/articles/hdy201692/metrics\" data-track=\"click\" data-track-action=\"view metrics\" data-track-label=\"link\" rel=\"nofollow\">Metrics <span class=\"u-visually-hidden\">details</span></a></p>\\n            </li>\\n        </ul>\\n    </div>\\n\\n                    \\n                </header>\\n\\n                \\n    <div class=\"u-js-hide\" data-component=\"article-subject-links\">\\n        <h3 class=\"c-article__sub-heading\">Subjects</h3>\\n        <ul class=\"c-article-subject-list\">\\n            <li class=\"c-article-subject-list__subject\"><a href=\"/subjects/evolutionary-genetics\" data-track=\"click\" data-track-action=\"view subject\" data-track-label=\"link\">Evolutionary genetics</a></li>\\n        </ul>\\n    </div>\\n\\n                \\n    \\n\\n    \\n\\n                \\n            </div>\\n\\n        <div class=\"c-article-body\">\\n            <section aria-labelledby=\"Abs1\" data-title=\"Abstract\" lang=\"en\"><div class=\"c-article-section\" id=\"Abs1-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Abs1\">Abstract</h2><div class=\"c-article-section__content\" id=\"Abs1-content\"><p>Quantitative genetics approaches, and particularly animal models, are widely used to assess the genetic (co)variance of key fitness related traits and infer adaptive potential of wild populations. Despite the importance of precision and accuracy of genetic variance estimates and their potential sensitivity to various ecological and population specific factors, their reliability is rarely tested explicitly. Here, we used simulations and empirical data collected from an 11-year study on tree swallow (<i>Tachycineta bicolor</i>), a species showing a high rate of extra-pair paternity and a low recruitment rate, to assess the importance of identity errors, structure and size of the pedigree on quantitative genetic estimates in our dataset. Our simulations revealed an important lack of precision in heritability and genetic-correlation estimates for most traits, a low power to detect significant effects and important identifiability problems. We also observed a large bias in heritability estimates when using the social pedigree instead of the genetic one (deflated heritabilities) or when not accounting for an important cause of resemblance among individuals (for example, permanent environment or brood effect) in model parameterizations for some traits (inflated heritabilities). We discuss the causes underlying the low reliability observed here and why they are also likely to occur in other study systems. Altogether, our results re-emphasize the difficulties of generalizing quantitative genetic estimates reliably from one study system to another and the importance of reporting simulation analyses to evaluate these important issues.</p></div></div></section>\\n            <noscript>\\n                \\n            </noscript>\\n\\n                \\n            \\n                <div class=\"main-content\">\\n                <section data-title=\"Introduction\"><div class=\"c-article-section\" id=\"Sec1-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec1\">Introduction</h2><div class=\"c-article-section__content\" id=\"Sec1-content\"><p>Understanding the genetic component underlying important fitness related traits is essential to infer the adaptive potential of wild populations (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Lynch and Walsh, 1998\" title=\"Lynch M, Walsh B . (1998) Genetics and Analysis of Quantitative Traits. Sinauer Associates: Sunderland, MA, USA.\" href=\"/articles/hdy201692#ref-CR33\" id=\"ref-link-section-d40968538e365\">Lynch and Walsh, 1998</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Hendry et al., 2011\" title=\"Hendry AP, Kinnison MT, Heino M, Day T, Smith TB, Fitt G et al. (2011). Evolutionary principles and their practical application. Evol Appl 4: 159\\xe2\\x80\\x93183.\" href=\"/articles/hdy201692#ref-CR16\" id=\"ref-link-section-d40968538e368\">Hendry et al., 2011</a>). However, as most traits are likely under the control of numerous genes with small effects (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Lande, 1981\" title=\"Lande R . (1981). The minimum number of genes contributing to quantitative variation between and within populations. Genetics 99: 541\\xe2\\x80\\x93553.\" href=\"/articles/hdy201692#ref-CR28\" id=\"ref-link-section-d40968538e371\">Lande, 1981</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Falconer and Mackay, 1996\" title=\"Falconer DS, Mackay TFC . (1996) Introduction to Quantitative Genetics, 4th edn. Pearson: Essex, UK.\" href=\"/articles/hdy201692#ref-CR8\" id=\"ref-link-section-d40968538e374\">Falconer and Mackay, 1996</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Husby et al., 2015\" title=\"Husby A, Kawakami T, R\\xc3\\xb6nneg\\xc3\\xa5rd L, Smeds L, Ellegren H, Qvarnstr\\xc3\\xb6m A . (2015). Genome-wide association mapping in a wild avian population identifies a link between genetic and phenotypic variation in a life-history trait. Proc R Soc B 282: 20150156.\" href=\"/articles/hdy201692#ref-CR21\" id=\"ref-link-section-d40968538e377\">Husby et al., 2015</a>), potentially interacting with the genetic components of other traits (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Lande and Arnold, 1983\" title=\"Lande R, Arnold SJ . (1983). The measurement of selection on correlated characters. Evolution 37: 1210\\xe2\\x80\\x931226.\" href=\"/articles/hdy201692#ref-CR29\" id=\"ref-link-section-d40968538e381\">Lande and Arnold, 1983</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Blows and Hoffmann, 2005\" title=\"Blows MW, Hoffmann AA . (2005). A reassessment of genetic limits to evoutionary change. Ecology 86: 1371\\xe2\\x80\\x931384.\" href=\"/articles/hdy201692#ref-CR3\" id=\"ref-link-section-d40968538e384\">Blows and Hoffmann, 2005</a>) and/or with the environment in which they are expressed (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Via and Lande, 1985\" title=\"Via S, Lande R . (1985). Genotype-environment interaction and the evolution of phenotypic plasticity. Evolution 39: 505\\xe2\\x80\\x93522.\" href=\"/articles/hdy201692#ref-CR42\" id=\"ref-link-section-d40968538e387\">Via and Lande, 1985</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Hoffmann and Meril\\xc3\\xa4, 1999\" title=\"Hoffmann AA, Meril\\xc3\\xa4 J . (1999). Heritable variation and evolution under favourable and unfavourable conditions. Trends Ecol Evol 14: 96\\xe2\\x80\\x93101.\" href=\"/articles/hdy201692#ref-CR17\" id=\"ref-link-section-d40968538e390\">Hoffmann and Meril\\xc3\\xa4, 1999</a>), this task remains challenging despite rapid advances in whole-genome analysis methods (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Vinkhuyzen et al., 2013\" title=\"Vinkhuyzen AAE, Wray NR, Yang J, Goddard ME, Visscher PM . (2013). Estimation and partition of heritability in human populations using whole-genome analysis methods. Annu Rev Genet 47: 75\\xe2\\x80\\x9395.\" href=\"/articles/hdy201692#ref-CR44\" id=\"ref-link-section-d40968538e393\">Vinkhuyzen et al., 2013</a>). For these reasons, quantitative genetic approaches\\xe2\\x80\\x94statistical methods using known relationships between individuals to assess the genetic and environmental components of a population phenotypic variance\\xe2\\x80\\x94remain efficient ways to infer the underlying genetic variation of focal traits (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Falconer and Mackay, 1996\" title=\"Falconer DS, Mackay TFC . (1996) Introduction to Quantitative Genetics, 4th edn. Pearson: Essex, UK.\" href=\"/articles/hdy201692#ref-CR8\" id=\"ref-link-section-d40968538e396\">Falconer and Mackay, 1996</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Kruuk et al., 2008\" title=\"Kruuk LEB, Slate J, Wilson AJ . (2008). New answers for old questions: the evolutionary quantitative genetics of wild animal populations. Annu Rev Ecol Evol Syst 39: 525\\xe2\\x80\\x93548.\" href=\"/articles/hdy201692#ref-CR26\" id=\"ref-link-section-d40968538e400\">Kruuk et al., 2008</a>). For this purpose, animal models are particularly suitable for wild population datasets because they allow the use of all relationships between individuals and can account for missing data, unbalanced designs and other potential biases (for example, environmental causes of phenotypic similarity among individuals rather than genetic ones) (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Henderson, 1984\" title=\"Henderson CR . (1984) Applications of Linear Models in Animal Breeding. University of Guelph: Guelph, ON, Canada.\" href=\"/articles/hdy201692#ref-CR15\" id=\"ref-link-section-d40968538e403\">Henderson, 1984</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Kruuk, 2004\" title=\"Kruuk LEB . (2004). Estimating genetic parameters in natural populations using the \\xe2\\x80\\x98animal model\\xe2\\x80\\x99. Philos Trans R Soc B 359: 873\\xe2\\x80\\x93890.\" href=\"/articles/hdy201692#ref-CR24\" id=\"ref-link-section-d40968538e406\">Kruuk, 2004</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Kruuk and Hadfield, 2007\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890\\xe2\\x80\\x931903.\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d40968538e409\">Kruuk and Hadfield, 2007</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wilson et al., 2010\" title=\"Wilson AJ, R\\xc3\\xa9ale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist\\xe2\\x80\\x99s guide to the animal model. J Anim Ecol 79: 13\\xe2\\x80\\x9326.\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d40968538e412\">Wilson et al., 2010</a>). Moreover, heritability estimates from wild populations obtained with animal models are generally more accurate than those estimated with traditional parent\\xe2\\x80\\x93offspring regressions (reviewed in <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Postma, 2014\" title=\"Postma E . (2014). Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates? In: Charmantier A, Garant D et al. (eds) Quantitative Genetics in the Wild. Oxford University Press, Oxford, UK, pp 16\\xe2\\x80\\x9333.\" href=\"/articles/hdy201692#ref-CR39\" id=\"ref-link-section-d40968538e415\">Postma, 2014</a>).</p><p>Several factors will influence the validity of the genetic variance estimates obtained in the wild, which in turn will affect our capacity to infer population responses to selective pressures (for example, through breeder\\xe2\\x80\\x99s or Robertson\\xe2\\x80\\x93Price equation, see <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Morrissey et al., 2010\" title=\"Morrissey MB, Kruuk LEB, Wilson AJ . (2010). The danger of applying the breeder\\xe2\\x80\\x99s equation in observational studies of natural populations. J Evol Biol 23: 2277\\xe2\\x80\\x932288.\" href=\"/articles/hdy201692#ref-CR35\" id=\"ref-link-section-d40968538e421\">Morrissey et al., 2010</a>). Precision\\xe2\\x80\\x94the repeatability of a measurement\\xe2\\x80\\x94is highly dependent on sample size both in terms of pedigree depth and completeness, but also on the complexity of the genetic architecture underlying the focal trait (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Morrissey et al., 2007\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309\\xe2\\x80\\x932321.\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d40968538e424\">Morrissey et al., 2007</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference de Villemereuil et al., 2013\" title=\"de Villemereuil P, Gimenez O, Doligez B . (2013). Comparing parent\\xe2\\x80\\x93offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits. Methods Ecol Evol 4: 260\\xe2\\x80\\x93275.\" href=\"/articles/hdy201692#ref-CR43\" id=\"ref-link-section-d40968538e427\">de Villemereuil et al., 2013</a>). For example, given two pedigrees with identical sample sizes, a pedigree with a weak degree of connectivity between individuals, for example, in populations with strong natal dispersion and/or immigration resulting in few related individuals, will be less precise than a pedigree with known relationships between all individuals (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wilson et al., 2010\" title=\"Wilson AJ, R\\xc3\\xa9ale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist\\xe2\\x80\\x99s guide to the animal model. J Anim Ecol 79: 13\\xe2\\x80\\x9326.\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d40968538e430\">Wilson et al., 2010</a>). However, precision is rarely tested <i>per se</i>, but rather deduced from the size of standard errors around estimates (for example, <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Charmantier and R\\xc3\\xa9ale, 2005\" title=\"Charmantier A, R\\xc3\\xa9ale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839\\xe2\\x80\\x932850.\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d40968538e437\">Charmantier and R\\xc3\\xa9ale, 2005</a>). Accuracy\\xe2\\x80\\x94the proximity between the estimated value and the true value\\xe2\\x80\\x94is affected mainly by pedigree errors (that is, erroneous links between individuals in the pedigree; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Charmantier and R\\xc3\\xa9ale, 2005\" title=\"Charmantier A, R\\xc3\\xa9ale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839\\xe2\\x80\\x932850.\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d40968538e440\">Charmantier and R\\xc3\\xa9ale, 2005</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Morrissey et al., 2007\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309\\xe2\\x80\\x932321.\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d40968538e443\">Morrissey et al., 2007</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Firth et al., 2015\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336\\xe2\\x80\\x931344.\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d40968538e446\">Firth et al., 2015</a>) and by model parameterizations (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Kruuk and Hadfield, 2007\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890\\xe2\\x80\\x931903.\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d40968538e449\">Kruuk and Hadfield, 2007</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wilson, 2008\" title=\"Wilson AJ . (2008). Why h2 does not always equal VA/VP? J Evol Biol 21: 647\\xe2\\x80\\x93650.\" href=\"/articles/hdy201692#ref-CR46\" id=\"ref-link-section-d40968538e452\">Wilson, 2008</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wolak et al., 2015\" title=\"Wolak ME, Roff DA, Fairbairn DJ . (2015). Are we underestimating the genetic variances of dimorphic traits? Ecol Evol 5: 590\\xe2\\x80\\x93597.\" href=\"/articles/hdy201692#ref-CR51\" id=\"ref-link-section-d40968538e456\">Wolak et al., 2015</a>). Pedigree errors could be particularly common in study systems where extra-pair copulations are important and parental links in pedigrees constructed solely based on social pair observations (that is, social pedigree). Theoretically, the presence of extra-pair paternity (EPP), if not accounted for, can bias downward the heritability estimates, even though evidence so far suggests that this bias is generally small in wild populations (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Charmantier and R\\xc3\\xa9ale, 2005\" title=\"Charmantier A, R\\xc3\\xa9ale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839\\xe2\\x80\\x932850.\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d40968538e459\">Charmantier and R\\xc3\\xa9ale, 2005</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference B\\xc3\\xa9r\\xc3\\xa9nos et al., 2014\" title=\"B\\xc3\\xa9r\\xc3\\xa9nos C, Ellis PA, Pilkington JG, Pemberton JM . (2014). Estimating quantitative genetic parameters in wild populations: a comparison of pedigree and genomic approaches. Mol Ecol 23: 3434\\xe2\\x80\\x933451.\" href=\"/articles/hdy201692#ref-CR2\" id=\"ref-link-section-d40968538e462\">B\\xc3\\xa9r\\xc3\\xa9nos et al., 2014</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Firth et al., 2015\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336\\xe2\\x80\\x931344.\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d40968538e465\">Firth et al., 2015</a>; but see <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Lee and Pollak, 1997\" title=\"Lee C, Pollak EJ . (1997). Influence of sire misidentification on sire x year interaction variance and direct-maternal genetic covariance for weaning weight in beef cattle. J Anim Sci 75: 2858\\xe2\\x80\\x932863.\" href=\"/articles/hdy201692#ref-CR30\" id=\"ref-link-section-d40968538e468\">Lee and Pollak, 1997</a> for reports of bias in the animal breeding litterature; see also <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Walther and Moore, 2005\" title=\"Walther BA, Moore JL . (2005). The concepts of bias, precision and accuracy, and their use in testing the performance of species richness estimators, with a literature review of estimator performance. Ecography 28: 815\\xe2\\x80\\x93829.\" href=\"/articles/hdy201692#ref-CR45\" id=\"ref-link-section-d40968538e471\">Walther and Moore, 2005</a> for a detailed discussion on the concepts of bias, precision and accuracy).</p><p>Despite previous knowledge of what can affect precision and accuracy of genetic variance estimates, extrapolating the reliability of these estimates from one study system to another is difficult given the large diversity of life history observed across species that inevitably modified dataset and pedigree structures. Therefore, simulation analyses are recommended for testing the limits of a particular dataset, pedigree and model to answer specific biological questions (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Morrissey et al., 2007\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309\\xe2\\x80\\x932321.\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d40968538e477\">Morrissey et al., 2007</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wilson et al., 2010\" title=\"Wilson AJ, R\\xc3\\xa9ale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist\\xe2\\x80\\x99s guide to the animal model. J Anim Ecol 79: 13\\xe2\\x80\\x9326.\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d40968538e480\">Wilson et al., 2010</a>). Simulation frameworks allow the assessment of power\\xe2\\x80\\x94the probability of detecting an effect given that this effect is true\\xe2\\x80\\x94of a particular dataset or model. In animal models, power is affected by the same factors as precision (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Quinn et al., 2006\" title=\"Quinn JL, Charmantier A, Garant D, Sheldon BC . (2006). Data depth, data completeness, and their influence on quantitative genetic estimation in two contrasting bird populations. J Evol Biol 19: 994\\xe2\\x80\\x931002.\" href=\"/articles/hdy201692#ref-CR40\" id=\"ref-link-section-d40968538e483\">Quinn et al., 2006</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Morrissey et al., 2007\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309\\xe2\\x80\\x932321.\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d40968538e486\">Morrissey et al., 2007</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference B\\xc3\\xa9r\\xc3\\xa9nos et al., 2014\" title=\"B\\xc3\\xa9r\\xc3\\xa9nos C, Ellis PA, Pilkington JG, Pemberton JM . (2014). Estimating quantitative genetic parameters in wild populations: a comparison of pedigree and genomic approaches. Mol Ecol 23: 3434\\xe2\\x80\\x933451.\" href=\"/articles/hdy201692#ref-CR2\" id=\"ref-link-section-d40968538e489\">B\\xc3\\xa9r\\xc3\\xa9nos et al., 2014</a>). However, performing a power analysis alone could provide an incomplete picture of the validity of genetic variance estimates. In fact, a particular model applied on a dataset could have a high power to estimate a given effect, but the model itself might be missing a critical variance term, leading to precise but inaccurate estimates (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Kruuk and Hadfield, 2007\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890\\xe2\\x80\\x931903.\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d40968538e493\">Kruuk and Hadfield, 2007</a>). Similarly, even if all important variance terms are included in a particular model, the model variance structure might not allow discrimination between two variance terms being included, a problem of statistical models related to their low \\xe2\\x80\\x98identifiability\\xe2\\x80\\x99 (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Bolker, 2008\" title=\"Bolker B . (2008) Ecological Models and Data in R. Princeton, University Princeton, NJ, USA.\" href=\"/articles/hdy201692#ref-CR4\" id=\"ref-link-section-d40968538e496\">Bolker, 2008</a>). Confounded parameters could be frequent when applying animal models to wild populations (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wilson et al., 2010\" title=\"Wilson AJ, R\\xc3\\xa9ale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist\\xe2\\x80\\x99s guide to the animal model. J Anim Ecol 79: 13\\xe2\\x80\\x9326.\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d40968538e499\">Wilson et al., 2010</a>), but the extent of identifiability problems is still unknown. Finally, reliability of quantitative genetics estimates is even less reported for multivariate models than univariate models, limiting our confidence on important parameter estimations such as genetic additive covariance/correlation among traits.</p><p>Here, we used an 11-year study on Tree swallow (<i>Tachycineta bicolor</i> (Vieillot, 1808)) to assess the effects of identity errors, structure and size of a pedigree on reliability of quantitative genetic estimates. Tree swallows are small migratory passerines and their breeding grounds are widely distributed across North America (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Winkler et al., 2011\" title=\"Winkler D, Hallinger K, Ardia D, Robertson R, Stutchbury B, Cohen R . (2011). Tree swallow (Tachycineta bicolor. Birds North Am Online. &#xA;                  http://bna.birds.cornelledu/bna/&#xA;                  &#xA;                . Accessed on 1 Oct 2013.\" href=\"/articles/hdy201692#ref-CR49\" id=\"ref-link-section-d40968538e508\">Winkler et al., 2011</a>). Similarly to other migrating species, tree swallow shows high natal dispersal, high mortality within the first year and low nestling recruitment rate (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Hosner and Winkler, 2007\" title=\"Hosner PA, Winkler DW . (2007). Dispersal distances of tree swallows estimated from continent-wide and limited-area data. J Field Ornithol 78: 290\\xe2\\x80\\x93297.\" href=\"/articles/hdy201692#ref-CR18\" id=\"ref-link-section-d40968538e511\">Hosner and Winkler, 2007</a>), which reduce kinship among individuals in monitored populations. This socially monogamous species also displays one of the highest rate of EPP documented in birds (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Griffith et al., 2002\" title=\"Griffith SC, Owens IPF, Thuman KA . (2002). Extra pair paternity in birds: a review of interspecific variation and adaptive function. Mol Ecol 11: 2195\\xe2\\x80\\x932212.\" href=\"/articles/hdy201692#ref-CR13\" id=\"ref-link-section-d40968538e514\">Griffith et al., 2002</a>), with more than 80% of nests containing at least one extra-pair offspring and overall around 50% of nestlings resulting from extra-pair copulations (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Dunn et al., 2001\" title=\"Dunn PO, Whittingham LA, Pitcher TE . (2001). Mating systems, sperm competition, and the evolution of sexual dimorphism in birds. Evolution 55: 161\\xe2\\x80\\x93175.\" href=\"/articles/hdy201692#ref-CR7\" id=\"ref-link-section-d40968538e517\">Dunn et al., 2001</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Lessard et al., 2014\" title=\"Lessard A, Bourret A, B\\xc3\\xa9lisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor. Behav Ecol Sociobiol 68: 733\\xe2\\x80\\x93742.\" href=\"/articles/hdy201692#ref-CR32\" id=\"ref-link-section-d40968538e521\">Lessard et al., 2014</a>). In the context of building a pedigree for quantitative genetics analyses, this means that half of paternal links would be erroneous if using a social pedigree. This represents a higher proportion of EPP than those tested in previous studies to assess their impacts on heritability estimates (for example, in <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Charmantier and R\\xc3\\xa9ale, 2005\" title=\"Charmantier A, R\\xc3\\xa9ale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839\\xe2\\x80\\x932850.\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d40968538e524\">Charmantier and R\\xc3\\xa9ale, 2005</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Firth et al., 2015\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336\\xe2\\x80\\x931344.\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d40968538e527\">Firth et al., 2015</a>), but it is a proportion that will be typically found in other similar study systems (see <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Griffith et al., 2002\" title=\"Griffith SC, Owens IPF, Thuman KA . (2002). Extra pair paternity in birds: a review of interspecific variation and adaptive function. Mol Ecol 11: 2195\\xe2\\x80\\x932212.\" href=\"/articles/hdy201692#ref-CR13\" id=\"ref-link-section-d40968538e530\">Griffith et al., 2002</a> and references therein). Furthermore, reliability of genetic variance estimates for this and similar species is difficult to predict based on previous results obtained on other species, as the typical poorly connected pedigree available for migratory species could be compensated by the natural half-sibs design resulting from the high rate of EPP occurring in tree swallows.</p><p>In this study, we used a social and a genetic pedigree to estimate and compare genetic additive (co-)variances and heritability of traits differing in their completeness through the pedigree structure. More specifically, we first used empirical data to assess the bias resulting from using a social pedigree over a genetic one. Then, we used simulated data to assess precision and accuracy of quantitative genetic estimates obtained with both pedigrees, as well as power of datasets and models, and identifiability among variance terms.</p></div></div></section><section data-title=\"Materials and methods\"><div class=\"c-article-section\" id=\"Sec2-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec2\">Materials and methods</h2><div class=\"c-article-section__content\" id=\"Sec2-content\"><h3 class=\"c-article__sub-heading\" id=\"Sec3\">Study species, system and phenotypic data</h3><p>Tree swallows breed in tree cavities or nest boxes, they produce only one clutch per year, containing on average five eggs and both parents provide care to nestlings (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Winkler et al., 2011\" title=\"Winkler D, Hallinger K, Ardia D, Robertson R, Stutchbury B, Cohen R . (2011). Tree swallow (Tachycineta bicolor. Birds North Am Online. &#xA;                  http://bna.birds.cornelledu/bna/&#xA;                  &#xA;                . Accessed on 1 Oct 2013.\" href=\"/articles/hdy201692#ref-CR49\" id=\"ref-link-section-d40968538e549\">Winkler et al., 2011</a>). Since 2004, we have intensively followed tree swallow activities during the breeding season at 400 nest boxes equally distributed across 40 farms in an area covering 10\\xe2\\x80\\x89200\\xe2\\x80\\x89km<sup>2</sup> in southern Qu\\xc3\\xa9bec, Canada (detailed in <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Ghilain and B\\xc3\\xa9lisle, 2008\" title=\"Ghilain A, B\\xc3\\xa9lisle M . (2008). Breeding success of tree swallows along a gradient of agricultural intensification. Ecol Appl 18: 1140\\xe2\\x80\\x931154.\" href=\"/articles/hdy201692#ref-CR11\" id=\"ref-link-section-d40968538e554\">Ghilain and B\\xc3\\xa9lisle, 2008</a>). Nest boxes were visited every 2 days from May to July to record nest box occupancy and important brood characteristics (for example, laying date, clutch size and hatching date). Adults and nestlings were individually marked with an aluminum band (US Fish and Wildlife Service). Adults were captured directly in their nest box by a trap system, during the incubation period and food provisioning period for females and males, respectively. Morphological measurements of body mass (\\xc2\\xb10.01\\xe2\\x80\\x89g), non-flattened wing length (\\xc2\\xb10.5\\xe2\\x80\\x89mm) and tarsus length (\\xc2\\xb10.02\\xe2\\x80\\x89mm) were taken on adult tree swallows during captures. Females were classified based on their plumage color as second-year (brown) or after-second-year (blue-green) (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Hussell, 1983\" title=\"Hussell DJT . (1983). Age and plumage color in female tree swallows. J Field Ornithol 54: 312\\xe2\\x80\\x93318.\" href=\"/articles/hdy201692#ref-CR22\" id=\"ref-link-section-d40968538e557\">Hussell, 1983</a>) and a minimal age was determined for all adults based on the year they were first observed in the study system. Nestlings were captured at 16 days old before they fledged (fledging occurs around 18\\xe2\\x80\\x9322 days) to record body mass (\\xc2\\xb10.01\\xe2\\x80\\x89g), primary length (hereafter wing length; \\xc2\\xb10.02\\xe2\\x80\\x89mm) and tarsus length (\\xc2\\xb10.02\\xe2\\x80\\x89mm). Since 2006, blood samples of adults and nestlings have been collected on a qualitative P8 grade filter paper (Thermo Fisher Scientific, Waltham, MA, USA) for molecular analysis (see below).</p><p>To reflect the large differences that can exist between traits in terms of sample size and amount of standing genetic variance, we focused on nine phenotypic traits grouped in three categories: (1) morphological traits, which included wing length, body mass and tarsus length of all adults; (2) reproductive traits, which were restricted to females, and included laying date (that is, date of the first egg laid), clutch size (that is, number of eggs laid within a nest box) and incubation duration (defined as hatching date\\xe2\\x88\\x92(laying date+clutch size\\xe2\\x88\\x921)); and (3) nestling traits, which included wing length, body mass and tarsus length measured at the age of 16 days. Most traits were measured since the beginning of the research program (that is, in 2004), but some traits were first measured later (nestling body mass: 2005; nestling wing length: 2006; adult/nestling tarsus length: 2007) creating differences in data completeness among traits (sample size per traits can be found in <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table A1</a>).</p><h3 class=\"c-article__sub-heading\" id=\"Sec4\">Molecular analysis and pedigree construction</h3><p>DNA extraction, molecular sexing and microsatellite data analyses are detailed in <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Lessard et al. (2014)\" title=\"Lessard A, Bourret A, B\\xc3\\xa9lisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor. Behav Ecol Sociobiol 68: 733\\xe2\\x80\\x93742.\" href=\"/articles/hdy201692#ref-CR32\" id=\"ref-link-section-d40968538e574\">Lessard et al. (2014)</a>. Briefly, DNA was extracted from blood samples following a standard salt-extraction method and DNA concentration was determined by migration on 2% agarose gels with a molecular weight standard. A molecular sexing technique was used to determine nestling sexes and to validate adult field observations. All DNA samples were characterized at six microsatellites loci using an AB3130xl automated DNA sequencer and allele lengths were determined using GeneMapper v4.1 (Applied Biosystems, Foster City, CA, USA).</p><p>We constructed a social pedigree using social male identities (that is, males caught in nest boxes while feeding the young) and a genetic pedigree using genetic father identities (that is, males assigned as fathers using genetic analyses\\xe2\\x80\\x94see below). In both pedigrees, dam identities were first determined by female captures during egg incubation and then verified genetically based on locus mismatches with nestlings (2.1% of broods with two females captured within the same nest box, from which only 3.6% were from mixed maternity; 11 nestling genotypes (0.15%) mismatched at more than two loci with their social mother genotype). Genetic fathers were determined by parental assignations with <span class=\"u-small-caps\">CERVUS</span> v3.0.7 (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Kalinowski et al., 2007\" title=\"Kalinowski ST, Taper ML, Marshall TC . (2007). Revising how the computer program cervus accommodates genotyping error increases success in paternity assignment. Mol Ecol 16: 1099\\xe2\\x80\\x931106.\" href=\"/articles/hdy201692#ref-CR23\" id=\"ref-link-section-d40968538e583\">Kalinowski et al., 2007</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Lemons et al., 2015\" title=\"Lemons PR, Marshall TC, McCloskey SE, Sethi SA, Schmutz JA, Sedinger JS . (2015). A likelihood-based approach for assessment of extra-pair paternity and conspecific brood parasitism in natural populations. Mol Ecol Resour 7: 107\\xe2\\x80\\x93116.\" href=\"/articles/hdy201692#ref-CR31\" id=\"ref-link-section-d40968538e586\">Lemons et al., 2015</a>) in a three step procedure. First, we proceeded to father assignments each year separately following a method slightly modified from <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Lessard et al. (2014)\" title=\"Lessard A, Bourret A, B\\xc3\\xa9lisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor. Behav Ecol Sociobiol 68: 733\\xe2\\x80\\x93742.\" href=\"/articles/hdy201692#ref-CR32\" id=\"ref-link-section-d40968538e589\">Lessard et al. (2014)</a>. Candidate fathers considered in analyses included all males captured during a given breeding season within 15\\xe2\\x80\\x89km of the nest box of interest (see <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Lessard et al., 2014\" title=\"Lessard A, Bourret A, B\\xc3\\xa9lisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor. Behav Ecol Sociobiol 68: 733\\xe2\\x80\\x93742.\" href=\"/articles/hdy201692#ref-CR32\" id=\"ref-link-section-d40968538e592\">Lessard et al., 2014</a> for the rationale behind this approach), but also all males not captured in a given year but suspected of being present outside of our nest box system (that is, captured on the same farm on both previous and following years). These assignations were based on a 90% confidence level, assuming a 2% mistyping error rate and we used the percentage of social fathers captured as the percentage of candidate fathers known (variable among years, range: 64\\xe2\\x80\\x9388%). Second, social fathers, when known, were tested for being genetic fathers of offspring in their nest using the likelihood-based approach of <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Lemons et al. (2015)\" title=\"Lemons PR, Marshall TC, McCloskey SE, Sethi SA, Schmutz JA, Sedinger JS . (2015). A likelihood-based approach for assessment of extra-pair paternity and conspecific brood parasitism in natural populations. Mol Ecol Resour 7: 107\\xe2\\x80\\x93116.\" href=\"/articles/hdy201692#ref-CR31\" id=\"ref-link-section-d40968538e596\">Lemons et al. (2015)</a>. Briefly, we re-ran the parental analysis with the social father as the unique candidate father for a given nestling, and we defined the proportion of sampled fathers as the proportion of nestlings without any locus mismatch with its social father (that is, the probability of the social male being the true father). We then extracted, for each nestling, the critical logarithm of odds score associated with 95% confidence that its social father was not its true father and we compared these scores with those observed in regular parental analyses. Males significantly assigned to nestlings in the initial parental analysis (that is, step 1) were considered as their genetic fathers. For the remaining nestlings (that is, without a significant male assignment at step 1), if their social fathers could not be excluded (that is, step 2), they were considered as their genetic fathers but otherwise no genetic father were assigned to them (see <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure A1</a> for the exact number of fathers assigned to a nestling at each step).</p><p>Summary statistics for both pedigrees were obtained with the package <span class=\"u-small-caps\">PEDANTICS</span> (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Morrissey et al., 2007\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309\\xe2\\x80\\x932321.\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d40968538e608\">Morrissey et al., 2007</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Morrissey and Wilson, 2010\" title=\"Morrissey MB, Wilson AJ . (2010). PEDANTICS: an R package for pedigree-based genetic simulation and pedigree manipulation, characterization and viewing. Mol Ecol Resour 10: 711\\xe2\\x80\\x93719.\" href=\"/articles/hdy201692#ref-CR37\" id=\"ref-link-section-d40968538e611\">Morrissey and Wilson, 2010</a>) in R v3.2.0 (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference R Core Team, 2015\" title=\"Core Team R . (2015). R: a language and environment for statistical computing &#xA;                  http://www.R-project.org/&#xA;                  &#xA;                .\" href=\"/articles/hdy201692#ref-CR41\" id=\"ref-link-section-d40968538e614\">R Core Team, 2015</a>). These statistics were computed for complete pedigrees, but also for pedigrees pruned to contain only informative individuals based on the availability of phenotypic data for each trait (hereafter pruned pedigrees) and are presented for each trait category in <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"table anchor\" href=\"/articles/hdy201692#Tab1\">Table 1</a> (see also <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table A2</a> for more information on each trait).</p><div class=\"c-article-table\" data-test=\"inline-table\" data-container-section=\"table\" id=\"table-1\"><figure><figcaption class=\"c-article-table__figcaption\"><b id=\"Tab1\" data-test=\"table-caption\">Table 1 Summary statistics for social and genetic pedigrees, in complete pedigrees (total) or pruned pedigrees detailed for each trait category (means for all morphological, reproductive and nestling traits) based on data collected between 2004 and 2014 in our study system in southern Qu\\xc3\\xa9bec</b></figcaption><div class=\"u-text-right u-hide-print\"><a class=\"c-article__pill-button\" data-test=\"table-link\" data-track=\"click\" data-track-action=\"view table\" data-track-label=\"button\" rel=\"nofollow\" href=\"/articles/hdy201692/tables/1\" aria-label=\"Full size table 1\"><span>Full size table</span><svg width=\"16\" height=\"16\" focusable=\"false\" role=\"img\" aria-hidden=\"true\" class=\"u-icon\"><use xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"#global-icon-chevron-right\"></use></svg></a></div></figure></div><h3 class=\"c-article__sub-heading\" id=\"Sec5\">Estimations of quantitative genetic parameters</h3><p>To estimate additive genetic (co)variances of our focal traits, we used both univariate and multivariate animal models (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Kruuk, 2004\" title=\"Kruuk LEB . (2004). Estimating genetic parameters in natural populations using the \\xe2\\x80\\x98animal model\\xe2\\x80\\x99. Philos Trans R Soc B 359: 873\\xe2\\x80\\x93890.\" href=\"/articles/hdy201692#ref-CR24\" id=\"ref-link-section-d40968538e1273\">Kruuk, 2004</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wilson et al., 2010\" title=\"Wilson AJ, R\\xc3\\xa9ale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist\\xe2\\x80\\x99s guide to the animal model. J Anim Ecol 79: 13\\xe2\\x80\\x9326.\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d40968538e1276\">Wilson et al., 2010</a>). Fixed effects (for example, age, sex) were included for some traits based on mixed model analyses detailed in <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Information</a>. For adult morphological and reproductive traits, full univariate animal models were constructed as follows:</p><div id=\"Equ1\" class=\"c-article-equation\"><div class=\"c-article-equation__content\"><img src=\"//media.springernature.com/lw457/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Equ1_HTML.gif\" class=\"u-display-block\" alt=\"\"></div></div><p>and for nestling traits:</p><div id=\"Equ2\" class=\"c-article-equation\"><div class=\"c-article-equation__content\"><img src=\"//media.springernature.com/lw457/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Equ2_HTML.gif\" class=\"u-display-block\" alt=\"\"></div></div><p>where <i>V</i><sub>P</sub> is the phenotypic variance after accounting for fixed effects, <i>V</i><sub>A</sub> is the additive genetic variance, <i>V</i><sub>PE</sub> is the permanent environmental effect, <i>V</i><sub>Y</sub> and <i>V</i><sub>BY</sub> are the variance among years and among birth years, respectively, <i>V</i><sub>B</sub> is the variance among broods and <i>V</i><sub>R</sub> the residual variance. All traits followed a Gaussian distribution and animal models were resolved with a restricted maximum likelihood method (REML), using both the social and the genetic pedigree. Final animal models were constructed by sequential model-building from a model with only residual variance to more complex ones, with a comparison of models at each step using a likelihood ratio test (LRT; see <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Tables C1</a> for the increasing levels of complexity). Only <i>V</i><sub>A</sub> in the incubation duration models did not significantly improve the model likelihood (<i>\\xcf\\x87</i><sup>2</sup>&lt;0.01, <i>P</i>&gt;0.99).</p><p>We also constructed three multivariate animal models, one for each trait category. For each of them, we first included the same variance terms as for univariate models. However, due to convergence problems when including <i>V</i><sub>PE</sub> for morphological and reproductive traits, we decided to use a model without the <i>V</i><sub>PE</sub> term on a reduced dataset comprising only one observation by individual (randomly chosen). Moreover, as we observed no <i>V</i><sub>A</sub> for incubation duration, this trait was not included in the multivariate analysis of reproductive traits. Covariances among traits for each variance components were estimated using unstructured variance models. Significance was tested by comparing a model including covariance estimation to a model where covariance was constrained to be equal to zero using LRT.</p><p>We estimated heritability (<i>h</i><sup>2</sup>=<i>V</i><sub>A</sub>/<i>V</i><sub>P</sub>, <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Falconer and Mackay, 1996\" title=\"Falconer DS, Mackay TFC . (1996) Introduction to Quantitative Genetics, 4th edn. Pearson: Essex, UK.\" href=\"/articles/hdy201692#ref-CR8\" id=\"ref-link-section-d40968538e1377\">Falconer and Mackay, 1996</a>) and coefficient of genetic variation (CV<sub>A</sub>=<img src=\"//media.springernature.com/lw89/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq1_HTML.gif\" style=\"width:89px;max-width:none;\" alt=\"\">, where <img src=\"//media.springernature.com/lw11/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq2_HTML.gif\" style=\"width:11px;max-width:none;\" alt=\"\"> is the trait mean, <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Houle, 1992\" title=\"Houle D . (1992). Comparing evolvability and variability of quantitative traits. Genetics 130: 195\\xe2\\x80\\x93204.\" href=\"/articles/hdy201692#ref-CR19\" id=\"ref-link-section-d40968538e1399\">Houle, 1992</a>) for all traits within each analysis. For multivariate analyses we also estimated additive genetic correlations (<i>r</i><sub>A</sub>) between each pair of traits. All animal model analyses were conducted with ASRemL v3.0.5 (VSN International Ltd, Hemel Hempstead, UK). Standard errors (SEs) for variance components and <i>h</i><sup>2</sup> estimates were computed directly by ASRemL. For CV<sub>A</sub>, standard deviations were approximated following the equation provided by Lynch and Walsh <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference (1998\" title=\"Lynch M, Walsh B . (1998) Genetics and Analysis of Quantitative Traits. Sinauer Associates: Sunderland, MA, USA.\" href=\"/articles/hdy201692#ref-CR33\" id=\"ref-link-section-d40968538e1413\">(1998</a>; equation A1.26c, Appendix 1).</p><h3 class=\"c-article__sub-heading\" id=\"Sec6\">Simulation analyses</h3><p>We simulated three different datasets of phenotypic data within both the social and genetic pedigree structures with different levels of complexity. In all cases, simulated traits were normally distributed among all individuals, with <i>V</i><sub>P</sub>=1, <img src=\"//media.springernature.com/lw11/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq3_HTML.gif\" style=\"width:11px;max-width:none;\" alt=\"\">=0 and <i>h</i><sup>2</sup> of 0.1, 0.3 and 0.5 for the focal trait. In dataset 1, we simulated one phenotype per individual with the parameters mentioned above to reflect the simplest scenario possible. In dataset 2, we simulated phenotypic traits with a more complex underlying structure based on equations (1) and (2) to better reflect our empirical dataset, with similar value of <i>h</i><sup><i>2</i></sup> as dataset 1. For nestling traits, we simulated phenotypes with <i>V</i><sub>Y</sub>=0.1 and <i>V</i><sub>B</sub>=0.3, whereas for morphological and reproductive traits we simulated phenotypic values with <i>V</i><sub>Y</sub>=0.1 and <i>V</i><sub>PE</sub>=0.1, thus implying multiple observations per individual. Finally, in dataset 3, different genetic and environmental correlations between traits were also implemented in simulations to assess the difference in power, precision and accuracy when using multivariate models. More specifically, we simulated phenotypes with <i>h</i><sup>2</sup> similar to dataset 1 for the focal trait, whereas fixing <i>h</i><sup>2</sup> of the two other traits at 0.3, for three different values of <i>r</i><sub>A</sub>, 0.1, 0.3 and 0.5, whereas fixing environmental correlation at 0.3. No other variance components were simulated. Simulations were performed in R, and breeding values were simulated with the package <span class=\"u-small-caps\">PEDANTICS</span>. Although simulations were performed within total social and genetic pedigree structures, only individuals with complete information in the empirical dataset were kept in these three simulated datasets.</p><p>Different analyses were performed on simulated datasets to answer two different questions. First, to assess if there was a bias created by not accounting for EPP, we used datasets simulated with the genetic pedigree in animal models using either the genetic pedigree (GG analysis) or the social pedigree (GS analysis), and then the results were compared. Second, to look at inherent differences in reliability caused by pedigree structures, datasets simulated with the social pedigree were analyzed using the social pedigree (SS analysis) and were compared with GG analysis. Animal models used for datasets 1 and 3 included only <i>V</i><sub>A</sub> and <i>V</i><sub>R</sub> as variance components (and covariance in dataset 3), whereas for dataset 2 they included all components described in equations (1) and (2). All these scenarios within each analysis were repeated 100 times for each trait, and animal model analyses were conducted with ASRemL.</p><p>Precision and accuracy for <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> estimates were checked visually with boxplots (that is, median for accuracy and distribution of estimates for precision) and also by computing the mean squared error (MSE) for each scenario. MSE is defined as <img src=\"//media.springernature.com/lw62/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq4_HTML.gif\" style=\"width:62px;max-width:none;\" alt=\"\">, where <i>d</i> is the true value (for example, the simulated parameter of <i>h</i><sup>2</sup> or <i>r</i><sub>A</sub>; these values are highly correlated (<i>r</i>&gt;0.99) with the realized value within our simulated population) and <img src=\"//media.springernature.com/lw7/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq5_HTML.gif\" style=\"width:7px;max-width:none;\" alt=\"\"> is the estimated value; a small MSE indicates high precision and accuracy (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Bolker, 2008\" title=\"Bolker B . (2008) Ecological Models and Data in R. Princeton, University Princeton, NJ, USA.\" href=\"/articles/hdy201692#ref-CR4\" id=\"ref-link-section-d40968538e1532\">Bolker, 2008</a>). Root MSE (RMSE) was used to allow a comparison at the scale of the estimates (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference de Villemereuil et al., 2013\" title=\"de Villemereuil P, Gimenez O, Doligez B . (2013). Comparing parent\\xe2\\x80\\x93offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits. Methods Ecol Evol 4: 260\\xe2\\x80\\x93275.\" href=\"/articles/hdy201692#ref-CR43\" id=\"ref-link-section-d40968538e1536\">de Villemereuil et al., 2013</a>). Power to detect significant <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> within each scenario were estimated using a \\xe2\\x80\\x98rule of thumb\\xe2\\x80\\x99 approach by computing the proportion of estimates that were two times larger than their SE. Although LRTs could have provided a formal hypothesis testing, they were not possible to implement in our simulations procedures. Instead, the \\xe2\\x80\\x98rule of thumb\\xe2\\x80\\x99 approach is a practical indicator of statistical testing that could be easily integrated to simulation analyses and lead to results comparable to a LRT approach (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table A3</a>; see also <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wilson et al., 2011\" title=\"Wilson AJ, Morrissey MB, Adams MJ, Walling CA, Guinness FE, Pemberton JM et al. (2011). Indirect genetics effects and evolutionary constraint: an analysis of social dominance in red deer, Cervus elaphus. J Evol Biol 24: 772\\xe2\\x80\\x93783.\" href=\"/articles/hdy201692#ref-CR47\" id=\"ref-link-section-d40968538e1550\">Wilson et al., 2011</a> for a similar approach). Finally, we also checked the identifiability of variance terms in dataset 2 using a Spearman\\xe2\\x80\\x99s rank correlation between estimates of phenotypic variance components. In this case, a low correlation between two components would indicate a high identifiability, whereas a high correlation would indicate a low identifiability.</p></div></div></section><section data-title=\"Results\"><div class=\"c-article-section\" id=\"Sec7-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec7\">Results</h2><div class=\"c-article-section__content\" id=\"Sec7-content\"><h3 class=\"c-article__sub-heading\" id=\"Sec8\">Pedigrees</h3><p>Our total dataset comprised 13\\xe2\\x80\\x89446 individuals, with 2839 individuals observed only as adults, 10\\xe2\\x80\\x89472 observed only as nestlings and 135 observed at both states (that is, recruits). From the field observations, we were able to determine the identity of the social father for 69.2% of nestlings (82.2% since 2006). Parental analyses allowed us to determine the genetic father for 53.3% of nestling (66.3% since 2006). On average, 1.74\\xc2\\xb10.82 (range: 1\\xe2\\x88\\x925) genetic fathers were assigned per brood, and each adult male had on average 4.81\\xc2\\xb15.44 (range: 0\\xe2\\x88\\x9237) nestlings assigned over its entire lifespan. Overall, 49.3% of nestlings successfully genotyped with a known social father were extra-pair young (<i>N</i><sub>total</sub>=6382).</p><p>Pedigree size varied among the type of pedigree and traits (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"table anchor\" href=\"/articles/hdy201692#Tab1\">Table 1</a>). As expected, the social pedigree always had a higher number of links related to father identities than its genetic counterpart, a difference mainly caused by unsampled males contributing to the genetic pool of nestlings. Nestling traits had a much higher number of links than morphological and reproductive traits, which were both of similar order of magnitude, but with slightly fewer links for the latter (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"table anchor\" href=\"/articles/hdy201692#Tab1\">Table 1</a>). Pedigree sizes were similar for each trait within a given category (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table A2</a>).</p><h3 class=\"c-article__sub-heading\" id=\"Sec9\">Estimations of quantitative genetic parameters (empirical data)</h3><p>Variance components, and resulting heritability values, estimated from empirical data varied among traits and also often importantly between pedigrees (social versus genetic) and between models (univariate versus multivariate) (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"figure anchor\" href=\"/articles/hdy201692#Fig1\">Figure 1</a>; <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"table anchor\" href=\"/articles/hdy201692#Tab2\">Table 2</a>). Biases in <i>h</i><sup>2</sup> and CV<sub>A</sub> resulting from using social instead of genetic pedigrees were moderate for morphological traits (<i>h</i><sup>2</sup>: \\xe2\\x88\\x920.21 to \\xe2\\x88\\x920.02; CV<sub>A</sub>: \\xe2\\x88\\x920.008 to \\xe2\\x88\\x920.001), almost null for reproductive traits (almost all around 0) and highly variable for nestling traits (<i>h</i><sup>2</sup>: \\xe2\\x88\\x920.04\\xe2\\x80\\x930.21; CV<sub>A</sub>: \\xe2\\x88\\x920.001\\xe2\\x80\\x930.051). Estimates of <i>r</i><sub>A</sub> obtained with the social and genetic pedigree were similar, except between nestling wing length and body mass where <i>r</i><sub>A</sub> was positive and significantly different from 0 with the social pedigree (<i>r</i><sub>A</sub>=0.56\\xc2\\xb10.13, <i>\\xcf\\x87</i><sup>2</sup>=9.92, <i>P</i>=0.002), but not with the genetic pedigree (<i>r</i><sub>A</sub>=0.25\\xc2\\xb10.19, \\xcf\\x87<sup>2</sup>=1.32, <i>P</i>=0.25; <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure C1</a>). Raw phenotypic variance for all traits can be found in <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table C4</a>.</p><div class=\"c-article-section__figure js-c-reading-companion-figures-item\" data-test=\"figure\" data-container-section=\"figure\" id=\"figure-1\" data-title=\"Figure 1\"><figure><figcaption><b id=\"Fig1\" class=\"c-article-section__figure-caption\" data-test=\"figure-caption-text\">Figure 1</b></figcaption><div class=\"c-article-section__figure-content\"><div class=\"c-article-section__figure-item\"><a class=\"c-article-section__figure-link\" data-test=\"img-link\" data-track=\"click\" data-track-label=\"image\" data-track-action=\"view figure\" href=\"/articles/hdy201692/figures/1\" rel=\"nofollow\"><picture><source type=\"image/webp\" srcset=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig1_HTML.jpg?as=webp\"><img aria-describedby=\"Fig1\" src=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig1_HTML.jpg\" alt=\"figure 1\" loading=\"lazy\" width=\"560\" height=\"705\"></picture></a></div><div class=\"c-article-section__figure-description\" data-test=\"bottom-caption\" id=\"figure-1-desc\"><p>Proportion of phenotypic variance estimated from final animal models on empirical dataset, for (<b>a</b>) morphological, (<b>b</b>) reproductive and (<b>c</b>) nestling traits. Different models were assessed for each trait: univariate models using social (SU) and genetic (GU) pedigree, and multivariate models using social (SM) and genetic (GM) pedigree.</p></div></div><div class=\"u-text-right u-hide-print\"><a class=\"c-article__pill-button\" data-test=\"article-link\" data-track=\"click\" data-track-label=\"button\" data-track-action=\"view figure\" href=\"/articles/hdy201692/figures/1\" data-track-dest=\"link:Figure1 Full size image\" aria-label=\"Full size image figure 1\" rel=\"nofollow\"><span>Full size image</span><svg width=\"16\" height=\"16\" focusable=\"false\" role=\"img\" aria-hidden=\"true\" class=\"u-icon\"><use xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"#global-icon-chevron-right\"></use></svg></a></div></figure></div><div class=\"c-article-table\" data-test=\"inline-table\" data-container-section=\"table\" id=\"table-2\"><figure><figcaption class=\"c-article-table__figcaption\"><b id=\"Tab2\" data-test=\"table-caption\">Table 2 Summary of final A) univariate and B) multivariate animal models using empirical datasets and either the social pedigree or the genetic pedigree</b></figcaption><div class=\"u-text-right u-hide-print\"><a class=\"c-article__pill-button\" data-test=\"table-link\" data-track=\"click\" data-track-action=\"view table\" data-track-label=\"button\" rel=\"nofollow\" href=\"/articles/hdy201692/tables/2\" aria-label=\"Full size table 2\"><span>Full size table</span><svg width=\"16\" height=\"16\" focusable=\"false\" role=\"img\" aria-hidden=\"true\" class=\"u-icon\"><use xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"#global-icon-chevron-right\"></use></svg></a></div></figure></div><h3 class=\"c-article__sub-heading\" id=\"Sec10\">Reliability of quantitative genetics estimates (simulated data)</h3><p>Visual inspection of boxplots (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"figure anchor\" href=\"/articles/hdy201692#Fig2\">Figures 2b, d, f</a> and <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"figure anchor\" href=\"/articles/hdy201692#Fig3\">3b</a>) and comparison of RMSE values (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figures D1</a>) showed complementary information on precision and accuracy of <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> estimates. Precision of <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> estimates varied greatly depending on trait category (nestling&gt;morphological&gt;reproductive; mean\\xc2\\xb1s.d. RMSE for <i>h</i><sup>2</sup>/<i>r</i><sub>A</sub>: <img src=\"//media.springernature.com/lw46/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq6_HTML.gif\" style=\"width:46px;max-width:none;\" alt=\"\">=0.055\\xc2\\xb10.045/0.081\\xc2\\xb10.025, <img src=\"//media.springernature.com/lw45/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq7_HTML.gif\" style=\"width:45px;max-width:none;\" alt=\"\">=0.122\\xc2\\xb10.026/0.342\\xc2\\xb10.061, <img src=\"//media.springernature.com/lw33/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq8_HTML.gif\" style=\"width:33px;max-width:none;\" alt=\"\">=0.200\\xc2\\xb10.045/0.430\\xc2\\xb10.025) but also among datasets used (dataset 2 (simulation of <i>h</i><sup>2</sup> and other source of resemblance among individuals)&gt;3 (simulation of <i>h</i><sup>2</sup> and <i>r</i><sub><i>A</i></sub>)&gt;1 (simulation of <i>h</i><sup>2</sup> only); mean\\xc2\\xb1s.d. RMSE for <i>h</i><sup>2</sup>: <img src=\"//media.springernature.com/lw49/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq9_HTML.gif\" style=\"width:49px;max-width:none;\" alt=\"\">=0.144\\xc2\\xb10.086, <img src=\"//media.springernature.com/lw45/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq10_HTML.gif\" style=\"width:45px;max-width:none;\" alt=\"\">=0.100\\xc2\\xb10.047, <img src=\"//media.springernature.com/lw45/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq11_HTML.gif\" style=\"width:45px;max-width:none;\" alt=\"\">=0.127\\xc2\\xb10.072). Accuracy for <i>h</i><sup>2</sup> estimates was high for all GG and SS analyses (except for reproductive traits in dataset 3), but downwardly biased in GS analyses for nestling and reproductive traits, an effect more important as simulated <i>h</i><sup>2</sup> increased (mean\\xc2\\xb1s.d. RMSE: <img src=\"//media.springernature.com/lw27/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq12_HTML.gif\" style=\"width:27px;max-width:none;\" alt=\"\">=0.117\\xc2\\xb10.076, <img src=\"//media.springernature.com/lw24/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq13_HTML.gif\" style=\"width:24px;max-width:none;\" alt=\"\">=0.143\\xc2\\xb10.061, <img src=\"//media.springernature.com/lw31/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq14_HTML.gif\" style=\"width:31px;max-width:none;\" alt=\"\">=0.115\\xc2\\xb10.075). Accuracy for <i>r</i><sub>A</sub> were similar for all type of analysis (mean\\xc2\\xb1s.d. RMSE: <img src=\"//media.springernature.com/lw31/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq15_HTML.gif\" style=\"width:31px;max-width:none;\" alt=\"\">=0.283\\xc2\\xb10.160, <img src=\"//media.springernature.com/lw29/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq16_HTML.gif\" style=\"width:29px;max-width:none;\" alt=\"\">=0.294\\xc2\\xb10.151, <img src=\"//media.springernature.com/lw21/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq17_HTML.gif\" style=\"width:21px;max-width:none;\" alt=\"\">=0.275\\xc2\\xb10.158).</p><div class=\"c-article-section__figure js-c-reading-companion-figures-item\" data-test=\"figure\" data-container-section=\"figure\" id=\"figure-2\" data-title=\"Figure 2\"><figure><figcaption><b id=\"Fig2\" class=\"c-article-section__figure-caption\" data-test=\"figure-caption-text\">Figure 2</b></figcaption><div class=\"c-article-section__figure-content\"><div class=\"c-article-section__figure-item\"><a class=\"c-article-section__figure-link\" data-test=\"img-link\" data-track=\"click\" data-track-label=\"image\" data-track-action=\"view figure\" href=\"/articles/hdy201692/figures/2\" rel=\"nofollow\"><picture><source type=\"image/webp\" srcset=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig2_HTML.jpg?as=webp\"><img aria-describedby=\"Fig2\" src=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig2_HTML.jpg\" alt=\"figure 2\" loading=\"lazy\" width=\"685\" height=\"871\"></picture></a></div><div class=\"c-article-section__figure-description\" data-test=\"bottom-caption\" id=\"figure-2-desc\"><p>Power, precision and accuracy of heritability estimated from animal models for dataset 1 (<b>a</b> and <b>b</b>; simulation of <i>h</i><sup>2</sup> only), dataset 2 (<b>c</b> and <b>d</b>; simulation of <i>h</i><sup>2</sup> and other source of resemblance among individuals) and dataset 3 (<b>e</b> and <b>f</b>; simulation of <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub>) simulated on the pedigree structure of morphological traits (dark gray lines), reproductive traits (light gray lines) and nestling traits (black lines). Power of detecting significant heritability over 300 simulated trait values (100 simulations per trait) are presented in <b>a</b>, <b>c</b> and <b>e</b>. Distributions of these estimates are represented by boxplots (1st quartile, median, 3rd quartile) for 2 levels of heritability (dotted lines represent the <i>h</i><sup>2</sup> simulated value of 0.1 and 0.5) in <b>b</b>, <b>d</b> and <b>f</b>. Analysis types refer to datasets simulated with the genetic pedigree and analyzed with the genetic (GG analysis, solid lines) or the social pedigree (GS analysis, dashed lines), and datasets simulated with the social pedigree and analyzed with the social pedigree (SS analysis, dotted lines). For graphical representation, <i>r</i><sub>A</sub> was fixed at 0.3 in <b>e</b> and <b>f</b>.</p></div></div><div class=\"u-text-right u-hide-print\"><a class=\"c-article__pill-button\" data-test=\"article-link\" data-track=\"click\" data-track-label=\"button\" data-track-action=\"view figure\" href=\"/articles/hdy201692/figures/2\" data-track-dest=\"link:Figure2 Full size image\" aria-label=\"Full size image figure 2\" rel=\"nofollow\"><span>Full size image</span><svg width=\"16\" height=\"16\" focusable=\"false\" role=\"img\" aria-hidden=\"true\" class=\"u-icon\"><use xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"#global-icon-chevron-right\"></use></svg></a></div></figure></div><div class=\"c-article-section__figure js-c-reading-companion-figures-item\" data-test=\"figure\" data-container-section=\"figure\" id=\"figure-3\" data-title=\"Figure 3\"><figure><figcaption><b id=\"Fig3\" class=\"c-article-section__figure-caption\" data-test=\"figure-caption-text\">Figure 3</b></figcaption><div class=\"c-article-section__figure-content\"><div class=\"c-article-section__figure-item\"><a class=\"c-article-section__figure-link\" data-test=\"img-link\" data-track=\"click\" data-track-label=\"image\" data-track-action=\"view figure\" href=\"/articles/hdy201692/figures/3\" rel=\"nofollow\"><picture><source type=\"image/webp\" srcset=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig3_HTML.jpg?as=webp\"><img aria-describedby=\"Fig3\" src=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig3_HTML.jpg\" alt=\"figure 3\" loading=\"lazy\" width=\"685\" height=\"292\"></picture></a></div><div class=\"c-article-section__figure-description\" data-test=\"bottom-caption\" id=\"figure-3-desc\"><p>Power, precision and accuracy of genetic additive correlation estimated from multivariate animal models, for dataset 3 (simulation of <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub>) simulated on the pedigree structure of morphological traits (dark gray lines), reproductive traits (light gray lines) and nestling traits (black lines). Power of detecting significant genetic correlation over 300 simulated trait values (100 simulations per trait), is presented in <b>a</b> and distributions of these estimates are represented by boxplots (1st quartile, median, 3rd quartile) in <b>b</b> (for 1 value of <i>r</i><sub>A</sub>\\xe2\\x80\\x94dotted lines represent the <i>r</i><sub>A</sub> true value of 0.5 simulated; 5 estimates&gt;1 are not presented). Analysis types refer to datasets simulated with the genetic pedigree and analyzed with the genetic (GG analysis, solid lines) or the social pedigree (GS analysis, dashed lines), and datasets simulated with the social pedigree and analyzed with the social pedigree (SS analysis, dotted lines). For graphical representation, <i>h</i><sup>2</sup> was fixed at 0.5 in <b>b</b>.</p></div></div><div class=\"u-text-right u-hide-print\"><a class=\"c-article__pill-button\" data-test=\"article-link\" data-track=\"click\" data-track-label=\"button\" data-track-action=\"view figure\" href=\"/articles/hdy201692/figures/3\" data-track-dest=\"link:Figure3 Full size image\" aria-label=\"Full size image figure 3\" rel=\"nofollow\"><span>Full size image</span><svg width=\"16\" height=\"16\" focusable=\"false\" role=\"img\" aria-hidden=\"true\" class=\"u-icon\"><use xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"#global-icon-chevron-right\"></use></svg></a></div></figure></div><p>Power to detect significant <i>h</i><sup>2</sup> or <i>r</i><sub>A</sub> was greater for traits with larger sample size (nestling&gt;morphological&gt;reproductive; <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"figure anchor\" href=\"/articles/hdy201692#Fig2\">Figures 2a, c, e</a> and <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"figure anchor\" href=\"/articles/hdy201692#Fig3\">3a</a>), but we observed no difference among traits within each category (data not shown). For multivariate animal models from dataset 3, power to detect <i>h</i><sup>2</sup> was similar for the different genetic correlations tested (data not shown), whereas power to detect genetic correlation increased with increasing heritability (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"figure anchor\" href=\"/articles/hdy201692#Fig3\">Figure 3a</a>). Within a trait category, power was similar between GG and SS analyses, except for nestling traits from dataset 2 were the simulated brood effect decreased power in SS analysis (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"figure anchor\" href=\"/articles/hdy201692#Fig2\">Figure 2c</a>). Finally, power comparison between GG and GS analyses revealed generally lower power in morphological and nestling trait for GS (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"figure anchor\" href=\"/articles/hdy201692#Fig2\">Figures 2a, c, e</a> and <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"figure anchor\" href=\"/articles/hdy201692#Fig3\">3a</a>).</p><p>For adult traits, <i>V</i><sub>A</sub> and <i>V</i><sub>PE</sub> estimates from dataset 2 were highly negatively correlated, suggesting that these terms were almost completely confounded (morphological traits: <i>r</i><sub>s</sub>=\\xe2\\x88\\x920.95, <i>P</i>&lt;0.001; reproductive traits: <i>r</i><sub>s</sub>=\\xe2\\x88\\x920.94, <i>P</i>&lt;0.001, results from GG analysis with <i>h</i><sup>2</sup>=0.3; <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figures D3</a>). For nestling traits, <i>V</i><sub>A</sub> and <i>V</i><sub>B</sub> estimates were also correlated, but to a lesser extent (<i>r</i><sub>s</sub>=\\xe2\\x88\\x920.42, <i>P</i>&lt;0.001; <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure D5</a>). All other variance terms showed correlations non-significantly different from 0 (all |<i>r</i><sub>s</sub>|&lt;0.09, <i>P</i>&gt;0.12).</p><p>Finally, given the large differences observed for some reproductive traits between SEs of <i>h</i><sup>2</sup> estimates from empirical and simulated dataset (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"table anchor\" href=\"/articles/hdy201692#Tab2\">Table 2</a>; <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure D6a</a>), we further inspected the impact of model parameters on SEs. More specifically, we assessed the impact of not accounting for an important cause of resemblance among individuals by running dataset 2 animal models and omitting <i>V</i><sub>PE</sub> or <i>V</i><sub>B</sub> components. By doing so, we observed a power of 1 at all traits, and all <i>h</i><sup>2</sup> estimates were very precise but showed a bias equivalent to the variance component not accounted for (that is, 0.1 for morphological and reproductive trait and 0.4 for nestling traits; <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"figure anchor\" href=\"/articles/hdy201692#Fig4\">Figure 4</a>). As suspected, SEs for these biased <i>h</i><sup>2</sup> estimates were small and similar to those obtained in empirical analyses, where <i>V</i><sub>PE</sub> was almost completely confounded with <i>V</i><sub>A</sub> (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure D6b</a>).</p><div class=\"c-article-section__figure js-c-reading-companion-figures-item\" data-test=\"figure\" data-container-section=\"figure\" id=\"figure-4\" data-title=\"Figure 4\"><figure><figcaption><b id=\"Fig4\" class=\"c-article-section__figure-caption\" data-test=\"figure-caption-text\">Figure 4</b></figcaption><div class=\"c-article-section__figure-content\"><div class=\"c-article-section__figure-item\"><a class=\"c-article-section__figure-link\" data-test=\"img-link\" data-track=\"click\" data-track-label=\"image\" data-track-action=\"view figure\" href=\"/articles/hdy201692/figures/4\" rel=\"nofollow\"><picture><source type=\"image/webp\" srcset=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig4_HTML.jpg?as=webp\"><img aria-describedby=\"Fig4\" src=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig4_HTML.jpg\" alt=\"figure 4\" loading=\"lazy\" width=\"556\" height=\"340\"></picture></a></div><div class=\"c-article-section__figure-description\" data-test=\"bottom-caption\" id=\"figure-4-desc\"><p>Precision and accuracy of heritability estimated from animal models omitting an important cause of resemblance among individuals (<i>V</i><sub>PE</sub> or <i>V</i><sub>B</sub>), for morphological traits, reproductive traits and nestling traits simulated in dataset 2. Distribution of these estimates are represented by boxplots (1st quartile, median, 3rd quartile) for 2 levels of heritability (dotted lines represent the <i>h</i><sup>2</sup> true value of 0.1 and 0.5 simulated). Analysis types refer to datasets simulated with the genetic pedigree and analyzed with both the genetic and the social pedigree (GG and GS analyses, respectively), and datasets simulated with the social pedigree and analyzed with the social pedigree (SS analysis).</p></div></div><div class=\"u-text-right u-hide-print\"><a class=\"c-article__pill-button\" data-test=\"article-link\" data-track=\"click\" data-track-label=\"button\" data-track-action=\"view figure\" href=\"/articles/hdy201692/figures/4\" data-track-dest=\"link:Figure4 Full size image\" aria-label=\"Full size image figure 4\" rel=\"nofollow\"><span>Full size image</span><svg width=\"16\" height=\"16\" focusable=\"false\" role=\"img\" aria-hidden=\"true\" class=\"u-icon\"><use xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"#global-icon-chevron-right\"></use></svg></a></div></figure></div></div></div></section><section data-title=\"Discussion\"><div class=\"c-article-section\" id=\"Sec11-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec11\">Discussion</h2><div class=\"c-article-section__content\" id=\"Sec11-content\"><p>Despite the importance of obtaining precise and accurate genetic (co)variance estimates when assessing a population\\xe2\\x80\\x99s adaptive potential, their reliability is rarely explicitly tested or reported in wild populations. Here, we formally assessed the reliability of genetic (co)variance estimates, in a species with a high rate of extra-pair reproduction and low recruitment, with a combination of empirical and simulated data. Altogether, our simulation analyses emphasized the limits of this particular dataset by revealing an important lack of precision in <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> estimates for all adult traits, a lack of power to detect significant effects and identifiability problems between <i>V</i><sub>A</sub> and V<sub>PE</sub>. Moreover, we observed a large bias in <i>h</i><sup><i>2</i></sup> when using the social pedigree instead of the genetic one, and also when non-genetic causes of resemblance among individuals (that is, repeated measurements or brood effects) were not accounted for in our analyses. We briefly discuss below (i) the factors contributing to estimate reliability, (ii) the caution needed with model parameterization, (iii) the impacts of high levels of EPP on genetic variance estimates and (iv) we finally conclude with remarks on the accessibility and utility of simulation analyses to address all these potential problems.</p><h3 class=\"c-article__sub-heading\" id=\"Sec12\">Estimate reliability and generalization among studies</h3><p>At first glance, our sample size seemed large enough to be powerful, with more than 10 years of sampling and a number of records within our pruned pedigrees that was larger than the median number of records typically reported for similar studies in the literature. Indeed, a recent compilation of quantitative genetics estimates obtained from wild populations by <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Postma (2014)\" title=\"Postma E . (2014). Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates? In: Charmantier A, Garant D et al. (eds) Quantitative Genetics in the Wild. Oxford University Press, Oxford, UK, pp 16\\xe2\\x80\\x9333.\" href=\"/articles/hdy201692#ref-CR39\" id=\"ref-link-section-d40968538e3130\">Postma (2014)</a> showed that for estimates obtained with animal models, the median number of records for life history traits was 377 (range 6\\xe2\\x80\\x934992; from 39 studies, covering 19 species) and 363 for morphological traits (range 50\\xe2\\x80\\x9338\\xe2\\x80\\x89024; from 47 studies, covering 22 species). This suggests that sample size alone is not sufficient to infer pedigree quality, as it does not reflect the connectivity of pedigrees (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wilson et al., 2010\" title=\"Wilson AJ, R\\xc3\\xa9ale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist\\xe2\\x80\\x99s guide to the animal model. J Anim Ecol 79: 13\\xe2\\x80\\x9326.\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d40968538e3133\">Wilson et al., 2010</a>). In our study system, the low recruitment of nestlings (1.3%) results in very few grandparent links within our pedigree, which greatly reduces its power. However, this problem is far less important for nestling traits as the high-EPP rate results in a genetic pedigree containing several half-sib families, which increase its power (see GG versus SS, <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"figure anchor\" href=\"/articles/hdy201692#Fig2\">Figure 2c</a>) despite a smaller number of observations compared with the social pedigree for all categories (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"table anchor\" href=\"/articles/hdy201692#Tab1\">Table 1</a>). Thus, given that pedigree structures are different from one study system to another, their impact on reliability of quantitative genetic estimates is best predicted from simulation analyses.</p><p>Various ecological and population specific factors could generate fluctuations in the precision and accuracy of quantitative genetic estimates among study systems. Here, our study system was characterized by a high rate of EPP and a low recruitment, two characteristics that affected the reliability of quantitative genetic estimates for some trait types. Although these characteristics are certainly common in other migratory species (for example, Common reed bunting (<i>Emberiza schoeniclus</i>) and Red-eyed vireo (<i>Vireo olivaceus</i>), <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Griffith et al., 2002\" title=\"Griffith SC, Owens IPF, Thuman KA . (2002). Extra pair paternity in birds: a review of interspecific variation and adaptive function. Mol Ecol 11: 2195\\xe2\\x80\\x932212.\" href=\"/articles/hdy201692#ref-CR13\" id=\"ref-link-section-d40968538e3151\">Griffith et al., 2002</a>), other characteristics, such as low-recapture rate between years in open study systems, a sex bias in sampling or large natal dispersal may also have important consequences on the precision of estimates. Moreover, differential mortality or capture rate might bias the sampling of some traits, and consequently affect the accuracy of estimates that can be obtained (for example, <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Dingemanse et al., 2009\" title=\"Dingemanse NJ, Kazem AJN, R\\xc3\\xa9ale D, Wright J . (2009). Behavioural reaction norms: animal personality meets individual plasticity. Trends Ecol Evol 25: 81\\xe2\\x80\\x9389.\" href=\"/articles/hdy201692#ref-CR6\" id=\"ref-link-section-d40968538e3154\">Dingemanse et al., 2009</a>). It is thus important to appropriately identify the limits in a particular system and assess their impacts on the reliability of quantitative genetic estimates by using simulations.</p><h3 class=\"c-article__sub-heading\" id=\"Sec13\">Caution needed with model parameterization</h3><p>The choice of which variance terms to include in a model is a crucial step that can have substantial impacts on reliability of genetic variance estimates (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Kruuk and Hadfield, 2007\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890\\xe2\\x80\\x931903.\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d40968538e3165\">Kruuk and Hadfield, 2007</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wilson, 2008\" title=\"Wilson AJ . (2008). Why h2 does not always equal VA/VP? J Evol Biol 21: 647\\xe2\\x80\\x93650.\" href=\"/articles/hdy201692#ref-CR46\" id=\"ref-link-section-d40968538e3168\">Wilson, 2008</a>). Here, the decision to include or not a particular variance term relied on LRTs, which is a widely used approach (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wilson et al., 2010\" title=\"Wilson AJ, R\\xc3\\xa9ale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist\\xe2\\x80\\x99s guide to the animal model. J Anim Ecol 79: 13\\xe2\\x80\\x9326.\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d40968538e3171\">Wilson et al., 2010</a>). However, in some cases even if the inclusion of a particular component is not improving the model likelihood, it may still have to be included (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wilson et al., 2010\" title=\"Wilson AJ, R\\xc3\\xa9ale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist\\xe2\\x80\\x99s guide to the animal model. J Anim Ecol 79: 13\\xe2\\x80\\x9326.\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d40968538e3174\">Wilson et al., 2010</a>). This is the case for component of variance attributable to repeated measurement of a given individual (<i>V</i><sub>PE</sub>). In our models, we had to account for multiple observations by fitting a <i>V</i><sub>PE</sub> term in our model, but this component was almost completely confounded with <i>V</i><sub>A</sub>. This situation probably occurred because of the low number of observations per individual (mean of 1.5 (averaged across traits; range: 1\\xe2\\x80\\x937) for morphological traits and 1.2 (range: 1\\xe2\\x80\\x937) for reproductive traits), a situation that should be present in other short-lived species datasets. Not accounting for repeated measurements in an animal model is equivalent to considering each repeated measurement as a clone of the same individual, which is inappropriate. In such cases, using a dataset with only one observation per individual may lead to more accurate estimates and easier model convergence.</p><p>Standard errors are often viewed as predictors of an estimate\\xe2\\x80\\x99s accuracy or significance, but this can be misleading (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Krzywinski and Altman, 2013\" title=\"Krzywinski M, Altman N . (2013). Points of significance: error bars. Nat Methods 10: 921\\xe2\\x80\\x93922.\" href=\"/articles/hdy201692#ref-CR27\" id=\"ref-link-section-d40968538e3193\">Krzywinski and Altman, 2013</a>). Estimates from animal models have generally smaller standard errors than those obtained with parent\\xe2\\x80\\x93offspring regressions (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Kruuk, 2004\" title=\"Kruuk LEB . (2004). Estimating genetic parameters in natural populations using the \\xe2\\x80\\x98animal model\\xe2\\x80\\x99. Philos Trans R Soc B 359: 873\\xe2\\x80\\x93890.\" href=\"/articles/hdy201692#ref-CR24\" id=\"ref-link-section-d40968538e3196\">Kruuk, 2004</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Postma, 2014\" title=\"Postma E . (2014). Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates? In: Charmantier A, Garant D et al. (eds) Quantitative Genetics in the Wild. Oxford University Press, Oxford, UK, pp 16\\xe2\\x80\\x9333.\" href=\"/articles/hdy201692#ref-CR39\" id=\"ref-link-section-d40968538e3199\">Postma, 2014</a>), partly due to their integration of multiple observations for a given individual (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference \\xc3\\x85kesson et al., 2008\" title=\"\\xc3\\x85kesson M, Bensch S, Hasselquist D, Tarka M, Hansson B, Akesson M . (2008). Estimating heritabilities and genetic correlations: comparing the \\xe2\\x80\\x98animal model\\xe2\\x80\\x99 with parent\\xe2\\x80\\x93offspring regression using data from a natural population. PLoS One 3: e1739.\" href=\"/articles/hdy201692#ref-CR1\" id=\"ref-link-section-d40968538e3202\">\\xc3\\x85kesson et al., 2008</a>). As previously stated, failing to account for multiple observations can create unpredictable bias in phenotypic variance component estimates (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Kruuk and Hadfield, 2007\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890\\xe2\\x80\\x931903.\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d40968538e3205\">Kruuk and Hadfield, 2007</a>) and considerably reduce standard errors around these biased estimates (<a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure D6</a>). Moreover, it seems that problems of identifiability could also result in reduced standard errors. For example, in our empirical analyses, small standard errors were estimated around <i>h</i><sup>2</sup> estimates for reproductive traits from univariate animal models with repeated measurements per individual (SE range: 0.03\\xe2\\x80\\x930.04), which could have led us to misleadingly conclude that our dataset was powerful and our estimates were precise for these traits.</p><p>This study was performed with a frequentist method (that is, REML) rather than a Bayesian method (that is, MCMC) because it generally gives more precise and accurate estimates for traits following a Gaussian distribution and it necessitates a shorter computation time (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference de Villemereuil et al., 2013\" title=\"de Villemereuil P, Gimenez O, Doligez B . (2013). Comparing parent\\xe2\\x80\\x93offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits. Methods Ecol Evol 4: 260\\xe2\\x80\\x93275.\" href=\"/articles/hdy201692#ref-CR43\" id=\"ref-link-section-d40968538e3219\">de Villemereuil et al., 2013</a>). Bayesian methods have other advantages, such as the use of posterior distributions for the calculation of derived parameters (for example, heritability) and associated standard errors and they are particularly useful for analyses of non-Gaussian traits (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Morrissey et al., 2014\" title=\"Morrissey MB, de Villemereuil P, Doligez B, Gimenez O . (2014). Bayesian approaches to the quantitative genetic analysis of natural populations. In: Charmantier A, Garant D et al. (eds) Quantitative Genetics in the Wild. Oxford University Press, Oxford, UK, pp 228\\xe2\\x80\\x93253.\" href=\"/articles/hdy201692#ref-CR36\" id=\"ref-link-section-d40968538e3222\">Morrissey et al., 2014</a>). In the present context, problems related to missing of an important source of resemblance among individuals are not method specific and could impact any model regardless of the method used for its estimation. However, it is less clear if identifiability problems could be better uncovered with Bayesian approach, for example through an appropriate MCMC checking. Simulation studies such as this one could be applied to animal models resolved with a Bayesian approach to better assess its limits.</p><h3 class=\"c-article__sub-heading\" id=\"Sec14\">Impact of high level of EPP on estimates</h3><p>In theory, EPPs, if not accounted for, could downwardly bias additive genetic variance and resulting heritability estimates (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Charmantier and R\\xc3\\xa9ale, 2005\" title=\"Charmantier A, R\\xc3\\xa9ale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839\\xe2\\x80\\x932850.\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d40968538e3233\">Charmantier and R\\xc3\\xa9ale, 2005</a>). Yet, previous simulation studies showed that even if biases increased with the importance of EPPs, with increasing heritability and when focal traits were directly related to the number of extra-pair young produced, underestimations were generally smaller than 15% (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Firth et al., 2015\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336\\xe2\\x80\\x931344.\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d40968538e3236\">Firth et al., 2015</a>). In our study, the rate of EPP was higher than those previously tested so far (up to 40% in <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Charmantier and R\\xc3\\xa9ale, 2005\" title=\"Charmantier A, R\\xc3\\xa9ale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839\\xe2\\x80\\x932850.\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d40968538e3239\">Charmantier and R\\xc3\\xa9ale, 2005</a>, 12.5% in <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Firth et al., 2015\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336\\xe2\\x80\\x931344.\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d40968538e3242\">Firth et al., 2015</a>), but its effect on quantitative genetics estimates was complex. First, our simulation analyses showed that the bias in <i>h</i><sup>2</sup> estimates resulting from using the social instead of the genetic pedigree changed depending on the trait category (that is, higher for nestling traits). Also, our empirical analyses suggested that the impact of not accounting for EPP was noticeable for morphological and nestling traits, but sometimes resulted in higher <i>h</i><sup>2</sup> and CV<sub>A</sub> when using the social pedigree (that is, for nestling wing length and body mass). A similar unexpected finding was previously reported in blue tits (<i>Cyanistes caeruleus</i>), where <i>h</i><sup>2</sup> estimates from a social pedigree were sometimes higher than those from a genetic pedigree (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Charmantier and R\\xc3\\xa9ale, 2005\" title=\"Charmantier A, R\\xc3\\xa9ale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839\\xe2\\x80\\x932850.\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d40968538e3263\">Charmantier and R\\xc3\\xa9ale, 2005</a>). These positive biases observed in empirical data could be due to the influence of the social father (for example through parental care), which could be captured in <i>V</i><sub>A</sub> components (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Griffith et al., 1999\" title=\"Griffith SC, Owens IPF, Burke T . (1999). Environmental determination of a sexually selected trait. Science 400: 358\\xe2\\x80\\x93360.\" href=\"/articles/hdy201692#ref-CR12\" id=\"ref-link-section-d40968538e3271\">Griffith et al., 1999</a>; <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Charmantier and R\\xc3\\xa9ale, 2005\" title=\"Charmantier A, R\\xc3\\xa9ale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839\\xe2\\x80\\x932850.\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d40968538e3274\">Charmantier and R\\xc3\\xa9ale, 2005</a>) rather than due to pedigree errors. To assess this possible problem, we performed an additional animal model analysis including mother and social father identities as additional variance components. We found that new <i>h</i><sup>2</sup> and CV<sub>A</sub> values obtained using the social pedigree were smaller than values obtained with the genetic one (change in estimates for body mass <i>h</i><sup>2</sup>=\\xe2\\x88\\x920.06, CV<sub>A</sub>=\\xe2\\x88\\x920.011; wing length <i>h</i><sup>2</sup>=\\xe2\\x88\\x920.04, CV<sub>A</sub>=\\xe2\\x88\\x920.041; difference calculated with univariate animal models; see <a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table C5</a> for details). This further emphasizes the importance of model parameterization on reliability of variance component estimates. Finally, even if the genetic pedigree better represents relationships among individuals than the social pedigree, it probably still contains a few incorrect links among individuals and thus the estimated heritabilities might also be slightly downwardly biased.</p></div></div></section><section data-title=\"Conclusion\"><div class=\"c-article-section\" id=\"Sec15-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec15\">Conclusion</h2><div class=\"c-article-section__content\" id=\"Sec15-content\"><p>Simulation analyses are now widely accessible to anyone with minimal programming skills in R, for instance by using the \\xe2\\x80\\x98<i>phensim</i>\\xe2\\x80\\x99 function within the package <span class=\"u-small-caps\">PEDANTICS</span>, the \\xe2\\x80\\x98<i>rbv</i>\\xe2\\x80\\x99 function within the package MCMCglmm (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Hadfield, 2010\" title=\"Hadfield JD . (2010). MCMC methods for multi-response generalized linear mixed models: the MCMCglmm R package. J Stat Softw 33: 1\\xe2\\x80\\x9322.\" href=\"/articles/hdy201692#ref-CR14\" id=\"ref-link-section-d40968538e3317\">Hadfield, 2010</a>) or the \\xe2\\x80\\x98<i>drfx</i>\\xe2\\x80\\x99 and the \\xe2\\x80\\x98<i>grfx</i>\\xe2\\x80\\x99 function within the package <span class=\"u-small-caps\">NADIV</span> (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wolak, 2012\" title=\"Wolak ME . (2012). nadiv : an R package to create relatedness matrices for estimating non-additive genetic variances in animal models. Methods Ecol Evol 3: 792\\xe2\\x80\\x93796.\" href=\"/articles/hdy201692#ref-CR50\" id=\"ref-link-section-d40968538e3330\">Wolak, 2012</a>). Moreover, other methods such as profile likelihoods (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Meyer, 2008\" title=\"Meyer K . (2008). Likelihood calculations to evaluate experimental designs to estimate genetic variances. Heredity 101: 212\\xe2\\x80\\x93221.\" href=\"/articles/hdy201692#ref-CR34\" id=\"ref-link-section-d40968538e3333\">Meyer, 2008</a>; implemented in the \\xe2\\x80\\x98proLik\\xe2\\x80\\x99 function within the package <span class=\"u-small-caps\">NADIV,</span> <a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Wolak, 2012\" title=\"Wolak ME . (2012). nadiv : an R package to create relatedness matrices for estimating non-additive genetic variances in animal models. Methods Ecol Evol 3: 792\\xe2\\x80\\x93796.\" href=\"/articles/hdy201692#ref-CR50\" id=\"ref-link-section-d40968538e3339\">Wolak, 2012</a>) are more accessible and can also improve our comprehension of a particular dataset limits (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Houle and Meyer, 2015\" title=\"Houle D, Meyer K . (2015). Estimating sampling error of evolutionary statistics based on genetic covariance matrices using maximum likelihood. J Evol Biol 28: 1542\\xe2\\x80\\x931549.\" href=\"/articles/hdy201692#ref-CR20\" id=\"ref-link-section-d40968538e3343\">Houle and Meyer, 2015</a>). As each study system is unique, in terms of pedigree structure, number of repeated measures per individual or potential causes of pedigree errors, it is difficult to predict the reliability of quantitative genetics estimates without testing it formally. Detailed simulations should be routinely included when reporting quantitative genetics analyses performed within a new study system, or when applying more complex modeling to previously studied populations, to explicitly assess the precision and accuracy of genetic variance components (<a data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" data-test=\"citation-ref\" aria-label=\"Reference Morrissey et al., 2007\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309\\xe2\\x80\\x932321.\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d40968538e3346\">Morrissey et al., 2007</a>).</p></div></div></section><section data-title=\"Data archiving\"><div class=\"c-article-section\" id=\"Sec16-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec16\">Data archiving</h2><div class=\"c-article-section__content\" id=\"Sec16-content\"><p>Data available from the Dryad Digital Repository: <a href=\"https://doi.org/10.5061/dryad.n8c57\">http://dx.doi.org/10.5061/dryad.n8c57</a>.</p></div></div></section>\\n                </div>\\n            \\n\\n            <div>\\n            <div id=\"MagazineFulltextArticleBodySuffix\"><section aria-labelledby=\"Bib1\" data-title=\"References\"><div class=\"c-article-section\" id=\"Bib1-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Bib1\">References</h2><div class=\"c-article-section__content\" id=\"Bib1-content\"><div data-container-section=\"references\"><ul class=\"c-article-references\" data-track-component=\"outbound reference\"><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR1\">\\xc3\\x85kesson M, Bensch S, Hasselquist D, Tarka M, Hansson B, Akesson M . (2008). Estimating heritabilities and genetic correlations: comparing the \\xe2\\x80\\x98animal model\\xe2\\x80\\x99 with parent\\xe2\\x80\\x93offspring regression using data from a natural population. <i>PLoS One</i> <b>3</b>: e1739.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1371/journal.pone.0001739\" data-track-action=\"article reference\" href=\"https://doi.org/10.1371%2Fjournal.pone.0001739\" aria-label=\"Article reference 1\" data-doi=\"10.1371/journal.pone.0001739\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18320057\" aria-label=\"PubMed reference 1\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed central reference\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2254494\" aria-label=\"PubMed Central reference 1\">PubMed Central</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 1\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimating%20heritabilities%20and%20genetic%20correlations%3A%20comparing%20the%20%E2%80%98animal%20model%E2%80%99%20with%20parent%E2%80%93offspring%20regression%20using%20data%20from%20a%20natural%20population&amp;journal=PLoS%20One&amp;doi=10.1371%2Fjournal.pone.0001739&amp;volume=3&amp;publication_year=2008&amp;author=%C3%85kesson%2CM&amp;author=Bensch%2CS&amp;author=Hasselquist%2CD&amp;author=Tarka%2CM&amp;author=Hansson%2CB&amp;author=Akesson%2CM\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR2\">B\\xc3\\xa9r\\xc3\\xa9nos C, Ellis PA, Pilkington JG, Pemberton JM . (2014). Estimating quantitative genetic parameters in wild populations: a comparison of pedigree and genomic approaches. <i>Mol Ecol</i> <b>23</b>: 3434\\xe2\\x80\\x933451.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/mec.12827\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fmec.12827\" aria-label=\"Article reference 2\" data-doi=\"10.1111/mec.12827\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=24917482\" aria-label=\"PubMed reference 2\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed central reference\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4149785\" aria-label=\"PubMed Central reference 2\">PubMed Central</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 2\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimating%20quantitative%20genetic%20parameters%20in%20wild%20populations%3A%20a%20comparison%20of%20pedigree%20and%20genomic%20approaches&amp;journal=Mol%20Ecol&amp;doi=10.1111%2Fmec.12827&amp;volume=23&amp;pages=3434-3451&amp;publication_year=2014&amp;author=B%C3%A9r%C3%A9nos%2CC&amp;author=Ellis%2CPA&amp;author=Pilkington%2CJG&amp;author=Pemberton%2CJM\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR3\">Blows MW, Hoffmann AA . (2005). A reassessment of genetic limits to evoutionary change. <i>Ecology</i> <b>86</b>: 1371\\xe2\\x80\\x931384.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1890/04-1209\" data-track-action=\"article reference\" href=\"https://doi.org/10.1890%2F04-1209\" aria-label=\"Article reference 3\" data-doi=\"10.1890/04-1209\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 3\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=A%20reassessment%20of%20genetic%20limits%20to%20evoutionary%20change&amp;journal=Ecology&amp;doi=10.1890%2F04-1209&amp;volume=86&amp;pages=1371-1384&amp;publication_year=2005&amp;author=Blows%2CMW&amp;author=Hoffmann%2CAA\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR4\">Bolker B . (2008) <i>Ecological Models and Data in R</i>. Princeton, University Princeton, NJ, USA.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.2307/j.ctvcm4g37\" data-track-action=\"book reference\" href=\"https://doi.org/10.2307%2Fj.ctvcm4g37\" aria-label=\"Book reference 4\" data-doi=\"10.2307/j.ctvcm4g37\">Book</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 4\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Ecological%20Models%20and%20Data%20in%20R&amp;doi=10.2307%2Fj.ctvcm4g37&amp;publication_year=2008&amp;author=Bolker%2CB\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR5\">Charmantier A, R\\xc3\\xa9ale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? <i>Mol Ecol</i> <b>14</b>: 2839\\xe2\\x80\\x932850.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1365-294X.2005.02619.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1365-294X.2005.02619.x\" aria-label=\"Article reference 5\" data-doi=\"10.1111/j.1365-294X.2005.02619.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:CAS:528:DC%2BD2MXpt1Cju7g%3D\" aria-label=\"CAS reference 5\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=16029482\" aria-label=\"PubMed reference 5\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 5\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=How%20do%20misassigned%20paternities%20affect%20the%20estimation%20of%20heritability%20in%20the%20wild%3F&amp;journal=Mol%20Ecol&amp;doi=10.1111%2Fj.1365-294X.2005.02619.x&amp;volume=14&amp;pages=2839-2850&amp;publication_year=2005&amp;author=Charmantier%2CA&amp;author=R%C3%A9ale%2CD\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR6\">Dingemanse NJ, Kazem AJN, R\\xc3\\xa9ale D, Wright J . (2009). Behavioural reaction norms: animal personality meets individual plasticity. <i>Trends Ecol Evol</i> <b>25</b>: 81\\xe2\\x80\\x9389.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1016/j.tree.2009.07.013\" data-track-action=\"article reference\" href=\"https://doi.org/10.1016%2Fj.tree.2009.07.013\" aria-label=\"Article reference 6\" data-doi=\"10.1016/j.tree.2009.07.013\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=19748700\" aria-label=\"PubMed reference 6\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 6\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Behavioural%20reaction%20norms%3A%20animal%20personality%20meets%20individual%20plasticity&amp;journal=Trends%20Ecol%20Evol&amp;doi=10.1016%2Fj.tree.2009.07.013&amp;volume=25&amp;pages=81-89&amp;publication_year=2009&amp;author=Dingemanse%2CNJ&amp;author=Kazem%2CAJN&amp;author=R%C3%A9ale%2CD&amp;author=Wright%2CJ\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR7\">Dunn PO, Whittingham LA, Pitcher TE . (2001). Mating systems, sperm competition, and the evolution of sexual dimorphism in birds. <i>Evolution</i> <b>55</b>: 161\\xe2\\x80\\x93175.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.0014-3820.2001.tb01281.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.0014-3820.2001.tb01281.x\" aria-label=\"Article reference 7\" data-doi=\"10.1111/j.0014-3820.2001.tb01281.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD3M7mt1Gmsw%3D%3D\" aria-label=\"CAS reference 7\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=11263736\" aria-label=\"PubMed reference 7\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 7\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Mating%20systems%2C%20sperm%20competition%2C%20and%20the%20evolution%20of%20sexual%20dimorphism%20in%20birds&amp;journal=Evolution&amp;doi=10.1111%2Fj.0014-3820.2001.tb01281.x&amp;volume=55&amp;pages=161-175&amp;publication_year=2001&amp;author=Dunn%2CPO&amp;author=Whittingham%2CLA&amp;author=Pitcher%2CTE\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR8\">Falconer DS, Mackay TFC . (1996) <i>Introduction to Quantitative Genetics</i>, 4th edn. Pearson: Essex, UK.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 8\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Introduction%20to%20Quantitative%20Genetics&amp;publication_year=1996&amp;author=Falconer%2CDS&amp;author=Mackay%2CTFC\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR9\">Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. <i>Evolution</i> <b>69</b>: 1336\\xe2\\x80\\x931344.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/evo.12649\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fevo.12649\" aria-label=\"Article reference 9\" data-doi=\"10.1111/evo.12649\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=25800997\" aria-label=\"PubMed reference 9\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed central reference\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4950017\" aria-label=\"PubMed Central reference 9\">PubMed Central</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 9\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20influence%20of%20nonrandom%20extra-pair%20paternity%20on%20heritability%20estimates%20derived%20from%20wild%20pedigrees&amp;journal=Evolution&amp;doi=10.1111%2Fevo.12649&amp;volume=69&amp;pages=1336-1344&amp;publication_year=2015&amp;author=Firth%2CJA&amp;author=Hadfield%2CJD&amp;author=Santure%2CAW&amp;author=Slate%2CJ&amp;author=Sheldon%2CBC\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR10\">Garcia-Gonzalez F, Simmons LW, Tomkins JL, Kotiaho JS, Evans JP . (2012). Comparing evolvabilities: common errors surrounding the calculation and use of coefficients of additive genetic variation. <i>Evolution</i> <b>66</b>: 2341\\xe2\\x80\\x932349.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1558-5646.2011.01565.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1558-5646.2011.01565.x\" aria-label=\"Article reference 10\" data-doi=\"10.1111/j.1558-5646.2011.01565.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=22834736\" aria-label=\"PubMed reference 10\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 10\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Comparing%20evolvabilities%3A%20common%20errors%20surrounding%20the%20calculation%20and%20use%20of%20coefficients%20of%20additive%20genetic%20variation&amp;journal=Evolution&amp;doi=10.1111%2Fj.1558-5646.2011.01565.x&amp;volume=66&amp;pages=2341-2349&amp;publication_year=2012&amp;author=Garcia-Gonzalez%2CF&amp;author=Simmons%2CLW&amp;author=Tomkins%2CJL&amp;author=Kotiaho%2CJS&amp;author=Evans%2CJP\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR11\">Ghilain A, B\\xc3\\xa9lisle M . (2008). Breeding success of tree swallows along a gradient of agricultural intensification. <i>Ecol Appl</i> <b>18</b>: 1140\\xe2\\x80\\x931154.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1890/07-1107.1\" data-track-action=\"article reference\" href=\"https://doi.org/10.1890%2F07-1107.1\" aria-label=\"Article reference 11\" data-doi=\"10.1890/07-1107.1\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18686577\" aria-label=\"PubMed reference 11\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 11\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Breeding%20success%20of%20tree%20swallows%20along%20a%20gradient%20of%20agricultural%20intensification&amp;journal=Ecol%20Appl&amp;doi=10.1890%2F07-1107.1&amp;volume=18&amp;pages=1140-1154&amp;publication_year=2008&amp;author=Ghilain%2CA&amp;author=B%C3%A9lisle%2CM\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR12\">Griffith SC, Owens IPF, Burke T . (1999). Environmental determination of a sexually selected trait. <i>Science</i> <b>400</b>: 358\\xe2\\x80\\x93360.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:CAS:528:DyaK1MXkvFOnsr4%3D\" aria-label=\"CAS reference 12\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 12\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Environmental%20determination%20of%20a%20sexually%20selected%20trait&amp;journal=Science&amp;volume=400&amp;pages=358-360&amp;publication_year=1999&amp;author=Griffith%2CSC&amp;author=Owens%2CIPF&amp;author=Burke%2CT\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR13\">Griffith SC, Owens IPF, Thuman KA . (2002). Extra pair paternity in birds: a review of interspecific variation and adaptive function. <i>Mol Ecol</i> <b>11</b>: 2195\\xe2\\x80\\x932212.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1046/j.1365-294X.2002.01613.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1046%2Fj.1365-294X.2002.01613.x\" aria-label=\"Article reference 13\" data-doi=\"10.1046/j.1365-294X.2002.01613.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:CAS:528:DC%2BD38XptlSmurY%3D\" aria-label=\"CAS reference 13\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=12406233\" aria-label=\"PubMed reference 13\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 13\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Extra%20pair%20paternity%20in%20birds%3A%20a%20review%20of%20interspecific%20variation%20and%20adaptive%20function&amp;journal=Mol%20Ecol&amp;doi=10.1046%2Fj.1365-294X.2002.01613.x&amp;volume=11&amp;pages=2195-2212&amp;publication_year=2002&amp;author=Griffith%2CSC&amp;author=Owens%2CIPF&amp;author=Thuman%2CKA\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR14\">Hadfield JD . (2010). MCMC methods for multi-response generalized linear mixed models: the MCMCglmm R package. <i>J Stat Softw</i> <b>33</b>: 1\\xe2\\x80\\x9322.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.18637/jss.v033.i02\" data-track-action=\"article reference\" href=\"https://doi.org/10.18637%2Fjss.v033.i02\" aria-label=\"Article reference 14\" data-doi=\"10.18637/jss.v033.i02\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 14\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=MCMC%20methods%20for%20multi-response%20generalized%20linear%20mixed%20models%3A%20the%20MCMCglmm%20R%20package&amp;journal=J%20Stat%20Softw&amp;doi=10.18637%2Fjss.v033.i02&amp;volume=33&amp;pages=1-22&amp;publication_year=2010&amp;author=Hadfield%2CJD\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR15\">Henderson CR . (1984) <i>Applications of Linear Models in Animal Breeding</i>. University of Guelph: Guelph, ON, Canada.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 15\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Applications%20of%20Linear%20Models%20in%20Animal%20Breeding&amp;publication_year=1984&amp;author=Henderson%2CCR\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR16\">Hendry AP, Kinnison MT, Heino M, Day T, Smith TB, Fitt G <i>et al</i>. (2011). Evolutionary principles and their practical application. <i>Evol Appl</i> <b>4</b>: 159\\xe2\\x80\\x93183.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1752-4571.2010.00165.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1752-4571.2010.00165.x\" aria-label=\"Article reference 16\" data-doi=\"10.1111/j.1752-4571.2010.00165.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=25567966\" aria-label=\"PubMed reference 16\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed central reference\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3352551\" aria-label=\"PubMed Central reference 16\">PubMed Central</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 16\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Evolutionary%20principles%20and%20their%20practical%20application&amp;journal=Evol%20Appl&amp;doi=10.1111%2Fj.1752-4571.2010.00165.x&amp;volume=4&amp;pages=159-183&amp;publication_year=2011&amp;author=Hendry%2CAP&amp;author=Kinnison%2CMT&amp;author=Heino%2CM&amp;author=Day%2CT&amp;author=Smith%2CTB&amp;author=Fitt%2CG\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR17\">Hoffmann AA, Meril\\xc3\\xa4 J . (1999). Heritable variation and evolution under favourable and unfavourable conditions. <i>Trends Ecol Evol</i> <b>14</b>: 96\\xe2\\x80\\x93101.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1016/S0169-5347(99)01595-5\" data-track-action=\"article reference\" href=\"https://doi.org/10.1016%2FS0169-5347%2899%2901595-5\" aria-label=\"Article reference 17\" data-doi=\"10.1016/S0169-5347(99)01595-5\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:STN:280:DC%2BC2sbgsVGquw%3D%3D\" aria-label=\"CAS reference 17\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=10322508\" aria-label=\"PubMed reference 17\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 17\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Heritable%20variation%20and%20evolution%20under%20favourable%20and%20unfavourable%20conditions&amp;journal=Trends%20Ecol%20Evol&amp;doi=10.1016%2FS0169-5347%2899%2901595-5&amp;volume=14&amp;pages=96-101&amp;publication_year=1999&amp;author=Hoffmann%2CAA&amp;author=Meril%C3%A4%2CJ\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR18\">Hosner PA, Winkler DW . (2007). Dispersal distances of tree swallows estimated from continent-wide and limited-area data. <i>J Field Ornithol</i> <b>78</b>: 290\\xe2\\x80\\x93297.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1557-9263.2007.00117.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1557-9263.2007.00117.x\" aria-label=\"Article reference 18\" data-doi=\"10.1111/j.1557-9263.2007.00117.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 18\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Dispersal%20distances%20of%20tree%20swallows%20estimated%20from%20continent-wide%20and%20limited-area%20data&amp;journal=J%20Field%20Ornithol&amp;doi=10.1111%2Fj.1557-9263.2007.00117.x&amp;volume=78&amp;pages=290-297&amp;publication_year=2007&amp;author=Hosner%2CPA&amp;author=Winkler%2CDW\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR19\">Houle D . (1992). Comparing evolvability and variability of quantitative traits. <i>Genetics</i> <b>130</b>: 195\\xe2\\x80\\x93204.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:STN:280:DyaK387itVGitQ%3D%3D\" aria-label=\"CAS reference 19\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=1732160\" aria-label=\"PubMed reference 19\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed central reference\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1204793\" aria-label=\"PubMed Central reference 19\">PubMed Central</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 19\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Comparing%20evolvability%20and%20variability%20of%20quantitative%20traits&amp;journal=Genetics&amp;volume=130&amp;pages=195-204&amp;publication_year=1992&amp;author=Houle%2CD\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR20\">Houle D, Meyer K . (2015). Estimating sampling error of evolutionary statistics based on genetic covariance matrices using maximum likelihood. <i>J Evol Biol</i> <b>28</b>: 1542\\xe2\\x80\\x931549.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/jeb.12674\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fjeb.12674\" aria-label=\"Article reference 20\" data-doi=\"10.1111/jeb.12674\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:STN:280:DC%2BC2MbjvVGguw%3D%3D\" aria-label=\"CAS reference 20\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=26079756\" aria-label=\"PubMed reference 20\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 20\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimating%20sampling%20error%20of%20evolutionary%20statistics%20based%20on%20genetic%20covariance%20matrices%20using%20maximum%20likelihood&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fjeb.12674&amp;volume=28&amp;pages=1542-1549&amp;publication_year=2015&amp;author=Houle%2CD&amp;author=Meyer%2CK\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR21\">Husby A, Kawakami T, R\\xc3\\xb6nneg\\xc3\\xa5rd L, Smeds L, Ellegren H, Qvarnstr\\xc3\\xb6m A . (2015). Genome-wide association mapping in a wild avian population identifies a link between genetic and phenotypic variation in a life-history trait. <i>Proc R Soc B</i> <b>282</b>: 20150156.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1098/rspb.2015.0156\" data-track-action=\"article reference\" href=\"https://doi.org/10.1098%2Frspb.2015.0156\" aria-label=\"Article reference 21\" data-doi=\"10.1098/rspb.2015.0156\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=25833857\" aria-label=\"PubMed reference 21\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed central reference\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4426624\" aria-label=\"PubMed Central reference 21\">PubMed Central</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 21\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Genome-wide%20association%20mapping%20in%20a%20wild%20avian%20population%20identifies%20a%20link%20between%20genetic%20and%20phenotypic%20variation%20in%20a%20life-history%20trait&amp;journal=Proc%20R%20Soc%20B&amp;doi=10.1098%2Frspb.2015.0156&amp;volume=282&amp;publication_year=2015&amp;author=Husby%2CA&amp;author=Kawakami%2CT&amp;author=R%C3%B6nneg%C3%A5rd%2CL&amp;author=Smeds%2CL&amp;author=Ellegren%2CH&amp;author=Qvarnstr%C3%B6m%2CA\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR22\">Hussell DJT . (1983). Age and plumage color in female tree swallows. <i>J Field Ornithol</i> <b>54</b>: 312\\xe2\\x80\\x93318.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 22\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Age%20and%20plumage%20color%20in%20female%20tree%20swallows&amp;journal=J%20Field%20Ornithol&amp;volume=54&amp;pages=312-318&amp;publication_year=1983&amp;author=Hussell%2CDJT\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR23\">Kalinowski ST, Taper ML, Marshall TC . (2007). Revising how the computer program cervus accommodates genotyping error increases success in paternity assignment. <i>Mol Ecol</i> <b>16</b>: 1099\\xe2\\x80\\x931106.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1365-294X.2007.03089.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1365-294X.2007.03089.x\" aria-label=\"Article reference 23\" data-doi=\"10.1111/j.1365-294X.2007.03089.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17305863\" aria-label=\"PubMed reference 23\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 23\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Revising%20how%20the%20computer%20program%20cervus%20accommodates%20genotyping%20error%20increases%20success%20in%20paternity%20assignment&amp;journal=Mol%20Ecol&amp;doi=10.1111%2Fj.1365-294X.2007.03089.x&amp;volume=16&amp;pages=1099-1106&amp;publication_year=2007&amp;author=Kalinowski%2CST&amp;author=Taper%2CML&amp;author=Marshall%2CTC\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR24\">Kruuk LEB . (2004). Estimating genetic parameters in natural populations using the \\xe2\\x80\\x98animal model\\xe2\\x80\\x99. <i>Philos Trans R Soc B</i> <b>359</b>: 873\\xe2\\x80\\x93890.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1098/rstb.2003.1437\" data-track-action=\"article reference\" href=\"https://doi.org/10.1098%2Frstb.2003.1437\" aria-label=\"Article reference 24\" data-doi=\"10.1098/rstb.2003.1437\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 24\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimating%20genetic%20parameters%20in%20natural%20populations%20using%20the%20%E2%80%98animal%20model%E2%80%99&amp;journal=Philos%20Trans%20R%20Soc%20B&amp;doi=10.1098%2Frstb.2003.1437&amp;volume=359&amp;pages=873-890&amp;publication_year=2004&amp;author=Kruuk%2CLEB\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR25\">Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. <i>J Evol Biol</i> <b>20</b>: 1890\\xe2\\x80\\x931903.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1420-9101.2007.01377.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2007.01377.x\" aria-label=\"Article reference 25\" data-doi=\"10.1111/j.1420-9101.2007.01377.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD2svpsFGmuw%3D%3D\" aria-label=\"CAS reference 25\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17714306\" aria-label=\"PubMed reference 25\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 25\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=How%20to%20separate%20genetic%20and%20environmental%20causes%20of%20similarity%20between%20relatives&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2007.01377.x&amp;volume=20&amp;pages=1890-1903&amp;publication_year=2007&amp;author=Kruuk%2CLEB&amp;author=Hadfield%2CJD\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR26\">Kruuk LEB, Slate J, Wilson AJ . (2008). New answers for old questions: the evolutionary quantitative genetics of wild animal populations. <i>Annu Rev Ecol Evol Syst</i> <b>39</b>: 525\\xe2\\x80\\x93548.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1146/annurev.ecolsys.39.110707.173542\" data-track-action=\"article reference\" href=\"https://doi.org/10.1146%2Fannurev.ecolsys.39.110707.173542\" aria-label=\"Article reference 26\" data-doi=\"10.1146/annurev.ecolsys.39.110707.173542\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 26\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=New%20answers%20for%20old%20questions%3A%20the%20evolutionary%20quantitative%20genetics%20of%20wild%20animal%20populations&amp;journal=Annu%20Rev%20Ecol%20Evol%20Syst&amp;doi=10.1146%2Fannurev.ecolsys.39.110707.173542&amp;volume=39&amp;pages=525-548&amp;publication_year=2008&amp;author=Kruuk%2CLEB&amp;author=Slate%2CJ&amp;author=Wilson%2CAJ\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR27\">Krzywinski M, Altman N . (2013). Points of significance: error bars. <i>Nat Methods</i> <b>10</b>: 921\\xe2\\x80\\x93922.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1038/nmeth.2659\" data-track-action=\"article reference\" href=\"https://doi.org/10.1038%2Fnmeth.2659\" aria-label=\"Article reference 27\" data-doi=\"10.1038/nmeth.2659\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:CAS:528:DC%2BC3sXhsFaksrzO\" aria-label=\"CAS reference 27\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=24161969\" aria-label=\"PubMed reference 27\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 27\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Points%20of%20significance%3A%20error%20bars&amp;journal=Nat%20Methods&amp;doi=10.1038%2Fnmeth.2659&amp;volume=10&amp;pages=921-922&amp;publication_year=2013&amp;author=Krzywinski%2CM&amp;author=Altman%2CN\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR28\">Lande R . (1981). The minimum number of genes contributing to quantitative variation between and within populations. <i>Genetics</i> <b>99</b>: 541\\xe2\\x80\\x93553.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:STN:280:DyaL383htFOmsQ%3D%3D\" aria-label=\"CAS reference 28\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=7343418\" aria-label=\"PubMed reference 28\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed central reference\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1214520\" aria-label=\"PubMed Central reference 28\">PubMed Central</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 28\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20minimum%20number%20of%20genes%20contributing%20to%20quantitative%20variation%20between%20and%20within%20populations&amp;journal=Genetics&amp;volume=99&amp;pages=541-553&amp;publication_year=1981&amp;author=Lande%2CR\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR29\">Lande R, Arnold SJ . (1983). The measurement of selection on correlated characters. <i>Evolution</i> <b>37</b>: 1210\\xe2\\x80\\x931226.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1558-5646.1983.tb00236.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1558-5646.1983.tb00236.x\" aria-label=\"Article reference 29\" data-doi=\"10.1111/j.1558-5646.1983.tb00236.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=28556011\" aria-label=\"PubMed reference 29\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 29\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20measurement%20of%20selection%20on%20correlated%20characters&amp;journal=Evolution&amp;doi=10.1111%2Fj.1558-5646.1983.tb00236.x&amp;volume=37&amp;pages=1210-1226&amp;publication_year=1983&amp;author=Lande%2CR&amp;author=Arnold%2CSJ\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR30\">Lee C, Pollak EJ . (1997). Influence of sire misidentification on sire x year interaction variance and direct-maternal genetic covariance for weaning weight in beef cattle. <i>J Anim Sci</i> <b>75</b>: 2858\\xe2\\x80\\x932863.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.2527/1997.75112858x\" data-track-action=\"article reference\" href=\"https://doi.org/10.2527%2F1997.75112858x\" aria-label=\"Article reference 30\" data-doi=\"10.2527/1997.75112858x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:CAS:528:DyaK2sXntFemurw%3D\" aria-label=\"CAS reference 30\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=9374297\" aria-label=\"PubMed reference 30\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 30\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Influence%20of%20sire%20misidentification%20on%20sire%20x%20year%20interaction%20variance%20and%20direct-maternal%20genetic%20covariance%20for%20weaning%20weight%20in%20beef%20cattle&amp;journal=J%20Anim%20Sci&amp;doi=10.2527%2F1997.75112858x&amp;volume=75&amp;pages=2858-2863&amp;publication_year=1997&amp;author=Lee%2CC&amp;author=Pollak%2CEJ\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR31\">Lemons PR, Marshall TC, McCloskey SE, Sethi SA, Schmutz JA, Sedinger JS . (2015). A likelihood-based approach for assessment of extra-pair paternity and conspecific brood parasitism in natural populations. <i>Mol Ecol Resour</i> <b>7</b>: 107\\xe2\\x80\\x93116.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/1755-0998.12287\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2F1755-0998.12287\" aria-label=\"Article reference 31\" data-doi=\"10.1111/1755-0998.12287\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 31\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=A%20likelihood-based%20approach%20for%20assessment%20of%20extra-pair%20paternity%20and%20conspecific%20brood%20parasitism%20in%20natural%20populations&amp;journal=Mol%20Ecol%20Resour&amp;doi=10.1111%2F1755-0998.12287&amp;volume=7&amp;pages=107-116&amp;publication_year=2015&amp;author=Lemons%2CPR&amp;author=Marshall%2CTC&amp;author=McCloskey%2CSE&amp;author=Sethi%2CSA&amp;author=Schmutz%2CJA&amp;author=Sedinger%2CJS\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR32\">Lessard A, Bourret A, B\\xc3\\xa9lisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (<i>Tachycineta bicolor</i>. <i>Behav Ecol Sociobiol</i> <b>68</b>: 733\\xe2\\x80\\x93742.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1007/s00265-014-1686-y\" data-track-action=\"article reference\" href=\"https://doi.org/10.1007%2Fs00265-014-1686-y\" aria-label=\"Article reference 32\" data-doi=\"10.1007/s00265-014-1686-y\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 32\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Individual%20and%20environmental%20determinants%20of%20reproductive%20success%20in%20male%20tree%20swallow%20%28Tachycineta%20bicolor&amp;journal=Behav%20Ecol%20Sociobiol&amp;doi=10.1007%2Fs00265-014-1686-y&amp;volume=68&amp;pages=733-742&amp;publication_year=2014&amp;author=Lessard%2CA&amp;author=Bourret%2CA&amp;author=B%C3%A9lisle%2CM&amp;author=Pelletier%2CF&amp;author=Garant%2CD\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR33\">Lynch M, Walsh B . (1998) <i>Genetics and Analysis of Quantitative Traits</i>. Sinauer Associates: Sunderland, MA, USA.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 33\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Genetics%20and%20Analysis%20of%20Quantitative%20Traits&amp;publication_year=1998&amp;author=Lynch%2CM&amp;author=Walsh%2CB\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR34\">Meyer K . (2008). Likelihood calculations to evaluate experimental designs to estimate genetic variances. <i>Heredity</i> <b>101</b>: 212\\xe2\\x80\\x93221.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1038/hdy.2008.46\" data-track-action=\"article reference\" href=\"https://doi.org/10.1038%2Fhdy.2008.46\" aria-label=\"Article reference 34\" data-doi=\"10.1038/hdy.2008.46\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD1crjtVSmug%3D%3D\" aria-label=\"CAS reference 34\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18523444\" aria-label=\"PubMed reference 34\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 34\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Likelihood%20calculations%20to%20evaluate%20experimental%20designs%20to%20estimate%20genetic%20variances&amp;journal=Heredity&amp;doi=10.1038%2Fhdy.2008.46&amp;volume=101&amp;pages=212-221&amp;publication_year=2008&amp;author=Meyer%2CK\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR35\">Morrissey MB, Kruuk LEB, Wilson AJ . (2010). The danger of applying the breeder\\xe2\\x80\\x99s equation in observational studies of natural populations. <i>J Evol Biol</i> <b>23</b>: 2277\\xe2\\x80\\x932288.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1420-9101.2010.02084.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2010.02084.x\" aria-label=\"Article reference 35\" data-doi=\"10.1111/j.1420-9101.2010.02084.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:STN:280:DC%2BC3cbhtFCksQ%3D%3D\" aria-label=\"CAS reference 35\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=20831731\" aria-label=\"PubMed reference 35\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 35\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20danger%20of%20applying%20the%20breeder%E2%80%99s%20equation%20in%20observational%20studies%20of%20natural%20populations&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2010.02084.x&amp;volume=23&amp;pages=2277-2288&amp;publication_year=2010&amp;author=Morrissey%2CMB&amp;author=Kruuk%2CLEB&amp;author=Wilson%2CAJ\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR36\">Morrissey MB, de Villemereuil P, Doligez B, Gimenez O . (2014). Bayesian approaches to the quantitative genetic analysis of natural populations. In: Charmantier A, Garant D <i>et al.</i> (eds) <i>Quantitative Genetics in the Wild</i>. Oxford University Press, Oxford, UK, pp 228\\xe2\\x80\\x93253.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1093/acprof:oso/9780199674237.003.0014\" data-track-action=\"chapter reference\" href=\"https://doi.org/10.1093%2Facprof%3Aoso%2F9780199674237.003.0014\" aria-label=\"Chapter reference 36\" data-doi=\"10.1093/acprof:oso/9780199674237.003.0014\">Chapter</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 36\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Bayesian%20approaches%20to%20the%20quantitative%20genetic%20analysis%20of%20natural%20populations&amp;doi=10.1093%2Facprof%3Aoso%2F9780199674237.003.0014&amp;pages=228-253&amp;publication_year=2014&amp;author=Morrissey%2CMichael%20B.&amp;author=de%20Villemereuil%2CPierre&amp;author=Doligez%2CBlandine&amp;author=Gimenez%2COlivier\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR37\">Morrissey MB, Wilson AJ . (2010). PEDANTICS: an R package for pedigree-based genetic simulation and pedigree manipulation, characterization and viewing. <i>Mol Ecol Resour</i> <b>10</b>: 711\\xe2\\x80\\x93719.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1755-0998.2009.02817.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1755-0998.2009.02817.x\" aria-label=\"Article reference 37\" data-doi=\"10.1111/j.1755-0998.2009.02817.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=21565076\" aria-label=\"PubMed reference 37\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 37\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=PEDANTICS%3A%20an%20R%20package%20for%20pedigree-based%20genetic%20simulation%20and%20pedigree%20manipulation%2C%20characterization%20and%20viewing&amp;journal=Mol%20Ecol%20Resour&amp;doi=10.1111%2Fj.1755-0998.2009.02817.x&amp;volume=10&amp;pages=711-719&amp;publication_year=2010&amp;author=Morrissey%2CMB&amp;author=Wilson%2CAJ\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR38\">Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (<i>Ovis aries</i>. <i>J Evol Biol</i> <b>20</b>: 2309\\xe2\\x80\\x932321.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1420-9101.2007.01412.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2007.01412.x\" aria-label=\"Article reference 38\" data-doi=\"10.1111/j.1420-9101.2007.01412.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD2snjt1eksQ%3D%3D\" aria-label=\"CAS reference 38\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17956393\" aria-label=\"PubMed reference 38\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 38\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=A%20framework%20for%20power%20and%20sensitivity%20analyses%20for%20quantitative%20genetic%20studies%20of%20natural%20populations%2C%20and%20case%20studies%20in%20Soay%20sheep%20%28Ovis%20aries&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2007.01412.x&amp;volume=20&amp;pages=2309-2321&amp;publication_year=2007&amp;author=Morrissey%2CMB&amp;author=Wilson%2CAJ&amp;author=Pemberton%2CJM&amp;author=Ferguson%2CMM\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR39\">Postma E . (2014). Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates? In: Charmantier A, Garant D <i>et al.</i> (eds) <i>Quantitative Genetics in the Wild.</i> Oxford University Press, Oxford, UK, pp 16\\xe2\\x80\\x9333.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1093/acprof:oso/9780199674237.003.0002\" data-track-action=\"chapter reference\" href=\"https://doi.org/10.1093%2Facprof%3Aoso%2F9780199674237.003.0002\" aria-label=\"Chapter reference 39\" data-doi=\"10.1093/acprof:oso/9780199674237.003.0002\">Chapter</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 39\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Four%20decades%20of%20estimating%20heritabilities%20in%20wild%20vertebrate%20populations%3A%20improved%20methods%2C%20more%20data%2C%20better%20estimates%3F&amp;doi=10.1093%2Facprof%3Aoso%2F9780199674237.003.0002&amp;pages=16-33&amp;publication_year=2014&amp;author=Postma%2CErik\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR40\">Quinn JL, Charmantier A, Garant D, Sheldon BC . (2006). Data depth, data completeness, and their influence on quantitative genetic estimation in two contrasting bird populations. <i>J Evol Biol</i> <b>19</b>: 994\\xe2\\x80\\x931002.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1420-9101.2006.01081.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2006.01081.x\" aria-label=\"Article reference 40\" data-doi=\"10.1111/j.1420-9101.2006.01081.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD283ltV2rtw%3D%3D\" aria-label=\"CAS reference 40\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=16674594\" aria-label=\"PubMed reference 40\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 40\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Data%20depth%2C%20data%20completeness%2C%20and%20their%20influence%20on%20quantitative%20genetic%20estimation%20in%20two%20contrasting%20bird%20populations&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2006.01081.x&amp;volume=19&amp;pages=994-1002&amp;publication_year=2006&amp;author=Quinn%2CJL&amp;author=Charmantier%2CA&amp;author=Garant%2CD&amp;author=Sheldon%2CBC\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR41\">Core Team R . (2015). R: a language and environment for statistical computing <a href=\"http://www.R-project.org/\">http://www.R-project.org/</a>.</p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR42\">Via S, Lande R . (1985). Genotype-environment interaction and the evolution of phenotypic plasticity. <i>Evolution</i> <b>39</b>: 505\\xe2\\x80\\x93522.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1558-5646.1985.tb00391.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1558-5646.1985.tb00391.x\" aria-label=\"Article reference 42\" data-doi=\"10.1111/j.1558-5646.1985.tb00391.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=28561964\" aria-label=\"PubMed reference 42\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 42\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Genotype-environment%20interaction%20and%20the%20evolution%20of%20phenotypic%20plasticity&amp;journal=Evolution&amp;doi=10.1111%2Fj.1558-5646.1985.tb00391.x&amp;volume=39&amp;pages=505-522&amp;publication_year=1985&amp;author=Via%2CS&amp;author=Lande%2CR\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR43\">de Villemereuil P, Gimenez O, Doligez B . (2013). Comparing parent\\xe2\\x80\\x93offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits. <i>Methods Ecol Evol</i> <b>4</b>: 260\\xe2\\x80\\x93275.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/2041-210X.12011\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2F2041-210X.12011\" aria-label=\"Article reference 43\" data-doi=\"10.1111/2041-210X.12011\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 43\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Comparing%20parent%E2%80%93offspring%20regression%20with%20frequentist%20and%20Bayesian%20animal%20models%20to%20estimate%20heritability%20in%20wild%20populations%3A%20A%20simulation%20study%20for%20Gaussian%20and%20binary%20traits&amp;journal=Methods%20Ecol%20Evol&amp;doi=10.1111%2F2041-210X.12011&amp;volume=4&amp;pages=260-275&amp;publication_year=2013&amp;author=de%20Villemereuil%2CP&amp;author=Gimenez%2CO&amp;author=Doligez%2CB\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR44\">Vinkhuyzen AAE, Wray NR, Yang J, Goddard ME, Visscher PM . (2013). Estimation and partition of heritability in human populations using whole-genome analysis methods. <i>Annu Rev Genet</i> <b>47</b>: 75\\xe2\\x80\\x9395.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1146/annurev-genet-111212-133258\" data-track-action=\"article reference\" href=\"https://doi.org/10.1146%2Fannurev-genet-111212-133258\" aria-label=\"Article reference 44\" data-doi=\"10.1146/annurev-genet-111212-133258\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:CAS:528:DC%2BC2cXhsFOjtw%3D%3D\" aria-label=\"CAS reference 44\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=23988118\" aria-label=\"PubMed reference 44\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed central reference\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4037293\" aria-label=\"PubMed Central reference 44\">PubMed Central</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 44\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimation%20and%20partition%20of%20heritability%20in%20human%20populations%20using%20whole-genome%20analysis%20methods&amp;journal=Annu%20Rev%20Genet&amp;doi=10.1146%2Fannurev-genet-111212-133258&amp;volume=47&amp;pages=75-95&amp;publication_year=2013&amp;author=Vinkhuyzen%2CAAE&amp;author=Wray%2CNR&amp;author=Yang%2CJ&amp;author=Goddard%2CME&amp;author=Visscher%2CPM\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR45\">Walther BA, Moore JL . (2005). The concepts of bias, precision and accuracy, and their use in testing the performance of species richness estimators, with a literature review of estimator performance. <i>Ecography</i> <b>28</b>: 815\\xe2\\x80\\x93829.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.2005.0906-7590.04112.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.2005.0906-7590.04112.x\" aria-label=\"Article reference 45\" data-doi=\"10.1111/j.2005.0906-7590.04112.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 45\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20concepts%20of%20bias%2C%20precision%20and%20accuracy%2C%20and%20their%20use%20in%20testing%20the%20performance%20of%20species%20richness%20estimators%2C%20with%20a%20literature%20review%20of%20estimator%20performance&amp;journal=Ecography&amp;doi=10.1111%2Fj.2005.0906-7590.04112.x&amp;volume=28&amp;pages=815-829&amp;publication_year=2005&amp;author=Walther%2CBA&amp;author=Moore%2CJL\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR46\">Wilson AJ . (2008). Why h<sup>2</sup> does not always equal VA/VP? <i>J Evol Biol</i> <b>21</b>: 647\\xe2\\x80\\x93650.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1420-9101.2008.01500.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2008.01500.x\" aria-label=\"Article reference 46\" data-doi=\"10.1111/j.1420-9101.2008.01500.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD1c3kvVKmtg%3D%3D\" aria-label=\"CAS reference 46\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18266683\" aria-label=\"PubMed reference 46\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 46\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Why%20h2%20does%20not%20always%20equal%20VA%2FVP%3F&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2008.01500.x&amp;volume=21&amp;pages=647-650&amp;publication_year=2008&amp;author=Wilson%2CAJ\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR47\">Wilson AJ, Morrissey MB, Adams MJ, Walling CA, Guinness FE, Pemberton JM <i>et al</i>. (2011). Indirect genetics effects and evolutionary constraint: an analysis of social dominance in red deer, <i>Cervus elaphus</i>. <i>J Evol Biol</i> <b>24</b>: 772\\xe2\\x80\\x93783.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1420-9101.2010.02212.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2010.02212.x\" aria-label=\"Article reference 47\" data-doi=\"10.1111/j.1420-9101.2010.02212.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"cas reference\" href=\"/articles/cas-redirect/1:STN:280:DC%2BC3M3msVCqtQ%3D%3D\" aria-label=\"CAS reference 47\">CAS</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=21288272\" aria-label=\"PubMed reference 47\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 47\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Indirect%20genetics%20effects%20and%20evolutionary%20constraint%3A%20an%20analysis%20of%20social%20dominance%20in%20red%20deer%2C%20Cervus%20elaphus&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2010.02212.x&amp;volume=24&amp;pages=772-783&amp;publication_year=2011&amp;author=Wilson%2CAJ&amp;author=Morrissey%2CMB&amp;author=Adams%2CMJ&amp;author=Walling%2CCA&amp;author=Guinness%2CFE&amp;author=Pemberton%2CJM\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR48\">Wilson AJ, R\\xc3\\xa9ale D, Clements MN, Morrissey MM, Postma E, Walling CA <i>et al</i>. (2010). An ecologist\\xe2\\x80\\x99s guide to the animal model. <i>J Anim Ecol</i> <b>79</b>: 13\\xe2\\x80\\x9326.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.1365-2656.2009.01639.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.1365-2656.2009.01639.x\" aria-label=\"Article reference 48\" data-doi=\"10.1111/j.1365-2656.2009.01639.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=20409158\" aria-label=\"PubMed reference 48\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 48\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=An%20ecologist%E2%80%99s%20guide%20to%20the%20animal%20model&amp;journal=J%20Anim%20Ecol&amp;doi=10.1111%2Fj.1365-2656.2009.01639.x&amp;volume=79&amp;pages=13-26&amp;publication_year=2010&amp;author=Wilson%2CAJ&amp;author=R%C3%A9ale%2CD&amp;author=Clements%2CMN&amp;author=Morrissey%2CMM&amp;author=Postma%2CE&amp;author=Walling%2CCA\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR49\">Winkler D, Hallinger K, Ardia D, Robertson R, Stutchbury B, Cohen R . (2011). Tree swallow (<i>Tachycineta bicolor</i>. Birds North Am Online. <a href=\"http://bna.birds.cornell.edu/bna/\">http://bna.birds.cornelledu/bna/</a>. Accessed on 1 Oct 2013.</p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR50\">Wolak ME . (2012). nadiv : an R package to create relatedness matrices for estimating non-additive genetic variances in animal models. <i>Methods Ecol Evol</i> <b>3</b>: 792\\xe2\\x80\\x93796.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1111/j.2041-210X.2012.00213.x\" data-track-action=\"article reference\" href=\"https://doi.org/10.1111%2Fj.2041-210X.2012.00213.x\" aria-label=\"Article reference 50\" data-doi=\"10.1111/j.2041-210X.2012.00213.x\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 50\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=nadiv%20%3A%20an%20R%20package%20to%20create%20relatedness%20matrices%20for%20estimating%20non-additive%20genetic%20variances%20in%20animal%20models&amp;journal=Methods%20Ecol%20Evol&amp;doi=10.1111%2Fj.2041-210X.2012.00213.x&amp;volume=3&amp;pages=792-796&amp;publication_year=2012&amp;author=Wolak%2CME\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR51\">Wolak ME, Roff DA, Fairbairn DJ . (2015). Are we underestimating the genetic variances of dimorphic traits? <i>Ecol Evol</i> <b>5</b>: 590\\xe2\\x80\\x93597.</p><p class=\"c-article-references__links u-hide-print\"><a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"10.1002/ece3.1361\" data-track-action=\"article reference\" href=\"https://doi.org/10.1002%2Fece3.1361\" aria-label=\"Article reference 51\" data-doi=\"10.1002/ece3.1361\">Article</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed reference\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=25691983\" aria-label=\"PubMed reference 51\">PubMed</a>\\xc2\\xa0\\n    <a data-track=\"click\" rel=\"nofollow noopener\" data-track-label=\"link\" data-track-action=\"pubmed central reference\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4328764\" aria-label=\"PubMed Central reference 51\">PubMed Central</a>\\xc2\\xa0\\n    <a data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" rel=\"nofollow noopener\" aria-label=\"Google Scholar reference 51\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Are%20we%20underestimating%20the%20genetic%20variances%20of%20dimorphic%20traits%3F&amp;journal=Ecol%20Evol&amp;doi=10.1002%2Fece3.1361&amp;volume=5&amp;pages=590-597&amp;publication_year=2015&amp;author=Wolak%2CME&amp;author=Roff%2CDA&amp;author=Fairbairn%2CDJ\">\\n                    Google Scholar</a>\\xc2\\xa0\\n                </p></li></ul><p class=\"c-article-references__download u-hide-print\"><a data-track=\"click\" data-track-action=\"download citation references\" data-track-label=\"link\" href=\"https://citation-needed.springer.com/v2/references/10.1038/hdy.2016.92?format=refman&amp;flavour=references\">Download references<svg width=\"16\" height=\"16\" focusable=\"false\" role=\"img\" aria-hidden=\"true\" class=\"u-icon\"><use xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"#global-icon-download\"></use></svg></a></p></div></div></div></section></div><section data-title=\"Acknowledgements\"><div class=\"c-article-section\" id=\"Ack1-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Ack1\">Acknowledgements</h2><div class=\"c-article-section__content\" id=\"Ack1-content\"><p>We thank the 40 farm owners for providing access to their lands and all graduate students and field and laboratory assistants who have contributed in gathering data in our system over the years. We also thank Fanie Pelletier and Marc B\\xc3\\xa9lisle for their long-term involvement in the southern Qu\\xc3\\xa9bec Tree swallow study. We also thank them as well as Denis R\\xc3\\xa9ale, Matthew Wolak, Michael Morrissey and two anonymous reviewers for constructive comments on a previous version of this manuscript. Animals were captured and handled in compliance with the Canadian Council on Animal Care, under the approval of the Universit\\xc3\\xa9 de Sherbrooke Animal Ethics Committee (Protocol Number: DG2010-01 and DG2014-01-Universit\\xc3\\xa9 de Sherbrooke). This work was supported by the Natural Sciences and Engineering Research Council of Canada (NSERC) discovery grants to DG, Fanie Pelletier and Marc B\\xc3\\xa9lisle and by the Canada Research Chairs program to Fanie Pelletier and Marc B\\xc3\\xa9lisle. AB was supported by a postgraduate NSERC scholarship.</p></div></div></section><section aria-labelledby=\"author-information\" data-title=\"Author information\"><div class=\"c-article-section\" id=\"author-information-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"author-information\">Author information</h2><div class=\"c-article-section__content\" id=\"author-information-content\"><h3 class=\"c-article__sub-heading\" id=\"affiliations\">Authors and Affiliations</h3><ol class=\"c-article-author-affiliation__list\"><li id=\"Aff1\"><p class=\"c-article-author-affiliation__address\">D\\xc3\\xa9partement de Biologie, Universit\\xc3\\xa9 de Sherbrooke, Sherbrooke, Quebec, Canada</p><p class=\"c-article-author-affiliation__authors-list\">A Bourret\\xc2\\xa0&amp;\\xc2\\xa0D Garant</p></li></ol><div class=\"u-js-hide u-hide-print\" data-test=\"author-info\"><span class=\"c-article__sub-heading\">Authors</span><ol class=\"c-article-authors-search u-list-reset\"><li id=\"auth-A-Bourret\"><span class=\"c-article-authors-search__title u-h3 js-search-name\">A Bourret</span><div class=\"c-article-authors-search__list\"><div class=\"c-article-authors-search__item c-article-authors-search__list-item--left\"><a href=\"/search?author=A%20Bourret\" class=\"c-article-button\" data-track=\"click\" data-track-action=\"author link - publication\" data-track-label=\"link\" rel=\"nofollow\">View author publications</a></div><div class=\"c-article-authors-search__item c-article-authors-search__list-item--right\"><p class=\"search-in-title-js c-article-authors-search__text\">You can also search for this author in\\n                        <span class=\"c-article-identifiers\"><a class=\"c-article-identifiers__item\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&amp;term=A%20Bourret\" data-track=\"click\" data-track-action=\"author link - pubmed\" data-track-label=\"link\" rel=\"nofollow\">PubMed</a><span class=\"u-hide\">\\xc2\\xa0</span><a class=\"c-article-identifiers__item\" href=\"http://scholar.google.co.uk/scholar?as_q=&amp;num=10&amp;btnG=Search+Scholar&amp;as_epq=&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=%22A%20Bourret%22&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;as_allsubj=all&amp;hl=en\" data-track=\"click\" data-track-action=\"author link - scholar\" data-track-label=\"link\" rel=\"nofollow\">Google Scholar</a></span></p></div></div></li><li id=\"auth-D-Garant\"><span class=\"c-article-authors-search__title u-h3 js-search-name\">D Garant</span><div class=\"c-article-authors-search__list\"><div class=\"c-article-authors-search__item c-article-authors-search__list-item--left\"><a href=\"/search?author=D%20Garant\" class=\"c-article-button\" data-track=\"click\" data-track-action=\"author link - publication\" data-track-label=\"link\" rel=\"nofollow\">View author publications</a></div><div class=\"c-article-authors-search__item c-article-authors-search__list-item--right\"><p class=\"search-in-title-js c-article-authors-search__text\">You can also search for this author in\\n                        <span class=\"c-article-identifiers\"><a class=\"c-article-identifiers__item\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&amp;term=D%20Garant\" data-track=\"click\" data-track-action=\"author link - pubmed\" data-track-label=\"link\" rel=\"nofollow\">PubMed</a><span class=\"u-hide\">\\xc2\\xa0</span><a class=\"c-article-identifiers__item\" href=\"http://scholar.google.co.uk/scholar?as_q=&amp;num=10&amp;btnG=Search+Scholar&amp;as_epq=&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=%22D%20Garant%22&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;as_allsubj=all&amp;hl=en\" data-track=\"click\" data-track-action=\"author link - scholar\" data-track-label=\"link\" rel=\"nofollow\">Google Scholar</a></span></p></div></div></li></ol></div><h3 class=\"c-article__sub-heading\" id=\"corresponding-author\">Corresponding author</h3><p id=\"corresponding-author-list\">Correspondence to\\n                <a id=\"corresp-c1\" href=\"mailto:Audrey.Bourret@USherbrooke.ca\">A Bourret</a>.</p></div></div></section><section data-title=\"Ethics declarations\"><div class=\"c-article-section\" id=\"ethics-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"ethics\">Ethics declarations</h2><div class=\"c-article-section__content\" id=\"ethics-content\">\\n              \\n                <h3 class=\"c-article__sub-heading\">Competing interests</h3>\\n                <p>The authors declare no conflict of interest.</p>\\n              \\n            </div></div></section><section data-title=\"Additional information\"><div class=\"c-article-section\" id=\"additional-information-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"additional-information\">Additional information</h2><div class=\"c-article-section__content\" id=\"additional-information-content\"><p><a data-track=\"click\" data-track-label=\"link\" data-track-action=\"supplementary material anchor\" href=\"/articles/hdy201692#MOESM34\">Supplementary Information</a> accompanies this paper on Heredity website</p></div></div></section><section data-title=\"Supplementary information\"><div class=\"c-article-section\" id=\"Sec17-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec17\">Supplementary information</h2><div class=\"c-article-section__content\" id=\"Sec17-content\"><div data-test=\"supplementary-info\"><div id=\"figshareContainer\" class=\"c-article-figshare-container\" data-test=\"figshare-container\"></div><div class=\"c-article-supplementary__item\" data-test=\"supp-item\" id=\"MOESM34\"><h3 class=\"c-article-supplementary__title u-h3\"><a class=\"print-link\" data-track=\"click\" data-track-action=\"view supplementary info\" data-track-label=\"link\" data-test=\"supp-info-link\" href=\"https://static-content.springer.com/esm/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_BFhdy201692_MOESM34_ESM.doc\" data-supp-info-image=\"\">Supplementary Information (DOC 711 kb)</a></h3></div></div></div></div></section><section data-title=\"Rights and permissions\"><div class=\"c-article-section\" id=\"rightslink-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"rightslink\">Rights and permissions</h2><div class=\"c-article-section__content\" id=\"rightslink-content\"><p class=\"c-article-rights\"><a data-track=\"click\" data-track-action=\"view rights and permissions\" data-track-label=\"link\" href=\"https://s100.copyright.com/AppDispatchServlet?title=An%20assessment%20of%20the%20reliability%20of%20quantitative%20genetics%20estimates%20in%20study%20systems%20with%20high%20rate%20of%20extra-pair%20reproduction%20and%20low%20recruitment&amp;author=A%20Bourret%20et%20al&amp;contentID=10.1038%2Fhdy.2016.92&amp;copyright=The%20Genetics%20Society%20Macmillan%20Publishers%20Limited%2C%20part%20of%20Springer%20Nature.&amp;publication=0018-067X&amp;publicationDate=2016-10-26&amp;publisherName=SpringerNature&amp;orderBeanReset=true\">Reprints and Permissions</a></p></div></div></section><section aria-labelledby=\"article-info\" data-title=\"About this article\"><div class=\"c-article-section\" id=\"article-info-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"article-info\">About this article</h2><div class=\"c-article-section__content\" id=\"article-info-content\"><div class=\"c-bibliographic-information\"><div class=\"u-hide-print c-bibliographic-information__column c-bibliographic-information__column--border\"><a data-crossmark=\"10.1038/hdy.2016.92\" target=\"_blank\" rel=\"noopener\" href=\"https://crossmark.crossref.org/dialog/?doi=10.1038/hdy.2016.92\" data-track=\"click\" data-track-action=\"Click Crossmark\" data-track-label=\"link\" data-test=\"crossmark\"><img width=\"57\" height=\"81\" alt=\"Verify currency and authenticity via CrossMark\" src=\"data:image/svg+xml;base64,<svg height="81" width="57" xmlns="http://www.w3.org/2000/svg"><g fill="none" fill-rule="evenodd"><path d="m17.35 35.45 21.3-14.2v-17.03h-21.3" fill="#989898"/><path d="m38.65 35.45-21.3-14.2v-17.03h21.3" fill="#747474"/><path d="m28 .5c-12.98 0-23.5 10.52-23.5 23.5s10.52 23.5 23.5 23.5 23.5-10.52 23.5-23.5c0-6.23-2.48-12.21-6.88-16.62-4.41-4.4-10.39-6.88-16.62-6.88zm0 41.25c-9.8 0-17.75-7.95-17.75-17.75s7.95-17.75 17.75-17.75 17.75 7.95 17.75 17.75c0 4.71-1.87 9.22-5.2 12.55s-7.84 5.2-12.55 5.2z" fill="#535353"/><path d="m41 36c-5.81 6.23-15.23 7.45-22.43 2.9-7.21-4.55-10.16-13.57-7.03-21.5l-4.92-3.11c-4.95 10.7-1.19 23.42 8.78 29.71 9.97 6.3 23.07 4.22 30.6-4.86z" fill="#9c9c9c"/><path d="m.2 58.45c0-.75.11-1.42.33-2.01s.52-1.09.91-1.5c.38-.41.83-.73 1.34-.94.51-.22 1.06-.32 1.65-.32.56 0 1.06.11 1.51.35.44.23.81.5 1.1.81l-.91 1.01c-.24-.24-.49-.42-.75-.56-.27-.13-.58-.2-.93-.2-.39 0-.73.08-1.05.23-.31.16-.58.37-.81.66-.23.28-.41.63-.53 1.04-.13.41-.19.88-.19 1.39 0 1.04.23 1.86.68 2.46.45.59 1.06.88 1.84.88.41 0 .77-.07 1.07-.23s.59-.39.85-.68l.91 1c-.38.43-.8.76-1.28.99-.47.22-1 .34-1.58.34-.59 0-1.13-.1-1.64-.31-.5-.2-.94-.51-1.31-.91-.38-.4-.67-.9-.88-1.48-.22-.59-.33-1.26-.33-2.02zm8.4-5.33h1.61v2.54l-.05 1.33c.29-.27.61-.51.96-.72s.76-.31 1.24-.31c.73 0 1.27.23 1.61.71.33.47.5 1.14.5 2.02v4.31h-1.61v-4.1c0-.57-.08-.97-.25-1.21-.17-.23-.45-.35-.83-.35-.3 0-.56.08-.79.22-.23.15-.49.36-.78.64v4.8h-1.61zm7.37 6.45c0-.56.09-1.06.26-1.51.18-.45.42-.83.71-1.14.29-.3.63-.54 1.01-.71.39-.17.78-.25 1.18-.25.47 0 .88.08 1.23.24.36.16.65.38.89.67s.42.63.54 1.03c.12.41.18.84.18 1.32 0 .32-.02.57-.07.76h-4.36c.07.62.29 1.1.65 1.44.36.33.82.5 1.38.5.29 0 .57-.04.83-.13s.51-.21.76-.37l.55 1.01c-.33.21-.69.39-1.09.53-.41.14-.83.21-1.26.21-.48 0-.92-.08-1.34-.25-.41-.16-.76-.4-1.07-.7-.31-.31-.55-.69-.72-1.13-.18-.44-.26-.95-.26-1.52zm4.6-.62c0-.55-.11-.98-.34-1.28-.23-.31-.58-.47-1.06-.47-.41 0-.77.15-1.07.45-.31.29-.5.73-.58 1.3zm2.5.62c0-.57.09-1.08.28-1.53.18-.44.43-.82.75-1.13s.69-.54 1.1-.71c.42-.16.85-.24 1.31-.24.45 0 .84.08 1.17.23s.61.34.85.57l-.77 1.02c-.19-.16-.38-.28-.56-.37-.19-.09-.39-.14-.61-.14-.56 0-1.01.21-1.35.63-.35.41-.52.97-.52 1.67 0 .69.17 1.24.51 1.66.34.41.78.62 1.32.62.28 0 .54-.06.78-.17.24-.12.45-.26.64-.42l.67 1.03c-.33.29-.69.51-1.08.65-.39.15-.78.23-1.18.23-.46 0-.9-.08-1.31-.24-.4-.16-.75-.39-1.05-.7s-.53-.69-.7-1.13c-.17-.45-.25-.96-.25-1.53zm6.91-6.45h1.58v6.17h.05l2.54-3.16h1.77l-2.35 2.8 2.59 4.07h-1.75l-1.77-2.98-1.08 1.23v1.75h-1.58zm13.69 1.27c-.25-.11-.5-.17-.75-.17-.58 0-.87.39-.87 1.16v.75h1.34v1.27h-1.34v5.6h-1.61v-5.6h-.92v-1.2l.92-.07v-.72c0-.35.04-.68.13-.98.08-.31.21-.57.4-.79s.42-.39.71-.51c.28-.12.63-.18 1.04-.18.24 0 .48.02.69.07.22.05.41.1.57.17zm.48 5.18c0-.57.09-1.08.27-1.53.17-.44.41-.82.72-1.13.3-.31.65-.54 1.04-.71.39-.16.8-.24 1.23-.24s.84.08 1.24.24c.4.17.74.4 1.04.71s.54.69.72 1.13c.19.45.28.96.28 1.53s-.09 1.08-.28 1.53c-.18.44-.42.82-.72 1.13s-.64.54-1.04.7-.81.24-1.24.24-.84-.08-1.23-.24-.74-.39-1.04-.7c-.31-.31-.55-.69-.72-1.13-.18-.45-.27-.96-.27-1.53zm1.65 0c0 .69.14 1.24.43 1.66.28.41.68.62 1.18.62.51 0 .9-.21 1.19-.62.29-.42.44-.97.44-1.66 0-.7-.15-1.26-.44-1.67-.29-.42-.68-.63-1.19-.63-.5 0-.9.21-1.18.63-.29.41-.43.97-.43 1.67zm6.48-3.44h1.33l.12 1.21h.05c.24-.44.54-.79.88-1.02.35-.24.7-.36 1.07-.36.32 0 .59.05.78.14l-.28 1.4-.33-.09c-.11-.01-.23-.02-.38-.02-.27 0-.56.1-.86.31s-.55.58-.77 1.1v4.2h-1.61zm-47.87 15h1.61v4.1c0 .57.08.97.25 1.2.17.24.44.35.81.35.3 0 .57-.07.8-.22.22-.15.47-.39.73-.73v-4.7h1.61v6.87h-1.32l-.12-1.01h-.04c-.3.36-.63.64-.98.86-.35.21-.76.32-1.24.32-.73 0-1.27-.24-1.61-.71-.33-.47-.5-1.14-.5-2.02zm9.46 7.43v2.16h-1.61v-9.59h1.33l.12.72h.05c.29-.24.61-.45.97-.63.35-.17.72-.26 1.1-.26.43 0 .81.08 1.15.24.33.17.61.4.84.71.24.31.41.68.53 1.11.13.42.19.91.19 1.44 0 .59-.09 1.11-.25 1.57-.16.47-.38.85-.65 1.16-.27.32-.58.56-.94.73-.35.16-.72.25-1.1.25-.3 0-.6-.07-.9-.2s-.59-.31-.87-.56zm0-2.3c.26.22.5.37.73.45.24.09.46.13.66.13.46 0 .84-.2 1.15-.6.31-.39.46-.98.46-1.77 0-.69-.12-1.22-.35-1.61-.23-.38-.61-.57-1.13-.57-.49 0-.99.26-1.52.77zm5.87-1.69c0-.56.08-1.06.25-1.51.16-.45.37-.83.65-1.14.27-.3.58-.54.93-.71s.71-.25 1.08-.25c.39 0 .73.07 1 .2.27.14.54.32.81.55l-.06-1.1v-2.49h1.61v9.88h-1.33l-.11-.74h-.06c-.25.25-.54.46-.88.64-.33.18-.69.27-1.06.27-.87 0-1.56-.32-2.07-.95s-.76-1.51-.76-2.65zm1.67-.01c0 .74.13 1.31.4 1.7.26.38.65.58 1.15.58.51 0 .99-.26 1.44-.77v-3.21c-.24-.21-.48-.36-.7-.45-.23-.08-.46-.12-.7-.12-.45 0-.82.19-1.13.59-.31.39-.46.95-.46 1.68zm6.35 1.59c0-.73.32-1.3.97-1.71.64-.4 1.67-.68 3.08-.84 0-.17-.02-.34-.07-.51-.05-.16-.12-.3-.22-.43s-.22-.22-.38-.3c-.15-.06-.34-.1-.58-.1-.34 0-.68.07-1 .2s-.63.29-.93.47l-.59-1.08c.39-.24.81-.45 1.28-.63.47-.17.99-.26 1.54-.26.86 0 1.51.25 1.93.76s.63 1.25.63 2.21v4.07h-1.32l-.12-.76h-.05c-.3.27-.63.48-.98.66s-.73.27-1.14.27c-.61 0-1.1-.19-1.48-.56-.38-.36-.57-.85-.57-1.46zm1.57-.12c0 .3.09.53.27.67.19.14.42.21.71.21.28 0 .54-.07.77-.2s.48-.31.73-.56v-1.54c-.47.06-.86.13-1.18.23-.31.09-.57.19-.76.31s-.33.25-.41.4c-.09.15-.13.31-.13.48zm6.29-3.63h-.98v-1.2l1.06-.07.2-1.88h1.34v1.88h1.75v1.27h-1.75v3.28c0 .8.32 1.2.97 1.2.12 0 .24-.01.37-.04.12-.03.24-.07.34-.11l.28 1.19c-.19.06-.4.12-.64.17-.23.05-.49.08-.76.08-.4 0-.74-.06-1.02-.18-.27-.13-.49-.3-.67-.52-.17-.21-.3-.48-.37-.78-.08-.3-.12-.64-.12-1.01zm4.36 2.17c0-.56.09-1.06.27-1.51s.41-.83.71-1.14c.29-.3.63-.54 1.01-.71.39-.17.78-.25 1.18-.25.47 0 .88.08 1.23.24.36.16.65.38.89.67s.42.63.54 1.03c.12.41.18.84.18 1.32 0 .32-.02.57-.07.76h-4.37c.08.62.29 1.1.65 1.44.36.33.82.5 1.38.5.3 0 .58-.04.84-.13.25-.09.51-.21.76-.37l.54 1.01c-.32.21-.69.39-1.09.53s-.82.21-1.26.21c-.47 0-.92-.08-1.33-.25-.41-.16-.77-.4-1.08-.7-.3-.31-.54-.69-.72-1.13-.17-.44-.26-.95-.26-1.52zm4.61-.62c0-.55-.11-.98-.34-1.28-.23-.31-.58-.47-1.06-.47-.41 0-.77.15-1.08.45-.31.29-.5.73-.57 1.3zm3.01 2.23c.31.24.61.43.92.57.3.13.63.2.98.2.38 0 .65-.08.83-.23s.27-.35.27-.6c0-.14-.05-.26-.13-.37-.08-.1-.2-.2-.34-.28-.14-.09-.29-.16-.47-.23l-.53-.22c-.23-.09-.46-.18-.69-.3-.23-.11-.44-.24-.62-.4s-.33-.35-.45-.55c-.12-.21-.18-.46-.18-.75 0-.61.23-1.1.68-1.49.44-.38 1.06-.57 1.83-.57.48 0 .91.08 1.29.25s.71.36.99.57l-.74.98c-.24-.17-.49-.32-.73-.42-.25-.11-.51-.16-.78-.16-.35 0-.6.07-.76.21-.17.15-.25.33-.25.54 0 .14.04.26.12.36s.18.18.31.26c.14.07.29.14.46.21l.54.19c.23.09.47.18.7.29s.44.24.64.4c.19.16.34.35.46.58.11.23.17.5.17.82 0 .3-.06.58-.17.83-.12.26-.29.48-.51.68-.23.19-.51.34-.84.45-.34.11-.72.17-1.15.17-.48 0-.95-.09-1.41-.27-.46-.19-.86-.41-1.2-.68z" fill="#535353"/></g></svg>\"></a></div><div class=\"c-bibliographic-information__column\"><h3 class=\"c-article__sub-heading\" id=\"citeas\">Cite this article</h3><p class=\"c-bibliographic-information__citation\">Bourret, A., Garant, D. An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment.\\n                    <i>Heredity</i> <b>118</b>, 229\\xe2\\x80\\x93238 (2017). https://doi.org/10.1038/hdy.2016.92</p><p class=\"c-bibliographic-information__download-citation u-hide-print\"><a data-test=\"citation-link\" data-track=\"click\" data-track-action=\"download article citation\" data-track-label=\"link\" data-track-external=\"\" href=\"https://citation-needed.springer.com/v2/references/10.1038/hdy.2016.92?format=refman&amp;flavour=citation\">Download citation<svg width=\"16\" height=\"16\" focusable=\"false\" role=\"img\" aria-hidden=\"true\" class=\"u-icon\"><use xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"#global-icon-download\"></use></svg></a></p><ul class=\"c-bibliographic-information__list\" data-test=\"publication-history\"><li class=\"c-bibliographic-information__list-item\"><p>Received<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2016-02-24\">24 February 2016</time></span></p></li><li class=\"c-bibliographic-information__list-item\"><p>Revised<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2016-06-26\">26 June 2016</time></span></p></li><li class=\"c-bibliographic-information__list-item\"><p>Accepted<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2016-08-19\">19 August 2016</time></span></p></li><li class=\"c-bibliographic-information__list-item\"><p>Published<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2016-10-26\">26 October 2016</time></span></p></li><li class=\"c-bibliographic-information__list-item\"><p>Issue Date<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2017-03\">March 2017</time></span></p></li><li class=\"c-bibliographic-information__list-item c-bibliographic-information__list-item--doi\"><p><abbr title=\"Digital Object Identifier\">DOI</abbr><span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\">https://doi.org/10.1038/hdy.2016.92</span></p></li></ul><div data-component=\"share-box\"><div class=\"c-article-share-box u-display-none\" hidden=\"\"><h3 class=\"c-article__sub-heading\">Share this article</h3><p class=\"c-article-share-box__description\">Anyone you share the following link with will be able to read this content:</p><button class=\"js-get-share-url c-article-share-box__button\" id=\"get-share-url\" data-track=\"click\" data-track-label=\"button\" data-track-external=\"\" data-track-action=\"get shareable link\">Get shareable link</button><div class=\"js-no-share-url-container u-display-none\" hidden=\"\"><p class=\"js-c-article-share-box__no-sharelink-info c-article-share-box__no-sharelink-info\">Sorry, a shareable link is not currently available for this article.</p></div><div class=\"js-share-url-container u-display-none\" hidden=\"\"><p class=\"js-share-url c-article-share-box__only-read-input\" id=\"share-url\" data-track=\"click\" data-track-label=\"button\" data-track-action=\"select share url\"></p><button class=\"js-copy-share-url c-article-share-box__button--link-like\" id=\"copy-share-url\" data-track=\"click\" data-track-label=\"button\" data-track-action=\"copy share url\" data-track-external=\"\">Copy to clipboard</button></div><p class=\"js-c-article-share-box__additional-info c-article-share-box__additional-info\">\\n                            Provided by the Springer Nature SharedIt content-sharing initiative\\n                        </p></div></div><div data-component=\"article-info-list\"></div></div></div></div></div></section>\\n            </div>\\n\\n            \\n        <section>\\n            <div class=\"c-article-section js-article-section\" id=\"further-reading-section\">\\n                <h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"further-reading\">This article is cited by</h2>\\n                <div class=\"c-article-section__content js-collapsible-section\" id=\"further-reading-content\">\\n                    <ul class=\"c-article-further-reading__list\" id=\"further-reading-list\">\\n                        \\n                            <li class=\"c-article-further-reading__item js-ref-item\">\\n                                <h3 class=\"c-article-further-reading__title\">\\n                                    <a class=\"print-link\" data-track=\"click\" data-track-action=\"view further reading article\"\\n                                       data-track-label=\"link:Heritability of plumage colour morph variation in a wild population of promiscuous, long-lived Australian magpies\" href=\"https://doi.org/10.1038/s41437-019-0212-4\">\\n                                        Heritability of plumage colour morph variation in a wild population of promiscuous, long-lived Australian magpies\\n                                    </a>\\n                                </h3>\\n                                \\n                                    <ul data-test=\"author-list\" class=\"c-author-list c-author-list--compact u-sans-serif u-mb-4 u-mt-auto\">\\n                                        <li>Ana E. Dobson</li><li>Daniel J. Schmidt</li><li>Jane M. Hughes</li>\\n                                    </ul>\\n                                \\n                                <p class=\"c-article-further-reading__journal-title\"><i>Heredity</i> (2019)</p>\\n                            </li>\\n                        \\n                    </ul>\\n                </div>\\n            </div>\\n        </section>\\n    \\n\\n            \\n\\n            <span data-recommended=\"jobs\"></span>\\n</div>\\n</article>\\n</main>\\n\\n<aside class=\"c-article-extras u-hide-print\" aria-label=\"Article navigation\" data-component-reading-companion data-container-type=\"reading-companion\" data-track-component=\"reading companion\">\\n    <div class=\"js-context-bar-sticky-point-desktop\">\\n        \\n\\n        \\n            \\n                \\n    \\n        <div class=\"c-pdf-download u-clear-both js-pdf-download\">\\n            <a href=\"/articles/hdy201692.pdf\" class=\"u-button u-button--full-width u-button--primary u-justify-content-space-between c-pdf-download__link\" data-article-pdf=\"true\" data-readcube-pdf-url=\"true\" data-test=\"download-pdf\" data-draft-ignore=\"true\" data-track=\"click\" data-track-action=\"download pdf\" data-track-label=\"link\" data-track-external download>\\n                <span class=\"c-pdf-download__text\">Download PDF</span>\\n                <svg aria-hidden=\"true\" focusable=\"false\" width=\"16\" height=\"16\" class=\"u-icon\"><use xlink:href=\"#global-icon-download\"/></svg>\\n            </a>\\n        </div>\\n    \\n\\n            \\n        \\n    </div>\\n\\n    \\n        \\n    \\n\\n    \\n\\n    <div class=\"c-reading-companion\">\\n        <div class=\"c-reading-companion__sticky\" data-component=\"reading-companion-sticky\" data-test=\"reading-companion-sticky\">\\n            <div class=\"c-reading-companion__panel c-reading-companion__sections c-reading-companion__panel--active\" id=\"tabpanel-sections\">\\n                <div class=\"u-lazy-ad-wrapper u-mt-16 u-hide\" data-component-mpu>\\n                    <div class=\"c-ad c-ad--300x250\">\\n                        <div class=\"c-ad__inner\">\\n                            <p class=\"c-ad__label\">Advertisement</p>\\n                            \\n    <div id=\"div-gpt-ad-right-2\"\\n         class=\"div-gpt-ad advert medium-rectangle js-ad text-center hide-print grade-c-hide\"\\n         data-ad-type=\"right\"\\n         data-test=\"right-ad\"\\n         data-pa11y-ignore\\n         data-gpt\\n         data-gpt-unitpath=\"/285/hdy.nature.com/article\"\\n         data-gpt-sizes=\"300x250\"\\n         data-gpt-targeting=\"type=article;pos=right;artid=hdy201692;doi=10.1038/hdy.2016.92;techmeta=23,29,45,77;subjmeta=181,2474,631;kwrd=Evolutionary+genetics\">\\n        <noscript>\\n            <a href=\"//pubads.g.doubleclick.net/gampad/jump?iu=/285/hdy.nature.com/article&amp;sz=300x250&amp;c=-2128937448&amp;t=pos%3Dright%26type%3Darticle%26artid%3Dhdy201692%26doi%3D10.1038/hdy.2016.92%26techmeta%3D23,29,45,77%26subjmeta%3D181,2474,631%26kwrd%3DEvolutionary+genetics\">\\n                <img data-test=\"gpt-advert-fallback-img\"\\n                     src=\"//pubads.g.doubleclick.net/gampad/ad?iu=/285/hdy.nature.com/article&amp;sz=300x250&amp;c=-2128937448&amp;t=pos%3Dright%26type%3Darticle%26artid%3Dhdy201692%26doi%3D10.1038/hdy.2016.92%26techmeta%3D23,29,45,77%26subjmeta%3D181,2474,631%26kwrd%3DEvolutionary+genetics\"\\n                     alt=\"Advertisement\"\\n                     width=\"300\"\\n                     height=\"250\"></a>\\n        </noscript>\\n    </div>\\n\\n                        </div>\\n                    </div>\\n                </div>\\n            </div>\\n            <div class=\"c-reading-companion__panel c-reading-companion__figures c-reading-companion__panel--full-width\" id=\"tabpanel-figures\"></div>\\n            <div class=\"c-reading-companion__panel c-reading-companion__references c-reading-companion__panel--full-width\" id=\"tabpanel-references\"></div>\\n        </div>\\n    </div>\\n</aside>\\n</div>\\n\\n\\n    \\n        <nav class=\"c-header__dropdown\" aria-labelledby=\"Explore-content\" data-test=\"Explore-content\" id=\"explore\" data-track-component=\"nature-150-split-header\">\\n            <div class=\"c-header__container\">\\n                <h2 id=\"Explore-content\" class=\"c-header__heading c-header__heading--js-hide\">Explore content</h2>\\n                <ul class=\"c-header__list c-header__list--js-stack\">\\n                    \\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/research-articles\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"research articles\"\\n                                   data-track-label=\"link\"\\n                                   data-test=\"explore-nav-item\">\\n                                    Research articles\\n                                </a>\\n                            </li>\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/reviews-and-analysis\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"reviews &amp; analysis\"\\n                                   data-track-label=\"link\"\\n                                   data-test=\"explore-nav-item\">\\n                                    Reviews &amp; Analysis\\n                                </a>\\n                            </li>\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/news-and-comment\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"news &amp; comment\"\\n                                   data-track-label=\"link\"\\n                                   data-test=\"explore-nav-item\">\\n                                    News &amp; Comment\\n                                </a>\\n                            </li>\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/podcast\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"podcasts\"\\n                                   data-track-label=\"link\"\\n                                   data-test=\"explore-nav-item\">\\n                                    Podcasts\\n                                </a>\\n                            </li>\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/current-issue\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"current issue\"\\n                                   data-track-label=\"link\"\\n                                   data-test=\"explore-nav-item\">\\n                                    Current issue\\n                                </a>\\n                            </li>\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/collections\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"collections\"\\n                                   data-track-label=\"link\"\\n                                   data-test=\"explore-nav-item\">\\n                                    Collections\\n                                </a>\\n                            </li>\\n                        \\n                    \\n                </ul>\\n                <ul class=\"c-header__list c-header__list--js-stack\">\\n                    \\n                    \\n                        <li class=\"c-header__item\">\\n                            <a class=\"c-header__link\"\\n                               href=\"https://twitter.com/HeredityJournal\"\\n                               data-track=\"click\"\\n                               data-track-action=\"twitter\"\\n                               data-track-label=\"link\">Follow us on Twitter\\n                            </a>\\n                        </li>\\n                    \\n                    \\n                    \\n                        <li class=\"c-header__item c-header__item--hide-lg\">\\n                            <a class=\"c-header__link\"\\n                               href=\"https://www.nature.com/my-account/alerts/subscribe-journal?list-id&#x3D;40\"\\n                               rel=\"nofollow\"\\n                               data-track=\"click\"\\n                               data-track-action=\"Sign up for alerts\"\\n                               data-track-external\\n                               data-track-label=\"link (mobile dropdown)\">Sign up for alerts<svg role=\"img\" aria-hidden=\"true\" focusable=\"false\" height=\"18\" viewBox=\"0 0 18 18\" width=\"18\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"m4 10h2.5c.27614237 0 .5.2238576.5.5s-.22385763.5-.5.5h-3.08578644l-1.12132034 1.1213203c-.18753638.1875364-.29289322.4418903-.29289322.7071068v.1715729h14v-.1715729c0-.2652165-.1053568-.5195704-.2928932-.7071068l-1.7071068-1.7071067v-3.4142136c0-2.76142375-2.2385763-5-5-5-2.76142375 0-5 2.23857625-5 5zm3 4c0 1.1045695.8954305 2 2 2s2-.8954305 2-2zm-5 0c-.55228475 0-1-.4477153-1-1v-.1715729c0-.530433.21071368-1.0391408.58578644-1.4142135l1.41421356-1.4142136v-3c0-3.3137085 2.6862915-6 6-6s6 2.6862915 6 6v3l1.4142136 1.4142136c.3750727.3750727.5857864.8837805.5857864 1.4142135v.1715729c0 .5522847-.4477153 1-1 1h-4c0 1.6568542-1.3431458 3-3 3-1.65685425 0-3-1.3431458-3-3z\" fill=\"#fff\"/></svg>\\n                            </a>\\n                        </li>\\n                    \\n                    \\n                        <li class=\"c-header__item c-header__item--hide-lg\">\\n                            <a class=\"c-header__link\"\\n                               href=\"https://www.nature.com/hdy.rss\"\\n                               data-track=\"click\"\\n                               data-track-action=\"rss feed\"\\n                               data-track-label=\"link\">\\n                                <span>RSS feed</span>\\n                            </a>\\n                        </li>\\n                    \\n                </ul>\\n            </div>\\n        </nav>\\n    \\n    \\n        \\n            <nav class=\"c-header__dropdown\" aria-labelledby=\"About-the-journal\" id=\"about-the-journal\" data-test=\"about-the-journal\" data-track-component=\"nature-150-split-header\">\\n                <div class=\"c-header__container\">\\n                    <h2 id=\"About-the-journal\" class=\"c-header__heading c-header__heading--js-hide\">About the journal</h2>\\n                    <ul class=\"c-header__list c-header__list--js-stack\">\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/journal-information\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"journal information\"\\n                                   data-track-label=\"link\">\\n                                    Journal Information\\n                                </a>\\n                            </li>\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/open-access\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"open access publishing\"\\n                                   data-track-label=\"link\">\\n                                    Open access publishing\\n                                </a>\\n                            </li>\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/editors\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"about the editors\"\\n                                   data-track-label=\"link\">\\n                                    About the Editors\\n                                </a>\\n                            </li>\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/contact\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"contact\"\\n                                   data-track-label=\"link\">\\n                                    Contact\\n                                </a>\\n                            </li>\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/partner\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"about the partner\"\\n                                   data-track-label=\"link\">\\n                                    About the Partner\\n                                </a>\\n                            </li>\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/advertising\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"for advertisers\"\\n                                   data-track-label=\"link\">\\n                                    For Advertisers\\n                                </a>\\n                            </li>\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/subscribe\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"subscribe\"\\n                                   data-track-label=\"link\">\\n                                    Subscribe\\n                                </a>\\n                            </li>\\n                        \\n                    </ul>\\n                </div>\\n            </nav>\\n        \\n\\n        \\n            <nav class=\"c-header__dropdown\" aria-labelledby=\"Publish-with-us-label\" id=\"publish-with-us\" data-test=\"publish-with-us\" data-track-component=\"nature-150-split-header\">\\n                <div class=\"c-header__container\">\\n                    <h2 id=\"Publish-with-us-label\" class=\"c-header__heading c-header__heading--js-hide\">Publish with us</h2>\\n                    <ul class=\"c-header__list c-header__list--js-stack\">\\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"/hdy/authors-and-referees\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"for authors &amp; referees\"\\n                                   data-track-label=\"link\">\\n                                    For Authors &amp; Referees\\n                                </a>\\n                            </li>\\n                        \\n                        \\n                            <li class=\"c-header__item\">\\n                                <a class=\"c-header__link\" data-test=\"nature-author-services\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"manuscript author services\"\\n                                   data-track-label=\"link manuscript author services\"\\n                                   href=\"https://authorservices.springernature.com/go/sn/?utm_source=For+Authors&utm_medium=Website_Nature&utm_campaign=Platform+Experimentation+2022&utm_id=PE2022\">\\n                                    Language editing services\\n                                </a>\\n                            </li>\\n                        \\n                        \\n                            <li class=\"c-header__item c-header__item--keyline\">\\n                                <a class=\"c-header__link\"\\n                                   href=\"https://mts-hdy.nature.com/cgi-bin/main.plex\"\\n                                   data-track=\"click\"\\n                                   data-track-action=\"submit manuscript\"\\n                                   data-track-label=\"link (publish with us dropdown menu)\"\\n                                   data-track-external>Submit manuscript<svg role=\"img\" aria-hidden=\"true\" focusable=\"false\" height=\"18\" viewBox=\"0 0 18 18\" width=\"18\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"m15 0c1.1045695 0 2 .8954305 2 2v5.5c0 .27614237-.2238576.5-.5.5s-.5-.22385763-.5-.5v-5.5c0-.51283584-.3860402-.93550716-.8833789-.99327227l-.1166211-.00672773h-9v3c0 1.1045695-.8954305 2-2 2h-3v10c0 .5128358.38604019.9355072.88337887.9932723l.11662113.0067277h7.5c.27614237 0 .5.2238576.5.5s-.22385763.5-.5.5h-7.5c-1.1045695 0-2-.8954305-2-2v-10.17157288c0-.53043297.21071368-1.0391408.58578644-1.41421356l3.82842712-3.82842712c.37507276-.37507276.88378059-.58578644 1.41421356-.58578644zm-.5442863 8.18867991 3.3545404 3.35454039c.2508994.2508994.2538696.6596433.0035959.909917-.2429543.2429542-.6561449.2462671-.9065387-.0089489l-2.2609825-2.3045251.0010427 7.2231989c0 .3569916-.2898381.6371378-.6473715.6371378-.3470771 0-.6473715-.2852563-.6473715-.6371378l-.0010428-7.2231995-2.2611222 2.3046654c-.2531661.2580415-.6562868.2592444-.9065605.0089707-.24295423-.2429542-.24865597-.6576651.0036132-.9099343l3.3546673-3.35466731c.2509089-.25090888.6612706-.25227691.9135302-.00001728zm-.9557137-3.18867991c.2761424 0 .5.22385763.5.5s-.2238576.5-.5.5h-6c-.27614237 0-.5-.22385763-.5-.5s.22385763-.5.5-.5zm-8.5-3.587-3.587 3.587h2.587c.55228475 0 1-.44771525 1-1zm8.5 1.587c.2761424 0 .5.22385763.5.5s-.2238576.5-.5.5h-6c-.27614237 0-.5-.22385763-.5-.5s.22385763-.5.5-.5z\" fill=\"#fff\"/></svg>\\n                                </a>\\n                            </li>\\n                        \\n                    </ul>\\n                </div>\\n            </nav>\\n        \\n    \\n\\n\\n<div id=\"search-menu\" class=\"c-header__dropdown c-header__dropdown--full-width\" data-track-component=\"nature-150-split-header\">\\n    <div class=\"c-header__container\">\\n        <h2 class=\"c-header__visually-hidden\">Search</h2>\\n        <form class=\"c-header__search-form\" action=\"/search\" method=\"get\" role=\"search\" autocomplete=\"off\" data-test=\"inline-search\">\\n            <label class=\"c-header__heading\" for=\"keywords\">Search articles by subject, keyword or author</label>\\n            <div class=\"c-header__search-layout c-header__search-layout--max-width\">\\n                <div>\\n                    <input type=\"text\" required=\"\" class=\"c-header__input\" id=\"keywords\" name=\"q\" value=\"\">\\n                </div>\\n                <div class=\"c-header__search-layout\">\\n                    <div>\\n                        <label for=\"results-from\" class=\"c-header__visually-hidden\">Show results from</label>\\n                        <select id=\"results-from\" name=\"journal\" class=\"c-header__select\">\\n                            \\n                                \\n                                    <option value=\"\" selected>All journals</option>\\n                                    <option value=\"hdy\">This journal</option>\\n                                \\n                            \\n                        </select>\\n                    </div>\\n                    <div>\\n                        <button type=\"submit\" class=\"c-header__search-button\">Search</button>\\n                    </div>\\n                </div>\\n\\n            </div>\\n        </form>\\n\\n        <div class=\"c-header__flush\">\\n            <a class=\"c-header__link\" href=\"/search/advanced\"\\n               data-track=\"click\" data-track-action=\"advanced search\" data-track-label=\"link\">\\n                Advanced search\\n            </a>\\n        </div>\\n\\n        <h3 class=\"c-header__heading c-header__heading--keyline\">Quick links</h3>\\n        <ul class=\"c-header__list\">\\n            <li><a class=\"c-header__link\" href=\"/subjects\" data-track=\"click\" data-track-action=\"explore articles by subject\" data-track-label=\"link\">Explore articles by subject</a></li>\\n            <li><a class=\"c-header__link\" href=\"/naturecareers\" data-track=\"click\" data-track-action=\"find a job\" data-track-label=\"link\">Find a job</a></li>\\n            <li><a class=\"c-header__link\" href=\"/authors/index.html\" data-track=\"click\" data-track-action=\"guide to authors\" data-track-label=\"link\">Guide to authors</a></li>\\n            <li><a class=\"c-header__link\" href=\"/authors/editorial_policies/\" data-track=\"click\" data-track-action=\"editorial policies\" data-track-label=\"link\">Editorial policies</a></li>\\n        </ul>\\n    </div>\\n</div>\\n\\n<footer class=\"composite-layer\" itemscope itemtype=\"http://schema.org/Periodical\">\\n        <meta itemprop=\"publisher\" content=\"Springer Nature\">\\n        \\n\\n        <div class=\"u-mt-16 u-mb-16\">\\n    <div class=\"u-container\">\\n        <div class=\"u-display-flex u-flex-wrap u-justify-content-space-between\">\\n            \\n\\n            <p class=\"c-meta u-ma-0 u-flex-shrink\">\\n                <span class=\"c-meta__item\">\\n                    Heredity (<i>Heredity</i>)\\n                </span>\\n                \\n    \\n    <span class=\"c-meta__item\">\\n        <abbr title=\"International Standard Serial Number\">ISSN</abbr> <span itemprop=\"onlineIssn\">1365-2540</span> (online)\\n    </span>\\n    \\n\\n\\n                \\n    \\n    <span class=\"c-meta__item\">\\n        <abbr title=\"International Standard Serial Number\">ISSN</abbr> <span itemprop=\"printIssn\">0018-067X</span> (print)\\n    </span>\\n    \\n\\n            </p>\\n        </div>\\n    </div>\\n</div>\\n\\n    <div class=\"c-footer\">\\n        <div class=\"u-hide-print\" data-track-component=\"footer\">\\n    <h2 class=\"u-visually-hidden\">nature.com sitemap</h2>\\n    <div class=\"c-footer__container\">\\n        <div class=\"c-footer__grid c-footer__group--separator\">\\n            <div class=\"c-footer__group\">\\n                <h3 class=\"c-footer__heading\">About Nature Portfolio</h3>\\n                <ul class=\"c-footer__list\">\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://www.nature.com/npg_/company_info/index.html\"\\n                                                  data-track=\"click\" data-track-action=\"about us\"\\n                                                  data-track-label=\"link\">About us</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://www.nature.com/npg_/press_room/press_releases.html\"\\n                                                  data-track=\"click\" data-track-action=\"press releases\"\\n                                                  data-track-label=\"link\">Press releases</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://press.nature.com/\"\\n                                                  data-track=\"click\" data-track-action=\"press office\"\\n                                                  data-track-label=\"link\">Press office</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://support.nature.com/support/home\"\\n                                                  data-track=\"click\" data-track-action=\"contact us\"\\n                                                  data-track-label=\"link\">Contact us</a></li>\\n                </ul>\\n            </div>\\n\\n            <div class=\"c-footer__group\">\\n                <h3 class=\"c-footer__heading\">Discover content</h3>\\n                <ul class=\"c-footer__list\">\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/siteindex\"\\n                                                  data-track=\"click\" data-track-action=\"journals a-z\"\\n                                                  data-track-label=\"link\">Journals A-Z</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/subjects/\"\\n                                                  data-track=\"click\" data-track-action=\"article by subject\"\\n                                                  data-track-label=\"link\">Articles by subject</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://nano.nature.com/\"\\n                                                  data-track=\"click\" data-track-action=\"nano\" data-track-label=\"link\">Nano</a>\\n                    </li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/protocolexchange/\"\\n                                                  data-track=\"click\" data-track-action=\"protocol exchange\"\\n                                                  data-track-label=\"link\">Protocol Exchange</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.natureindex.com/\"\\n                                                  data-track=\"click\" data-track-action=\"nature index\"\\n                                                  data-track-label=\"link\">Nature Index</a></li>\\n                </ul>\\n            </div>\\n\\n            <div class=\"c-footer__group\">\\n                <h3 class=\"c-footer__heading\">Publishing policies</h3>\\n                <ul class=\"c-footer__list\">\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://www.nature.com/authors/editorial_policies/\"\\n                                                  data-track=\"click\" data-track-action=\"Nature portfolio policies\"\\n                                                  data-track-label=\"link\">Nature portfolio policies</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://www.nature.com/nature-research/open-access\"\\n                                                  data-track=\"click\" data-track-action=\"open access\"\\n                                                  data-track-label=\"link\">Open access</a></li>\\n                </ul>\\n            </div>\\n\\n            <div class=\"c-footer__group\">\\n                <h3 class=\"c-footer__heading\">Author &amp; Researcher services</h3>\\n                <ul class=\"c-footer__list\">\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/reprints/\"\\n                                                  data-track=\"click\" data-track-action=\"reprints and permissions\"\\n                                                  data-track-label=\"link\">Reprints &amp; permissions</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://www.springernature.com/gp/authors/research-data\"\\n                                                  data-track=\"click\" data-track-action=\"data research service\"\\n                                                  data-track-label=\"link\">Research data</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://authorservices.springernature.com/language-editing/\"\\n                                                  data-track=\"click\" data-track-action=\"language editing\"\\n                                                  data-track-label=\"link\">Language editing</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://authorservices.springernature.com/scientific-editing/\"\\n                                                  data-track=\"click\" data-track-action=\"scientific editing\"\\n                                                  data-track-label=\"link\">Scientific editing</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://masterclasses.nature.com/\"\\n                                                  data-track=\"click\" data-track-action=\"nature masterclasses\"\\n                                                  data-track-label=\"link\">Nature Masterclasses</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://partnerships.nature.com/product/researcher-training/\"\\n                                                  data-track=\"click\" data-track-action=\"nature research academies\"\\n                                                  data-track-label=\"link\">Nature Research Academies</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://solutions.springernature.com/\"\\n                                                  data-track=\"click\" data-track-action=\"research solutions\"\\n                                                  data-track-label=\"link\">Research Solutions</a></li>\\n                </ul>\\n            </div>\\n\\n            <div class=\"c-footer__group\">\\n                <h3 class=\"c-footer__heading\">Libraries &amp; institutions</h3>\\n                <ul class=\"c-footer__list\">\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://www.springernature.com/gp/librarians/tools-services\"\\n                                                  data-track=\"click\" data-track-action=\"librarian service and tools\"\\n                                                  data-track-label=\"link\">Librarian service &amp; tools</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://www.springernature.com/gp/librarians/manage-your-account/librarianportal\"\\n                                                  data-track=\"click\" data-track-action=\"librarian portal\"\\n                                                  data-track-label=\"link\">Librarian portal</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://www.nature.com/openresearch/about-open-access/information-for-institutions/\"\\n                                                  data-track=\"click\" data-track-action=\"open research\"\\n                                                  data-track-label=\"link\">Open research</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://www.springernature.com/gp/librarians/recommend-to-your-library\"\\n                                                  data-track=\"click\" data-track-action=\"Recommend to library\"\\n                                                  data-track-label=\"link\">Recommend to library</a></li>\\n                </ul>\\n            </div>\\n\\n            <div class=\"c-footer__group\">\\n                <h3 class=\"c-footer__heading\">Advertising &amp; partnerships</h3>\\n                <ul class=\"c-footer__list\">\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://partnerships.nature.com/product/digital-advertising/\"\\n                                                  data-track=\"click\" data-track-action=\"advertising\"\\n                                                  data-track-label=\"link\">Advertising</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://partnerships.nature.com/\"\\n                                                  data-track=\"click\" data-track-action=\"partnerships and services\"\\n                                                  data-track-label=\"link\">Partnerships &amp; Services</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://partnerships.nature.com/media-kits/\" data-track=\"click\"\\n                                                  data-track-action=\"media kits\" data-track-label=\"link\">Media kits</a>\\n                    </li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                                  href=\"https://partnerships.nature.com/product/branded-content-native-advertising/\"\\n                                                  data-track-action=\"branded content\" data-track-label=\"link\">Branded\\n                        content</a></li>\\n                </ul>\\n            </div>\\n\\n            <div class=\"c-footer__group\">\\n                <h3 class=\"c-footer__heading\">Career development</h3>\\n                <ul class=\"c-footer__list\">\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/naturecareers\"\\n                                                  data-track=\"click\" data-track-action=\"nature careers\"\\n                                                  data-track-label=\"link\">Nature Careers</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://conferences.nature.com\"\\n                                                  data-track=\"click\" data-track-action=\"nature conferences\"\\n                                                  data-track-label=\"link\">Nature<span class=\"u-visually-hidden\"> </span>\\n                        Conferences</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/natureevents/\"\\n                                                  data-track=\"click\" data-track-action=\"nature events\"\\n                                                  data-track-label=\"link\">Nature<span class=\"u-visually-hidden\"> </span>\\n                        events</a></li>\\n                </ul>\\n            </div>\\n\\n            <div class=\"c-footer__group\">\\n                <h3 class=\"c-footer__heading\">Regional websites</h3>\\n                <ul class=\"c-footer__list\">\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/natafrica\"\\n                                                  data-track=\"click\" data-track-action=\"nature africa\"\\n                                                  data-track-label=\"link\">Nature Africa</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"http://www.naturechina.com\"\\n                                                  data-track=\"click\" data-track-action=\"nature china\"\\n                                                  data-track-label=\"link\">Nature China</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/nindia\"\\n                                                  data-track=\"click\" data-track-action=\"nature india\"\\n                                                  data-track-label=\"link\">Nature India</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/natitaly\"\\n                                                  data-track=\"click\" data-track-action=\"nature Italy\"\\n                                                  data-track-label=\"link\">Nature Italy</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.natureasia.com/ja-jp/\"\\n                                                  data-track=\"click\" data-track-action=\"nature japan\"\\n                                                  data-track-label=\"link\">Nature Japan</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.natureasia.com/ko-kr/\"\\n                                                  data-track=\"click\" data-track-action=\"nature korea\"\\n                                                  data-track-label=\"link\">Nature Korea</a></li>\\n                    <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/nmiddleeast/\"\\n                                                  data-track=\"click\" data-track-action=\"nature middle east\"\\n                                                  data-track-label=\"link\">Nature Middle East</a></li>\\n                </ul>\\n            </div>\\n\\n        </div>\\n    </div>\\n    <div class=\"c-footer__container\">\\n        <ul class=\"c-footer__links\">\\n            <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/info/privacy\"\\n                                          data-track=\"click\" data-track-action=\"privacy policy\" data-track-label=\"link\">Privacy\\n                Policy</a></li>\\n            <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/info/cookies\"\\n                                          data-track=\"click\" data-track-action=\"use of cookies\" data-track-label=\"link\">Use\\n                of cookies</a></li>\\n            <li class=\"c-footer__item\">\\n                <button class=\"optanon-toggle-display c-footer__link\" onclick=\"javascript:;\"\\n                        data-cc-action=\"preferences\" data-track=\"click\" data-track-action=\"manage cookies\"\\n                        data-track-label=\"link\">Manage cookies/Do not sell my data\\n                </button>\\n            </li>\\n            <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/info/legal-notice\"\\n                                          data-track=\"click\" data-track-action=\"legal notice\" data-track-label=\"link\">Legal\\n                notice</a></li>\\n            <li class=\"c-footer__item\"><a class=\"c-footer__link\"\\n                                          href=\"https://www.nature.com/info/accessibility-statement\" data-track=\"click\"\\n                                          data-track-action=\"accessibility statement\" data-track-label=\"link\">Accessibility\\n                statement</a></li>\\n            <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.nature.com/info/terms-and-conditions\"\\n                                          data-track=\"click\" data-track-action=\"terms and conditions\"\\n                                          data-track-label=\"link\">Terms &amp; Conditions</a></li>\\n            <li class=\"c-footer__item\"><a class=\"c-footer__link\" href=\"https://www.springernature.com/ccpa\"\\n                                          data-track=\"click\" data-track-action=\"california privacy statement\"\\n                                          data-track-label=\"link\">California Privacy Statement</a></li>\\n            \\n        </ul>\\n    </div>\\n</div>\\n\\n\\n        <div class=\"c-footer__container\">\\n    <a href=\"https://www.springernature.com/\" class=\"c-footer__link\">\\n        <img src=\"/static/images/logos/sn-logo-white-ea63208b81.svg\" alt=\"Springer Nature\" loading=\"lazy\" width=\"200\" height=\"20\"/>\\n    </a>\\n    <p class=\"c-footer__legal\" data-test=\"copyright\">&copy; 2022 Springer Nature Limited</p>\\n</div>\\n\\n    </div>\\n    \\n    <svg class=\"u-hide hide\">\\n        <symbol id=\"global-icon-chevron-right\" viewBox=\"0 0 10 10\" xmlns=\"http://www.w3.org/2000/svg\">\\n            <path d=\"m5.96738168 4.70639573 2.39518594-2.41447274c.37913917-.38219212.98637524-.38972225 1.35419292-.01894278.37750606.38054586.37784436.99719163-.00013556 1.37821513l-4.03074001 4.06319683c-.37758093.38062133-.98937525.38100976-1.367372-.00003075l-4.03091981-4.06337806c-.37759778-.38063832-.38381821-.99150444-.01600053-1.3622839.37750607-.38054587.98772445-.38240057 1.37006824.00302197l2.39538588 2.4146743.96295325.98624457z\" fill=\"currentColor\" fill-rule=\"evenodd\" transform=\"matrix(0 -1 1 0 0 10)\"/>\\n        </symbol>\\n        <symbol id=\"global-icon-download\" viewBox=\"0 0 16 16\">\\n            <path d=\"M2 14c0-.556.449-1 1.002-1h9.996a.999.999 0 110 2H3.002A1.006 1.006 0 012 14zM9 2v6.8l2.482-2.482c.392-.392 1.022-.4 1.403-.02a1.001 1.001 0 010 1.417l-4.177 4.177a1.001 1.001 0 01-1.416 0L3.115 7.715a.991.991 0 01-.016-1.4 1 1 0 011.42.003L7 8.8V2c0-.55.444-.996 1-.996.552 0 1 .445 1 .996z\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"global-icon-email\" viewBox=\"0 0 18 18\">\\n            <path d=\"M1.995 2h14.01A2 2 0 0118 4.006v9.988A2 2 0 0116.005 16H1.995A2 2 0 010 13.994V4.006A2 2 0 011.995 2zM1 13.994A1 1 0 001.995 15h14.01A1 1 0 0017 13.994V4.006A1 1 0 0016.005 3H1.995A1 1 0 001 4.006zM9 11L2 7V5.557l7 4 7-4V7z\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"global-icon-institution\" viewBox=\"0 0 18 18\">\\n            <path d=\"M14 8a1 1 0 011 1v6h1.5a.5.5 0 01.5.5v.5h.5a.5.5 0 01.5.5V18H0v-1.5a.5.5 0 01.5-.5H1v-.5a.5.5 0 01.5-.5H3V9a1 1 0 112 0v6h8V9a1 1 0 011-1zM6 8l2 1v4l-2 1zm6 0v6l-2-1V9zM9.573.401l7.036 4.925A.92.92 0 0116.081 7H1.92a.92.92 0 01-.528-1.674L8.427.401a1 1 0 011.146 0zM9 2.441L5.345 5h7.31z\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"global-icon-search\" viewBox=\"0 0 22 22\">\\n            <path fill-rule=\"evenodd\" d=\"M21.697 20.261a1.028 1.028 0 01.01 1.448 1.034 1.034 0 01-1.448-.01l-4.267-4.267A9.812 9.811 0 010 9.812a9.812 9.811 0 1117.43 6.182zM9.812 18.222A8.41 8.41 0 109.81 1.403a8.41 8.41 0 000 16.82z\"/>\\n        </symbol>\\n        <symbol id=\"icon-info\" viewBox=\"0 0 18 18\">\\n            <path d=\"m9 0c4.9705627 0 9 4.02943725 9 9 0 4.9705627-4.0294373 9-9 9-4.97056275 0-9-4.0294373-9-9 0-4.97056275 4.02943725-9 9-9zm0 7h-1.5l-.11662113.00672773c-.49733868.05776511-.88337887.48043643-.88337887.99327227 0 .47338693.32893365.86994729.77070917.97358929l.1126697.01968298.11662113.00672773h.5v3h-.5l-.11662113.0067277c-.42082504.0488782-.76196299.3590206-.85696816.7639815l-.01968298.1126697-.00672773.1166211.00672773.1166211c.04887817.4208251.35902055.761963.76398144.8569682l.1126697.019683.11662113.0067277h3l.1166211-.0067277c.4973387-.0577651.8833789-.4804365.8833789-.9932723 0-.4733869-.3289337-.8699473-.7707092-.9735893l-.1126697-.019683-.1166211-.0067277h-.5v-4l-.00672773-.11662113c-.04887817-.42082504-.35902055-.76196299-.76398144-.85696816l-.1126697-.01968298zm0-3.25c-.69035594 0-1.25.55964406-1.25 1.25s.55964406 1.25 1.25 1.25 1.25-.55964406 1.25-1.25-.55964406-1.25-1.25-1.25z\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"icon-success\" viewBox=\"0 0 18 18\">\\n            <path d=\"M9 0a9 9 0 110 18A9 9 0 019 0zm3.486 4.982l-4.718 5.506L5.14 8.465a.991.991 0 00-1.423.133 1.06 1.06 0 00.13 1.463l3.407 2.733a1 1 0 001.387-.133l5.385-6.334a1.06 1.06 0 00-.116-1.464.991.991 0 00-1.424.119z\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"icon-chevron-down\" viewBox=\"0 0 16 16\">\\n            <path d=\"m5.58578644 3-3.29289322-3.29289322c-.39052429-.39052429-.39052429-1.02368927 0-1.41421356s1.02368927-.39052429 1.41421356 0l4 4c.39052429.39052429.39052429 1.02368927 0 1.41421356l-4 4c-.39052429.39052429-1.02368927.39052429-1.41421356 0s-.39052429-1.02368927 0-1.41421356z\" fill-rule=\"evenodd\" transform=\"matrix(0 1 -1 0 11 1)\"/>\\n        </symbol>\\n        <symbol id=\"icon-warning\" viewBox=\"0 0 18 18\">\\n            <path d=\"m9 11.75c.69035594 0 1.25.5596441 1.25 1.25s-.55964406 1.25-1.25 1.25-1.25-.5596441-1.25-1.25.55964406-1.25 1.25-1.25zm.41320045-7.75c.55228475 0 1.00000005.44771525 1.00000005 1l-.0034543.08304548-.3333333 4c-.043191.51829212-.47645714.91695452-.99654578.91695452h-.15973424c-.52008864 0-.95335475-.3986624-.99654576-.91695452l-.33333333-4c-.04586475-.55037702.36312325-1.03372649.91350028-1.07959124l.04148683-.00259031zm-.41320045 14c-4.97056275 0-9-4.0294373-9-9 0-4.97056275 4.02943725-9 9-9 4.9705627 0 9 4.02943725 9 9 0 4.9705627-4.0294373 9-9 9z\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"icon-plus\" viewBox=\"0 0 16 16\">\\n            <path d=\"m2.00087166 7h4.99912834v-4.99912834c0-.55276616.44386482-1.00087166 1-1.00087166.55228475 0 1 .44463086 1 1.00087166v4.99912834h4.9991283c.5527662 0 1.0008717.44386482 1.0008717 1 0 .55228475-.4446309 1-1.0008717 1h-4.9991283v4.9991283c0 .5527662-.44386482 1.0008717-1 1.0008717-.55228475 0-1-.4446309-1-1.0008717v-4.9991283h-4.99912834c-.55276616 0-1.00087166-.44386482-1.00087166-1 0-.55228475.44463086-1 1.00087166-1z\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"icon-minus\" viewBox=\"0 0 16 16\">\\n            <path d=\"m2.00087166 7h11.99825664c.5527662 0 1.0008717.44386482 1.0008717 1 0 .55228475-.4446309 1-1.0008717 1h-11.99825664c-.55276616 0-1.00087166-.44386482-1.00087166-1 0-.55228475.44463086-1 1.00087166-1z\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"icon-error\" viewBox=\"0 0 18 18\">\\n            <path d=\"m9 0c4.9705627 0 9 4.02943725 9 9 0 4.9705627-4.0294373 9-9 9-4.97056275 0-9-4.0294373-9-9 0-4.97056275 4.02943725-9 9-9zm2.8630343 4.71100931-2.8630343 2.86303426-2.86303426-2.86303426c-.39658757-.39658757-1.03281091-.39438847-1.4265779-.00062147-.39651227.39651226-.39348876 1.03246767.00062147 1.4265779l2.86303426 2.86303426-2.86303426 2.8630343c-.39658757.3965875-.39438847 1.0328109-.00062147 1.4265779.39651226.3965122 1.03246767.3934887 1.4265779-.0006215l2.86303426-2.8630343 2.8630343 2.8630343c.3965875.3965876 1.0328109.3943885 1.4265779.0006215.3965122-.3965123.3934887-1.0324677-.0006215-1.4265779l-2.8630343-2.8630343 2.8630343-2.86303426c.3965876-.39658757.3943885-1.03281091.0006215-1.4265779-.3965123-.39651227-1.0324677-.39348876-1.4265779.00062147z\" fill=\"currentColor\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"icon-springer-arrow-left\">\\n            <path d=\"M15 7a1 1 0 000-2H3.385l2.482-2.482a.994.994 0 00.02-1.403 1.001 1.001 0 00-1.417 0L.294 5.292a1.001 1.001 0 000 1.416l4.176 4.177a.991.991 0 001.4.016 1 1 0 00-.003-1.42L3.385 7H15z\"/>\\n        </symbol>\\n        <symbol id=\"icon-springer-arrow-right\">\\n            <path d=\"M1 7a1 1 0 010-2h11.615l-2.482-2.482a.994.994 0 01-.02-1.403 1.001 1.001 0 011.417 0l4.176 4.177a1.001 1.001 0 010 1.416l-4.176 4.177a.991.991 0 01-1.4.016 1 1 0 01.003-1.42L12.615 7H1z\"/>\\n        </symbol>\\n        <symbol id=\"icon-arrow-up\" viewBox=\"0 0 16 16\">\\n            <path d=\"m12.716625 4.46975946-4.03074003-4.17620792c-.37758093-.39120768-.98937525-.39160691-1.367372.0000316l-4.03091981 4.1763942c-.37759778.39122514-.38381821 1.01908149-.01600053 1.40017357.37750607.39113012.98772445.3930364 1.37006824-.00310603l2.39538588-2.48183446v11.61478958l.00649339.1166211c.055753.4973387.46370161.8833789.95867408.8833789.49497246 0 .90292107-.3860402.95867408-.8833789l.00649338-.1166211v-11.61478958l2.39518592 2.4816273c.3791392.39282216.9863753.40056173 1.3541929.01946965.3775061-.39113012.3778444-1.02492687-.0001355-1.41654791z\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"icon-tick\" viewBox=\"0 0 24 24\">\\n            <path d=\"M12,24 C5.372583,24 0,18.627417 0,12 C0,5.372583 5.372583,0 12,0 C18.627417,0 24,5.372583 24,12 C24,18.627417 18.627417,24 12,24 Z M7.657,10.79 C7.45285634,10.6137568 7.18569967,10.5283283 6.91717333,10.5534259 C6.648647,10.5785236 6.40194824,10.7119794 6.234,10.923 C5.87705269,11.3666969 5.93445559,12.0131419 6.364,12.387 L10.261,15.754 C10.6765468,16.112859 11.3037113,16.0695601 11.666,15.657 L17.759,8.713 C18.120307,8.27302248 18.0695334,7.62621189 17.644,7.248 C17.4414817,7.06995024 17.1751516,6.9821166 16.9064461,7.00476032 C16.6377406,7.02740404 16.3898655,7.15856958 16.22,7.368 L10.768,13.489 L7.657,10.79 Z\"/>\\n        </symbol>\\n        <symbol id=\"icon-expand-image\" viewBox=\"0 0 18 18\">\\n            <path d=\"m7.49754099 11.9178212c.38955542-.3895554.38761957-1.0207846-.00290473-1.4113089-.39324695-.3932469-1.02238878-.3918247-1.41130883-.0029047l-4.10273549 4.1027355.00055454-3.5103985c.00008852-.5603185-.44832171-1.006032-1.00155062-1.0059446-.53903074.0000852-.97857527.4487442-.97866268 1.0021075l-.00093318 5.9072465c-.00008751.553948.44841131 1.001882 1.00174994 1.0017946l5.906983-.0009331c.5539233-.0000875 1.00197907-.4486389 1.00206646-1.0018679.00008515-.5390307-.45026621-.9784332-1.00588841-.9783454l-3.51010549.0005545zm3.00571741-5.83449376c-.3895554.38955541-.3876196 1.02078454.0029047 1.41130883.393247.39324696 1.0223888.39182478 1.4113089.00290473l4.1027355-4.10273549-.0005546 3.5103985c-.0000885.56031852.4483217 1.006032 1.0015506 1.00594461.5390308-.00008516.9785753-.44874418.9786627-1.00210749l.0009332-5.9072465c.0000875-.553948-.4484113-1.00188204-1.0017499-1.00179463l-5.906983.00093313c-.5539233.00008751-1.0019791.44863892-1.0020665 1.00186784-.0000852.53903074.4502662.97843325 1.0058884.97834547l3.5101055-.00055449z\" fill=\"currentColor\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"icon-close\" viewBox=\"0 0 16 16\">\\n            <path d=\"m2.29679575 12.2772478c-.39658757.3965876-.39438847 1.0328109-.00062148 1.4265779.39651227.3965123 1.03246768.3934888 1.42657791-.0006214l4.27724782-4.27724787 4.2772478 4.27724787c.3965876.3965875 1.0328109.3943884 1.4265779.0006214.3965123-.3965122.3934888-1.0324677-.0006214-1.4265779l-4.27724787-4.2772478 4.27724787-4.27724782c.3965875-.39658757.3943884-1.03281091.0006214-1.42657791-.3965122-.39651226-1.0324677-.39348875-1.4265779.00062148l-4.2772478 4.27724782-4.27724782-4.27724782c-.39658757-.39658757-1.03281091-.39438847-1.42657791-.00062148-.39651226.39651227-.39348875 1.03246768.00062148 1.42657791l4.27724782 4.27724782z\" fill=\"currentColor\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"icon-chevron-right\" viewBox=\"0 0 7 12\">\\n            <path d=\"M2.782 5 .3 2.518c-.393-.392-.4-1.022-.02-1.403a1.001 1.001 0 0 1 1.417 0l4.176 4.177a1.001 1.001 0 0 1 0 1.416l-4.176 4.177a.991.991 0 0 1-1.4.016A1 1 0 0 1 .3 9.481L2.782 7l1.013-.998L2.782 5Z\" fill-rule=\"evenodd\"/>\\n        </symbol>\\n        <symbol id=\"icon-checklist-banner\" viewBox=\"0 0 56.69 56.69\">\\n            <style type=\"text/css\">.st0 {\\n                fill: none;\\n            }\\n\\n            .st1 {\\n                clip-path: url(#SVGID_2_);\\n                fill: none;\\n                stroke: #01324B;\\n                stroke-width: 2;\\n                stroke-linecap: round;\\n            }\\n\\n            .st2 {\\n                clip-path: url(#SVGID_2_);\\n                fill: none;\\n                stroke: #01324B;\\n                stroke-width: 2;\\n                stroke-linecap: round;\\n                stroke-linejoin: round;\\n            }</style>\\n            <rect id=\"transparent_background\" class=\"st0\" width=\"56.69\" height=\"56.69\"/>\\n            <g>\\n                <clipPath id=\"SVGID_2_\">\\n                    <use xlink:href=\"#SVGID_1_\" style=\"overflow:visible\"/>\\n                </clipPath>\\n                <path class=\"st1\" d=\"M21.14,34.46c0-6.77,5.48-12.26,12.24-12.26c6.76,0,12.24,5.49,12.24,12.26c0,6.77-5.48,12.26-12.24,12.26\\n\\t\\tC26.62,46.71,21.14,41.23,21.14,34.46z M40.47,45.12l10.23,9.22c0,0,1.21,1.09,2.3-0.12l2.09-2.32c0,0,1.09-1.21-0.12-2.3\\n\\t\\tl-10.23-9.22 M25.45,34.46c0-4.38,3.55-7.94,7.93-7.94c4.38,0,7.93,3.55,7.93,7.94c0,4.38-3.55,7.94-7.93,7.94\\n\\t\\tC29,42.39,25.45,38.84,25.45,34.46z M43.03,47.45l4.14-4.81\"/>\\n                <path class=\"st2\" d=\"M8.26,9.75H28.6 M8.26,15.98H28.6 M8.26,22.18h12.5 M35.18,16.98V4.86c0,0,0-2.93-2.93-2.93H4.13\\n\\t\\tc0,0-2.93,0-2.93,2.93v37.57c0,0,0,2.93,2.93,2.93h15.01 M8.26,9.75H28.6 M8.26,15.98H28.6 M8.26,22.18h12.5\"/>\\n            </g>\\n            <g/>\\n            <g/>\\n            <g/>\\n            <g/>\\n            <g/>\\n            <g/>\\n        </symbol>\\n    </svg>\\n\\n</footer>\\n\\n\\n\\n\\n\\n\\n\\n<noscript>\\n    <img hidden src=\"https://verify.nature.com/verify/nature.png\" width=\"0\" height=\"0\" style=\"display: none\" alt=\"\">\\n</noscript>\\n\\n\\n\\n\\n<script src=\"//content.readcube.com/ping?doi=10.1038/hdy.2016.92&amp;format=js&amp;last_modified=2017-03-01\" async></script>\\n<img src=\"/z2rvyg7r/article/hdy201692\" width=\"1\" height=\"1\" alt=\"\" class=\"u-visually-hidden\">\\n</body>\\n</html>'\n"
     ]
    }
   ],
   "source": [
    "import urllib.request\n",
    "\n",
    "with urllib.request.urlopen(link) as url:\n",
    "    s = url.read()\n",
    "    # I'm guessing this would output the html source code ?\n",
    "    print(s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "id": "8beba8ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "URL =\"https://doi.org/10.1038/hdy.2016.92\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "id": "141bb7a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment | Heredity\n"
     ]
    }
   ],
   "source": [
    "URL =\"https://doi.org/10.1038/hdy.2016.92\"\n",
    "page = requests.get(URL)\n",
    "soup = BeautifulSoup(page.content, \"html.parser\")\n",
    "title = soup.find('title')\n",
    "print(title.string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 362,
   "id": "dcf82245",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Quantitative genetics approaches, and particularly animal models, are widely used to assess the genetic (co)variance of key fitness related traits and infer adaptive potential of wild populations. Despite the importance of precision and accuracy of genetic variance estimates and their potential sensitivity to various ecological and population specific factors, their reliability is rarely tested explicitly. Here, we used simulations and empirical data collected from an 11-year study on tree swallow (Tachycineta bicolor), a species showing a high rate of extra-pair paternity and a low recruitment rate, to assess the importance of identity errors, structure and size of the pedigree on quantitative genetic estimates in our dataset. Our simulations revealed an important lack of precision in heritability and genetic-correlation estimates for most traits, a low power to detect significant effects and important identifiability problems. We also observed a large bias in heritability estimates when using the social pedigree instead of the genetic one (deflated heritabilities) or when not accounting for an important cause of resemblance among individuals (for example, permanent environment or brood effect) in model parameterizations for some traits (inflated heritabilities). We discuss the causes underlying the low reliability observed here and why they are also likely to occur in other study systems. Altogether, our results re-emphasize the difficulties of generalizing quantitative genetic estimates reliably from one study system to another and the importance of reporting simulation analyses to evaluate these important issues.\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "id": "1c035a9b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment\n",
      "Quantitative genetics approaches, and particularly animal models, are widely used to assess the genetic (co)variance of key fitness related traits and infer adaptive potential of wild populations. Despite the importance of precision and accuracy of genetic variance estimates and their potential sensitivity to various ecological and population specific factors, their reliability is rarely tested explicitly. Here, we used simulations and empirical data collected from an 11-year study on tree swallow (Tachycineta bicolor), a species showing a high rate of extra-pair paternity and a low recruitment rate, to assess the importance of identity errors, structure and size of the pedigree on quantitative genetic estimates in our dataset. Our simulations revealed an important lack of precision in heritability and genetic-correlation estimates for most traits, a low power to detect significant effects and important identifiability problems. We also observed a large bias in heritability estimates when using the social pedigree instead of the genetic one (deflated heritabilities) or when not accounting for an important cause of resemblance among individuals (for example, permanent environment or brood effect) in model parameterizations for some traits (inflated heritabilities). We discuss the causes underlying the low reliability observed here and why they are also likely to occur in other study systems. Altogether, our results re-emphasize the difficulties of generalizing quantitative genetic estimates reliably from one study system to another and the importance of reporting simulation analyses to evaluate these important issues.\n",
      "Understanding the genetic component underlying important fitness related traits is essential to infer the adaptive potential of wild populations (Lynch and Walsh, 1998; Hendry et al., 2011). However, as most traits are likely under the control of numerous genes with small effects (Lande, 1981; Falconer and Mackay, 1996; Husby et al., 2015), potentially interacting with the genetic components of other traits (Lande and Arnold, 1983; Blows and Hoffmann, 2005) and/or with the environment in which they are expressed (Via and Lande, 1985; Hoffmann and Merilä, 1999), this task remains challenging despite rapid advances in whole-genome analysis methods (Vinkhuyzen et al., 2013). For these reasons, quantitative genetic approaches—statistical methods using known relationships between individuals to assess the genetic and environmental components of a population phenotypic variance—remain efficient ways to infer the underlying genetic variation of focal traits (Falconer and Mackay, 1996; Kruuk et al., 2008). For this purpose, animal models are particularly suitable for wild population datasets because they allow the use of all relationships between individuals and can account for missing data, unbalanced designs and other potential biases (for example, environmental causes of phenotypic similarity among individuals rather than genetic ones) (Henderson, 1984; Kruuk, 2004; Kruuk and Hadfield, 2007; Wilson et al., 2010). Moreover, heritability estimates from wild populations obtained with animal models are generally more accurate than those estimated with traditional parent–offspring regressions (reviewed in Postma, 2014).Several factors will influence the validity of the genetic variance estimates obtained in the wild, which in turn will affect our capacity to infer population responses to selective pressures (for example, through breeder’s or Robertson–Price equation, see Morrissey et al., 2010). Precision—the repeatability of a measurement—is highly dependent on sample size both in terms of pedigree depth and completeness, but also on the complexity of the genetic architecture underlying the focal trait (Morrissey et al., 2007; de Villemereuil et al., 2013). For example, given two pedigrees with identical sample sizes, a pedigree with a weak degree of connectivity between individuals, for example, in populations with strong natal dispersion and/or immigration resulting in few related individuals, will be less precise than a pedigree with known relationships between all individuals (Wilson et al., 2010). However, precision is rarely tested per se, but rather deduced from the size of standard errors around estimates (for example, Charmantier and Réale, 2005). Accuracy—the proximity between the estimated value and the true value—is affected mainly by pedigree errors (that is, erroneous links between individuals in the pedigree; Charmantier and Réale, 2005; Morrissey et al., 2007; Firth et al., 2015) and by model parameterizations (Kruuk and Hadfield, 2007; Wilson, 2008; Wolak et al., 2015). Pedigree errors could be particularly common in study systems where extra-pair copulations are important and parental links in pedigrees constructed solely based on social pair observations (that is, social pedigree). Theoretically, the presence of extra-pair paternity (EPP), if not accounted for, can bias downward the heritability estimates, even though evidence so far suggests that this bias is generally small in wild populations (Charmantier and Réale, 2005; Bérénos et al., 2014; Firth et al., 2015; but see Lee and Pollak, 1997 for reports of bias in the animal breeding litterature; see also Walther and Moore, 2005 for a detailed discussion on the concepts of bias, precision and accuracy).Despite previous knowledge of what can affect precision and accuracy of genetic variance estimates, extrapolating the reliability of these estimates from one study system to another is difficult given the large diversity of life history observed across species that inevitably modified dataset and pedigree structures. Therefore, simulation analyses are recommended for testing the limits of a particular dataset, pedigree and model to answer specific biological questions (Morrissey et al., 2007; Wilson et al., 2010). Simulation frameworks allow the assessment of power—the probability of detecting an effect given that this effect is true—of a particular dataset or model. In animal models, power is affected by the same factors as precision (Quinn et al., 2006; Morrissey et al., 2007; Bérénos et al., 2014). However, performing a power analysis alone could provide an incomplete picture of the validity of genetic variance estimates. In fact, a particular model applied on a dataset could have a high power to estimate a given effect, but the model itself might be missing a critical variance term, leading to precise but inaccurate estimates (Kruuk and Hadfield, 2007). Similarly, even if all important variance terms are included in a particular model, the model variance structure might not allow discrimination between two variance terms being included, a problem of statistical models related to their low ‘identifiability’ (Bolker, 2008). Confounded parameters could be frequent when applying animal models to wild populations (Wilson et al., 2010), but the extent of identifiability problems is still unknown. Finally, reliability of quantitative genetics estimates is even less reported for multivariate models than univariate models, limiting our confidence on important parameter estimations such as genetic additive covariance/correlation among traits.Here, we used an 11-year study on Tree swallow (Tachycineta bicolor (Vieillot, 1808)) to assess the effects of identity errors, structure and size of a pedigree on reliability of quantitative genetic estimates. Tree swallows are small migratory passerines and their breeding grounds are widely distributed across North America (Winkler et al., 2011). Similarly to other migrating species, tree swallow shows high natal dispersal, high mortality within the first year and low nestling recruitment rate (Hosner and Winkler, 2007), which reduce kinship among individuals in monitored populations. This socially monogamous species also displays one of the highest rate of EPP documented in birds (Griffith et al., 2002), with more than 80% of nests containing at least one extra-pair offspring and overall around 50% of nestlings resulting from extra-pair copulations (Dunn et al., 2001; Lessard et al., 2014). In the context of building a pedigree for quantitative genetics analyses, this means that half of paternal links would be erroneous if using a social pedigree. This represents a higher proportion of EPP than those tested in previous studies to assess their impacts on heritability estimates (for example, in Charmantier and Réale, 2005; Firth et al., 2015), but it is a proportion that will be typically found in other similar study systems (see Griffith et al., 2002 and references therein). Furthermore, reliability of genetic variance estimates for this and similar species is difficult to predict based on previous results obtained on other species, as the typical poorly connected pedigree available for migratory species could be compensated by the natural half-sibs design resulting from the high rate of EPP occurring in tree swallows.In this study, we used a social and a genetic pedigree to estimate and compare genetic additive (co-)variances and heritability of traits differing in their completeness through the pedigree structure. More specifically, we first used empirical data to assess the bias resulting from using a social pedigree over a genetic one. Then, we used simulated data to assess precision and accuracy of quantitative genetic estimates obtained with both pedigrees, as well as power of datasets and models, and identifiability among variance terms.\n"
     ]
    }
   ],
   "source": [
    "res = soup.find_all(\"h1\", class_=\"c-article-title\")\n",
    "title = res[0].text\n",
    "print(title)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 310,
   "id": "af9cffdc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<div class=\"c-article-section__content\" id=\"Abs1-content\"><p>Quantitative genetics approaches, and particularly animal models, are widely used to assess the genetic (co)variance of key fitness related traits and infer adaptive potential of wild populations. Despite the importance of precision and accuracy of genetic variance estimates and their potential sensitivity to various ecological and population specific factors, their reliability is rarely tested explicitly. Here, we used simulations and empirical data collected from an 11-year study on tree swallow (<i>Tachycineta bicolor</i>), a species showing a high rate of extra-pair paternity and a low recruitment rate, to assess the importance of identity errors, structure and size of the pedigree on quantitative genetic estimates in our dataset. Our simulations revealed an important lack of precision in heritability and genetic-correlation estimates for most traits, a low power to detect significant effects and important identifiability problems. We also observed a large bias in heritability estimates when using the social pedigree instead of the genetic one (deflated heritabilities) or when not accounting for an important cause of resemblance among individuals (for example, permanent environment or brood effect) in model parameterizations for some traits (inflated heritabilities). We discuss the causes underlying the low reliability observed here and why they are also likely to occur in other study systems. Altogether, our results re-emphasize the difficulties of generalizing quantitative genetic estimates reliably from one study system to another and the importance of reporting simulation analyses to evaluate these important issues.</p></div>]"
      ]
     },
     "execution_count": 310,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "id": "20685c5b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<div class=\"c-article-section__content\" id=\"Abs1-content\"><p>Quantitative genetics approaches, and particularly animal models, are widely used to assess the genetic (co)variance of key fitness related traits and infer adaptive potential of wild populations. Despite the importance of precision and accuracy of genetic variance estimates and their potential sensitivity to various ecological and population specific factors, their reliability is rarely tested explicitly. Here, we used simulations and empirical data collected from an 11-year study on tree swallow (<i>Tachycineta bicolor</i>), a species showing a high rate of extra-pair paternity and a low recruitment rate, to assess the importance of identity errors, structure and size of the pedigree on quantitative genetic estimates in our dataset. Our simulations revealed an important lack of precision in heritability and genetic-correlation estimates for most traits, a low power to detect significant effects and important identifiability problems. We also observed a large bias in heritability estimates when using the social pedigree instead of the genetic one (deflated heritabilities) or when not accounting for an important cause of resemblance among individuals (for example, permanent environment or brood effect) in model parameterizations for some traits (inflated heritabilities). We discuss the causes underlying the low reliability observed here and why they are also likely to occur in other study systems. Altogether, our results re-emphasize the difficulties of generalizing quantitative genetic estimates reliably from one study system to another and the importance of reporting simulation analyses to evaluate these important issues.</p></div>\n",
      "<div class=\"c-article-section__content\" id=\"Sec1-content\"><p>Understanding the genetic component underlying important fitness related traits is essential to infer the adaptive potential of wild populations (<a aria-label=\"Reference Lynch and Walsh, 1998\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR33\" id=\"ref-link-section-d28693445e365\" title=\"Lynch M, Walsh B . (1998) Genetics and Analysis of Quantitative Traits. Sinauer Associates: Sunderland, MA, USA.\">Lynch and Walsh, 1998</a>; <a aria-label=\"Reference Hendry et al., 2011\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR16\" id=\"ref-link-section-d28693445e368\" title=\"Hendry AP, Kinnison MT, Heino M, Day T, Smith TB, Fitt G et al. (2011). Evolutionary principles and their practical application. Evol Appl 4: 159–183.\">Hendry et al., 2011</a>). However, as most traits are likely under the control of numerous genes with small effects (<a aria-label=\"Reference Lande, 1981\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR28\" id=\"ref-link-section-d28693445e371\" title=\"Lande R . (1981). The minimum number of genes contributing to quantitative variation between and within populations. Genetics 99: 541–553.\">Lande, 1981</a>; <a aria-label=\"Reference Falconer and Mackay, 1996\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR8\" id=\"ref-link-section-d28693445e374\" title=\"Falconer DS, Mackay TFC . (1996) Introduction to Quantitative Genetics, 4th edn. Pearson: Essex, UK.\">Falconer and Mackay, 1996</a>; <a aria-label=\"Reference Husby et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR21\" id=\"ref-link-section-d28693445e377\" title=\"Husby A, Kawakami T, Rönnegård L, Smeds L, Ellegren H, Qvarnström A . (2015). Genome-wide association mapping in a wild avian population identifies a link between genetic and phenotypic variation in a life-history trait. Proc R Soc B 282: 20150156.\">Husby et al., 2015</a>), potentially interacting with the genetic components of other traits (<a aria-label=\"Reference Lande and Arnold, 1983\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR29\" id=\"ref-link-section-d28693445e381\" title=\"Lande R, Arnold SJ . (1983). The measurement of selection on correlated characters. Evolution 37: 1210–1226.\">Lande and Arnold, 1983</a>; <a aria-label=\"Reference Blows and Hoffmann, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR3\" id=\"ref-link-section-d28693445e384\" title=\"Blows MW, Hoffmann AA . (2005). A reassessment of genetic limits to evoutionary change. Ecology 86: 1371–1384.\">Blows and Hoffmann, 2005</a>) and/or with the environment in which they are expressed (<a aria-label=\"Reference Via and Lande, 1985\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR42\" id=\"ref-link-section-d28693445e387\" title=\"Via S, Lande R . (1985). Genotype-environment interaction and the evolution of phenotypic plasticity. Evolution 39: 505–522.\">Via and Lande, 1985</a>; <a aria-label=\"Reference Hoffmann and Merilä, 1999\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR17\" id=\"ref-link-section-d28693445e390\" title=\"Hoffmann AA, Merilä J . (1999). Heritable variation and evolution under favourable and unfavourable conditions. Trends Ecol Evol 14: 96–101.\">Hoffmann and Merilä, 1999</a>), this task remains challenging despite rapid advances in whole-genome analysis methods (<a aria-label=\"Reference Vinkhuyzen et al., 2013\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR44\" id=\"ref-link-section-d28693445e393\" title=\"Vinkhuyzen AAE, Wray NR, Yang J, Goddard ME, Visscher PM . (2013). Estimation and partition of heritability in human populations using whole-genome analysis methods. Annu Rev Genet 47: 75–95.\">Vinkhuyzen et al., 2013</a>). For these reasons, quantitative genetic approaches—statistical methods using known relationships between individuals to assess the genetic and environmental components of a population phenotypic variance—remain efficient ways to infer the underlying genetic variation of focal traits (<a aria-label=\"Reference Falconer and Mackay, 1996\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR8\" id=\"ref-link-section-d28693445e396\" title=\"Falconer DS, Mackay TFC . (1996) Introduction to Quantitative Genetics, 4th edn. Pearson: Essex, UK.\">Falconer and Mackay, 1996</a>; <a aria-label=\"Reference Kruuk et al., 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR26\" id=\"ref-link-section-d28693445e400\" title=\"Kruuk LEB, Slate J, Wilson AJ . (2008). New answers for old questions: the evolutionary quantitative genetics of wild animal populations. Annu Rev Ecol Evol Syst 39: 525–548.\">Kruuk et al., 2008</a>). For this purpose, animal models are particularly suitable for wild population datasets because they allow the use of all relationships between individuals and can account for missing data, unbalanced designs and other potential biases (for example, environmental causes of phenotypic similarity among individuals rather than genetic ones) (<a aria-label=\"Reference Henderson, 1984\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR15\" id=\"ref-link-section-d28693445e403\" title=\"Henderson CR . (1984) Applications of Linear Models in Animal Breeding. University of Guelph: Guelph, ON, Canada.\">Henderson, 1984</a>; <a aria-label=\"Reference Kruuk, 2004\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR24\" id=\"ref-link-section-d28693445e406\" title=\"Kruuk LEB . (2004). Estimating genetic parameters in natural populations using the ‘animal model’. Philos Trans R Soc B 359: 873–890.\">Kruuk, 2004</a>; <a aria-label=\"Reference Kruuk and Hadfield, 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d28693445e409\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890–1903.\">Kruuk and Hadfield, 2007</a>; <a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e412\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). Moreover, heritability estimates from wild populations obtained with animal models are generally more accurate than those estimated with traditional parent–offspring regressions (reviewed in <a aria-label=\"Reference Postma, 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR39\" id=\"ref-link-section-d28693445e415\" title=\"Postma E . (2014). Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates? In: Charmantier A, Garant D et al. (eds) Quantitative Genetics in the Wild. Oxford University Press, Oxford, UK, pp 16–33.\">Postma, 2014</a>).</p><p>Several factors will influence the validity of the genetic variance estimates obtained in the wild, which in turn will affect our capacity to infer population responses to selective pressures (for example, through breeder’s or Robertson–Price equation, see <a aria-label=\"Reference Morrissey et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR35\" id=\"ref-link-section-d28693445e421\" title=\"Morrissey MB, Kruuk LEB, Wilson AJ . (2010). The danger of applying the breeder’s equation in observational studies of natural populations. J Evol Biol 23: 2277–2288.\">Morrissey et al., 2010</a>). Precision—the repeatability of a measurement—is highly dependent on sample size both in terms of pedigree depth and completeness, but also on the complexity of the genetic architecture underlying the focal trait (<a aria-label=\"Reference Morrissey et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d28693445e424\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309–2321.\">Morrissey et al., 2007</a>; <a aria-label=\"Reference de Villemereuil et al., 2013\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR43\" id=\"ref-link-section-d28693445e427\" title=\"de Villemereuil P, Gimenez O, Doligez B . (2013). Comparing parent–offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits. Methods Ecol Evol 4: 260–275.\">de Villemereuil et al., 2013</a>). For example, given two pedigrees with identical sample sizes, a pedigree with a weak degree of connectivity between individuals, for example, in populations with strong natal dispersion and/or immigration resulting in few related individuals, will be less precise than a pedigree with known relationships between all individuals (<a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e430\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). However, precision is rarely tested <i>per se</i>, but rather deduced from the size of standard errors around estimates (for example, <a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e437\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>). Accuracy—the proximity between the estimated value and the true value—is affected mainly by pedigree errors (that is, erroneous links between individuals in the pedigree; <a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e440\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>; <a aria-label=\"Reference Morrissey et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d28693445e443\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309–2321.\">Morrissey et al., 2007</a>; <a aria-label=\"Reference Firth et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d28693445e446\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336–1344.\">Firth et al., 2015</a>) and by model parameterizations (<a aria-label=\"Reference Kruuk and Hadfield, 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d28693445e449\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890–1903.\">Kruuk and Hadfield, 2007</a>; <a aria-label=\"Reference Wilson, 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR46\" id=\"ref-link-section-d28693445e452\" title=\"Wilson AJ . (2008). Why h2 does not always equal VA/VP? J Evol Biol 21: 647–650.\">Wilson, 2008</a>; <a aria-label=\"Reference Wolak et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR51\" id=\"ref-link-section-d28693445e456\" title=\"Wolak ME, Roff DA, Fairbairn DJ . (2015). Are we underestimating the genetic variances of dimorphic traits? Ecol Evol 5: 590–597.\">Wolak et al., 2015</a>). Pedigree errors could be particularly common in study systems where extra-pair copulations are important and parental links in pedigrees constructed solely based on social pair observations (that is, social pedigree). Theoretically, the presence of extra-pair paternity (EPP), if not accounted for, can bias downward the heritability estimates, even though evidence so far suggests that this bias is generally small in wild populations (<a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e459\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>; <a aria-label=\"Reference Bérénos et al., 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR2\" id=\"ref-link-section-d28693445e462\" title=\"Bérénos C, Ellis PA, Pilkington JG, Pemberton JM . (2014). Estimating quantitative genetic parameters in wild populations: a comparison of pedigree and genomic approaches. Mol Ecol 23: 3434–3451.\">Bérénos et al., 2014</a>; <a aria-label=\"Reference Firth et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d28693445e465\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336–1344.\">Firth et al., 2015</a>; but see <a aria-label=\"Reference Lee and Pollak, 1997\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR30\" id=\"ref-link-section-d28693445e468\" title=\"Lee C, Pollak EJ . (1997). Influence of sire misidentification on sire x year interaction variance and direct-maternal genetic covariance for weaning weight in beef cattle. J Anim Sci 75: 2858–2863.\">Lee and Pollak, 1997</a> for reports of bias in the animal breeding litterature; see also <a aria-label=\"Reference Walther and Moore, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR45\" id=\"ref-link-section-d28693445e471\" title=\"Walther BA, Moore JL . (2005). The concepts of bias, precision and accuracy, and their use in testing the performance of species richness estimators, with a literature review of estimator performance. Ecography 28: 815–829.\">Walther and Moore, 2005</a> for a detailed discussion on the concepts of bias, precision and accuracy).</p><p>Despite previous knowledge of what can affect precision and accuracy of genetic variance estimates, extrapolating the reliability of these estimates from one study system to another is difficult given the large diversity of life history observed across species that inevitably modified dataset and pedigree structures. Therefore, simulation analyses are recommended for testing the limits of a particular dataset, pedigree and model to answer specific biological questions (<a aria-label=\"Reference Morrissey et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d28693445e477\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309–2321.\">Morrissey et al., 2007</a>; <a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e480\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). Simulation frameworks allow the assessment of power—the probability of detecting an effect given that this effect is true—of a particular dataset or model. In animal models, power is affected by the same factors as precision (<a aria-label=\"Reference Quinn et al., 2006\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR40\" id=\"ref-link-section-d28693445e483\" title=\"Quinn JL, Charmantier A, Garant D, Sheldon BC . (2006). Data depth, data completeness, and their influence on quantitative genetic estimation in two contrasting bird populations. J Evol Biol 19: 994–1002.\">Quinn et al., 2006</a>; <a aria-label=\"Reference Morrissey et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d28693445e486\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309–2321.\">Morrissey et al., 2007</a>; <a aria-label=\"Reference Bérénos et al., 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR2\" id=\"ref-link-section-d28693445e489\" title=\"Bérénos C, Ellis PA, Pilkington JG, Pemberton JM . (2014). Estimating quantitative genetic parameters in wild populations: a comparison of pedigree and genomic approaches. Mol Ecol 23: 3434–3451.\">Bérénos et al., 2014</a>). However, performing a power analysis alone could provide an incomplete picture of the validity of genetic variance estimates. In fact, a particular model applied on a dataset could have a high power to estimate a given effect, but the model itself might be missing a critical variance term, leading to precise but inaccurate estimates (<a aria-label=\"Reference Kruuk and Hadfield, 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d28693445e493\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890–1903.\">Kruuk and Hadfield, 2007</a>). Similarly, even if all important variance terms are included in a particular model, the model variance structure might not allow discrimination between two variance terms being included, a problem of statistical models related to their low ‘identifiability’ (<a aria-label=\"Reference Bolker, 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR4\" id=\"ref-link-section-d28693445e496\" title=\"Bolker B . (2008) Ecological Models and Data in R. Princeton, University Princeton, NJ, USA.\">Bolker, 2008</a>). Confounded parameters could be frequent when applying animal models to wild populations (<a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e499\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>), but the extent of identifiability problems is still unknown. Finally, reliability of quantitative genetics estimates is even less reported for multivariate models than univariate models, limiting our confidence on important parameter estimations such as genetic additive covariance/correlation among traits.</p><p>Here, we used an 11-year study on Tree swallow (<i>Tachycineta bicolor</i> (Vieillot, 1808)) to assess the effects of identity errors, structure and size of a pedigree on reliability of quantitative genetic estimates. Tree swallows are small migratory passerines and their breeding grounds are widely distributed across North America (<a aria-label=\"Reference Winkler et al., 2011\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR49\" id=\"ref-link-section-d28693445e508\" title=\"Winkler D, Hallinger K, Ardia D, Robertson R, Stutchbury B, Cohen R . (2011). Tree swallow (Tachycineta bicolor. Birds North Am Online. \n",
      "                  http://bna.birds.cornelledu/bna/\n",
      "                  \n",
      "                . Accessed on 1 Oct 2013.\">Winkler et al., 2011</a>). Similarly to other migrating species, tree swallow shows high natal dispersal, high mortality within the first year and low nestling recruitment rate (<a aria-label=\"Reference Hosner and Winkler, 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR18\" id=\"ref-link-section-d28693445e511\" title=\"Hosner PA, Winkler DW . (2007). Dispersal distances of tree swallows estimated from continent-wide and limited-area data. J Field Ornithol 78: 290–297.\">Hosner and Winkler, 2007</a>), which reduce kinship among individuals in monitored populations. This socially monogamous species also displays one of the highest rate of EPP documented in birds (<a aria-label=\"Reference Griffith et al., 2002\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR13\" id=\"ref-link-section-d28693445e514\" title=\"Griffith SC, Owens IPF, Thuman KA . (2002). Extra pair paternity in birds: a review of interspecific variation and adaptive function. Mol Ecol 11: 2195–2212.\">Griffith et al., 2002</a>), with more than 80% of nests containing at least one extra-pair offspring and overall around 50% of nestlings resulting from extra-pair copulations (<a aria-label=\"Reference Dunn et al., 2001\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR7\" id=\"ref-link-section-d28693445e517\" title=\"Dunn PO, Whittingham LA, Pitcher TE . (2001). Mating systems, sperm competition, and the evolution of sexual dimorphism in birds. Evolution 55: 161–175.\">Dunn et al., 2001</a>; <a aria-label=\"Reference Lessard et al., 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR32\" id=\"ref-link-section-d28693445e521\" title=\"Lessard A, Bourret A, Bélisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor. Behav Ecol Sociobiol 68: 733–742.\">Lessard et al., 2014</a>). In the context of building a pedigree for quantitative genetics analyses, this means that half of paternal links would be erroneous if using a social pedigree. This represents a higher proportion of EPP than those tested in previous studies to assess their impacts on heritability estimates (for example, in <a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e524\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>; <a aria-label=\"Reference Firth et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d28693445e527\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336–1344.\">Firth et al., 2015</a>), but it is a proportion that will be typically found in other similar study systems (see <a aria-label=\"Reference Griffith et al., 2002\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR13\" id=\"ref-link-section-d28693445e530\" title=\"Griffith SC, Owens IPF, Thuman KA . (2002). Extra pair paternity in birds: a review of interspecific variation and adaptive function. Mol Ecol 11: 2195–2212.\">Griffith et al., 2002</a> and references therein). Furthermore, reliability of genetic variance estimates for this and similar species is difficult to predict based on previous results obtained on other species, as the typical poorly connected pedigree available for migratory species could be compensated by the natural half-sibs design resulting from the high rate of EPP occurring in tree swallows.</p><p>In this study, we used a social and a genetic pedigree to estimate and compare genetic additive (co-)variances and heritability of traits differing in their completeness through the pedigree structure. More specifically, we first used empirical data to assess the bias resulting from using a social pedigree over a genetic one. Then, we used simulated data to assess precision and accuracy of quantitative genetic estimates obtained with both pedigrees, as well as power of datasets and models, and identifiability among variance terms.</p></div>\n",
      "<div class=\"c-article-section__content\" id=\"Sec2-content\"><h3 class=\"c-article__sub-heading\" id=\"Sec3\">Study species, system and phenotypic data</h3><p>Tree swallows breed in tree cavities or nest boxes, they produce only one clutch per year, containing on average five eggs and both parents provide care to nestlings (<a aria-label=\"Reference Winkler et al., 2011\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR49\" id=\"ref-link-section-d28693445e549\" title=\"Winkler D, Hallinger K, Ardia D, Robertson R, Stutchbury B, Cohen R . (2011). Tree swallow (Tachycineta bicolor. Birds North Am Online. \n",
      "                  http://bna.birds.cornelledu/bna/\n",
      "                  \n",
      "                . Accessed on 1 Oct 2013.\">Winkler et al., 2011</a>). Since 2004, we have intensively followed tree swallow activities during the breeding season at 400 nest boxes equally distributed across 40 farms in an area covering 10 200 km<sup>2</sup> in southern Québec, Canada (detailed in <a aria-label=\"Reference Ghilain and Bélisle, 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR11\" id=\"ref-link-section-d28693445e554\" title=\"Ghilain A, Bélisle M . (2008). Breeding success of tree swallows along a gradient of agricultural intensification. Ecol Appl 18: 1140–1154.\">Ghilain and Bélisle, 2008</a>). Nest boxes were visited every 2 days from May to July to record nest box occupancy and important brood characteristics (for example, laying date, clutch size and hatching date). Adults and nestlings were individually marked with an aluminum band (US Fish and Wildlife Service). Adults were captured directly in their nest box by a trap system, during the incubation period and food provisioning period for females and males, respectively. Morphological measurements of body mass (±0.01 g), non-flattened wing length (±0.5 mm) and tarsus length (±0.02 mm) were taken on adult tree swallows during captures. Females were classified based on their plumage color as second-year (brown) or after-second-year (blue-green) (<a aria-label=\"Reference Hussell, 1983\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR22\" id=\"ref-link-section-d28693445e557\" title=\"Hussell DJT . (1983). Age and plumage color in female tree swallows. J Field Ornithol 54: 312–318.\">Hussell, 1983</a>) and a minimal age was determined for all adults based on the year they were first observed in the study system. Nestlings were captured at 16 days old before they fledged (fledging occurs around 18–22 days) to record body mass (±0.01 g), primary length (hereafter wing length; ±0.02 mm) and tarsus length (±0.02 mm). Since 2006, blood samples of adults and nestlings have been collected on a qualitative P8 grade filter paper (Thermo Fisher Scientific, Waltham, MA, USA) for molecular analysis (see below).</p><p>To reflect the large differences that can exist between traits in terms of sample size and amount of standing genetic variance, we focused on nine phenotypic traits grouped in three categories: (1) morphological traits, which included wing length, body mass and tarsus length of all adults; (2) reproductive traits, which were restricted to females, and included laying date (that is, date of the first egg laid), clutch size (that is, number of eggs laid within a nest box) and incubation duration (defined as hatching date−(laying date+clutch size−1)); and (3) nestling traits, which included wing length, body mass and tarsus length measured at the age of 16 days. Most traits were measured since the beginning of the research program (that is, in 2004), but some traits were first measured later (nestling body mass: 2005; nestling wing length: 2006; adult/nestling tarsus length: 2007) creating differences in data completeness among traits (sample size per traits can be found in <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table A1</a>).</p><h3 class=\"c-article__sub-heading\" id=\"Sec4\">Molecular analysis and pedigree construction</h3><p>DNA extraction, molecular sexing and microsatellite data analyses are detailed in <a aria-label=\"Reference Lessard et al. (2014)\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR32\" id=\"ref-link-section-d28693445e574\" title=\"Lessard A, Bourret A, Bélisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor. Behav Ecol Sociobiol 68: 733–742.\">Lessard et al. (2014)</a>. Briefly, DNA was extracted from blood samples following a standard salt-extraction method and DNA concentration was determined by migration on 2% agarose gels with a molecular weight standard. A molecular sexing technique was used to determine nestling sexes and to validate adult field observations. All DNA samples were characterized at six microsatellites loci using an AB3130xl automated DNA sequencer and allele lengths were determined using GeneMapper v4.1 (Applied Biosystems, Foster City, CA, USA).</p><p>We constructed a social pedigree using social male identities (that is, males caught in nest boxes while feeding the young) and a genetic pedigree using genetic father identities (that is, males assigned as fathers using genetic analyses—see below). In both pedigrees, dam identities were first determined by female captures during egg incubation and then verified genetically based on locus mismatches with nestlings (2.1% of broods with two females captured within the same nest box, from which only 3.6% were from mixed maternity; 11 nestling genotypes (0.15%) mismatched at more than two loci with their social mother genotype). Genetic fathers were determined by parental assignations with <span class=\"u-small-caps\">CERVUS</span> v3.0.7 (<a aria-label=\"Reference Kalinowski et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR23\" id=\"ref-link-section-d28693445e583\" title=\"Kalinowski ST, Taper ML, Marshall TC . (2007). Revising how the computer program cervus accommodates genotyping error increases success in paternity assignment. Mol Ecol 16: 1099–1106.\">Kalinowski et al., 2007</a>; <a aria-label=\"Reference Lemons et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR31\" id=\"ref-link-section-d28693445e586\" title=\"Lemons PR, Marshall TC, McCloskey SE, Sethi SA, Schmutz JA, Sedinger JS . (2015). A likelihood-based approach for assessment of extra-pair paternity and conspecific brood parasitism in natural populations. Mol Ecol Resour 7: 107–116.\">Lemons et al., 2015</a>) in a three step procedure. First, we proceeded to father assignments each year separately following a method slightly modified from <a aria-label=\"Reference Lessard et al. (2014)\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR32\" id=\"ref-link-section-d28693445e589\" title=\"Lessard A, Bourret A, Bélisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor. Behav Ecol Sociobiol 68: 733–742.\">Lessard et al. (2014)</a>. Candidate fathers considered in analyses included all males captured during a given breeding season within 15 km of the nest box of interest (see <a aria-label=\"Reference Lessard et al., 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR32\" id=\"ref-link-section-d28693445e592\" title=\"Lessard A, Bourret A, Bélisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor. Behav Ecol Sociobiol 68: 733–742.\">Lessard et al., 2014</a> for the rationale behind this approach), but also all males not captured in a given year but suspected of being present outside of our nest box system (that is, captured on the same farm on both previous and following years). These assignations were based on a 90% confidence level, assuming a 2% mistyping error rate and we used the percentage of social fathers captured as the percentage of candidate fathers known (variable among years, range: 64–88%). Second, social fathers, when known, were tested for being genetic fathers of offspring in their nest using the likelihood-based approach of <a aria-label=\"Reference Lemons et al. (2015)\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR31\" id=\"ref-link-section-d28693445e596\" title=\"Lemons PR, Marshall TC, McCloskey SE, Sethi SA, Schmutz JA, Sedinger JS . (2015). A likelihood-based approach for assessment of extra-pair paternity and conspecific brood parasitism in natural populations. Mol Ecol Resour 7: 107–116.\">Lemons et al. (2015)</a>. Briefly, we re-ran the parental analysis with the social father as the unique candidate father for a given nestling, and we defined the proportion of sampled fathers as the proportion of nestlings without any locus mismatch with its social father (that is, the probability of the social male being the true father). We then extracted, for each nestling, the critical logarithm of odds score associated with 95% confidence that its social father was not its true father and we compared these scores with those observed in regular parental analyses. Males significantly assigned to nestlings in the initial parental analysis (that is, step 1) were considered as their genetic fathers. For the remaining nestlings (that is, without a significant male assignment at step 1), if their social fathers could not be excluded (that is, step 2), they were considered as their genetic fathers but otherwise no genetic father were assigned to them (see <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure A1</a> for the exact number of fathers assigned to a nestling at each step).</p><p>Summary statistics for both pedigrees were obtained with the package <span class=\"u-small-caps\">PEDANTICS</span> (<a aria-label=\"Reference Morrissey et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d28693445e608\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309–2321.\">Morrissey et al., 2007</a>; <a aria-label=\"Reference Morrissey and Wilson, 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR37\" id=\"ref-link-section-d28693445e611\" title=\"Morrissey MB, Wilson AJ . (2010). PEDANTICS: an R package for pedigree-based genetic simulation and pedigree manipulation, characterization and viewing. Mol Ecol Resour 10: 711–719.\">Morrissey and Wilson, 2010</a>) in R v3.2.0 (<a aria-label=\"Reference R Core Team, 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR41\" id=\"ref-link-section-d28693445e614\" title=\"Core Team R . (2015). R: a language and environment for statistical computing \n",
      "                  http://www.R-project.org/\n",
      "                  \n",
      "                .\">R Core Team, 2015</a>). These statistics were computed for complete pedigrees, but also for pedigrees pruned to contain only informative individuals based on the availability of phenotypic data for each trait (hereafter pruned pedigrees) and are presented for each trait category in <a data-track=\"click\" data-track-action=\"table anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Tab1\">Table 1</a> (see also <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table A2</a> for more information on each trait).</p><div class=\"c-article-table\" data-container-section=\"table\" data-test=\"inline-table\" id=\"table-1\"><figure><figcaption class=\"c-article-table__figcaption\"><b data-test=\"table-caption\" id=\"Tab1\">Table 1 Summary statistics for social and genetic pedigrees, in complete pedigrees (total) or pruned pedigrees detailed for each trait category (means for all morphological, reproductive and nestling traits) based on data collected between 2004 and 2014 in our study system in southern Québec</b></figcaption><div class=\"u-text-right u-hide-print\"><a aria-label=\"Full size table 1\" class=\"c-article__pill-button\" data-test=\"table-link\" data-track=\"click\" data-track-action=\"view table\" data-track-label=\"button\" href=\"/articles/hdy201692/tables/1\" rel=\"nofollow\"><span>Full size table</span><svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-chevron-right\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></div></figure></div><h3 class=\"c-article__sub-heading\" id=\"Sec5\">Estimations of quantitative genetic parameters</h3><p>To estimate additive genetic (co)variances of our focal traits, we used both univariate and multivariate animal models (<a aria-label=\"Reference Kruuk, 2004\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR24\" id=\"ref-link-section-d28693445e1273\" title=\"Kruuk LEB . (2004). Estimating genetic parameters in natural populations using the ‘animal model’. Philos Trans R Soc B 359: 873–890.\">Kruuk, 2004</a>; <a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e1276\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). Fixed effects (for example, age, sex) were included for some traits based on mixed model analyses detailed in <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Information</a>. For adult morphological and reproductive traits, full univariate animal models were constructed as follows:</p><div class=\"c-article-equation\" id=\"Equ1\"><div class=\"c-article-equation__content\"><img alt=\"\" class=\"u-display-block\" src=\"//media.springernature.com/lw457/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Equ1_HTML.gif\"/></div></div><p>and for nestling traits:</p><div class=\"c-article-equation\" id=\"Equ2\"><div class=\"c-article-equation__content\"><img alt=\"\" class=\"u-display-block\" src=\"//media.springernature.com/lw457/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Equ2_HTML.gif\"/></div></div><p>where <i>V</i><sub>P</sub> is the phenotypic variance after accounting for fixed effects, <i>V</i><sub>A</sub> is the additive genetic variance, <i>V</i><sub>PE</sub> is the permanent environmental effect, <i>V</i><sub>Y</sub> and <i>V</i><sub>BY</sub> are the variance among years and among birth years, respectively, <i>V</i><sub>B</sub> is the variance among broods and <i>V</i><sub>R</sub> the residual variance. All traits followed a Gaussian distribution and animal models were resolved with a restricted maximum likelihood method (REML), using both the social and the genetic pedigree. Final animal models were constructed by sequential model-building from a model with only residual variance to more complex ones, with a comparison of models at each step using a likelihood ratio test (LRT; see <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Tables C1</a> for the increasing levels of complexity). Only <i>V</i><sub>A</sub> in the incubation duration models did not significantly improve the model likelihood (<i>χ</i><sup>2</sup>&lt;0.01, <i>P</i>&gt;0.99).</p><p>We also constructed three multivariate animal models, one for each trait category. For each of them, we first included the same variance terms as for univariate models. However, due to convergence problems when including <i>V</i><sub>PE</sub> for morphological and reproductive traits, we decided to use a model without the <i>V</i><sub>PE</sub> term on a reduced dataset comprising only one observation by individual (randomly chosen). Moreover, as we observed no <i>V</i><sub>A</sub> for incubation duration, this trait was not included in the multivariate analysis of reproductive traits. Covariances among traits for each variance components were estimated using unstructured variance models. Significance was tested by comparing a model including covariance estimation to a model where covariance was constrained to be equal to zero using LRT.</p><p>We estimated heritability (<i>h</i><sup>2</sup>=<i>V</i><sub>A</sub>/<i>V</i><sub>P</sub>, <a aria-label=\"Reference Falconer and Mackay, 1996\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR8\" id=\"ref-link-section-d28693445e1377\" title=\"Falconer DS, Mackay TFC . (1996) Introduction to Quantitative Genetics, 4th edn. Pearson: Essex, UK.\">Falconer and Mackay, 1996</a>) and coefficient of genetic variation (CV<sub>A</sub>=<img alt=\"\" src=\"//media.springernature.com/lw89/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq1_HTML.gif\" style=\"width:89px;max-width:none;\"/>, where <img alt=\"\" src=\"//media.springernature.com/lw11/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq2_HTML.gif\" style=\"width:11px;max-width:none;\"/> is the trait mean, <a aria-label=\"Reference Houle, 1992\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR19\" id=\"ref-link-section-d28693445e1399\" title=\"Houle D . (1992). Comparing evolvability and variability of quantitative traits. Genetics 130: 195–204.\">Houle, 1992</a>) for all traits within each analysis. For multivariate analyses we also estimated additive genetic correlations (<i>r</i><sub>A</sub>) between each pair of traits. All animal model analyses were conducted with ASRemL v3.0.5 (VSN International Ltd, Hemel Hempstead, UK). Standard errors (SEs) for variance components and <i>h</i><sup>2</sup> estimates were computed directly by ASRemL. For CV<sub>A</sub>, standard deviations were approximated following the equation provided by Lynch and Walsh <a aria-label=\"Reference (1998\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR33\" id=\"ref-link-section-d28693445e1413\" title=\"Lynch M, Walsh B . (1998) Genetics and Analysis of Quantitative Traits. Sinauer Associates: Sunderland, MA, USA.\">(1998</a>; equation A1.26c, Appendix 1).</p><h3 class=\"c-article__sub-heading\" id=\"Sec6\">Simulation analyses</h3><p>We simulated three different datasets of phenotypic data within both the social and genetic pedigree structures with different levels of complexity. In all cases, simulated traits were normally distributed among all individuals, with <i>V</i><sub>P</sub>=1, <img alt=\"\" src=\"//media.springernature.com/lw11/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq3_HTML.gif\" style=\"width:11px;max-width:none;\"/>=0 and <i>h</i><sup>2</sup> of 0.1, 0.3 and 0.5 for the focal trait. In dataset 1, we simulated one phenotype per individual with the parameters mentioned above to reflect the simplest scenario possible. In dataset 2, we simulated phenotypic traits with a more complex underlying structure based on equations (1) and (2) to better reflect our empirical dataset, with similar value of <i>h</i><sup><i>2</i></sup> as dataset 1. For nestling traits, we simulated phenotypes with <i>V</i><sub>Y</sub>=0.1 and <i>V</i><sub>B</sub>=0.3, whereas for morphological and reproductive traits we simulated phenotypic values with <i>V</i><sub>Y</sub>=0.1 and <i>V</i><sub>PE</sub>=0.1, thus implying multiple observations per individual. Finally, in dataset 3, different genetic and environmental correlations between traits were also implemented in simulations to assess the difference in power, precision and accuracy when using multivariate models. More specifically, we simulated phenotypes with <i>h</i><sup>2</sup> similar to dataset 1 for the focal trait, whereas fixing <i>h</i><sup>2</sup> of the two other traits at 0.3, for three different values of <i>r</i><sub>A</sub>, 0.1, 0.3 and 0.5, whereas fixing environmental correlation at 0.3. No other variance components were simulated. Simulations were performed in R, and breeding values were simulated with the package <span class=\"u-small-caps\">PEDANTICS</span>. Although simulations were performed within total social and genetic pedigree structures, only individuals with complete information in the empirical dataset were kept in these three simulated datasets.</p><p>Different analyses were performed on simulated datasets to answer two different questions. First, to assess if there was a bias created by not accounting for EPP, we used datasets simulated with the genetic pedigree in animal models using either the genetic pedigree (GG analysis) or the social pedigree (GS analysis), and then the results were compared. Second, to look at inherent differences in reliability caused by pedigree structures, datasets simulated with the social pedigree were analyzed using the social pedigree (SS analysis) and were compared with GG analysis. Animal models used for datasets 1 and 3 included only <i>V</i><sub>A</sub> and <i>V</i><sub>R</sub> as variance components (and covariance in dataset 3), whereas for dataset 2 they included all components described in equations (1) and (2). All these scenarios within each analysis were repeated 100 times for each trait, and animal model analyses were conducted with ASRemL.</p><p>Precision and accuracy for <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> estimates were checked visually with boxplots (that is, median for accuracy and distribution of estimates for precision) and also by computing the mean squared error (MSE) for each scenario. MSE is defined as <img alt=\"\" src=\"//media.springernature.com/lw62/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq4_HTML.gif\" style=\"width:62px;max-width:none;\"/>, where <i>d</i> is the true value (for example, the simulated parameter of <i>h</i><sup>2</sup> or <i>r</i><sub>A</sub>; these values are highly correlated (<i>r</i>&gt;0.99) with the realized value within our simulated population) and <img alt=\"\" src=\"//media.springernature.com/lw7/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq5_HTML.gif\" style=\"width:7px;max-width:none;\"/> is the estimated value; a small MSE indicates high precision and accuracy (<a aria-label=\"Reference Bolker, 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR4\" id=\"ref-link-section-d28693445e1532\" title=\"Bolker B . (2008) Ecological Models and Data in R. Princeton, University Princeton, NJ, USA.\">Bolker, 2008</a>). Root MSE (RMSE) was used to allow a comparison at the scale of the estimates (<a aria-label=\"Reference de Villemereuil et al., 2013\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR43\" id=\"ref-link-section-d28693445e1536\" title=\"de Villemereuil P, Gimenez O, Doligez B . (2013). Comparing parent–offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits. Methods Ecol Evol 4: 260–275.\">de Villemereuil et al., 2013</a>). Power to detect significant <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> within each scenario were estimated using a ‘rule of thumb’ approach by computing the proportion of estimates that were two times larger than their SE. Although LRTs could have provided a formal hypothesis testing, they were not possible to implement in our simulations procedures. Instead, the ‘rule of thumb’ approach is a practical indicator of statistical testing that could be easily integrated to simulation analyses and lead to results comparable to a LRT approach (<a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table A3</a>; see also <a aria-label=\"Reference Wilson et al., 2011\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR47\" id=\"ref-link-section-d28693445e1550\" title=\"Wilson AJ, Morrissey MB, Adams MJ, Walling CA, Guinness FE, Pemberton JM et al. (2011). Indirect genetics effects and evolutionary constraint: an analysis of social dominance in red deer, Cervus elaphus. J Evol Biol 24: 772–783.\">Wilson et al., 2011</a> for a similar approach). Finally, we also checked the identifiability of variance terms in dataset 2 using a Spearman’s rank correlation between estimates of phenotypic variance components. In this case, a low correlation between two components would indicate a high identifiability, whereas a high correlation would indicate a low identifiability.</p></div>\n",
      "<div class=\"c-article-section__content\" id=\"Sec7-content\"><h3 class=\"c-article__sub-heading\" id=\"Sec8\">Pedigrees</h3><p>Our total dataset comprised 13 446 individuals, with 2839 individuals observed only as adults, 10 472 observed only as nestlings and 135 observed at both states (that is, recruits). From the field observations, we were able to determine the identity of the social father for 69.2% of nestlings (82.2% since 2006). Parental analyses allowed us to determine the genetic father for 53.3% of nestling (66.3% since 2006). On average, 1.74±0.82 (range: 1−5) genetic fathers were assigned per brood, and each adult male had on average 4.81±5.44 (range: 0−37) nestlings assigned over its entire lifespan. Overall, 49.3% of nestlings successfully genotyped with a known social father were extra-pair young (<i>N</i><sub>total</sub>=6382).</p><p>Pedigree size varied among the type of pedigree and traits (<a data-track=\"click\" data-track-action=\"table anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Tab1\">Table 1</a>). As expected, the social pedigree always had a higher number of links related to father identities than its genetic counterpart, a difference mainly caused by unsampled males contributing to the genetic pool of nestlings. Nestling traits had a much higher number of links than morphological and reproductive traits, which were both of similar order of magnitude, but with slightly fewer links for the latter (<a data-track=\"click\" data-track-action=\"table anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Tab1\">Table 1</a>). Pedigree sizes were similar for each trait within a given category (<a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table A2</a>).</p><h3 class=\"c-article__sub-heading\" id=\"Sec9\">Estimations of quantitative genetic parameters (empirical data)</h3><p>Variance components, and resulting heritability values, estimated from empirical data varied among traits and also often importantly between pedigrees (social versus genetic) and between models (univariate versus multivariate) (<a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig1\">Figure 1</a>; <a data-track=\"click\" data-track-action=\"table anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Tab2\">Table 2</a>). Biases in <i>h</i><sup>2</sup> and CV<sub>A</sub> resulting from using social instead of genetic pedigrees were moderate for morphological traits (<i>h</i><sup>2</sup>: −0.21 to −0.02; CV<sub>A</sub>: −0.008 to −0.001), almost null for reproductive traits (almost all around 0) and highly variable for nestling traits (<i>h</i><sup>2</sup>: −0.04–0.21; CV<sub>A</sub>: −0.001–0.051). Estimates of <i>r</i><sub>A</sub> obtained with the social and genetic pedigree were similar, except between nestling wing length and body mass where <i>r</i><sub>A</sub> was positive and significantly different from 0 with the social pedigree (<i>r</i><sub>A</sub>=0.56±0.13, <i>χ</i><sup>2</sup>=9.92, <i>P</i>=0.002), but not with the genetic pedigree (<i>r</i><sub>A</sub>=0.25±0.19, χ<sup>2</sup>=1.32, <i>P</i>=0.25; <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure C1</a>). Raw phenotypic variance for all traits can be found in <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table C4</a>.</p><div class=\"c-article-section__figure js-c-reading-companion-figures-item\" data-container-section=\"figure\" data-test=\"figure\" data-title=\"Figure 1\" id=\"figure-1\"><figure><figcaption><b class=\"c-article-section__figure-caption\" data-test=\"figure-caption-text\" id=\"Fig1\">Figure 1</b></figcaption><div class=\"c-article-section__figure-content\"><div class=\"c-article-section__figure-item\"><a class=\"c-article-section__figure-link\" data-test=\"img-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-label=\"image\" href=\"/articles/hdy201692/figures/1\" rel=\"nofollow\"><picture><source srcset=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig1_HTML.jpg?as=webp\" type=\"image/webp\"/><img alt=\"figure 1\" aria-describedby=\"Fig1\" height=\"705\" loading=\"lazy\" src=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig1_HTML.jpg\" width=\"560\"/></picture></a></div><div class=\"c-article-section__figure-description\" data-test=\"bottom-caption\" id=\"figure-1-desc\"><p>Proportion of phenotypic variance estimated from final animal models on empirical dataset, for (<b>a</b>) morphological, (<b>b</b>) reproductive and (<b>c</b>) nestling traits. Different models were assessed for each trait: univariate models using social (SU) and genetic (GU) pedigree, and multivariate models using social (SM) and genetic (GM) pedigree.</p></div></div><div class=\"u-text-right u-hide-print\"><a aria-label=\"Full size image figure 1\" class=\"c-article__pill-button\" data-test=\"article-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-dest=\"link:Figure1 Full size image\" data-track-label=\"button\" href=\"/articles/hdy201692/figures/1\" rel=\"nofollow\"><span>Full size image</span><svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-chevron-right\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></div></figure></div><div class=\"c-article-table\" data-container-section=\"table\" data-test=\"inline-table\" id=\"table-2\"><figure><figcaption class=\"c-article-table__figcaption\"><b data-test=\"table-caption\" id=\"Tab2\">Table 2 Summary of final A) univariate and B) multivariate animal models using empirical datasets and either the social pedigree or the genetic pedigree</b></figcaption><div class=\"u-text-right u-hide-print\"><a aria-label=\"Full size table 2\" class=\"c-article__pill-button\" data-test=\"table-link\" data-track=\"click\" data-track-action=\"view table\" data-track-label=\"button\" href=\"/articles/hdy201692/tables/2\" rel=\"nofollow\"><span>Full size table</span><svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-chevron-right\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></div></figure></div><h3 class=\"c-article__sub-heading\" id=\"Sec10\">Reliability of quantitative genetics estimates (simulated data)</h3><p>Visual inspection of boxplots (<a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig2\">Figures 2b, d, f</a> and <a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig3\">3b</a>) and comparison of RMSE values (<a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figures D1</a>) showed complementary information on precision and accuracy of <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> estimates. Precision of <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> estimates varied greatly depending on trait category (nestling&gt;morphological&gt;reproductive; mean±s.d. RMSE for <i>h</i><sup>2</sup>/<i>r</i><sub>A</sub>: <img alt=\"\" src=\"//media.springernature.com/lw46/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq6_HTML.gif\" style=\"width:46px;max-width:none;\"/>=0.055±0.045/0.081±0.025, <img alt=\"\" src=\"//media.springernature.com/lw45/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq7_HTML.gif\" style=\"width:45px;max-width:none;\"/>=0.122±0.026/0.342±0.061, <img alt=\"\" src=\"//media.springernature.com/lw33/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq8_HTML.gif\" style=\"width:33px;max-width:none;\"/>=0.200±0.045/0.430±0.025) but also among datasets used (dataset 2 (simulation of <i>h</i><sup>2</sup> and other source of resemblance among individuals)&gt;3 (simulation of <i>h</i><sup>2</sup> and <i>r</i><sub><i>A</i></sub>)&gt;1 (simulation of <i>h</i><sup>2</sup> only); mean±s.d. RMSE for <i>h</i><sup>2</sup>: <img alt=\"\" src=\"//media.springernature.com/lw49/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq9_HTML.gif\" style=\"width:49px;max-width:none;\"/>=0.144±0.086, <img alt=\"\" src=\"//media.springernature.com/lw45/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq10_HTML.gif\" style=\"width:45px;max-width:none;\"/>=0.100±0.047, <img alt=\"\" src=\"//media.springernature.com/lw45/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq11_HTML.gif\" style=\"width:45px;max-width:none;\"/>=0.127±0.072). Accuracy for <i>h</i><sup>2</sup> estimates was high for all GG and SS analyses (except for reproductive traits in dataset 3), but downwardly biased in GS analyses for nestling and reproductive traits, an effect more important as simulated <i>h</i><sup>2</sup> increased (mean±s.d. RMSE: <img alt=\"\" src=\"//media.springernature.com/lw27/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq12_HTML.gif\" style=\"width:27px;max-width:none;\"/>=0.117±0.076, <img alt=\"\" src=\"//media.springernature.com/lw24/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq13_HTML.gif\" style=\"width:24px;max-width:none;\"/>=0.143±0.061, <img alt=\"\" src=\"//media.springernature.com/lw31/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq14_HTML.gif\" style=\"width:31px;max-width:none;\"/>=0.115±0.075). Accuracy for <i>r</i><sub>A</sub> were similar for all type of analysis (mean±s.d. RMSE: <img alt=\"\" src=\"//media.springernature.com/lw31/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq15_HTML.gif\" style=\"width:31px;max-width:none;\"/>=0.283±0.160, <img alt=\"\" src=\"//media.springernature.com/lw29/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq16_HTML.gif\" style=\"width:29px;max-width:none;\"/>=0.294±0.151, <img alt=\"\" src=\"//media.springernature.com/lw21/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq17_HTML.gif\" style=\"width:21px;max-width:none;\"/>=0.275±0.158).</p><div class=\"c-article-section__figure js-c-reading-companion-figures-item\" data-container-section=\"figure\" data-test=\"figure\" data-title=\"Figure 2\" id=\"figure-2\"><figure><figcaption><b class=\"c-article-section__figure-caption\" data-test=\"figure-caption-text\" id=\"Fig2\">Figure 2</b></figcaption><div class=\"c-article-section__figure-content\"><div class=\"c-article-section__figure-item\"><a class=\"c-article-section__figure-link\" data-test=\"img-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-label=\"image\" href=\"/articles/hdy201692/figures/2\" rel=\"nofollow\"><picture><source srcset=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig2_HTML.jpg?as=webp\" type=\"image/webp\"/><img alt=\"figure 2\" aria-describedby=\"Fig2\" height=\"871\" loading=\"lazy\" src=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig2_HTML.jpg\" width=\"685\"/></picture></a></div><div class=\"c-article-section__figure-description\" data-test=\"bottom-caption\" id=\"figure-2-desc\"><p>Power, precision and accuracy of heritability estimated from animal models for dataset 1 (<b>a</b> and <b>b</b>; simulation of <i>h</i><sup>2</sup> only), dataset 2 (<b>c</b> and <b>d</b>; simulation of <i>h</i><sup>2</sup> and other source of resemblance among individuals) and dataset 3 (<b>e</b> and <b>f</b>; simulation of <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub>) simulated on the pedigree structure of morphological traits (dark gray lines), reproductive traits (light gray lines) and nestling traits (black lines). Power of detecting significant heritability over 300 simulated trait values (100 simulations per trait) are presented in <b>a</b>, <b>c</b> and <b>e</b>. Distributions of these estimates are represented by boxplots (1st quartile, median, 3rd quartile) for 2 levels of heritability (dotted lines represent the <i>h</i><sup>2</sup> simulated value of 0.1 and 0.5) in <b>b</b>, <b>d</b> and <b>f</b>. Analysis types refer to datasets simulated with the genetic pedigree and analyzed with the genetic (GG analysis, solid lines) or the social pedigree (GS analysis, dashed lines), and datasets simulated with the social pedigree and analyzed with the social pedigree (SS analysis, dotted lines). For graphical representation, <i>r</i><sub>A</sub> was fixed at 0.3 in <b>e</b> and <b>f</b>.</p></div></div><div class=\"u-text-right u-hide-print\"><a aria-label=\"Full size image figure 2\" class=\"c-article__pill-button\" data-test=\"article-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-dest=\"link:Figure2 Full size image\" data-track-label=\"button\" href=\"/articles/hdy201692/figures/2\" rel=\"nofollow\"><span>Full size image</span><svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-chevron-right\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></div></figure></div><div class=\"c-article-section__figure js-c-reading-companion-figures-item\" data-container-section=\"figure\" data-test=\"figure\" data-title=\"Figure 3\" id=\"figure-3\"><figure><figcaption><b class=\"c-article-section__figure-caption\" data-test=\"figure-caption-text\" id=\"Fig3\">Figure 3</b></figcaption><div class=\"c-article-section__figure-content\"><div class=\"c-article-section__figure-item\"><a class=\"c-article-section__figure-link\" data-test=\"img-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-label=\"image\" href=\"/articles/hdy201692/figures/3\" rel=\"nofollow\"><picture><source srcset=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig3_HTML.jpg?as=webp\" type=\"image/webp\"/><img alt=\"figure 3\" aria-describedby=\"Fig3\" height=\"292\" loading=\"lazy\" src=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig3_HTML.jpg\" width=\"685\"/></picture></a></div><div class=\"c-article-section__figure-description\" data-test=\"bottom-caption\" id=\"figure-3-desc\"><p>Power, precision and accuracy of genetic additive correlation estimated from multivariate animal models, for dataset 3 (simulation of <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub>) simulated on the pedigree structure of morphological traits (dark gray lines), reproductive traits (light gray lines) and nestling traits (black lines). Power of detecting significant genetic correlation over 300 simulated trait values (100 simulations per trait), is presented in <b>a</b> and distributions of these estimates are represented by boxplots (1st quartile, median, 3rd quartile) in <b>b</b> (for 1 value of <i>r</i><sub>A</sub>—dotted lines represent the <i>r</i><sub>A</sub> true value of 0.5 simulated; 5 estimates&gt;1 are not presented). Analysis types refer to datasets simulated with the genetic pedigree and analyzed with the genetic (GG analysis, solid lines) or the social pedigree (GS analysis, dashed lines), and datasets simulated with the social pedigree and analyzed with the social pedigree (SS analysis, dotted lines). For graphical representation, <i>h</i><sup>2</sup> was fixed at 0.5 in <b>b</b>.</p></div></div><div class=\"u-text-right u-hide-print\"><a aria-label=\"Full size image figure 3\" class=\"c-article__pill-button\" data-test=\"article-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-dest=\"link:Figure3 Full size image\" data-track-label=\"button\" href=\"/articles/hdy201692/figures/3\" rel=\"nofollow\"><span>Full size image</span><svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-chevron-right\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></div></figure></div><p>Power to detect significant <i>h</i><sup>2</sup> or <i>r</i><sub>A</sub> was greater for traits with larger sample size (nestling&gt;morphological&gt;reproductive; <a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig2\">Figures 2a, c, e</a> and <a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig3\">3a</a>), but we observed no difference among traits within each category (data not shown). For multivariate animal models from dataset 3, power to detect <i>h</i><sup>2</sup> was similar for the different genetic correlations tested (data not shown), whereas power to detect genetic correlation increased with increasing heritability (<a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig3\">Figure 3a</a>). Within a trait category, power was similar between GG and SS analyses, except for nestling traits from dataset 2 were the simulated brood effect decreased power in SS analysis (<a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig2\">Figure 2c</a>). Finally, power comparison between GG and GS analyses revealed generally lower power in morphological and nestling trait for GS (<a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig2\">Figures 2a, c, e</a> and <a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig3\">3a</a>).</p><p>For adult traits, <i>V</i><sub>A</sub> and <i>V</i><sub>PE</sub> estimates from dataset 2 were highly negatively correlated, suggesting that these terms were almost completely confounded (morphological traits: <i>r</i><sub>s</sub>=−0.95, <i>P</i>&lt;0.001; reproductive traits: <i>r</i><sub>s</sub>=−0.94, <i>P</i>&lt;0.001, results from GG analysis with <i>h</i><sup>2</sup>=0.3; <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figures D3</a>). For nestling traits, <i>V</i><sub>A</sub> and <i>V</i><sub>B</sub> estimates were also correlated, but to a lesser extent (<i>r</i><sub>s</sub>=−0.42, <i>P</i>&lt;0.001; <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure D5</a>). All other variance terms showed correlations non-significantly different from 0 (all |<i>r</i><sub>s</sub>|&lt;0.09, <i>P</i>&gt;0.12).</p><p>Finally, given the large differences observed for some reproductive traits between SEs of <i>h</i><sup>2</sup> estimates from empirical and simulated dataset (<a data-track=\"click\" data-track-action=\"table anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Tab2\">Table 2</a>; <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure D6a</a>), we further inspected the impact of model parameters on SEs. More specifically, we assessed the impact of not accounting for an important cause of resemblance among individuals by running dataset 2 animal models and omitting <i>V</i><sub>PE</sub> or <i>V</i><sub>B</sub> components. By doing so, we observed a power of 1 at all traits, and all <i>h</i><sup>2</sup> estimates were very precise but showed a bias equivalent to the variance component not accounted for (that is, 0.1 for morphological and reproductive trait and 0.4 for nestling traits; <a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig4\">Figure 4</a>). As suspected, SEs for these biased <i>h</i><sup>2</sup> estimates were small and similar to those obtained in empirical analyses, where <i>V</i><sub>PE</sub> was almost completely confounded with <i>V</i><sub>A</sub> (<a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure D6b</a>).</p><div class=\"c-article-section__figure js-c-reading-companion-figures-item\" data-container-section=\"figure\" data-test=\"figure\" data-title=\"Figure 4\" id=\"figure-4\"><figure><figcaption><b class=\"c-article-section__figure-caption\" data-test=\"figure-caption-text\" id=\"Fig4\">Figure 4</b></figcaption><div class=\"c-article-section__figure-content\"><div class=\"c-article-section__figure-item\"><a class=\"c-article-section__figure-link\" data-test=\"img-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-label=\"image\" href=\"/articles/hdy201692/figures/4\" rel=\"nofollow\"><picture><source srcset=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig4_HTML.jpg?as=webp\" type=\"image/webp\"/><img alt=\"figure 4\" aria-describedby=\"Fig4\" height=\"340\" loading=\"lazy\" src=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig4_HTML.jpg\" width=\"556\"/></picture></a></div><div class=\"c-article-section__figure-description\" data-test=\"bottom-caption\" id=\"figure-4-desc\"><p>Precision and accuracy of heritability estimated from animal models omitting an important cause of resemblance among individuals (<i>V</i><sub>PE</sub> or <i>V</i><sub>B</sub>), for morphological traits, reproductive traits and nestling traits simulated in dataset 2. Distribution of these estimates are represented by boxplots (1st quartile, median, 3rd quartile) for 2 levels of heritability (dotted lines represent the <i>h</i><sup>2</sup> true value of 0.1 and 0.5 simulated). Analysis types refer to datasets simulated with the genetic pedigree and analyzed with both the genetic and the social pedigree (GG and GS analyses, respectively), and datasets simulated with the social pedigree and analyzed with the social pedigree (SS analysis).</p></div></div><div class=\"u-text-right u-hide-print\"><a aria-label=\"Full size image figure 4\" class=\"c-article__pill-button\" data-test=\"article-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-dest=\"link:Figure4 Full size image\" data-track-label=\"button\" href=\"/articles/hdy201692/figures/4\" rel=\"nofollow\"><span>Full size image</span><svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-chevron-right\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></div></figure></div></div>\n",
      "<div class=\"c-article-section__content\" id=\"Sec11-content\"><p>Despite the importance of obtaining precise and accurate genetic (co)variance estimates when assessing a population’s adaptive potential, their reliability is rarely explicitly tested or reported in wild populations. Here, we formally assessed the reliability of genetic (co)variance estimates, in a species with a high rate of extra-pair reproduction and low recruitment, with a combination of empirical and simulated data. Altogether, our simulation analyses emphasized the limits of this particular dataset by revealing an important lack of precision in <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> estimates for all adult traits, a lack of power to detect significant effects and identifiability problems between <i>V</i><sub>A</sub> and V<sub>PE</sub>. Moreover, we observed a large bias in <i>h</i><sup><i>2</i></sup> when using the social pedigree instead of the genetic one, and also when non-genetic causes of resemblance among individuals (that is, repeated measurements or brood effects) were not accounted for in our analyses. We briefly discuss below (i) the factors contributing to estimate reliability, (ii) the caution needed with model parameterization, (iii) the impacts of high levels of EPP on genetic variance estimates and (iv) we finally conclude with remarks on the accessibility and utility of simulation analyses to address all these potential problems.</p><h3 class=\"c-article__sub-heading\" id=\"Sec12\">Estimate reliability and generalization among studies</h3><p>At first glance, our sample size seemed large enough to be powerful, with more than 10 years of sampling and a number of records within our pruned pedigrees that was larger than the median number of records typically reported for similar studies in the literature. Indeed, a recent compilation of quantitative genetics estimates obtained from wild populations by <a aria-label=\"Reference Postma (2014)\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR39\" id=\"ref-link-section-d28693445e3130\" title=\"Postma E . (2014). Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates? In: Charmantier A, Garant D et al. (eds) Quantitative Genetics in the Wild. Oxford University Press, Oxford, UK, pp 16–33.\">Postma (2014)</a> showed that for estimates obtained with animal models, the median number of records for life history traits was 377 (range 6–4992; from 39 studies, covering 19 species) and 363 for morphological traits (range 50–38 024; from 47 studies, covering 22 species). This suggests that sample size alone is not sufficient to infer pedigree quality, as it does not reflect the connectivity of pedigrees (<a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e3133\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). In our study system, the low recruitment of nestlings (1.3%) results in very few grandparent links within our pedigree, which greatly reduces its power. However, this problem is far less important for nestling traits as the high-EPP rate results in a genetic pedigree containing several half-sib families, which increase its power (see GG versus SS, <a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig2\">Figure 2c</a>) despite a smaller number of observations compared with the social pedigree for all categories (<a data-track=\"click\" data-track-action=\"table anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Tab1\">Table 1</a>). Thus, given that pedigree structures are different from one study system to another, their impact on reliability of quantitative genetic estimates is best predicted from simulation analyses.</p><p>Various ecological and population specific factors could generate fluctuations in the precision and accuracy of quantitative genetic estimates among study systems. Here, our study system was characterized by a high rate of EPP and a low recruitment, two characteristics that affected the reliability of quantitative genetic estimates for some trait types. Although these characteristics are certainly common in other migratory species (for example, Common reed bunting (<i>Emberiza schoeniclus</i>) and Red-eyed vireo (<i>Vireo olivaceus</i>), <a aria-label=\"Reference Griffith et al., 2002\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR13\" id=\"ref-link-section-d28693445e3151\" title=\"Griffith SC, Owens IPF, Thuman KA . (2002). Extra pair paternity in birds: a review of interspecific variation and adaptive function. Mol Ecol 11: 2195–2212.\">Griffith et al., 2002</a>), other characteristics, such as low-recapture rate between years in open study systems, a sex bias in sampling or large natal dispersal may also have important consequences on the precision of estimates. Moreover, differential mortality or capture rate might bias the sampling of some traits, and consequently affect the accuracy of estimates that can be obtained (for example, <a aria-label=\"Reference Dingemanse et al., 2009\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR6\" id=\"ref-link-section-d28693445e3154\" title=\"Dingemanse NJ, Kazem AJN, Réale D, Wright J . (2009). Behavioural reaction norms: animal personality meets individual plasticity. Trends Ecol Evol 25: 81–89.\">Dingemanse et al., 2009</a>). It is thus important to appropriately identify the limits in a particular system and assess their impacts on the reliability of quantitative genetic estimates by using simulations.</p><h3 class=\"c-article__sub-heading\" id=\"Sec13\">Caution needed with model parameterization</h3><p>The choice of which variance terms to include in a model is a crucial step that can have substantial impacts on reliability of genetic variance estimates (<a aria-label=\"Reference Kruuk and Hadfield, 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d28693445e3165\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890–1903.\">Kruuk and Hadfield, 2007</a>; <a aria-label=\"Reference Wilson, 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR46\" id=\"ref-link-section-d28693445e3168\" title=\"Wilson AJ . (2008). Why h2 does not always equal VA/VP? J Evol Biol 21: 647–650.\">Wilson, 2008</a>). Here, the decision to include or not a particular variance term relied on LRTs, which is a widely used approach (<a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e3171\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). However, in some cases even if the inclusion of a particular component is not improving the model likelihood, it may still have to be included (<a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e3174\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). This is the case for component of variance attributable to repeated measurement of a given individual (<i>V</i><sub>PE</sub>). In our models, we had to account for multiple observations by fitting a <i>V</i><sub>PE</sub> term in our model, but this component was almost completely confounded with <i>V</i><sub>A</sub>. This situation probably occurred because of the low number of observations per individual (mean of 1.5 (averaged across traits; range: 1–7) for morphological traits and 1.2 (range: 1–7) for reproductive traits), a situation that should be present in other short-lived species datasets. Not accounting for repeated measurements in an animal model is equivalent to considering each repeated measurement as a clone of the same individual, which is inappropriate. In such cases, using a dataset with only one observation per individual may lead to more accurate estimates and easier model convergence.</p><p>Standard errors are often viewed as predictors of an estimate’s accuracy or significance, but this can be misleading (<a aria-label=\"Reference Krzywinski and Altman, 2013\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR27\" id=\"ref-link-section-d28693445e3193\" title=\"Krzywinski M, Altman N . (2013). Points of significance: error bars. Nat Methods 10: 921–922.\">Krzywinski and Altman, 2013</a>). Estimates from animal models have generally smaller standard errors than those obtained with parent–offspring regressions (<a aria-label=\"Reference Kruuk, 2004\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR24\" id=\"ref-link-section-d28693445e3196\" title=\"Kruuk LEB . (2004). Estimating genetic parameters in natural populations using the ‘animal model’. Philos Trans R Soc B 359: 873–890.\">Kruuk, 2004</a>; <a aria-label=\"Reference Postma, 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR39\" id=\"ref-link-section-d28693445e3199\" title=\"Postma E . (2014). Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates? In: Charmantier A, Garant D et al. (eds) Quantitative Genetics in the Wild. Oxford University Press, Oxford, UK, pp 16–33.\">Postma, 2014</a>), partly due to their integration of multiple observations for a given individual (<a aria-label=\"Reference Åkesson et al., 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR1\" id=\"ref-link-section-d28693445e3202\" title=\"Åkesson M, Bensch S, Hasselquist D, Tarka M, Hansson B, Akesson M . (2008). Estimating heritabilities and genetic correlations: comparing the ‘animal model’ with parent–offspring regression using data from a natural population. PLoS One 3: e1739.\">Åkesson et al., 2008</a>). As previously stated, failing to account for multiple observations can create unpredictable bias in phenotypic variance component estimates (<a aria-label=\"Reference Kruuk and Hadfield, 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d28693445e3205\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890–1903.\">Kruuk and Hadfield, 2007</a>) and considerably reduce standard errors around these biased estimates (<a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure D6</a>). Moreover, it seems that problems of identifiability could also result in reduced standard errors. For example, in our empirical analyses, small standard errors were estimated around <i>h</i><sup>2</sup> estimates for reproductive traits from univariate animal models with repeated measurements per individual (SE range: 0.03–0.04), which could have led us to misleadingly conclude that our dataset was powerful and our estimates were precise for these traits.</p><p>This study was performed with a frequentist method (that is, REML) rather than a Bayesian method (that is, MCMC) because it generally gives more precise and accurate estimates for traits following a Gaussian distribution and it necessitates a shorter computation time (<a aria-label=\"Reference de Villemereuil et al., 2013\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR43\" id=\"ref-link-section-d28693445e3219\" title=\"de Villemereuil P, Gimenez O, Doligez B . (2013). Comparing parent–offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits. Methods Ecol Evol 4: 260–275.\">de Villemereuil et al., 2013</a>). Bayesian methods have other advantages, such as the use of posterior distributions for the calculation of derived parameters (for example, heritability) and associated standard errors and they are particularly useful for analyses of non-Gaussian traits (<a aria-label=\"Reference Morrissey et al., 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR36\" id=\"ref-link-section-d28693445e3222\" title=\"Morrissey MB, de Villemereuil P, Doligez B, Gimenez O . (2014). Bayesian approaches to the quantitative genetic analysis of natural populations. In: Charmantier A, Garant D et al. (eds) Quantitative Genetics in the Wild. Oxford University Press, Oxford, UK, pp 228–253.\">Morrissey et al., 2014</a>). In the present context, problems related to missing of an important source of resemblance among individuals are not method specific and could impact any model regardless of the method used for its estimation. However, it is less clear if identifiability problems could be better uncovered with Bayesian approach, for example through an appropriate MCMC checking. Simulation studies such as this one could be applied to animal models resolved with a Bayesian approach to better assess its limits.</p><h3 class=\"c-article__sub-heading\" id=\"Sec14\">Impact of high level of EPP on estimates</h3><p>In theory, EPPs, if not accounted for, could downwardly bias additive genetic variance and resulting heritability estimates (<a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e3233\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>). Yet, previous simulation studies showed that even if biases increased with the importance of EPPs, with increasing heritability and when focal traits were directly related to the number of extra-pair young produced, underestimations were generally smaller than 15% (<a aria-label=\"Reference Firth et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d28693445e3236\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336–1344.\">Firth et al., 2015</a>). In our study, the rate of EPP was higher than those previously tested so far (up to 40% in <a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e3239\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>, 12.5% in <a aria-label=\"Reference Firth et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d28693445e3242\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336–1344.\">Firth et al., 2015</a>), but its effect on quantitative genetics estimates was complex. First, our simulation analyses showed that the bias in <i>h</i><sup>2</sup> estimates resulting from using the social instead of the genetic pedigree changed depending on the trait category (that is, higher for nestling traits). Also, our empirical analyses suggested that the impact of not accounting for EPP was noticeable for morphological and nestling traits, but sometimes resulted in higher <i>h</i><sup>2</sup> and CV<sub>A</sub> when using the social pedigree (that is, for nestling wing length and body mass). A similar unexpected finding was previously reported in blue tits (<i>Cyanistes caeruleus</i>), where <i>h</i><sup>2</sup> estimates from a social pedigree were sometimes higher than those from a genetic pedigree (<a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e3263\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>). These positive biases observed in empirical data could be due to the influence of the social father (for example through parental care), which could be captured in <i>V</i><sub>A</sub> components (<a aria-label=\"Reference Griffith et al., 1999\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR12\" id=\"ref-link-section-d28693445e3271\" title=\"Griffith SC, Owens IPF, Burke T . (1999). Environmental determination of a sexually selected trait. Science 400: 358–360.\">Griffith et al., 1999</a>; <a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e3274\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>) rather than due to pedigree errors. To assess this possible problem, we performed an additional animal model analysis including mother and social father identities as additional variance components. We found that new <i>h</i><sup>2</sup> and CV<sub>A</sub> values obtained using the social pedigree were smaller than values obtained with the genetic one (change in estimates for body mass <i>h</i><sup>2</sup>=−0.06, CV<sub>A</sub>=−0.011; wing length <i>h</i><sup>2</sup>=−0.04, CV<sub>A</sub>=−0.041; difference calculated with univariate animal models; see <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table C5</a> for details). This further emphasizes the importance of model parameterization on reliability of variance component estimates. Finally, even if the genetic pedigree better represents relationships among individuals than the social pedigree, it probably still contains a few incorrect links among individuals and thus the estimated heritabilities might also be slightly downwardly biased.</p></div>\n",
      "<div class=\"c-article-section__content\" id=\"Sec15-content\"><p>Simulation analyses are now widely accessible to anyone with minimal programming skills in R, for instance by using the ‘<i>phensim</i>’ function within the package <span class=\"u-small-caps\">PEDANTICS</span>, the ‘<i>rbv</i>’ function within the package MCMCglmm (<a aria-label=\"Reference Hadfield, 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR14\" id=\"ref-link-section-d28693445e3317\" title=\"Hadfield JD . (2010). MCMC methods for multi-response generalized linear mixed models: the MCMCglmm R package. J Stat Softw 33: 1–22.\">Hadfield, 2010</a>) or the ‘<i>drfx</i>’ and the ‘<i>grfx</i>’ function within the package <span class=\"u-small-caps\">NADIV</span> (<a aria-label=\"Reference Wolak, 2012\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR50\" id=\"ref-link-section-d28693445e3330\" title=\"Wolak ME . (2012). nadiv : an R package to create relatedness matrices for estimating non-additive genetic variances in animal models. Methods Ecol Evol 3: 792–796.\">Wolak, 2012</a>). Moreover, other methods such as profile likelihoods (<a aria-label=\"Reference Meyer, 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR34\" id=\"ref-link-section-d28693445e3333\" title=\"Meyer K . (2008). Likelihood calculations to evaluate experimental designs to estimate genetic variances. Heredity 101: 212–221.\">Meyer, 2008</a>; implemented in the ‘proLik’ function within the package <span class=\"u-small-caps\">NADIV,</span> <a aria-label=\"Reference Wolak, 2012\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR50\" id=\"ref-link-section-d28693445e3339\" title=\"Wolak ME . (2012). nadiv : an R package to create relatedness matrices for estimating non-additive genetic variances in animal models. Methods Ecol Evol 3: 792–796.\">Wolak, 2012</a>) are more accessible and can also improve our comprehension of a particular dataset limits (<a aria-label=\"Reference Houle and Meyer, 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR20\" id=\"ref-link-section-d28693445e3343\" title=\"Houle D, Meyer K . (2015). Estimating sampling error of evolutionary statistics based on genetic covariance matrices using maximum likelihood. J Evol Biol 28: 1542–1549.\">Houle and Meyer, 2015</a>). As each study system is unique, in terms of pedigree structure, number of repeated measures per individual or potential causes of pedigree errors, it is difficult to predict the reliability of quantitative genetics estimates without testing it formally. Detailed simulations should be routinely included when reporting quantitative genetics analyses performed within a new study system, or when applying more complex modeling to previously studied populations, to explicitly assess the precision and accuracy of genetic variance components (<a aria-label=\"Reference Morrissey et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d28693445e3346\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309–2321.\">Morrissey et al., 2007</a>).</p></div>\n",
      "<div class=\"c-article-section__content\" id=\"Sec16-content\"><p>Data available from the Dryad Digital Repository: <a href=\"https://doi.org/10.5061/dryad.n8c57\">http://dx.doi.org/10.5061/dryad.n8c57</a>.</p></div>\n",
      "<div class=\"c-article-section__content\" id=\"Bib1-content\"><div data-container-section=\"references\"><ul class=\"c-article-references\" data-track-component=\"outbound reference\"><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR1\">Åkesson M, Bensch S, Hasselquist D, Tarka M, Hansson B, Akesson M . (2008). Estimating heritabilities and genetic correlations: comparing the ‘animal model’ with parent–offspring regression using data from a natural population. <i>PLoS One</i> <b>3</b>: e1739.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 1\" data-doi=\"10.1371/journal.pone.0001739\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1371/journal.pone.0001739\" href=\"https://doi.org/10.1371%2Fjournal.pone.0001739\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 1\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18320057\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 1\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2254494\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 1\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimating%20heritabilities%20and%20genetic%20correlations%3A%20comparing%20the%20%E2%80%98animal%20model%E2%80%99%20with%20parent%E2%80%93offspring%20regression%20using%20data%20from%20a%20natural%20population&amp;journal=PLoS%20One&amp;doi=10.1371%2Fjournal.pone.0001739&amp;volume=3&amp;publication_year=2008&amp;author=%C3%85kesson%2CM&amp;author=Bensch%2CS&amp;author=Hasselquist%2CD&amp;author=Tarka%2CM&amp;author=Hansson%2CB&amp;author=Akesson%2CM\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR2\">Bérénos C, Ellis PA, Pilkington JG, Pemberton JM . (2014). Estimating quantitative genetic parameters in wild populations: a comparison of pedigree and genomic approaches. <i>Mol Ecol</i> <b>23</b>: 3434–3451.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 2\" data-doi=\"10.1111/mec.12827\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/mec.12827\" href=\"https://doi.org/10.1111%2Fmec.12827\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 2\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=24917482\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 2\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4149785\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 2\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimating%20quantitative%20genetic%20parameters%20in%20wild%20populations%3A%20a%20comparison%20of%20pedigree%20and%20genomic%20approaches&amp;journal=Mol%20Ecol&amp;doi=10.1111%2Fmec.12827&amp;volume=23&amp;pages=3434-3451&amp;publication_year=2014&amp;author=B%C3%A9r%C3%A9nos%2CC&amp;author=Ellis%2CPA&amp;author=Pilkington%2CJG&amp;author=Pemberton%2CJM\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR3\">Blows MW, Hoffmann AA . (2005). A reassessment of genetic limits to evoutionary change. <i>Ecology</i> <b>86</b>: 1371–1384.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 3\" data-doi=\"10.1890/04-1209\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1890/04-1209\" href=\"https://doi.org/10.1890%2F04-1209\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 3\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=A%20reassessment%20of%20genetic%20limits%20to%20evoutionary%20change&amp;journal=Ecology&amp;doi=10.1890%2F04-1209&amp;volume=86&amp;pages=1371-1384&amp;publication_year=2005&amp;author=Blows%2CMW&amp;author=Hoffmann%2CAA\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR4\">Bolker B . (2008) <i>Ecological Models and Data in R</i>. Princeton, University Princeton, NJ, USA.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Book reference 4\" data-doi=\"10.2307/j.ctvcm4g37\" data-track=\"click\" data-track-action=\"book reference\" data-track-label=\"10.2307/j.ctvcm4g37\" href=\"https://doi.org/10.2307%2Fj.ctvcm4g37\" rel=\"nofollow noopener\">Book</a> \n",
      "    <a aria-label=\"Google Scholar reference 4\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Ecological%20Models%20and%20Data%20in%20R&amp;doi=10.2307%2Fj.ctvcm4g37&amp;publication_year=2008&amp;author=Bolker%2CB\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR5\">Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? <i>Mol Ecol</i> <b>14</b>: 2839–2850.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 5\" data-doi=\"10.1111/j.1365-294X.2005.02619.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1365-294X.2005.02619.x\" href=\"https://doi.org/10.1111%2Fj.1365-294X.2005.02619.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 5\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:CAS:528:DC%2BD2MXpt1Cju7g%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 5\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=16029482\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 5\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=How%20do%20misassigned%20paternities%20affect%20the%20estimation%20of%20heritability%20in%20the%20wild%3F&amp;journal=Mol%20Ecol&amp;doi=10.1111%2Fj.1365-294X.2005.02619.x&amp;volume=14&amp;pages=2839-2850&amp;publication_year=2005&amp;author=Charmantier%2CA&amp;author=R%C3%A9ale%2CD\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR6\">Dingemanse NJ, Kazem AJN, Réale D, Wright J . (2009). Behavioural reaction norms: animal personality meets individual plasticity. <i>Trends Ecol Evol</i> <b>25</b>: 81–89.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 6\" data-doi=\"10.1016/j.tree.2009.07.013\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1016/j.tree.2009.07.013\" href=\"https://doi.org/10.1016%2Fj.tree.2009.07.013\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 6\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=19748700\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 6\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Behavioural%20reaction%20norms%3A%20animal%20personality%20meets%20individual%20plasticity&amp;journal=Trends%20Ecol%20Evol&amp;doi=10.1016%2Fj.tree.2009.07.013&amp;volume=25&amp;pages=81-89&amp;publication_year=2009&amp;author=Dingemanse%2CNJ&amp;author=Kazem%2CAJN&amp;author=R%C3%A9ale%2CD&amp;author=Wright%2CJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR7\">Dunn PO, Whittingham LA, Pitcher TE . (2001). Mating systems, sperm competition, and the evolution of sexual dimorphism in birds. <i>Evolution</i> <b>55</b>: 161–175.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 7\" data-doi=\"10.1111/j.0014-3820.2001.tb01281.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.0014-3820.2001.tb01281.x\" href=\"https://doi.org/10.1111%2Fj.0014-3820.2001.tb01281.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 7\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD3M7mt1Gmsw%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 7\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=11263736\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 7\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Mating%20systems%2C%20sperm%20competition%2C%20and%20the%20evolution%20of%20sexual%20dimorphism%20in%20birds&amp;journal=Evolution&amp;doi=10.1111%2Fj.0014-3820.2001.tb01281.x&amp;volume=55&amp;pages=161-175&amp;publication_year=2001&amp;author=Dunn%2CPO&amp;author=Whittingham%2CLA&amp;author=Pitcher%2CTE\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR8\">Falconer DS, Mackay TFC . (1996) <i>Introduction to Quantitative Genetics</i>, 4th edn. Pearson: Essex, UK.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Google Scholar reference 8\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Introduction%20to%20Quantitative%20Genetics&amp;publication_year=1996&amp;author=Falconer%2CDS&amp;author=Mackay%2CTFC\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR9\">Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. <i>Evolution</i> <b>69</b>: 1336–1344.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 9\" data-doi=\"10.1111/evo.12649\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/evo.12649\" href=\"https://doi.org/10.1111%2Fevo.12649\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 9\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=25800997\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 9\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4950017\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 9\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20influence%20of%20nonrandom%20extra-pair%20paternity%20on%20heritability%20estimates%20derived%20from%20wild%20pedigrees&amp;journal=Evolution&amp;doi=10.1111%2Fevo.12649&amp;volume=69&amp;pages=1336-1344&amp;publication_year=2015&amp;author=Firth%2CJA&amp;author=Hadfield%2CJD&amp;author=Santure%2CAW&amp;author=Slate%2CJ&amp;author=Sheldon%2CBC\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR10\">Garcia-Gonzalez F, Simmons LW, Tomkins JL, Kotiaho JS, Evans JP . (2012). Comparing evolvabilities: common errors surrounding the calculation and use of coefficients of additive genetic variation. <i>Evolution</i> <b>66</b>: 2341–2349.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 10\" data-doi=\"10.1111/j.1558-5646.2011.01565.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1558-5646.2011.01565.x\" href=\"https://doi.org/10.1111%2Fj.1558-5646.2011.01565.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 10\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=22834736\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 10\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Comparing%20evolvabilities%3A%20common%20errors%20surrounding%20the%20calculation%20and%20use%20of%20coefficients%20of%20additive%20genetic%20variation&amp;journal=Evolution&amp;doi=10.1111%2Fj.1558-5646.2011.01565.x&amp;volume=66&amp;pages=2341-2349&amp;publication_year=2012&amp;author=Garcia-Gonzalez%2CF&amp;author=Simmons%2CLW&amp;author=Tomkins%2CJL&amp;author=Kotiaho%2CJS&amp;author=Evans%2CJP\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR11\">Ghilain A, Bélisle M . (2008). Breeding success of tree swallows along a gradient of agricultural intensification. <i>Ecol Appl</i> <b>18</b>: 1140–1154.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 11\" data-doi=\"10.1890/07-1107.1\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1890/07-1107.1\" href=\"https://doi.org/10.1890%2F07-1107.1\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 11\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18686577\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 11\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Breeding%20success%20of%20tree%20swallows%20along%20a%20gradient%20of%20agricultural%20intensification&amp;journal=Ecol%20Appl&amp;doi=10.1890%2F07-1107.1&amp;volume=18&amp;pages=1140-1154&amp;publication_year=2008&amp;author=Ghilain%2CA&amp;author=B%C3%A9lisle%2CM\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR12\">Griffith SC, Owens IPF, Burke T . (1999). Environmental determination of a sexually selected trait. <i>Science</i> <b>400</b>: 358–360.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"CAS reference 12\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:CAS:528:DyaK1MXkvFOnsr4%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"Google Scholar reference 12\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Environmental%20determination%20of%20a%20sexually%20selected%20trait&amp;journal=Science&amp;volume=400&amp;pages=358-360&amp;publication_year=1999&amp;author=Griffith%2CSC&amp;author=Owens%2CIPF&amp;author=Burke%2CT\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR13\">Griffith SC, Owens IPF, Thuman KA . (2002). Extra pair paternity in birds: a review of interspecific variation and adaptive function. <i>Mol Ecol</i> <b>11</b>: 2195–2212.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 13\" data-doi=\"10.1046/j.1365-294X.2002.01613.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1046/j.1365-294X.2002.01613.x\" href=\"https://doi.org/10.1046%2Fj.1365-294X.2002.01613.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 13\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:CAS:528:DC%2BD38XptlSmurY%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 13\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=12406233\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 13\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Extra%20pair%20paternity%20in%20birds%3A%20a%20review%20of%20interspecific%20variation%20and%20adaptive%20function&amp;journal=Mol%20Ecol&amp;doi=10.1046%2Fj.1365-294X.2002.01613.x&amp;volume=11&amp;pages=2195-2212&amp;publication_year=2002&amp;author=Griffith%2CSC&amp;author=Owens%2CIPF&amp;author=Thuman%2CKA\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR14\">Hadfield JD . (2010). MCMC methods for multi-response generalized linear mixed models: the MCMCglmm R package. <i>J Stat Softw</i> <b>33</b>: 1–22.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 14\" data-doi=\"10.18637/jss.v033.i02\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.18637/jss.v033.i02\" href=\"https://doi.org/10.18637%2Fjss.v033.i02\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 14\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=MCMC%20methods%20for%20multi-response%20generalized%20linear%20mixed%20models%3A%20the%20MCMCglmm%20R%20package&amp;journal=J%20Stat%20Softw&amp;doi=10.18637%2Fjss.v033.i02&amp;volume=33&amp;pages=1-22&amp;publication_year=2010&amp;author=Hadfield%2CJD\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR15\">Henderson CR . (1984) <i>Applications of Linear Models in Animal Breeding</i>. University of Guelph: Guelph, ON, Canada.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Google Scholar reference 15\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Applications%20of%20Linear%20Models%20in%20Animal%20Breeding&amp;publication_year=1984&amp;author=Henderson%2CCR\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR16\">Hendry AP, Kinnison MT, Heino M, Day T, Smith TB, Fitt G <i>et al</i>. (2011). Evolutionary principles and their practical application. <i>Evol Appl</i> <b>4</b>: 159–183.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 16\" data-doi=\"10.1111/j.1752-4571.2010.00165.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1752-4571.2010.00165.x\" href=\"https://doi.org/10.1111%2Fj.1752-4571.2010.00165.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 16\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=25567966\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 16\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3352551\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 16\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Evolutionary%20principles%20and%20their%20practical%20application&amp;journal=Evol%20Appl&amp;doi=10.1111%2Fj.1752-4571.2010.00165.x&amp;volume=4&amp;pages=159-183&amp;publication_year=2011&amp;author=Hendry%2CAP&amp;author=Kinnison%2CMT&amp;author=Heino%2CM&amp;author=Day%2CT&amp;author=Smith%2CTB&amp;author=Fitt%2CG\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR17\">Hoffmann AA, Merilä J . (1999). Heritable variation and evolution under favourable and unfavourable conditions. <i>Trends Ecol Evol</i> <b>14</b>: 96–101.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 17\" data-doi=\"10.1016/S0169-5347(99)01595-5\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1016/S0169-5347(99)01595-5\" href=\"https://doi.org/10.1016%2FS0169-5347%2899%2901595-5\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 17\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BC2sbgsVGquw%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 17\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=10322508\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 17\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Heritable%20variation%20and%20evolution%20under%20favourable%20and%20unfavourable%20conditions&amp;journal=Trends%20Ecol%20Evol&amp;doi=10.1016%2FS0169-5347%2899%2901595-5&amp;volume=14&amp;pages=96-101&amp;publication_year=1999&amp;author=Hoffmann%2CAA&amp;author=Meril%C3%A4%2CJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR18\">Hosner PA, Winkler DW . (2007). Dispersal distances of tree swallows estimated from continent-wide and limited-area data. <i>J Field Ornithol</i> <b>78</b>: 290–297.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 18\" data-doi=\"10.1111/j.1557-9263.2007.00117.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1557-9263.2007.00117.x\" href=\"https://doi.org/10.1111%2Fj.1557-9263.2007.00117.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 18\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Dispersal%20distances%20of%20tree%20swallows%20estimated%20from%20continent-wide%20and%20limited-area%20data&amp;journal=J%20Field%20Ornithol&amp;doi=10.1111%2Fj.1557-9263.2007.00117.x&amp;volume=78&amp;pages=290-297&amp;publication_year=2007&amp;author=Hosner%2CPA&amp;author=Winkler%2CDW\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR19\">Houle D . (1992). Comparing evolvability and variability of quantitative traits. <i>Genetics</i> <b>130</b>: 195–204.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"CAS reference 19\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DyaK387itVGitQ%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 19\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=1732160\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 19\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1204793\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 19\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Comparing%20evolvability%20and%20variability%20of%20quantitative%20traits&amp;journal=Genetics&amp;volume=130&amp;pages=195-204&amp;publication_year=1992&amp;author=Houle%2CD\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR20\">Houle D, Meyer K . (2015). Estimating sampling error of evolutionary statistics based on genetic covariance matrices using maximum likelihood. <i>J Evol Biol</i> <b>28</b>: 1542–1549.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 20\" data-doi=\"10.1111/jeb.12674\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/jeb.12674\" href=\"https://doi.org/10.1111%2Fjeb.12674\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 20\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BC2MbjvVGguw%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 20\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=26079756\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 20\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimating%20sampling%20error%20of%20evolutionary%20statistics%20based%20on%20genetic%20covariance%20matrices%20using%20maximum%20likelihood&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fjeb.12674&amp;volume=28&amp;pages=1542-1549&amp;publication_year=2015&amp;author=Houle%2CD&amp;author=Meyer%2CK\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR21\">Husby A, Kawakami T, Rönnegård L, Smeds L, Ellegren H, Qvarnström A . (2015). Genome-wide association mapping in a wild avian population identifies a link between genetic and phenotypic variation in a life-history trait. <i>Proc R Soc B</i> <b>282</b>: 20150156.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 21\" data-doi=\"10.1098/rspb.2015.0156\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1098/rspb.2015.0156\" href=\"https://doi.org/10.1098%2Frspb.2015.0156\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 21\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=25833857\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 21\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4426624\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 21\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Genome-wide%20association%20mapping%20in%20a%20wild%20avian%20population%20identifies%20a%20link%20between%20genetic%20and%20phenotypic%20variation%20in%20a%20life-history%20trait&amp;journal=Proc%20R%20Soc%20B&amp;doi=10.1098%2Frspb.2015.0156&amp;volume=282&amp;publication_year=2015&amp;author=Husby%2CA&amp;author=Kawakami%2CT&amp;author=R%C3%B6nneg%C3%A5rd%2CL&amp;author=Smeds%2CL&amp;author=Ellegren%2CH&amp;author=Qvarnstr%C3%B6m%2CA\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR22\">Hussell DJT . (1983). Age and plumage color in female tree swallows. <i>J Field Ornithol</i> <b>54</b>: 312–318.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Google Scholar reference 22\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Age%20and%20plumage%20color%20in%20female%20tree%20swallows&amp;journal=J%20Field%20Ornithol&amp;volume=54&amp;pages=312-318&amp;publication_year=1983&amp;author=Hussell%2CDJT\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR23\">Kalinowski ST, Taper ML, Marshall TC . (2007). Revising how the computer program cervus accommodates genotyping error increases success in paternity assignment. <i>Mol Ecol</i> <b>16</b>: 1099–1106.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 23\" data-doi=\"10.1111/j.1365-294X.2007.03089.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1365-294X.2007.03089.x\" href=\"https://doi.org/10.1111%2Fj.1365-294X.2007.03089.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 23\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17305863\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 23\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Revising%20how%20the%20computer%20program%20cervus%20accommodates%20genotyping%20error%20increases%20success%20in%20paternity%20assignment&amp;journal=Mol%20Ecol&amp;doi=10.1111%2Fj.1365-294X.2007.03089.x&amp;volume=16&amp;pages=1099-1106&amp;publication_year=2007&amp;author=Kalinowski%2CST&amp;author=Taper%2CML&amp;author=Marshall%2CTC\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR24\">Kruuk LEB . (2004). Estimating genetic parameters in natural populations using the ‘animal model’. <i>Philos Trans R Soc B</i> <b>359</b>: 873–890.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 24\" data-doi=\"10.1098/rstb.2003.1437\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1098/rstb.2003.1437\" href=\"https://doi.org/10.1098%2Frstb.2003.1437\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 24\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimating%20genetic%20parameters%20in%20natural%20populations%20using%20the%20%E2%80%98animal%20model%E2%80%99&amp;journal=Philos%20Trans%20R%20Soc%20B&amp;doi=10.1098%2Frstb.2003.1437&amp;volume=359&amp;pages=873-890&amp;publication_year=2004&amp;author=Kruuk%2CLEB\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR25\">Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. <i>J Evol Biol</i> <b>20</b>: 1890–1903.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 25\" data-doi=\"10.1111/j.1420-9101.2007.01377.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1420-9101.2007.01377.x\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2007.01377.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 25\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD2svpsFGmuw%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 25\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17714306\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 25\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=How%20to%20separate%20genetic%20and%20environmental%20causes%20of%20similarity%20between%20relatives&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2007.01377.x&amp;volume=20&amp;pages=1890-1903&amp;publication_year=2007&amp;author=Kruuk%2CLEB&amp;author=Hadfield%2CJD\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR26\">Kruuk LEB, Slate J, Wilson AJ . (2008). New answers for old questions: the evolutionary quantitative genetics of wild animal populations. <i>Annu Rev Ecol Evol Syst</i> <b>39</b>: 525–548.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 26\" data-doi=\"10.1146/annurev.ecolsys.39.110707.173542\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1146/annurev.ecolsys.39.110707.173542\" href=\"https://doi.org/10.1146%2Fannurev.ecolsys.39.110707.173542\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 26\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=New%20answers%20for%20old%20questions%3A%20the%20evolutionary%20quantitative%20genetics%20of%20wild%20animal%20populations&amp;journal=Annu%20Rev%20Ecol%20Evol%20Syst&amp;doi=10.1146%2Fannurev.ecolsys.39.110707.173542&amp;volume=39&amp;pages=525-548&amp;publication_year=2008&amp;author=Kruuk%2CLEB&amp;author=Slate%2CJ&amp;author=Wilson%2CAJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR27\">Krzywinski M, Altman N . (2013). Points of significance: error bars. <i>Nat Methods</i> <b>10</b>: 921–922.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 27\" data-doi=\"10.1038/nmeth.2659\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1038/nmeth.2659\" href=\"https://doi.org/10.1038%2Fnmeth.2659\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 27\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:CAS:528:DC%2BC3sXhsFaksrzO\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 27\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=24161969\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 27\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Points%20of%20significance%3A%20error%20bars&amp;journal=Nat%20Methods&amp;doi=10.1038%2Fnmeth.2659&amp;volume=10&amp;pages=921-922&amp;publication_year=2013&amp;author=Krzywinski%2CM&amp;author=Altman%2CN\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR28\">Lande R . (1981). The minimum number of genes contributing to quantitative variation between and within populations. <i>Genetics</i> <b>99</b>: 541–553.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"CAS reference 28\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DyaL383htFOmsQ%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 28\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=7343418\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 28\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1214520\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 28\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20minimum%20number%20of%20genes%20contributing%20to%20quantitative%20variation%20between%20and%20within%20populations&amp;journal=Genetics&amp;volume=99&amp;pages=541-553&amp;publication_year=1981&amp;author=Lande%2CR\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR29\">Lande R, Arnold SJ . (1983). The measurement of selection on correlated characters. <i>Evolution</i> <b>37</b>: 1210–1226.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 29\" data-doi=\"10.1111/j.1558-5646.1983.tb00236.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1558-5646.1983.tb00236.x\" href=\"https://doi.org/10.1111%2Fj.1558-5646.1983.tb00236.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 29\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=28556011\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 29\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20measurement%20of%20selection%20on%20correlated%20characters&amp;journal=Evolution&amp;doi=10.1111%2Fj.1558-5646.1983.tb00236.x&amp;volume=37&amp;pages=1210-1226&amp;publication_year=1983&amp;author=Lande%2CR&amp;author=Arnold%2CSJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR30\">Lee C, Pollak EJ . (1997). Influence of sire misidentification on sire x year interaction variance and direct-maternal genetic covariance for weaning weight in beef cattle. <i>J Anim Sci</i> <b>75</b>: 2858–2863.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 30\" data-doi=\"10.2527/1997.75112858x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.2527/1997.75112858x\" href=\"https://doi.org/10.2527%2F1997.75112858x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 30\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:CAS:528:DyaK2sXntFemurw%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 30\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=9374297\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 30\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Influence%20of%20sire%20misidentification%20on%20sire%20x%20year%20interaction%20variance%20and%20direct-maternal%20genetic%20covariance%20for%20weaning%20weight%20in%20beef%20cattle&amp;journal=J%20Anim%20Sci&amp;doi=10.2527%2F1997.75112858x&amp;volume=75&amp;pages=2858-2863&amp;publication_year=1997&amp;author=Lee%2CC&amp;author=Pollak%2CEJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR31\">Lemons PR, Marshall TC, McCloskey SE, Sethi SA, Schmutz JA, Sedinger JS . (2015). A likelihood-based approach for assessment of extra-pair paternity and conspecific brood parasitism in natural populations. <i>Mol Ecol Resour</i> <b>7</b>: 107–116.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 31\" data-doi=\"10.1111/1755-0998.12287\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/1755-0998.12287\" href=\"https://doi.org/10.1111%2F1755-0998.12287\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 31\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=A%20likelihood-based%20approach%20for%20assessment%20of%20extra-pair%20paternity%20and%20conspecific%20brood%20parasitism%20in%20natural%20populations&amp;journal=Mol%20Ecol%20Resour&amp;doi=10.1111%2F1755-0998.12287&amp;volume=7&amp;pages=107-116&amp;publication_year=2015&amp;author=Lemons%2CPR&amp;author=Marshall%2CTC&amp;author=McCloskey%2CSE&amp;author=Sethi%2CSA&amp;author=Schmutz%2CJA&amp;author=Sedinger%2CJS\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR32\">Lessard A, Bourret A, Bélisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (<i>Tachycineta bicolor</i>. <i>Behav Ecol Sociobiol</i> <b>68</b>: 733–742.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 32\" data-doi=\"10.1007/s00265-014-1686-y\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1007/s00265-014-1686-y\" href=\"https://doi.org/10.1007%2Fs00265-014-1686-y\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 32\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Individual%20and%20environmental%20determinants%20of%20reproductive%20success%20in%20male%20tree%20swallow%20%28Tachycineta%20bicolor&amp;journal=Behav%20Ecol%20Sociobiol&amp;doi=10.1007%2Fs00265-014-1686-y&amp;volume=68&amp;pages=733-742&amp;publication_year=2014&amp;author=Lessard%2CA&amp;author=Bourret%2CA&amp;author=B%C3%A9lisle%2CM&amp;author=Pelletier%2CF&amp;author=Garant%2CD\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR33\">Lynch M, Walsh B . (1998) <i>Genetics and Analysis of Quantitative Traits</i>. Sinauer Associates: Sunderland, MA, USA.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Google Scholar reference 33\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Genetics%20and%20Analysis%20of%20Quantitative%20Traits&amp;publication_year=1998&amp;author=Lynch%2CM&amp;author=Walsh%2CB\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR34\">Meyer K . (2008). Likelihood calculations to evaluate experimental designs to estimate genetic variances. <i>Heredity</i> <b>101</b>: 212–221.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 34\" data-doi=\"10.1038/hdy.2008.46\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1038/hdy.2008.46\" href=\"https://doi.org/10.1038%2Fhdy.2008.46\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 34\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD1crjtVSmug%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 34\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18523444\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 34\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Likelihood%20calculations%20to%20evaluate%20experimental%20designs%20to%20estimate%20genetic%20variances&amp;journal=Heredity&amp;doi=10.1038%2Fhdy.2008.46&amp;volume=101&amp;pages=212-221&amp;publication_year=2008&amp;author=Meyer%2CK\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR35\">Morrissey MB, Kruuk LEB, Wilson AJ . (2010). The danger of applying the breeder’s equation in observational studies of natural populations. <i>J Evol Biol</i> <b>23</b>: 2277–2288.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 35\" data-doi=\"10.1111/j.1420-9101.2010.02084.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1420-9101.2010.02084.x\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2010.02084.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 35\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BC3cbhtFCksQ%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 35\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=20831731\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 35\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20danger%20of%20applying%20the%20breeder%E2%80%99s%20equation%20in%20observational%20studies%20of%20natural%20populations&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2010.02084.x&amp;volume=23&amp;pages=2277-2288&amp;publication_year=2010&amp;author=Morrissey%2CMB&amp;author=Kruuk%2CLEB&amp;author=Wilson%2CAJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR36\">Morrissey MB, de Villemereuil P, Doligez B, Gimenez O . (2014). Bayesian approaches to the quantitative genetic analysis of natural populations. In: Charmantier A, Garant D <i>et al.</i> (eds) <i>Quantitative Genetics in the Wild</i>. Oxford University Press, Oxford, UK, pp 228–253.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Chapter reference 36\" data-doi=\"10.1093/acprof:oso/9780199674237.003.0014\" data-track=\"click\" data-track-action=\"chapter reference\" data-track-label=\"10.1093/acprof:oso/9780199674237.003.0014\" href=\"https://doi.org/10.1093%2Facprof%3Aoso%2F9780199674237.003.0014\" rel=\"nofollow noopener\">Chapter</a> \n",
      "    <a aria-label=\"Google Scholar reference 36\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Bayesian%20approaches%20to%20the%20quantitative%20genetic%20analysis%20of%20natural%20populations&amp;doi=10.1093%2Facprof%3Aoso%2F9780199674237.003.0014&amp;pages=228-253&amp;publication_year=2014&amp;author=Morrissey%2CMichael%20B.&amp;author=de%20Villemereuil%2CPierre&amp;author=Doligez%2CBlandine&amp;author=Gimenez%2COlivier\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR37\">Morrissey MB, Wilson AJ . (2010). PEDANTICS: an R package for pedigree-based genetic simulation and pedigree manipulation, characterization and viewing. <i>Mol Ecol Resour</i> <b>10</b>: 711–719.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 37\" data-doi=\"10.1111/j.1755-0998.2009.02817.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1755-0998.2009.02817.x\" href=\"https://doi.org/10.1111%2Fj.1755-0998.2009.02817.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 37\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=21565076\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 37\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=PEDANTICS%3A%20an%20R%20package%20for%20pedigree-based%20genetic%20simulation%20and%20pedigree%20manipulation%2C%20characterization%20and%20viewing&amp;journal=Mol%20Ecol%20Resour&amp;doi=10.1111%2Fj.1755-0998.2009.02817.x&amp;volume=10&amp;pages=711-719&amp;publication_year=2010&amp;author=Morrissey%2CMB&amp;author=Wilson%2CAJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR38\">Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (<i>Ovis aries</i>. <i>J Evol Biol</i> <b>20</b>: 2309–2321.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 38\" data-doi=\"10.1111/j.1420-9101.2007.01412.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1420-9101.2007.01412.x\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2007.01412.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 38\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD2snjt1eksQ%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 38\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17956393\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 38\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=A%20framework%20for%20power%20and%20sensitivity%20analyses%20for%20quantitative%20genetic%20studies%20of%20natural%20populations%2C%20and%20case%20studies%20in%20Soay%20sheep%20%28Ovis%20aries&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2007.01412.x&amp;volume=20&amp;pages=2309-2321&amp;publication_year=2007&amp;author=Morrissey%2CMB&amp;author=Wilson%2CAJ&amp;author=Pemberton%2CJM&amp;author=Ferguson%2CMM\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR39\">Postma E . (2014). Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates? In: Charmantier A, Garant D <i>et al.</i> (eds) <i>Quantitative Genetics in the Wild.</i> Oxford University Press, Oxford, UK, pp 16–33.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Chapter reference 39\" data-doi=\"10.1093/acprof:oso/9780199674237.003.0002\" data-track=\"click\" data-track-action=\"chapter reference\" data-track-label=\"10.1093/acprof:oso/9780199674237.003.0002\" href=\"https://doi.org/10.1093%2Facprof%3Aoso%2F9780199674237.003.0002\" rel=\"nofollow noopener\">Chapter</a> \n",
      "    <a aria-label=\"Google Scholar reference 39\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Four%20decades%20of%20estimating%20heritabilities%20in%20wild%20vertebrate%20populations%3A%20improved%20methods%2C%20more%20data%2C%20better%20estimates%3F&amp;doi=10.1093%2Facprof%3Aoso%2F9780199674237.003.0002&amp;pages=16-33&amp;publication_year=2014&amp;author=Postma%2CErik\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR40\">Quinn JL, Charmantier A, Garant D, Sheldon BC . (2006). Data depth, data completeness, and their influence on quantitative genetic estimation in two contrasting bird populations. <i>J Evol Biol</i> <b>19</b>: 994–1002.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 40\" data-doi=\"10.1111/j.1420-9101.2006.01081.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1420-9101.2006.01081.x\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2006.01081.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 40\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD283ltV2rtw%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 40\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=16674594\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 40\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Data%20depth%2C%20data%20completeness%2C%20and%20their%20influence%20on%20quantitative%20genetic%20estimation%20in%20two%20contrasting%20bird%20populations&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2006.01081.x&amp;volume=19&amp;pages=994-1002&amp;publication_year=2006&amp;author=Quinn%2CJL&amp;author=Charmantier%2CA&amp;author=Garant%2CD&amp;author=Sheldon%2CBC\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR41\">Core Team R . (2015). R: a language and environment for statistical computing <a href=\"http://www.R-project.org/\">http://www.R-project.org/</a>.</p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR42\">Via S, Lande R . (1985). Genotype-environment interaction and the evolution of phenotypic plasticity. <i>Evolution</i> <b>39</b>: 505–522.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 42\" data-doi=\"10.1111/j.1558-5646.1985.tb00391.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1558-5646.1985.tb00391.x\" href=\"https://doi.org/10.1111%2Fj.1558-5646.1985.tb00391.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 42\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=28561964\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 42\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Genotype-environment%20interaction%20and%20the%20evolution%20of%20phenotypic%20plasticity&amp;journal=Evolution&amp;doi=10.1111%2Fj.1558-5646.1985.tb00391.x&amp;volume=39&amp;pages=505-522&amp;publication_year=1985&amp;author=Via%2CS&amp;author=Lande%2CR\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR43\">de Villemereuil P, Gimenez O, Doligez B . (2013). Comparing parent–offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits. <i>Methods Ecol Evol</i> <b>4</b>: 260–275.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 43\" data-doi=\"10.1111/2041-210X.12011\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/2041-210X.12011\" href=\"https://doi.org/10.1111%2F2041-210X.12011\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 43\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Comparing%20parent%E2%80%93offspring%20regression%20with%20frequentist%20and%20Bayesian%20animal%20models%20to%20estimate%20heritability%20in%20wild%20populations%3A%20A%20simulation%20study%20for%20Gaussian%20and%20binary%20traits&amp;journal=Methods%20Ecol%20Evol&amp;doi=10.1111%2F2041-210X.12011&amp;volume=4&amp;pages=260-275&amp;publication_year=2013&amp;author=de%20Villemereuil%2CP&amp;author=Gimenez%2CO&amp;author=Doligez%2CB\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR44\">Vinkhuyzen AAE, Wray NR, Yang J, Goddard ME, Visscher PM . (2013). Estimation and partition of heritability in human populations using whole-genome analysis methods. <i>Annu Rev Genet</i> <b>47</b>: 75–95.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 44\" data-doi=\"10.1146/annurev-genet-111212-133258\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1146/annurev-genet-111212-133258\" href=\"https://doi.org/10.1146%2Fannurev-genet-111212-133258\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 44\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:CAS:528:DC%2BC2cXhsFOjtw%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 44\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=23988118\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 44\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4037293\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 44\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimation%20and%20partition%20of%20heritability%20in%20human%20populations%20using%20whole-genome%20analysis%20methods&amp;journal=Annu%20Rev%20Genet&amp;doi=10.1146%2Fannurev-genet-111212-133258&amp;volume=47&amp;pages=75-95&amp;publication_year=2013&amp;author=Vinkhuyzen%2CAAE&amp;author=Wray%2CNR&amp;author=Yang%2CJ&amp;author=Goddard%2CME&amp;author=Visscher%2CPM\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR45\">Walther BA, Moore JL . (2005). The concepts of bias, precision and accuracy, and their use in testing the performance of species richness estimators, with a literature review of estimator performance. <i>Ecography</i> <b>28</b>: 815–829.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 45\" data-doi=\"10.1111/j.2005.0906-7590.04112.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.2005.0906-7590.04112.x\" href=\"https://doi.org/10.1111%2Fj.2005.0906-7590.04112.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 45\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20concepts%20of%20bias%2C%20precision%20and%20accuracy%2C%20and%20their%20use%20in%20testing%20the%20performance%20of%20species%20richness%20estimators%2C%20with%20a%20literature%20review%20of%20estimator%20performance&amp;journal=Ecography&amp;doi=10.1111%2Fj.2005.0906-7590.04112.x&amp;volume=28&amp;pages=815-829&amp;publication_year=2005&amp;author=Walther%2CBA&amp;author=Moore%2CJL\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR46\">Wilson AJ . (2008). Why h<sup>2</sup> does not always equal VA/VP? <i>J Evol Biol</i> <b>21</b>: 647–650.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 46\" data-doi=\"10.1111/j.1420-9101.2008.01500.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1420-9101.2008.01500.x\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2008.01500.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 46\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD1c3kvVKmtg%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 46\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18266683\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 46\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Why%20h2%20does%20not%20always%20equal%20VA%2FVP%3F&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2008.01500.x&amp;volume=21&amp;pages=647-650&amp;publication_year=2008&amp;author=Wilson%2CAJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR47\">Wilson AJ, Morrissey MB, Adams MJ, Walling CA, Guinness FE, Pemberton JM <i>et al</i>. (2011). Indirect genetics effects and evolutionary constraint: an analysis of social dominance in red deer, <i>Cervus elaphus</i>. <i>J Evol Biol</i> <b>24</b>: 772–783.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 47\" data-doi=\"10.1111/j.1420-9101.2010.02212.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1420-9101.2010.02212.x\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2010.02212.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 47\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BC3M3msVCqtQ%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 47\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=21288272\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 47\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Indirect%20genetics%20effects%20and%20evolutionary%20constraint%3A%20an%20analysis%20of%20social%20dominance%20in%20red%20deer%2C%20Cervus%20elaphus&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2010.02212.x&amp;volume=24&amp;pages=772-783&amp;publication_year=2011&amp;author=Wilson%2CAJ&amp;author=Morrissey%2CMB&amp;author=Adams%2CMJ&amp;author=Walling%2CCA&amp;author=Guinness%2CFE&amp;author=Pemberton%2CJM\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR48\">Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA <i>et al</i>. (2010). An ecologist’s guide to the animal model. <i>J Anim Ecol</i> <b>79</b>: 13–26.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 48\" data-doi=\"10.1111/j.1365-2656.2009.01639.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1365-2656.2009.01639.x\" href=\"https://doi.org/10.1111%2Fj.1365-2656.2009.01639.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 48\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=20409158\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 48\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=An%20ecologist%E2%80%99s%20guide%20to%20the%20animal%20model&amp;journal=J%20Anim%20Ecol&amp;doi=10.1111%2Fj.1365-2656.2009.01639.x&amp;volume=79&amp;pages=13-26&amp;publication_year=2010&amp;author=Wilson%2CAJ&amp;author=R%C3%A9ale%2CD&amp;author=Clements%2CMN&amp;author=Morrissey%2CMM&amp;author=Postma%2CE&amp;author=Walling%2CCA\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR49\">Winkler D, Hallinger K, Ardia D, Robertson R, Stutchbury B, Cohen R . (2011). Tree swallow (<i>Tachycineta bicolor</i>. Birds North Am Online. <a href=\"http://bna.birds.cornell.edu/bna/\">http://bna.birds.cornelledu/bna/</a>. Accessed on 1 Oct 2013.</p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR50\">Wolak ME . (2012). nadiv : an R package to create relatedness matrices for estimating non-additive genetic variances in animal models. <i>Methods Ecol Evol</i> <b>3</b>: 792–796.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 50\" data-doi=\"10.1111/j.2041-210X.2012.00213.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.2041-210X.2012.00213.x\" href=\"https://doi.org/10.1111%2Fj.2041-210X.2012.00213.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 50\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=nadiv%20%3A%20an%20R%20package%20to%20create%20relatedness%20matrices%20for%20estimating%20non-additive%20genetic%20variances%20in%20animal%20models&amp;journal=Methods%20Ecol%20Evol&amp;doi=10.1111%2Fj.2041-210X.2012.00213.x&amp;volume=3&amp;pages=792-796&amp;publication_year=2012&amp;author=Wolak%2CME\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR51\">Wolak ME, Roff DA, Fairbairn DJ . (2015). Are we underestimating the genetic variances of dimorphic traits? <i>Ecol Evol</i> <b>5</b>: 590–597.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 51\" data-doi=\"10.1002/ece3.1361\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1002/ece3.1361\" href=\"https://doi.org/10.1002%2Fece3.1361\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 51\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=25691983\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 51\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4328764\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 51\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Are%20we%20underestimating%20the%20genetic%20variances%20of%20dimorphic%20traits%3F&amp;journal=Ecol%20Evol&amp;doi=10.1002%2Fece3.1361&amp;volume=5&amp;pages=590-597&amp;publication_year=2015&amp;author=Wolak%2CME&amp;author=Roff%2CDA&amp;author=Fairbairn%2CDJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li></ul><p class=\"c-article-references__download u-hide-print\"><a data-track=\"click\" data-track-action=\"download citation references\" data-track-label=\"link\" href=\"https://citation-needed.springer.com/v2/references/10.1038/hdy.2016.92?format=refman&amp;flavour=references\">Download references<svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-download\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></p></div></div>\n",
      "<div class=\"c-article-section__content\" id=\"Ack1-content\"><p>We thank the 40 farm owners for providing access to their lands and all graduate students and field and laboratory assistants who have contributed in gathering data in our system over the years. We also thank Fanie Pelletier and Marc Bélisle for their long-term involvement in the southern Québec Tree swallow study. We also thank them as well as Denis Réale, Matthew Wolak, Michael Morrissey and two anonymous reviewers for constructive comments on a previous version of this manuscript. Animals were captured and handled in compliance with the Canadian Council on Animal Care, under the approval of the Université de Sherbrooke Animal Ethics Committee (Protocol Number: DG2010-01 and DG2014-01-Université de Sherbrooke). This work was supported by the Natural Sciences and Engineering Research Council of Canada (NSERC) discovery grants to DG, Fanie Pelletier and Marc Bélisle and by the Canada Research Chairs program to Fanie Pelletier and Marc Bélisle. AB was supported by a postgraduate NSERC scholarship.</p></div>\n",
      "<div class=\"c-article-section__content\" id=\"author-information-content\"><h3 class=\"c-article__sub-heading\" id=\"affiliations\">Authors and Affiliations</h3><ol class=\"c-article-author-affiliation__list\"><li id=\"Aff1\"><p class=\"c-article-author-affiliation__address\">Département de Biologie, Université de Sherbrooke, Sherbrooke, Quebec, Canada</p><p class=\"c-article-author-affiliation__authors-list\">A Bourret &amp; D Garant</p></li></ol><div class=\"u-js-hide u-hide-print\" data-test=\"author-info\"><span class=\"c-article__sub-heading\">Authors</span><ol class=\"c-article-authors-search u-list-reset\"><li id=\"auth-A-Bourret\"><span class=\"c-article-authors-search__title u-h3 js-search-name\">A Bourret</span><div class=\"c-article-authors-search__list\"><div class=\"c-article-authors-search__item c-article-authors-search__list-item--left\"><a class=\"c-article-button\" data-track=\"click\" data-track-action=\"author link - publication\" data-track-label=\"link\" href=\"/search?author=A%20Bourret\" rel=\"nofollow\">View author publications</a></div><div class=\"c-article-authors-search__item c-article-authors-search__list-item--right\"><p class=\"search-in-title-js c-article-authors-search__text\">You can also search for this author in\n",
      "                        <span class=\"c-article-identifiers\"><a class=\"c-article-identifiers__item\" data-track=\"click\" data-track-action=\"author link - pubmed\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&amp;term=A%20Bourret\" rel=\"nofollow\">PubMed</a><span class=\"u-hide\"> </span><a class=\"c-article-identifiers__item\" data-track=\"click\" data-track-action=\"author link - scholar\" data-track-label=\"link\" href=\"http://scholar.google.co.uk/scholar?as_q=&amp;num=10&amp;btnG=Search+Scholar&amp;as_epq=&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=%22A%20Bourret%22&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;as_allsubj=all&amp;hl=en\" rel=\"nofollow\">Google Scholar</a></span></p></div></div></li><li id=\"auth-D-Garant\"><span class=\"c-article-authors-search__title u-h3 js-search-name\">D Garant</span><div class=\"c-article-authors-search__list\"><div class=\"c-article-authors-search__item c-article-authors-search__list-item--left\"><a class=\"c-article-button\" data-track=\"click\" data-track-action=\"author link - publication\" data-track-label=\"link\" href=\"/search?author=D%20Garant\" rel=\"nofollow\">View author publications</a></div><div class=\"c-article-authors-search__item c-article-authors-search__list-item--right\"><p class=\"search-in-title-js c-article-authors-search__text\">You can also search for this author in\n",
      "                        <span class=\"c-article-identifiers\"><a class=\"c-article-identifiers__item\" data-track=\"click\" data-track-action=\"author link - pubmed\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&amp;term=D%20Garant\" rel=\"nofollow\">PubMed</a><span class=\"u-hide\"> </span><a class=\"c-article-identifiers__item\" data-track=\"click\" data-track-action=\"author link - scholar\" data-track-label=\"link\" href=\"http://scholar.google.co.uk/scholar?as_q=&amp;num=10&amp;btnG=Search+Scholar&amp;as_epq=&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=%22D%20Garant%22&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;as_allsubj=all&amp;hl=en\" rel=\"nofollow\">Google Scholar</a></span></p></div></div></li></ol></div><h3 class=\"c-article__sub-heading\" id=\"corresponding-author\">Corresponding author</h3><p id=\"corresponding-author-list\">Correspondence to\n",
      "                <a href=\"mailto:Audrey.Bourret@USherbrooke.ca\" id=\"corresp-c1\">A Bourret</a>.</p></div>\n",
      "<div class=\"c-article-section__content\" id=\"ethics-content\">\n",
      "<h3 class=\"c-article__sub-heading\">Competing interests</h3>\n",
      "<p>The authors declare no conflict of interest.</p>\n",
      "</div>\n",
      "<div class=\"c-article-section__content\" id=\"additional-information-content\"><p><a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Information</a> accompanies this paper on Heredity website</p></div>\n",
      "<div class=\"c-article-section__content\" id=\"Sec17-content\"><div data-test=\"supplementary-info\"><div class=\"c-article-figshare-container\" data-test=\"figshare-container\" id=\"figshareContainer\"></div><div class=\"c-article-supplementary__item\" data-test=\"supp-item\" id=\"MOESM34\"><h3 class=\"c-article-supplementary__title u-h3\"><a class=\"print-link\" data-supp-info-image=\"\" data-test=\"supp-info-link\" data-track=\"click\" data-track-action=\"view supplementary info\" data-track-label=\"link\" href=\"https://static-content.springer.com/esm/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_BFhdy201692_MOESM34_ESM.doc\">Supplementary Information (DOC 711 kb)</a></h3></div></div></div>\n",
      "<div class=\"c-article-section__content\" id=\"rightslink-content\"><p class=\"c-article-rights\"><a data-track=\"click\" data-track-action=\"view rights and permissions\" data-track-label=\"link\" href=\"https://s100.copyright.com/AppDispatchServlet?title=An%20assessment%20of%20the%20reliability%20of%20quantitative%20genetics%20estimates%20in%20study%20systems%20with%20high%20rate%20of%20extra-pair%20reproduction%20and%20low%20recruitment&amp;author=A%20Bourret%20et%20al&amp;contentID=10.1038%2Fhdy.2016.92&amp;copyright=The%20Genetics%20Society%20Macmillan%20Publishers%20Limited%2C%20part%20of%20Springer%20Nature.&amp;publication=0018-067X&amp;publicationDate=2016-10-26&amp;publisherName=SpringerNature&amp;orderBeanReset=true\">Reprints and Permissions</a></p></div>\n",
      "<div class=\"c-article-section__content\" id=\"article-info-content\"><div class=\"c-bibliographic-information\"><div class=\"u-hide-print c-bibliographic-information__column c-bibliographic-information__column--border\"><a data-crossmark=\"10.1038/hdy.2016.92\" data-test=\"crossmark\" data-track=\"click\" data-track-action=\"Click Crossmark\" data-track-label=\"link\" href=\"https://crossmark.crossref.org/dialog/?doi=10.1038/hdy.2016.92\" rel=\"noopener\" target=\"_blank\"><img alt=\"Verify currency and authenticity via CrossMark\" height=\"81\" src=\"data:image/svg+xml;base64,<svg height="81" width="57" xmlns="http://www.w3.org/2000/svg"><g fill="none" fill-rule="evenodd"><path d="m17.35 35.45 21.3-14.2v-17.03h-21.3" fill="#989898"/><path d="m38.65 35.45-21.3-14.2v-17.03h21.3" fill="#747474"/><path d="m28 .5c-12.98 0-23.5 10.52-23.5 23.5s10.52 23.5 23.5 23.5 23.5-10.52 23.5-23.5c0-6.23-2.48-12.21-6.88-16.62-4.41-4.4-10.39-6.88-16.62-6.88zm0 41.25c-9.8 0-17.75-7.95-17.75-17.75s7.95-17.75 17.75-17.75 17.75 7.95 17.75 17.75c0 4.71-1.87 9.22-5.2 12.55s-7.84 5.2-12.55 5.2z" fill="#535353"/><path d="m41 36c-5.81 6.23-15.23 7.45-22.43 2.9-7.21-4.55-10.16-13.57-7.03-21.5l-4.92-3.11c-4.95 10.7-1.19 23.42 8.78 29.71 9.97 6.3 23.07 4.22 30.6-4.86z" fill="#9c9c9c"/><path d="m.2 58.45c0-.75.11-1.42.33-2.01s.52-1.09.91-1.5c.38-.41.83-.73 1.34-.94.51-.22 1.06-.32 1.65-.32.56 0 1.06.11 1.51.35.44.23.81.5 1.1.81l-.91 1.01c-.24-.24-.49-.42-.75-.56-.27-.13-.58-.2-.93-.2-.39 0-.73.08-1.05.23-.31.16-.58.37-.81.66-.23.28-.41.63-.53 1.04-.13.41-.19.88-.19 1.39 0 1.04.23 1.86.68 2.46.45.59 1.06.88 1.84.88.41 0 .77-.07 1.07-.23s.59-.39.85-.68l.91 1c-.38.43-.8.76-1.28.99-.47.22-1 .34-1.58.34-.59 0-1.13-.1-1.64-.31-.5-.2-.94-.51-1.31-.91-.38-.4-.67-.9-.88-1.48-.22-.59-.33-1.26-.33-2.02zm8.4-5.33h1.61v2.54l-.05 1.33c.29-.27.61-.51.96-.72s.76-.31 1.24-.31c.73 0 1.27.23 1.61.71.33.47.5 1.14.5 2.02v4.31h-1.61v-4.1c0-.57-.08-.97-.25-1.21-.17-.23-.45-.35-.83-.35-.3 0-.56.08-.79.22-.23.15-.49.36-.78.64v4.8h-1.61zm7.37 6.45c0-.56.09-1.06.26-1.51.18-.45.42-.83.71-1.14.29-.3.63-.54 1.01-.71.39-.17.78-.25 1.18-.25.47 0 .88.08 1.23.24.36.16.65.38.89.67s.42.63.54 1.03c.12.41.18.84.18 1.32 0 .32-.02.57-.07.76h-4.36c.07.62.29 1.1.65 1.44.36.33.82.5 1.38.5.29 0 .57-.04.83-.13s.51-.21.76-.37l.55 1.01c-.33.21-.69.39-1.09.53-.41.14-.83.21-1.26.21-.48 0-.92-.08-1.34-.25-.41-.16-.76-.4-1.07-.7-.31-.31-.55-.69-.72-1.13-.18-.44-.26-.95-.26-1.52zm4.6-.62c0-.55-.11-.98-.34-1.28-.23-.31-.58-.47-1.06-.47-.41 0-.77.15-1.07.45-.31.29-.5.73-.58 1.3zm2.5.62c0-.57.09-1.08.28-1.53.18-.44.43-.82.75-1.13s.69-.54 1.1-.71c.42-.16.85-.24 1.31-.24.45 0 .84.08 1.17.23s.61.34.85.57l-.77 1.02c-.19-.16-.38-.28-.56-.37-.19-.09-.39-.14-.61-.14-.56 0-1.01.21-1.35.63-.35.41-.52.97-.52 1.67 0 .69.17 1.24.51 1.66.34.41.78.62 1.32.62.28 0 .54-.06.78-.17.24-.12.45-.26.64-.42l.67 1.03c-.33.29-.69.51-1.08.65-.39.15-.78.23-1.18.23-.46 0-.9-.08-1.31-.24-.4-.16-.75-.39-1.05-.7s-.53-.69-.7-1.13c-.17-.45-.25-.96-.25-1.53zm6.91-6.45h1.58v6.17h.05l2.54-3.16h1.77l-2.35 2.8 2.59 4.07h-1.75l-1.77-2.98-1.08 1.23v1.75h-1.58zm13.69 1.27c-.25-.11-.5-.17-.75-.17-.58 0-.87.39-.87 1.16v.75h1.34v1.27h-1.34v5.6h-1.61v-5.6h-.92v-1.2l.92-.07v-.72c0-.35.04-.68.13-.98.08-.31.21-.57.4-.79s.42-.39.71-.51c.28-.12.63-.18 1.04-.18.24 0 .48.02.69.07.22.05.41.1.57.17zm.48 5.18c0-.57.09-1.08.27-1.53.17-.44.41-.82.72-1.13.3-.31.65-.54 1.04-.71.39-.16.8-.24 1.23-.24s.84.08 1.24.24c.4.17.74.4 1.04.71s.54.69.72 1.13c.19.45.28.96.28 1.53s-.09 1.08-.28 1.53c-.18.44-.42.82-.72 1.13s-.64.54-1.04.7-.81.24-1.24.24-.84-.08-1.23-.24-.74-.39-1.04-.7c-.31-.31-.55-.69-.72-1.13-.18-.45-.27-.96-.27-1.53zm1.65 0c0 .69.14 1.24.43 1.66.28.41.68.62 1.18.62.51 0 .9-.21 1.19-.62.29-.42.44-.97.44-1.66 0-.7-.15-1.26-.44-1.67-.29-.42-.68-.63-1.19-.63-.5 0-.9.21-1.18.63-.29.41-.43.97-.43 1.67zm6.48-3.44h1.33l.12 1.21h.05c.24-.44.54-.79.88-1.02.35-.24.7-.36 1.07-.36.32 0 .59.05.78.14l-.28 1.4-.33-.09c-.11-.01-.23-.02-.38-.02-.27 0-.56.1-.86.31s-.55.58-.77 1.1v4.2h-1.61zm-47.87 15h1.61v4.1c0 .57.08.97.25 1.2.17.24.44.35.81.35.3 0 .57-.07.8-.22.22-.15.47-.39.73-.73v-4.7h1.61v6.87h-1.32l-.12-1.01h-.04c-.3.36-.63.64-.98.86-.35.21-.76.32-1.24.32-.73 0-1.27-.24-1.61-.71-.33-.47-.5-1.14-.5-2.02zm9.46 7.43v2.16h-1.61v-9.59h1.33l.12.72h.05c.29-.24.61-.45.97-.63.35-.17.72-.26 1.1-.26.43 0 .81.08 1.15.24.33.17.61.4.84.71.24.31.41.68.53 1.11.13.42.19.91.19 1.44 0 .59-.09 1.11-.25 1.57-.16.47-.38.85-.65 1.16-.27.32-.58.56-.94.73-.35.16-.72.25-1.1.25-.3 0-.6-.07-.9-.2s-.59-.31-.87-.56zm0-2.3c.26.22.5.37.73.45.24.09.46.13.66.13.46 0 .84-.2 1.15-.6.31-.39.46-.98.46-1.77 0-.69-.12-1.22-.35-1.61-.23-.38-.61-.57-1.13-.57-.49 0-.99.26-1.52.77zm5.87-1.69c0-.56.08-1.06.25-1.51.16-.45.37-.83.65-1.14.27-.3.58-.54.93-.71s.71-.25 1.08-.25c.39 0 .73.07 1 .2.27.14.54.32.81.55l-.06-1.1v-2.49h1.61v9.88h-1.33l-.11-.74h-.06c-.25.25-.54.46-.88.64-.33.18-.69.27-1.06.27-.87 0-1.56-.32-2.07-.95s-.76-1.51-.76-2.65zm1.67-.01c0 .74.13 1.31.4 1.7.26.38.65.58 1.15.58.51 0 .99-.26 1.44-.77v-3.21c-.24-.21-.48-.36-.7-.45-.23-.08-.46-.12-.7-.12-.45 0-.82.19-1.13.59-.31.39-.46.95-.46 1.68zm6.35 1.59c0-.73.32-1.3.97-1.71.64-.4 1.67-.68 3.08-.84 0-.17-.02-.34-.07-.51-.05-.16-.12-.3-.22-.43s-.22-.22-.38-.3c-.15-.06-.34-.1-.58-.1-.34 0-.68.07-1 .2s-.63.29-.93.47l-.59-1.08c.39-.24.81-.45 1.28-.63.47-.17.99-.26 1.54-.26.86 0 1.51.25 1.93.76s.63 1.25.63 2.21v4.07h-1.32l-.12-.76h-.05c-.3.27-.63.48-.98.66s-.73.27-1.14.27c-.61 0-1.1-.19-1.48-.56-.38-.36-.57-.85-.57-1.46zm1.57-.12c0 .3.09.53.27.67.19.14.42.21.71.21.28 0 .54-.07.77-.2s.48-.31.73-.56v-1.54c-.47.06-.86.13-1.18.23-.31.09-.57.19-.76.31s-.33.25-.41.4c-.09.15-.13.31-.13.48zm6.29-3.63h-.98v-1.2l1.06-.07.2-1.88h1.34v1.88h1.75v1.27h-1.75v3.28c0 .8.32 1.2.97 1.2.12 0 .24-.01.37-.04.12-.03.24-.07.34-.11l.28 1.19c-.19.06-.4.12-.64.17-.23.05-.49.08-.76.08-.4 0-.74-.06-1.02-.18-.27-.13-.49-.3-.67-.52-.17-.21-.3-.48-.37-.78-.08-.3-.12-.64-.12-1.01zm4.36 2.17c0-.56.09-1.06.27-1.51s.41-.83.71-1.14c.29-.3.63-.54 1.01-.71.39-.17.78-.25 1.18-.25.47 0 .88.08 1.23.24.36.16.65.38.89.67s.42.63.54 1.03c.12.41.18.84.18 1.32 0 .32-.02.57-.07.76h-4.37c.08.62.29 1.1.65 1.44.36.33.82.5 1.38.5.3 0 .58-.04.84-.13.25-.09.51-.21.76-.37l.54 1.01c-.32.21-.69.39-1.09.53s-.82.21-1.26.21c-.47 0-.92-.08-1.33-.25-.41-.16-.77-.4-1.08-.7-.3-.31-.54-.69-.72-1.13-.17-.44-.26-.95-.26-1.52zm4.61-.62c0-.55-.11-.98-.34-1.28-.23-.31-.58-.47-1.06-.47-.41 0-.77.15-1.08.45-.31.29-.5.73-.57 1.3zm3.01 2.23c.31.24.61.43.92.57.3.13.63.2.98.2.38 0 .65-.08.83-.23s.27-.35.27-.6c0-.14-.05-.26-.13-.37-.08-.1-.2-.2-.34-.28-.14-.09-.29-.16-.47-.23l-.53-.22c-.23-.09-.46-.18-.69-.3-.23-.11-.44-.24-.62-.4s-.33-.35-.45-.55c-.12-.21-.18-.46-.18-.75 0-.61.23-1.1.68-1.49.44-.38 1.06-.57 1.83-.57.48 0 .91.08 1.29.25s.71.36.99.57l-.74.98c-.24-.17-.49-.32-.73-.42-.25-.11-.51-.16-.78-.16-.35 0-.6.07-.76.21-.17.15-.25.33-.25.54 0 .14.04.26.12.36s.18.18.31.26c.14.07.29.14.46.21l.54.19c.23.09.47.18.7.29s.44.24.64.4c.19.16.34.35.46.58.11.23.17.5.17.82 0 .3-.06.58-.17.83-.12.26-.29.48-.51.68-.23.19-.51.34-.84.45-.34.11-.72.17-1.15.17-.48 0-.95-.09-1.41-.27-.46-.19-.86-.41-1.2-.68z" fill="#535353"/></g></svg>\" width=\"57\"/></a></div><div class=\"c-bibliographic-information__column\"><h3 class=\"c-article__sub-heading\" id=\"citeas\">Cite this article</h3><p class=\"c-bibliographic-information__citation\">Bourret, A., Garant, D. An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment.\n",
      "                    <i>Heredity</i> <b>118</b>, 229–238 (2017). https://doi.org/10.1038/hdy.2016.92</p><p class=\"c-bibliographic-information__download-citation u-hide-print\"><a data-test=\"citation-link\" data-track=\"click\" data-track-action=\"download article citation\" data-track-external=\"\" data-track-label=\"link\" href=\"https://citation-needed.springer.com/v2/references/10.1038/hdy.2016.92?format=refman&amp;flavour=citation\">Download citation<svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-download\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></p><ul class=\"c-bibliographic-information__list\" data-test=\"publication-history\"><li class=\"c-bibliographic-information__list-item\"><p>Received<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2016-02-24\">24 February 2016</time></span></p></li><li class=\"c-bibliographic-information__list-item\"><p>Revised<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2016-06-26\">26 June 2016</time></span></p></li><li class=\"c-bibliographic-information__list-item\"><p>Accepted<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2016-08-19\">19 August 2016</time></span></p></li><li class=\"c-bibliographic-information__list-item\"><p>Published<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2016-10-26\">26 October 2016</time></span></p></li><li class=\"c-bibliographic-information__list-item\"><p>Issue Date<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2017-03\">March 2017</time></span></p></li><li class=\"c-bibliographic-information__list-item c-bibliographic-information__list-item--doi\"><p><abbr title=\"Digital Object Identifier\">DOI</abbr><span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\">https://doi.org/10.1038/hdy.2016.92</span></p></li></ul><div data-component=\"share-box\"><div class=\"c-article-share-box u-display-none\" hidden=\"\"><h3 class=\"c-article__sub-heading\">Share this article</h3><p class=\"c-article-share-box__description\">Anyone you share the following link with will be able to read this content:</p><button class=\"js-get-share-url c-article-share-box__button\" data-track=\"click\" data-track-action=\"get shareable link\" data-track-external=\"\" data-track-label=\"button\" id=\"get-share-url\">Get shareable link</button><div class=\"js-no-share-url-container u-display-none\" hidden=\"\"><p class=\"js-c-article-share-box__no-sharelink-info c-article-share-box__no-sharelink-info\">Sorry, a shareable link is not currently available for this article.</p></div><div class=\"js-share-url-container u-display-none\" hidden=\"\"><p class=\"js-share-url c-article-share-box__only-read-input\" data-track=\"click\" data-track-action=\"select share url\" data-track-label=\"button\" id=\"share-url\"></p><button class=\"js-copy-share-url c-article-share-box__button--link-like\" data-track=\"click\" data-track-action=\"copy share url\" data-track-external=\"\" data-track-label=\"button\" id=\"copy-share-url\">Copy to clipboard</button></div><p class=\"js-c-article-share-box__additional-info c-article-share-box__additional-info\">\n",
      "                            Provided by the Springer Nature SharedIt content-sharing initiative\n",
      "                        </p></div></div><div data-component=\"article-info-list\"></div></div></div></div>\n",
      "<div class=\"c-article-section__content js-collapsible-section\" id=\"further-reading-content\">\n",
      "<ul class=\"c-article-further-reading__list\" id=\"further-reading-list\">\n",
      "<li class=\"c-article-further-reading__item js-ref-item\">\n",
      "<h3 class=\"c-article-further-reading__title\">\n",
      "<a class=\"print-link\" data-track=\"click\" data-track-action=\"view further reading article\" data-track-label=\"link:Heritability of plumage colour morph variation in a wild population of promiscuous, long-lived Australian magpies\" href=\"https://doi.org/10.1038/s41437-019-0212-4\">\n",
      "                                        Heritability of plumage colour morph variation in a wild population of promiscuous, long-lived Australian magpies\n",
      "                                    </a>\n",
      "</h3>\n",
      "<ul class=\"c-author-list c-author-list--compact u-sans-serif u-mb-4 u-mt-auto\" data-test=\"author-list\">\n",
      "<li>Ana E. Dobson</li><li>Daniel J. Schmidt</li><li>Jane M. Hughes</li>\n",
      "</ul>\n",
      "<p class=\"c-article-further-reading__journal-title\"><i>Heredity</i> (2019)</p>\n",
      "</li>\n",
      "</ul>\n",
      "</div>\n"
     ]
    }
   ],
   "source": [
    "res = soup.find_all(\"div\", class_=\"c-article-body\")\n",
    "for i in res:\n",
    "    res2 = i.find_all(\"div\", class_=\"c-article-section__content\")\n",
    "    for j in res2:\n",
    "        l = j.find(attrs={'id' :\"Abs1-content\"})\n",
    "        print(j)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "id": "7cc9c100",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<div class=\"c-article-section__content js-collapsible-section\" id=\"further-reading-content\">\n",
       "<ul class=\"c-article-further-reading__list\" id=\"further-reading-list\">\n",
       "<li class=\"c-article-further-reading__item js-ref-item\">\n",
       "<h3 class=\"c-article-further-reading__title\">\n",
       "<a class=\"print-link\" data-track=\"click\" data-track-action=\"view further reading article\" data-track-label=\"link:Heritability of plumage colour morph variation in a wild population of promiscuous, long-lived Australian magpies\" href=\"https://doi.org/10.1038/s41437-019-0212-4\">\n",
       "                                        Heritability of plumage colour morph variation in a wild population of promiscuous, long-lived Australian magpies\n",
       "                                    </a>\n",
       "</h3>\n",
       "<ul class=\"c-author-list c-author-list--compact u-sans-serif u-mb-4 u-mt-auto\" data-test=\"author-list\">\n",
       "<li>Ana E. Dobson</li><li>Daniel J. Schmidt</li><li>Jane M. Hughes</li>\n",
       "</ul>\n",
       "<p class=\"c-article-further-reading__journal-title\"><i>Heredity</i> (2019)</p>\n",
       "</li>\n",
       "</ul>\n",
       "</div>"
      ]
     },
     "execution_count": 303,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "j"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e11d65d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "results = soup.find_all(\"div\", class_=\"c-article-section__content\", id_=\"Abs1-content\")\n",
    "for job_element in results:\n",
    "    print(job_element)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "id": "fa28c837",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'NoneType' object has no attribute 'text'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-298-a079614b854a>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0ml\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtext\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m: 'NoneType' object has no attribute 'text'"
     ]
    }
   ],
   "source": [
    "l.text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "id": "0d597243",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<title>Aspicilia bicensis (Megasporaceae), a new sterile, pustulose lichen from eastern Canada</title>\n",
      "Aspicilia bicensis (Megasporaceae), a new sterile, pustulose lichen from eastern Canada\n"
     ]
    }
   ],
   "source": [
    "URL=\"https://doi.org/10.1639/0007-2745-119.1.008\"\n",
    "#page = requests.get(URL)\n",
    "dr.get(URL)\n",
    "soup = BeautifulSoup(dr.page_source, \"html.parser\")\n",
    "results = soup.find_all(\"text\", class_=\"ArticleContentText\")\n",
    "text = \"\"\n",
    "for job_element in results:\n",
    "    text = text + job_element.get_text()\n",
    "title = soup.find('title')\n",
    "\n",
    "print(title) # Prints the tag\n",
    "print(title.string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 323,
   "id": "4b1a54e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Just a moment...\n"
     ]
    }
   ],
   "source": [
    "URL =\"https://doi.org/10.1002/ece3.3947\"\n",
    "page = requests.get(URL)\n",
    "soup = BeautifulSoup(page.content, \"html.parser\")\n",
    "title = soup.find('title')\n",
    "print(title.string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "id": "ea5032fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "headers = {'user-agent':'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36'}\n",
    "JAGS7_result = requests.get(\n",
    "    URL,\n",
    "    headers=headers,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5af3c6a5",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'BeautifulSoup' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-1-1264e469d04f>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0msoup\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mBeautifulSoup\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mJAGS7_result\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcontent\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"html.parser\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[0mtitle\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0msoup\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'title'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: name 'BeautifulSoup' is not defined"
     ]
    }
   ],
   "source": [
    "soup = BeautifulSoup(JAGS7_result.content, \"html.parser\")\n",
    "title = soup.find('title')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 447,
   "id": "75b3180a",
   "metadata": {},
   "outputs": [],
   "source": [
    "from selenium import webdriver\n",
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "import math\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 448,
   "id": "976e5ded",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "https://doi.org/10.5281/zenodo.4548262\n",
      "https://doi.org/10.1111/1755-0998.12412\n",
      "https://doi.org/10.1038/hdy.2016.92\n",
      "https://doi.org/10.1098/rsos.150669\n",
      "https://doi.org/10.1639/0007-2745-119.1.008\n",
      "https://doi.org/10.1111/1365-2745.13123\n",
      "https://doi.org/10.1002/ece3.3947\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-448-3629709c0e42>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      4\u001b[0m         \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0murl\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m         \u001b[0mdr\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mwebdriver\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mChrome\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 6\u001b[1;33m         \u001b[0mdr\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0murl\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      7\u001b[0m         \u001b[0mbs\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mBeautifulSoup\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdr\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpage_source\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m\"html.parser\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      8\u001b[0m         \u001b[0mtitle\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mbs\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'title'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\u001b[0m in \u001b[0;36mget\u001b[1;34m(self, url)\u001b[0m\n\u001b[0;32m    453\u001b[0m         \u001b[0mLoads\u001b[0m \u001b[0ma\u001b[0m \u001b[0mweb\u001b[0m \u001b[0mpage\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mcurrent\u001b[0m \u001b[0mbrowser\u001b[0m \u001b[0msession\u001b[0m\u001b[1;33m.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    454\u001b[0m         \"\"\"\n\u001b[1;32m--> 455\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mexecute\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mCommand\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mGET\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m{\u001b[0m\u001b[1;34m\"url\"\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    456\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    457\u001b[0m     \u001b[1;33m@\u001b[0m\u001b[0mproperty\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\u001b[0m in \u001b[0;36mexecute\u001b[1;34m(self, driver_command, params)\u001b[0m\n\u001b[0;32m    440\u001b[0m                 \u001b[0mparams\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"sessionId\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msession_id\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    441\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 442\u001b[1;33m         \u001b[0mresponse\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcommand_executor\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mexecute\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdriver_command\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    443\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mresponse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    444\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0merror_handler\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcheck_response\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresponse\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\remote_connection.py\u001b[0m in \u001b[0;36mexecute\u001b[1;34m(self, command, params)\u001b[0m\n\u001b[0;32m    292\u001b[0m         \u001b[0mdata\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mutils\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdump_json\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mparams\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    293\u001b[0m         \u001b[0murl\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34mf\"{self._url}{path}\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 294\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_request\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcommand_info\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mbody\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdata\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    295\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    296\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_request\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mbody\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\remote_connection.py\u001b[0m in \u001b[0;36m_request\u001b[1;34m(self, method, url, body)\u001b[0m\n\u001b[0;32m    314\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    315\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mkeep_alive\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 316\u001b[1;33m             \u001b[0mresponse\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_conn\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mbody\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mbody\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mheaders\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    317\u001b[0m             \u001b[0mstatuscode\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mresponse\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstatus\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    318\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\request.py\u001b[0m in \u001b[0;36mrequest\u001b[1;34m(self, method, url, fields, headers, **urlopen_kw)\u001b[0m\n\u001b[0;32m     76\u001b[0m             )\n\u001b[0;32m     77\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 78\u001b[1;33m             return self.request_encode_body(\n\u001b[0m\u001b[0;32m     79\u001b[0m                 \u001b[0mmethod\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfields\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mfields\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mheaders\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0murlopen_kw\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     80\u001b[0m             )\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\request.py\u001b[0m in \u001b[0;36mrequest_encode_body\u001b[1;34m(self, method, url, fields, headers, encode_multipart, multipart_boundary, **urlopen_kw)\u001b[0m\n\u001b[0;32m    168\u001b[0m         \u001b[0mextra_kw\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0murlopen_kw\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    169\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 170\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0murlopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mextra_kw\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\poolmanager.py\u001b[0m in \u001b[0;36murlopen\u001b[1;34m(self, method, url, redirect, **kw)\u001b[0m\n\u001b[0;32m    374\u001b[0m             \u001b[0mresponse\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mconn\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0murlopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkw\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    375\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 376\u001b[1;33m             \u001b[0mresponse\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mconn\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0murlopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mu\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrequest_uri\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkw\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    377\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    378\u001b[0m         \u001b[0mredirect_location\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mredirect\u001b[0m \u001b[1;32mand\u001b[0m \u001b[0mresponse\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_redirect_location\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[0;32m    701\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    702\u001b[0m             \u001b[1;31m# Make the request on the httplib connection object.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 703\u001b[1;33m             httplib_response = self._make_request(\n\u001b[0m\u001b[0;32m    704\u001b[0m                 \u001b[0mconn\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    705\u001b[0m                 \u001b[0mmethod\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[1;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[0;32m    447\u001b[0m                     \u001b[1;31m# Python 3 (including for exceptions like SystemExit).\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    448\u001b[0m                     \u001b[1;31m# Otherwise it looks like a bug in the code.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 449\u001b[1;33m                     \u001b[0msix\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mraise_from\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0me\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    450\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mSocketTimeout\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mBaseSSLError\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mSocketError\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    451\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_raise_timeout\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0merr\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0me\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtimeout_value\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mread_timeout\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\packages\\six.py\u001b[0m in \u001b[0;36mraise_from\u001b[1;34m(value, from_value)\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[1;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[0;32m    442\u001b[0m                 \u001b[1;31m# Python 3\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    443\u001b[0m                 \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 444\u001b[1;33m                     \u001b[0mhttplib_response\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mconn\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgetresponse\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    445\u001b[0m                 \u001b[1;32mexcept\u001b[0m \u001b[0mBaseException\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    446\u001b[0m                     \u001b[1;31m# Remove the TypeError from the exception chain in\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\http\\client.py\u001b[0m in \u001b[0;36mgetresponse\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m   1369\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1370\u001b[0m             \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1371\u001b[1;33m                 \u001b[0mresponse\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbegin\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1372\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0mConnectionError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1373\u001b[0m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\http\\client.py\u001b[0m in \u001b[0;36mbegin\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    317\u001b[0m         \u001b[1;31m# read until we get a non-100 response\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    318\u001b[0m         \u001b[1;32mwhile\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 319\u001b[1;33m             \u001b[0mversion\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mstatus\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mreason\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_read_status\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    320\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mstatus\u001b[0m \u001b[1;33m!=\u001b[0m \u001b[0mCONTINUE\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    321\u001b[0m                 \u001b[1;32mbreak\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\http\\client.py\u001b[0m in \u001b[0;36m_read_status\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    278\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    279\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_read_status\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 280\u001b[1;33m         \u001b[0mline\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreadline\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0m_MAXLINE\u001b[0m \u001b[1;33m+\u001b[0m \u001b[1;36m1\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"iso-8859-1\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    281\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mline\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m>\u001b[0m \u001b[0m_MAXLINE\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    282\u001b[0m             \u001b[1;32mraise\u001b[0m \u001b[0mLineTooLong\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"status line\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\Anaconda3\\lib\\socket.py\u001b[0m in \u001b[0;36mreadinto\u001b[1;34m(self, b)\u001b[0m\n\u001b[0;32m    702\u001b[0m         \u001b[1;32mwhile\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    703\u001b[0m             \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 704\u001b[1;33m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_sock\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mb\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    705\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    706\u001b[0m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_timeout_occurred\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "abst, comp, titles, full_texts = [], [], [], []\n",
    "for url in df_datasets.cited_articles.tolist():\n",
    "    if (str(url)!=\"nan\"):\n",
    "        print(url)\n",
    "        dr = webdriver.Chrome()\n",
    "        dr.get(url)\n",
    "        bs = BeautifulSoup(dr.page_source,\"html.parser\")\n",
    "        title = bs.find('title')\n",
    "        title = title.text\n",
    "        titles.append(title)\n",
    "        full_texts.append(bs.text)\n",
    "        abstract = \"\"\n",
    "        complete = False\n",
    "        section_abstract = [bs.find_all(\"div\", class_=\"article-section__content en main\"),\n",
    "                           bs.find_all(\"div\", class_=\"c-article-section__content\", attrs={'id' :\"Abs1-content\"})]\n",
    "        for res in section_abstract:\n",
    "            if (len(res)>0) :\n",
    "                abstract = res[0].text\n",
    "        abst.append(abstract)\n",
    "        if \"introduction\" in bs.text.lower():\n",
    "            complete = True\n",
    "        comp.append(complete)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 485,
   "id": "cc3e7224",
   "metadata": {},
   "outputs": [],
   "source": [
    "i = full_texts[1].lower().find(\"abstract\")\n",
    "j = full_texts[1].lower().find(\"introduction\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 488,
   "id": "616adb03",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"Abstract\\n\\nCyst nematodes are important agricultural pests responsible for billions of dollars of losses each year. Plant resistance is the most effective management tool, but it requires a close monitoring of population genetics. Current technologies for pathotyping and genotyping cyst nematodes are time-consuming, expensive and imprecise. In this study, we capitalized on the reproduction mode of cyst nematodes to develop a simple population genetic analysis pipeline based on genotyping-by-sequencing and Pool-Seq. This method yielded thousands of SNPs and allowed us to study the relationships between populations of different origins or pathotypes. Validation of the method on well-characterized populations also demonstrated that it was a powerful and accurate tool for population genetics. The genomewide allele frequencies of 23 populations of golden nematode, from nine countries and representing the five known pathotypes, were compared. A clear separation of the pathotypes and fine genetic relationships between and among global populations were obtained using this method. In addition to being powerful, this tool has proven to be very time- and cost-efficient and could be applied to other cyst nematode species.\\n\\nCiting Literature\\n\\n\\nSupporting Information\\n\\n\\n\\n\\nFilename\\nDescription\\n\\n\\n\\nmen12412-sup-0001-FigS1.aiPS document, 87 KB\\nFig. S1 Relationship between the median coverage (median number of reads/locus/population supporting each SNP) and the number of SNPs retained by the two pipelines at a minimum minor allele frequency of 1% (MAF\\xa0=\\xa00.01) and 5% (MAF\\xa0=\\xa00.05).\\n\\n\\nmen12412-sup-0002-FigS2.aiPS document, 113.9 KB\\nFig. S2 Number of SNPs retained by UNEAK and PoPoolation2 pipelines at different parameter values on an ApeKI library from 23 worldwide populations of Globodera\\xa0rostochiensis.\\n\\n\\nmen12412-sup-0003-FigS3.aiPS document, 153.9 KB\\nFig. S3 Principal component analyses of 23 worldwide populations of Globodera\\xa0rostochiensis from different pathotypes (Ro1, Ro2, Ro3, Ro4, Ro5 and a mixed population of Ro2/Ro3).\\n\\n\\nmen12412-sup-0004-FigS4.aiPS document, 1.1 MB\\nFig. S4 Principal component analysis of 23 worldwide populations of Globodera\\xa0rostochiensis.\\n\\n\\nmen12412-sup-0005-TableS1.docxWord document, 17.7 KB\\nTable S1 Population differentiation (FST) of 23 worldwide populations of Globodera\\xa0rostochiensis estimated from genome-wide allele frequencies at 604 loci.\\n\\n\\n\\nPlease note: The publisher is not responsible for the content or functionality of any supporting information supplied by the authors. Any queries (other than missing content) should be directed to the corresponding author for the article.\\n\\n\\n\\n\\nVolume15, Issue6November 2015Pages 1356-1365\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\nRelatedInformation\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\nClose Figure ViewerReturn to FigurePrevious FigureNext FigureCaption\\n\\nDownload PDF\\n\\n\\n\\n\\n\\n\\n\\n\\nback\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n                Additional links\\n            \\n\\n\\nAbout Wiley Online Library\\n\\nPrivacy Policy\\nTerms of Use\\nAbout Cookies\\nManage Cookies\\nAccessibility\\nWiley Research DE&I Statement and Publishing Policies\\nDeveloping World Access\\n\\n\\n\\n\\n\\nHelp & Support\\n\\nContact Us\\nTraining and Support\\nDMCA & Reporting Piracy\\n\\n\\n\\n\\nOpportunities\\n\\nSubscription Agents\\nAdvertisers & Corporate Partners\\n\\n\\n\\n\\nConnect with Wiley\\n\\nThe Wiley Network\\nWiley Press Room\\n\\n\\n\\n\\n\\n\\n\\nCopyright © 1999-2022 John Wiley & Sons, Inc. All rights reserved\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n  The full text of this article hosted at iucr.org is unavailable due to technical difficulties.  \\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\nLog in to Wiley Online Library\\n\\n\\n\\n\\n\\n\\n\\nEmail or Customer ID\\n\\n\\n\\n\\n\\n\\n\\nPassword\\n\\n\\n\\n\\n\\n\\n                                Forgot password?\\n                            \\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n                        \\n                        NEW USER >\\n\\n\\n                            \\n                            INSTITUTIONAL LOGIN >\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\nChange Password\\n\\n\\n\\n\\n\\n\\n\\nOld Password\\n\\n\\n\\n\\n\\n\\nNew Password\\n\\n\\n\\n\\nToo Short\\nWeak\\nMedium\\nStrong\\nVery Strong\\nToo Long\\n\\n\\n\\n\\n\\n\\nPassword Changed Successfully\\nYour password has been changed\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\nCreate a new account\\n\\n\\n\\n\\n\\nEmail\\n\\n\\n\\n\\n\\n\\n\\n\\n                    \\n                    Returning user\\n                \\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\nForgot your password?\\nEnter your email address below. \\n\\n\\n\\n\\n\\n\\n\\nEmail\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\nPlease check your email for instructions on resetting your password.\\n                    If you do not receive an email within 10 minutes, your email address may not be registered,\\n                    and you may need to create a new Wiley Online Library account.\\n\\n\\n\\n\\n\\n\\n\\n\\nRequest Username\\n\\nCan't sign in? Forgot your username?\\nEnter your email address below and we will send you your username\\n\\n\\n\\n\\nEmail\\n\\n\\n\\n\\n\\n\\n\\nClose\\n\\n\\n\\nIf the address matches an existing account you will receive an email with instructions to retrieve your username\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\nClose crossmark popup\""
      ]
     },
     "execution_count": 488,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_texts[1][i:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 491,
   "id": "f7f06342",
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_article(url):\n",
    "    if (str(url)!=\"nan\"):\n",
    "        dr = webdriver.Chrome()\n",
    "        dr.get(url)\n",
    "        bs = BeautifulSoup(dr.page_source,\"html.parser\")\n",
    "        title = bs.find('title')\n",
    "        title = title.text\n",
    "        text = bs.text\n",
    "        section_abstract = [bs.find_all(\"div\", class_=\"article-section__content en main\"),\n",
    "                           bs.find_all(\"div\", class_=\"c-article-section__content\", attrs={'id' :\"Abs1-content\"})]\n",
    "        for res in section_abstract:\n",
    "            if (len(res)>0) :\n",
    "                abstract = res[0].text\n",
    "        if \"introduction\" in bs.text.lower():\n",
    "            if \"material and methods\" in text.lower():\n",
    "                i = text.lower().find(\"material and methods\")\n",
    "            elif \"materials and methods\" in text.lower():\n",
    "                i = text.lower().find(\"materials and methods\")\n",
    "            elif \"methods\" in text.lower():\n",
    "                i = text.lower().find(\"methods\")\n",
    "            else:\n",
    "                i = 0\n",
    "            j = text.lower().find(\"results\")\n",
    "            j = max(0, j)\n",
    "            material = text[i:j]\n",
    "            print(url)\n",
    "            print(material)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 601,
   "id": "1e6068d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_article(url):\n",
    "    title, abstract, material, text = \"\",\"\",\"\",\"\"\n",
    "    if (str(url)!=\"nan\"):\n",
    "        dr = webdriver.Chrome()\n",
    "        dr.get(url)\n",
    "        bs = BeautifulSoup(dr.page_source,\"html.parser\")\n",
    "        title = bs.find('title')\n",
    "        title = title.text\n",
    "        text = bs.text\n",
    "        section_abstract = [bs.find_all(\"div\", class_=\"article-section__content en main\"),\n",
    "                            bs.find_all(\"section\", class_=\"abstract\"),\n",
    "                           bs.find_all(\"div\", class_=\"c-article-section__content\", attrs={'id' :\"Abs1-content\"})]\n",
    "        for res in section_abstract:\n",
    "            if (len(res)>0) :\n",
    "                abstract = res[0].text\n",
    "        section_abstract = bs.find(\"meta\", {\"name\":\"description\"})\n",
    "        if section_abstract is not None:\n",
    "            abstract =  section_abstract[\"content\"]\n",
    "        i = 0\n",
    "        if \"introduction\" in bs.text.lower():\n",
    "            if \"material and methods\" in text.lower():\n",
    "                result = re.finditer(r\"material and methods\", text.lower())\n",
    "            elif \"materials and methods\" in text.lower():\n",
    "                 result = re.finditer(r\"materials and methods\", text.lower())\n",
    "            elif \"methods\" in text.lower():\n",
    "                 result = re.finditer(r\"methods\", text.lower())\n",
    "            else:\n",
    "                 result = []\n",
    "\n",
    "            s = []\n",
    "            for res in result:\n",
    "                s.append(res.start(0))\n",
    "            if len(s)>1:\n",
    "                i = s[1]\n",
    "            elif len(s) == 1 :\n",
    "                i = s[0]\n",
    "\n",
    "            r =  re.finditer(r\"results\\n\", text.lower())\n",
    "            y = []\n",
    "            for res in r:\n",
    "                if res.start(0) > i:\n",
    "                    y.append(res.start(0))\n",
    "            if len(y) >0:\n",
    "                j = min(y)\n",
    "            else:\n",
    "                j= 0\n",
    "            material = text[i:j]\n",
    "    return({\"title\": title, \"abstract\": abstract, \"material\": material, \"text\": text})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 602,
   "id": "dac62413",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nan\n",
      "nan\n",
      "nan\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.1111/1755-0998.12412\n",
      "https://doi.org/10.1038/hdy.2016.92\n",
      "https://doi.org/10.1098/rsos.150669\n",
      "https://doi.org/10.1639/0007-2745-119.1.008\n",
      "https://doi.org/10.1111/1365-2745.13123\n",
      "https://doi.org/10.1002/ece3.3947\n",
      "nan\n",
      "https://doi.org/10.1038/s41598-018-34822-9\n",
      "https://doi.org/10.1002/eap.1713\n",
      "https://doi.org/10.1371/journal.pone.0176706\n",
      "https://doi.org/10.1111/mec.13847\n",
      "https://doi.org/10.1371/journal.pone.0077514\n",
      "https://doi.org/10.1111/mec.14361\n",
      "https://doi.org/10.1371/journal.pone.0107929\n",
      "https://doi.org/10.1600/036364416x692514\n",
      "https://doi.org/10.1002/ece3.1476\n",
      "https://doi.org/10.3389/fpls.2019.00932\n",
      "https://doi.org/10.1186/s12862-016-0810-1\n",
      "https://doi.org/10.1002/ece3.1029\n",
      "https://doi.org/10.1002/ece3.1028\n",
      "https://doi.org/10.1002/ece3.1266\n",
      "https://doi.org/10.1111/mec.12197\n",
      "https://doi.org/10.1111/eva.12464\n",
      "https://doi.org/10.1111/1365-2656.12269\n",
      "https://doi.org/10.1002/ece3.3397\n",
      "https://doi.org/10.1007/s10592-019-01170-8\n",
      "https://doi.org/10.1111/1365-2664.12397\n",
      "https://doi.org/10.1371/journal.pone.0073695\n",
      "https://doi.org/https://mffp.gouv.qc.ca/publications/forets/connaissances/Norme-PET.pdf; https://doi.org/https://mffp.gouv.qc.ca/publications/forets/connaissances/Norme-PEP.pdf; https://doi.org/10.7809/b-e.00219; https://doi.org/http://charcoal.cnre.vt.edu/climate/\n",
      "https://doi.org/10.1002/ece3.3906\n",
      "https://doi.org/10.1111/mec.13795\n",
      "nan\n",
      "https://doi.org/10.1111/geb.12829\n",
      "https://doi.org/10.1890/12-2118.1\n",
      "https://doi.org/https://www.ncbi.nlm.nih.gov/bioproject/PRJNA725248; https://doi.org/https://www.bodc.ac.uk/data/bodc_database/nodb/cruise/16981/\n",
      "https://doi.org/10.1111/eva.12315\n",
      "https://doi.org/10.1111/2041-210X.12470\n",
      "https://doi.org/10.1002/ece3.4685\n",
      "https://doi.org/10.1098/rspb.2014.1779\n",
      "https://doi.org/10.1111/mec.12500\n",
      "https://doi.org/10.1111/oik.03868\n",
      "https://doi.org/10.1111/ddi.12496\n",
      "https://doi.org/10.1093/jhered/esx103\n",
      "https://doi.org/10.1111/j.1365-294X.2009.04370.x\n",
      "https://doi.org/10.1093/jhered/esx005\n",
      "https://doi.org/10.1111/mec.12142\n",
      "https://doi.org/10.1111/1365-2664.12598\n",
      "https://doi.org/10.1111/eva.12028\n",
      "https://doi.org/10.1111/j.1420-9101.2011.02418.x\n",
      "https://doi.org/10.1371/journal.pone.0128238\n",
      "https://doi.org/10.1002/ece3.7260\n",
      "https://doi.org/10.1098/rspb.2014.0649\n",
      "https://doi.org/10.1111/j.1420-9101.2010.02223.x\n",
      "https://doi.org/10.1111/1365-2745.12473\n",
      "https://doi.org/10.1371/journal.pone.0162325\n",
      "https://doi.org/10.1371/journal.pone.0204445\n",
      "https://doi.org/10.1098/rspb.2014.0502\n",
      "https://doi.org/10.1093/jhered/esw073\n",
      "nan\n",
      "https://doi.org/10.1111/1365-2664.12517\n",
      "nan\n",
      "https://doi.org/10.1111/1365-2664.12675\n",
      "https://doi.org/10.1098/rspb.2019.0856\n",
      "https://doi.org/10.1111/jav.01273\n",
      "https://doi.org/10.1111/jav.01273\n",
      "https://doi.org/10.1002/ece3.1086\n",
      "https://doi.org/10.1038/hdy.2013.54\n",
      "nan\n",
      "nan\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.1101/2021.12.01.469845; https://doi.org/10.5281/zenodo.6478222\n",
      "https://doi.org/10.5281/zenodo.5898699\n",
      "nan\n",
      "https://doi.org/10.1101/2020.10.07.329961\n",
      "https://doi.org/https://github.com/LauricReynes\n",
      "nan\n",
      "https://doi.org/10.5281/zenodo.4555787; https://doi.org/10.5281/zenodo.5245301\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.5281/zenodo.5661999\n",
      "https://doi.org/10.5281/zenodo.5831101\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.5751/ACE-00446-060105; https://doi.org/10.1016/S0006-3207(01)00079-9; https://doi.org/10.1016/S0006-3207(01)00079-9; https://doi.org/10.1016/j.agee.2017.04.009\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.1093/jhered/esaa023\n",
      "nan\n",
      "https://doi.org/10.1101/2020.04.21.042820; https://doi.org/https://github.com/CristescuLab/YAAP\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.1093/ee/nvy113\n",
      "nan\n",
      "https://doi.org/10.1111/oik.06668\n",
      "nan\n",
      "https://doi.org/10.5281/zenodo.5140459\n",
      "https://doi.org/10.5281/zenodo.4701262; https://doi.org/10.5281/zenodo.6246852\n",
      "https://doi.org/10.1111/1365-2745.13087\n",
      "https://doi.org/https://escholarship.org/uc/item/6r2801jn#main\n",
      "nan\n",
      "https://doi.org/10.1038/s41477-020-0647-x\n",
      "https://doi.org/10.1002/ece3.3581\n",
      "nan\n",
      "https://doi.org/10.1371/journal.pone.0129857\n",
      "nan\n",
      "https://doi.org/10.1111/1365-2745.13194\n",
      "https://doi.org/10.5281/zenodo.2231046\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.5281/zenodo.4697655; https://doi.org/10.1139/f87-301; https://doi.org/10.5281/zenodo.4701261\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.1111/eva.12791\n",
      "nan\n",
      "https://doi.org/10.1111/1365-2745.12459\n",
      "nan\n",
      "https://doi.org/10.5281/zenodo.4701262; https://doi.org/10.5281/zenodo.6209389\n",
      "https://doi.org/10.1002/edn3.7\n",
      "https://doi.org/10.1111/j.1600-0706.2013.00631.x\n",
      "https://doi.org/10.1002/ecs2.1788\n",
      "https://doi.org/10.1111/oik.03948\n",
      "https://doi.org/10.1111/1365-2664.12685\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.1101/2021.06.11.448048; https://doi.org/10.5281/zenodo.6075087\n",
      "https://doi.org/10.1101/2020.10.16.342725; https://doi.org/10.1002/ecs2.3858\n",
      "nan\n",
      "https://doi.org/10.1371/journal.pone.0155253\n",
      "https://doi.org/10.22541/au.161832268.87346989/v1\n",
      "https://doi.org/10.1371/journal.pone.0152495\n",
      "nan\n",
      "nan\n",
      "nan\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.1111/oik.04126\n",
      "https://doi.org/10.1111/1365-2656.12645\n",
      "https://doi.org/10.1101/005900; https://doi.org/10.6084/m9.figshare.1044306; https://doi.org/https://github.com/kgturner/Cdiff_genome; https://doi.org/https://github.com/kgturner/Cdiff_GWAS; https://doi.org/https://www.ncbi.nlm.nih.gov/sra/SRX1355843; https://doi.org/https://www.ncbi.nlm.nih.gov/nuccore/KJ690264; https://doi.org/https://www.ncbi.nlm.nih.gov/sra/PRJNA681918\n",
      "nan\n",
      "https://doi.org/10.1016/j.jglr.2019.12.009\n",
      "nan\n",
      "https://doi.org/10.1534/g3.119.400498\n",
      "nan\n",
      "https://doi.org/10.1111/mec.12022\n",
      "nan\n",
      "https://doi.org/10.1111/j.1365-294X.2011.05069.x\n",
      "https://doi.org/10.1038/hdy.2012.113\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.1371/journal.pgen.1006517\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.1111/1365-2664.13497\n",
      "nan\n",
      "https://doi.org/10.1111/jav.01766\n",
      "nan\n",
      "https://doi.org/10.1111/mec.13395\n",
      "nan\n",
      "https://doi.org/10.1111/jav.00824\n",
      "https://doi.org/10.1111/oik.07311\n",
      "https://doi.org/10.1093/plankt/fbs042\n",
      "https://doi.org/10.1002/ece3.3313\n",
      "nan\n",
      "https://doi.org/10.1111/1365-2435.12804\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.1093/jxb/erab127\n",
      "nan\n",
      "https://doi.org/10.6084/m9.figshare.13118132.v1\n",
      "https://doi.org/10.5281/zenodo.6862863\n",
      "nan\n",
      "https://doi.org/10.1111/1365-2435.13257\n",
      "https://doi.org/10.1093/treephys/tpu026; https://doi.org/10.3389/fpls.2015.00730; https://doi.org/10.1016/j.scitotenv.2021.149968; https://doi.org/10.1078/1125-7865-00034; https://doi.org/10.5281/zenodo.6675187; https://doi.org/10.5281/zenodo.6675189\n",
      "nan\n",
      "https://doi.org/10.1111/1365-2745.12728\n",
      "https://doi.org/10.1111/1365-2656.12619\n",
      "nan\n",
      "https://doi.org/10.1111/j.1558-5646.2011.01435.x\n",
      "https://doi.org/10.1111/2041-210X.12391\n",
      "nan\n",
      "nan\n",
      "https://doi.org/10.3897/zookeys.1041.64460\n",
      "https://doi.org/10.1098/rstb.2019.0035\n",
      "nan\n"
     ]
    }
   ],
   "source": [
    "scientific_dic = {}\n",
    "for url in df_datasets.cited_articles.tolist()[2:]:\n",
    "    print(url)\n",
    "    d = parse_article(url)\n",
    "    scientific_dic[url] = d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 598,
   "id": "19187111",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[nan,\n",
       " 'https://doi.org/10.1098/rsos.150669',\n",
       " 'https://doi.org/10.1639/0007-2745-119.1.008',\n",
       " 'https://doi.org/10.1371/journal.pone.0176706',\n",
       " 'https://doi.org/10.1371/journal.pone.0077514',\n",
       " 'https://doi.org/10.1371/journal.pone.0107929',\n",
       " 'https://doi.org/10.1600/036364416x692514',\n",
       " 'https://doi.org/10.3389/fpls.2019.00932',\n",
       " 'https://doi.org/10.1371/journal.pone.0073695',\n",
       " 'https://doi.org/https://mffp.gouv.qc.ca/publications/forets/connaissances/Norme-PET.pdf; https://doi.org/https://mffp.gouv.qc.ca/publications/forets/connaissances/Norme-PEP.pdf; https://doi.org/10.7809/b-e.00219; https://doi.org/http://charcoal.cnre.vt.edu/climate/',\n",
       " 'https://doi.org/https://www.ncbi.nlm.nih.gov/bioproject/PRJNA725248; https://doi.org/https://www.bodc.ac.uk/data/bodc_database/nodb/cruise/16981/',\n",
       " 'https://doi.org/10.1098/rspb.2014.1779',\n",
       " 'https://doi.org/10.1093/jhered/esx103',\n",
       " 'https://doi.org/10.1093/jhered/esx005',\n",
       " 'https://doi.org/10.1371/journal.pone.0128238',\n",
       " 'https://doi.org/10.1098/rspb.2014.0649',\n",
       " 'https://doi.org/10.1371/journal.pone.0162325',\n",
       " 'https://doi.org/10.1371/journal.pone.0204445',\n",
       " 'https://doi.org/10.1098/rspb.2014.0502',\n",
       " 'https://doi.org/10.1093/jhered/esw073',\n",
       " 'https://doi.org/10.1098/rspb.2019.0856',\n",
       " 'https://doi.org/10.1111/jav.01273',\n",
       " 'https://doi.org/10.1101/2021.12.01.469845; https://doi.org/10.5281/zenodo.6478222',\n",
       " 'https://doi.org/10.5281/zenodo.5898699',\n",
       " 'https://doi.org/10.1101/2020.10.07.329961',\n",
       " 'https://doi.org/https://github.com/LauricReynes',\n",
       " 'https://doi.org/10.5281/zenodo.4555787; https://doi.org/10.5281/zenodo.5245301',\n",
       " 'https://doi.org/10.5281/zenodo.5661999',\n",
       " 'https://doi.org/10.5281/zenodo.5831101',\n",
       " 'https://doi.org/10.5751/ACE-00446-060105; https://doi.org/10.1016/S0006-3207(01)00079-9; https://doi.org/10.1016/S0006-3207(01)00079-9; https://doi.org/10.1016/j.agee.2017.04.009',\n",
       " 'https://doi.org/10.1093/jhered/esaa023',\n",
       " 'https://doi.org/10.1101/2020.04.21.042820; https://doi.org/https://github.com/CristescuLab/YAAP',\n",
       " 'https://doi.org/10.1093/ee/nvy113',\n",
       " 'https://doi.org/10.5281/zenodo.5140459',\n",
       " 'https://doi.org/10.5281/zenodo.4701262; https://doi.org/10.5281/zenodo.6246852',\n",
       " 'https://doi.org/https://escholarship.org/uc/item/6r2801jn#main',\n",
       " 'https://doi.org/10.1371/journal.pone.0129857',\n",
       " 'https://doi.org/10.5281/zenodo.2231046',\n",
       " 'https://doi.org/10.5281/zenodo.4697655; https://doi.org/10.1139/f87-301; https://doi.org/10.5281/zenodo.4701261',\n",
       " 'https://doi.org/10.5281/zenodo.4701262; https://doi.org/10.5281/zenodo.6209389',\n",
       " 'https://doi.org/10.1101/2021.06.11.448048; https://doi.org/10.5281/zenodo.6075087',\n",
       " 'https://doi.org/10.1101/2020.10.16.342725; https://doi.org/10.1002/ecs2.3858',\n",
       " 'https://doi.org/10.1371/journal.pone.0155253',\n",
       " 'https://doi.org/10.22541/au.161832268.87346989/v1',\n",
       " 'https://doi.org/10.1371/journal.pone.0152495',\n",
       " 'https://doi.org/10.1101/005900; https://doi.org/10.6084/m9.figshare.1044306; https://doi.org/https://github.com/kgturner/Cdiff_genome; https://doi.org/https://github.com/kgturner/Cdiff_GWAS; https://doi.org/https://www.ncbi.nlm.nih.gov/sra/SRX1355843; https://doi.org/https://www.ncbi.nlm.nih.gov/nuccore/KJ690264; https://doi.org/https://www.ncbi.nlm.nih.gov/sra/PRJNA681918',\n",
       " 'https://doi.org/10.1016/j.jglr.2019.12.009',\n",
       " 'https://doi.org/10.1534/g3.119.400498',\n",
       " 'https://doi.org/10.1371/journal.pgen.1006517',\n",
       " 'https://doi.org/10.1111/jav.01766',\n",
       " 'https://doi.org/10.1111/jav.00824',\n",
       " 'https://doi.org/10.1093/plankt/fbs042',\n",
       " 'https://doi.org/10.1093/jxb/erab127',\n",
       " 'https://doi.org/10.6084/m9.figshare.13118132.v1',\n",
       " 'https://doi.org/10.5281/zenodo.6862863',\n",
       " 'https://doi.org/10.1093/treephys/tpu026; https://doi.org/10.3389/fpls.2015.00730; https://doi.org/10.1016/j.scitotenv.2021.149968; https://doi.org/10.1078/1125-7865-00034; https://doi.org/10.5281/zenodo.6675187; https://doi.org/10.5281/zenodo.6675189',\n",
       " 'https://doi.org/10.3897/zookeys.1041.64460',\n",
       " 'https://doi.org/10.1098/rstb.2019.0035']"
      ]
     },
     "execution_count": 598,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "found"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 599,
   "id": "02d174f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "url = \"https://doi.org/10.3389/fpls.2019.00932\"\n",
    "dr = webdriver.Chrome()\n",
    "dr.get(url)\n",
    "bs = BeautifulSoup(dr.page_source,\"html.parser\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 600,
   "id": "37709a9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<html lang=\"en\"><head><script src=\"https://m.addthis.com/live/red_lojson/300lo.json?si=639a4b7c7eb0dc80&amp;bkl=0&amp;bl=1&amp;pdt=2452&amp;sid=639a4b7c7eb0dc80&amp;pub=ra-535f7e222fc3ca72&amp;rev=v8.28.8-wp&amp;ln=en&amp;pc=men&amp;cb=0&amp;ab=-&amp;dp=www.frontiersin.org&amp;fp=articles%2F10.3389%2Ffpls.2019.00932%2Ffull&amp;fr=&amp;of=0&amp;pd=0&amp;irt=0&amp;vcl=0&amp;md=0&amp;ct=1&amp;tct=0&amp;abt=0&amp;cdn=0&amp;pi=1&amp;rb=0&amp;gen=100&amp;chr=UTF-8&amp;mk=Abies%20balsamea%2Cboreal%20forest%2CEctomycorrhiza%20(EcM)%2CEricaceous%20shrubs%2Cfacilitation%2CPicea%20mariana%2CPopulus%20tremuloides&amp;colc=1671056252863&amp;jsl=8193&amp;uvs=639a4b7c34154032000&amp;skipb=1&amp;callback=addthis.cbs.jsonp__486234041588388340\" type=\"text/javascript\"></script><script src=\"https://v1.addthisedge.com/live/boost/ra-535f7e222fc3ca72/_ate.track.config_resp\" type=\"text/javascript\"></script><script src=\"https://z.moatads.com/addthismoatframe568911941483/moatframe.js\" type=\"text/javascript\"></script>\n",
       "<!-- Google Tag Manager -->\n",
       "<script async=\"\" src=\"https://snap.licdn.com/li.lms-analytics/insight.beta.min.js\"></script><script async=\"\" src=\"https://s.saleswingsapp.com/sw.prod.min.js\" type=\"text/javascript\"></script><script async=\"\" src=\"https://www.googletagmanager.com/gtag/js?id=G-MF9DGCL3QL&amp;l=dataLayer&amp;cx=c\" type=\"text/javascript\"></script><script async=\"\" src=\"https://connect.facebook.net/signals/config/2098247140231737?v=2.9.90&amp;r=stable\"></script><script async=\"\" src=\"https://connect.facebook.net/en_US/fbevents.js\"></script><script async=\"\" src=\"https://static.ads-twitter.com/uwt.js\" type=\"text/javascript\"></script><script async=\"\" src=\"https://www.google-analytics.com/analytics.js\" type=\"text/javascript\"></script><script async=\"\" src=\"https://snap.licdn.com/li.lms-analytics/insight.min.js\" type=\"text/javascript\"></script><script async=\"\" src=\"https://cdnjs.cloudflare.com/ajax/libs/snowplow/2.18.2/sp.min.js\"></script><script async=\"\" src=\"https://www.googletagmanager.com/gtm.js?id=GTM-N279F7B&amp;gtm_auth=jfKETtlWamfZ4l02YiFCQw&amp;gtm_preview=env-1&amp;gtm_cookies_win=x\"></script><script async=\"\" src=\"https://www.googletagmanager.com/gtm.js?id=GTM-TFKV632&amp;gtm_auth=NsJH3Ts1-89I8lZURwQYmw&amp;gtm_preview=env-1&amp;gtm_cookies_win=x\"></script><script>\n",
       "        (function (w, d, s, l, i) {\n",
       "            w[l] = w[l] || []; w[l].push({ 'gtm.start': new Date().getTime(), event: 'gtm.js' });\n",
       "            var f = d.getElementsByTagName(s)[0], j = d.createElement(s), dl = l != 'dataLayer' ? '&l=' + l : ''; j.async = true;\n",
       "            j.src = 'https://www.googletagmanager.com/gtm.js?id=' + i + dl + '&gtm_auth=NsJH3Ts1-89I8lZURwQYmw&gtm_preview=env-1&gtm_cookies_win=x';\n",
       "            f.parentNode.insertBefore(j, f);\n",
       "        })(window, document, 'script', 'dataLayer', 'GTM-TFKV632');\n",
       "    </script>\n",
       "<script>\n",
       "        (function (w, d, s, l, i) {\n",
       "            w[l] = w[l] || []; w[l].push({ 'gtm.start': new Date().getTime(), event: 'gtm.js' });\n",
       "            var f = d.getElementsByTagName(s)[0], j = d.createElement(s), dl = l != 'dataLayer' ? '&l=' + l : ''; j.async = true;\n",
       "            j.src = 'https://www.googletagmanager.com/gtm.js?id=' + i + dl + '&gtm_auth=jfKETtlWamfZ4l02YiFCQw&gtm_preview=env-1&gtm_cookies_win=x';\n",
       "            f.parentNode.insertBefore(j, f);\n",
       "        })(window, document, 'script', 'dataLayer', 'GTM-N279F7B');\n",
       "    </script>\n",
       "<!-- End Google Tag Manager -->\n",
       "<script src=\"https://unpkg.com/vue@2.6.14/dist/vue.min.js\"></script>\n",
       "<!--Snowplow Tag-->\n",
       "<script type=\"text/javascript\">\n",
       "            (function (p, l, o, w, i, n, g) {\n",
       "                if (!p[i]) {\n",
       "                    p.GlobalSnowplowNamespace = p.GlobalSnowplowNamespace || [];\n",
       "                    p.GlobalSnowplowNamespace.push(i); p[i] = function () { (p[i].q = p[i].q || []).push(arguments) }; p[i].q = p[i].q || [];\n",
       "                    n = l.createElement(o); g = l.getElementsByTagName(o)[0]; n.async = 1; n.src = w;\n",
       "                    g.parentNode.insertBefore(n, g)\n",
       "                }\n",
       "            }(window, document, \"script\", \"https://cdnjs.cloudflare.com/ajax/libs/snowplow/2.18.2/sp.min.js\", \"snowplow\"));\n",
       "\n",
       "            snowplow('newTracker', 'spTracker-inline-live', 'sp-live-collector.cdp.frontiersin.io', {\n",
       "                appId: '1',\n",
       "                eventMethod: 'post',\n",
       "                anonymousTracking: true,\n",
       "                stateStorageStrategy: 'none',\n",
       "                contexts: {\n",
       "                    webPage: true\n",
       "                }\n",
       "            });\n",
       "            snowplow('enableActivityTracking', 30, 10);\n",
       "            window.snowplow('trackPageView');\n",
       "            snowplow('enableAnonymousTracking', { withServerAnonymisation: true });\n",
       "            snowplow('clearUserData');\n",
       "            snowplow('setUserId', '0');\n",
       "    </script>\n",
       "<!--End Snowplow Tag-->\n",
       "<meta charset=\"utf-8\"/>\n",
       "<meta content=\"IE=edge\" http-equiv=\"X-UA-Compatible\"/><script type=\"text/javascript\">window.NREUM||(NREUM={});NREUM.info = {\"beacon\":\"bam.nr-data.net\",\"errorBeacon\":\"bam.nr-data.net\",\"licenseKey\":\"598a124f17\",\"applicationID\":\"3007887\",\"transactionName\":\"MQcDMkECCkNSW0YMWghNLDBwTCVCR1FRCVAlDQ8SQQwIXFZKHSNACg41A0sXJkl3d3s=\",\"queueTime\":0,\"applicationTime\":289,\"agent\":\"\",\"atts\":\"\"}</script><script type=\"text/javascript\">(window.NREUM||(NREUM={})).init={ajax:{deny_list:[\"bam.nr-data.net\"]}};(window.NREUM||(NREUM={})).loader_config={xpid:\"VgUHUl5WGwAAVFZaDwY=\",licenseKey:\"598a124f17\",applicationID:\"3007887\"};/*! For license information please see nr-loader-full-1220.min.js.LICENSE.txt */\n",
       "!function(t,e){\"object\"==typeof exports&&\"object\"==typeof module?module.exports=e():\"function\"==typeof define&&define.amd?define([],e):\"object\"==typeof exports?exports.NRBA=e():t.NRBA=e()}(self,(function(){return function(){var t,e,n={9034:function(t,e,n){\"use strict\";var r=n(4168);e.Z=(0,r.ky)(16)},5973:function(t,e,n){\"use strict\";n.d(e,{I:function(){return r}});var r=0,i=navigator.userAgent.match(/Firefox[\\/\\s](\\d+\\.\\d+)/);i&&(r=+i[1])},4280:function(t,e,n){\"use strict\";n.d(e,{H:function(){return i}});var r=document.createElement(\"div\");r.innerHTML=\"\\x3c!--[if lte IE 6]><div></div><![endif]--\\x3e\\x3c!--[if lte IE 7]><div></div><![endif]--\\x3e\\x3c!--[if lte IE 8]><div></div><![endif]--\\x3e\\x3c!--[if lte IE 9]><div></div><![endif]--\\x3e\";var i,o=r.getElementsByTagName(\"div\").length;i=4===o?6:3===o?7:2===o?8:1===o?9:0},5955:function(t,e,n){\"use strict\";n.d(e,{I:function(){return r}});var r=function(t,e){var n=this;return t&&\"object\"==typeof t?e&&\"object\"==typeof e?(Object.assign(this,e),void Object.entries(t).forEach((function(t){var e=t[0],r=t[1];n[e]=r}))):console.error(\"setting a Configurable requires a model to set its initial properties\"):console.error(\"setting a Configurable requires an object as input\")}},441:function(t,e,n){\"use strict\";n.d(e,{C:function(){return c},L:function(){return u}});var r=n(1424),i=n(5955),o={beacon:r.ce.beacon,errorBeacon:r.ce.errorBeacon,licenseKey:void 0,applicationID:void 0,sa:void 0,queueTime:void 0,applicationTime:void 0,ttGuid:void 0,user:void 0,account:void 0,product:void 0,extra:void 0,jsAttributes:{},userAttributes:void 0,atts:void 0,transactionName:void 0,tNamePlain:void 0},a={};function c(t){if(!t)throw new Error(\"All info objects require an agent identifier!\");if(!a[t])throw new Error(\"Info for \"+t+\" was never set\");return a[t]}function u(t,e){if(!t)throw new Error(\"All info objects require an agent identifier!\");a[t]=new i.I(e,o),(0,r.Qy)(t,a[t],\"info\")}},1476:function(t,e,n){\"use strict\";n.d(e,{Dg:function(){return u},Mt:function(){return s},P_:function(){return c}});var r=n(1424),i=n(5955),o={privacy:{cookies_enabled:!0},ajax:{deny_list:void 0,enabled:!0},distributed_tracing:{enabled:void 0,exclude_newrelic_header:void 0,cors_use_newrelic_header:void 0,cors_use_tracecontext_headers:void 0,allowed_origins:void 0},ssl:void 0,obfuscate:void 0,jserrors:{enabled:!0},metrics:{enabled:!0},page_action:{enabled:!0},page_view_event:{enabled:!0},page_view_timing:{enabled:!0},session_trace:{enabled:!0},spa:{enabled:!0}},a={};function c(t){if(!t)throw new Error(\"All configuration objects require an agent identifier!\");if(!a[t])throw new Error(\"Configuration for \"+t+\" was never set\");return a[t]}function u(t,e){if(!t)throw new Error(\"All configuration objects require an agent identifier!\");a[t]=new i.I(e,o),(0,r.Qy)(t,a[t],\"config\")}function s(t,e){if(!t)throw new Error(\"All configuration objects require an agent identifier!\");var n=c(t);if(n){for(var r=e.split(\".\"),i=0;i<r.length-1;i++)if(\"object\"!=typeof(n=n[r[i]]))return;n=n[r[r.length-1]]}return n}},2085:function(t,e,n){\"use strict\";n.d(e,{Y:function(){return r}});var r=(0,n(1424).mF)().o},1220:function(t,e,n){\"use strict\";n.d(e,{O:function(){return b},s:function(){return O}});var r={};n.r(r),n.d(r,{agent:function(){return a},match:function(){return f},version:function(){return c}});var i=n(4280),o=n(6959),a=null,c=null;if(navigator.userAgent){var u=navigator.userAgent,s=u.match(/Version\\/(\\S+)\\s+Safari/);s&&-1===u.indexOf(\"Chrome\")&&-1===u.indexOf(\"Chromium\")&&(a=\"Safari\",c=s[1])}function f(t,e){if(!a)return!1;if(t!==a)return!1;if(!e)return!0;if(!c)return!1;for(var n=c.split(\".\"),r=e.split(\".\"),i=0;i<r.length;i++)if(r[i]!==n[i])return!1;return!0}var d=n(5955),l=n(1424),p=n(4168),h=window.sessionStorage,v=\"NRBA_SESSION_ID\";var g=n(1476),m=window.XMLHttpRequest,y=m&&m.prototype,w={};function b(t){if(!t)throw new Error(\"All runtime objects require an agent identifier!\");if(!w[t])throw new Error(\"Runtime for \"+t+\" was never set\");return w[t]}function O(t,e){if(!t)throw new Error(\"All runtime objects require an agent identifier!\");var n,a;w[t]=new d.I(e,(n=t,{customTransaction:void 0,disabled:!1,features:{},maxBytes:6===i.H?2e3:3e4,offset:(0,o.yf)(),onerror:void 0,origin:\"\"+window.location,ptid:void 0,releaseIds:{},sessionId:!0===(0,g.Mt)(n,\"privacy.cookies_enabled\")?(null===(a=h.getItem(v))&&(a=(0,p.ky)(16),h.setItem(v,a)),a):\"0\",xhrWrappable:m&&y&&y.addEventListener&&!/CriOS/.test(navigator.userAgent),userAgent:r})),(0,l.Qy)(t,w[t],\"runtime\")}},158:function(t,e,n){\"use strict\";n.d(e,{q:function(){return r}});var r=[\"1220\",\"PROD\"].filter((function(t){return t})).join(\".\")},3707:function(t,e,n){\"use strict\";n.d(e,{w:function(){return i}});var r={agentIdentifier:\"\"},i=function(t){var e=this;if(\"object\"!=typeof t)return console.error(\"shared context requires an object as input\");this.sharedContext={},Object.assign(this.sharedContext,r),Object.entries(t).forEach((function(t){var n=t[0],i=t[1];Object.keys(r).includes(n)&&(e.sharedContext[n]=i)}))}},1776:function(t,e,n){\"use strict\";n.d(e,{ee:function(){return r}});var r,i=n(1424),o=n(4217),a=n(357),c=\"nr@context\",u=(0,i.fP)();function s(){}function f(){return new s}function d(){(r.backlog.api||r.backlog.feature)&&(r.aborted=!0,r.backlog={})}u.ee?r=u.ee:(r=function t(e,n){var i={},u={},l={},p={on:g,addEventListener:g,removeEventListener:m,emit:v,get:w,listeners:y,context:h,buffer:b,abort:d,aborted:!1,isBuffering:O,debugId:n,backlog:e&&e.backlog?e.backlog:{}};return p;function h(t){return t&&t instanceof s?t:t?(0,o.X)(t,c,f):f()}function v(t,n,i,o,a){if(!1!==a&&(a=!0),!r.aborted||o){e&&a&&e.emit(t,n,i);for(var c=h(i),s=y(t),f=s.length,d=0;d<f;d++)s[d].apply(c,n);var l=x()[u[t]];return l&&l.push([p,t,n,c]),c}}function g(t,e){i[t]=y(t).concat(e)}function m(t,e){var n=i[t];if(n)for(var r=0;r<n.length;r++)n[r]===e&&n.splice(r,1)}function y(t){return i[t]||[]}function w(e){return l[e]=l[e]||t(p,e)}function b(t,e){var n=x();p.aborted||(0,a.D)(t,(function(t,r){e=e||\"feature\",u[r]=e,e in n||(n[e]=[])}))}function O(t){return!!x()[u[t]]}function x(){return p.backlog}}(void 0,\"globalEE\"),u.ee=r)},7361:function(t,e,n){\"use strict\";n.d(e,{E:function(){return r},p:function(){return i}});var r=n(1776).ee.get(\"handle\");function i(t,e,n,i,o){o?(o.buffer([t],i),o.emit(t,e,n)):(r.buffer([t],i),r.emit(t,e,n))}},3350:function(t,e,n){\"use strict\";n.d(e,{X:function(){return o}});var r=n(7361);o.on=a;var i=o.handlers={};function o(t,e,n,o){a(o||r.E,i,t,e,n)}function a(t,e,n,i,o){o||(o=\"feature\"),t||(t=r.E);var a=e[o]=e[o]||{};(a[n]=a[n]||[]).push([t,i])}},4408:function(t,e,n){\"use strict\";n.d(e,{m:function(){return o}});var r=!1;try{var i=Object.defineProperty({},\"passive\",{get:function(){r=!0}});window.addEventListener(\"testPassive\",null,i),window.removeEventListener(\"testPassive\",null,i)}catch(t){}function o(t){return r?{passive:!0,capture:!!t}:!!t}},4168:function(t,e,n){\"use strict\";function r(){var t=null,e=0,n=window.crypto||window.msCrypto;function r(){return t?15&t[e++]:16*Math.random()|0}n&&n.getRandomValues&&(t=n.getRandomValues(new Uint8Array(31)));for(var i,o=\"xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx\",a=\"\",c=0;c<o.length;c++)a+=\"x\"===(i=o[c])?r().toString(16):\"y\"===i?(i=3&r()|8).toString(16):i;return a}function i(){return a(16)}function o(){return a(32)}function a(t){var e=null,n=0,r=window.crypto||window.msCrypto;r&&r.getRandomValues&&Uint8Array&&(e=r.getRandomValues(new Uint8Array(31)));for(var i=[],o=0;o<t;o++)i.push(a().toString(16));return i.join(\"\");function a(){return e?15&e[n++]:16*Math.random()|0}}n.d(e,{Ht:function(){return o},M:function(){return i},Rl:function(){return r},ky:function(){return a}})},6959:function(t,e,n){\"use strict\";n.d(e,{nb:function(){return u},os:function(){return s},yf:function(){return c},zO:function(){return a}});var r=n(2364),i=(new Date).getTime(),o=i;function a(){return r.G&&performance.now?Math.round(performance.now()):(i=Math.max((new Date).getTime(),i))-o}function c(){return i}function u(t){o=t}function s(){return o}},2364:function(t,e,n){\"use strict\";n.d(e,{G:function(){return r}});var r=void 0!==window.performance&&window.performance.timing&&void 0!==window.performance.timing.navigationStart},1793:function(t,e,n){\"use strict\";function r(t){var e,n=0;for(e=0;e<t.length;e++)n+=(e+1)*t.charCodeAt(e);return Math.abs(n)}n.d(e,{v:function(){return s},s:function(){return u}});var i=n(6972),o=n(5973),a=n(6959),c=n(2364),u=!0;function s(t){var e=function(){if(o.I&&o.I<9)return;if(c.G)return u=!1,window.performance.timing.navigationStart}()||function(){for(var t=document.cookie.split(\" \"),e=0;e<t.length;e++)if(0===t[e].indexOf(\"NREUM=\")){for(var n,i,o,a,c=t[e].substring(\"NREUM=\".length).split(\"&\"),u=0;u<c.length;u++)0===c[u].indexOf(\"s=\")?o=c[u].substring(2):0===c[u].indexOf(\"p=\")?\";\"===(i=c[u].substring(2)).charAt(i.length-1)&&(i=i.substr(0,i.length-1)):0===c[u].indexOf(\"r=\")&&\";\"===(n=c[u].substring(2)).charAt(n.length-1)&&(n=n.substr(0,n.length-1));if(n){var s=r(document.referrer);(a=s==n)||(a=r(document.location.href)==n&&s==i)}if(a&&o){if((new Date).getTime()-o>6e4)return;return o}}}();e&&((0,i.B)(t,\"starttime\",e),(0,a.nb)(e))}},6972:function(t,e,n){\"use strict\";n.d(e,{B:function(){return o},L:function(){return a}});var r=n(6959),i={};function o(t,e,n){void 0===n&&(n=(0,r.zO)()+(0,r.os)()),i[t]=i[t]||{},i[t][e]=n}function a(t,e,n,r){var o,a,c=t.sharedContext.agentIdentifier,u=null==(o=i[c])?void 0:o[n],s=null==(a=i[c])?void 0:a[r];void 0!==u&&void 0!==s&&t.store(\"measures\",e,{value:s-u})}},7001:function(t,e,n){\"use strict\";n.d(e,{e:function(){return i}});var r={};function i(t){if(t in r)return r[t];if(0===(t||\"\").indexOf(\"data:\"))return{protocol:\"data\"};var e=document.createElement(\"a\"),n=window.location,i={};e.href=t,i.port=e.port;var o=e.href.split(\"://\");!i.port&&o[1]&&(i.port=o[1].split(\"/\")[0].split(\"@\").pop().split(\":\")[1]),i.port&&\"0\"!==i.port||(i.port=\"https\"===o[0]?\"443\":\"80\"),i.hostname=e.hostname||n.hostname,i.pathname=e.pathname,i.protocol=o[0],\"/\"!==i.pathname.charAt(0)&&(i.pathname=\"/\"+i.pathname);var a=!e.protocol||\":\"===e.protocol||e.protocol===n.protocol,c=e.hostname===document.domain&&e.port===n.port;return i.sameOrigin=a&&(!e.hostname||c),\"/\"===i.pathname&&(r[t]=i),i}},7299:function(t,e,n){\"use strict\";n.d(e,{T:function(){return a}});var r=window,i=r;function o(){return i}var a={isFileProtocol:function(){var t=o(),e=!(!t.location||!t.location.protocol||\"file:\"!==t.location.protocol);e&&(a.supportabilityMetricSent=!0);return e},supportabilityMetricSent:!1}},847:function(t,e,n){\"use strict\";n.d(e,{K:function(){return a}});var r=n(1220),i=n(1476),o=[\"ajax\",\"jserrors\",\"metrics\",\"page_action\",\"page_view_event\",\"page_view_timing\",\"session_trace\",\"spa\"];function a(t){var e={};return o.forEach((function(n){e[n]=function(t,e){return!0!==(0,r.O)(e).disabled&&!1!==(0,i.Mt)(e,t+\".enabled\")}(n,t)})),e}},5023:function(t,e,n){\"use strict\";n.d(e,{W:function(){return i}});var r=n(1776),i=function(t,e,n){void 0===n&&(n=[]),this.agentIdentifier=t,this.aggregator=e,this.ee=r.ee.get(t),this.externalFeatures=n}},4217:function(t,e,n){\"use strict\";n.d(e,{X:function(){return i}});var r=Object.prototype.hasOwnProperty;function i(t,e,n){if(r.call(t,e))return t[e];var i=n();if(Object.defineProperty&&Object.keys)try{return Object.defineProperty(t,e,{value:i,writable:!0,enumerable:!1}),i}catch(t){}return t[e]=i,i}},357:function(t,e,n){\"use strict\";n.d(e,{D:function(){return i}});var r=Object.prototype.hasOwnProperty;function i(t,e){var n=[],i=\"\",o=0;for(i in t)r.call(t,i)&&(n[o]=e(i,t[i]),o+=1);return n}},603:function(t,e,n){\"use strict\";n.d(e,{$c:function(){return s},Ng:function(){return f},RR:function(){return u}});var r=n(1476),i=n(3707),o=n(7299);function a(t,e){return a=Object.setPrototypeOf?Object.setPrototypeOf.bind():function(t,e){return t.__proto__=e,t},a(t,e)}var c={regex:/^file:\\/\\/(.*)/,replacement:\"file://OBFUSCATED\"},u=function(t){var e,n;function r(e){return t.call(this,e)||this}n=t,(e=r).prototype=Object.create(n.prototype),e.prototype.constructor=e,a(e,n);var i=r.prototype;return i.shouldObfuscate=function(){return s(this.sharedContext.agentIdentifier).length>0},i.obfuscateString=function(t){if(!t||\"string\"!=typeof t)return t;for(var e=s(this.sharedContext.agentIdentifier),n=t,r=0;r<e.length;r++){var i=e[r].regex,o=e[r].replacement||\"*\";n=n.replace(i,o)}return n},r}(i.w);function s(t){var e=[],n=(0,r.Mt)(t,\"obfuscate\")||[];return e=e.concat(n),o.T.isFileProtocol()&&e.push(c),e}function f(t){for(var e=!1,n=!1,r=0;r<t.length;r++){\"regex\"in t[r]?\"string\"!=typeof t[r].regex&&t[r].regex.constructor!==RegExp&&(console&&console.warn&&console.warn('An obfuscation replacement rule contains a \"regex\" value with an invalid type (must be a string or RegExp)'),n=!0):(console&&console.warn&&console.warn('An obfuscation replacement rule was detected missing a \"regex\" value.'),n=!0);var i=t[r].replacement;i&&\"string\"!=typeof i&&(console&&console.warn&&console.warn('An obfuscation replacement rule contains a \"replacement\" value with an invalid type (must be a string)'),e=!0)}return!e&&!n}},1424:function(t,e,n){\"use strict\";n.d(e,{EZ:function(){return u},Qy:function(){return c},ce:function(){return i},fP:function(){return o},gG:function(){return s},mF:function(){return a}});var r=n(6959),i={beacon:\"bam.nr-data.net\",errorBeacon:\"bam.nr-data.net\"};function o(){return window.NREUM||(window.NREUM={}),void 0===window.newrelic&&(window.newrelic=window.NREUM),window.NREUM}function a(){var t=o();if(!t.o){var e=window,n=e.XMLHttpRequest;t.o={ST:setTimeout,SI:e.setImmediate,CT:clearTimeout,XHR:n,REQ:e.Request,EV:e.Event,PR:e.Promise,MO:e.MutationObserver,FETCH:e.fetch}}return t}function c(t,e,n){var i,a,c=o(),u=c.initializedAgents||{},s=u[t]||{};return Object.keys(s).length||(s.initializedAt={ms:(0,r.zO)(),date:new Date}),c.initializedAgents=Object.assign({},u,((a={})[t]=Object.assign({},s,((i={})[n]=e,i)),a)),c}function u(t,e){o()[t]=e}function s(){var t,e;return t=o(),e=t.info||{},t.info=Object.assign({beacon:i.beacon,errorBeacon:i.errorBeacon},e),function(){var t=o(),e=t.init||{};t.init=Object.assign({},e)}(),a(),function(){var t=o(),e=t.loader_config||{};t.loader_config=Object.assign({},e)}(),o()}},5098:function(t,e,n){\"use strict\";function r(){return\"PerformanceObserver\"in window&&\"function\"==typeof window.PerformanceObserver}n.d(e,{W:function(){return r}})},8539:function(t){t.exports=function(t,e,n){e||(e=0),void 0===n&&(n=t?t.length:0);for(var r=-1,i=n-e||0,o=Array(i<0?0:i);++r<i;)o[r]=t[e+r];return o}},8118:function(t){t.exports=function(t,e,n){e||(e=0),void 0===n&&(n=t?t.length:0);for(var r=-1,i=n-e||0,o=Array(i<0?0:i);++r<i;)o[r]=t[e+r];return o}}},r={};function i(t){var e=r[t];if(void 0!==e)return e.exports;var o=r[t]={exports:{}};return n[t](o,o.exports,i),o.exports}i.m=n,i.n=function(t){var e=t&&t.__esModule?function(){return t.default}:function(){return t};return i.d(e,{a:e}),e},i.d=function(t,e){for(var n in e)i.o(e,n)&&!i.o(t,n)&&Object.defineProperty(t,n,{enumerable:!0,get:e[n]})},i.f={},i.e=function(t){return Promise.all(Object.keys(i.f).reduce((function(e,n){return i.f[n](t,e),e}),[]))},i.u=function(t){return t+\".\"+i.h().slice(0,8)+\"-1220.js\"},i.h=function(){return\"2d6a2503b7f18a5b77dd\"},i.o=function(t,e){return Object.prototype.hasOwnProperty.call(t,e)},t={},e=\"NRBA:\",i.l=function(n,r,o,a){if(t[n])t[n].push(r);else{var c,u;if(void 0!==o)for(var s=document.getElementsByTagName(\"script\"),f=0;f<s.length;f++){var d=s[f];if(d.getAttribute(\"src\")==n||d.getAttribute(\"data-webpack\")==e+o){c=d;break}}c||(u=!0,(c=document.createElement(\"script\")).charset=\"utf-8\",c.timeout=120,i.nc&&c.setAttribute(\"nonce\",i.nc),c.setAttribute(\"data-webpack\",e+o),c.src=n),t[n]=[r];var l=function(e,r){c.onerror=c.onload=null,clearTimeout(p);var i=t[n];if(delete t[n],c.parentNode&&c.parentNode.removeChild(c),i&&i.forEach((function(t){return t(r)})),e)return e(r)},p=setTimeout(l.bind(null,void 0,{type:\"timeout\",target:c}),12e4);c.onerror=l.bind(null,c.onerror),c.onload=l.bind(null,c.onload),u&&document.head.appendChild(c)}},i.r=function(t){\"undefined\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(t,Symbol.toStringTag,{value:\"Module\"}),Object.defineProperty(t,\"__esModule\",{value:!0})},i.p=\"https://js-agent.newrelic.com/\",function(){var t={908:0,172:0};i.f.j=function(e,n){var r=i.o(t,e)?t[e]:void 0;if(0!==r)if(r)n.push(r[2]);else{var o=new Promise((function(n,i){r=t[e]=[n,i]}));n.push(r[2]=o);var a=i.p+i.u(e),c=new Error;i.l(a,(function(n){if(i.o(t,e)&&(0!==(r=t[e])&&(t[e]=void 0),r)){var o=n&&(\"load\"===n.type?\"missing\":n.type),a=n&&n.target&&n.target.src;c.message=\"Loading chunk \"+e+\" failed.\\n(\"+o+\": \"+a+\")\",c.name=\"ChunkLoadError\",c.type=o,c.request=a,r[1](c)}}),\"chunk-\"+e,e)}};var e=function(e,n){var r,o,a=n[0],c=n[1],u=n[2],s=0;if(a.some((function(e){return 0!==t[e]}))){for(r in c)i.o(c,r)&&(i.m[r]=c[r]);if(u)u(i)}for(e&&e(n);s<a.length;s++)o=a[s],i.o(t,o)&&t[o]&&t[o][0](),t[o]=0},n=self.webpackChunkNRBA=self.webpackChunkNRBA||[];n.forEach(e.bind(null,0)),n.push=e.bind(null,n.push.bind(n))}();var o={};return function(){\"use strict\";i.r(o);var t=i(9034),e=i(4408),n=window,r=n.document;function a(t){\"complete\"===r.readyState&&t()}function c(t){a(t),r.addEventListener?n.addEventListener(\"load\",t,(0,e.m)(!1)):n.attachEvent(\"onload\",t)}function u(t){a(t),r.addEventListener?r.addEventListener(\"DOMContentLoaded\",t,(0,e.m)(!1)):r.attachEvent(\"onreadystatechange\",a)}var s=i(1776);function f(){f=function(){return t};var t={},e=Object.prototype,n=e.hasOwnProperty,r=\"function\"==typeof Symbol?Symbol:{},i=r.iterator||\"@@iterator\",o=r.asyncIterator||\"@@asyncIterator\",a=r.toStringTag||\"@@toStringTag\";function c(t,e,n){return Object.defineProperty(t,e,{value:n,enumerable:!0,configurable:!0,writable:!0}),t[e]}try{c({},\"\")}catch(t){c=function(t,e,n){return t[e]=n}}function u(t,e,n,r){var i=e&&e.prototype instanceof l?e:l,o=Object.create(i.prototype),a=new j(r||[]);return o._invoke=function(t,e,n){var r=\"suspendedStart\";return function(i,o){if(\"executing\"===r)throw new Error(\"Generator is already running\");if(\"completed\"===r){if(\"throw\"===i)throw o;return P()}for(n.method=i,n.arg=o;;){var a=n.delegate;if(a){var c=O(a,n);if(c){if(c===d)continue;return c}}if(\"next\"===n.method)n.sent=n._sent=n.arg;else if(\"throw\"===n.method){if(\"suspendedStart\"===r)throw r=\"completed\",n.arg;n.dispatchException(n.arg)}else\"return\"===n.method&&n.abrupt(\"return\",n.arg);r=\"executing\";var u=s(t,e,n);if(\"normal\"===u.type){if(r=n.done?\"completed\":\"suspendedYield\",u.arg===d)continue;return{value:u.arg,done:n.done}}\"throw\"===u.type&&(r=\"completed\",n.method=\"throw\",n.arg=u.arg)}}}(t,n,a),o}function s(t,e,n){try{return{type:\"normal\",arg:t.call(e,n)}}catch(t){return{type:\"throw\",arg:t}}}t.wrap=u;var d={};function l(){}function p(){}function h(){}var v={};c(v,i,(function(){return this}));var g=Object.getPrototypeOf,m=g&&g(g(_([])));m&&m!==e&&n.call(m,i)&&(v=m);var y=h.prototype=l.prototype=Object.create(v);function w(t){[\"next\",\"throw\",\"return\"].forEach((function(e){c(t,e,(function(t){return this._invoke(e,t)}))}))}function b(t,e){function r(i,o,a,c){var u=s(t[i],t,o);if(\"throw\"!==u.type){var f=u.arg,d=f.value;return d&&\"object\"==typeof d&&n.call(d,\"__await\")?e.resolve(d.__await).then((function(t){r(\"next\",t,a,c)}),(function(t){r(\"throw\",t,a,c)})):e.resolve(d).then((function(t){f.value=t,a(f)}),(function(t){return r(\"throw\",t,a,c)}))}c(u.arg)}var i;this._invoke=function(t,n){function o(){return new e((function(e,i){r(t,n,e,i)}))}return i=i?i.then(o,o):o()}}function O(t,e){var n=t.iterator[e.method];if(void 0===n){if(e.delegate=null,\"throw\"===e.method){if(t.iterator.return&&(e.method=\"return\",e.arg=void 0,O(t,e),\"throw\"===e.method))return d;e.method=\"throw\",e.arg=new TypeError(\"The iterator does not provide a 'throw' method\")}return d}var r=s(n,t.iterator,e.arg);if(\"throw\"===r.type)return e.method=\"throw\",e.arg=r.arg,e.delegate=null,d;var i=r.arg;return i?i.done?(e[t.resultName]=i.value,e.next=t.nextLoc,\"return\"!==e.method&&(e.method=\"next\",e.arg=void 0),e.delegate=null,d):i:(e.method=\"throw\",e.arg=new TypeError(\"iterator result is not an object\"),e.delegate=null,d)}function x(t){var e={tryLoc:t[0]};1 in t&&(e.catchLoc=t[1]),2 in t&&(e.finallyLoc=t[2],e.afterLoc=t[3]),this.tryEntries.push(e)}function E(t){var e=t.completion||{};e.type=\"normal\",delete e.arg,t.completion=e}function j(t){this.tryEntries=[{tryLoc:\"root\"}],t.forEach(x,this),this.reset(!0)}function _(t){if(t){var e=t[i];if(e)return e.call(t);if(\"function\"==typeof t.next)return t;if(!isNaN(t.length)){var r=-1,o=function e(){for(;++r<t.length;)if(n.call(t,r))return e.value=t[r],e.done=!1,e;return e.value=void 0,e.done=!0,e};return o.next=o}}return{next:P}}function P(){return{value:void 0,done:!0}}return p.prototype=h,c(y,\"constructor\",h),c(h,\"constructor\",p),p.displayName=c(h,a,\"GeneratorFunction\"),t.isGeneratorFunction=function(t){var e=\"function\"==typeof t&&t.constructor;return!!e&&(e===p||\"GeneratorFunction\"===(e.displayName||e.name))},t.mark=function(t){return Object.setPrototypeOf?Object.setPrototypeOf(t,h):(t.__proto__=h,c(t,a,\"GeneratorFunction\")),t.prototype=Object.create(y),t},t.awrap=function(t){return{__await:t}},w(b.prototype),c(b.prototype,o,(function(){return this})),t.AsyncIterator=b,t.async=function(e,n,r,i,o){void 0===o&&(o=Promise);var a=new b(u(e,n,r,i),o);return t.isGeneratorFunction(n)?a:a.next().then((function(t){return t.done?t.value:a.next()}))},w(y),c(y,a,\"Generator\"),c(y,i,(function(){return this})),c(y,\"toString\",(function(){return\"[object Generator]\"})),t.keys=function(t){var e=[];for(var n in t)e.push(n);return e.reverse(),function n(){for(;e.length;){var r=e.pop();if(r in t)return n.value=r,n.done=!1,n}return n.done=!0,n}},t.values=_,j.prototype={constructor:j,reset:function(t){if(this.prev=0,this.next=0,this.sent=this._sent=void 0,this.done=!1,this.delegate=null,this.method=\"next\",this.arg=void 0,this.tryEntries.forEach(E),!t)for(var e in this)\"t\"===e.charAt(0)&&n.call(this,e)&&!isNaN(+e.slice(1))&&(this[e]=void 0)},stop:function(){this.done=!0;var t=this.tryEntries[0].completion;if(\"throw\"===t.type)throw t.arg;return this.rval},dispatchException:function(t){if(this.done)throw t;var e=this;function r(n,r){return a.type=\"throw\",a.arg=t,e.next=n,r&&(e.method=\"next\",e.arg=void 0),!!r}for(var i=this.tryEntries.length-1;i>=0;--i){var o=this.tryEntries[i],a=o.completion;if(\"root\"===o.tryLoc)return r(\"end\");if(o.tryLoc<=this.prev){var c=n.call(o,\"catchLoc\"),u=n.call(o,\"finallyLoc\");if(c&&u){if(this.prev<o.catchLoc)return r(o.catchLoc,!0);if(this.prev<o.finallyLoc)return r(o.finallyLoc)}else if(c){if(this.prev<o.catchLoc)return r(o.catchLoc,!0)}else{if(!u)throw new Error(\"try statement without catch or finally\");if(this.prev<o.finallyLoc)return r(o.finallyLoc)}}}},abrupt:function(t,e){for(var r=this.tryEntries.length-1;r>=0;--r){var i=this.tryEntries[r];if(i.tryLoc<=this.prev&&n.call(i,\"finallyLoc\")&&this.prev<i.finallyLoc){var o=i;break}}o&&(\"break\"===t||\"continue\"===t)&&o.tryLoc<=e&&e<=o.finallyLoc&&(o=null);var a=o?o.completion:{};return a.type=t,a.arg=e,o?(this.method=\"next\",this.next=o.finallyLoc,d):this.complete(a)},complete:function(t,e){if(\"throw\"===t.type)throw t.arg;return\"break\"===t.type||\"continue\"===t.type?this.next=t.arg:\"return\"===t.type?(this.rval=this.arg=t.arg,this.method=\"return\",this.next=\"end\"):\"normal\"===t.type&&e&&(this.next=e),d},finish:function(t){for(var e=this.tryEntries.length-1;e>=0;--e){var n=this.tryEntries[e];if(n.finallyLoc===t)return this.complete(n.completion,n.afterLoc),E(n),d}},catch:function(t){for(var e=this.tryEntries.length-1;e>=0;--e){var n=this.tryEntries[e];if(n.tryLoc===t){var r=n.completion;if(\"throw\"===r.type){var i=r.arg;E(n)}return i}}throw new Error(\"illegal catch attempt\")},delegateYield:function(t,e,n){return this.delegate={iterator:_(t),resultName:e,nextLoc:n},\"next\"===this.method&&(this.arg=void 0),d}},t}function d(t,e,n,r,i,o,a){try{var c=t[o](a),u=c.value}catch(t){return void n(t)}c.done?e(u):Promise.resolve(u).then(r,i)}var l=0;function p(t){var e;(e=f().mark((function e(){var n,r;return f().wrap((function(e){for(;;)switch(e.prev=e.next){case 0:if(!l++){e.next=2;break}return e.abrupt(\"return\");case 2:return e.prev=2,e.next=5,i.e(552).then(i.bind(i,5552));case 5:return n=e.sent,r=n.aggregator,e.next=9,r(t);case 9:e.next=15;break;case 11:e.prev=11,e.t0=e.catch(2),console.error(\"Failed to successfully load all aggregators. Aborting...\\n\",e.t0),s.ee.abort();case 15:case\"end\":return e.stop()}}),e,null,[[2,11]])})),function(){var t=this,n=arguments;return new Promise((function(r,i){var o=e.apply(t,n);function a(t){d(o,r,i,a,c,\"next\",t)}function c(t){d(o,r,i,a,c,\"throw\",t)}a(void 0)}))})()}var h=i(8539),v=i.n(h),g=i(1424),m=i(1220),y=i(441),w=i(7361),b=i(357),O=i(6959);var x=i(1476),E=i(5955),j={accountID:void 0,trustKey:void 0,agentID:void 0,licenseKey:void 0,applicationID:void 0,xpid:void 0},_={};function P(t){if(!t)throw new Error(\"All loader-config objects require an agent identifier!\");if(!_[t])throw new Error(\"LoaderConfig for \"+t+\" was never set\");return _[t]}var T=!1;var S=i(6972),L=i(1793),C=i(5023);function I(t,e){return I=Object.setPrototypeOf?Object.setPrototypeOf.bind():function(t,e){return t.__proto__=e,t},I(t,e)}var A,R,k,H=function(t){var e,n;function r(e){var n;return n=t.call(this,e)||this,(0,L.v)(e),(0,S.B)(e,\"firstbyte\",(0,O.yf)()),c((function(){return n.measureWindowLoaded()})),u((function(){return n.measureDomContentLoaded()})),n}n=t,(e=r).prototype=Object.create(n.prototype),e.prototype.constructor=e,I(e,n);var i=r.prototype;return i.measureWindowLoaded=function(){var t=(0,O.zO)();(0,S.B)(this.agentIdentifier,\"onload\",t+(0,O.os)()),(0,w.p)(\"timing\",[\"load\",t],void 0,void 0,this.ee)},i.measureDomContentLoaded=function(){(0,S.B)(this.agentIdentifier,\"domContent\",(0,O.zO)()+(0,O.os)())},r}(C.W);void 0!==document.hidden?(A=\"hidden\",R=\"visibilitychange\",k=\"visibilityState\"):void 0!==document.msHidden?(A=\"msHidden\",R=\"msvisibilitychange\"):void 0!==document.webkitHidden&&(A=\"webkitHidden\",R=\"webkitvisibilitychange\",k=\"webkitVisibilityState\");var z=i(2085);function N(t,e){return N=Object.setPrototypeOf?Object.setPrototypeOf.bind():function(t,e){return t.__proto__=e,t},N(t,e)}var M=function(t){var n,r;function i(n){var r,i;if((r=t.call(this,n)||this).pageHiddenTime=\"hidden\"===document.visibilityState?-1:1/0,r.performanceObserver,r.lcpPerformanceObserver,r.clsPerformanceObserver,r.fiRecorded=!1,!r.isEnabled())return function(t){if(void 0===t)throw new ReferenceError(\"this hasn't been initialised - super() hasn't been called\");return t}(r);if(\"PerformanceObserver\"in window&&\"function\"==typeof window.PerformanceObserver){r.performanceObserver=new PerformanceObserver((function(){var t;return(t=r).perfObserver.apply(t,arguments)}));try{r.performanceObserver.observe({entryTypes:[\"paint\"]})}catch(t){}r.lcpPerformanceObserver=new PerformanceObserver((function(){var t;return(t=r).lcpObserver.apply(t,arguments)}));try{r.lcpPerformanceObserver.observe({entryTypes:[\"largest-contentful-paint\"]})}catch(t){}r.clsPerformanceObserver=new PerformanceObserver((function(){var t;return(t=r).clsObserver.apply(t,arguments)}));try{r.clsPerformanceObserver.observe({type:\"layout-shift\",buffered:!0})}catch(t){}}if(\"addEventListener\"in document){r.fiRecorded=!1;[\"click\",\"keydown\",\"mousedown\",\"pointerdown\",\"touchstart\"].forEach((function(t){document.addEventListener(t,(function(){var t;return(t=r).captureInteraction.apply(t,arguments)}),(0,e.m)(!1))}))}return i=function(){var t;return(t=r).captureVisibilityChange.apply(t,arguments)},\"addEventListener\"in document&&R&&document.addEventListener(R,(function(){k&&document[k]?i(document[k]):document[A]?i(\"hidden\"):i(\"visible\")}),(0,e.m)(!1)),r}r=t,(n=i).prototype=Object.create(r.prototype),n.prototype.constructor=n,N(n,r);var o=i.prototype;return o.isEnabled=function(){return!1!==(0,x.Mt)(this.agentIdentifier,\"page_view_timing.enabled\")},o.perfObserver=function(t,e){var n=this;t.getEntries().forEach((function(t){\"first-paint\"===t.name?(0,w.p)(\"timing\",[\"fp\",Math.floor(t.startTime)],void 0,void 0,n.ee):\"first-contentful-paint\"===t.name&&(0,w.p)(\"timing\",[\"fcp\",Math.floor(t.startTime)],void 0,void 0,n.ee)}))},o.lcpObserver=function(t,e){var n=t.getEntries();if(n.length>0){var r=n[n.length-1];if(this.pageHiddenTime<r.startTime)return;var i=[r],o=this.addConnectionAttributes({});o&&i.push(o),(0,w.p)(\"lcp\",i,void 0,void 0,this.ee)}},o.clsObserver=function(t){var e=this;t.getEntries().forEach((function(t){t.hadRecentInput||(0,w.p)(\"cls\",[t],void 0,void 0,e.ee)}))},o.addConnectionAttributes=function(t){var e=navigator.connection||navigator.mozConnection||navigator.webkitConnection;if(e)return e.type&&(t[\"net-type\"]=e.type),e.effectiveType&&(t[\"net-etype\"]=e.effectiveType),e.rtt&&(t[\"net-rtt\"]=e.rtt),e.downlink&&(t[\"net-dlink\"]=e.downlink),t},o.captureInteraction=function(t){if(t instanceof z.Y.EV&&!this.fiRecorded){var e=Math.round(t.timeStamp),n={type:t.type};this.addConnectionAttributes(n),e<=(0,O.zO)()?n.fid=(0,O.zO)()-e:e>(0,O.os)()&&e<=Date.now()?(e-=(0,O.os)(),n.fid=(0,O.zO)()-e):e=(0,O.zO)(),this.fiRecorded=!0,(0,w.p)(\"timing\",[\"fi\",e,n],void 0,void 0,this.ee)}},o.captureVisibilityChange=function(t){\"hidden\"===t&&(this.pageHiddenTime=(0,O.zO)(),(0,w.p)(\"pageHide\",[this.pageHiddenTime],void 0,void 0,this.ee))},i}(C.W),q=i(3350),D=\"React\",B=\"Angular\",F=\"AngularJS\",G=\"Backbone\",U=\"Ember\",X=\"Vue\",Z=\"Meteor\",V=\"Zepto\",W=\"Jquery\";function Y(){var t=[];try{(function(){try{if(window.React||window.ReactDOM||window.ReactRedux)return!0;if(document.querySelector(\"[data-reactroot], [data-reactid]\"))return!0;for(var t=document.querySelectorAll(\"body > div\"),e=0;e<t.length;e++)if(Object.keys(t[e]).indexOf(\"_reactRootContainer\")>=0)return!0;return!1}catch(t){return!1}})()&&t.push(D),function(){try{return!!window.angular||(!!document.querySelector(\".ng-binding, [ng-app], [data-ng-app], [ng-controller], [data-ng-controller], [ng-repeat], [data-ng-repeat]\")||!!document.querySelector('script[src*=\"angular.js\"], script[src*=\"angular.min.js\"]'))}catch(t){return!1}}()&&t.push(F),function(){try{return!!(window.hasOwnProperty(\"ng\")&&window.ng.hasOwnProperty(\"coreTokens\")&&window.ng.coreTokens.hasOwnProperty(\"NgZone\"))||!!document.querySelectorAll(\"[ng-version]\").length}catch(t){return!1}}()&&t.push(B),window.Backbone&&t.push(G),window.Ember&&t.push(U),window.Vue&&t.push(X),window.Meteor&&t.push(Z),window.Zepto&&t.push(V),window.jQuery&&t.push(W)}catch(t){}return t}var Q=i(7299),K=i(603),J=i(158);function $(t,e){return $=Object.setPrototypeOf?Object.setPrototypeOf.bind():function(t,e){return t.__proto__=e,t},$(t,e)}var tt=function(t){var e,n;function r(e){var n;return(n=t.call(this,e)||this).singleChecks(),(0,q.X)(\"record-supportability\",(function(){var t;return(t=n).recordSupportability.apply(t,arguments)}),void 0,n.ee),(0,q.X)(\"record-custom\",(function(){var t;return(t=n).recordCustom.apply(t,arguments)}),void 0,n.ee),n}n=t,(e=r).prototype=Object.create(n.prototype),e.prototype.constructor=e,$(e,n);var i=r.prototype;return i.recordSupportability=function(t,e){var n=[\"sm\",t,{name:t},e];return(0,w.p)(\"storeMetric\",n,null,void 0,this.ee),n},i.recordCustom=function(t,e){var n=[\"cm\",t,{name:t},e];return(0,w.p)(\"storeEventMetrics\",n,null,void 0,this.ee),n},i.singleChecks=function(){var t=this;this.recordSupportability(\"Generic/Version/\"+J.q+\"/Detected\"),u((function(){Y().forEach((function(e){t.recordSupportability(\"Framework/\"+e+\"/Detected\")}))})),Q.T.isFileProtocol()&&(this.recordSupportability(\"Generic/FileProtocol/Detected\"),Q.T.supportabilityMetricSent=!0);var e=(0,K.$c)(this.agentIdentifier);e.length>0&&this.recordSupportability(\"Generic/Obfuscate/Detected\"),e.length>0&&!(0,K.Ng)(e)&&this.recordSupportability(\"Generic/Obfuscate/Invalid\")},r}(C.W),et=i(4217),nt=i(8118),rt=i.n(nt),it=window,ot=\"fetch-\",at=\"fetch-body-\",ct=[\"arrayBuffer\",\"blob\",\"json\",\"text\",\"formData\"],ut=it.Request,st=it.Response,ft=\"prototype\",dt=\"nr@context\",lt={};function pt(t){var e=function(t){return(t||s.ee).get(\"fetch\")}(t);if(!(ut&&st&&window.fetch))return e;if(lt[e.debugId])return e;function n(t,n,r){var i=t[n];\"function\"==typeof i&&(t[n]=function(){var t,n=rt()(arguments),o={};e.emit(r+\"before-start\",[n],o),o[dt]&&o[dt].dt&&(t=o[dt].dt);var a=i.apply(this,n);return e.emit(r+\"start\",[n,t],a),a.then((function(t){return e.emit(r+\"end\",[null,t],a),t}),(function(t){throw e.emit(r+\"end\",[t],a),t}))})}return lt[e.debugId]=!0,(0,b.D)(ct,(function(t,e){n(ut[ft],e,at),n(st[ft],e,at)})),n(it,\"fetch\",ot),e.on(\"fetch-end\",(function(t,n){var r=this;if(n){var i=n.headers.get(\"content-length\");null!==i&&(r.rxSize=i),e.emit(\"fetch-done\",[null,n],r)}else e.emit(\"fetch-done\",[t],r)})),e}var ht=\"nr@original\",vt=Object.prototype.hasOwnProperty,gt=!1;function mt(t,e){return t||(t=s.ee),n.inPlace=function(t,e,r,i,o){r||(r=\"\");var a,c,u,s=\"-\"===r.charAt(0);for(u=0;u<e.length;u++)c=e[u],bt(a=t[c])||(t[c]=n(a,s?c+r:r,i,c,o))},n.flag=ht,n;function n(e,n,i,o,a){return bt(e)?e:(n||(n=\"\"),c[ht]=e,wt(e,c,t),c);function c(){var c,u,s,f;try{u=this,c=rt()(arguments),s=\"function\"==typeof i?i(c,u):i||{}}catch(e){yt([e,\"\",[c,u,o],s],t)}r(n+\"start\",[c,u,o],s,a);try{return f=e.apply(u,c)}catch(t){throw r(n+\"err\",[c,u,t],s,a),t}finally{r(n+\"end\",[c,u,f],s,a)}}}function r(n,r,i,o){if(!gt||e){var a=gt;gt=!0;try{t.emit(n,r,i,e,o)}catch(e){yt([e,n,r,i],t)}gt=a}}}function yt(t,e){e||(e=s.ee);try{e.emit(\"internal-error\",t)}catch(t){}}function wt(t,e,n){if(Object.defineProperty&&Object.keys)try{return Object.keys(t).forEach((function(n){Object.defineProperty(e,n,{get:function(){return t[n]},set:function(e){return t[n]=e,e}})})),e}catch(t){yt([t],n)}for(var r in t)vt.call(t,r)&&(e[r]=t[r]);return e}function bt(t){return!(t&&t instanceof Function&&t.apply&&!t[ht])}var Ot={};function xt(t){var e=function(t){return(t||s.ee).get(\"timer\")}(t);if(Ot[e.debugId])return e;Ot[e.debugId]=!0;var n=mt(e),r=\"setTimeout\",i=\"setInterval\",o=\"clearTimeout\",a=\"-start\";return n.inPlace(window,[r,\"setImmediate\"],\"setTimeout-\"),n.inPlace(window,[i],\"setInterval-\"),n.inPlace(window,[o,\"clearImmediate\"],\"clearTimeout-\"),e.on(i+a,(function(t,e,r){t[0]=n(t[0],\"fn-\",null,r)})),e.on(r+a,(function(t,e,r){this.method=r,this.timerDuration=isNaN(t[1])?0:+t[1],t[0]=n(t[0],\"fn-\",this,r)})),e}var Et={};function jt(t){var e=function(t){return(t||s.ee).get(\"raf\")}(t);if(Et[e.debugId])return e;Et[e.debugId]=!0;var n=mt(e);return n.inPlace(window,[\"requestAnimationFrame\",\"mozRequestAnimationFrame\",\"webkitRequestAnimationFrame\",\"msRequestAnimationFrame\"],\"raf-\"),e.on(\"raf-start\",(function(t){t[0]=n(t[0],\"fn-\")})),e}var _t={};function Pt(t){var e=function(t){return(t||s.ee).get(\"history\")}(t);if(_t[e.debugId])return e;_t[e.debugId]=!0;var n=mt(e),r=window.history&&window.history.constructor&&window.history.constructor.prototype,i=window.history;return r&&r.pushState&&r.replaceState&&(i=r),n.inPlace(i,[\"pushState\",\"replaceState\"],\"-\"),e}var Tt={};function St(t){var e=function(t){return(t||s.ee).get(\"events\")}(t);if(Tt[e.debugId])return e;Tt[e.debugId]=!0;var n=mt(e,!0),r=XMLHttpRequest,i=\"addEventListener\",o=\"removeEventListener\";function a(t){for(var e=t;e&&!e.hasOwnProperty(i);)e=Object.getPrototypeOf(e);e&&c(e)}function c(t){n.inPlace(t,[i,o],\"-\",u)}function u(t,e){return t[1]}return\"getPrototypeOf\"in Object?(a(document),a(window),a(r.prototype)):r.prototype.hasOwnProperty(i)&&(c(window),c(r.prototype)),e.on(\"addEventListener-start\",(function(t,e){var r=t[1];if(null!==r&&(\"function\"==typeof r||\"object\"==typeof r)){var i=(0,et.X)(r,\"nr@wrapped\",(function(){var t={object:function(){if(\"function\"!=typeof r.handleEvent)return;return r.handleEvent.apply(r,arguments)},function:r}[typeof r];return t?n(t,\"fn-\",null,t.name||\"anonymous\"):r}));this.wrapped=t[1]=i}})),e.on(\"removeEventListener-start\",(function(t){t[1]=this.wrapped||t[1]})),e}var Lt={};function Ct(t){var n=t||s.ee,r=function(t){return(t||s.ee).get(\"xhr\")}(n);if(Lt[r.debugId])return r;Lt[r.debugId]=!0,St(n);var i=mt(r),o=z.Y.XHR,a=z.Y.MO,c=z.Y.PR,u=z.Y.SI,f=\"readystatechange\",d=[\"onload\",\"onerror\",\"onabort\",\"onloadstart\",\"onloadend\",\"onprogress\",\"ontimeout\"],l=[],p=window.XMLHttpRequest.listeners,h=window.XMLHttpRequest=function(t){var n=new o(t);function i(){try{r.emit(\"new-xhr\",[n],n),n.addEventListener(f,g,(0,e.m)(!1))}catch(t){console.error(t);try{r.emit(\"internal-error\",[t])}catch(t){}}}return this.listeners=p?[].concat(p,[i]):[i],this.listeners.forEach((function(t){return t()})),n};function v(t,e){i.inPlace(e,[\"onreadystatechange\"],\"fn-\",O)}function g(){var t=this,e=r.context(t);t.readyState>3&&!e.resolved&&(e.resolved=!0,r.emit(\"xhr-resolved\",[],t)),i.inPlace(t,d,\"fn-\",O)}if(function(t,e){for(var n in t)e[n]=t[n]}(o,h),h.prototype=o.prototype,i.inPlace(h.prototype,[\"open\",\"send\"],\"-xhr-\",O),r.on(\"send-xhr-start\",(function(t,e){v(t,e),function(t){l.push(t),a&&(m?m.then(b):u?u(b):(y=-y,w.data=y))}(e)})),r.on(\"open-xhr-start\",v),a){var m=c&&c.resolve();if(!u&&!c){var y=1,w=document.createTextNode(y);new a(b).observe(w,{characterData:!0})}}else n.on(\"fn-end\",(function(t){t[0]&&t[0].type===f||b()}));function b(){for(var t=0;t<l.length;t++)v(0,l[t]);l.length&&(l=[])}function O(t,e){return e}return r}function It(t){return St(t)}function At(t){return jt(t)}function Rt(t){return xt(t)}function kt(t){return Ct(t)}function Ht(t,e){return Ht=Object.setPrototypeOf?Object.setPrototypeOf.bind():function(t,e){return t.__proto__=e,t},Ht(t,e)}var zt=\"nr@seenError\",Nt=function(t){var e,n;function r(e){var n;(n=t.call(this,e)||this).skipNext=0,n.handleErrors=!1,n.origOnerror=window.onerror;var r=function(t){if(void 0===t)throw new ReferenceError(\"this hasn't been initialised - super() hasn't been called\");return t}(n),i=(0,m.O)(n.agentIdentifier);i.features.err=!0,r.ee.on(\"fn-start\",(function(t,e,n){r.handleErrors&&(r.skipNext+=1)})),r.ee.on(\"fn-err\",(function(t,e,n){r.handleErrors&&!n[zt]&&((0,et.X)(n,zt,(function(){return!0})),this.thrown=!0,qt(n,void 0,r.ee))})),r.ee.on(\"fn-end\",(function(){r.handleErrors&&!this.thrown&&r.skipNext>0&&(r.skipNext-=1)})),r.ee.on(\"internal-error\",(function(t){(0,w.p)(\"ierr\",[t,(0,O.zO)(),!0],void 0,void 0,r.ee)}));var o=window.onerror;window.onerror=function(){var t;return o&&o.apply(void 0,arguments),(t=n).onerrorHandler.apply(t,arguments),!1};try{window.addEventListener(\"unhandledrejection\",(function(t){var e=new Error(\"\"+t.reason);(0,w.p)(\"err\",[e,(0,O.zO)(),!1,{unhandledPromiseRejection:1}],void 0,void 0,n.ee)}))}catch(t){}try{throw new Error}catch(t){\"stack\"in t&&(Rt(n.ee),At(n.ee),\"addEventListener\"in window&&It(n.ee),i.xhrWrappable&&kt(n.ee),r.handleErrors=!0)}return n}return n=t,(e=r).prototype=Object.create(n.prototype),e.prototype.constructor=e,Ht(e,n),r.prototype.onerrorHandler=function(t,e,n,r,i){try{this.skipNext?this.skipNext-=1:qt(i||new Mt(t,e,n),!0,this.ee)}catch(t){try{(0,w.p)(\"ierr\",[t,(0,O.zO)(),!0],void 0,void 0,this.ee)}catch(t){}}return\"function\"==typeof this.origOnerror&&this.origOnerror.apply(this,rt()(arguments))},r}(C.W);function Mt(t,e,n){this.message=t||\"Uncaught error with no additional information\",this.sourceURL=e,this.line=n}function qt(t,e,n){var r=e?null:(0,O.zO)();(0,w.p)(\"err\",[t,r],void 0,void 0,n)}var Dt=1;function Bt(t){var e=typeof t;return!t||\"object\"!==e&&\"function\"!==e?-1:t===window?0:(0,et.X)(t,\"nr@id\",(function(){return Dt++}))}var Ft=i(5973);function Gt(t){if(\"string\"==typeof t&&t.length)return t.length;if(\"object\"==typeof t){if(\"undefined\"!=typeof ArrayBuffer&&t instanceof ArrayBuffer&&t.byteLength)return t.byteLength;if(\"undefined\"!=typeof Blob&&t instanceof Blob&&t.size)return t.size;if(!(\"undefined\"!=typeof FormData&&t instanceof FormData))try{return JSON.stringify(t).length}catch(t){return}}}var Ut=i(7001),Xt=i(4168),Zt=function(){function t(t){this.agentIdentifier=t,this.generateTracePayload=this.generateTracePayload.bind(this),this.shouldGenerateTrace=this.shouldGenerateTrace.bind(this)}var e=t.prototype;return e.generateTracePayload=function(t){if(!this.shouldGenerateTrace(t))return null;var e=P(this.agentIdentifier);if(!e)return null;var n=(e.accountID||\"\").toString()||null,r=(e.agentID||\"\").toString()||null,i=(e.trustKey||\"\").toString()||null;if(!n||!r)return null;var o=(0,Xt.M)(),a=(0,Xt.Ht)(),c=Date.now(),u={spanId:o,traceId:a,timestamp:c};return(t.sameOrigin||this.isAllowedOrigin(t)&&this.useTraceContextHeadersForCors())&&(u.traceContextParentHeader=this.generateTraceContextParentHeader(o,a),u.traceContextStateHeader=this.generateTraceContextStateHeader(o,c,n,r,i)),(t.sameOrigin&&!this.excludeNewrelicHeader()||!t.sameOrigin&&this.isAllowedOrigin(t)&&this.useNewrelicHeaderForCors())&&(u.newrelicHeader=this.generateTraceHeader(o,a,c,n,r,i)),u},e.generateTraceContextParentHeader=function(t,e){return\"00-\"+e+\"-\"+t+\"-01\"},e.generateTraceContextStateHeader=function(t,e,n,r,i){return i+\"@nr=0-1-\"+n+\"-\"+r+\"-\"+t+\"----\"+e},e.generateTraceHeader=function(t,e,n,r,i,o){if(!(\"btoa\"in window&&\"function\"==typeof window.btoa))return null;var a={v:[0,1],d:{ty:\"Browser\",ac:r,ap:i,id:t,tr:e,ti:n}};return o&&r!==o&&(a.d.tk=o),btoa(JSON.stringify(a))},e.shouldGenerateTrace=function(t){return this.isDtEnabled()&&this.isAllowedOrigin(t)},e.isAllowedOrigin=function(t){var e=!1,n={};if((0,x.Mt)(this.agentIdentifier,\"distributed_tracing\")&&(n=(0,x.P_)(this.agentIdentifier).distributed_tracing),t.sameOrigin)e=!0;else if(n.allowed_origins instanceof Array)for(var r=0;r<n.allowed_origins.length;r++){var i=(0,Ut.e)(n.allowed_origins[r]);if(t.hostname===i.hostname&&t.protocol===i.protocol&&t.port===i.port){e=!0;break}}return e},e.isDtEnabled=function(){var t=(0,x.Mt)(this.agentIdentifier,\"distributed_tracing\");return!!t&&!!t.enabled},e.excludeNewrelicHeader=function(){var t=(0,x.Mt)(this.agentIdentifier,\"distributed_tracing\");return!!t&&!!t.exclude_newrelic_header},e.useNewrelicHeaderForCors=function(){var t=(0,x.Mt)(this.agentIdentifier,\"distributed_tracing\");return!!t&&!1!==t.cors_use_newrelic_header},e.useTraceContextHeadersForCors=function(){var t=(0,x.Mt)(this.agentIdentifier,\"distributed_tracing\");return!!t&&!!t.cors_use_tracecontext_headers},t}();function Vt(t,e){return Vt=Object.setPrototypeOf?Object.setPrototypeOf.bind():function(t,e){return t.__proto__=e,t},Vt(t,e)}var Wt=[\"load\",\"error\",\"abort\",\"timeout\"],Yt=Wt.length,Qt=z.Y.REQ,Kt=window.XMLHttpRequest,Jt=function(t){var n,r;function i(n){var r;r=t.call(this,n)||this;var i=(0,m.O)(r.agentIdentifier);return!i.xhrWrappable||i.disabled?function(t){if(void 0===t)throw new ReferenceError(\"this hasn't been initialised - super() hasn't been called\");return t}(r):(i.features.xhr=!0,r.dt=new Zt(r.agentIdentifier),r.handler=function(t,e,n,i){return(0,w.p)(t,e,n,i,r.ee)},r.wrappedFetch=function(t,e){return n=t,pt(n);var n}(r.ee),kt(r.ee),function(t,n,r,i){function o(t){var n=this;n.totalCbs=0,n.called=0,n.cbTime=0,n.end=b,n.ended=!1,n.xhrGuids={},n.lastSize=null,n.loadCaptureCalled=!1,n.params=this.params||{},n.metrics=this.metrics||{},t.addEventListener(\"load\",(function(e){E(n,t)}),(0,e.m)(!1)),Ft.I&&(Ft.I>34||Ft.I<10)||t.addEventListener(\"progress\",(function(t){n.lastSize=t.loaded}),(0,e.m)(!1))}function a(t){this.params={method:t[0]},x(this,t[1]),this.metrics={}}function c(e,n){var r=P(t);\"xpid\"in r&&this.sameOrigin&&n.setRequestHeader(\"X-NewRelic-ID\",r.xpid);var o=i.generateTracePayload(this.parsedOrigin);if(o){var a=!1;o.newrelicHeader&&(n.setRequestHeader(\"newrelic\",o.newrelicHeader),a=!0),o.traceContextParentHeader&&(n.setRequestHeader(\"traceparent\",o.traceContextParentHeader),o.traceContextStateHeader&&n.setRequestHeader(\"tracestate\",o.traceContextStateHeader),a=!0),a&&(this.dt=o)}}function u(t,r){var i=this.metrics,o=t[0],a=this;if(i&&o){var c=Gt(o);c&&(i.txSize=c)}this.startTime=(0,O.zO)(),this.listener=function(t){try{\"abort\"!==t.type||a.loadCaptureCalled||(a.params.aborted=!0),(\"load\"!==t.type||a.called===a.totalCbs&&(a.onloadCalled||\"function\"!=typeof r.onload))&&a.end(r)}catch(t){try{n.emit(\"internal-error\",[t])}catch(t){}}};for(var u=0;u<Yt;u++)r.addEventListener(Wt[u],this.listener,(0,e.m)(!1))}function s(t,e,n){this.cbTime+=t,e?this.onloadCalled=!0:this.called+=1,this.called!==this.totalCbs||!this.onloadCalled&&\"function\"==typeof n.onload||this.end(n)}function f(t,e){var n=\"\"+Bt(t)+!!e;this.xhrGuids&&!this.xhrGuids[n]&&(this.xhrGuids[n]=!0,this.totalCbs+=1)}function d(t,e){var n=\"\"+Bt(t)+!!e;this.xhrGuids&&this.xhrGuids[n]&&(delete this.xhrGuids[n],this.totalCbs-=1)}function l(){this.endTime=(0,O.zO)()}function p(t,e){e instanceof Kt&&\"load\"===t[0]&&n.emit(\"xhr-load-added\",[t[1],t[2]],e)}function h(t,e){e instanceof Kt&&\"load\"===t[0]&&n.emit(\"xhr-load-removed\",[t[1],t[2]],e)}function v(t,e,n){e instanceof Kt&&(\"onload\"===n&&(this.onload=!0),(\"load\"===(t[0]&&t[0].type)||this.onload)&&(this.xhrCbStart=(0,O.zO)()))}function g(t,e){this.xhrCbStart&&n.emit(\"xhr-cb-time\",[(0,O.zO)()-this.xhrCbStart,this.onload,e],e)}function m(t){var e,n=t[1]||{};\"string\"==typeof t[0]?e=t[0]:t[0]&&t[0].url?e=t[0].url:window.URL&&t[0]&&t[0]instanceof URL&&(e=t[0].href),e&&(this.parsedOrigin=(0,Ut.e)(e),this.sameOrigin=this.parsedOrigin.sameOrigin);var r=i.generateTracePayload(this.parsedOrigin);if(r&&(r.newrelicHeader||r.traceContextParentHeader))if(\"string\"==typeof t[0]||window.URL&&t[0]&&t[0]instanceof URL){var o={};for(var a in n)o[a]=n[a];o.headers=new Headers(n.headers||{}),c(o.headers,r)&&(this.dt=r),t.length>1?t[1]=o:t.push(o)}else t[0]&&t[0].headers&&c(t[0].headers,r)&&(this.dt=r);function c(t,e){var n=!1;return e.newrelicHeader&&(t.set(\"newrelic\",e.newrelicHeader),n=!0),e.traceContextParentHeader&&(t.set(\"traceparent\",e.traceContextParentHeader),e.traceContextStateHeader&&t.set(\"tracestate\",e.traceContextStateHeader),n=!0),n}}function y(t,e){this.params={},this.metrics={},this.startTime=(0,O.zO)(),this.dt=e,t.length>=1&&(this.target=t[0]),t.length>=2&&(this.opts=t[1]);var n,r=this.opts||{},i=this.target;\"string\"==typeof i?n=i:\"object\"==typeof i&&i instanceof Qt?n=i.url:window.URL&&\"object\"==typeof i&&i instanceof URL&&(n=i.href),x(this,n);var o=(\"\"+(i&&i instanceof Qt&&i.method||r.method||\"GET\")).toUpperCase();this.params.method=o,this.txSize=Gt(r.body)||0}function w(t,e){var n;this.endTime=(0,O.zO)(),this.params||(this.params={}),this.params.status=e?e.status:0,\"string\"==typeof this.rxSize&&this.rxSize.length>0&&(n=+this.rxSize);var i={txSize:this.txSize,rxSize:n,duration:(0,O.zO)()-this.startTime};r(\"xhr\",[this.params,i,this.startTime,this.endTime,\"fetch\"],this)}function b(t){var e=this.params,n=this.metrics;if(!this.ended){this.ended=!0;for(var i=0;i<Yt;i++)t.removeEventListener(Wt[i],this.listener,!1);e.aborted||(n.duration=(0,O.zO)()-this.startTime,this.loadCaptureCalled||4!==t.readyState?null==e.status&&(e.status=0):E(this,t),n.cbTime=this.cbTime,r(\"xhr\",[e,n,this.startTime,this.endTime,\"xhr\"],this))}}function x(t,e){var n=(0,Ut.e)(e),r=t.params;r.hostname=n.hostname,r.port=n.port,r.protocol=n.protocol,r.host=n.hostname+\":\"+n.port,r.pathname=n.pathname,t.parsedOrigin=n,t.sameOrigin=n.sameOrigin}function E(t,e){t.params.status=e.status;var n=function(t,e){var n=t.responseType;return\"json\"===n&&null!==e?e:\"arraybuffer\"===n||\"blob\"===n||\"json\"===n?Gt(t.response):\"text\"===n||\"\"===n||void 0===n?Gt(t.responseText):void 0}(e,t.lastSize);if(n&&(t.metrics.rxSize=n),t.sameOrigin){var r=e.getResponseHeader(\"X-NewRelic-App-Data\");r&&(t.params.cat=r.split(\", \").pop())}t.loadCaptureCalled=!0}n.on(\"new-xhr\",o),n.on(\"open-xhr-start\",a),n.on(\"open-xhr-end\",c),n.on(\"send-xhr-start\",u),n.on(\"xhr-cb-time\",s),n.on(\"xhr-load-added\",f),n.on(\"xhr-load-removed\",d),n.on(\"xhr-resolved\",l),n.on(\"addEventListener-end\",p),n.on(\"removeEventListener-end\",h),n.on(\"fn-end\",g),n.on(\"fetch-before-start\",m),n.on(\"fetch-start\",y),n.on(\"fn-start\",v),n.on(\"fetch-done\",w)}(r.agentIdentifier,r.ee,r.handler,r.dt),r)}return r=t,(n=i).prototype=Object.create(r.prototype),n.prototype.constructor=n,Vt(n,r),i}(C.W);var $t=i(5098);function te(t,e){return te=Object.setPrototypeOf?Object.setPrototypeOf.bind():function(t,e){return t.__proto__=e,t},te(t,e)}var ee=\"addEventListener\",ne=\"resourcetimingbufferfull\",re=\"bstResource\",ie=\"-start\",oe=\"-end\",ae=\"fn-start\",ce=\"fn-end\",ue=\"bstTimer\",se=\"pushState\",fe=z.Y.EV,de=function(t){var n,r;function i(n){var r;if(r=t.call(this,n)||this,!(window.performance&&window.performance.timing&&window.performance.getEntriesByType))return function(t){if(void 0===t)throw new ReferenceError(\"this hasn't been initialised - super() hasn't been called\");return t}(r);(0,m.O)(r.agentIdentifier).features.stn=!0;var i=r.ee;function o(t){if((0,w.p)(re,[window.performance.getEntriesByType(\"resource\")],void 0,void 0,i),window.performance.clearResourceTimings)try{window.performance.removeEventListener(ne,o,!1)}catch(t){}else try{window.performance.removeEventListener(\"webkitresourcetimingbufferfull\",o,!1)}catch(t){}}return r.timerEE=Rt(r.ee),r.rafEE=At(r.ee),Pt(r.ee),It(r.ee),r.ee.on(ae,(function(t,e){t[0]instanceof fe&&(this.bstStart=(0,O.zO)())})),r.ee.on(ce,(function(t,e){var n=t[0];n instanceof fe&&(0,w.p)(\"bst\",[n,e,this.bstStart,(0,O.zO)()],void 0,void 0,i)})),r.timerEE.on(ae,(function(t,e,n){this.bstStart=(0,O.zO)(),this.bstType=n})),r.timerEE.on(ce,(function(t,e){(0,w.p)(ue,[e,this.bstStart,(0,O.zO)(),this.bstType],void 0,void 0,i)})),r.rafEE.on(ae,(function(){this.bstStart=(0,O.zO)()})),r.rafEE.on(ce,(function(t,e){(0,w.p)(ue,[e,this.bstStart,(0,O.zO)(),\"requestAnimationFrame\"],void 0,void 0,i)})),r.ee.on(se+ie,(function(t){this.time=(0,O.zO)(),this.startPath=location.pathname+location.hash})),r.ee.on(se+oe,(function(t){(0,w.p)(\"bstHist\",[location.pathname+location.hash,this.startPath,this.time],void 0,void 0,i)})),(0,$t.W)()?((0,w.p)(re,[window.performance.getEntriesByType(\"resource\")],void 0,void 0,i),function(){var t=new PerformanceObserver((function(t,e){var n=t.getEntries();(0,w.p)(re,[n],void 0,void 0,i)}));try{t.observe({entryTypes:[\"resource\"]})}catch(t){}}()):ee in window.performance&&(window.performance.clearResourceTimings?window.performance.addEventListener(ne,o,(0,e.m)(!1)):window.performance.addEventListener(\"webkitresourcetimingbufferfull\",o,(0,e.m)(!1))),document.addEventListener(\"scroll\",r.noOp,(0,e.m)(!1)),document.addEventListener(\"keypress\",r.noOp,(0,e.m)(!1)),document.addEventListener(\"click\",r.noOp,(0,e.m)(!1)),r}return r=t,(n=i).prototype=Object.create(r.prototype),n.prototype.constructor=n,te(n,r),i.prototype.noOp=function(t){},i}(C.W);function le(t,e){return le=Object.setPrototypeOf?Object.setPrototypeOf.bind():function(t,e){return t.__proto__=e,t},le(t,e)}var pe=function(t){var e,n;function r(e){var n;return n=t.call(this,e)||this,(0,m.O)(n.agentIdentifier).features.ins=!0,n}return n=t,(e=r).prototype=Object.create(n.prototype),e.prototype.constructor=e,le(e,n),r}(C.W),he=i(847);new Promise((function(e,n){if(T)e(T);else{var r=(0,g.gG)();try{(0,y.L)(t.Z,r.info),(0,x.Dg)(t.Z,r.init),function(t,e){if(!t)throw new Error(\"All loader-config objects require an agent identifier!\");_[t]=new E.I(e,j),(0,g.Qy)(t,_[t],\"loader_config\")}(t.Z,r.loader_config),(0,m.s)(t.Z,{}),function(t){var e=(0,g.fP)(),n=s.ee.get(t),r=n.get(\"tracer\"),i=\"api-\",o=\"api-ixn-\";function a(){}(0,b.D)([\"setErrorHandler\",\"finished\",\"addToTrace\",\"inlineHit\",\"addRelease\"],(function(t,n){e[n]=u(i,n,!0,\"api\")})),e.addPageAction=u(i,\"addPageAction\",!0),e.setCurrentRouteName=u(i,\"routeName\",!0),e.setPageViewName=function(e,n){if(\"string\"==typeof e)return\"/\"!==e.charAt(0)&&(e=\"/\"+e),(0,m.O)(t).customTransaction=(n||\"http://custom.transaction\")+e,u(i,\"setPageViewName\",!0,\"api\")()},e.setCustomAttribute=function(e,n){var r,o=(0,y.C)(t);return(0,y.L)(t,Object.assign({},o,{jsAttributes:Object.assign({},o.jsAttributes,(r={},r[e]=n,r))})),u(i,\"setCustomAttribute\",!0,\"api\")()},e.interaction=function(){return(new a).get()};var c=a.prototype={createTracer:function(t,e){var i={},o=this,a=\"function\"==typeof e;return(0,w.p)(\"api-ixn-tracer\",[(0,O.zO)(),t,i],o,void 0,n),function(){if(r.emit((a?\"\":\"no-\")+\"fn-start\",[(0,O.zO)(),o,a],i),a)try{return e.apply(this,arguments)}catch(t){throw r.emit(\"fn-err\",[arguments,this,\"string\"==typeof t?new Error(t):t],i),t}finally{r.emit(\"fn-end\",[(0,O.zO)()],i)}}}};function u(t,e,r,i){return function(){return(0,w.p)(\"record-supportability\",[\"API/\"+e+\"/called\"],void 0,void 0,n),(0,w.p)(t+e,[(0,O.zO)()].concat(v()(arguments)),r?null:this,i,n),r?void 0:this}}(0,b.D)(\"actionText,setName,setAttribute,save,ignore,onEnd,getContext,end,get\".split(\",\"),(function(t,e){c[e]=u(o,e)})),e.noticeError=function(t,e){\"string\"==typeof t&&(t=new Error(t)),(0,w.p)(\"record-supportability\",[\"API/noticeError/called\"],void 0,void 0,n),(0,w.p)(\"err\",[t,(0,O.zO)(),!1,e],void 0,void 0,n)}}(t.Z),e(T=!0)}catch(t){n(t)}}})).then((function(){var e,n,r,i=(0,he.K)(t.Z);i.page_view_event&&new H(t.Z),i.page_view_timing&&new M(t.Z),i.metrics&&new tt(t.Z),i.jserrors&&new Nt(t.Z),i.ajax&&new Jt(t.Z),i.session_trace&&new de(t.Z),i.page_action&&new pe(t.Z),e=\"pro\",n?setTimeout((function(){return p(e)}),r||1e3):c((function(){return p(e)}))}))}(),o}()}));</script>\n",
       "<meta content=\"width=device-width, initial-scale=1, maximum-scale=1\" name=\"viewport\"/>\n",
       "<title>Frontiers | Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?</title>\n",
       "<link href=\"https://3718aeafc638f96f5bd6-d4a9ca15fc46ba40e71f94dec0aad28c.ssl.cf1.rackcdn.com/favicon_16x16.ico\" rel=\"shortcut icon\" type=\"image/x-icon\"/>\n",
       "<meta content=\"article\" property=\"og:type\"/>\n",
       "<meta content=\"Article\" property=\"frontiers:type\"/>\n",
       "<meta content=\"Frontiers\" name=\"site_name\" property=\"og:site_name\"/>\n",
       "<meta content=\"Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?\" name=\"Title\" property=\"og:title\"/>\n",
       "<meta content=\"Succession is generally well described above-ground in the boreal forest, and several studies have demonstrated the role of interspecific facilitation in tree species establishment. However, the role of mycorrhizal communities for tree establishment and interspecific facilitation, has been little explored. At the ecotone between the mixed boreal forest, dominated by balsam fir and hardwood species, and the boreal forest, dominated by black spruce, several stands of trembling aspen can be found, surrounded by black spruce forest. Regeneration of balsam fir seems to have increased in the recent decades within the boreal forest, and it seems better adapted to grow in trembling aspen stands than in black spruce stands, even when located in similar abiotic conditions. As black spruce stands are also covered by ericaceous shrubs, we investigated if differences in soil fungal communities and ericaceous shrubs abundance could explain the differences observed in balsam fir growth and nutrition. We conducted a study centered on individual saplings to link growth and foliar nutrient concentrations to local vegetation cover, mycorrhization rate, and mycorrhizal communities associated with balsam fir roots. We found that foliar nutrient concentrations and ramification indices (colonization by mycorrhiza per length of root) were greater in trembling aspen stands and were positively correlated to apical and lateral growth of balsam fir saplings. In black spruce stands, the presence of er...\" name=\"Description\" property=\"og:description\"/>\n",
       "<meta content=\"https://www.frontiersin.org/articles/10.3389/fpls.2019.00932/full\" name=\"url\" property=\"og:url\"/>\n",
       "<meta content=\"https://www.frontiersin.org/files/MyHome%20Article%20Library/464100/464100_Thumb_400.jpg\" property=\"og:image\"/>\n",
       "<meta content=\"10\" name=\"citation_volume\"/>\n",
       "<meta content=\"Frontiers in Plant Science\" name=\"citation_journal_title\"/>\n",
       "<meta content=\"Frontiers\" name=\"citation_publisher\"/>\n",
       "<meta content=\"Front. Plant Sci.\" name=\"citation_journal_abbrev\"/>\n",
       "<meta content=\"1664-462X\" name=\"citation_issn\"/>\n",
       "<meta content=\"10.3389/fpls.2019.00932\" name=\"citation_doi\"/>\n",
       "<meta content=\"932\" name=\"citation_pages\"/>\n",
       "<meta content=\"English\" name=\"citation_language\"/>\n",
       "<meta content=\"Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?\" name=\"citation_title\"/>\n",
       "<meta content=\"Abies balsamea; boreal forest; Ectomycorrhiza (EcM); Ericaceous shrubs; facilitation; Picea mariana; Populus tremuloides\" name=\"citation_keywords\"/>\n",
       "<meta content=\"Succession is generally well described above-ground in the boreal forest, and several studies have demonstrated the role of interspecific facilitation in tree species establishment. However the role of mycorrhizal communities for tree establishment and interspecific facilitation, has been little explored. At the ecotone between the mixed boreal forest, dominated by balsam fir and hardwood species, and the boreal forest, dominated by black spruce, several stands of trembling aspen can be found, surrounded by black spruce forest. Regeneration of balsam fir seems to have increased in the recent decades within the boreal forest, and it seems better adapted to grow in trembling aspen stands than in black spruce stands, even when located in similar abiotic conditions. As black spruce stands are also covered by ericaceous shrubs, we investigated if differences in soil fungal communities and ericaceous shrubs abundance could explain the differences observed in balsam fir growth and nutrition. We conducted a study centered on individual saplings to link growth and foliar nutrient concentrations to local vegetation cover, mycorrhization rate and mycorrhizal communities associated with balsam fir roots. We found that foliar nutrient concentrations and ramification indice (colonization by mycorrhiza per length of root) were greater in trembling aspen stands and were positively correlated to apical and lateral growth of balsam fir saplings. In black spruce stands, the presence of ericaceous shrubs near balsam fir saplings affected ectomycorrhizal communities associated with tree roots which in turn negatively affected N foliar concentrations. Our results reveal that fungal communities observed under aspen are drivers of balsam fir early growth and nutrition in boreal forest stands and may facilitate ecotone migration in a context of climate change.\" name=\"citation_abstract\"/>\n",
       "<meta content=\"Succession is generally well described above-ground in the boreal forest, and several studies have demonstrated the role of interspecific facilitation in tree species establishment. However the role of mycorrhizal communities for tree establishment and interspecific facilitation, has been little explored. At the ecotone between the mixed boreal forest, dominated by balsam fir and hardwood species, and the boreal forest, dominated by black spruce, several stands of trembling aspen can be found, surrounded by black spruce forest. Regeneration of balsam fir seems to have increased in the recent decades within the boreal forest, and it seems better adapted to grow in trembling aspen stands than in black spruce stands, even when located in similar abiotic conditions. As black spruce stands are also covered by ericaceous shrubs, we investigated if differences in soil fungal communities and ericaceous shrubs abundance could explain the differences observed in balsam fir growth and nutrition. We conducted a study centered on individual saplings to link growth and foliar nutrient concentrations to local vegetation cover, mycorrhization rate and mycorrhizal communities associated with balsam fir roots. We found that foliar nutrient concentrations and ramification indice (colonization by mycorrhiza per length of root) were greater in trembling aspen stands and were positively correlated to apical and lateral growth of balsam fir saplings. In black spruce stands, the presence of ericaceous shrubs near balsam fir saplings affected ectomycorrhizal communities associated with tree roots which in turn negatively affected N foliar concentrations. Our results reveal that fungal communities observed under aspen are drivers of balsam fir early growth and nutrition in boreal forest stands and may facilitate ecotone migration in a context of climate change.\" name=\"description\"/>\n",
       "<meta content=\"2019/07/03\" name=\"citation_online_date\"/>\n",
       "<meta content=\"2019/07/18\" name=\"citation_publication_date\"/>\n",
       "<meta content=\"Nagati, Mélissande\" name=\"citation_author\"/>\n",
       "<meta content=\"UQAT-UQAM Industrial Chair in Sustainable Forest Management, Forest Research Institute, University of Québec in Abitibi-Témiscamingue, Canada\" name=\"citation_author_institution\"/>\n",
       "<meta content=\"UMR5174, Laboratory Evolution and Biological Diversity, Centre National de la Recherche Scientifique – IRD, Université Paul Sabatier, France\" name=\"citation_author_institution\"/>\n",
       "<meta content=\"Roy, Mélanie\" name=\"citation_author\"/>\n",
       "<meta content=\"UMR5174, Laboratory Evolution and Biological Diversity, Centre National de la Recherche Scientifique – IRD, Université Paul Sabatier, France\" name=\"citation_author_institution\"/>\n",
       "<meta content=\"Desrochers, Annie\" name=\"citation_author\"/>\n",
       "<meta content=\"UQAT-UQAM Industrial Chair in Sustainable Forest Management, Forest Research Institute, University of Québec in Abitibi-Témiscamingue, Canada\" name=\"citation_author_institution\"/>\n",
       "<meta content=\"Manzi, Sophie\" name=\"citation_author\"/>\n",
       "<meta content=\"UMR5174, Laboratory Evolution and Biological Diversity, Centre National de la Recherche Scientifique – IRD, Université Paul Sabatier, France\" name=\"citation_author_institution\"/>\n",
       "<meta content=\"Bergeron, Yves\" name=\"citation_author\"/>\n",
       "<meta content=\"UQAT-UQAM Industrial Chair in Sustainable Forest Management, Forest Research Institute, University of Québec in Abitibi-Témiscamingue, Canada\" name=\"citation_author_institution\"/>\n",
       "<meta content=\"Gardes, Monique\" name=\"citation_author\"/>\n",
       "<meta content=\"UMR5174, Laboratory Evolution and Biological Diversity, Centre National de la Recherche Scientifique – IRD, Université Paul Sabatier, France\" name=\"citation_author_institution\"/>\n",
       "<meta content=\"Abies balsamea, boreal forest, Ectomycorrhiza (EcM), Ericaceous shrubs, facilitation, Picea mariana, Populus tremuloides\" name=\"Keywords\"/>\n",
       "<meta content=\"doi:10.3389/fpls.2019.00932\" name=\"dc.identifier\"/>\n",
       "<!--CrossMark widget-->\n",
       "<script type=\"text/javascript\">\n",
       "\n",
       "        var CurrentIBarMenu = 'bysubjects';\n",
       "        var CurrentPageCode = 'ARTICLE_PAGE_NEW';\n",
       "\n",
       "        var FRConfiguration = (function () {\n",
       "            return {\n",
       "                Environment: 'Live',\n",
       "                SANVirtualPath: 'https://www.frontiersin.org/files/',\n",
       "                SharepointWebsiteUrl: 'https://www.frontiersin.org',\n",
       "                FrontiersJournalUIUrl: 'https://www.frontiersin.org',\n",
       "                FrontiersJournalAPIUrl: 'https://api-journal.frontiersin.org',\n",
       "                FrontiersReviewUIUrl: '',\n",
       "                FrontiersReviewAPIUrl: '',\n",
       "                FrontiersCookie: 'frontiersN',\n",
       "                FrontiersCookieRememberMe: 'frontiersNt',\n",
       "                FrontiersLoginUrl: 'https://www.frontiersin.org/Login.aspx',\n",
       "                FrontiersRegistrationUrl: 'http://www.frontiersin.org/Registration/Register.aspx',\n",
       "                IsCommentVisible: 'True',\n",
       "                IsPreview: 'False',\n",
       "                TenantName: 'Frontiers'\n",
       "            }\n",
       "        })();\n",
       "\n",
       "        var FRLanguage = (function() {\n",
       "            var languageSet = {\"article_accepteddate\":\"Accepted: \",\"article_analyticstooltip\":\"The total view count is updated once a day, so not to worry if you don\\u0027t see immediate results.\",\"article_analyticsviewimpact\":\"View Article Impact\",\"article_bibtex\":\"BibTex\",\"article_copyrighttext\":\"This is an open-access article distributed under the terms of the \\u003ca href=\\\"http://creativecommons.org/licenses/by/4.0/\\\"\\u003eCreative Commons Attribution License (CC BY)\\u003c/a\\u003e. The use, distribution or reproduction in other forums is permitted, provided the original author(s) or licensor are credited and that the original publication in this journal is cited, in accordance with accepted academic practice. No use, distribution or reproduction is permitted which does not comply with these terms.\",\"article_downloadarticle\":\"Download Article\",\"article_downloadpdf\":\"Download PDF\",\"article_downloadprovisionalarticle\":\"Download Provisional Article\",\"article_endnote\":\"EndNote\",\"article_epub\":\"EPUB\",\"article_exportcitation\":\"Export Citation\",\"article_jats\":\"JATS\",\"article_nlm\":\"XML (NLM)\",\"article_pdf\":\"PDF\",\"article_provisionalpdf\":\"Provisional PDF\",\"article_publisheddate\":\"Published online: \",\"article_readfulltext\":\"Read Full Text\",\"article_receiveddate\":\"Received:\",\"article_referencemanager\":\"Reference Manager\",\"article_shareon\":\"SHARE ON\",\"article_simpletextfile\":\"Simple TEXT file\",\"article_supplementaldata\":\"SUPPLEMENTAL DATA\",\"article_tableofcontent\":\"TABLE OF CONTENTS\",\"article_viewenhancedpdf\":\"ReadCube\",\"article_xml\":\"XML\",\"browserwarningtext\":\"\\u003ch2\\u003eWarning!\\u003c/h2\\u003e\\u003cp\\u003eYou are using an \\u003cstrong\\u003eoutdated\\u003c/strong\\u003e browser. This page doesn\\u0027t support Internet Explorer 6, 7 and 8.\\u003cbr /\\u003ePlease \\u003ca class=\\\"blue\\\" href=\\\"http://browsehappy.com/\\\"\\u003eupgrade your browser\\u003c/a\\u003e or \\u003ca class=\\\"blue\\\" href=\\\"http://www.google.com/chromeframe/?redirect=true\\\"\\u003eactivate Google Chrome Frame\\u003c/a\\u003e to improve your experience.\\u003c/p\\u003e\",\"impact_backtoarticle\":\"Back to article\",\"people_also_lookedat\":\"People also looked at\",\"article_analyticstotalviews\":\"total views\"};\n",
       "\n",
       "            return {\n",
       "                value: function(key) {\n",
       "                    if (languageSet[key]) {\n",
       "                        return languageSet[key];\n",
       "                    } else {\n",
       "                        throw new Error('Unable to get the value status from the language set'); // Use Error, not FRError\n",
       "                    }\n",
       "                }\n",
       "            };\n",
       "\n",
       "        })();\n",
       "\n",
       "        var FRJournalDetails = (function() {\n",
       "            return {\n",
       "                JournalType: 'section',\n",
       "                JournalId: '373',\n",
       "                SectionId: '485'\n",
       "            };\n",
       "        })();\n",
       "\n",
       "        var FRArticleDetails = (function() {\n",
       "            return {\n",
       "                ArticleId: '464100',\n",
       "                DisclaimerText: 'All claims expressed in this article are solely those of the authors and do not necessarily represent those of their affiliated organizations, or those of the publisher, the editors and the reviewers. Any product that may be evaluated in this article or claim that may be made by its manufacturer is not guaranteed or endorsed by the publisher.'\n",
       "            };\n",
       "        })();\n",
       "        var FRArticleRecaptchaSettings = (function() {\n",
       "            return {\n",
       "                RecaptchaSiteKey: '6LdG3i0UAAAAAOC4qUh35ubHgJotEHp_STXHgr_v'\n",
       "            };\n",
       "        })();\n",
       "\n",
       "    </script>\n",
       "<link href=\"https://209cd62b0febf2a55d40-715eb384bc6027b876e93ad33d5c5ec3.ssl.cf3.rackcdn.com/font-awesome-4.3.0/css/font-awesome.min.css\" rel=\"stylesheet\"/>\n",
       "<link href=\"https://9d0dd7a648345f19af83-877d2ecaf11b88d5e17327c758e17ef6.ssl.cf2.rackcdn.com/museo-sans-1.0.1/css/museo-sans.css\" rel=\"stylesheet\"/>\n",
       "<link href=\"https://static.frontiersin.org/areas/articles/css/app?v=keEodGDvcIZ2_1y3nDxMv_krWtMWyGsbWbqae4pMq7M1\" rel=\"stylesheet\"/>\n",
       "<style type=\"text/css\"> .journal-plant-science {\n",
       " background: #fff url('https://3718aeafc638f96f5bd6-d4a9ca15fc46ba40e71f94dec0aad28c.ssl.cf1.rackcdn.com/journal-plant-science.gif') no-repeat 110% 0px;\n",
       "}\n",
       "\n",
       "/* Medium Devices, Desktops ---------- */  @media only screen and (max-width : 992px) { {\n",
       " .journal-plant-science {\n",
       "  background: #fff url('https://3718aeafc638f96f5bd6-d4a9ca15fc46ba40e71f94dec0aad28c.ssl.cf1.rackcdn.com/journal-plant-science.gif') no-repeat -50px 0px;\n",
       " }}\n",
       "\n",
       "/* Small Devices, Tablets ------------ */  @media only screen and (max-width : 768px) {\n",
       ".journal-plant-science {\n",
       "  background: #fff url('https://3718aeafc638f96f5bd6-d4a9ca15fc46ba40e71f94dec0aad28c.ssl.cf1.rackcdn.com/journal-plant-science.gif') no-repeat -150px 0px;\n",
       " }}\n",
       "\n",
       "/* Small Devices, Tablets  --------- */ @media only screen and (max-width : 640px) {\n",
       ".journal-plant-science {\n",
       "  background: #fff url('https://3718aeafc638f96f5bd6-d4a9ca15fc46ba40e71f94dec0aad28c.ssl.cf1.rackcdn.com/journal-plant-science.gif') no-repeat -250px 0px;\n",
       " }}\n",
       "/* Extra Small Devices, Phones-----------  */  @media only screen and (max-width : 480px) {\n",
       ".journal-plant-science {\n",
       "  background: #fff url('https://3718aeafc638f96f5bd6-d4a9ca15fc46ba40e71f94dec0aad28c.ssl.cf1.rackcdn.com/journal-plant-science.gif') no-repeat -330px 0px;\n",
       " }}\n",
       " </style>\n",
       "<script id=\"altmetric-embed-js\" src=\"https://d1bxh8uas1mnw7.cloudfront.net/assets/altmetric_badges-4b75721c66722d87e7882ff5b18f12d295f4ac71b1a804b2c06ae36a76542439.js\"></script><style type=\"text/css\">.MathJax_Hover_Frame {border-radius: .25em; -webkit-border-radius: .25em; -moz-border-radius: .25em; -khtml-border-radius: .25em; box-shadow: 0px 0px 15px #83A; -webkit-box-shadow: 0px 0px 15px #83A; -moz-box-shadow: 0px 0px 15px #83A; -khtml-box-shadow: 0px 0px 15px #83A; border: 1px solid #A6D ! important; display: inline-block; position: absolute}\n",
       ".MathJax_Menu_Button .MathJax_Hover_Arrow {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 4px; -webkit-border-radius: 4px; -moz-border-radius: 4px; -khtml-border-radius: 4px; font-family: 'Courier New',Courier; font-size: 9px; color: #F0F0F0}\n",
       ".MathJax_Menu_Button .MathJax_Hover_Arrow span {display: block; background-color: #AAA; border: 1px solid; border-radius: 3px; line-height: 0; padding: 4px}\n",
       ".MathJax_Hover_Arrow:hover {color: white!important; border: 2px solid #CCC!important}\n",
       ".MathJax_Hover_Arrow:hover span {background-color: #CCC!important}\n",
       "</style><style type=\"text/css\">#MathJax_About {position: fixed; left: 50%; width: auto; text-align: center; border: 3px outset; padding: 1em 2em; background-color: #DDDDDD; color: black; cursor: default; font-family: message-box; font-size: 120%; font-style: normal; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; border-radius: 15px; -webkit-border-radius: 15px; -moz-border-radius: 15px; -khtml-border-radius: 15px; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}\n",
       "#MathJax_About.MathJax_MousePost {outline: none}\n",
       ".MathJax_Menu {position: absolute; background-color: white; color: black; width: auto; padding: 2px; border: 1px solid #CCCCCC; margin: 0; cursor: default; font: menu; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}\n",
       ".MathJax_MenuItem {padding: 2px 2em; background: transparent}\n",
       ".MathJax_MenuArrow {position: absolute; right: .5em; padding-top: .25em; color: #666666; font-size: .75em}\n",
       ".MathJax_MenuActive .MathJax_MenuArrow {color: white}\n",
       ".MathJax_MenuArrow.RTL {left: .5em; right: auto}\n",
       ".MathJax_MenuCheck {position: absolute; left: .7em}\n",
       ".MathJax_MenuCheck.RTL {right: .7em; left: auto}\n",
       ".MathJax_MenuRadioCheck {position: absolute; left: 1em}\n",
       ".MathJax_MenuRadioCheck.RTL {right: 1em; left: auto}\n",
       ".MathJax_MenuLabel {padding: 2px 2em 4px 1.33em; font-style: italic}\n",
       ".MathJax_MenuRule {border-top: 1px solid #CCCCCC; margin: 4px 1px 0px}\n",
       ".MathJax_MenuDisabled {color: GrayText}\n",
       ".MathJax_MenuActive {background-color: Highlight; color: HighlightText}\n",
       ".MathJax_MenuDisabled:focus, .MathJax_MenuLabel:focus {background-color: #E8E8E8}\n",
       ".MathJax_ContextMenu:focus {outline: none}\n",
       ".MathJax_ContextMenu .MathJax_MenuItem:focus {outline: none}\n",
       "#MathJax_AboutClose {top: .2em; right: .2em}\n",
       ".MathJax_Menu .MathJax_MenuClose {top: -10px; left: -10px}\n",
       ".MathJax_MenuClose {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; font-family: 'Courier New',Courier; font-size: 24px; color: #F0F0F0}\n",
       ".MathJax_MenuClose span {display: block; background-color: #AAA; border: 1.5px solid; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; line-height: 0; padding: 8px 0 6px}\n",
       ".MathJax_MenuClose:hover {color: white!important; border: 2px solid #CCC!important}\n",
       ".MathJax_MenuClose:hover span {background-color: #CCC!important}\n",
       ".MathJax_MenuClose:hover:focus {outline: none}\n",
       "</style><style type=\"text/css\">.MathJax_Preview .MJXf-math {color: inherit!important}\n",
       "</style><style type=\"text/css\">.MJX_Assistive_MathML {position: absolute!important; top: 0; left: 0; clip: rect(1px, 1px, 1px, 1px); padding: 1px 0 0 0!important; border: 0!important; height: 1px!important; width: 1px!important; overflow: hidden!important; display: block!important; -webkit-touch-callout: none; -webkit-user-select: none; -khtml-user-select: none; -moz-user-select: none; -ms-user-select: none; user-select: none}\n",
       ".MJX_Assistive_MathML.MJX_Assistive_MathML_Block {width: 100%!important}\n",
       "</style><style type=\"text/css\">#MathJax_Zoom {position: absolute; background-color: #F0F0F0; overflow: auto; display: block; z-index: 301; padding: .5em; border: 1px solid black; margin: 0; font-weight: normal; font-style: normal; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; -webkit-box-sizing: content-box; -moz-box-sizing: content-box; box-sizing: content-box; box-shadow: 5px 5px 15px #AAAAAA; -webkit-box-shadow: 5px 5px 15px #AAAAAA; -moz-box-shadow: 5px 5px 15px #AAAAAA; -khtml-box-shadow: 5px 5px 15px #AAAAAA; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}\n",
       "#MathJax_ZoomOverlay {position: absolute; left: 0; top: 0; z-index: 300; display: inline-block; width: 100%; height: 100%; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}\n",
       "#MathJax_ZoomFrame {position: relative; display: inline-block; height: 0; width: 0}\n",
       "#MathJax_ZoomEventTrap {position: absolute; left: 0; top: 0; z-index: 302; display: inline-block; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}\n",
       "</style><style type=\"text/css\">.MathJax_Preview {color: #888}\n",
       "#MathJax_Message {position: fixed; left: 1em; bottom: 1.5em; background-color: #E6E6E6; border: 1px solid #959595; margin: 0px; padding: 2px 8px; z-index: 102; color: black; font-size: 80%; width: auto; white-space: nowrap}\n",
       "#MathJax_MSIE_Frame {position: absolute; top: 0; left: 0; width: 0px; z-index: 101; border: 0px; margin: 0px; padding: 0px}\n",
       ".MathJax_Error {color: #CC0000; font-style: italic}\n",
       "</style><style type=\"text/css\">.MJXp-script {font-size: .8em}\n",
       ".MJXp-right {-webkit-transform-origin: right; -moz-transform-origin: right; -ms-transform-origin: right; -o-transform-origin: right; transform-origin: right}\n",
       ".MJXp-bold {font-weight: bold}\n",
       ".MJXp-italic {font-style: italic}\n",
       ".MJXp-scr {font-family: MathJax_Script,'Times New Roman',Times,STIXGeneral,serif}\n",
       ".MJXp-frak {font-family: MathJax_Fraktur,'Times New Roman',Times,STIXGeneral,serif}\n",
       ".MJXp-sf {font-family: MathJax_SansSerif,'Times New Roman',Times,STIXGeneral,serif}\n",
       ".MJXp-cal {font-family: MathJax_Caligraphic,'Times New Roman',Times,STIXGeneral,serif}\n",
       ".MJXp-mono {font-family: MathJax_Typewriter,'Times New Roman',Times,STIXGeneral,serif}\n",
       ".MJXp-largeop {font-size: 150%}\n",
       ".MJXp-largeop.MJXp-int {vertical-align: -.2em}\n",
       ".MJXp-math {display: inline-block; line-height: 1.2; text-indent: 0; font-family: 'Times New Roman',Times,STIXGeneral,serif; white-space: nowrap; border-collapse: collapse}\n",
       ".MJXp-display {display: block; text-align: center; margin: 1em 0}\n",
       ".MJXp-math span {display: inline-block}\n",
       ".MJXp-box {display: block!important; text-align: center}\n",
       ".MJXp-box:after {content: \" \"}\n",
       ".MJXp-rule {display: block!important; margin-top: .1em}\n",
       ".MJXp-char {display: block!important}\n",
       ".MJXp-mo {margin: 0 .15em}\n",
       ".MJXp-mfrac {margin: 0 .125em; vertical-align: .25em}\n",
       ".MJXp-denom {display: inline-table!important; width: 100%}\n",
       ".MJXp-denom > * {display: table-row!important}\n",
       ".MJXp-surd {vertical-align: top}\n",
       ".MJXp-surd > * {display: block!important}\n",
       ".MJXp-script-box > *  {display: table!important; height: 50%}\n",
       ".MJXp-script-box > * > * {display: table-cell!important; vertical-align: top}\n",
       ".MJXp-script-box > *:last-child > * {vertical-align: bottom}\n",
       ".MJXp-script-box > * > * > * {display: block!important}\n",
       ".MJXp-mphantom {visibility: hidden}\n",
       ".MJXp-munderover {display: inline-table!important}\n",
       ".MJXp-over {display: inline-block!important; text-align: center}\n",
       ".MJXp-over > * {display: block!important}\n",
       ".MJXp-munderover > * {display: table-row!important}\n",
       ".MJXp-mtable {vertical-align: .25em; margin: 0 .125em}\n",
       ".MJXp-mtable > * {display: inline-table!important; vertical-align: middle}\n",
       ".MJXp-mtr {display: table-row!important}\n",
       ".MJXp-mtd {display: table-cell!important; text-align: center; padding: .5em 0 0 .5em}\n",
       ".MJXp-mtr > .MJXp-mtd:first-child {padding-left: 0}\n",
       ".MJXp-mtr:first-child > .MJXp-mtd {padding-top: 0}\n",
       ".MJXp-mlabeledtr {display: table-row!important}\n",
       ".MJXp-mlabeledtr > .MJXp-mtd:first-child {padding-left: 0}\n",
       ".MJXp-mlabeledtr:first-child > .MJXp-mtd {padding-top: 0}\n",
       ".MJXp-merror {background-color: #FFFF88; color: #CC0000; border: 1px solid #CC0000; padding: 1px 3px; font-style: normal; font-size: 90%}\n",
       ".MJXp-scale0 {-webkit-transform: scaleX(.0); -moz-transform: scaleX(.0); -ms-transform: scaleX(.0); -o-transform: scaleX(.0); transform: scaleX(.0)}\n",
       ".MJXp-scale1 {-webkit-transform: scaleX(.1); -moz-transform: scaleX(.1); -ms-transform: scaleX(.1); -o-transform: scaleX(.1); transform: scaleX(.1)}\n",
       ".MJXp-scale2 {-webkit-transform: scaleX(.2); -moz-transform: scaleX(.2); -ms-transform: scaleX(.2); -o-transform: scaleX(.2); transform: scaleX(.2)}\n",
       ".MJXp-scale3 {-webkit-transform: scaleX(.3); -moz-transform: scaleX(.3); -ms-transform: scaleX(.3); -o-transform: scaleX(.3); transform: scaleX(.3)}\n",
       ".MJXp-scale4 {-webkit-transform: scaleX(.4); -moz-transform: scaleX(.4); -ms-transform: scaleX(.4); -o-transform: scaleX(.4); transform: scaleX(.4)}\n",
       ".MJXp-scale5 {-webkit-transform: scaleX(.5); -moz-transform: scaleX(.5); -ms-transform: scaleX(.5); -o-transform: scaleX(.5); transform: scaleX(.5)}\n",
       ".MJXp-scale6 {-webkit-transform: scaleX(.6); -moz-transform: scaleX(.6); -ms-transform: scaleX(.6); -o-transform: scaleX(.6); transform: scaleX(.6)}\n",
       ".MJXp-scale7 {-webkit-transform: scaleX(.7); -moz-transform: scaleX(.7); -ms-transform: scaleX(.7); -o-transform: scaleX(.7); transform: scaleX(.7)}\n",
       ".MJXp-scale8 {-webkit-transform: scaleX(.8); -moz-transform: scaleX(.8); -ms-transform: scaleX(.8); -o-transform: scaleX(.8); transform: scaleX(.8)}\n",
       ".MJXp-scale9 {-webkit-transform: scaleX(.9); -moz-transform: scaleX(.9); -ms-transform: scaleX(.9); -o-transform: scaleX(.9); transform: scaleX(.9)}\n",
       ".MathJax_PHTML .noError {vertical-align: ; font-size: 90%; text-align: left; color: black; padding: 1px 3px; border: 1px solid}\n",
       "</style><link href=\"https://d1bxh8uas1mnw7.cloudfront.net/assets/embed-31802d5163e7112bb932c6b069663a78644b40bdeb2e6ca3a53456853965d98d.css\" id=\"altmetric-embed-css\" rel=\"stylesheet\" type=\"text/css\"/><script type=\"application/ld+json\">{\n",
       "    \"@context\": \"http://schema.org\",\n",
       "    \"@id\": \"https://www.frontiersin.org/articles/10.3389/fpls.2019.00932\",\n",
       "    \"@type\": \"ScholarlyArticle\",\n",
       "    \"headline\": \"Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?\",\n",
       "    \"image\": \"https://www.frontiersin.org/files/myhome article library/464100/464100_Thumb_400.jpg\",\n",
       "    \"author\": [\n",
       "        {\n",
       "            \"@id\": \"https://loop.frontiersin.org/people/639840\",\n",
       "            \"@type\": \"Person\",\n",
       "            \"name\": \"Nagati, Mélissande\",\n",
       "            \"affiliation\": [\n",
       "                {\n",
       "                    \"@type\": \"Organization\",\n",
       "                    \"name\": \"UQAT-UQAM Industrial Chair in Sustainable Forest Management, Forest Research Institute, University of Québec in Abitibi-Témiscamingue\",\n",
       "                    \"location\": {\n",
       "                        \"@type\": \"Place\",\n",
       "                        \"address\": {\n",
       "                            \"@type\": \"PostalAddress\",\n",
       "                            \"addressCountry\": \"Canada\"\n",
       "                        }\n",
       "                    }\n",
       "                },\n",
       "                {\n",
       "                    \"@type\": \"Organization\",\n",
       "                    \"name\": \"UMR5174, Laboratory Evolution and Biological Diversity, Centre National de la Recherche Scientifique – IRD, Université Paul Sabatier\",\n",
       "                    \"location\": {\n",
       "                        \"@type\": \"Place\",\n",
       "                        \"address\": {\n",
       "                            \"@type\": \"PostalAddress\",\n",
       "                            \"addressCountry\": \"France\"\n",
       "                        }\n",
       "                    }\n",
       "                }\n",
       "            ]\n",
       "        },\n",
       "        {\n",
       "            \"@id\": \"https://loop.frontiersin.org/people/399792\",\n",
       "            \"@type\": \"Person\",\n",
       "            \"name\": \"Roy, Mélanie\",\n",
       "            \"affiliation\": [\n",
       "                {\n",
       "                    \"@type\": \"Organization\",\n",
       "                    \"name\": \"UMR5174, Laboratory Evolution and Biological Diversity, Centre National de la Recherche Scientifique – IRD, Université Paul Sabatier\",\n",
       "                    \"location\": {\n",
       "                        \"@type\": \"Place\",\n",
       "                        \"address\": {\n",
       "                            \"@type\": \"PostalAddress\",\n",
       "                            \"addressCountry\": \"France\"\n",
       "                        }\n",
       "                    }\n",
       "                }\n",
       "            ]\n",
       "        },\n",
       "        {\n",
       "            \"@type\": \"Person\",\n",
       "            \"name\": \"Desrochers, Annie\",\n",
       "            \"affiliation\": [\n",
       "                {\n",
       "                    \"@type\": \"Organization\",\n",
       "                    \"name\": \"UQAT-UQAM_Industrial_Chair_in_Sustainable_Forest_Management,_Forest_Research_Institute,_University_of_Québec_in_Abitibi-Témiscamingue\",\n",
       "                    \"location\": {\n",
       "                        \"@type\": \"Place\",\n",
       "                        \"address\": {\n",
       "                            \"@type\": \"PostalAddress\",\n",
       "                            \"addressCountry\": \"Canada\"\n",
       "                        }\n",
       "                    }\n",
       "                }\n",
       "            ]\n",
       "        },\n",
       "        {\n",
       "            \"@type\": \"Person\",\n",
       "            \"name\": \"Manzi, Sophie\",\n",
       "            \"affiliation\": [\n",
       "                {\n",
       "                    \"@type\": \"Organization\",\n",
       "                    \"name\": \"UMR5174,_Laboratory_Evolution_and_Biological_Diversity,_Centre_National_de_la_Recherche_Scientifique_–_IRD,_Université_Paul_Sabatier\",\n",
       "                    \"location\": {\n",
       "                        \"@type\": \"Place\",\n",
       "                        \"address\": {\n",
       "                            \"@type\": \"PostalAddress\",\n",
       "                            \"addressCountry\": \"France\"\n",
       "                        }\n",
       "                    }\n",
       "                }\n",
       "            ]\n",
       "        },\n",
       "        {\n",
       "            \"@id\": \"https://loop.frontiersin.org/people/112075\",\n",
       "            \"@type\": \"Person\",\n",
       "            \"name\": \"Bergeron, Yves\",\n",
       "            \"affiliation\": [\n",
       "                {\n",
       "                    \"@type\": \"Organization\",\n",
       "                    \"name\": \"UQAT-UQAM Industrial Chair in Sustainable Forest Management, Forest Research Institute, University of Québec in Abitibi-Témiscamingue\",\n",
       "                    \"location\": {\n",
       "                        \"@type\": \"Place\",\n",
       "                        \"address\": {\n",
       "                            \"@type\": \"PostalAddress\",\n",
       "                            \"addressCountry\": \"Canada\"\n",
       "                        }\n",
       "                    }\n",
       "                }\n",
       "            ]\n",
       "        },\n",
       "        {\n",
       "            \"@id\": \"https://loop.frontiersin.org/people/766853\",\n",
       "            \"@type\": \"Person\",\n",
       "            \"name\": \"Gardes, Monique\",\n",
       "            \"affiliation\": [\n",
       "                {\n",
       "                    \"@type\": \"Organization\",\n",
       "                    \"name\": \"UMR5174, Laboratory Evolution and Biological Diversity, Centre National de la Recherche Scientifique – IRD, Université Paul Sabatier\",\n",
       "                    \"location\": {\n",
       "                        \"@type\": \"Place\",\n",
       "                        \"address\": {\n",
       "                            \"@type\": \"PostalAddress\",\n",
       "                            \"addressCountry\": \"France\"\n",
       "                        }\n",
       "                    }\n",
       "                }\n",
       "            ]\n",
       "        }\n",
       "    ],\n",
       "    \"isPartOf\": [\n",
       "        {\n",
       "            \"@id\": \"https://www.frontiersin.org/journals/plant-science\",\n",
       "            \"@type\": \"Periodical\"\n",
       "        },\n",
       "        {\n",
       "            \"@id\": \"https://www.frontiersin.org/journals/plant-science/sections/plant-pathogen-interactions\",\n",
       "            \"@type\": \"PublicationIssue\"\n",
       "        },\n",
       "        {\n",
       "            \"@id\": \"https://www.frontiersin.org/research-topics/8849/importance-of-root-symbiomes-for-plant-nutrition-new-insights-perspectives-and-future-challenges\",\n",
       "            \"@type\": \"Collection\"\n",
       "        }\n",
       "    ],\n",
       "    \"datePublished\": \"2019-07-18\",\n",
       "    \"publisher\": {\n",
       "        \"@id\": \"https://www.frontiersin.org/#organization\",\n",
       "        \"@type\": \"NewsMediaOrganization\",\n",
       "        \"name\": \"Frontiers Media S.A.\",\n",
       "        \"logo\": {\n",
       "            \"@type\": \"ImageObject\",\n",
       "            \"url\": \"https://www.frontiersin.org/Areas/Home/Content/Images/logo-home.png\"\n",
       "        }\n",
       "    },\n",
       "    \"name\": \"Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?\",\n",
       "    \"keywords\": \"Abies balsamea,boreal forest,Ectomycorrhiza (EcM),Ericaceous shrubs,facilitation,Picea mariana,Populus tremuloides\",\n",
       "    \"articleSection\": \"Plant Pathogen Interactions\",\n",
       "    \"interactionStatistic\": [\n",
       "        {\n",
       "            \"@type\": \"InteractionCounter\",\n",
       "            \"interactionType\": \"http://schema.org/ReadAction\",\n",
       "            \"userInteractionCount\": 4419\n",
       "        },\n",
       "        {\n",
       "            \"@type\": \"InteractionCounter\",\n",
       "            \"interactionType\": \"http://schema.org/DownloadAction\",\n",
       "            \"userInteractionCount\": 811\n",
       "        }\n",
       "    ],\n",
       "    \"license\": \"CC BY 4.0\",\n",
       "    \"sameAs\": \"https://doi.org/10.3389/fpls.2019.00932\"\n",
       "}</script><script async=\"\" src=\"https://cdn.cookielaw.org/scripttemplates/6.19.0/otBannerSdk.js\" type=\"text/javascript\"></script><meta content=\"A751Xsk4ZW3DVQ8WZng2Dk5s3YzAyqncTzgv+VaE6wavgTY0QHkDvUTET1o7HanhuJO8lgv1Vvc88Ij78W1FIAAAAAB7eyJvcmlnaW4iOiJodHRwczovL3d3dy5nb29nbGV0YWdtYW5hZ2VyLmNvbTo0NDMiLCJmZWF0dXJlIjoiUHJpdmFjeVNhbmRib3hBZHNBUElzIiwiZXhwaXJ5IjoxNjgwNjUyNzk5LCJpc1RoaXJkUGFydHkiOnRydWV9\" http-equiv=\"origin-trial\"/><script async=\"\" src=\"https://googleads.g.doubleclick.net/pagead/viewthroughconversion/AW-804005817/?random=1671056251959&amp;cv=11&amp;fst=1671056251959&amp;bg=ffffff&amp;guid=ON&amp;async=1&amp;gtm=2wgbu0&amp;u_w=1920&amp;u_h=1080&amp;hn=www.googleadservices.com&amp;frm=0&amp;url=https%3A%2F%2Fwww.frontiersin.org%2Farticles%2F10.3389%2Ffpls.2019.00932%2Ffull&amp;tiba=Frontiers%20%7C%20Facilitation%20of%20Balsam%20Fir%20by%20Trembling%20Aspen%20in%20the%20Boreal%20Forest%3A%20Do%20Ectomycorrhizal%20Communities%20Matter%3F&amp;auid=212888393.1671056251&amp;uaa=x86&amp;uab=64&amp;uafvl=Google%2520Chrome%3B107.0.5304.107%7CChromium%3B107.0.5304.107%7CNot%253DA%253FBrand%3B24.0.0.0&amp;uap=Windows&amp;uapv=10.0.0&amp;uaw=0&amp;rfmt=3&amp;fmt=4\" type=\"text/javascript\"></script><style id=\"onetrust-style\">#onetrust-banner-sdk{-ms-text-size-adjust:100%;-webkit-text-size-adjust:100%}#onetrust-banner-sdk .onetrust-vendors-list-handler{cursor:pointer;color:#1f96db;font-size:inherit;font-weight:bold;text-decoration:none;margin-left:5px}#onetrust-banner-sdk .onetrust-vendors-list-handler:hover{color:#1f96db}#onetrust-banner-sdk:focus{outline:2px solid #000;outline-offset:-2px}#onetrust-banner-sdk a:focus{outline:2px solid #000}#onetrust-banner-sdk .ot-close-icon,#onetrust-pc-sdk .ot-close-icon,#ot-sync-ntfy .ot-close-icon{background-image:url(\"data:image/svg+xml;base64,PHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IiB3aWR0aD0iMzQ4LjMzM3B4IiBoZWlnaHQ9IjM0OC4zMzNweCIgdmlld0JveD0iMCAwIDM0OC4zMzMgMzQ4LjMzNCIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgMzQ4LjMzMyAzNDguMzM0OyIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSI+PGc+PHBhdGggZmlsbD0iIzU2NTY1NiIgZD0iTTMzNi41NTksNjguNjExTDIzMS4wMTYsMTc0LjE2NWwxMDUuNTQzLDEwNS41NDljMTUuNjk5LDE1LjcwNSwxNS42OTksNDEuMTQ1LDAsNTYuODVjLTcuODQ0LDcuODQ0LTE4LjEyOCwxMS43NjktMjguNDA3LDExLjc2OWMtMTAuMjk2LDAtMjAuNTgxLTMuOTE5LTI4LjQxOS0xMS43NjlMMTc0LjE2NywyMzEuMDAzTDY4LjYwOSwzMzYuNTYzYy03Ljg0Myw3Ljg0NC0xOC4xMjgsMTEuNzY5LTI4LjQxNiwxMS43NjljLTEwLjI4NSwwLTIwLjU2My0zLjkxOS0yOC40MTMtMTEuNzY5Yy0xNS42OTktMTUuNjk4LTE1LjY5OS00MS4xMzksMC01Ni44NWwxMDUuNTQtMTA1LjU0OUwxMS43NzQsNjguNjExYy0xNS42OTktMTUuNjk5LTE1LjY5OS00MS4xNDUsMC01Ni44NDRjMTUuNjk2LTE1LjY4Nyw0MS4xMjctMTUuNjg3LDU2LjgyOSwwbDEwNS41NjMsMTA1LjU1NEwyNzkuNzIxLDExLjc2N2MxNS43MDUtMTUuNjg3LDQxLjEzOS0xNS42ODcsNTYuODMyLDBDMzUyLjI1OCwyNy40NjYsMzUyLjI1OCw1Mi45MTIsMzM2LjU1OSw2OC42MTF6Ii8+PC9nPjwvc3ZnPg==\");background-size:contain;background-repeat:no-repeat;background-position:center;height:12px;width:12px}#onetrust-banner-sdk .powered-by-logo,#onetrust-banner-sdk .ot-pc-footer-logo a,#onetrust-pc-sdk .powered-by-logo,#onetrust-pc-sdk .ot-pc-footer-logo a,#ot-sync-ntfy .powered-by-logo,#ot-sync-ntfy .ot-pc-footer-logo a{background-size:contain;background-repeat:no-repeat;background-position:center;height:25px;width:152px;display:block}#onetrust-banner-sdk h3 *,#onetrust-banner-sdk h4 *,#onetrust-banner-sdk h6 *,#onetrust-banner-sdk button *,#onetrust-banner-sdk a[data-parent-id] *,#onetrust-pc-sdk h3 *,#onetrust-pc-sdk h4 *,#onetrust-pc-sdk h6 *,#onetrust-pc-sdk button *,#onetrust-pc-sdk a[data-parent-id] *,#ot-sync-ntfy h3 *,#ot-sync-ntfy h4 *,#ot-sync-ntfy h6 *,#ot-sync-ntfy button *,#ot-sync-ntfy a[data-parent-id] *{font-size:inherit;font-weight:inherit;color:inherit}#onetrust-banner-sdk .ot-hide,#onetrust-pc-sdk .ot-hide,#ot-sync-ntfy .ot-hide{display:none !important}#onetrust-pc-sdk .ot-sdk-row .ot-sdk-column{padding:0}#onetrust-pc-sdk .ot-sdk-container{padding-right:0}#onetrust-pc-sdk .ot-sdk-row{flex-direction:initial;width:100%}#onetrust-pc-sdk [type=\"checkbox\"]:checked,#onetrust-pc-sdk [type=\"checkbox\"]:not(:checked){pointer-events:initial}#onetrust-pc-sdk [type=\"checkbox\"]:disabled+label::before,#onetrust-pc-sdk [type=\"checkbox\"]:disabled+label:after,#onetrust-pc-sdk [type=\"checkbox\"]:disabled+label{pointer-events:none;opacity:0.7}#onetrust-pc-sdk #vendor-list-content{transform:translate3d(0, 0, 0)}#onetrust-pc-sdk li input[type=\"checkbox\"]{z-index:1}#onetrust-pc-sdk li .ot-checkbox label{z-index:2}#onetrust-pc-sdk li .ot-checkbox input[type=\"checkbox\"]{height:auto;width:auto}#onetrust-pc-sdk li .host-title a,#onetrust-pc-sdk li .ot-host-name a,#onetrust-pc-sdk li .accordion-text,#onetrust-pc-sdk li .ot-acc-txt{z-index:2;position:relative}#onetrust-pc-sdk input{margin:3px 0.1ex}#onetrust-pc-sdk .toggle-always-active{opacity:0.6;cursor:default}#onetrust-pc-sdk .pc-logo,#onetrust-pc-sdk .ot-pc-logo{height:60px;width:180px;background-position:center;background-size:contain;background-repeat:no-repeat}#onetrust-pc-sdk .ot-tooltip .ot-tooltiptext{visibility:hidden;width:120px;background-color:#555;color:#fff;text-align:center;padding:5px 0;border-radius:6px;position:absolute;z-index:1;bottom:125%;left:50%;margin-left:-60px;opacity:0;transition:opacity 0.3s}#onetrust-pc-sdk .ot-tooltip .ot-tooltiptext::after{content:\"\";position:absolute;top:100%;left:50%;margin-left:-5px;border-width:5px;border-style:solid;border-color:#555 transparent transparent transparent}#onetrust-pc-sdk .ot-tooltip:hover .ot-tooltiptext{visibility:visible;opacity:1}#onetrust-pc-sdk .ot-tooltip{position:relative;display:inline-block;z-index:3}#onetrust-pc-sdk .ot-tooltip svg{color:grey;height:20px;width:20px}#onetrust-pc-sdk .screen-reader-only,#onetrust-pc-sdk .ot-scrn-rdr,.ot-sdk-cookie-policy .screen-reader-only,.ot-sdk-cookie-policy .ot-scrn-rdr{border:0;clip:rect(0 0 0 0);height:1px;margin:-1px;overflow:hidden;padding:0;position:absolute;width:1px}#onetrust-pc-sdk.ot-fade-in,.onetrust-pc-dark-filter.ot-fade-in,#onetrust-banner-sdk.ot-fade-in{animation-name:onetrust-fade-in;animation-duration:400ms;animation-timing-function:ease-in-out}#onetrust-pc-sdk.ot-hide{display:none !important}.onetrust-pc-dark-filter.ot-hide{display:none !important}#ot-sdk-btn.ot-sdk-show-settings,#ot-sdk-btn.optanon-show-settings{color:#68b631;border:1px solid #68b631;height:auto;white-space:normal;word-wrap:break-word;padding:0.8em 2em;font-size:0.8em;line-height:1.2;cursor:pointer;-moz-transition:0.1s ease;-o-transition:0.1s ease;-webkit-transition:1s ease;transition:0.1s ease}#ot-sdk-btn.ot-sdk-show-settings:hover,#ot-sdk-btn.optanon-show-settings:hover{color:#fff;background-color:#68b631}.onetrust-pc-dark-filter{background:rgba(0,0,0,0.5);z-index:2147483646;width:100%;height:100%;overflow:hidden;position:fixed;top:0;bottom:0;left:0}@keyframes onetrust-fade-in{0%{opacity:0}100%{opacity:1}}@media only screen and (min-width: 426px) and (max-width: 896px) and (orientation: landscape){#onetrust-pc-sdk p{font-size:0.75em}}#onetrust-banner-sdk .banner-option-input:focus+label{outline:1px solid #000;outline-style:auto}\n",
       "#onetrust-banner-sdk,#onetrust-pc-sdk,#ot-sdk-cookie-policy,#ot-sync-ntfy{font-size:16px}#onetrust-banner-sdk *,#onetrust-banner-sdk ::after,#onetrust-banner-sdk ::before,#onetrust-pc-sdk *,#onetrust-pc-sdk ::after,#onetrust-pc-sdk ::before,#ot-sdk-cookie-policy *,#ot-sdk-cookie-policy ::after,#ot-sdk-cookie-policy ::before,#ot-sync-ntfy *,#ot-sync-ntfy ::after,#ot-sync-ntfy ::before{-webkit-box-sizing:content-box;-moz-box-sizing:content-box;box-sizing:content-box}#onetrust-banner-sdk div,#onetrust-banner-sdk span,#onetrust-banner-sdk h1,#onetrust-banner-sdk h2,#onetrust-banner-sdk h3,#onetrust-banner-sdk h4,#onetrust-banner-sdk h5,#onetrust-banner-sdk h6,#onetrust-banner-sdk p,#onetrust-banner-sdk img,#onetrust-banner-sdk svg,#onetrust-banner-sdk button,#onetrust-banner-sdk section,#onetrust-banner-sdk a,#onetrust-banner-sdk label,#onetrust-banner-sdk input,#onetrust-banner-sdk ul,#onetrust-banner-sdk li,#onetrust-banner-sdk nav,#onetrust-banner-sdk table,#onetrust-banner-sdk thead,#onetrust-banner-sdk tr,#onetrust-banner-sdk td,#onetrust-banner-sdk tbody,#onetrust-banner-sdk .ot-main-content,#onetrust-banner-sdk .ot-toggle,#onetrust-banner-sdk #ot-content,#onetrust-banner-sdk #ot-pc-content,#onetrust-banner-sdk .checkbox,#onetrust-pc-sdk div,#onetrust-pc-sdk span,#onetrust-pc-sdk h1,#onetrust-pc-sdk h2,#onetrust-pc-sdk h3,#onetrust-pc-sdk h4,#onetrust-pc-sdk h5,#onetrust-pc-sdk h6,#onetrust-pc-sdk p,#onetrust-pc-sdk img,#onetrust-pc-sdk svg,#onetrust-pc-sdk button,#onetrust-pc-sdk section,#onetrust-pc-sdk a,#onetrust-pc-sdk label,#onetrust-pc-sdk input,#onetrust-pc-sdk ul,#onetrust-pc-sdk li,#onetrust-pc-sdk nav,#onetrust-pc-sdk table,#onetrust-pc-sdk thead,#onetrust-pc-sdk tr,#onetrust-pc-sdk td,#onetrust-pc-sdk tbody,#onetrust-pc-sdk .ot-main-content,#onetrust-pc-sdk .ot-toggle,#onetrust-pc-sdk #ot-content,#onetrust-pc-sdk #ot-pc-content,#onetrust-pc-sdk .checkbox,#ot-sdk-cookie-policy div,#ot-sdk-cookie-policy span,#ot-sdk-cookie-policy h1,#ot-sdk-cookie-policy h2,#ot-sdk-cookie-policy h3,#ot-sdk-cookie-policy h4,#ot-sdk-cookie-policy h5,#ot-sdk-cookie-policy h6,#ot-sdk-cookie-policy p,#ot-sdk-cookie-policy img,#ot-sdk-cookie-policy svg,#ot-sdk-cookie-policy button,#ot-sdk-cookie-policy section,#ot-sdk-cookie-policy a,#ot-sdk-cookie-policy label,#ot-sdk-cookie-policy input,#ot-sdk-cookie-policy ul,#ot-sdk-cookie-policy li,#ot-sdk-cookie-policy nav,#ot-sdk-cookie-policy table,#ot-sdk-cookie-policy thead,#ot-sdk-cookie-policy tr,#ot-sdk-cookie-policy td,#ot-sdk-cookie-policy tbody,#ot-sdk-cookie-policy .ot-main-content,#ot-sdk-cookie-policy .ot-toggle,#ot-sdk-cookie-policy #ot-content,#ot-sdk-cookie-policy #ot-pc-content,#ot-sdk-cookie-policy .checkbox,#ot-sync-ntfy div,#ot-sync-ntfy span,#ot-sync-ntfy h1,#ot-sync-ntfy h2,#ot-sync-ntfy h3,#ot-sync-ntfy h4,#ot-sync-ntfy h5,#ot-sync-ntfy h6,#ot-sync-ntfy p,#ot-sync-ntfy img,#ot-sync-ntfy svg,#ot-sync-ntfy button,#ot-sync-ntfy section,#ot-sync-ntfy a,#ot-sync-ntfy label,#ot-sync-ntfy input,#ot-sync-ntfy ul,#ot-sync-ntfy li,#ot-sync-ntfy nav,#ot-sync-ntfy table,#ot-sync-ntfy thead,#ot-sync-ntfy tr,#ot-sync-ntfy td,#ot-sync-ntfy tbody,#ot-sync-ntfy .ot-main-content,#ot-sync-ntfy .ot-toggle,#ot-sync-ntfy #ot-content,#ot-sync-ntfy #ot-pc-content,#ot-sync-ntfy .checkbox{font-family:inherit;font-weight:normal;-webkit-font-smoothing:auto;letter-spacing:normal;line-height:normal;padding:0;margin:0;height:auto;min-height:0;max-height:none;width:auto;min-width:0;max-width:none;border-radius:0;border:none;clear:none;float:none;position:static;bottom:auto;left:auto;right:auto;top:auto;text-align:left;text-decoration:none;text-indent:0;text-shadow:none;text-transform:none;white-space:normal;background:none;overflow:visible;vertical-align:baseline;visibility:visible;z-index:auto;box-shadow:none}#onetrust-banner-sdk label:before,#onetrust-banner-sdk label:after,#onetrust-banner-sdk .checkbox:after,#onetrust-banner-sdk .checkbox:before,#onetrust-pc-sdk label:before,#onetrust-pc-sdk label:after,#onetrust-pc-sdk .checkbox:after,#onetrust-pc-sdk .checkbox:before,#ot-sdk-cookie-policy label:before,#ot-sdk-cookie-policy label:after,#ot-sdk-cookie-policy .checkbox:after,#ot-sdk-cookie-policy .checkbox:before,#ot-sync-ntfy label:before,#ot-sync-ntfy label:after,#ot-sync-ntfy .checkbox:after,#ot-sync-ntfy .checkbox:before{content:\"\";content:none}\n",
       "#onetrust-banner-sdk .ot-sdk-container,#onetrust-pc-sdk .ot-sdk-container,#ot-sdk-cookie-policy .ot-sdk-container{position:relative;width:100%;max-width:100%;margin:0 auto;padding:0 20px;box-sizing:border-box}#onetrust-banner-sdk .ot-sdk-column,#onetrust-banner-sdk .ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-column,#onetrust-pc-sdk .ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-columns{width:100%;float:left;box-sizing:border-box;padding:0;display:initial}@media (min-width: 400px){#onetrust-banner-sdk .ot-sdk-container,#onetrust-pc-sdk .ot-sdk-container,#ot-sdk-cookie-policy .ot-sdk-container{width:90%;padding:0}}@media (min-width: 550px){#onetrust-banner-sdk .ot-sdk-container,#onetrust-pc-sdk .ot-sdk-container,#ot-sdk-cookie-policy .ot-sdk-container{width:100%}#onetrust-banner-sdk .ot-sdk-column,#onetrust-banner-sdk .ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-column,#onetrust-pc-sdk .ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-columns{margin-left:4%}#onetrust-banner-sdk .ot-sdk-column:first-child,#onetrust-banner-sdk .ot-sdk-columns:first-child,#onetrust-pc-sdk .ot-sdk-column:first-child,#onetrust-pc-sdk .ot-sdk-columns:first-child,#ot-sdk-cookie-policy .ot-sdk-column:first-child,#ot-sdk-cookie-policy .ot-sdk-columns:first-child{margin-left:0}#onetrust-banner-sdk .ot-sdk-one.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-one.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-one.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-one.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-one.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-one.ot-sdk-columns{width:4.66666666667%}#onetrust-banner-sdk .ot-sdk-two.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-two.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-two.ot-sdk-columns{width:13.3333333333%}#onetrust-banner-sdk .ot-sdk-three.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-three.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-three.ot-sdk-columns{width:22%}#onetrust-banner-sdk .ot-sdk-four.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-four.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-four.ot-sdk-columns{width:30.6666666667%}#onetrust-banner-sdk .ot-sdk-five.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-five.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-five.ot-sdk-columns{width:39.3333333333%}#onetrust-banner-sdk .ot-sdk-six.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-six.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-six.ot-sdk-columns{width:48%}#onetrust-banner-sdk .ot-sdk-seven.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-seven.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-seven.ot-sdk-columns{width:56.6666666667%}#onetrust-banner-sdk .ot-sdk-eight.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-eight.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-eight.ot-sdk-columns{width:65.3333333333%}#onetrust-banner-sdk .ot-sdk-nine.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-nine.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-nine.ot-sdk-columns{width:74%}#onetrust-banner-sdk .ot-sdk-ten.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-ten.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-ten.ot-sdk-columns{width:82.6666666667%}#onetrust-banner-sdk .ot-sdk-eleven.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-eleven.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-eleven.ot-sdk-columns{width:91.3333333333%}#onetrust-banner-sdk .ot-sdk-twelve.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-twelve.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-twelve.ot-sdk-columns{width:100%;margin-left:0}#onetrust-banner-sdk .ot-sdk-one-third.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-one-third.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-one-third.ot-sdk-column{width:30.6666666667%}#onetrust-banner-sdk .ot-sdk-two-thirds.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-two-thirds.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-two-thirds.ot-sdk-column{width:65.3333333333%}#onetrust-banner-sdk .ot-sdk-one-half.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-one-half.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-one-half.ot-sdk-column{width:48%}#onetrust-banner-sdk .ot-sdk-offset-by-one.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-one.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-one.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-one.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-one.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-one.ot-sdk-columns{margin-left:8.66666666667%}#onetrust-banner-sdk .ot-sdk-offset-by-two.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-two.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-two.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-two.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-two.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-two.ot-sdk-columns{margin-left:17.3333333333%}#onetrust-banner-sdk .ot-sdk-offset-by-three.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-three.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-three.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-three.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-three.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-three.ot-sdk-columns{margin-left:26%}#onetrust-banner-sdk .ot-sdk-offset-by-four.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-four.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-four.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-four.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-four.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-four.ot-sdk-columns{margin-left:34.6666666667%}#onetrust-banner-sdk .ot-sdk-offset-by-five.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-five.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-five.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-five.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-five.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-five.ot-sdk-columns{margin-left:43.3333333333%}#onetrust-banner-sdk .ot-sdk-offset-by-six.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-six.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-six.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-six.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-six.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-six.ot-sdk-columns{margin-left:52%}#onetrust-banner-sdk .ot-sdk-offset-by-seven.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-seven.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-seven.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-seven.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-seven.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-seven.ot-sdk-columns{margin-left:60.6666666667%}#onetrust-banner-sdk .ot-sdk-offset-by-eight.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-eight.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-eight.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-eight.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-eight.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-eight.ot-sdk-columns{margin-left:69.3333333333%}#onetrust-banner-sdk .ot-sdk-offset-by-nine.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-nine.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-nine.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-nine.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-nine.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-nine.ot-sdk-columns{margin-left:78%}#onetrust-banner-sdk .ot-sdk-offset-by-ten.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-ten.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-ten.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-ten.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-ten.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-ten.ot-sdk-columns{margin-left:86.6666666667%}#onetrust-banner-sdk .ot-sdk-offset-by-eleven.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-eleven.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-eleven.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-eleven.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-eleven.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-eleven.ot-sdk-columns{margin-left:95.3333333333%}#onetrust-banner-sdk .ot-sdk-offset-by-one-third.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-one-third.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-one-third.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-one-third.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-one-third.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-one-third.ot-sdk-columns{margin-left:34.6666666667%}#onetrust-banner-sdk .ot-sdk-offset-by-two-thirds.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-two-thirds.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-two-thirds.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-two-thirds.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-two-thirds.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-two-thirds.ot-sdk-columns{margin-left:69.3333333333%}#onetrust-banner-sdk .ot-sdk-offset-by-one-half.ot-sdk-column,#onetrust-banner-sdk .ot-sdk-offset-by-one-half.ot-sdk-columns,#onetrust-pc-sdk .ot-sdk-offset-by-one-half.ot-sdk-column,#onetrust-pc-sdk .ot-sdk-offset-by-one-half.ot-sdk-columns,#ot-sdk-cookie-policy .ot-sdk-offset-by-one-half.ot-sdk-column,#ot-sdk-cookie-policy .ot-sdk-offset-by-one-half.ot-sdk-columns{margin-left:52%}}#onetrust-banner-sdk h1,#onetrust-banner-sdk h2,#onetrust-banner-sdk h3,#onetrust-banner-sdk h4,#onetrust-banner-sdk h5,#onetrust-banner-sdk h6,#onetrust-pc-sdk h1,#onetrust-pc-sdk h2,#onetrust-pc-sdk h3,#onetrust-pc-sdk h4,#onetrust-pc-sdk h5,#onetrust-pc-sdk h6,#ot-sdk-cookie-policy h1,#ot-sdk-cookie-policy h2,#ot-sdk-cookie-policy h3,#ot-sdk-cookie-policy h4,#ot-sdk-cookie-policy h5,#ot-sdk-cookie-policy h6{margin-top:0;font-weight:600;font-family:inherit}#onetrust-banner-sdk h1,#onetrust-pc-sdk h1,#ot-sdk-cookie-policy h1{font-size:1.5rem;line-height:1.2}#onetrust-banner-sdk h2,#onetrust-pc-sdk h2,#ot-sdk-cookie-policy h2{font-size:1.5rem;line-height:1.25}#onetrust-banner-sdk h3,#onetrust-pc-sdk h3,#ot-sdk-cookie-policy h3{font-size:1.5rem;line-height:1.3}#onetrust-banner-sdk h4,#onetrust-pc-sdk h4,#ot-sdk-cookie-policy h4{font-size:1.5rem;line-height:1.35}#onetrust-banner-sdk h5,#onetrust-pc-sdk h5,#ot-sdk-cookie-policy h5{font-size:1.5rem;line-height:1.5}#onetrust-banner-sdk h6,#onetrust-pc-sdk h6,#ot-sdk-cookie-policy h6{font-size:1.5rem;line-height:1.6}@media (min-width: 550px){#onetrust-banner-sdk h1,#onetrust-pc-sdk h1,#ot-sdk-cookie-policy h1{font-size:1.5rem}#onetrust-banner-sdk h2,#onetrust-pc-sdk h2,#ot-sdk-cookie-policy h2{font-size:1.5rem}#onetrust-banner-sdk h3,#onetrust-pc-sdk h3,#ot-sdk-cookie-policy h3{font-size:1.5rem}#onetrust-banner-sdk h4,#onetrust-pc-sdk h4,#ot-sdk-cookie-policy h4{font-size:1.5rem}#onetrust-banner-sdk h5,#onetrust-pc-sdk h5,#ot-sdk-cookie-policy h5{font-size:1.5rem}#onetrust-banner-sdk h6,#onetrust-pc-sdk h6,#ot-sdk-cookie-policy h6{font-size:1.5rem}}#onetrust-banner-sdk p,#onetrust-pc-sdk p,#ot-sdk-cookie-policy p{margin:0 0 1em 0;font-family:inherit;line-height:normal}#onetrust-banner-sdk a,#onetrust-pc-sdk a,#ot-sdk-cookie-policy a{color:#565656;text-decoration:underline}#onetrust-banner-sdk a:hover,#onetrust-pc-sdk a:hover,#ot-sdk-cookie-policy a:hover{color:#565656;text-decoration:none}#onetrust-banner-sdk .ot-sdk-button,#onetrust-banner-sdk button,#onetrust-pc-sdk .ot-sdk-button,#onetrust-pc-sdk button,#ot-sdk-cookie-policy .ot-sdk-button,#ot-sdk-cookie-policy button{margin-bottom:1rem;font-family:inherit}#onetrust-banner-sdk .ot-sdk-button,#onetrust-banner-sdk button,#onetrust-banner-sdk input[type=\"submit\"],#onetrust-banner-sdk input[type=\"reset\"],#onetrust-banner-sdk input[type=\"button\"],#onetrust-pc-sdk .ot-sdk-button,#onetrust-pc-sdk button,#onetrust-pc-sdk input[type=\"submit\"],#onetrust-pc-sdk input[type=\"reset\"],#onetrust-pc-sdk input[type=\"button\"],#ot-sdk-cookie-policy .ot-sdk-button,#ot-sdk-cookie-policy button,#ot-sdk-cookie-policy input[type=\"submit\"],#ot-sdk-cookie-policy input[type=\"reset\"],#ot-sdk-cookie-policy input[type=\"button\"]{display:inline-block;height:38px;padding:0 30px;color:#555;text-align:center;font-size:0.9em;font-weight:400;line-height:38px;letter-spacing:0.01em;text-decoration:none;white-space:nowrap;background-color:transparent;border-radius:2px;border:1px solid #bbb;cursor:pointer;box-sizing:border-box}#onetrust-banner-sdk .ot-sdk-button:hover,#onetrust-banner-sdk :not(.ot-leg-btn-container)>button:hover,#onetrust-banner-sdk input[type=\"submit\"]:hover,#onetrust-banner-sdk input[type=\"reset\"]:hover,#onetrust-banner-sdk input[type=\"button\"]:hover,#onetrust-banner-sdk .ot-sdk-button:focus,#onetrust-banner-sdk :not(.ot-leg-btn-container)>button:focus,#onetrust-banner-sdk input[type=\"submit\"]:focus,#onetrust-banner-sdk input[type=\"reset\"]:focus,#onetrust-banner-sdk input[type=\"button\"]:focus,#onetrust-pc-sdk .ot-sdk-button:hover,#onetrust-pc-sdk :not(.ot-leg-btn-container)>button:hover,#onetrust-pc-sdk input[type=\"submit\"]:hover,#onetrust-pc-sdk input[type=\"reset\"]:hover,#onetrust-pc-sdk input[type=\"button\"]:hover,#onetrust-pc-sdk .ot-sdk-button:focus,#onetrust-pc-sdk :not(.ot-leg-btn-container)>button:focus,#onetrust-pc-sdk input[type=\"submit\"]:focus,#onetrust-pc-sdk input[type=\"reset\"]:focus,#onetrust-pc-sdk input[type=\"button\"]:focus,#ot-sdk-cookie-policy .ot-sdk-button:hover,#ot-sdk-cookie-policy :not(.ot-leg-btn-container)>button:hover,#ot-sdk-cookie-policy input[type=\"submit\"]:hover,#ot-sdk-cookie-policy input[type=\"reset\"]:hover,#ot-sdk-cookie-policy input[type=\"button\"]:hover,#ot-sdk-cookie-policy .ot-sdk-button:focus,#ot-sdk-cookie-policy :not(.ot-leg-btn-container)>button:focus,#ot-sdk-cookie-policy input[type=\"submit\"]:focus,#ot-sdk-cookie-policy input[type=\"reset\"]:focus,#ot-sdk-cookie-policy input[type=\"button\"]:focus{color:#333;border-color:#888;opacity:0.7}#onetrust-banner-sdk .ot-sdk-button:focus,#onetrust-banner-sdk :not(.ot-leg-btn-container)>button:focus,#onetrust-banner-sdk input[type=\"submit\"]:focus,#onetrust-banner-sdk input[type=\"reset\"]:focus,#onetrust-banner-sdk input[type=\"button\"]:focus,#onetrust-pc-sdk .ot-sdk-button:focus,#onetrust-pc-sdk :not(.ot-leg-btn-container)>button:focus,#onetrust-pc-sdk input[type=\"submit\"]:focus,#onetrust-pc-sdk input[type=\"reset\"]:focus,#onetrust-pc-sdk input[type=\"button\"]:focus,#ot-sdk-cookie-policy .ot-sdk-button:focus,#ot-sdk-cookie-policy :not(.ot-leg-btn-container)>button:focus,#ot-sdk-cookie-policy input[type=\"submit\"]:focus,#ot-sdk-cookie-policy input[type=\"reset\"]:focus,#ot-sdk-cookie-policy input[type=\"button\"]:focus{outline:2px solid #000}#onetrust-banner-sdk .ot-sdk-button.ot-sdk-button-primary,#onetrust-banner-sdk button.ot-sdk-button-primary,#onetrust-banner-sdk input[type=\"submit\"].ot-sdk-button-primary,#onetrust-banner-sdk input[type=\"reset\"].ot-sdk-button-primary,#onetrust-banner-sdk input[type=\"button\"].ot-sdk-button-primary,#onetrust-pc-sdk .ot-sdk-button.ot-sdk-button-primary,#onetrust-pc-sdk button.ot-sdk-button-primary,#onetrust-pc-sdk input[type=\"submit\"].ot-sdk-button-primary,#onetrust-pc-sdk input[type=\"reset\"].ot-sdk-button-primary,#onetrust-pc-sdk input[type=\"button\"].ot-sdk-button-primary,#ot-sdk-cookie-policy .ot-sdk-button.ot-sdk-button-primary,#ot-sdk-cookie-policy button.ot-sdk-button-primary,#ot-sdk-cookie-policy input[type=\"submit\"].ot-sdk-button-primary,#ot-sdk-cookie-policy input[type=\"reset\"].ot-sdk-button-primary,#ot-sdk-cookie-policy input[type=\"button\"].ot-sdk-button-primary{color:#fff;background-color:#33c3f0;border-color:#33c3f0}#onetrust-banner-sdk .ot-sdk-button.ot-sdk-button-primary:hover,#onetrust-banner-sdk button.ot-sdk-button-primary:hover,#onetrust-banner-sdk input[type=\"submit\"].ot-sdk-button-primary:hover,#onetrust-banner-sdk input[type=\"reset\"].ot-sdk-button-primary:hover,#onetrust-banner-sdk input[type=\"button\"].ot-sdk-button-primary:hover,#onetrust-banner-sdk .ot-sdk-button.ot-sdk-button-primary:focus,#onetrust-banner-sdk button.ot-sdk-button-primary:focus,#onetrust-banner-sdk input[type=\"submit\"].ot-sdk-button-primary:focus,#onetrust-banner-sdk input[type=\"reset\"].ot-sdk-button-primary:focus,#onetrust-banner-sdk input[type=\"button\"].ot-sdk-button-primary:focus,#onetrust-pc-sdk .ot-sdk-button.ot-sdk-button-primary:hover,#onetrust-pc-sdk button.ot-sdk-button-primary:hover,#onetrust-pc-sdk input[type=\"submit\"].ot-sdk-button-primary:hover,#onetrust-pc-sdk input[type=\"reset\"].ot-sdk-button-primary:hover,#onetrust-pc-sdk input[type=\"button\"].ot-sdk-button-primary:hover,#onetrust-pc-sdk .ot-sdk-button.ot-sdk-button-primary:focus,#onetrust-pc-sdk button.ot-sdk-button-primary:focus,#onetrust-pc-sdk input[type=\"submit\"].ot-sdk-button-primary:focus,#onetrust-pc-sdk input[type=\"reset\"].ot-sdk-button-primary:focus,#onetrust-pc-sdk input[type=\"button\"].ot-sdk-button-primary:focus,#ot-sdk-cookie-policy .ot-sdk-button.ot-sdk-button-primary:hover,#ot-sdk-cookie-policy button.ot-sdk-button-primary:hover,#ot-sdk-cookie-policy input[type=\"submit\"].ot-sdk-button-primary:hover,#ot-sdk-cookie-policy input[type=\"reset\"].ot-sdk-button-primary:hover,#ot-sdk-cookie-policy input[type=\"button\"].ot-sdk-button-primary:hover,#ot-sdk-cookie-policy .ot-sdk-button.ot-sdk-button-primary:focus,#ot-sdk-cookie-policy button.ot-sdk-button-primary:focus,#ot-sdk-cookie-policy input[type=\"submit\"].ot-sdk-button-primary:focus,#ot-sdk-cookie-policy input[type=\"reset\"].ot-sdk-button-primary:focus,#ot-sdk-cookie-policy input[type=\"button\"].ot-sdk-button-primary:focus{color:#fff;background-color:#1eaedb;border-color:#1eaedb}#onetrust-banner-sdk input[type=\"email\"],#onetrust-banner-sdk input[type=\"number\"],#onetrust-banner-sdk input[type=\"search\"],#onetrust-banner-sdk input[type=\"text\"],#onetrust-banner-sdk input[type=\"tel\"],#onetrust-banner-sdk input[type=\"url\"],#onetrust-banner-sdk input[type=\"password\"],#onetrust-banner-sdk textarea,#onetrust-banner-sdk select,#onetrust-pc-sdk input[type=\"email\"],#onetrust-pc-sdk input[type=\"number\"],#onetrust-pc-sdk input[type=\"search\"],#onetrust-pc-sdk input[type=\"text\"],#onetrust-pc-sdk input[type=\"tel\"],#onetrust-pc-sdk input[type=\"url\"],#onetrust-pc-sdk input[type=\"password\"],#onetrust-pc-sdk textarea,#onetrust-pc-sdk select,#ot-sdk-cookie-policy input[type=\"email\"],#ot-sdk-cookie-policy input[type=\"number\"],#ot-sdk-cookie-policy input[type=\"search\"],#ot-sdk-cookie-policy input[type=\"text\"],#ot-sdk-cookie-policy input[type=\"tel\"],#ot-sdk-cookie-policy input[type=\"url\"],#ot-sdk-cookie-policy input[type=\"password\"],#ot-sdk-cookie-policy textarea,#ot-sdk-cookie-policy select{height:38px;padding:6px 10px;background-color:#fff;border:1px solid #d1d1d1;border-radius:4px;box-shadow:none;box-sizing:border-box}#onetrust-banner-sdk input[type=\"email\"],#onetrust-banner-sdk input[type=\"number\"],#onetrust-banner-sdk input[type=\"search\"],#onetrust-banner-sdk input[type=\"text\"],#onetrust-banner-sdk input[type=\"tel\"],#onetrust-banner-sdk input[type=\"url\"],#onetrust-banner-sdk input[type=\"password\"],#onetrust-banner-sdk textarea,#onetrust-pc-sdk input[type=\"email\"],#onetrust-pc-sdk input[type=\"number\"],#onetrust-pc-sdk input[type=\"search\"],#onetrust-pc-sdk input[type=\"text\"],#onetrust-pc-sdk input[type=\"tel\"],#onetrust-pc-sdk input[type=\"url\"],#onetrust-pc-sdk input[type=\"password\"],#onetrust-pc-sdk textarea,#ot-sdk-cookie-policy input[type=\"email\"],#ot-sdk-cookie-policy input[type=\"number\"],#ot-sdk-cookie-policy input[type=\"search\"],#ot-sdk-cookie-policy input[type=\"text\"],#ot-sdk-cookie-policy input[type=\"tel\"],#ot-sdk-cookie-policy input[type=\"url\"],#ot-sdk-cookie-policy input[type=\"password\"],#ot-sdk-cookie-policy textarea{-webkit-appearance:none;-moz-appearance:none;appearance:none}#onetrust-banner-sdk textarea,#onetrust-pc-sdk textarea,#ot-sdk-cookie-policy textarea{min-height:65px;padding-top:6px;padding-bottom:6px}#onetrust-banner-sdk input[type=\"email\"]:focus,#onetrust-banner-sdk input[type=\"number\"]:focus,#onetrust-banner-sdk input[type=\"search\"]:focus,#onetrust-banner-sdk input[type=\"text\"]:focus,#onetrust-banner-sdk input[type=\"tel\"]:focus,#onetrust-banner-sdk input[type=\"url\"]:focus,#onetrust-banner-sdk input[type=\"password\"]:focus,#onetrust-banner-sdk textarea:focus,#onetrust-banner-sdk select:focus,#onetrust-pc-sdk input[type=\"email\"]:focus,#onetrust-pc-sdk input[type=\"number\"]:focus,#onetrust-pc-sdk input[type=\"search\"]:focus,#onetrust-pc-sdk input[type=\"text\"]:focus,#onetrust-pc-sdk input[type=\"tel\"]:focus,#onetrust-pc-sdk input[type=\"url\"]:focus,#onetrust-pc-sdk input[type=\"password\"]:focus,#onetrust-pc-sdk textarea:focus,#onetrust-pc-sdk select:focus,#ot-sdk-cookie-policy input[type=\"email\"]:focus,#ot-sdk-cookie-policy input[type=\"number\"]:focus,#ot-sdk-cookie-policy input[type=\"search\"]:focus,#ot-sdk-cookie-policy input[type=\"text\"]:focus,#ot-sdk-cookie-policy input[type=\"tel\"]:focus,#ot-sdk-cookie-policy input[type=\"url\"]:focus,#ot-sdk-cookie-policy input[type=\"password\"]:focus,#ot-sdk-cookie-policy textarea:focus,#ot-sdk-cookie-policy select:focus{border:1px solid #000;outline:0}#onetrust-banner-sdk label,#onetrust-banner-sdk legend,#onetrust-pc-sdk label,#onetrust-pc-sdk legend,#ot-sdk-cookie-policy label,#ot-sdk-cookie-policy legend{display:block;margin-bottom:0.5rem;font-weight:600}#onetrust-banner-sdk fieldset,#onetrust-pc-sdk fieldset,#ot-sdk-cookie-policy fieldset{padding:0;border-width:0}#onetrust-banner-sdk input[type=\"checkbox\"],#onetrust-banner-sdk input[type=\"radio\"],#onetrust-pc-sdk input[type=\"checkbox\"],#onetrust-pc-sdk input[type=\"radio\"],#ot-sdk-cookie-policy input[type=\"checkbox\"],#ot-sdk-cookie-policy input[type=\"radio\"]{display:inline}#onetrust-banner-sdk label>.label-body,#onetrust-pc-sdk label>.label-body,#ot-sdk-cookie-policy label>.label-body{display:inline-block;margin-left:0.5rem;font-weight:normal}#onetrust-banner-sdk ul,#onetrust-pc-sdk ul,#ot-sdk-cookie-policy ul{list-style:circle inside}#onetrust-banner-sdk ol,#onetrust-pc-sdk ol,#ot-sdk-cookie-policy ol{list-style:decimal inside}#onetrust-banner-sdk ol,#onetrust-banner-sdk ul,#onetrust-pc-sdk ol,#onetrust-pc-sdk ul,#ot-sdk-cookie-policy ol,#ot-sdk-cookie-policy ul{padding-left:0;margin-top:0}#onetrust-banner-sdk ul ul,#onetrust-banner-sdk ul ol,#onetrust-banner-sdk ol ol,#onetrust-banner-sdk ol ul,#onetrust-pc-sdk ul ul,#onetrust-pc-sdk ul ol,#onetrust-pc-sdk ol ol,#onetrust-pc-sdk ol ul,#ot-sdk-cookie-policy ul ul,#ot-sdk-cookie-policy ul ol,#ot-sdk-cookie-policy ol ol,#ot-sdk-cookie-policy ol ul{margin:1.5rem 0 1.5rem 3rem;font-size:90%}#onetrust-banner-sdk li,#onetrust-pc-sdk li,#ot-sdk-cookie-policy li{margin-bottom:1rem}#onetrust-banner-sdk code,#onetrust-pc-sdk code,#ot-sdk-cookie-policy code{padding:0.2rem 0.5rem;margin:0 0.2rem;font-size:90%;white-space:nowrap;background:#f1f1f1;border:1px solid #e1e1e1;border-radius:4px}#onetrust-banner-sdk pre>code,#onetrust-pc-sdk pre>code,#ot-sdk-cookie-policy pre>code{display:block;padding:1rem 1.5rem;white-space:pre}#onetrust-banner-sdk th,#onetrust-banner-sdk td,#onetrust-pc-sdk th,#onetrust-pc-sdk td,#ot-sdk-cookie-policy th,#ot-sdk-cookie-policy td{padding:12px 15px;text-align:left;border-bottom:1px solid #e1e1e1}#onetrust-banner-sdk .ot-sdk-u-full-width,#onetrust-pc-sdk .ot-sdk-u-full-width,#ot-sdk-cookie-policy .ot-sdk-u-full-width{width:100%;box-sizing:border-box}#onetrust-banner-sdk .ot-sdk-u-max-full-width,#onetrust-pc-sdk .ot-sdk-u-max-full-width,#ot-sdk-cookie-policy .ot-sdk-u-max-full-width{max-width:100%;box-sizing:border-box}#onetrust-banner-sdk .ot-sdk-u-pull-right,#onetrust-pc-sdk .ot-sdk-u-pull-right,#ot-sdk-cookie-policy .ot-sdk-u-pull-right{float:right}#onetrust-banner-sdk .ot-sdk-u-pull-left,#onetrust-pc-sdk .ot-sdk-u-pull-left,#ot-sdk-cookie-policy .ot-sdk-u-pull-left{float:left}#onetrust-banner-sdk hr,#onetrust-pc-sdk hr,#ot-sdk-cookie-policy hr{margin-top:3rem;margin-bottom:3.5rem;border-width:0;border-top:1px solid #e1e1e1}#onetrust-banner-sdk .ot-sdk-container:after,#onetrust-banner-sdk .ot-sdk-row:after,#onetrust-banner-sdk .ot-sdk-u-cf,#onetrust-pc-sdk .ot-sdk-container:after,#onetrust-pc-sdk .ot-sdk-row:after,#onetrust-pc-sdk .ot-sdk-u-cf,#ot-sdk-cookie-policy .ot-sdk-container:after,#ot-sdk-cookie-policy .ot-sdk-row:after,#ot-sdk-cookie-policy .ot-sdk-u-cf{content:\"\";display:table;clear:both}#onetrust-banner-sdk .ot-sdk-row,#onetrust-pc-sdk .ot-sdk-row,#ot-sdk-cookie-policy .ot-sdk-row{margin:0;max-width:none;display:block}\n",
       "#onetrust-banner-sdk{box-shadow:0 0 18px rgba(0,0,0,.2)}#onetrust-banner-sdk.otFlat{position:fixed;z-index:2147483645;bottom:0;right:0;left:0;background-color:#fff;max-height:90%;overflow-x:hidden;overflow-y:auto}#onetrust-banner-sdk.otRelFont{font-size:1rem}#onetrust-banner-sdk>.ot-sdk-container{overflow:hidden}#onetrust-banner-sdk::-webkit-scrollbar{width:11px}#onetrust-banner-sdk::-webkit-scrollbar-thumb{border-radius:10px;background:#c1c1c1}#onetrust-banner-sdk{scrollbar-arrow-color:#c1c1c1;scrollbar-darkshadow-color:#c1c1c1;scrollbar-face-color:#c1c1c1;scrollbar-shadow-color:#c1c1c1}#onetrust-banner-sdk #onetrust-policy{margin:1.25em 0 .625em 2em;overflow:hidden}#onetrust-banner-sdk #onetrust-policy .ot-gv-list-handler{float:left;font-size:.82em;padding:0;margin-bottom:0;border:0;line-height:normal;height:auto;width:auto}#onetrust-banner-sdk #onetrust-policy-title{font-size:1.2em;line-height:1.3;margin-bottom:10px}#onetrust-banner-sdk #onetrust-policy-text{clear:both;text-align:left;font-size:.88em;line-height:1.4}#onetrust-banner-sdk #onetrust-policy-text *{font-size:inherit;line-height:inherit}#onetrust-banner-sdk #onetrust-policy-text a{font-weight:bold;margin-left:5px}#onetrust-banner-sdk #onetrust-policy-title,#onetrust-banner-sdk #onetrust-policy-text{color:dimgray;float:left}#onetrust-banner-sdk #onetrust-button-group-parent{min-height:1px;text-align:center}#onetrust-banner-sdk #onetrust-button-group{display:inline-block}#onetrust-banner-sdk #onetrust-accept-btn-handler,#onetrust-banner-sdk #onetrust-reject-all-handler,#onetrust-banner-sdk #onetrust-pc-btn-handler{background-color:#68b631;color:#fff;border-color:#68b631;margin-right:1em;min-width:125px;height:auto;white-space:normal;word-break:break-word;word-wrap:break-word;padding:12px 10px;line-height:1.2;font-size:.813em;font-weight:600}#onetrust-banner-sdk #onetrust-pc-btn-handler.cookie-setting-link{background-color:#fff;border:none;color:#68b631;text-decoration:underline;padding-left:0;padding-right:0}#onetrust-banner-sdk .onetrust-close-btn-ui{width:44px;height:44px;background-size:12px;border:none;position:relative;margin:auto;padding:0}#onetrust-banner-sdk .banner_logo{display:none}#onetrust-banner-sdk .ot-b-addl-desc{clear:both;float:left;display:block}#onetrust-banner-sdk #banner-options{float:left;display:table;margin-right:0;margin-left:1em;width:calc(100% - 1em)}#onetrust-banner-sdk .banner-option-input{cursor:pointer;width:auto;height:auto;border:none;padding:0;padding-right:3px;margin:0 0 10px;font-size:.82em;line-height:1.4}#onetrust-banner-sdk .banner-option-input *{pointer-events:none;font-size:inherit;line-height:inherit}#onetrust-banner-sdk .banner-option-input[aria-expanded=true]~.banner-option-details{display:block;height:auto}#onetrust-banner-sdk .banner-option-input[aria-expanded=true] .ot-arrow-container{transform:rotate(90deg)}#onetrust-banner-sdk .banner-option{margin-bottom:12px;margin-left:0;border:none;float:left;padding:0}#onetrust-banner-sdk .banner-option:first-child{padding-left:2px}#onetrust-banner-sdk .banner-option:not(:first-child){padding:0;border:none}#onetrust-banner-sdk .banner-option-header{cursor:pointer;display:inline-block}#onetrust-banner-sdk .banner-option-header :first-child{color:dimgray;font-weight:bold;float:left}#onetrust-banner-sdk .banner-option-header .ot-arrow-container{display:inline-block;border-top:6px solid transparent;border-bottom:6px solid transparent;border-left:6px solid dimgray;margin-left:10px;vertical-align:middle}#onetrust-banner-sdk .banner-option-details{display:none;font-size:.83em;line-height:1.5;padding:10px 0px 5px 10px;margin-right:10px;height:0px}#onetrust-banner-sdk .banner-option-details *{font-size:inherit;line-height:inherit;color:dimgray}#onetrust-banner-sdk .ot-arrow-container,#onetrust-banner-sdk .banner-option-details{transition:all 300ms ease-in 0s;-webkit-transition:all 300ms ease-in 0s;-moz-transition:all 300ms ease-in 0s;-o-transition:all 300ms ease-in 0s}#onetrust-banner-sdk .ot-dpd-container{float:left}#onetrust-banner-sdk .ot-dpd-title{margin-bottom:10px}#onetrust-banner-sdk .ot-dpd-title,#onetrust-banner-sdk .ot-dpd-desc{font-size:.88em;line-height:1.4;color:dimgray}#onetrust-banner-sdk .ot-dpd-title *,#onetrust-banner-sdk .ot-dpd-desc *{font-size:inherit;line-height:inherit}#onetrust-banner-sdk.ot-iab-2 #onetrust-policy-text *{margin-bottom:0}#onetrust-banner-sdk.ot-iab-2 .onetrust-vendors-list-handler{display:block;margin-left:0;margin-top:5px;clear:both;margin-bottom:0;padding:0;border:0;height:auto;width:auto}#onetrust-banner-sdk.ot-iab-2 #onetrust-button-group button{display:block}#onetrust-banner-sdk.ot-close-btn-link{padding-top:25px}#onetrust-banner-sdk.ot-close-btn-link #onetrust-close-btn-container{top:15px;transform:none;right:15px}#onetrust-banner-sdk.ot-close-btn-link #onetrust-close-btn-container button{padding:0;white-space:pre-wrap;border:none;height:auto;line-height:1.5;text-decoration:underline;font-size:.69em}#onetrust-banner-sdk #onetrust-policy-text,#onetrust-banner-sdk .ot-dpd-desc,#onetrust-banner-sdk .ot-b-addl-desc{font-size:.813em;line-height:1.5}#onetrust-banner-sdk .ot-dpd-desc{margin-bottom:10px}#onetrust-banner-sdk .ot-dpd-desc>.ot-b-addl-desc{margin-top:10px;margin-bottom:10px;font-size:1em}@media only screen and (max-width: 425px){#onetrust-banner-sdk #onetrust-close-btn-container{position:absolute;top:10px;right:10px}#onetrust-banner-sdk #onetrust-policy{margin-left:0}#onetrust-banner-sdk #onetrust-button-group{display:block}#onetrust-banner-sdk #onetrust-accept-btn-handler,#onetrust-banner-sdk #onetrust-reject-all-handler,#onetrust-banner-sdk #onetrust-pc-btn-handler{width:100%}#onetrust-banner-sdk .onetrust-close-btn-ui{top:auto;transform:none}#onetrust-banner-sdk #onetrust-policy-title{display:inline;float:none}#onetrust-banner-sdk #banner-options{margin:0;padding:0;width:100%}}@media only screen and (min-width: 426px)and (max-width: 896px){#onetrust-banner-sdk #onetrust-close-btn-container{position:absolute;top:0;right:0}#onetrust-banner-sdk #onetrust-policy{margin-left:1em;margin-right:1em}#onetrust-banner-sdk .onetrust-close-btn-ui{top:10px;right:10px}#onetrust-banner-sdk:not(.ot-iab-2) #onetrust-group-container{width:95%}#onetrust-banner-sdk.ot-iab-2 #onetrust-group-container{width:100%}#onetrust-banner-sdk #onetrust-button-group-parent{width:100%;position:relative;margin-left:0}#onetrust-banner-sdk #onetrust-button-group button{display:inline-block}#onetrust-banner-sdk #onetrust-button-group{margin-right:0;text-align:center}#onetrust-banner-sdk .has-reject-all-button #onetrust-pc-btn-handler{float:left}#onetrust-banner-sdk .has-reject-all-button #onetrust-reject-all-handler,#onetrust-banner-sdk .has-reject-all-button #onetrust-accept-btn-handler{float:right}#onetrust-banner-sdk .has-reject-all-button #onetrust-button-group{width:calc(100% - 2em);margin-right:0}#onetrust-banner-sdk .has-reject-all-button #onetrust-pc-btn-handler.cookie-setting-link{padding-left:0px;text-align:left}#onetrust-banner-sdk.ot-buttons-fw .ot-sdk-three button{width:100%;text-align:center}#onetrust-banner-sdk.ot-buttons-fw #onetrust-button-group-parent button{float:none}#onetrust-banner-sdk.ot-buttons-fw #onetrust-pc-btn-handler.cookie-setting-link{text-align:center}}@media only screen and (min-width: 550px){#onetrust-banner-sdk .banner-option:not(:first-child){border-left:1px solid #d8d8d8;padding-left:25px}}@media only screen and (min-width: 425px)and (max-width: 550px){#onetrust-banner-sdk.ot-iab-2 #onetrust-button-group,#onetrust-banner-sdk.ot-iab-2 #onetrust-policy,#onetrust-banner-sdk.ot-iab-2 .banner-option{width:100%}}@media only screen and (min-width: 769px){#onetrust-banner-sdk #onetrust-button-group{margin-right:30%}#onetrust-banner-sdk #banner-options{margin-left:2em;margin-right:5em;margin-bottom:1.25em;width:calc(100% - 7em)}}@media only screen and (min-width: 897px)and (max-width: 1023px){#onetrust-banner-sdk.vertical-align-content #onetrust-button-group-parent{position:absolute;top:50%;left:75%;transform:translateY(-50%)}#onetrust-banner-sdk #onetrust-close-btn-container{top:50%;margin:auto;transform:translate(-50%, -50%);position:absolute;padding:0;right:0}#onetrust-banner-sdk #onetrust-close-btn-container button{position:relative;margin:0;right:-22px;top:2px}}@media only screen and (min-width: 1024px){#onetrust-banner-sdk #onetrust-close-btn-container{top:50%;margin:auto;transform:translate(-50%, -50%);position:absolute;right:0}#onetrust-banner-sdk #onetrust-close-btn-container button{right:-12px}#onetrust-banner-sdk #onetrust-policy{margin-left:2em}#onetrust-banner-sdk.vertical-align-content #onetrust-button-group-parent{position:absolute;top:50%;left:60%;transform:translateY(-50%)}#onetrust-banner-sdk.ot-iab-2 #onetrust-policy-title{width:50%}#onetrust-banner-sdk.ot-iab-2 #onetrust-policy-text,#onetrust-banner-sdk.ot-iab-2 :not(.ot-dpd-desc)>.ot-b-addl-desc{margin-bottom:1em;width:50%;border-right:1px solid #d8d8d8;padding-right:1rem}#onetrust-banner-sdk.ot-iab-2 #onetrust-policy-text{margin-bottom:0;padding-bottom:1em}#onetrust-banner-sdk.ot-iab-2 :not(.ot-dpd-desc)>.ot-b-addl-desc{margin-bottom:0;padding-bottom:1em}#onetrust-banner-sdk.ot-iab-2 .ot-dpd-container{width:45%;padding-left:1rem;display:inline-block;float:none}#onetrust-banner-sdk.ot-iab-2 .ot-dpd-title{line-height:1.7}#onetrust-banner-sdk.ot-iab-2 #onetrust-button-group-parent{left:auto;right:4%;margin-left:0}#onetrust-banner-sdk.ot-iab-2 #onetrust-button-group button{display:block}#onetrust-banner-sdk:not(.ot-iab-2) #onetrust-button-group-parent{margin:auto;width:30%}#onetrust-banner-sdk:not(.ot-iab-2) #onetrust-group-container{width:60%}#onetrust-banner-sdk #onetrust-button-group{margin-right:auto}#onetrust-banner-sdk #onetrust-accept-btn-handler,#onetrust-banner-sdk #onetrust-reject-all-handler,#onetrust-banner-sdk #onetrust-pc-btn-handler{margin-top:1em}}@media only screen and (min-width: 890px){#onetrust-banner-sdk.ot-buttons-fw:not(.ot-iab-2) #onetrust-button-group-parent{padding-left:3%;padding-right:4%;margin-left:0}#onetrust-banner-sdk.ot-buttons-fw:not(.ot-iab-2) #onetrust-button-group{margin-right:0;margin-top:1.25em;width:100%}#onetrust-banner-sdk.ot-buttons-fw:not(.ot-iab-2) #onetrust-button-group button{width:100%;margin-bottom:5px;margin-top:5px}#onetrust-banner-sdk.ot-buttons-fw:not(.ot-iab-2) #onetrust-button-group button:last-of-type{margin-bottom:20px}}@media only screen and (min-width: 1280px){#onetrust-banner-sdk:not(.ot-iab-2) #onetrust-group-container{width:55%}#onetrust-banner-sdk:not(.ot-iab-2) #onetrust-button-group-parent{width:44%;padding-left:2%;padding-right:2%}#onetrust-banner-sdk:not(.ot-iab-2).vertical-align-content #onetrust-button-group-parent{position:absolute;left:55%}}\n",
       "        #onetrust-consent-sdk #onetrust-banner-sdk {background-color: #FFFFFF;}\n",
       "            #onetrust-consent-sdk #onetrust-policy-title,\n",
       "                    #onetrust-consent-sdk #onetrust-policy-text,\n",
       "                    #onetrust-consent-sdk .ot-b-addl-desc,\n",
       "                    #onetrust-consent-sdk .ot-dpd-desc,\n",
       "                    #onetrust-consent-sdk .ot-dpd-title,\n",
       "                    #onetrust-consent-sdk #onetrust-policy-text *:not(.onetrust-vendors-list-handler),\n",
       "                    #onetrust-consent-sdk .ot-dpd-desc *:not(.onetrust-vendors-list-handler),\n",
       "                    #onetrust-consent-sdk #onetrust-banner-sdk #banner-options *,\n",
       "                    #onetrust-banner-sdk .ot-cat-header {\n",
       "                        color: #545454;\n",
       "                    }\n",
       "            #onetrust-consent-sdk #onetrust-banner-sdk .banner-option-details {\n",
       "                    background-color: #E5E5E5;}\n",
       "             #onetrust-consent-sdk #onetrust-banner-sdk a[href],\n",
       "                    #onetrust-consent-sdk #onetrust-banner-sdk a[href] font,\n",
       "                    #onetrust-consent-sdk #onetrust-banner-sdk .ot-link-btn\n",
       "                        {\n",
       "                            color: #0C4DED;\n",
       "                        }#onetrust-consent-sdk #onetrust-accept-btn-handler,\n",
       "                         #onetrust-banner-sdk #onetrust-reject-all-handler {\n",
       "                            background-color: #0C4DED;border-color: #0C4DED;\n",
       "                            color: #ffffff;\n",
       "                        }\n",
       "            #onetrust-consent-sdk #onetrust-pc-btn-handler,\n",
       "            #onetrust-consent-sdk #onetrust-pc-btn-handler.cookie-setting-link {\n",
       "                color: #0C4DED; border-color: #0C4DED;\n",
       "                background-color: \n",
       "                #FFFFFF;\n",
       "            }#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container .ot-close-icon,#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container-mobile .ot-close-icon,#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-header .ot-close-icon{background-image:url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4NCjxwYXRoIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNMTIuMDIyIDEzLjM0N0w1LjIxODAzIDIwLjEyODVDNS4xNzM0NCAyMC4xNzMyIDUuMTIwNDkgMjAuMjA4NiA1LjA2MjE5IDIwLjIzMjhDNS4wMDM4OSAyMC4yNTcgNC45NDEzOSAyMC4yNjk0IDQuODc4MjggMjAuMjY5NEM0LjgxNTE2IDIwLjI2OTQgNC43NTI2NyAyMC4yNTcgNC42OTQzNyAyMC4yMzI4QzQuNjM2MDcgMjAuMjA4NiA0LjU4MzExIDIwLjE3MzIgNC41Mzg1MyAyMC4xMjg1TDMuODYwNTMgMTkuNDQ5QzMuODE1OSAxOS40MDQ1IDMuNzgwNDkgMTkuMzUxNyAzLjc1NjMzIDE5LjI5MzVDMy43MzIxNyAxOS4yMzU0IDMuNzE5NzMgMTkuMTczIDMuNzE5NzMgMTkuMTFDMy43MTk3MyAxOS4wNDcgMy43MzIxNyAxOC45ODQ2IDMuNzU2MzMgMTguOTI2NUMzLjc4MDQ5IDE4Ljg2ODMgMy44MTU5IDE4LjgxNTUgMy44NjA1MyAxOC43NzFMMTAuNjY0NSAxMS45ODk1TDMuODc3MDMgNS4yMzIwMUMzLjgzMjQgNS4xODc1NSAzLjc5Njk5IDUuMTM0NzMgMy43NzI4MyA1LjA3NjU1QzMuNzQ4NjcgNS4wMTgzOCAzLjczNjIzIDQuOTU2MDEgMy43MzYyMyA0Ljg5MzAxQzMuNzM2MjMgNC44MzAwMiAzLjc0ODY3IDQuNzY3NjUgMy43NzI4MyA0LjcwOTQ3QzMuNzk2OTkgNC42NTEzIDMuODMyNCA0LjU5ODQ3IDMuODc3MDMgNC41NTQwMkw0LjU1NjUzIDMuODc0NTFDNC42MDA5OCAzLjgyOTg4IDQuNjUzODIgMy43OTQ0NiA0LjcxMTk5IDMuNzcwM0M0Ljc3MDE2IDMuNzQ2MTQgNC44MzI1MyAzLjczMzcgNC44OTU1MyAzLjczMzdDNC45NTg1MiAzLjczMzcgNS4wMjA4OSAzLjc0NjE0IDUuMDc5MDYgMy43NzAzQzUuMTM3MjQgMy43OTQ0NiA1LjE5MDA3IDMuODI5ODggNS4yMzQ1MyAzLjg3NDUxTDEyLjAyMiAxMC42MzJMMTguNzgyNSAzLjg3MTUxQzE4LjgyNzEgMy44MjY4NCAxOC44ODAxIDMuNzkxMzkgMTguOTM4NCAzLjc2NzIxQzE4Ljk5NjcgMy43NDMwMyAxOS4wNTkyIDMuNzMwNTkgMTkuMTIyMyAzLjczMDU5QzE5LjE4NTQgMy43MzA1OSAxOS4yNDc5IDMuNzQzMDMgMTkuMzA2MiAzLjc2NzIxQzE5LjM2NDUgMy43OTEzOSAxOS40MTc0IDMuODI2ODQgMTkuNDYyIDMuODcxNTFMMjAuMTQgNC41NTEwMkMyMC4xODQ3IDQuNTk1NDcgMjAuMjIwMSA0LjY0ODMgMjAuMjQ0MiA0LjcwNjQ3QzIwLjI2ODQgNC43NjQ2NSAyMC4yODA4IDQuODI3MDIgMjAuMjgwOCA0Ljg5MDAxQzIwLjI4MDggNC45NTMwMSAyMC4yNjg0IDUuMDE1MzggMjAuMjQ0MiA1LjA3MzU1QzIwLjIyMDEgNS4xMzE3MyAyMC4xODQ3IDUuMTg0NTUgMjAuMTQgNS4yMjkwMUwxMy4zNzk1IDExLjk4OTVMMjAuMTI5NSAxOC43NzFDMjAuMTc0MiAxOC44MTU1IDIwLjIwOTYgMTguODY4MyAyMC4yMzM3IDE4LjkyNjVDMjAuMjU3OSAxOC45ODQ2IDIwLjI3MDMgMTkuMDQ3IDIwLjI3MDMgMTkuMTFDMjAuMjcwMyAxOS4xNzMgMjAuMjU3OSAxOS4yMzU0IDIwLjIzMzcgMTkuMjkzNUMyMC4yMDk2IDE5LjM1MTcgMjAuMTc0MiAxOS40MDQ1IDIwLjEyOTUgMTkuNDQ5TDE5LjQ1IDIwLjEyODVDMTkuNDA1NiAyMC4xNzMxIDE5LjM1MjcgMjAuMjA4NiAxOS4yOTQ2IDIwLjIzMjdDMTkuMjM2NCAyMC4yNTY5IDE5LjE3NCAyMC4yNjkzIDE5LjExMSAyMC4yNjkzQzE5LjA0OCAyMC4yNjkzIDE4Ljk4NTcgMjAuMjU2OSAxOC45Mjc1IDIwLjIzMjdDMTguODY5MyAyMC4yMDg2IDE4LjgxNjUgMjAuMTczMSAxOC43NzIgMjAuMTI4NUwxMi4wMjIgMTMuMzQ3WiIgZmlsbD0iIzU0NTQ1NCIvPg0KPC9zdmc+DQo=);background-size:20px;background-repeat:no-repeat;background-position:center}#onetrust-consent-sdk{font-family:MuseoSans,Helvetica,Arial,sans-serif!important;box-shadow:0 -12px 17px rgba(40,40,40,.12),0 -5px 22px rgba(40,40,40,.08),0 -7px 8px rgba(40,40,40,.16)}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group{margin-right:0;margin-left:auto;display:flex}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group-parent{display:flex;position:relative;top:unset;left:unset;align-items:center;transform:none;margin:0;min-width:490px;width:55%;padding-left:0}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button{--padding-horizontal:24px;--padding-iconButton:32px;--padding-vertical:8px;font-size:1rem;line-height:1.375rem;font-weight:300;align-items:center;background-color:var(--maincolor40,#0c4ded);border-radius:32px;border:1px solid var(--maincolor40,#0c4ded);box-sizing:border-box;color:var(--white,#fff);display:inline-flex;justify-content:center;padding:var(--padding-vertical) var(--padding-horizontal);position:relative;margin:0;height:40px}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button:hover,#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container .save-preference-btn-handler.onetrust-close-btn-handler:hover{background-color:var(--maincolor50,#003bde);border-color:var(--maincolor50,#003bde)}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button:focus,#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content #accept-recommended-btn-handler:focus,#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item button,#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container .save-preference-btn-handler.onetrust-close-btn-handler:focus{outline:0}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button.cookie-setting-link{margin-right:24px;text-decoration:none;background-color:transparent!important;border-color:var(--maincolor40,#0c4ded)!important;color:var(--maincolor40,#0c4ded)!important}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button.cookie-setting-link:hover{color:var(--maincolor50,#003bde)!important;border-color:var(--maincolor50,#003bde)!important}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button#onetrust-accept-btn-handler{margin-right:72px}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container .ot-close-icon,#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container-mobile .ot-close-icon{margin:0}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container{top:50%;margin:auto;transform:translate(-50%,-50%);position:absolute;right:0}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container .ot-close-icon{top:auto;right:-25px}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container-mobile{position:absolute;right:-12px;top:8px}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-policy{margin:0;padding:32px 40px 32px 16px}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-policy h3{color:var(--black,#282828);font-size:1.125rem;line-height:1.75rem;font-weight:500;margin:0 0 8px}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-policy p{font-size:1rem;line-height:1.375rem;font-weight:300;margin:0}@media only screen and (max-width:1441px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-policy p{font-size:.875rem;line-height:1.25rem}}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-group-container{width:auto;position:relative}#onetrust-consent-sdk #onetrust-pc-sdk.otPcCenter{border-radius:8px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-header{border:none;height:0;padding:0}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-header .ot-close-icon{z-index:8}#onetrust-consent-sdk #onetrust-pc-sdk .ot-cat-grp .ot-acc-hdr .ot-plus-minus span,#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-logo{display:none}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer{border:none;background-color:var(--white,#fff);z-index:1}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer-logo{padding:16px 32px;box-sizing:border-box;height:48px}@media only screen and (max-width:1280px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group-parent{padding-right:32px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer-logo{padding:16px}}@media only screen and (max-width:768px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group,#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button{width:100%}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group-parent{padding:0;width:100%}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container{display:none}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container-mobile{display:inline;float:none}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-policy{padding:24px 0}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-policy p{font-size:1rem;line-height:1.375rem}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer-logo{padding:16px 32px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container{display:flex}}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer-logo a{margin:0;width:155px;height:16px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item{border-color:var(--grey40,#d5d5d5)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item:first-of-type{border-top-left-radius:8px;border-top-right-radius:8px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item:last-of-type{border-bottom-left-radius:8px;border-bottom-right-radius:8px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item .ot-cat-header{font-size:1rem;line-height:1.375rem;font-weight:500;color:var(--grey80,#545454)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item .ot-category-desc{font-size:1rem;line-height:1.375rem;font-weight:300;color:var(--grey80,#545454)}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content #accept-recommended-btn-handler,#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container .save-preference-btn-handler.onetrust-close-btn-handler{--padding-horizontal:24px;--padding-iconButton:32px;--padding-vertical:8px;background-color:var(--maincolor40,#0c4ded);box-sizing:border-box;color:var(--white,#fff);display:inline-flex;position:relative;height:40px;line-height:1.375rem}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item .ot-acc-grpcntr.ot-acc-txt,#onetrust-consent-sdk #onetrust-pc-sdk .ot-switch-nob::before,#onetrust-consent-sdk #onetrust-pc-sdk .ot-tgl input:checked+.ot-switch .ot-switch-nob::before{background-color:var(--white,#fff)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-switch-nob{background-color:var(--grey50,#bfbfbf);border-color:var(--grey50,#bfbfbf)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-tgl input:checked+.ot-switch .ot-switch-nob{background-color:var(--black,#282828);border:1px solid var(--black,#282828)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container{text-align:left}#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container .save-preference-btn-handler.onetrust-close-btn-handler{font-size:1rem;font-weight:300;align-items:center;border-radius:32px;border:1px solid var(--maincolor40,#0c4ded);justify-content:center;padding:var(--padding-vertical) var(--padding-horizontal);margin:16px 32px;background-color:var(--maincolor40,#0c4ded)!important;border-color:var(--maincolor40,#0c4ded)!important;color:var(--white,#fff)!important}@media only screen and (max-width:768px){#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container>button{width:100%;max-width:calc(50% - 16px)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container>button:first-child{margin-right:16px}#onetrust-banner-sdk .ot-sdk-container{padding:0 16px 24px!important}}#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container>:first-child{margin-left:0}#onetrust-consent-sdk #onetrust-pc-sdk .ot-cat-grp{margin-bottom:40px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-cat-grp .ot-acc-hdr .ot-plus-minus{background-image:url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIGhlaWdodD0iMTYiIHZpZXdCb3g9IjAgMCAxNiAxNiIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4NCjxwYXRoIGQ9Ik01LjY0MDcyIDIuMDk3TDUuMTg4NzIgMi41NDhDNS4xNTg4MSAyLjU3NzUzIDUuMTM1MSAyLjYxMjczIDUuMTE4OTYgMi42NTE1NEM1LjEwMjgyIDIuNjkwMzQgNS4wOTQ1OCAyLjczMTk3IDUuMDk0NzMgMi43NzRDNS4wOTQ2NyAyLjgxNjE3IDUuMTAyODMgMi44NTc5NSA1LjExODczIDIuODk3QzUuMTM1MzkgMi45MzQ1NCA1LjE1OTE1IDIuOTY4NTEgNS4xODg3MiAyLjk5NzAxTDEwLjE4ODcgNy45OTcwMUw1LjE4ODcyIDEyLjk5N0M1LjE1OTQ1IDEzLjAyNTUgNS4xMzYwMiAxMy4wNTk1IDUuMTE5NzIgMTMuMDk3QzUuMTAzNSAxMy4xMzYgNS4wOTUwMiAxMy4xNzc4IDUuMDk0NzMgMTMuMjJDNS4wOTUwMiAxMy4yNjE5IDUuMTAzNSAxMy4zMDM0IDUuMTE5NzIgMTMuMzQyQzUuMTM2MDIgMTMuMzc5NSA1LjE1OTQ1IDEzLjQxMzUgNS4xODg3MiAxMy40NDJMNS42NDE3MyAxMy44OTNDNS43MDE2NyAxMy45NTI1IDUuNzgyNzMgMTMuOTg2IDUuODY3MjIgMTMuOTg2QzUuOTUxNzEgMTMuOTg2IDYuMDMyNzggMTMuOTUyNSA2LjA5MjczIDEzLjg5M0wxMS45OTI3IDcuOTkzMDFMNi4wODU3MyAyLjA5M0M2LjAyNTYxIDIuMDMzNCA1Ljk0NDM4IDEuOTk5OTcgNS44NTk3MyAxLjk5OTk3QzUuNzc1MDcgMS45OTk5NyA1LjY5Mzg1IDIuMDMzNCA1LjYzMzczIDIuMDkzTDUuNjQwNzIgMi4wOTdaIiBmaWxsPSIjNTQ1NDU0Ii8+DQo8L3N2Zz4NCg==);background-repeat:no-repeat;background-position:center;margin-right:8px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-cat-grp button[aria-expanded=true]~.ot-acc-hdr .ot-plus-minus{transform:rotate(90deg)}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-title{font-size:1.25rem;line-height:1.75rem;font-weight:700;color:var(--black,#282828)!important;position:sticky;z-index:5;top:0;left:0;margin:0;padding:32px 0 8px;background-color:var(--white,#fff)}#onetrust-consent-sdk #onetrust-pc-sdk #ot-category-title{font-size:1.125rem;line-height:1.75rem;font-weight:500;color:var(--black,#282828)!important;margin:56px 0 24px}@media only screen and (max-width:1441px){#onetrust-consent-sdk #onetrust-pc-sdk #ot-category-title{font-size:1rem;line-height:1.375rem}}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content{margin:0;padding:0 32px;width:calc(100% - 68px);position:relative;top:0;height:calc(100% - 140px)}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content::-webkit-scrollbar{background:var(--white);-webkit-appearance:none;width:4px;height:4px}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content::-webkit-scrollbar-thumb{border-radius:4px;background-color:var(--grey50);-webkit-box-shadow:0 0 1px rgba(255,255,255,.5);border:none}@-moz-document url-prefix(){#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content{scrollbar-width:thin}}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content .privacy-notice-link{text-decoration:none;color:var(--maincolor50,#003bde);font-size:1rem;line-height:1.375rem;font-weight:500;position:relative;margin-top:16px;display:block}@media only screen and (max-width:1024px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button#onetrust-accept-btn-handler{margin-right:40px}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content .privacy-notice-link{font-size:.875rem;line-height:1.25rem}}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content #accept-recommended-btn-handler{font-size:1rem;font-weight:300;align-items:center;border-radius:32px;border:1px solid var(--maincolor40,#0c4ded);justify-content:center;padding:var(--padding-vertical) var(--padding-horizontal);margin:16px 0 0;background-color:transparent!important;border-color:var(--maincolor40,#0c4ded)!important;color:var(--maincolor40,#0c4ded)!important}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content #accept-recommended-btn-handler:hover{background-color:var(--maincolor50,#003bde);border-color:var(--maincolor50,#003bde);color:var(--maincolor50,#003bde)!important;border-color:var(--maincolor50,#003bde)!important}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-desc{font-size:1rem;line-height:1.375rem;font-weight:300;color:var(--grey80,#545454);margin:0}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group{flex-flow:column;flex-direction:column-reverse}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group-parent{min-width:auto}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button.cookie-setting-link,#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container .save-preference-btn-handler.onetrust-close-btn-handler{margin:0}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button#onetrust-accept-btn-handler{margin:0 0 16px}#onetrust-consent-sdk #onetrust-pc-sdk.otPcCenter{left:0;min-width:100%;top:0;border-radius:0;height:100%!important}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer .ot-btn-container{padding:16px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer-logo{padding:8px 16px;height:32px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container>button{max-width:none}#onetrust-consent-sdk #onetrust-pc-sdk .ot-cat-grp{margin-bottom:32px}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-title{font-size:1rem;line-height:1.375rem;padding:24px 0}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-title::before{content:\"\";position:absolute;display:block;height:1px;left:-16px;right:-16px;bottom:0;background-color:var(--grey40,#d5d5d5)}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content{padding:0 16px;width:calc(100% - 34px)}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-desc{padding-top:16px}}#onetrust-banner-sdk .ot-sdk-container>.ot-sdk-row{display:flex}@media only screen and (max-width:768px){#onetrust-banner-sdk .ot-sdk-container>.ot-sdk-row{flex-flow:column}}#onetrust-pc-sdk.otPcCenter{overflow:hidden;position:fixed;margin:0 auto;top:5%;right:0;left:0;width:40%;max-width:575px;min-width:575px;border-radius:2.5px;z-index:2147483647;background-color:#fff;-webkit-box-shadow:0px 2px 10px -3px #999;-moz-box-shadow:0px 2px 10px -3px #999;box-shadow:0px 2px 10px -3px #999}#onetrust-pc-sdk.otPcCenter[dir=rtl]{right:0;left:0}#onetrust-pc-sdk.otRelFont{font-size:1rem}#onetrust-pc-sdk #ot-addtl-venlst .ot-arw-cntr,#onetrust-pc-sdk #ot-addtl-venlst .ot-plus-minus,#onetrust-pc-sdk .ot-hide-tgl{visibility:hidden}#onetrust-pc-sdk #ot-addtl-venlst .ot-arw-cntr *,#onetrust-pc-sdk #ot-addtl-venlst .ot-plus-minus *,#onetrust-pc-sdk .ot-hide-tgl *{visibility:hidden}#onetrust-pc-sdk #ot-gn-venlst .ot-ven-item .ot-acc-hdr{min-height:40px}#onetrust-pc-sdk .ot-pc-header{height:39px;padding:10px 0 10px 30px;border-bottom:1px solid #e9e9e9}#onetrust-pc-sdk #ot-pc-title,#onetrust-pc-sdk #ot-category-title,#onetrust-pc-sdk .ot-cat-header,#onetrust-pc-sdk #ot-lst-title,#onetrust-pc-sdk .ot-ven-hdr .ot-ven-name,#onetrust-pc-sdk .ot-always-active{font-weight:bold;color:dimgray}#onetrust-pc-sdk .ot-cat-header{float:left;font-weight:600;font-size:.875em;line-height:1.5;max-width:90%;vertical-align:middle}#onetrust-pc-sdk .ot-always-active-group .ot-cat-header{width:55%;font-weight:700}#onetrust-pc-sdk .ot-cat-item p{clear:both;float:left;margin-top:10px;margin-bottom:5px;line-height:1.5;font-size:.812em;color:dimgray}#onetrust-pc-sdk .ot-close-icon{height:44px;width:44px;background-size:10px}#onetrust-pc-sdk #ot-pc-title{float:left;font-size:1em;line-height:1.5;margin-bottom:10px;margin-top:10px;width:100%}#onetrust-pc-sdk #accept-recommended-btn-handler{margin-right:10px;margin-bottom:25px;outline-offset:-1px}#onetrust-pc-sdk #ot-pc-desc{clear:both;width:100%;font-size:.812em;line-height:1.5;margin-bottom:25px}#onetrust-pc-sdk #ot-pc-desc a{margin-left:5px}#onetrust-pc-sdk #ot-pc-desc *{font-size:inherit;line-height:inherit}#onetrust-pc-sdk #ot-pc-desc ul li{padding:10px 0px}#onetrust-pc-sdk a{color:#656565;cursor:pointer}#onetrust-pc-sdk a:hover{color:#3860be}#onetrust-pc-sdk label{margin-bottom:0}#onetrust-pc-sdk #vdr-lst-dsc{font-size:.812em;line-height:1.5;padding:10px 15px 5px 15px}#onetrust-pc-sdk button{max-width:394px;padding:12px 30px;line-height:1;word-break:break-word;word-wrap:break-word;white-space:normal;font-weight:bold;height:auto}#onetrust-pc-sdk .ot-link-btn{padding:0;margin-bottom:0;border:0;font-weight:normal;line-height:normal;width:auto;height:auto}#onetrust-pc-sdk #ot-pc-content{position:absolute;overflow-y:scroll;padding-left:0px;padding-right:30px;top:60px;bottom:110px;margin:1px 3px 0 30px;width:calc(100% - 63px)}#onetrust-pc-sdk .ot-cat-grp .ot-always-active{float:right;clear:none;color:#3860be;margin:0;font-size:.813em;line-height:1.3}#onetrust-pc-sdk .ot-pc-scrollbar::-webkit-scrollbar-track{margin-right:20px}#onetrust-pc-sdk .ot-pc-scrollbar::-webkit-scrollbar{width:11px}#onetrust-pc-sdk .ot-pc-scrollbar::-webkit-scrollbar-thumb{border-radius:10px;background:#d8d8d8}#onetrust-pc-sdk input[type=checkbox]:focus+.ot-acc-hdr{outline:#000 1px solid}#onetrust-pc-sdk .ot-pc-scrollbar{scrollbar-arrow-color:#d8d8d8;scrollbar-darkshadow-color:#d8d8d8;scrollbar-face-color:#d8d8d8;scrollbar-shadow-color:#d8d8d8}#onetrust-pc-sdk .save-preference-btn-handler{margin-right:20px}#onetrust-pc-sdk .ot-pc-refuse-all-handler{margin-right:10px}#onetrust-pc-sdk #ot-pc-desc .privacy-notice-link{margin-left:0}#onetrust-pc-sdk .ot-subgrp-cntr{display:inline-block;clear:both;width:100%;padding-top:15px}#onetrust-pc-sdk .ot-switch+.ot-subgrp-cntr{padding-top:10px}#onetrust-pc-sdk ul.ot-subgrps{margin:0;font-size:initial}#onetrust-pc-sdk ul.ot-subgrps li p,#onetrust-pc-sdk ul.ot-subgrps li h5{font-size:.813em;line-height:1.4;color:dimgray}#onetrust-pc-sdk ul.ot-subgrps .ot-switch{min-height:auto}#onetrust-pc-sdk ul.ot-subgrps .ot-switch-nob{top:0}#onetrust-pc-sdk ul.ot-subgrps .ot-acc-hdr{display:inline-block;width:100%}#onetrust-pc-sdk ul.ot-subgrps .ot-acc-txt{margin:0}#onetrust-pc-sdk ul.ot-subgrps li{padding:0;border:none}#onetrust-pc-sdk ul.ot-subgrps li h5{position:relative;top:5px;font-weight:bold;margin-bottom:0;float:left}#onetrust-pc-sdk li.ot-subgrp{margin-left:20px;overflow:auto}#onetrust-pc-sdk li.ot-subgrp>h5{width:calc(100% - 100px)}#onetrust-pc-sdk .ot-cat-item p>ul,#onetrust-pc-sdk li.ot-subgrp p>ul{margin:0px;list-style:disc;margin-left:15px;font-size:inherit}#onetrust-pc-sdk .ot-cat-item p>ul li,#onetrust-pc-sdk li.ot-subgrp p>ul li{font-size:inherit;padding-top:10px;padding-left:0px;padding-right:0px;border:none}#onetrust-pc-sdk .ot-cat-item p>ul li:last-child,#onetrust-pc-sdk li.ot-subgrp p>ul li:last-child{padding-bottom:10px}#onetrust-pc-sdk .ot-pc-logo{height:40px;width:120px;display:inline-block}#onetrust-pc-sdk .ot-pc-footer{position:absolute;bottom:0px;width:100%;max-height:160px;border-top:1px solid #d8d8d8}#onetrust-pc-sdk.ot-ftr-stacked .ot-pc-refuse-all-handler{margin-bottom:0px}#onetrust-pc-sdk.ot-ftr-stacked #ot-pc-content{bottom:160px}#onetrust-pc-sdk.ot-ftr-stacked .ot-pc-footer button{width:100%;max-width:none}#onetrust-pc-sdk.ot-ftr-stacked .ot-btn-container{margin:0 30px;width:calc(100% - 60px);padding-right:0}#onetrust-pc-sdk .ot-pc-footer-logo{height:30px;width:100%;text-align:right;background:#f4f4f4}#onetrust-pc-sdk .ot-pc-footer-logo a{display:inline-block;margin-top:5px;margin-right:10px}#onetrust-pc-sdk[dir=rtl] .ot-pc-footer-logo{direction:rtl}#onetrust-pc-sdk[dir=rtl] .ot-pc-footer-logo a{margin-right:25px}#onetrust-pc-sdk .ot-tgl{float:right;position:relative;z-index:1}#onetrust-pc-sdk .ot-tgl input:checked+.ot-switch .ot-switch-nob{background-color:#cddcf2;border:1px solid #3860be}#onetrust-pc-sdk .ot-tgl input:checked+.ot-switch .ot-switch-nob:before{-webkit-transform:translateX(20px);-ms-transform:translateX(20px);transform:translateX(20px);background-color:#3860be;border-color:#3860be}#onetrust-pc-sdk .ot-tgl input:focus+.ot-switch{outline:#000 solid 1px}#onetrust-pc-sdk .ot-switch{position:relative;display:inline-block;width:45px;height:25px}#onetrust-pc-sdk .ot-switch-nob{position:absolute;cursor:pointer;top:0;left:0;right:0;bottom:0;background-color:#f2f1f1;border:1px solid #ddd;transition:all .2s ease-in 0s;-moz-transition:all .2s ease-in 0s;-o-transition:all .2s ease-in 0s;-webkit-transition:all .2s ease-in 0s;border-radius:20px}#onetrust-pc-sdk .ot-switch-nob:before{position:absolute;content:\"\";height:21px;width:21px;bottom:1px;background-color:#7d7d7d;-webkit-transition:.4s;transition:.4s;border-radius:20px}#onetrust-pc-sdk .ot-chkbox input:checked~label::before{background-color:#3860be}#onetrust-pc-sdk .ot-chkbox input+label::after{content:none;color:#fff}#onetrust-pc-sdk .ot-chkbox input:checked+label::after{content:\"\"}#onetrust-pc-sdk .ot-chkbox input:focus+label::before{outline-style:solid;outline-width:2px;outline-style:auto}#onetrust-pc-sdk .ot-chkbox label{position:relative;display:inline-block;padding-left:30px;cursor:pointer;font-weight:500}#onetrust-pc-sdk .ot-chkbox label::before,#onetrust-pc-sdk .ot-chkbox label::after{position:absolute;content:\"\";display:inline-block;border-radius:3px}#onetrust-pc-sdk .ot-chkbox label::before{height:18px;width:18px;border:1px solid #3860be;left:0px;top:auto}#onetrust-pc-sdk .ot-chkbox label::after{height:5px;width:9px;border-left:3px solid;border-bottom:3px solid;transform:rotate(-45deg);-o-transform:rotate(-45deg);-ms-transform:rotate(-45deg);-webkit-transform:rotate(-45deg);left:4px;top:5px}#onetrust-pc-sdk .ot-label-txt{display:none}#onetrust-pc-sdk .ot-chkbox input,#onetrust-pc-sdk .ot-tgl input{position:absolute;opacity:0;width:0;height:0}#onetrust-pc-sdk .ot-arw-cntr{float:right;position:relative}#onetrust-pc-sdk .ot-arw-cntr .ot-arw{width:16px;height:16px;margin-left:5px;color:dimgray;display:inline-block;vertical-align:middle;-webkit-transition:all 150ms ease-in 0s;-moz-transition:all 150ms ease-in 0s;-o-transition:all 150ms ease-in 0s;transition:all 150ms ease-in 0s}#onetrust-pc-sdk input:checked~.ot-acc-hdr .ot-arw{transform:rotate(90deg);-o-transform:rotate(90deg);-ms-transform:rotate(90deg);-webkit-transform:rotate(90deg)}#onetrust-pc-sdk input[type=checkbox]:focus+.ot-acc-hdr{outline:#000 1px solid}#onetrust-pc-sdk .ot-tgl-cntr,#onetrust-pc-sdk .ot-arw-cntr{display:inline-block}#onetrust-pc-sdk .ot-tgl-cntr{width:45px;float:right;margin-top:2px}#onetrust-pc-sdk #ot-lst-cnt .ot-tgl-cntr{margin-top:10px}#onetrust-pc-sdk .ot-always-active-subgroup{width:auto;padding-left:0px !important;top:3px;position:relative}#onetrust-pc-sdk .ot-label-status{padding-left:5px;font-size:.75em;display:none}#onetrust-pc-sdk .ot-arw-cntr{margin-top:-1px}#onetrust-pc-sdk .ot-arw-cntr svg{-webkit-transition:all 300ms ease-in 0s;-moz-transition:all 300ms ease-in 0s;-o-transition:all 300ms ease-in 0s;transition:all 300ms ease-in 0s;height:10px;width:10px}#onetrust-pc-sdk input:checked~.ot-acc-hdr .ot-arw{transform:rotate(90deg);-o-transform:rotate(90deg);-ms-transform:rotate(90deg);-webkit-transform:rotate(90deg)}#onetrust-pc-sdk .ot-arw{width:10px;margin-left:15px;transition:all 300ms ease-in 0s;-webkit-transition:all 300ms ease-in 0s;-moz-transition:all 300ms ease-in 0s;-o-transition:all 300ms ease-in 0s}#onetrust-pc-sdk .ot-vlst-cntr{margin-bottom:0}#onetrust-pc-sdk .ot-hlst-cntr{margin-top:5px;display:inline-block;width:100%}#onetrust-pc-sdk .category-vendors-list-handler,#onetrust-pc-sdk .category-vendors-list-handler+a,#onetrust-pc-sdk .category-host-list-handler{clear:both;color:#3860be;margin-left:0;font-size:.813em;text-decoration:none;float:left;overflow:hidden}#onetrust-pc-sdk .category-vendors-list-handler:hover,#onetrust-pc-sdk .category-vendors-list-handler+a:hover,#onetrust-pc-sdk .category-host-list-handler:hover{color:#1883fd}#onetrust-pc-sdk .category-vendors-list-handler+a{clear:none}#onetrust-pc-sdk .category-vendors-list-handler+a::after{content:\"\";height:15px;width:15px;background-repeat:no-repeat;margin-left:5px;float:right;background-image:url(\"data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 511.626 511.627'%3E%3Cg fill='%231276CE'%3E%3Cpath d='M392.857 292.354h-18.274c-2.669 0-4.859.855-6.563 2.573-1.718 1.708-2.573 3.897-2.573 6.563v91.361c0 12.563-4.47 23.315-13.415 32.262-8.945 8.945-19.701 13.414-32.264 13.414H82.224c-12.562 0-23.317-4.469-32.264-13.414-8.945-8.946-13.417-19.698-13.417-32.262V155.31c0-12.562 4.471-23.313 13.417-32.259 8.947-8.947 19.702-13.418 32.264-13.418h200.994c2.669 0 4.859-.859 6.57-2.57 1.711-1.713 2.566-3.9 2.566-6.567V82.221c0-2.662-.855-4.853-2.566-6.563-1.711-1.713-3.901-2.568-6.57-2.568H82.224c-22.648 0-42.016 8.042-58.102 24.125C8.042 113.297 0 132.665 0 155.313v237.542c0 22.647 8.042 42.018 24.123 58.095 16.086 16.084 35.454 24.13 58.102 24.13h237.543c22.647 0 42.017-8.046 58.101-24.13 16.085-16.077 24.127-35.447 24.127-58.095v-91.358c0-2.669-.856-4.859-2.574-6.57-1.713-1.718-3.903-2.573-6.565-2.573z'/%3E%3Cpath d='M506.199 41.971c-3.617-3.617-7.905-5.424-12.85-5.424H347.171c-4.948 0-9.233 1.807-12.847 5.424-3.617 3.615-5.428 7.898-5.428 12.847s1.811 9.233 5.428 12.85l50.247 50.248-186.147 186.151c-1.906 1.903-2.856 4.093-2.856 6.563 0 2.479.953 4.668 2.856 6.571l32.548 32.544c1.903 1.903 4.093 2.852 6.567 2.852s4.665-.948 6.567-2.852l186.148-186.148 50.251 50.248c3.614 3.617 7.898 5.426 12.847 5.426s9.233-1.809 12.851-5.426c3.617-3.616 5.424-7.898 5.424-12.847V54.818c-.001-4.952-1.814-9.232-5.428-12.847z'/%3E%3C/g%3E%3C/svg%3E\")}#onetrust-pc-sdk .back-btn-handler{font-size:1em;text-decoration:none}#onetrust-pc-sdk .back-btn-handler:hover{opacity:.6}#onetrust-pc-sdk #ot-lst-title span{display:inline-block;word-break:break-word;word-wrap:break-word;margin-bottom:0;color:#656565;font-size:1em;font-weight:bold;margin-left:15px}#onetrust-pc-sdk #ot-lst-title{margin:10px 0 10px 0px;font-size:1em;text-align:left}#onetrust-pc-sdk #ot-pc-hdr{margin:0 0 0 30px;height:auto;width:auto}#onetrust-pc-sdk #ot-pc-hdr input::placeholder{color:#d4d4d4;font-style:italic}#onetrust-pc-sdk #vendor-search-handler{height:31px;width:100%;border-radius:50px;font-size:.8em;padding-right:35px;padding-left:15px;float:left;margin-left:15px}#onetrust-pc-sdk .ot-ven-name{display:block;width:auto;padding-right:5px}#onetrust-pc-sdk #ot-lst-cnt{overflow-y:auto;margin-left:20px;margin-right:7px;width:calc(100% - 27px);max-height:calc(100% - 80px);height:100%;transform:translate3d(0, 0, 0)}#onetrust-pc-sdk #ot-pc-lst{width:100%;bottom:100px;position:absolute;top:60px}#onetrust-pc-sdk #ot-pc-lst:not(.ot-enbl-chr) .ot-tgl-cntr .ot-arw-cntr,#onetrust-pc-sdk #ot-pc-lst:not(.ot-enbl-chr) .ot-tgl-cntr .ot-arw-cntr *{visibility:hidden}#onetrust-pc-sdk #ot-pc-lst .ot-tgl-cntr{right:12px;position:absolute}#onetrust-pc-sdk #ot-pc-lst .ot-arw-cntr{float:right;position:relative}#onetrust-pc-sdk #ot-pc-lst .ot-arw{margin-left:10px}#onetrust-pc-sdk #ot-pc-lst .ot-acc-hdr{overflow:hidden;cursor:pointer}#onetrust-pc-sdk .ot-vlst-cntr{overflow:hidden}#onetrust-pc-sdk #ot-sel-blk{overflow:hidden;width:100%;position:sticky;position:-webkit-sticky;top:0;z-index:3}#onetrust-pc-sdk #ot-back-arw{height:12px;width:12px}#onetrust-pc-sdk .ot-lst-subhdr{width:100%;display:inline-block}#onetrust-pc-sdk .ot-search-cntr{float:left;width:78%;position:relative}#onetrust-pc-sdk .ot-search-cntr>svg{width:30px;height:30px;position:absolute;float:left;right:-15px}#onetrust-pc-sdk .ot-fltr-cntr{float:right;right:50px;position:relative}#onetrust-pc-sdk #filter-btn-handler{background-color:#3860be;border-radius:17px;display:inline-block;position:relative;width:32px;height:32px;-moz-transition:.1s ease;-o-transition:.1s ease;-webkit-transition:1s ease;transition:.1s ease;padding:0;margin:0}#onetrust-pc-sdk #filter-btn-handler:hover{background-color:#3860be}#onetrust-pc-sdk #filter-btn-handler svg{width:12px;height:12px;margin:3px 10px 0 10px;display:block;position:static;right:auto;top:auto}#onetrust-pc-sdk .ot-ven-link{color:#3860be;text-decoration:none;font-weight:100;display:inline-block;padding-top:10px;transform:translate(0, 1%);-o-transform:translate(0, 1%);-ms-transform:translate(0, 1%);-webkit-transform:translate(0, 1%);position:relative;z-index:2}#onetrust-pc-sdk .ot-ven-link *{font-size:inherit}#onetrust-pc-sdk .ot-ven-link:hover{text-decoration:underline}#onetrust-pc-sdk .ot-ven-hdr{width:calc(100% - 160px);height:auto;float:left;word-break:break-word;word-wrap:break-word;vertical-align:middle;padding-bottom:3px}#onetrust-pc-sdk .ot-ven-link{letter-spacing:.03em;font-size:.75em;font-weight:400}#onetrust-pc-sdk .ot-ven-dets{border-radius:2px;background-color:#f8f8f8}#onetrust-pc-sdk .ot-ven-dets div:first-child p:first-child{border-top:none}#onetrust-pc-sdk .ot-ven-dets .ot-ven-disc:not(:first-child){border-top:1px solid #e9e9e9}#onetrust-pc-sdk .ot-ven-dets .ot-ven-disc:nth-child(n+3) p{display:inline-block}#onetrust-pc-sdk .ot-ven-dets .ot-ven-disc:nth-child(n+3) p:nth-of-type(odd){width:30%}#onetrust-pc-sdk .ot-ven-dets .ot-ven-disc:nth-child(n+3) p:nth-of-type(even){width:50%;word-break:break-word;word-wrap:break-word}#onetrust-pc-sdk .ot-ven-dets .ot-ven-disc p{padding-top:5px;padding-bottom:5px;display:block}#onetrust-pc-sdk .ot-ven-dets p{font-size:.69em;text-align:left;vertical-align:middle;word-break:break-word;word-wrap:break-word;margin:0;padding-bottom:10px;padding-left:15px;color:#2e3644}#onetrust-pc-sdk .ot-ven-dets p:first-child{padding-top:5px}#onetrust-pc-sdk .ot-ven-dets .ot-ven-pur p:first-child{border-top:1px solid #e9e9e9;border-bottom:1px solid #e9e9e9;padding-bottom:5px;margin-bottom:5px;font-weight:bold}#onetrust-pc-sdk #ot-host-lst .ot-sel-all{float:right;position:relative;margin-right:42px;top:10px}#onetrust-pc-sdk #ot-host-lst .ot-sel-all input[type=checkbox]{width:auto;height:auto}#onetrust-pc-sdk #ot-host-lst .ot-sel-all label{height:20px;width:20px;padding-left:0px}#onetrust-pc-sdk #ot-host-lst .ot-acc-txt{overflow:hidden;width:95%}#onetrust-pc-sdk .ot-host-hdr{position:relative;z-index:1;pointer-events:none;width:calc(100% - 125px);float:left}#onetrust-pc-sdk .ot-host-name,#onetrust-pc-sdk .ot-host-desc{display:inline-block;width:90%}#onetrust-pc-sdk .ot-host-name{pointer-events:none}#onetrust-pc-sdk .ot-host-hdr>a{text-decoration:underline;font-size:.82em;position:relative;z-index:2;float:left;margin-bottom:5px;pointer-events:initial}#onetrust-pc-sdk .ot-host-name+a{margin-top:5px}#onetrust-pc-sdk .ot-host-name,#onetrust-pc-sdk .ot-host-name a,#onetrust-pc-sdk .ot-host-desc,#onetrust-pc-sdk .ot-host-info{color:dimgray;word-break:break-word;word-wrap:break-word}#onetrust-pc-sdk .ot-host-name,#onetrust-pc-sdk .ot-host-name a{font-weight:bold;font-size:.82em;line-height:1.3}#onetrust-pc-sdk .ot-host-name a{font-size:1em}#onetrust-pc-sdk .ot-host-expand{margin-top:3px;margin-bottom:3px;clear:both;display:block;color:#3860be;font-size:.72em;font-weight:normal}#onetrust-pc-sdk .ot-host-expand *{font-size:inherit}#onetrust-pc-sdk .ot-host-desc,#onetrust-pc-sdk .ot-host-info{font-size:.688em;line-height:1.4;font-weight:normal}#onetrust-pc-sdk .ot-host-desc{margin-top:10px}#onetrust-pc-sdk .ot-host-opt{margin:0;font-size:inherit;display:inline-block;width:100%}#onetrust-pc-sdk .ot-host-opt li>div div{font-size:.8em;padding:5px 0}#onetrust-pc-sdk .ot-host-opt li>div div:nth-child(1){width:30%;float:left}#onetrust-pc-sdk .ot-host-opt li>div div:nth-child(2){width:70%;float:left;word-break:break-word;word-wrap:break-word}#onetrust-pc-sdk .ot-host-info{border:none;display:inline-block;width:calc(100% - 10px);padding:10px;margin-bottom:10px;background-color:#f8f8f8}#onetrust-pc-sdk .ot-host-info>div{overflow:auto}#onetrust-pc-sdk #no-results{text-align:center;margin-top:30px}#onetrust-pc-sdk #no-results p{font-size:1em;color:#2e3644;word-break:break-word;word-wrap:break-word}#onetrust-pc-sdk #no-results p span{font-weight:bold}#onetrust-pc-sdk #ot-fltr-modal{width:100%;height:auto;display:none;-moz-transition:.2s ease;-o-transition:.2s ease;-webkit-transition:2s ease;transition:.2s ease;overflow:hidden;opacity:1;right:0}#onetrust-pc-sdk #ot-fltr-modal .ot-label-txt{display:inline-block;font-size:.85em;color:dimgray}#onetrust-pc-sdk #ot-fltr-cnt{z-index:2147483646;background-color:#fff;position:absolute;height:90%;max-height:300px;width:325px;left:210px;margin-top:10px;margin-bottom:20px;padding-right:10px;border-radius:3px;-webkit-box-shadow:0px 0px 12px 2px #c7c5c7;-moz-box-shadow:0px 0px 12px 2px #c7c5c7;box-shadow:0px 0px 12px 2px #c7c5c7}#onetrust-pc-sdk .ot-fltr-scrlcnt{overflow-y:auto;overflow-x:hidden;clear:both;max-height:calc(100% - 60px)}#onetrust-pc-sdk #ot-anchor{border:12px solid transparent;display:none;position:absolute;z-index:2147483647;right:55px;top:75px;transform:rotate(45deg);-o-transform:rotate(45deg);-ms-transform:rotate(45deg);-webkit-transform:rotate(45deg);background-color:#fff;-webkit-box-shadow:-3px -3px 5px -2px #c7c5c7;-moz-box-shadow:-3px -3px 5px -2px #c7c5c7;box-shadow:-3px -3px 5px -2px #c7c5c7}#onetrust-pc-sdk .ot-fltr-btns{margin-left:15px}#onetrust-pc-sdk #filter-apply-handler{margin-right:15px}#onetrust-pc-sdk .ot-fltr-opt{margin-bottom:25px;margin-left:15px;width:75%;position:relative}#onetrust-pc-sdk .ot-fltr-opt p{display:inline-block;margin:0;font-size:.9em;color:#2e3644}#onetrust-pc-sdk .ot-chkbox label span{font-size:.85em;color:dimgray}#onetrust-pc-sdk .ot-chkbox input[type=checkbox]+label::after{content:none;color:#fff}#onetrust-pc-sdk .ot-chkbox input[type=checkbox]:checked+label::after{content:\"\"}#onetrust-pc-sdk .ot-chkbox input[type=checkbox]:focus+label::before{outline-style:solid;outline-width:2px;outline-style:auto}#onetrust-pc-sdk #ot-selall-vencntr,#onetrust-pc-sdk #ot-selall-adtlvencntr,#onetrust-pc-sdk #ot-selall-hostcntr,#onetrust-pc-sdk #ot-selall-licntr,#onetrust-pc-sdk #ot-selall-gnvencntr{right:15px;position:relative;width:20px;height:20px;float:right}#onetrust-pc-sdk #ot-selall-vencntr label,#onetrust-pc-sdk #ot-selall-adtlvencntr label,#onetrust-pc-sdk #ot-selall-hostcntr label,#onetrust-pc-sdk #ot-selall-licntr label,#onetrust-pc-sdk #ot-selall-gnvencntr label{float:left;padding-left:0}#onetrust-pc-sdk #ot-ven-lst:first-child{border-top:1px solid #e2e2e2}#onetrust-pc-sdk ul{list-style:none;padding:0}#onetrust-pc-sdk ul li{position:relative;margin:0;padding:15px 15px 15px 10px;border-bottom:1px solid #e2e2e2}#onetrust-pc-sdk ul li h3{font-size:.75em;color:#656565;margin:0;display:inline-block;width:70%;height:auto;word-break:break-word;word-wrap:break-word}#onetrust-pc-sdk ul li p{margin:0;font-size:.7em}#onetrust-pc-sdk ul li input[type=checkbox]{position:absolute;cursor:pointer;width:100%;height:100%;opacity:0;margin:0;top:0;left:0}#onetrust-pc-sdk .ot-cat-item>button:focus,#onetrust-pc-sdk .ot-acc-cntr>button:focus,#onetrust-pc-sdk li>button:focus{outline:#000 solid 2px}#onetrust-pc-sdk .ot-cat-item>button,#onetrust-pc-sdk .ot-acc-cntr>button,#onetrust-pc-sdk li>button{position:absolute;cursor:pointer;width:100%;height:100%;margin:0;top:0;left:0;z-index:1;max-width:none;border:none}#onetrust-pc-sdk .ot-cat-item>button[aria-expanded=false]~.ot-acc-txt,#onetrust-pc-sdk .ot-acc-cntr>button[aria-expanded=false]~.ot-acc-txt,#onetrust-pc-sdk li>button[aria-expanded=false]~.ot-acc-txt{margin-top:0;max-height:0;opacity:0;overflow:hidden;width:100%;transition:.25s ease-out;display:none}#onetrust-pc-sdk .ot-cat-item>button[aria-expanded=true]~.ot-acc-txt,#onetrust-pc-sdk .ot-acc-cntr>button[aria-expanded=true]~.ot-acc-txt,#onetrust-pc-sdk li>button[aria-expanded=true]~.ot-acc-txt{transition:.1s ease-in;margin-top:10px;width:100%;overflow:auto;display:block}#onetrust-pc-sdk .ot-cat-item>button[aria-expanded=true]~.ot-acc-grpcntr,#onetrust-pc-sdk .ot-acc-cntr>button[aria-expanded=true]~.ot-acc-grpcntr,#onetrust-pc-sdk li>button[aria-expanded=true]~.ot-acc-grpcntr{width:auto;margin-top:0px;padding-bottom:10px}#onetrust-pc-sdk .ot-host-item>button:focus,#onetrust-pc-sdk .ot-ven-item>button:focus{outline:0;border:2px solid #000}#onetrust-pc-sdk .ot-hide-acc>button{pointer-events:none}#onetrust-pc-sdk .ot-hide-acc .ot-plus-minus>*,#onetrust-pc-sdk .ot-hide-acc .ot-arw-cntr>*{visibility:hidden}#onetrust-pc-sdk .ot-hide-acc .ot-acc-hdr{min-height:30px}#onetrust-pc-sdk.ot-addtl-vendors #ot-lst-cnt:not(.ot-host-cnt){padding-right:10px;width:calc(100% - 37px);margin-top:10px;max-height:calc(100% - 90px)}#onetrust-pc-sdk.ot-addtl-vendors #ot-lst-cnt:not(.ot-host-cnt) #ot-sel-blk{background-color:#f9f9fc;border:1px solid #e2e2e2;width:calc(100% - 2px);padding-bottom:5px;padding-top:5px}#onetrust-pc-sdk.ot-addtl-vendors #ot-lst-cnt:not(.ot-host-cnt) .ot-sel-all{padding-right:34px}#onetrust-pc-sdk.ot-addtl-vendors #ot-lst-cnt:not(.ot-host-cnt) .ot-sel-all-chkbox{width:auto}#onetrust-pc-sdk.ot-addtl-vendors #ot-lst-cnt:not(.ot-host-cnt) ul li{border:1px solid #e2e2e2;margin-bottom:10px}#onetrust-pc-sdk.ot-addtl-vendors #ot-lst-cnt:not(.ot-host-cnt) .ot-acc-cntr>.ot-acc-hdr{padding:10px 0 10px 15px}#onetrust-pc-sdk.ot-addtl-vendors .ot-sel-all-chkbox{float:right}#onetrust-pc-sdk.ot-addtl-vendors .ot-plus-minus~.ot-sel-all-chkbox{right:34px}#onetrust-pc-sdk.ot-addtl-vendors #ot-ven-lst:first-child{border-top:none}#onetrust-pc-sdk .ot-acc-cntr{position:relative;border-left:1px solid #e2e2e2;border-right:1px solid #e2e2e2;border-bottom:1px solid #e2e2e2}#onetrust-pc-sdk .ot-acc-cntr input{z-index:1}#onetrust-pc-sdk .ot-acc-cntr>.ot-acc-hdr{background-color:#f9f9fc;padding:5px 0 5px 15px;width:auto}#onetrust-pc-sdk .ot-acc-cntr>.ot-acc-hdr .ot-plus-minus{vertical-align:middle;top:auto}#onetrust-pc-sdk .ot-acc-cntr>.ot-acc-hdr .ot-arw-cntr{right:10px}#onetrust-pc-sdk .ot-acc-cntr>.ot-acc-hdr input{z-index:2}#onetrust-pc-sdk .ot-acc-cntr>input[type=checkbox]:checked~.ot-acc-hdr{border-bottom:1px solid #e2e2e2}#onetrust-pc-sdk .ot-acc-cntr>.ot-acc-txt{padding-left:10px;padding-right:10px}#onetrust-pc-sdk .ot-acc-cntr button[aria-expanded=true]~.ot-acc-txt{width:auto}#onetrust-pc-sdk .ot-acc-cntr .ot-addtl-venbox{display:none}#onetrust-pc-sdk .ot-vlst-cntr{margin-bottom:0;width:100%}#onetrust-pc-sdk .ot-vensec-title{font-size:.813em;vertical-align:middle;display:inline-block}#onetrust-pc-sdk .category-vendors-list-handler,#onetrust-pc-sdk .category-vendors-list-handler+a{margin-left:0;margin-top:10px}#onetrust-pc-sdk #ot-selall-vencntr.line-through label::after,#onetrust-pc-sdk #ot-selall-adtlvencntr.line-through label::after,#onetrust-pc-sdk #ot-selall-licntr.line-through label::after,#onetrust-pc-sdk #ot-selall-hostcntr.line-through label::after,#onetrust-pc-sdk #ot-selall-gnvencntr.line-through label::after{height:auto;border-left:0;transform:none;-o-transform:none;-ms-transform:none;-webkit-transform:none;left:5px;top:9px}#onetrust-pc-sdk #ot-category-title{float:left;padding-bottom:10px;font-size:1em;width:100%}#onetrust-pc-sdk .ot-cat-grp{margin-top:10px}#onetrust-pc-sdk .ot-cat-item{line-height:1.1;margin-top:10px;display:inline-block;width:100%}#onetrust-pc-sdk .ot-btn-container{text-align:right}#onetrust-pc-sdk .ot-btn-container button{display:inline-block;font-size:.75em;letter-spacing:.08em;margin-top:19px}#onetrust-pc-sdk #close-pc-btn-handler.ot-close-icon{position:absolute;top:10px;right:0;z-index:1;padding:0;background-color:transparent;border:none}#onetrust-pc-sdk #close-pc-btn-handler.ot-close-icon:hover{opacity:.7}#onetrust-pc-sdk #close-pc-btn-handler.ot-close-icon svg{display:block;height:10px;width:10px}#onetrust-pc-sdk #clear-filters-handler{margin-top:20px;margin-bottom:10px;float:right;max-width:200px;text-decoration:none;color:#3860be;font-size:.9em;font-weight:bold;background-color:transparent;border-color:transparent;padding:1px}#onetrust-pc-sdk #clear-filters-handler:hover{color:#2285f7}#onetrust-pc-sdk #clear-filters-handler:focus{outline:#000 solid 1px}#onetrust-pc-sdk .ot-accordion-layout.ot-cat-item{position:relative;border-radius:2px;margin:0;padding:0;border:1px solid #d8d8d8;border-top:none;width:calc(100% - 2px);float:left}#onetrust-pc-sdk .ot-accordion-layout.ot-cat-item:first-of-type{margin-top:10px;border-top:1px solid #d8d8d8}#onetrust-pc-sdk .ot-accordion-layout .ot-acc-grpdesc{padding-left:20px;padding-right:20px;width:calc(100% - 40px);font-size:.812em;margin-bottom:10px;margin-top:15px}#onetrust-pc-sdk .ot-accordion-layout .ot-acc-grpdesc>ul{padding-top:10px}#onetrust-pc-sdk .ot-accordion-layout .ot-acc-grpdesc>ul li{padding-top:0;line-height:1.5;padding-bottom:10px}#onetrust-pc-sdk .ot-accordion-layout div+.ot-acc-grpdesc{margin-top:5px}#onetrust-pc-sdk .ot-accordion-layout .ot-vlst-cntr:first-child{margin-top:10px}#onetrust-pc-sdk .ot-accordion-layout .ot-vlst-cntr:last-child,#onetrust-pc-sdk .ot-accordion-layout .ot-hlst-cntr:last-child{margin-bottom:5px}#onetrust-pc-sdk .ot-accordion-layout .ot-acc-hdr{padding-top:11.5px;padding-bottom:11.5px;padding-left:20px;padding-right:20px;width:calc(100% - 40px);display:inline-block}#onetrust-pc-sdk .ot-accordion-layout .ot-acc-txt{width:100%;padding:0px}#onetrust-pc-sdk .ot-accordion-layout .ot-subgrp-cntr{padding-left:20px;padding-right:15px;padding-bottom:0;width:calc(100% - 35px)}#onetrust-pc-sdk .ot-accordion-layout .ot-subgrp{padding-right:5px}#onetrust-pc-sdk .ot-accordion-layout .ot-acc-grpcntr{z-index:1;position:relative}#onetrust-pc-sdk .ot-accordion-layout .ot-cat-header+.ot-arw-cntr{position:absolute;top:50%;transform:translateY(-50%);right:20px;margin-top:-2px}#onetrust-pc-sdk .ot-accordion-layout .ot-cat-header+.ot-arw-cntr .ot-arw{width:15px;height:20px;margin-left:5px;color:dimgray}#onetrust-pc-sdk .ot-accordion-layout .ot-cat-header{float:none;color:#2e3644;margin:0;display:inline-block;height:auto;word-wrap:break-word;min-height:inherit}#onetrust-pc-sdk .ot-accordion-layout .ot-vlst-cntr,#onetrust-pc-sdk .ot-accordion-layout .ot-hlst-cntr{padding-left:20px;width:calc(100% - 20px);display:inline-block;margin-top:0px;padding-bottom:2px}#onetrust-pc-sdk .ot-accordion-layout .ot-acc-hdr{position:relative;min-height:25px}#onetrust-pc-sdk .ot-accordion-layout h4~.ot-tgl,#onetrust-pc-sdk .ot-accordion-layout h4~.ot-always-active{position:absolute;top:50%;transform:translateY(-50%);right:20px}#onetrust-pc-sdk .ot-accordion-layout h4~.ot-tgl+.ot-tgl{right:95px}#onetrust-pc-sdk .ot-accordion-layout .category-vendors-list-handler,#onetrust-pc-sdk .ot-accordion-layout .category-vendors-list-handler+a{margin-top:5px}#onetrust-pc-sdk .ot-enbl-chr h4~.ot-tgl,#onetrust-pc-sdk .ot-enbl-chr h4~.ot-always-active{right:45px}#onetrust-pc-sdk .ot-enbl-chr h4~.ot-tgl+.ot-tgl{right:120px}#onetrust-pc-sdk .ot-enbl-chr .ot-pli-hdr.ot-leg-border-color span:first-child{width:90px}#onetrust-pc-sdk .ot-enbl-chr li.ot-subgrp>h5+.ot-tgl-cntr{padding-right:25px}#onetrust-pc-sdk .ot-plus-minus{width:20px;height:20px;font-size:1.5em;position:relative;display:inline-block;margin-right:5px;top:3px}#onetrust-pc-sdk .ot-plus-minus span{position:absolute;background:#27455c;border-radius:1px}#onetrust-pc-sdk .ot-plus-minus span:first-of-type{top:25%;bottom:25%;width:10%;left:45%}#onetrust-pc-sdk .ot-plus-minus span:last-of-type{left:25%;right:25%;height:10%;top:45%}#onetrust-pc-sdk button[aria-expanded=true]~.ot-acc-hdr .ot-arw,#onetrust-pc-sdk button[aria-expanded=true]~.ot-acc-hdr .ot-plus-minus span:first-of-type,#onetrust-pc-sdk button[aria-expanded=true]~.ot-acc-hdr .ot-plus-minus span:last-of-type{transform:rotate(90deg)}#onetrust-pc-sdk button[aria-expanded=true]~.ot-acc-hdr .ot-plus-minus span:last-of-type{left:50%;right:50%}#onetrust-pc-sdk #ot-selall-vencntr label,#onetrust-pc-sdk #ot-selall-adtlvencntr label,#onetrust-pc-sdk #ot-selall-hostcntr label,#onetrust-pc-sdk #ot-selall-licntr label{position:relative;display:inline-block;width:20px;height:20px}#onetrust-pc-sdk .ot-host-item .ot-plus-minus,#onetrust-pc-sdk .ot-ven-item .ot-plus-minus{float:left;margin-right:8px;top:10px}#onetrust-pc-sdk .ot-pli-hdr{color:#77808e;overflow:hidden;padding-top:7.5px;padding-bottom:7.5px;width:calc(100% - 2px);border-top-left-radius:3px;border-top-right-radius:3px}#onetrust-pc-sdk .ot-pli-hdr span:first-child{top:50%;transform:translateY(50%);max-width:90px}#onetrust-pc-sdk .ot-pli-hdr span:last-child{padding-right:10px;max-width:95px;text-align:center}#onetrust-pc-sdk .ot-li-title{float:right;font-size:.813em}#onetrust-pc-sdk .ot-pli-hdr.ot-leg-border-color{background-color:#f4f4f4;border:1px solid #d8d8d8}#onetrust-pc-sdk .ot-pli-hdr.ot-leg-border-color span:first-child{text-align:left;width:70px}#onetrust-pc-sdk li.ot-subgrp>h5,#onetrust-pc-sdk .ot-cat-header{width:calc(100% - 130px)}#onetrust-pc-sdk li.ot-subgrp>h5+.ot-tgl-cntr{padding-left:13px}#onetrust-pc-sdk .ot-acc-grpcntr .ot-acc-grpdesc{margin-bottom:5px}#onetrust-pc-sdk .ot-acc-grpcntr .ot-subgrp-cntr{border-top:1px solid #d8d8d8}#onetrust-pc-sdk .ot-acc-grpcntr .ot-vlst-cntr+.ot-subgrp-cntr{border-top:none}#onetrust-pc-sdk .ot-acc-hdr .ot-arw-cntr+.ot-tgl-cntr,#onetrust-pc-sdk .ot-acc-txt h4+.ot-tgl-cntr{padding-left:13px}#onetrust-pc-sdk .ot-pli-hdr~.ot-cat-item .ot-subgrp>h5,#onetrust-pc-sdk .ot-pli-hdr~.ot-cat-item .ot-cat-header{width:calc(100% - 145px)}#onetrust-pc-sdk .ot-pli-hdr~.ot-cat-item h5+.ot-tgl-cntr,#onetrust-pc-sdk .ot-pli-hdr~.ot-cat-item .ot-cat-header+.ot-tgl{padding-left:28px}#onetrust-pc-sdk .ot-sel-all-hdr,#onetrust-pc-sdk .ot-sel-all-chkbox{display:inline-block;width:100%;position:relative}#onetrust-pc-sdk .ot-sel-all-chkbox{z-index:1}#onetrust-pc-sdk .ot-sel-all{margin:0;position:relative;padding-right:23px;float:right}#onetrust-pc-sdk .ot-consent-hdr,#onetrust-pc-sdk .ot-li-hdr{float:right;font-size:.812em;line-height:normal;text-align:center;word-break:break-word;word-wrap:break-word}#onetrust-pc-sdk .ot-li-hdr{max-width:100px;padding-right:10px}#onetrust-pc-sdk .ot-consent-hdr{max-width:55px}#onetrust-pc-sdk #ot-selall-licntr{display:block;width:21px;height:auto;float:right;position:relative;right:80px}#onetrust-pc-sdk #ot-selall-licntr label{position:absolute}#onetrust-pc-sdk .ot-ven-ctgl{margin-left:66px}#onetrust-pc-sdk .ot-ven-litgl+.ot-arw-cntr{margin-left:81px}#onetrust-pc-sdk .ot-enbl-chr .ot-host-cnt .ot-tgl-cntr{width:auto}#onetrust-pc-sdk #ot-lst-cnt:not(.ot-host-cnt) .ot-tgl-cntr{width:auto;top:auto;height:20px}#onetrust-pc-sdk #ot-lst-cnt .ot-chkbox{position:relative;display:inline-block;width:20px;height:20px}#onetrust-pc-sdk #ot-lst-cnt .ot-chkbox label{position:absolute;padding:0;width:20px;height:20px}#onetrust-pc-sdk .ot-acc-grpdesc+.ot-leg-btn-container{padding-left:20px;padding-right:20px;width:calc(100% - 40px);margin-bottom:5px}#onetrust-pc-sdk .ot-subgrp .ot-leg-btn-container{margin-bottom:5px}#onetrust-pc-sdk #ot-ven-lst .ot-leg-btn-container{margin-top:10px}#onetrust-pc-sdk .ot-leg-btn-container{display:inline-block;width:100%;margin-bottom:10px}#onetrust-pc-sdk .ot-leg-btn-container button{height:auto;padding:6.5px 8px;margin-bottom:0;letter-spacing:0;font-size:.75em;line-height:normal}#onetrust-pc-sdk .ot-leg-btn-container svg{display:none;height:14px;width:14px;padding-right:5px;vertical-align:sub}#onetrust-pc-sdk .ot-active-leg-btn{cursor:default;pointer-events:none}#onetrust-pc-sdk .ot-active-leg-btn svg{display:inline-block}#onetrust-pc-sdk .ot-remove-objection-handler{text-decoration:underline;padding:0;font-size:.75em;font-weight:600;line-height:1;padding-left:10px}#onetrust-pc-sdk .ot-obj-leg-btn-handler span{font-weight:bold;text-align:center;font-size:inherit;line-height:1.5}#onetrust-pc-sdk.ot-close-btn-link #close-pc-btn-handler{border:none;height:auto;line-height:1.5;text-decoration:underline;font-size:.69em;background:none;right:15px;top:15px;width:auto;font-weight:normal}#onetrust-pc-sdk[dir=rtl] #ot-back-arw,#onetrust-pc-sdk[dir=rtl] input~.ot-acc-hdr .ot-arw{transform:rotate(180deg);-o-transform:rotate(180deg);-ms-transform:rotate(180deg);-webkit-transform:rotate(180deg)}#onetrust-pc-sdk[dir=rtl] input:checked~.ot-acc-hdr .ot-arw{transform:rotate(270deg);-o-transform:rotate(270deg);-ms-transform:rotate(270deg);-webkit-transform:rotate(270deg)}#onetrust-pc-sdk[dir=rtl] .ot-chkbox label::after{transform:rotate(45deg);-webkit-transform:rotate(45deg);-o-transform:rotate(45deg);-ms-transform:rotate(45deg);border-left:0;border-right:3px solid}#onetrust-pc-sdk[dir=rtl] .ot-search-cntr>svg{right:0}@media only screen and (max-width: 600px){#onetrust-pc-sdk.otPcCenter{left:0;min-width:100%;height:100%;top:0;border-radius:0}#onetrust-pc-sdk #ot-pc-content,#onetrust-pc-sdk.ot-ftr-stacked .ot-btn-container{margin:1px 3px 0 10px;padding-right:10px;width:calc(100% - 23px)}#onetrust-pc-sdk .ot-btn-container button{max-width:none;letter-spacing:.01em}#onetrust-pc-sdk #close-pc-btn-handler{top:10px;right:17px}#onetrust-pc-sdk p{font-size:.7em}#onetrust-pc-sdk #ot-pc-hdr{margin:10px 10px 0 5px;width:calc(100% - 15px)}#onetrust-pc-sdk .vendor-search-handler{font-size:1em}#onetrust-pc-sdk #ot-back-arw{margin-left:12px}#onetrust-pc-sdk #ot-lst-cnt{margin:0;padding:0 5px 0 10px;min-width:95%}#onetrust-pc-sdk .switch+p{max-width:80%}#onetrust-pc-sdk .ot-ftr-stacked button{width:100%}#onetrust-pc-sdk #ot-fltr-cnt{max-width:320px;width:90%;border-top-right-radius:0;border-bottom-right-radius:0;margin:0;margin-left:15px;left:auto;right:40px;top:85px}#onetrust-pc-sdk .ot-fltr-opt{margin-left:25px;margin-bottom:10px}#onetrust-pc-sdk .ot-pc-refuse-all-handler{margin-bottom:0}#onetrust-pc-sdk #ot-fltr-cnt{right:40px}}@media only screen and (max-width: 476px){#onetrust-pc-sdk .ot-fltr-cntr,#onetrust-pc-sdk #ot-fltr-cnt{right:10px}#onetrust-pc-sdk #ot-anchor{right:25px}#onetrust-pc-sdk button{width:100%}#onetrust-pc-sdk:not(.ot-addtl-vendors) #ot-pc-lst:not(.ot-enbl-chr) .ot-sel-all{padding-right:9px}#onetrust-pc-sdk:not(.ot-addtl-vendors) #ot-pc-lst:not(.ot-enbl-chr) .ot-tgl-cntr{right:0}}@media only screen and (max-width: 896px)and (max-height: 425px)and (orientation: landscape){#onetrust-pc-sdk.otPcCenter{left:0;top:0;min-width:100%;height:100%;border-radius:0}#onetrust-pc-sdk #ot-anchor{left:initial;right:50px}#onetrust-pc-sdk #ot-lst-title{margin-top:12px}#onetrust-pc-sdk #ot-lst-title *{font-size:inherit}#onetrust-pc-sdk #ot-pc-hdr input{margin-right:0;padding-right:45px}#onetrust-pc-sdk .switch+p{max-width:85%}#onetrust-pc-sdk #ot-sel-blk{position:static}#onetrust-pc-sdk #ot-pc-lst{overflow:auto}#onetrust-pc-sdk .ot-pc-footer-logo{display:none}#onetrust-pc-sdk #ot-lst-cnt{max-height:none;overflow:initial}#onetrust-pc-sdk #ot-lst-cnt.no-results{height:auto}#onetrust-pc-sdk input{font-size:1em !important}#onetrust-pc-sdk p{font-size:.6em}#onetrust-pc-sdk #ot-fltr-modal{width:100%;top:0}#onetrust-pc-sdk ul li p,#onetrust-pc-sdk .category-vendors-list-handler,#onetrust-pc-sdk .category-vendors-list-handler+a,#onetrust-pc-sdk .category-host-list-handler{font-size:.6em}#onetrust-pc-sdk.ot-shw-fltr #ot-anchor{display:none !important}#onetrust-pc-sdk.ot-shw-fltr #ot-pc-lst{height:100% !important;overflow:hidden;top:0px}#onetrust-pc-sdk.ot-shw-fltr #ot-fltr-cnt{margin:0;height:100%;max-height:none;padding:10px;top:0;width:calc(100% - 20px);position:absolute;right:0;left:0;max-width:none}#onetrust-pc-sdk.ot-shw-fltr .ot-fltr-scrlcnt{max-height:calc(100% - 65px)}}\n",
       "            #onetrust-consent-sdk #onetrust-pc-sdk,\n",
       "                #onetrust-consent-sdk #ot-search-cntr,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk .ot-switch.ot-toggle,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk ot-grp-hdr1 .checkbox,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-title:after\n",
       "                ,#onetrust-consent-sdk #onetrust-pc-sdk #ot-sel-blk,\n",
       "                        #onetrust-consent-sdk #onetrust-pc-sdk #ot-fltr-cnt,\n",
       "                        #onetrust-consent-sdk #onetrust-pc-sdk #ot-anchor {\n",
       "                    background-color: #FFFFFF;\n",
       "                }\n",
       "               \n",
       "            #onetrust-consent-sdk #onetrust-pc-sdk h3,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk h4,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk h5,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk h6,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk p,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk #ot-ven-lst .ot-ven-opts p,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-desc,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-title,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk .ot-li-title,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk .ot-sel-all-hdr span,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk #ot-host-lst .ot-host-info,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk #ot-fltr-modal #modal-header,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk .ot-checkbox label span,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-lst #ot-sel-blk p,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-lst #ot-lst-title span,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-lst .back-btn-handler p,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-lst .ot-ven-name,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-lst #ot-ven-lst .consent-category,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk .ot-leg-btn-container .ot-inactive-leg-btn,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk .ot-label-status,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk .ot-chkbox label span,\n",
       "                #onetrust-consent-sdk #onetrust-pc-sdk #clear-filters-handler \n",
       "                {\n",
       "                    color: #696969;\n",
       "                }\n",
       "             #onetrust-consent-sdk #onetrust-pc-sdk .privacy-notice-link,\n",
       "                    #onetrust-consent-sdk #onetrust-pc-sdk .category-vendors-list-handler,\n",
       "                    #onetrust-consent-sdk #onetrust-pc-sdk .category-vendors-list-handler + a,\n",
       "                    #onetrust-consent-sdk #onetrust-pc-sdk .category-host-list-handler,\n",
       "                    #onetrust-consent-sdk #onetrust-pc-sdk .ot-ven-link,\n",
       "                    #onetrust-consent-sdk #onetrust-pc-sdk #ot-host-lst .ot-host-name a,\n",
       "                    #onetrust-consent-sdk #onetrust-pc-sdk #ot-host-lst .ot-acc-hdr .ot-host-expand,\n",
       "                    #onetrust-consent-sdk #onetrust-pc-sdk #ot-host-lst .ot-host-info a\n",
       "                    {\n",
       "                        color: #00a0d2;\n",
       "                    }\n",
       "            #onetrust-consent-sdk #onetrust-pc-sdk .category-vendors-list-handler:hover { opacity: .7;}\n",
       "            #onetrust-consent-sdk #onetrust-pc-sdk .ot-acc-grpcntr.ot-acc-txt,\n",
       "            #onetrust-consent-sdk #onetrust-pc-sdk .ot-acc-txt .ot-subgrp-tgl .ot-switch.ot-toggle\n",
       "             {\n",
       "                background-color: #F8F8F8;\n",
       "            }\n",
       "             #onetrust-consent-sdk #onetrust-pc-sdk #ot-host-lst .ot-host-info,\n",
       "                    #onetrust-consent-sdk #onetrust-pc-sdk .ot-acc-txt .ot-ven-dets\n",
       "                            {\n",
       "                                background-color: #F8F8F8;\n",
       "                            }\n",
       "        #onetrust-consent-sdk #onetrust-pc-sdk \n",
       "            button:not(#clear-filters-handler):not(.ot-close-icon):not(#filter-btn-handler):not(.ot-remove-objection-handler):not(.ot-obj-leg-btn-handler):not([aria-expanded]):not(.ot-link-btn),\n",
       "            #onetrust-consent-sdk #onetrust-pc-sdk .ot-leg-btn-container .ot-active-leg-btn {\n",
       "                background-color: #00a0d2;border-color: #00a0d2;\n",
       "                color: #FFFFFF;\n",
       "            }\n",
       "            #onetrust-consent-sdk #onetrust-pc-sdk .ot-active-menu {\n",
       "                border-color: #00a0d2;\n",
       "            }\n",
       "            \n",
       "            #onetrust-consent-sdk #onetrust-pc-sdk .ot-leg-btn-container .ot-remove-objection-handler{\n",
       "                background-color: transparent;\n",
       "                border:1px solid transparent;\n",
       "            }\n",
       "            #onetrust-consent-sdk #onetrust-pc-sdk .ot-leg-btn-container .ot-inactive-leg-btn {\n",
       "                background-color: #FFFFFF;\n",
       "                color: #78808E; border-color: #78808E;\n",
       "            }\n",
       "            #onetrust-consent-sdk{font-family:MuseoSans,Helvetica,Arial,sans-serif!important;box-shadow:0 -12px 17px rgba(40,40,40,.12),0 -5px 22px rgba(40,40,40,.08),0 -7px 8px rgba(40,40,40,.16)}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group{margin-right:0;margin-left:auto;display:flex}@media only screen and (max-width:768px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group{width:100%}}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group{flex-flow:column;flex-direction:column-reverse}}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group-parent{display:flex;position:relative;top:unset;left:unset;align-items:center;transform:none;margin:0;min-width:490px}@media only screen and (max-width:1280px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group-parent{padding-right:32px}}@media only screen and (max-width:768px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group-parent{padding:0}}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group-parent{min-width:auto}}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button{--padding-horizontal:24px;--padding-iconButton:32px;--padding-vertical:8px;font-size:1rem;line-height:1.375rem;font-weight:300;align-items:center;background-color:var(--maincolor40,#0c4ded);border-radius:32px;border:1px solid var(--maincolor40,#0c4ded);box-sizing:border-box;color:var(--white,#fff);display:inline-flex;justify-content:center;padding:var(--padding-vertical) var(--padding-horizontal);position:relative;margin:0;height:40px}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button:hover{background-color:var(--maincolor50,#003bde);border-color:var(--maincolor50,#003bde)}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button:focus{outline:0}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button.cookie-setting-link{margin-right:24px;text-decoration:none;background-color:transparent!important;border-color:var(--maincolor40,#0c4ded)!important;color:var(--maincolor40,#0c4ded)!important}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button.cookie-setting-link:hover{color:var(--maincolor50,#003bde)!important;border-color:var(--maincolor50,#003bde)!important}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button.cookie-setting-link{margin:0}}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button#onetrust-accept-btn-handler{margin-right:72px}@media only screen and (max-width:1024px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button#onetrust-accept-btn-handler{margin-right:40px}}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button#onetrust-accept-btn-handler{margin:0 0 16px}}@media only screen and (max-width:768px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-button-group>button{width:100%}}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container .ot-close-icon,#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container-mobile .ot-close-icon{background-image:url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4NCjxwYXRoIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNMTIuMDIyIDEzLjM0N0w1LjIxODAzIDIwLjEyODVDNS4xNzM0NCAyMC4xNzMyIDUuMTIwNDkgMjAuMjA4NiA1LjA2MjE5IDIwLjIzMjhDNS4wMDM4OSAyMC4yNTcgNC45NDEzOSAyMC4yNjk0IDQuODc4MjggMjAuMjY5NEM0LjgxNTE2IDIwLjI2OTQgNC43NTI2NyAyMC4yNTcgNC42OTQzNyAyMC4yMzI4QzQuNjM2MDcgMjAuMjA4NiA0LjU4MzExIDIwLjE3MzIgNC41Mzg1MyAyMC4xMjg1TDMuODYwNTMgMTkuNDQ5QzMuODE1OSAxOS40MDQ1IDMuNzgwNDkgMTkuMzUxNyAzLjc1NjMzIDE5LjI5MzVDMy43MzIxNyAxOS4yMzU0IDMuNzE5NzMgMTkuMTczIDMuNzE5NzMgMTkuMTFDMy43MTk3MyAxOS4wNDcgMy43MzIxNyAxOC45ODQ2IDMuNzU2MzMgMTguOTI2NUMzLjc4MDQ5IDE4Ljg2ODMgMy44MTU5IDE4LjgxNTUgMy44NjA1MyAxOC43NzFMMTAuNjY0NSAxMS45ODk1TDMuODc3MDMgNS4yMzIwMUMzLjgzMjQgNS4xODc1NSAzLjc5Njk5IDUuMTM0NzMgMy43NzI4MyA1LjA3NjU1QzMuNzQ4NjcgNS4wMTgzOCAzLjczNjIzIDQuOTU2MDEgMy43MzYyMyA0Ljg5MzAxQzMuNzM2MjMgNC44MzAwMiAzLjc0ODY3IDQuNzY3NjUgMy43NzI4MyA0LjcwOTQ3QzMuNzk2OTkgNC42NTEzIDMuODMyNCA0LjU5ODQ3IDMuODc3MDMgNC41NTQwMkw0LjU1NjUzIDMuODc0NTFDNC42MDA5OCAzLjgyOTg4IDQuNjUzODIgMy43OTQ0NiA0LjcxMTk5IDMuNzcwM0M0Ljc3MDE2IDMuNzQ2MTQgNC44MzI1MyAzLjczMzcgNC44OTU1MyAzLjczMzdDNC45NTg1MiAzLjczMzcgNS4wMjA4OSAzLjc0NjE0IDUuMDc5MDYgMy43NzAzQzUuMTM3MjQgMy43OTQ0NiA1LjE5MDA3IDMuODI5ODggNS4yMzQ1MyAzLjg3NDUxTDEyLjAyMiAxMC42MzJMMTguNzgyNSAzLjg3MTUxQzE4LjgyNzEgMy44MjY4NCAxOC44ODAxIDMuNzkxMzkgMTguOTM4NCAzLjc2NzIxQzE4Ljk5NjcgMy43NDMwMyAxOS4wNTkyIDMuNzMwNTkgMTkuMTIyMyAzLjczMDU5QzE5LjE4NTQgMy43MzA1OSAxOS4yNDc5IDMuNzQzMDMgMTkuMzA2MiAzLjc2NzIxQzE5LjM2NDUgMy43OTEzOSAxOS40MTc0IDMuODI2ODQgMTkuNDYyIDMuODcxNTFMMjAuMTQgNC41NTEwMkMyMC4xODQ3IDQuNTk1NDcgMjAuMjIwMSA0LjY0ODMgMjAuMjQ0MiA0LjcwNjQ3QzIwLjI2ODQgNC43NjQ2NSAyMC4yODA4IDQuODI3MDIgMjAuMjgwOCA0Ljg5MDAxQzIwLjI4MDggNC45NTMwMSAyMC4yNjg0IDUuMDE1MzggMjAuMjQ0MiA1LjA3MzU1QzIwLjIyMDEgNS4xMzE3MyAyMC4xODQ3IDUuMTg0NTUgMjAuMTQgNS4yMjkwMUwxMy4zNzk1IDExLjk4OTVMMjAuMTI5NSAxOC43NzFDMjAuMTc0MiAxOC44MTU1IDIwLjIwOTYgMTguODY4MyAyMC4yMzM3IDE4LjkyNjVDMjAuMjU3OSAxOC45ODQ2IDIwLjI3MDMgMTkuMDQ3IDIwLjI3MDMgMTkuMTFDMjAuMjcwMyAxOS4xNzMgMjAuMjU3OSAxOS4yMzU0IDIwLjIzMzcgMTkuMjkzNUMyMC4yMDk2IDE5LjM1MTcgMjAuMTc0MiAxOS40MDQ1IDIwLjEyOTUgMTkuNDQ5TDE5LjQ1IDIwLjEyODVDMTkuNDA1NiAyMC4xNzMxIDE5LjM1MjcgMjAuMjA4NiAxOS4yOTQ2IDIwLjIzMjdDMTkuMjM2NCAyMC4yNTY5IDE5LjE3NCAyMC4yNjkzIDE5LjExMSAyMC4yNjkzQzE5LjA0OCAyMC4yNjkzIDE4Ljk4NTcgMjAuMjU2OSAxOC45Mjc1IDIwLjIzMjdDMTguODY5MyAyMC4yMDg2IDE4LjgxNjUgMjAuMTczMSAxOC43NzIgMjAuMTI4NUwxMi4wMjIgMTMuMzQ3WiIgZmlsbD0iIzU0NTQ1NCIvPg0KPC9zdmc+DQo=);background-position:center;background-repeat:no-repeat;background-size:20px;margin:0}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container{top:50%;margin:auto;transform:translate(-50%,-50%);position:absolute;right:0}@media only screen and (max-width:768px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container{display:none}}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container .ot-close-icon{top:auto;right:-25px}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container-mobile{position:absolute;right:-12px;top:8px}@media only screen and (max-width:768px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-close-btn-container-mobile{display:inline;float:none}}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-policy{margin:0;padding:32px 40px 32px 16px}@media only screen and (max-width:768px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-policy{padding:24px 0}}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-policy h3{color:var(--black,#282828);font-size:1.125rem;line-height:1.75rem;font-weight:500;margin:0 0 8px}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-policy p{font-size:1rem;line-height:1.375rem;font-weight:300;margin:0}@media only screen and (max-width:1441px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-policy p{font-size:.875rem;line-height:1.25rem}}@media only screen and (max-width:768px){#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-policy p{font-size:1rem;line-height:1.375rem}}#onetrust-consent-sdk #onetrust-banner-sdk #onetrust-group-container{width:auto;position:relative}#onetrust-consent-sdk #onetrust-pc-sdk.otPcCenter{border-radius:8px}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-pc-sdk.otPcCenter{left:0;min-width:100%;height:100%;top:0;border-radius:0;height:100%!important}}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-header{border:none;height:0;padding:0}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-header .ot-close-icon{background-image:url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4NCjxwYXRoIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNMTIuMDIyIDEzLjM0N0w1LjIxODAzIDIwLjEyODVDNS4xNzM0NCAyMC4xNzMyIDUuMTIwNDkgMjAuMjA4NiA1LjA2MjE5IDIwLjIzMjhDNS4wMDM4OSAyMC4yNTcgNC45NDEzOSAyMC4yNjk0IDQuODc4MjggMjAuMjY5NEM0LjgxNTE2IDIwLjI2OTQgNC43NTI2NyAyMC4yNTcgNC42OTQzNyAyMC4yMzI4QzQuNjM2MDcgMjAuMjA4NiA0LjU4MzExIDIwLjE3MzIgNC41Mzg1MyAyMC4xMjg1TDMuODYwNTMgMTkuNDQ5QzMuODE1OSAxOS40MDQ1IDMuNzgwNDkgMTkuMzUxNyAzLjc1NjMzIDE5LjI5MzVDMy43MzIxNyAxOS4yMzU0IDMuNzE5NzMgMTkuMTczIDMuNzE5NzMgMTkuMTFDMy43MTk3MyAxOS4wNDcgMy43MzIxNyAxOC45ODQ2IDMuNzU2MzMgMTguOTI2NUMzLjc4MDQ5IDE4Ljg2ODMgMy44MTU5IDE4LjgxNTUgMy44NjA1MyAxOC43NzFMMTAuNjY0NSAxMS45ODk1TDMuODc3MDMgNS4yMzIwMUMzLjgzMjQgNS4xODc1NSAzLjc5Njk5IDUuMTM0NzMgMy43NzI4MyA1LjA3NjU1QzMuNzQ4NjcgNS4wMTgzOCAzLjczNjIzIDQuOTU2MDEgMy43MzYyMyA0Ljg5MzAxQzMuNzM2MjMgNC44MzAwMiAzLjc0ODY3IDQuNzY3NjUgMy43NzI4MyA0LjcwOTQ3QzMuNzk2OTkgNC42NTEzIDMuODMyNCA0LjU5ODQ3IDMuODc3MDMgNC41NTQwMkw0LjU1NjUzIDMuODc0NTFDNC42MDA5OCAzLjgyOTg4IDQuNjUzODIgMy43OTQ0NiA0LjcxMTk5IDMuNzcwM0M0Ljc3MDE2IDMuNzQ2MTQgNC44MzI1MyAzLjczMzcgNC44OTU1MyAzLjczMzdDNC45NTg1MiAzLjczMzcgNS4wMjA4OSAzLjc0NjE0IDUuMDc5MDYgMy43NzAzQzUuMTM3MjQgMy43OTQ0NiA1LjE5MDA3IDMuODI5ODggNS4yMzQ1MyAzLjg3NDUxTDEyLjAyMiAxMC42MzJMMTguNzgyNSAzLjg3MTUxQzE4LjgyNzEgMy44MjY4NCAxOC44ODAxIDMuNzkxMzkgMTguOTM4NCAzLjc2NzIxQzE4Ljk5NjcgMy43NDMwMyAxOS4wNTkyIDMuNzMwNTkgMTkuMTIyMyAzLjczMDU5QzE5LjE4NTQgMy43MzA1OSAxOS4yNDc5IDMuNzQzMDMgMTkuMzA2MiAzLjc2NzIxQzE5LjM2NDUgMy43OTEzOSAxOS40MTc0IDMuODI2ODQgMTkuNDYyIDMuODcxNTFMMjAuMTQgNC41NTEwMkMyMC4xODQ3IDQuNTk1NDcgMjAuMjIwMSA0LjY0ODMgMjAuMjQ0MiA0LjcwNjQ3QzIwLjI2ODQgNC43NjQ2NSAyMC4yODA4IDQuODI3MDIgMjAuMjgwOCA0Ljg5MDAxQzIwLjI4MDggNC45NTMwMSAyMC4yNjg0IDUuMDE1MzggMjAuMjQ0MiA1LjA3MzU1QzIwLjIyMDEgNS4xMzE3MyAyMC4xODQ3IDUuMTg0NTUgMjAuMTQgNS4yMjkwMUwxMy4zNzk1IDExLjk4OTVMMjAuMTI5NSAxOC43NzFDMjAuMTc0MiAxOC44MTU1IDIwLjIwOTYgMTguODY4MyAyMC4yMzM3IDE4LjkyNjVDMjAuMjU3OSAxOC45ODQ2IDIwLjI3MDMgMTkuMDQ3IDIwLjI3MDMgMTkuMTFDMjAuMjcwMyAxOS4xNzMgMjAuMjU3OSAxOS4yMzU0IDIwLjIzMzcgMTkuMjkzNUMyMC4yMDk2IDE5LjM1MTcgMjAuMTc0MiAxOS40MDQ1IDIwLjEyOTUgMTkuNDQ5TDE5LjQ1IDIwLjEyODVDMTkuNDA1NiAyMC4xNzMxIDE5LjM1MjcgMjAuMjA4NiAxOS4yOTQ2IDIwLjIzMjdDMTkuMjM2NCAyMC4yNTY5IDE5LjE3NCAyMC4yNjkzIDE5LjExMSAyMC4yNjkzQzE5LjA0OCAyMC4yNjkzIDE4Ljk4NTcgMjAuMjU2OSAxOC45Mjc1IDIwLjIzMjdDMTguODY5MyAyMC4yMDg2IDE4LjgxNjUgMjAuMTczMSAxOC43NzIgMjAuMTI4NUwxMi4wMjIgMTMuMzQ3WiIgZmlsbD0iIzU0NTQ1NCIvPg0KPC9zdmc+DQo=);background-position:center;background-repeat:no-repeat;background-size:20px;z-index:8}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-logo{display:none}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer{border:none;background-color:var(--white,#fff);z-index:1}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer .ot-btn-container{padding:16px}}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer-logo{padding:16px 32px;box-sizing:border-box;height:48px}@media only screen and (max-width:1280px){#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer-logo{padding:16px}}@media only screen and (max-width:768px){#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer-logo{padding:16px 32px}}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer-logo{padding:8px 16px;height:32px}}#onetrust-consent-sdk #onetrust-pc-sdk .ot-pc-footer-logo a{margin:0;width:155px;height:16px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item{border-color:var(--grey40,#d5d5d5)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item button{outline:0}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item:first-of-type{border-top-left-radius:8px;border-top-right-radius:8px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item:last-of-type{border-bottom-left-radius:8px;border-bottom-right-radius:8px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item .ot-cat-header{font-size:1rem;line-height:1.375rem;font-weight:500;color:var(--grey80,#545454)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item .ot-category-desc{font-size:1rem;line-height:1.375rem;font-weight:300;color:var(--grey80,#545454)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-accordion-layout.ot-cat-item .ot-acc-grpcntr.ot-acc-txt{background-color:var(--white,#fff)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-switch-nob{background-color:var(--grey50,#bfbfbf);border-color:var(--grey50,#bfbfbf)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-switch-nob::before{background-color:var(--white,#fff)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-tgl input:checked+.ot-switch .ot-switch-nob{background-color:var(--black,#282828);border:1px solid var(--black,#282828)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-tgl input:checked+.ot-switch .ot-switch-nob::before{background-color:var(--white,#fff)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container{text-align:left}@media only screen and (max-width:768px){#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container{display:flex}}#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container .save-preference-btn-handler.onetrust-close-btn-handler{--padding-horizontal:24px;--padding-iconButton:32px;--padding-vertical:8px;font-size:1rem;line-height:1.375rem;font-weight:300;align-items:center;background-color:var(--maincolor40,#0c4ded);border-radius:32px;border:1px solid var(--maincolor40,#0c4ded);box-sizing:border-box;color:var(--white,#fff);display:inline-flex;justify-content:center;padding:var(--padding-vertical) var(--padding-horizontal);position:relative;margin:0;height:40px;margin:16px 32px;background-color:var(--maincolor40,#0c4ded)!important;border-color:var(--maincolor40,#0c4ded)!important;color:var(--white,#fff)!important}#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container .save-preference-btn-handler.onetrust-close-btn-handler:hover{background-color:var(--maincolor50,#003bde);border-color:var(--maincolor50,#003bde)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container .save-preference-btn-handler.onetrust-close-btn-handler:focus{outline:0}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container .save-preference-btn-handler.onetrust-close-btn-handler{margin:0}}@media only screen and (max-width:768px){#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container>button{width:100%;max-width:calc(50% - 16px)}#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container>button:first-child{margin-right:16px}}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container>button{max-width:none}}#onetrust-consent-sdk #onetrust-pc-sdk .ot-btn-container>:first-child{margin-left:0}#onetrust-consent-sdk #onetrust-pc-sdk .ot-cat-grp{margin-bottom:40px}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-pc-sdk .ot-cat-grp{margin-bottom:32px}}#onetrust-consent-sdk #onetrust-pc-sdk .ot-cat-grp .ot-acc-hdr .ot-plus-minus{background-image:url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIGhlaWdodD0iMTYiIHZpZXdCb3g9IjAgMCAxNiAxNiIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4NCjxwYXRoIGQ9Ik01LjY0MDcyIDIuMDk3TDUuMTg4NzIgMi41NDhDNS4xNTg4MSAyLjU3NzUzIDUuMTM1MSAyLjYxMjczIDUuMTE4OTYgMi42NTE1NEM1LjEwMjgyIDIuNjkwMzQgNS4wOTQ1OCAyLjczMTk3IDUuMDk0NzMgMi43NzRDNS4wOTQ2NyAyLjgxNjE3IDUuMTAyODMgMi44NTc5NSA1LjExODczIDIuODk3QzUuMTM1MzkgMi45MzQ1NCA1LjE1OTE1IDIuOTY4NTEgNS4xODg3MiAyLjk5NzAxTDEwLjE4ODcgNy45OTcwMUw1LjE4ODcyIDEyLjk5N0M1LjE1OTQ1IDEzLjAyNTUgNS4xMzYwMiAxMy4wNTk1IDUuMTE5NzIgMTMuMDk3QzUuMTAzNSAxMy4xMzYgNS4wOTUwMiAxMy4xNzc4IDUuMDk0NzMgMTMuMjJDNS4wOTUwMiAxMy4yNjE5IDUuMTAzNSAxMy4zMDM0IDUuMTE5NzIgMTMuMzQyQzUuMTM2MDIgMTMuMzc5NSA1LjE1OTQ1IDEzLjQxMzUgNS4xODg3MiAxMy40NDJMNS42NDE3MyAxMy44OTNDNS43MDE2NyAxMy45NTI1IDUuNzgyNzMgMTMuOTg2IDUuODY3MjIgMTMuOTg2QzUuOTUxNzEgMTMuOTg2IDYuMDMyNzggMTMuOTUyNSA2LjA5MjczIDEzLjg5M0wxMS45OTI3IDcuOTkzMDFMNi4wODU3MyAyLjA5M0M2LjAyNTYxIDIuMDMzNCA1Ljk0NDM4IDEuOTk5OTcgNS44NTk3MyAxLjk5OTk3QzUuNzc1MDcgMS45OTk5NyA1LjY5Mzg1IDIuMDMzNCA1LjYzMzczIDIuMDkzTDUuNjQwNzIgMi4wOTdaIiBmaWxsPSIjNTQ1NDU0Ii8+DQo8L3N2Zz4NCg==);background-repeat:no-repeat;background-position:center;margin-right:8px}#onetrust-consent-sdk #onetrust-pc-sdk .ot-cat-grp .ot-acc-hdr .ot-plus-minus span{display:none}#onetrust-consent-sdk #onetrust-pc-sdk .ot-cat-grp button[aria-expanded=true]~.ot-acc-hdr .ot-plus-minus{transform:rotate(90deg)}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-title{font-size:1.25rem;line-height:1.75rem;font-weight:700;color:var(--black,#282828)!important;position:sticky;z-index:5;top:0;left:0;margin:0;padding:32px 0 8px;background-color:var(--white,#fff)}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-title{font-size:1rem;line-height:1.375rem;padding:24px 0}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-title::before{content:\"\";position:absolute;display:block;height:1px;left:-16px;right:-16px;bottom:0;background-color:var(--grey40,#d5d5d5)}}#onetrust-consent-sdk #onetrust-pc-sdk #ot-category-title{font-size:1.125rem;line-height:1.75rem;font-weight:500;color:var(--black,#282828)!important;margin:56px 0 24px}@media only screen and (max-width:1441px){#onetrust-consent-sdk #onetrust-pc-sdk #ot-category-title{font-size:1rem;line-height:1.375rem}}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content{margin:0;padding:0 32px;width:calc(100% - 68px);position:relative;top:0;height:calc(100% - 140px)}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content::-webkit-scrollbar{background:var(--white);-webkit-appearance:none;width:4px;height:4px}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content::-webkit-scrollbar-thumb{border-radius:0;background-color:var(--grey50);-webkit-box-shadow:0 0 1px rgba(255,255,255,.5);border-radius:4px;border:none}@-moz-document url-prefix(){#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content{scrollbar-width:thin}}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content{padding:0 16px;width:calc(100% - 34px)}}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content .privacy-notice-link{text-decoration:none;color:var(--maincolor50,#003bde);font-size:1rem;line-height:1.375rem;font-weight:500;position:relative;margin-top:16px;display:block}@media only screen and (max-width:1024px){#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content .privacy-notice-link{font-size:.875rem;line-height:1.25rem}}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content #accept-recommended-btn-handler{--padding-horizontal:24px;--padding-iconButton:32px;--padding-vertical:8px;font-size:1rem;line-height:1.375rem;font-weight:300;align-items:center;background-color:var(--maincolor40,#0c4ded);border-radius:32px;border:1px solid var(--maincolor40,#0c4ded);box-sizing:border-box;color:var(--white,#fff);display:inline-flex;justify-content:center;padding:var(--padding-vertical) var(--padding-horizontal);position:relative;margin:0;height:40px;background-color:transparent!important;border-color:var(--maincolor40,#0c4ded)!important;color:var(--maincolor40,#0c4ded)!important;margin-top:16px}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content #accept-recommended-btn-handler:hover{background-color:var(--maincolor50,#003bde);border-color:var(--maincolor50,#003bde)}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content #accept-recommended-btn-handler:focus{outline:0}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-content #accept-recommended-btn-handler:hover{color:var(--maincolor50,#003bde)!important;border-color:var(--maincolor50,#003bde)!important}#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-desc{font-size:1rem;line-height:1.375rem;font-weight:300;color:var(--grey80,#545454);margin:0}@media only screen and (max-width:563px){#onetrust-consent-sdk #onetrust-pc-sdk #ot-pc-desc{padding-top:16px}}@media only screen and (max-width:768px){#onetrust-banner-sdk .ot-sdk-container{padding:0 16px 24px!important}}#onetrust-banner-sdk .ot-sdk-container>.ot-sdk-row{display:flex}@media only screen and (max-width:768px){#onetrust-banner-sdk .ot-sdk-container>.ot-sdk-row{flex-flow:column}}.ot-sdk-cookie-policy{font-family:inherit;font-size:16px}.ot-sdk-cookie-policy.otRelFont{font-size:1rem}.ot-sdk-cookie-policy h3,.ot-sdk-cookie-policy h4,.ot-sdk-cookie-policy h6,.ot-sdk-cookie-policy p,.ot-sdk-cookie-policy li,.ot-sdk-cookie-policy a,.ot-sdk-cookie-policy th,.ot-sdk-cookie-policy #cookie-policy-description,.ot-sdk-cookie-policy .ot-sdk-cookie-policy-group,.ot-sdk-cookie-policy #cookie-policy-title{color:dimgray}.ot-sdk-cookie-policy #cookie-policy-description{margin-bottom:1em}.ot-sdk-cookie-policy h4{font-size:1.2em}.ot-sdk-cookie-policy h6{font-size:1em;margin-top:2em}.ot-sdk-cookie-policy th{min-width:75px}.ot-sdk-cookie-policy a,.ot-sdk-cookie-policy a:hover{background:#fff}.ot-sdk-cookie-policy thead{background-color:#f6f6f4;font-weight:bold}.ot-sdk-cookie-policy .ot-mobile-border{display:none}.ot-sdk-cookie-policy section{margin-bottom:2em}.ot-sdk-cookie-policy table{border-collapse:inherit}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy{font-family:inherit;font-size:1rem}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy h3,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy h4,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy h6,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy p,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy li,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy a,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy th,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy #cookie-policy-description,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy .ot-sdk-cookie-policy-group,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy #cookie-policy-title{color:dimgray}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy #cookie-policy-description{margin-bottom:1em}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy .ot-sdk-subgroup{margin-left:1.5em}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy #cookie-policy-description,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy .ot-sdk-cookie-policy-group-desc,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy .ot-table-header,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy a,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy span,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy td{font-size:.9em}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy td span,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy td a{font-size:inherit}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy .ot-sdk-cookie-policy-group{font-size:1em;margin-bottom:.6em}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy .ot-sdk-cookie-policy-title{margin-bottom:1.2em}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy>section{margin-bottom:1em}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy th{min-width:75px}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy a,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy a:hover{background:#fff}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy thead{background-color:#f6f6f4;font-weight:bold}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy .ot-mobile-border{display:none}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy section{margin-bottom:2em}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy .ot-sdk-subgroup ul li{list-style:disc;margin-left:1.5em}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy .ot-sdk-subgroup ul li h4{display:inline-block}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table{border-collapse:inherit;margin:auto;border:1px solid #d7d7d7;border-radius:5px;border-spacing:initial;width:100%;overflow:hidden}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table th,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table td{border-bottom:1px solid #d7d7d7;border-right:1px solid #d7d7d7}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table tr:last-child td{border-bottom:0px}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table tr th:last-child,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table tr td:last-child{border-right:0px}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table .ot-host,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table .ot-cookies-type{width:25%}.ot-sdk-cookie-policy[dir=rtl]{text-align:left}#ot-sdk-cookie-policy h3{font-size:1.5em}@media only screen and (max-width: 530px){.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) table,.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) thead,.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) tbody,.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) th,.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) td,.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) tr{display:block}.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) thead tr{position:absolute;top:-9999px;left:-9999px}.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) tr{margin:0 0 1em 0}.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) tr:nth-child(odd),.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) tr:nth-child(odd) a{background:#f6f6f4}.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) td{border:none;border-bottom:1px solid #eee;position:relative;padding-left:50%}.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) td:before{position:absolute;height:100%;left:6px;width:40%;padding-right:10px}.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) .ot-mobile-border{display:inline-block;background-color:#e4e4e4;position:absolute;height:100%;top:0;left:45%;width:2px}.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) td:before{content:attr(data-label);font-weight:bold}.ot-sdk-cookie-policy:not(#ot-sdk-cookie-policy-v2) li{word-break:break-word;word-wrap:break-word}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table{overflow:hidden}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table td{border:none;border-bottom:1px solid #d7d7d7}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy thead,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy tbody,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy th,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy td,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy tr{display:block}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table .ot-host,#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table .ot-cookies-type{width:auto}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy tr{margin:0 0 1em 0}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy td:before{height:100%;width:40%;padding-right:10px}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy td:before{content:attr(data-label);font-weight:bold}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy li{word-break:break-word;word-wrap:break-word}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy thead tr{position:absolute;top:-9999px;left:-9999px;z-index:-9999}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table tr:last-child td{border-bottom:1px solid #d7d7d7;border-right:0px}#ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table tr:last-child td:last-child{border-bottom:0px}}\n",
       "                \n",
       "                    #ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy h5,\n",
       "                    #ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy h6,\n",
       "                    #ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy li,\n",
       "                    #ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy p,\n",
       "                    #ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy a,\n",
       "                    #ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy span,\n",
       "                    #ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy td,\n",
       "                    #ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy #cookie-policy-description {\n",
       "                        color: #696969;\n",
       "                    }\n",
       "                    #ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy th {\n",
       "                        color: #696969;\n",
       "                    }\n",
       "                    #ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy .ot-sdk-cookie-policy-group {\n",
       "                        color: #696969;\n",
       "                    }\n",
       "                    \n",
       "                    #ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy #cookie-policy-title {\n",
       "                            color: #696969;\n",
       "                        }\n",
       "                    \n",
       "            \n",
       "                    #ot-sdk-cookie-policy-v2.ot-sdk-cookie-policy table th {\n",
       "                            background-color: #F8F8F8;\n",
       "                        }\n",
       "                    \n",
       "            \n",
       "                      @keyframes slide-down-custom {\n",
       "                          0% {\n",
       "                              bottom: 887px !important;\n",
       "                          }\n",
       "                          100% {\n",
       "                              bottom: 0px;\n",
       "                          }\n",
       "                      }\n",
       "                      @-webkit-keyframes slide-down-custom {\n",
       "                          0% {\n",
       "                              bottom: 887px !important;\n",
       "                          }\n",
       "                          100% {\n",
       "                              bottom: 0px;\n",
       "                          }\n",
       "                      }\n",
       "                      @-moz-keyframes slide-down-custom {\n",
       "                          0% {\n",
       "                              bottom: 887px !important;\n",
       "                          }\n",
       "                          100% {\n",
       "                              bottom: 0px;\n",
       "                          }\n",
       "                      }\n",
       "                      </style><style type=\"text/css\">.at-icon{fill:#fff;border:0}.at-icon-wrapper{display:inline-block;overflow:hidden}a .at-icon-wrapper{cursor:pointer}.at-rounded,.at-rounded-element .at-icon-wrapper{border-radius:12%}.at-circular,.at-circular-element .at-icon-wrapper{border-radius:50%}.addthis_32x32_style .at-icon{width:2pc;height:2pc}.addthis_24x24_style .at-icon{width:24px;height:24px}.addthis_20x20_style .at-icon{width:20px;height:20px}.addthis_16x16_style .at-icon{width:1pc;height:1pc}#at16lb{display:none;position:absolute;top:0;left:0;width:100%;height:100%;z-index:1001;background-color:#000;opacity:.001}#at_complete,#at_error,#at_share,#at_success{position:static!important}.at15dn{display:none}#at15s,#at16p,#at16p form input,#at16p label,#at16p textarea,#at_share .at_item{font-family:arial,helvetica,tahoma,verdana,sans-serif!important;font-size:9pt!important;outline-style:none;outline-width:0;line-height:1em}* html #at15s.mmborder{position:absolute!important}#at15s.mmborder{position:fixed!important;width:250px!important}#at15s{background:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAYAAACNMs+9AAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAABtJREFUeNpiZGBgaGAgAjAxEAlGFVJHIUCAAQDcngCUgqGMqwAAAABJRU5ErkJggg==);float:none;line-height:1em;margin:0;overflow:visible;padding:5px;text-align:left;position:absolute}#at15s a,#at15s span{outline:0;direction:ltr;text-transform:none}#at15s .at-label{margin-left:5px}#at15s .at-icon-wrapper{width:1pc;height:1pc;vertical-align:middle}#at15s .at-icon{width:1pc;height:1pc}.at4-icon{display:inline-block;background-repeat:no-repeat;background-position:top left;margin:0;overflow:hidden;cursor:pointer}.addthis_16x16_style .at4-icon,.addthis_default_style .at4-icon,.at4-icon,.at-16x16{width:1pc;height:1pc;line-height:1pc;background-size:1pc!important}.addthis_32x32_style .at4-icon,.at-32x32{width:2pc;height:2pc;line-height:2pc;background-size:2pc!important}.addthis_24x24_style .at4-icon,.at-24x24{width:24px;height:24px;line-height:24px;background-size:24px!important}.addthis_20x20_style .at4-icon,.at-20x20{width:20px;height:20px;line-height:20px;background-size:20px!important}.at4-icon.circular,.circular .at4-icon,.circular.aticon{border-radius:50%}.at4-icon.rounded,.rounded .at4-icon{border-radius:4px}.at4-icon-left{float:left}#at15s .at4-icon{text-indent:20px;padding:0;overflow:visible;white-space:nowrap;background-size:1pc;width:1pc;height:1pc;background-position:top left;display:inline-block;line-height:1pc}.addthis_vertical_style .at4-icon,.at4-follow-container .at4-icon{margin-right:5px}html>body #at15s{width:250px!important}#at15s.atm{background:none!important;padding:0!important;width:10pc!important}#at15s_inner{background:#fff;border:1px solid #fff;margin:0}#at15s_head{position:relative;background:#f2f2f2;padding:4px;cursor:default;border-bottom:1px solid #e5e5e5}.at15s_head_success{background:#cafd99!important;border-bottom:1px solid #a9d582!important}.at15s_head_success a,.at15s_head_success span{color:#000!important;text-decoration:none}#at15s_brand,#at15sptx,#at16_brand{position:absolute}#at15s_brand{top:4px;right:4px}.at15s_brandx{right:20px!important}a#at15sptx{top:4px;right:4px;text-decoration:none;color:#4c4c4c;font-weight:700}#at15sptx:hover{text-decoration:underline}#at16_brand{top:5px;right:30px;cursor:default}#at_hover{padding:4px}#at_hover .at_item,#at_share .at_item{background:#fff!important;float:left!important;color:#4c4c4c!important}#at_share .at_item .at-icon-wrapper{margin-right:5px}#at_hover .at_bold{font-weight:700;color:#000!important}#at_hover .at_item{width:7pc!important;padding:2px 3px!important;margin:1px;text-decoration:none!important}#at_hover .at_item.athov,#at_hover .at_item:focus,#at_hover .at_item:hover{margin:0!important}#at_hover .at_item.athov,#at_hover .at_item:focus,#at_hover .at_item:hover,#at_share .at_item.athov,#at_share .at_item:hover{background:#f2f2f2!important;border:1px solid #e5e5e5;color:#000!important;text-decoration:none}.ipad #at_hover .at_item:focus{background:#fff!important;border:1px solid #fff}.at15t{display:block!important;height:1pc!important;line-height:1pc!important;padding-left:20px!important;background-position:0 0;text-align:left}.addthis_button,.at15t{cursor:pointer}.addthis_toolbox a.at300b,.addthis_toolbox a.at300m{width:auto}.addthis_toolbox a{margin-bottom:5px;line-height:initial}.addthis_toolbox.addthis_vertical_style{width:200px}.addthis_button_facebook_like .fb_iframe_widget{line-height:100%}.addthis_button_facebook_like iframe.fb_iframe_widget_lift{max-width:none}.addthis_toolbox a.addthis_button_counter,.addthis_toolbox a.addthis_button_facebook_like,.addthis_toolbox a.addthis_button_facebook_send,.addthis_toolbox a.addthis_button_facebook_share,.addthis_toolbox a.addthis_button_foursquare,.addthis_toolbox a.addthis_button_linkedin_counter,.addthis_toolbox a.addthis_button_pinterest_pinit,.addthis_toolbox a.addthis_button_tweet{display:inline-block}.addthis_toolbox span.addthis_follow_label{display:none}.addthis_toolbox.addthis_vertical_style span.addthis_follow_label{display:block;white-space:nowrap}.addthis_toolbox.addthis_vertical_style a{display:block}.addthis_toolbox.addthis_vertical_style.addthis_32x32_style a{line-height:2pc;height:2pc}.addthis_toolbox.addthis_vertical_style .at300bs{margin-right:4px;float:left}.addthis_toolbox.addthis_20x20_style span{line-height:20px}.addthis_toolbox.addthis_32x32_style span{line-height:2pc}.addthis_toolbox.addthis_pill_combo_style .addthis_button_compact .at15t_compact,.addthis_toolbox.addthis_pill_combo_style a{float:left}.addthis_toolbox.addthis_pill_combo_style a.addthis_button_tweet{margin-top:-2px}.addthis_toolbox.addthis_pill_combo_style .addthis_button_compact .at15t_compact{margin-right:4px}.addthis_default_style .addthis_separator{margin:0 5px;display:inline}div.atclear{clear:both}.addthis_default_style .addthis_separator,.addthis_default_style .at4-icon,.addthis_default_style .at300b,.addthis_default_style .at300bo,.addthis_default_style .at300bs,.addthis_default_style .at300m{float:left}.at300b img,.at300bo img{border:0}a.at300b .at4-icon,a.at300m .at4-icon{display:block}.addthis_default_style .at300b,.addthis_default_style .at300bo,.addthis_default_style .at300m{padding:0 2px}.at300b,.at300bo,.at300bs,.at300m{cursor:pointer}.addthis_button_facebook_like.at300b:hover,.addthis_button_facebook_like.at300bs:hover,.addthis_button_facebook_send.at300b:hover,.addthis_button_facebook_send.at300bs:hover{opacity:1}.addthis_20x20_style .at15t,.addthis_20x20_style .at300bs{overflow:hidden;display:block;height:20px!important;width:20px!important;line-height:20px!important}.addthis_32x32_style .at15t,.addthis_32x32_style .at300bs{overflow:hidden;display:block;height:2pc!important;width:2pc!important;line-height:2pc!important}.at300bs{overflow:hidden;display:block;background-position:0 0;height:1pc;width:1pc;line-height:1pc!important}.addthis_default_style .at15t_compact,.addthis_default_style .at15t_expanded{margin-right:4px}#at_share .at_item{width:123px!important;padding:4px;margin-right:2px;border:1px solid #fff}#at16p{background:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAYAAACNMs+9AAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAABtJREFUeNpiZGBgaGAgAjAxEAlGFVJHIUCAAQDcngCUgqGMqwAAAABJRU5ErkJggg==);z-index:10000001;position:absolute;top:50%;left:50%;width:300px;padding:10px;margin:0 auto;margin-top:-185px;margin-left:-155px;font-family:arial,helvetica,tahoma,verdana,sans-serif;font-size:9pt;color:#5e5e5e}#at_share{margin:0;padding:0}#at16pt{position:relative;background:#f2f2f2;height:13px;padding:5px 10px}#at16pt a,#at16pt h4{font-weight:700}#at16pt h4{display:inline;margin:0;padding:0;font-size:9pt;color:#4c4c4c;cursor:default}#at16pt a{position:absolute;top:5px;right:10px;color:#4c4c4c;text-decoration:none;padding:2px}#at15sptx:focus,#at16pt a:focus{outline:thin dotted}#at15s #at16pf a{top:1px}#_atssh{width:1px!important;height:1px!important;border:0!important}.atm{width:10pc!important;padding:0;margin:0;line-height:9pt;letter-spacing:normal;font-family:arial,helvetica,tahoma,verdana,sans-serif;font-size:9pt;color:#444;background:url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAYAAACNMs+9AAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAABtJREFUeNpiZGBgaGAgAjAxEAlGFVJHIUCAAQDcngCUgqGMqwAAAABJRU5ErkJggg==);padding:4px}.atm-f{text-align:right;border-top:1px solid #ddd;padding:5px 8px}.atm-i{background:#fff;border:1px solid #d5d6d6;padding:0;margin:0;box-shadow:1px 1px 5px rgba(0,0,0,.15)}.atm-s{margin:0!important;padding:0!important}.atm-s a:focus{border:transparent;outline:0;transition:none}#at_hover.atm-s a,.atm-s a{display:block;text-decoration:none;padding:4px 10px;color:#235dab!important;font-weight:400;font-style:normal;transition:none}#at_hover.atm-s .at_bold{color:#235dab!important}#at_hover.atm-s a:hover,.atm-s a:hover{background:#2095f0;text-decoration:none;color:#fff!important}#at_hover.atm-s .at_bold{font-weight:700}#at_hover.atm-s a:hover .at_bold{color:#fff!important}.atm-s a .at-label{vertical-align:middle;margin-left:5px;direction:ltr}.at_PinItButton{display:block;width:40px;height:20px;padding:0;margin:0;background-image:url(//s7.addthis.com/static/t00/pinit00.png);background-repeat:no-repeat}.at_PinItButton:hover{background-position:0 -20px}.addthis_toolbox .addthis_button_pinterest_pinit{position:relative}.at-share-tbx-element .fb_iframe_widget span{vertical-align:baseline!important}#at16pf{height:auto;text-align:right;padding:4px 8px}.at-privacy-info{position:absolute;left:7px;bottom:7px;cursor:pointer;text-decoration:none;font-family:helvetica,arial,sans-serif;font-size:10px;line-height:9pt;letter-spacing:.2px;color:#666}.at-privacy-info:hover{color:#000}.body .wsb-social-share .wsb-social-share-button-vert{padding-top:0;padding-bottom:0}.body .wsb-social-share.addthis_counter_style .addthis_button_tweet.wsb-social-share-button{padding-top:40px}.body .wsb-social-share.addthis_counter_style .addthis_button_facebook_like.wsb-social-share-button{padding-top:21px}@media print{#at4-follow,#at4-share,#at4-thankyou,#at4-whatsnext,#at4m-mobile,#at15s,.at4,.at4-recommended{display:none!important}}@media screen and (max-width:400px){.at4win{width:100%}}@media screen and (max-height:700px) and (max-width:400px){.at4-thankyou-inner .at4-recommended-container{height:122px;overflow:hidden}.at4-thankyou-inner .at4-recommended .at4-recommended-item:first-child{border-bottom:1px solid #c5c5c5}}</style><style type=\"text/css\">.at-branding-logo{font-family:helvetica,arial,sans-serif;text-decoration:none;font-size:10px;display:inline-block;margin:2px 0;letter-spacing:.2px}.at-branding-logo .at-branding-icon{background-image:url(\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAMAAAC67D+PAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAAZQTFRF////+GlNUkcc1QAAAB1JREFUeNpiYIQDBjQmAwMmkwEM0JnY1WIxFyDAABGeAFEudiZsAAAAAElFTkSuQmCC\")}.at-branding-logo .at-branding-icon,.at-branding-logo .at-privacy-icon{display:inline-block;height:10px;width:10px;margin-left:4px;margin-right:3px;margin-bottom:-1px;background-repeat:no-repeat}.at-branding-logo .at-privacy-icon{background-image:url(\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAkAAAAKCAMAAABR24SMAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAABhQTFRF8fr9ot/xXcfn2/P5AKva////////AKTWodjhjAAAAAd0Uk5T////////ABpLA0YAAAA6SURBVHjaJMzBDQAwCAJAQaj7b9xifV0kUKJ9ciWxlzWEWI5gMF65KUTv0VKkjVeTerqE/x7+9BVgAEXbAWI8QDcfAAAAAElFTkSuQmCC\")}.at-branding-logo span{text-decoration:none}.at-branding-logo .at-branding-addthis,.at-branding-logo .at-branding-powered-by{color:#666}.at-branding-logo .at-branding-addthis:hover{color:#333}.at-cv-with-image .at-branding-addthis,.at-cv-with-image .at-branding-addthis:hover{color:#fff}a.at-branding-logo:visited{color:initial}.at-branding-info{display:inline-block;padding:0 5px;color:#666;border:1px solid #666;border-radius:50%;font-size:10px;line-height:9pt;opacity:.7;transition:all .3s ease;text-decoration:none}.at-branding-info span{border:0;clip:rect(0 0 0 0);height:1px;margin:-1px;overflow:hidden;padding:0;position:absolute;width:1px}.at-branding-info:before{content:'i';font-family:Times New Roman}.at-branding-info:hover{color:#0780df;border-color:#0780df}</style><script charset=\"utf-8\" data-webpack=\"NRBA:chunk-552\" src=\"https://js-agent.newrelic.com/552.2d6a2503-1220.js\"></script></head>\n",
       "<body class=\"journal-plant-science section-plant-pathogen-interactions\" style=\"\"><div id=\"MathJax_Message\" style=\"display: none;\"></div>\n",
       "<button class=\"bypassBlock-firstEl\"></button>\n",
       "<a class=\"bypassBlock-wrapper\" href=\"#main-content\" id=\"bypass\">\n",
       "<span class=\"bypassBlock-button\">Skip to main content</span>\n",
       "</a>\n",
       "<script type=\"text/javascript\">\n",
       "\n",
       "        const container = document.getElementById(\"bypass\");\n",
       "\n",
       "        //eventListeners\n",
       "        container.addEventListener('focus', function () {\n",
       "            document.documentElement.setAttribute(\"bypass-focus\", \"true\");\n",
       "        }, true);\n",
       "\n",
       "        container.addEventListener('blur', function () {\n",
       "            document.documentElement.setAttribute(\"bypass-focus\", \"false\");\n",
       "        }, true);\n",
       "\n",
       "    </script><script id=\"\" type=\"text/javascript\">console.log(\"the environment name is : \"+google_tag_manager[\"GTM-N279F7B\"].macro(6));</script>\n",
       "<!-- Google Tag Manager (noscript) -->\n",
       "<noscript>\n",
       "<iframe height=\"0\" src=\"https://www.googletagmanager.com/ns.html?id=GTM-TFKV632&gt;m_auth=NsJH3Ts1-89I8lZURwQYmw&gt;m_preview=env-1&gt;m_cookies_win=x\" style=\"display:none;visibility:hidden\" width=\"0\"></iframe>\n",
       "</noscript>\n",
       "<!-- Google Tag Manager (noscript) -->\n",
       "<noscript>\n",
       "<iframe height=\"0\" src=\"https://www.googletagmanager.com/ns.html?id=GTM-N279F7B&gt;m_auth=jfKETtlWamfZ4l02YiFCQw&gt;m_preview=env-1&gt;m_cookies_win=x\" style=\"display:none;visibility:hidden\" width=\"0\"></iframe>\n",
       "</noscript>\n",
       "<!-- End Google Tag Manager (noscript) -->\n",
       "<script src=\"https://static.frontiersin.org/areas/articles/js/vendors?v=Tk-waxr_iWGXyIfXPNClZsfLGaaFNLI99xR9M9Gkzeg1\"></script>\n",
       "<frontiers-ibar data=\"{&quot;aboutUs&quot;:[{&quot;title&quot;:&quot;Who we are&quot;,&quot;links&quot;:[{&quot;text&quot;:&quot;Mission&quot;,&quot;url&quot;:&quot;/about/mission&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Values&quot;,&quot;url&quot;:&quot;/about/values&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;History&quot;,&quot;url&quot;:&quot;/about/history&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Leadership&quot;,&quot;url&quot;:&quot;/about/leadership&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Awards&quot;,&quot;url&quot;:&quot;/about/awards&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null}]},{&quot;title&quot;:&quot;Impact and progress&quot;,&quot;links&quot;:[{&quot;text&quot;:&quot;Frontiers' impact&quot;,&quot;url&quot;:&quot;/about/impact&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;2021 Progress Report&quot;,&quot;url&quot;:&quot;https://progressreport.frontiersin.org/?utm_source=fweb&amp;utm_medium=frep&amp;utm_campaign=pr20&quot;,&quot;target&quot;:&quot;_blank&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;All progress reports&quot;,&quot;url&quot;:&quot;/about/progress-reports&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null}]},{&quot;title&quot;:&quot;Publishing model&quot;,&quot;links&quot;:[{&quot;text&quot;:&quot;How we publish&quot;,&quot;url&quot;:&quot;/about/how-we-publish&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Open access&quot;,&quot;url&quot;:&quot;/about/open-access&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Fee policy&quot;,&quot;url&quot;:&quot;/about/fee-policy&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Peer review&quot;,&quot;url&quot;:&quot;/about/peer-review&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Research Topics&quot;,&quot;url&quot;:&quot;/about/research-topics&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null}]},{&quot;title&quot;:&quot;Services&quot;,&quot;links&quot;:[{&quot;text&quot;:&quot;Societies&quot;,&quot;url&quot;:&quot;https://publishingpartnerships.frontiersin.org/&quot;,&quot;target&quot;:&quot;_blank&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;National consortia&quot;,&quot;url&quot;:&quot;/about/national-consortia&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Institutional partnerships&quot;,&quot;url&quot;:&quot;/about/institutional-partnerships&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Collaborators&quot;,&quot;url&quot;:&quot;/about/collaborators&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null}]},{&quot;title&quot;:&quot;More from Frontiers&quot;,&quot;links&quot;:[{&quot;text&quot;:&quot;Frontiers Forum&quot;,&quot;url&quot;:&quot;https://forum.frontiersin.org/&quot;,&quot;target&quot;:&quot;_blank&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Press office&quot;,&quot;url&quot;:&quot;https://pressoffice.frontiersin.org/&quot;,&quot;target&quot;:&quot;_blank&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Career opportunities&quot;,&quot;url&quot;:&quot;https://careers.frontiersin.org/&quot;,&quot;target&quot;:&quot;_blank&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Contact us&quot;,&quot;url&quot;:&quot;/about/contact&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null}]}],&quot;submitUrl&quot;:&quot;/submission/submit?domainid=1&amp;fieldid=66&amp;specialtyid=0&amp;entitytype=2&amp;entityid=373&quot;,&quot;showSubmitButton&quot;:true,&quot;journal&quot;:{&quot;id&quot;:373,&quot;name&quot;:&quot;Frontiers in Plant Science&quot;,&quot;slug&quot;:&quot;plant-science&quot;,&quot;sections&quot;:[{&quot;id&quot;:1356,&quot;name&quot;:&quot;Crop and Product Physiology&quot;,&quot;slug&quot;:&quot;crop-and-product-physiology&quot;},{&quot;id&quot;:467,&quot;name&quot;:&quot;Functional Plant Ecology&quot;,&quot;slug&quot;:&quot;functional-plant-ecology&quot;},{&quot;id&quot;:2844,&quot;name&quot;:&quot;Functional and Applied Plant Genomics&quot;,&quot;slug&quot;:&quot;functional-and-applied-plant-genomics&quot;},{&quot;id&quot;:1553,&quot;name&quot;:&quot;Marine and Freshwater Plants&quot;,&quot;slug&quot;:&quot;marine-and-freshwater-plants&quot;},{&quot;id&quot;:2843,&quot;name&quot;:&quot;Photosynthesis and Photobiology&quot;,&quot;slug&quot;:&quot;photosynthesis-and-photobiology&quot;},{&quot;id&quot;:1312,&quot;name&quot;:&quot;Plant Abiotic Stress&quot;,&quot;slug&quot;:&quot;plant-abiotic-stress&quot;},{&quot;id&quot;:2183,&quot;name&quot;:&quot;Plant Bioinformatics&quot;,&quot;slug&quot;:&quot;plant-bioinformatics&quot;},{&quot;id&quot;:589,&quot;name&quot;:&quot;Plant Biophysics and Modeling&quot;,&quot;slug&quot;:&quot;plant-biophysics-and-modeling&quot;},{&quot;id&quot;:560,&quot;name&quot;:&quot;Plant Biotechnology&quot;,&quot;slug&quot;:&quot;plant-biotechnology&quot;},{&quot;id&quot;:468,&quot;name&quot;:&quot;Plant Breeding&quot;,&quot;slug&quot;:&quot;plant-breeding&quot;},{&quot;id&quot;:577,&quot;name&quot;:&quot;Plant Cell Biology&quot;,&quot;slug&quot;:&quot;plant-cell-biology&quot;},{&quot;id&quot;:474,&quot;name&quot;:&quot;Plant Development and EvoDevo&quot;,&quot;slug&quot;:&quot;plant-development-and-evodevo&quot;},{&quot;id&quot;:2845,&quot;name&quot;:&quot;Plant Genetics, Epigenetics and Chromosome Biology&quot;,&quot;slug&quot;:&quot;plant-genetics-epigenetics-and-chromosome-biology&quot;},{&quot;id&quot;:479,&quot;name&quot;:&quot;Plant Membrane Traffic and Transport&quot;,&quot;slug&quot;:&quot;plant-membrane-traffic-and-transport&quot;},{&quot;id&quot;:481,&quot;name&quot;:&quot;Plant Metabolism and Chemodiversity&quot;,&quot;slug&quot;:&quot;plant-metabolism-and-chemodiversity&quot;},{&quot;id&quot;:486,&quot;name&quot;:&quot;Plant Nutrition&quot;,&quot;slug&quot;:&quot;plant-nutrition&quot;},{&quot;id&quot;:485,&quot;name&quot;:&quot;Plant Pathogen Interactions&quot;,&quot;slug&quot;:&quot;plant-pathogen-interactions&quot;},{&quot;id&quot;:226,&quot;name&quot;:&quot;Plant Physiology&quot;,&quot;slug&quot;:&quot;plant-physiology&quot;},{&quot;id&quot;:580,&quot;name&quot;:&quot;Plant Proteomics and Protein Structural Biology&quot;,&quot;slug&quot;:&quot;plant-proteomics-and-protein-structural-biology&quot;},{&quot;id&quot;:1570,&quot;name&quot;:&quot;Plant Symbiotic Interactions&quot;,&quot;slug&quot;:&quot;plant-symbiotic-interactions&quot;},{&quot;id&quot;:1428,&quot;name&quot;:&quot;Plant Systematics and Evolution&quot;,&quot;slug&quot;:&quot;plant-systematics-and-evolution&quot;},{&quot;id&quot;:483,&quot;name&quot;:&quot;Plant Systems and Synthetic Biology&quot;,&quot;slug&quot;:&quot;plant-systems-and-synthetic-biology&quot;},{&quot;id&quot;:2277,&quot;name&quot;:&quot;Sustainable and Intelligent Phytoprotection&quot;,&quot;slug&quot;:&quot;sustainable-and-intelligent-phytoprotection&quot;},{&quot;id&quot;:484,&quot;name&quot;:&quot;Technical Advances in Plant Science&quot;,&quot;slug&quot;:&quot;technical-advances-in-plant-science&quot;}]},&quot;aboutJournal&quot;:[{&quot;title&quot;:&quot;Scope&quot;,&quot;links&quot;:[{&quot;text&quot;:&quot;Field chief editor&quot;,&quot;url&quot;:&quot;/journals/plant-science/about#about-editors&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Mission&quot;,&quot;url&quot;:&quot;/journals/plant-science/about#about-scope&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Facts&quot;,&quot;url&quot;:&quot;/journals/plant-science/about#about-facts&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Submission&quot;,&quot;url&quot;:&quot;/journals/plant-science/about#about-submission&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Open access statement&quot;,&quot;url&quot;:&quot;/journals/plant-science/about#about-open&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Copyright statement&quot;,&quot;url&quot;:&quot;/journals/plant-science/about#about-copyright&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Quality&quot;,&quot;url&quot;:&quot;/journals/plant-science/about#about-quality&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null}]},{&quot;title&quot;:&quot;For authors&quot;,&quot;links&quot;:[{&quot;text&quot;:&quot;Why submit?&quot;,&quot;url&quot;:&quot;/journals/plant-science/for-authors/why-submit&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Article types&quot;,&quot;url&quot;:&quot;/journals/plant-science/for-authors/article-types&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Author guidelines&quot;,&quot;url&quot;:&quot;/journals/plant-science/for-authors/author-guidelines&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Editor guidelines&quot;,&quot;url&quot;:&quot;/journals/plant-science/for-authors/editor-guidelines&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Publishing fees&quot;,&quot;url&quot;:&quot;/journals/plant-science/for-authors/publishing-fees&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Submission checklist&quot;,&quot;url&quot;:&quot;/journals/plant-science/for-authors/submission-checklist&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;Contact editorial office&quot;,&quot;url&quot;:&quot;/journals/plant-science/for-authors/contact-editorial-office&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null}]}],&quot;mainLinks&quot;:[{&quot;text&quot;:&quot;All journals&quot;,&quot;url&quot;:&quot;/journals&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null},{&quot;text&quot;:&quot;All articles&quot;,&quot;url&quot;:&quot;/articles&quot;,&quot;target&quot;:&quot;_self&quot;,&quot;ariaLabel&quot;:null}],&quot;journalLinks&quot;:[]}\" main-domain=\"false\"></frontiers-ibar>\n",
       "<div class=\"page-container sidemenu-collapsed\" id=\"main-content\">\n",
       "<div class=\"boxed white no-padding\" id=\"article\">\n",
       "<div class=\"container-fluid main-container-xxl\">\n",
       "<div class=\"row\" id=\"similar-articles\">\n",
       "<div class=\"new-wrapper\">\n",
       "<style>\n",
       "    .disabled-supplementary-btn {\n",
       "        cursor: not-allowed;\n",
       "        pointer-events: none;\n",
       "        opacity: .65;\n",
       "        filter: alpha(opacity=65);\n",
       "        -webkit-box-shadow: none;\n",
       "        box-shadow: none;\n",
       "    }\n",
       "</style>\n",
       "<aside class=\"right-container\">\n",
       "<div class=\"side-article-download clearfix\">\n",
       "<ul class=\"list-unstyled list-inline clearfix\">\n",
       "<li class=\"dropdown text-center paper\">\n",
       "<button aria-expanded=\"false\" class=\"dropdown-toggle\" data-target=\"#\" data-test-id=\"download-button\" data-toggle=\"dropdown\" role=\"button\">\n",
       "<span class=\"icon-label\" data-event=\"download-button-click\">Download Article</span>\n",
       "</button>\n",
       "<div class=\"dropdown-menu-wrapper\">\n",
       "<div class=\"dropdown-menu-mobile-title\">\n",
       "                                Download Article\n",
       "                            </div>\n",
       "<ul class=\"dropdown-menu\" role=\"menu\">\n",
       "<li>\n",
       "<a class=\"download-files-pdf action-link\" data-event=\"downloadpdf-button-click\" data-test-id=\"article-downloadpdf\" href=\"/articles/10.3389/fpls.2019.00932/pdf\">\n",
       "                                            Download PDF\n",
       "                                        </a>\n",
       "</li>\n",
       "<li>\n",
       "<a class=\"download-files-readcube\" data-event=\"downloadreadcube-button-click\" data-test-id=\"article-viewenhancedpdf\" href=\"http://www.readcube.com/articles/10.3389/fpls.2019.00932\">\n",
       "                                            ReadCube\n",
       "                                        </a>\n",
       "</li>\n",
       "<li>\n",
       "<a class=\"download-files-epub\" data-event=\"downloadepub-button-click\" data-test-id=\"article-epub\" href=\"/articles/10.3389/fpls.2019.00932/epub\">\n",
       "                                            EPUB\n",
       "                                        </a>\n",
       "</li>\n",
       "<li>\n",
       "<a class=\"download-files-nlm\" data-event=\"downloadnlm-button-click\" data-test-id=\"article-nlm\" href=\"/articles/10.3389/fpls.2019.00932/xml/nlm\">\n",
       "                                            XML (NLM)\n",
       "                                        </a>\n",
       "</li>\n",
       "<li>\n",
       "<a class=\"supplemental-data-disabled\" data-event=\"supplementary-button-click\" data-test-id=\"supplementary-button\" id=\"supplementary_view\">\n",
       "\n",
       "                                            Supplementary\n",
       "                                            <br/>\n",
       "                                            Material\n",
       "                                        </a>\n",
       "</li>\n",
       "</ul>\n",
       "<button aria-label=\"Close\" class=\"dropdown-menu-mobile-close-button\" data-toggle=\"dropdown\"></button>\n",
       "</div>\n",
       "</li>\n",
       "<li>\n",
       "<button class=\"btn btn-open-supplemental\" data-event=\"supplementary-button-click\" id=\"supplementary_view\">Supplemental data</button>\n",
       "</li>\n",
       "<li class=\"dropdown\">\n",
       "<button aria-expanded=\"false\" class=\"navbar-toggle navbar-share\" data-target=\"#\" data-toggle=\"dropdown\" type=\"button\"></button>\n",
       "<div class=\"dropdown-menu-wrapper\">\n",
       "<div class=\"dropdown-menu-mobile-title\">\n",
       "                        Share on\n",
       "                    </div>\n",
       "<ul class=\"dropdown-menu\" role=\"menu\">\n",
       "<li class=\"social_block\">\n",
       "<div class=\"atButton\">\n",
       "<a addthis:description=\"Succession is generally well described above-ground in the boreal forest, and several studies have demonstrated the role of interspecific facilitation in tree species establishment. However, the role of mycorrhizal communities for tree establishment and interspecific facilitation, has been little explored. At the ecotone between the mixed boreal forest, dominated by balsam fir and hardwood species, and the boreal forest, dominated by black spruce, several stands of trembling aspen can be found, surrounded by black spruce forest. Regeneration of balsam fir seems to have increased in the recent decades within the boreal forest, and it seems better adapted to grow in trembling aspen stands than in black spruce stands, even when located in similar abiotic conditions. As black spruce stands are also covered by ericaceous shrubs, we investigated if differences in soil fungal communities and ericaceous shrubs abundance could explain the differences observed in balsam fir growth and nutrition. We conducted a study centered on individual saplings to link growth and foliar nutrient concentrations to local vegetation cover, mycorrhization rate, and mycorrhizal communities associated with balsam fir roots. We found that foliar nutrient concentrations and ramification indices (colonization by mycorrhiza per length of root) were greater in trembling aspen stands and were positively correlated to apical and lateral growth of balsam fir saplings. In black spruce stands, the presence of er...\" addthis:title=\"Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?\" addthis:url=\"https://www.frontiersin.org/articles/10.3389/fpls.2019.00932\" class=\"addthis_button_twitter at300b\" href=\"#\" title=\"Tweet\">\n",
       "<span class=\"at-icon-wrapper\"></span>\n",
       "</a>\n",
       "</div>\n",
       "</li>\n",
       "<li class=\"social_block\">\n",
       "<div class=\"atButton\">\n",
       "<a addthis:description=\"Succession is generally well described above-ground in the boreal forest, and several studies have demonstrated the role of interspecific facilitation in tree species establishment. However, the role of mycorrhizal communities for tree establishment and interspecific facilitation, has been little explored. At the ecotone between the mixed boreal forest, dominated by balsam fir and hardwood species, and the boreal forest, dominated by black spruce, several stands of trembling aspen can be found, surrounded by black spruce forest. Regeneration of balsam fir seems to have increased in the recent decades within the boreal forest, and it seems better adapted to grow in trembling aspen stands than in black spruce stands, even when located in similar abiotic conditions. As black spruce stands are also covered by ericaceous shrubs, we investigated if differences in soil fungal communities and ericaceous shrubs abundance could explain the differences observed in balsam fir growth and nutrition. We conducted a study centered on individual saplings to link growth and foliar nutrient concentrations to local vegetation cover, mycorrhization rate, and mycorrhizal communities associated with balsam fir roots. We found that foliar nutrient concentrations and ramification indices (colonization by mycorrhiza per length of root) were greater in trembling aspen stands and were positively correlated to apical and lateral growth of balsam fir saplings. In black spruce stands, the presence of er...\" addthis:title=\"Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?\" addthis:url=\"https://www.frontiersin.org/articles/10.3389/fpls.2019.00932\" class=\"addthis_button_linkedin at300b\" href=\"#\" target=\"_blank\" title=\"LinkedIn\">\n",
       "<span class=\"at-icon-wrapper\"></span>\n",
       "</a>\n",
       "</div>\n",
       "</li>\n",
       "<li class=\"social_block\">\n",
       "<div class=\"atButton\">\n",
       "<a addthis:description=\"Succession is generally well described above-ground in the boreal forest, and several studies have demonstrated the role of interspecific facilitation in tree species establishment. However, the role of mycorrhizal communities for tree establishment and interspecific facilitation, has been little explored. At the ecotone between the mixed boreal forest, dominated by balsam fir and hardwood species, and the boreal forest, dominated by black spruce, several stands of trembling aspen can be found, surrounded by black spruce forest. Regeneration of balsam fir seems to have increased in the recent decades within the boreal forest, and it seems better adapted to grow in trembling aspen stands than in black spruce stands, even when located in similar abiotic conditions. As black spruce stands are also covered by ericaceous shrubs, we investigated if differences in soil fungal communities and ericaceous shrubs abundance could explain the differences observed in balsam fir growth and nutrition. We conducted a study centered on individual saplings to link growth and foliar nutrient concentrations to local vegetation cover, mycorrhization rate, and mycorrhizal communities associated with balsam fir roots. We found that foliar nutrient concentrations and ramification indices (colonization by mycorrhiza per length of root) were greater in trembling aspen stands and were positively correlated to apical and lateral growth of balsam fir saplings. In black spruce stands, the presence of er...\" addthis:title=\"Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?\" addthis:url=\"https://www.frontiersin.org/articles/10.3389/fpls.2019.00932\" class=\"addthis_button_facebook at300b\" href=\"#\" title=\"Facebook\">\n",
       "<span class=\"at-icon-wrapper\"></span>\n",
       "</a>\n",
       "</div>\n",
       "</li>\n",
       "</ul>\n",
       "<button aria-label=\"Close\" class=\"dropdown-menu-mobile-close-button\" data-toggle=\"dropdown\"></button>\n",
       "</div>\n",
       "</li>\n",
       "<li class=\"dropdown\">\n",
       "<button aria-expanded=\"false\" class=\"navbar-toggle navbar-citations\" data-event=\"citation-button-click\" data-target=\"#\" data-toggle=\"dropdown\" type=\"button\">\n",
       "</button>\n",
       "<div class=\"dropdown-menu-wrapper\">\n",
       "<div class=\"dropdown-menu-mobile-title\">\n",
       "                        Export citation\n",
       "                    </div>\n",
       "<ul class=\"dropdown-menu\" role=\"menu\">\n",
       "<li>\n",
       "<a data-event=\"endnotedownload-button-click\" data-test-id=\"article-endnote\" href=\"/articles/10.3389/fpls.2019.00932/endNote\">\n",
       "                                    EndNote\n",
       "                                </a>\n",
       "</li>\n",
       "<li>\n",
       "<a data-event=\"referencemanagerdownload-button-click\" data-test-id=\"article-referencemanager\" href=\"/articles/10.3389/fpls.2019.00932/reference\">\n",
       "                                    Reference Manager\n",
       "                                </a>\n",
       "</li>\n",
       "<li>\n",
       "<a data-event=\"simpletextfiledownload-button-click\" data-test-id=\"article-simpletextfile\" href=\"/articles/10.3389/fpls.2019.00932/text\">\n",
       "                                    Simple TEXT file\n",
       "                                </a>\n",
       "</li>\n",
       "<li>\n",
       "<a data-event=\"bibtexdownload-button-click\" data-test-id=\"article-bibtex\" href=\"/articles/10.3389/fpls.2019.00932/bibTex\">\n",
       "                                    BibTex\n",
       "                                </a>\n",
       "</li>\n",
       "</ul>\n",
       "<button aria-label=\"Close\" class=\"dropdown-menu-mobile-close-button\" data-toggle=\"dropdown\"></button>\n",
       "</div>\n",
       "</li>\n",
       "</ul>\n",
       "</div>\n",
       "<div class=\"side-article-impact\">\n",
       "<ul class=\"nav\">\n",
       "<li class=\"impact-data\" data-original-title=\"The total view count is updated once a day, so not to worry if you don't see immediate results.\" title=\"\">\n",
       "<span class=\"title-number\">4,419</span>\n",
       "<span class=\"title-text\">total views</span>\n",
       "</li>\n",
       "<li class=\"altmetric-wrapper\">\n",
       "<div class=\"altmetric-embed\" data-badge-popover=\"bottom\" data-badge-type=\"donut\" data-condensed=\"true\" data-doi=\"10.3389/fpls.2019.00932\" data-event=\"altmetric-button-click\" data-uuid=\"02fd5df8-1e98-a739-d2f8-15d40a4df176\"><a data-badge-popover=\"bottom\" data-content=\"&lt;div&gt;    &lt;div style='padding-left: 10px; line-height:18px; border-left: 16px solid #FF0000;'&gt;\n",
       "      &lt;a class='link-to-altmetric-details-tab' target='_self' href='https://www.altmetric.com/details.php?domain=www.frontiersin.org&amp;citation_id=63734010&amp;tab=news'&gt;\n",
       "          News (1)\n",
       "      &lt;/a&gt;\n",
       "    &lt;/div&gt;\n",
       "    &lt;div style='padding-left: 10px; line-height:18px; border-left: 16px solid #74CFED;'&gt;\n",
       "      &lt;a class='link-to-altmetric-details-tab' target='_self' href='https://www.altmetric.com/details.php?domain=www.frontiersin.org&amp;citation_id=63734010&amp;tab=twitter'&gt;\n",
       "          Twitter (16)\n",
       "      &lt;/a&gt;\n",
       "    &lt;/div&gt;\n",
       "    &lt;div style='padding-left: 10px; line-height:18px; border-left: 16px solid #2445bd;'&gt;\n",
       "      &lt;a class='link-to-altmetric-details-tab' target='_self' href='https://www.altmetric.com/details.php?domain=www.frontiersin.org&amp;citation_id=63734010&amp;tab=facebook'&gt;\n",
       "          Facebook (1)\n",
       "      &lt;/a&gt;\n",
       "    &lt;/div&gt;\n",
       "\n",
       "    &lt;div class='altmetric-embed readers' style='margin-top: 10px;'&gt;\n",
       "          &lt;div class='altmetric-embed tip_mendeley'\n",
       "               style='padding-left: 10px; line-height:18px; border-left: 16px solid #A60000;'&gt;\n",
       "              Mendeley (32)\n",
       "          &lt;/div&gt;\n",
       "    &lt;/div&gt;\n",
       "    &lt;div style='margin-top: 10px; text-align: center;'&gt;\n",
       "        &lt;a class='altmetric_details' target='_self' href='https://www.altmetric.com/details.php?domain=www.frontiersin.org&amp;citation_id=63734010'&gt;\n",
       "            See more details\n",
       "        &lt;/a&gt;\n",
       "    &lt;/div&gt;\n",
       "&lt;/div&gt;\" href=\"https://www.altmetric.com/details.php?domain=www.frontiersin.org&amp;citation_id=63734010\" rel=\"popover\" style=\"display:inline-block;\" target=\"_self\">\n",
       "<img alt=\"Article has an altmetric score of 17\" height=\"64\" src=\"https://badges.altmetric.com/?size=128&amp;score=17&amp;types=mttttttf\" style=\"border:0; margin:0; max-width: none;\" width=\"64\"/>\n",
       "</a></div>\n",
       "</li>\n",
       "</ul>\n",
       "<a class=\"btn btn-default btn-impact\" data-event=\"impact-button-click\" data-test-id=\"view-article-impact\" href=\"http://loop-impact.frontiersin.org/impact/article/464100#views\" target=\"_blank\" type=\"button\">\n",
       "<span></span> View Article Impact\n",
       "            </a>\n",
       "</div>\n",
       "<div class=\"side-article-share\" style=\"display: flex;\">\n",
       "<h5 class=\"like-h4\">SHARE ON</h5>\n",
       "<ul class=\"share-media clearfix\" style=\"padding: 0;\">\n",
       "<li class=\"social_block\">\n",
       "<div class=\"atButton\">\n",
       "<a addthis:description=\"Succession is generally well described above-ground in the boreal forest, and several studies have demonstrated the role of interspecific facilitation in tree species establishment. However, the role of mycorrhizal communities for tree establishment and interspecific facilitation, has been little explored. At the ecotone between the mixed boreal forest, dominated by balsam fir and hardwood species, and the boreal forest, dominated by black spruce, several stands of trembling aspen can be found, surrounded by black spruce forest. Regeneration of balsam fir seems to have increased in the recent decades within the boreal forest, and it seems better adapted to grow in trembling aspen stands than in black spruce stands, even when located in similar abiotic conditions. As black spruce stands are also covered by ericaceous shrubs, we investigated if differences in soil fungal communities and ericaceous shrubs abundance could explain the differences observed in balsam fir growth and nutrition. We conducted a study centered on individual saplings to link growth and foliar nutrient concentrations to local vegetation cover, mycorrhization rate, and mycorrhizal communities associated with balsam fir roots. We found that foliar nutrient concentrations and ramification indices (colonization by mycorrhiza per length of root) were greater in trembling aspen stands and were positively correlated to apical and lateral growth of balsam fir saplings. In black spruce stands, the presence of er...\" addthis:title=\"Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?\" addthis:url=\"https://www.frontiersin.org/articles/10.3389/fpls.2019.00932\" class=\"addthis_button_facebook at300b\" data-event=\"shareicon-button-click\" title=\"Facebook\"><span class=\"at-icon-wrapper\" style=\"\n",
       "                              background-color: rgb(29, 161, 242);\n",
       "                              line-height: 16px;\n",
       "                              height: 16px;\n",
       "                              width: 16px;\n",
       "                            \"></span></a>\n",
       "</div>\n",
       "<a><span class=\"facebook_count\">0</span></a>\n",
       "</li>\n",
       "<li class=\"social_block\">\n",
       "<div class=\"atButton\">\n",
       "<a addthis:description=\"Succession is generally well described above-ground in the boreal forest, and several studies have demonstrated the role of interspecific facilitation in tree species establishment. However, the role of mycorrhizal communities for tree establishment and interspecific facilitation, has been little explored. At the ecotone between the mixed boreal forest, dominated by balsam fir and hardwood species, and the boreal forest, dominated by black spruce, several stands of trembling aspen can be found, surrounded by black spruce forest. Regeneration of balsam fir seems to have increased in the recent decades within the boreal forest, and it seems better adapted to grow in trembling aspen stands than in black spruce stands, even when located in similar abiotic conditions. As black spruce stands are also covered by ericaceous shrubs, we investigated if differences in soil fungal communities and ericaceous shrubs abundance could explain the differences observed in balsam fir growth and nutrition. We conducted a study centered on individual saplings to link growth and foliar nutrient concentrations to local vegetation cover, mycorrhization rate, and mycorrhizal communities associated with balsam fir roots. We found that foliar nutrient concentrations and ramification indices (colonization by mycorrhiza per length of root) were greater in trembling aspen stands and were positively correlated to apical and lateral growth of balsam fir saplings. In black spruce stands, the presence of er...\" addthis:title=\"Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?\" addthis:url=\"https://www.frontiersin.org/articles/10.3389/fpls.2019.00932\" class=\"addthis_button_twitter at300b\" data-event=\"shareicon-button-click\" title=\"Tweet\"><span class=\"at-icon-wrapper\" style=\"\n",
       "                              background-color: rgb(29, 161, 242);\n",
       "                              line-height: 16px;\n",
       "                              height: 16px;\n",
       "                              width: 16px;\n",
       "                            \"></span></a>\n",
       "</div>\n",
       "<a><span class=\"twitter_count\">0</span></a>\n",
       "</li>\n",
       "<li class=\"social_block\">\n",
       "<div class=\"atButton\">\n",
       "<a addthis:description=\"Succession is generally well described above-ground in the boreal forest, and several studies have demonstrated the role of interspecific facilitation in tree species establishment. However, the role of mycorrhizal communities for tree establishment and interspecific facilitation, has been little explored. At the ecotone between the mixed boreal forest, dominated by balsam fir and hardwood species, and the boreal forest, dominated by black spruce, several stands of trembling aspen can be found, surrounded by black spruce forest. Regeneration of balsam fir seems to have increased in the recent decades within the boreal forest, and it seems better adapted to grow in trembling aspen stands than in black spruce stands, even when located in similar abiotic conditions. As black spruce stands are also covered by ericaceous shrubs, we investigated if differences in soil fungal communities and ericaceous shrubs abundance could explain the differences observed in balsam fir growth and nutrition. We conducted a study centered on individual saplings to link growth and foliar nutrient concentrations to local vegetation cover, mycorrhization rate, and mycorrhizal communities associated with balsam fir roots. We found that foliar nutrient concentrations and ramification indices (colonization by mycorrhiza per length of root) were greater in trembling aspen stands and were positively correlated to apical and lateral growth of balsam fir saplings. In black spruce stands, the presence of er...\" addthis:title=\"Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?\" addthis:url=\"https://www.frontiersin.org/articles/10.3389/fpls.2019.00932\" class=\"addthis_button_google at300b\" data-event=\"shareicon-button-click\" title=\"Google Bookmark\"><span class=\"at-icon-wrapper\" style=\"\n",
       "                              background-color: rgb(29, 161, 242);\n",
       "                              line-height: 16px;\n",
       "                              height: 16px;\n",
       "                              width: 16px;\n",
       "                            \"></span></a>\n",
       "</div>\n",
       "<a><span class=\"twitter_count\">0</span></a>\n",
       "</li>\n",
       "<li class=\"social_block\">\n",
       "<div class=\"atButton\">\n",
       "<a addthis:description=\"Succession is generally well described above-ground in the boreal forest, and several studies have demonstrated the role of interspecific facilitation in tree species establishment. However, the role of mycorrhizal communities for tree establishment and interspecific facilitation, has been little explored. At the ecotone between the mixed boreal forest, dominated by balsam fir and hardwood species, and the boreal forest, dominated by black spruce, several stands of trembling aspen can be found, surrounded by black spruce forest. Regeneration of balsam fir seems to have increased in the recent decades within the boreal forest, and it seems better adapted to grow in trembling aspen stands than in black spruce stands, even when located in similar abiotic conditions. As black spruce stands are also covered by ericaceous shrubs, we investigated if differences in soil fungal communities and ericaceous shrubs abundance could explain the differences observed in balsam fir growth and nutrition. We conducted a study centered on individual saplings to link growth and foliar nutrient concentrations to local vegetation cover, mycorrhization rate, and mycorrhizal communities associated with balsam fir roots. We found that foliar nutrient concentrations and ramification indices (colonization by mycorrhiza per length of root) were greater in trembling aspen stands and were positively correlated to apical and lateral growth of balsam fir saplings. In black spruce stands, the presence of er...\" addthis:title=\"Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?\" addthis:url=\"https://www.frontiersin.org/articles/10.3389/fpls.2019.00932\" class=\"addthis_button_linkedin at300b\" data-event=\"shareicon-button-click\" title=\"LinkedIn\"><span class=\"at-icon-wrapper\" style=\"\n",
       "                              background-color: rgb(29, 161, 242);\n",
       "                              line-height: 16px;\n",
       "                              height: 16px;\n",
       "                              width: 16px;\n",
       "                            \"></span></a>\n",
       "</div>\n",
       "<a><span class=\"linkedin_count\">0</span></a>\n",
       "</li>\n",
       "<li class=\"social_block\">\n",
       "<div class=\"atButton\">\n",
       "<a addthis:description=\"Succession is generally well described above-ground in the boreal forest, and several studies have demonstrated the role of interspecific facilitation in tree species establishment. However, the role of mycorrhizal communities for tree establishment and interspecific facilitation, has been little explored. At the ecotone between the mixed boreal forest, dominated by balsam fir and hardwood species, and the boreal forest, dominated by black spruce, several stands of trembling aspen can be found, surrounded by black spruce forest. Regeneration of balsam fir seems to have increased in the recent decades within the boreal forest, and it seems better adapted to grow in trembling aspen stands than in black spruce stands, even when located in similar abiotic conditions. As black spruce stands are also covered by ericaceous shrubs, we investigated if differences in soil fungal communities and ericaceous shrubs abundance could explain the differences observed in balsam fir growth and nutrition. We conducted a study centered on individual saplings to link growth and foliar nutrient concentrations to local vegetation cover, mycorrhization rate, and mycorrhizal communities associated with balsam fir roots. We found that foliar nutrient concentrations and ramification indices (colonization by mycorrhiza per length of root) were greater in trembling aspen stands and were positively correlated to apical and lateral growth of balsam fir saplings. In black spruce stands, the presence of er...\" addthis:title=\"Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?\" addthis:url=\"https://www.frontiersin.org/articles/10.3389/fpls.2019.00932\" class=\"addthis_button_more at300b\" data-event=\"shareicon-button-click\" title=\"View more services\"><span class=\"at-icon-wrapper\" style=\"\n",
       "                              background-color: rgb(29, 161, 242);\n",
       "                              line-height: 16px;\n",
       "                              height: 16px;\n",
       "                              width: 16px;\n",
       "                            \"></span></a>\n",
       "</div>\n",
       "<a><span class=\"total_count\">New</span></a>\n",
       "</li>\n",
       "</ul>\n",
       "</div>\n",
       "<aside class=\"pull-left table-of-contents side-article\" id=\"anchors\">\n",
       "<div class=\"side-people hidden-sm hidden-xs\">\n",
       "<section class=\"side-article-editors\">\n",
       "<header><h5 class=\"like-h4\">Edited by</h5></header>\n",
       "<a class=\"authors\" href=\"https://loop.frontiersin.org/people/119073/overview\">\n",
       "<img alt=\" \" class=\"pr5 pull-left\" onerror=\"this.src = '/Areas/Articles/Images/Frontiers/Common/Profile/default-loop-profile.jpg'\" src=\"https://loop.frontiersin.org/images/profile/119073/24\"/>\n",
       "<h6 class=\"author-link-aside\">Kevin Garcia</h6>\n",
       "</a>\n",
       "<div class=\"clearfix\"></div>\n",
       "<p><span class=\"notes\">North Carolina State University, United States</span></p>\n",
       "<header><h5 class=\"like-h4\"> Reviewed by</h5></header>\n",
       "<a class=\"authors\" href=\"https://loop.frontiersin.org/people/113197/overview\">\n",
       "<img alt=\" \" class=\"pr5 pull-left\" onerror=\"this.src = '/Areas/Articles/Images/Frontiers/Common/Profile/default-loop-profile.jpg'\" src=\"https://loop.frontiersin.org/images/profile/113197/24\"/>\n",
       "<h6 class=\"author-link-aside\">Jaqueline Hess</h6>\n",
       "</a>\n",
       "<div class=\"clearfix\"></div>\n",
       "<p><span class=\"notes\">Helmholtz Centre for Environmental Research, Helmholtz Association of German Research Centres (HZ), Germany</span></p>\n",
       "<a class=\"authors\" href=\"https://loop.frontiersin.org/people/122827/overview\">\n",
       "<img alt=\" \" class=\"pr5 pull-left\" onerror=\"this.src = '/Areas/Articles/Images/Frontiers/Common/Profile/default-loop-profile.jpg'\" src=\"https://loop.frontiersin.org/images/profile/122827/24\"/>\n",
       "<h6 class=\"author-link-aside\">Plassard Claude</h6>\n",
       "</a>\n",
       "<div class=\"clearfix\"></div>\n",
       "<p><span class=\"notes\">IRD UMR210 Ecologie fonctionnelle et biogéochimie des sols et des agro-écosystèmes (Eco&amp;Sols), France</span></p>\n",
       "</section>\n",
       "</div>\n",
       "<nav>\n",
       "<header><h5 class=\"like-h4\" style=\"padding-top: 14px; padding-left: 6px; margin-bottom: 6px;\">TABLE OF CONTENTS</h5></header>\n",
       "<ul class=\"nav nav-list list-unstyled contents\" data-event=\"toc-link-click\">\n",
       "<li>\n",
       "<ul class=\"flyoutJournal\">\n",
       "<li><a href=\"#h1\">Abstract</a></li>\n",
       "<li><a href=\"#h2\">Introduction</a></li>\n",
       "<li><a href=\"#h3\">Materials and Methods</a></li>\n",
       "<li><a href=\"#h4\">Results</a></li>\n",
       "<li><a href=\"#h5\">Discussion</a></li>\n",
       "<li><a href=\"#h6\">Data Availability</a></li>\n",
       "<li><a href=\"#h7\">Author Contributions</a></li>\n",
       "<li><a href=\"#fun1\">Funding</a></li>\n",
       "<li><a href=\"#conf1\">Conflict of Interest Statement</a></li>\n",
       "<li><a href=\"#ack1\">Acknowledgments</a></li>\n",
       "<li><a href=\"#S9\">Supplementary Material</a></li>\n",
       "<li><a href=\"#FNT1\">Footnotes</a></li>\n",
       "<li><a href=\"#refer1\">References</a></li>\n",
       "</ul> </li>\n",
       "</ul>\n",
       "</nav>\n",
       "</aside>\n",
       "<div class=\"clearfix\"></div>\n",
       "<div class=\"side-article-download side-export clearfix\">\n",
       "<ul class=\"list-unstyled list-inline clearfix\">\n",
       "<li>\n",
       "<button class=\"btn btn-open-supplemental\" data-event=\"supplementary-button-click\" id=\"supplementary_view\">Open supplemental data</button>\n",
       "</li>\n",
       "<li class=\"dropdown text-center citation\">\n",
       "<button aria-expanded=\"false\" data-target=\"#\" data-test-id=\"citation-button\" data-toggle=\"dropdown\" role=\"button\">\n",
       "<span class=\"icon-label\" data-event=\"citation-button-click\">Export citation</span>\n",
       "</button>\n",
       "<ul class=\"dropdown-menu\" role=\"menu\">\n",
       "<li>\n",
       "<a data-event=\"endnotedownload-button-click\" data-test-id=\"article-endnote\" href=\"/articles/10.3389/fpls.2019.00932/endNote\">\n",
       "                                        EndNote\n",
       "                                    </a>\n",
       "</li>\n",
       "<li>\n",
       "<a data-event=\"referencemanagerdownload-button-click\" data-test-id=\"article-referencemanager\" href=\"/articles/10.3389/fpls.2019.00932/reference\">\n",
       "                                        Reference Manager\n",
       "                                    </a>\n",
       "</li>\n",
       "<li>\n",
       "<a data-event=\"simpletextfiledownload-button-click\" data-test-id=\"article-simpletextfile\" href=\"/articles/10.3389/fpls.2019.00932/text\">\n",
       "                                        Simple TEXT file\n",
       "                                    </a>\n",
       "</li>\n",
       "<li>\n",
       "<a data-event=\"bibtexdownload-button-click\" data-test-id=\"article-bibtex\" href=\"/articles/10.3389/fpls.2019.00932/bibTex\">\n",
       "                                        BibTex\n",
       "                                    </a>\n",
       "</li>\n",
       "</ul>\n",
       "</li>\n",
       "</ul>\n",
       "</div>\n",
       "<div class=\"side-crossmark\">\n",
       "<script src=\"https://crossmark-cdn.crossref.org/widget/v2.0/widget.js\"></script>\n",
       "<a class=\"crossmark\" data-target=\"crossmark\" style=\"cursor: pointer;\">\n",
       "<figure>\n",
       "<img alt=\"\" id=\"crossmark-icon\" src=\"https://crossmark-cdn.crossref.org/widget/v2.0/logos/CROSSMARK_BW_square_no_text.svg\" style=\"border: 0; width:64px; height:64px;\" title=\"\"/>\n",
       "</figure>\n",
       "<div id=\"crossmark-icon\">Check for updates</div>\n",
       "</a>\n",
       "</div>\n",
       "<article class=\"widget-listing people-also-looked-at side-article-related\">\n",
       "<h5 class=\"like-h4\" data-event=\"peoplelookedat-link-click\">People also looked at</h5>\n",
       "<div class=\"teaser\"><div class=\"heading-container\"><h2 class=\"teaser-heading\" data-event=\"peoplelookedat-link-click\"><a data-event=\"peoplelookedat-link-click\" href=\"/articles/10.3389/fpls.2022.759801/full\">A Closer Examination of the ‘Abundant-Center’ for Ectomycorrhizal Fungal Community Associated With Picea crassifolia in China</a></h2><p class=\"teaser-authors\" data-event=\"peoplelookedat-link-click\"><a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/1673684/overview\">Xiaobing Wang</a> and <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/1444600/overview\">Qisheng Han</a></p></div></div><div class=\"teaser\"><div class=\"heading-container\"><h2 class=\"teaser-heading\" data-event=\"peoplelookedat-link-click\"><a data-event=\"peoplelookedat-link-click\" href=\"/articles/10.3389/fpls.2019.00895/full\">Aphids Influence Soil Fungal Communities in Conventional Agricultural Systems</a></h2><p class=\"teaser-authors\" data-event=\"peoplelookedat-link-click\"><a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/639218/overview\">Thomas D. J. Wilkinson</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/736625/overview\">Jean-Pascal Miranda</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/699069/overview\">Julia Ferrari</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/122729/overview\">Sue E. Hartley</a> and <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/699023/overview\">Angela Hodge</a></p></div></div><div class=\"teaser\"><div class=\"heading-container\"><h2 class=\"teaser-heading\" data-event=\"peoplelookedat-link-click\"><a data-event=\"peoplelookedat-link-click\" href=\"/articles/10.3389/fmicb.2019.01888/full\">Perspectives on Endosymbiosis in Coralloid Roots: Association of Cycads and Cyanobacteria</a></h2><p class=\"teaser-authors\" data-event=\"peoplelookedat-link-click\"><a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/691177/overview\">Aimee Caye G. Chang</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/786490/overview\">Tao Chen</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/690638/overview\">Nan Li</a> and <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/201185/overview\">Jun Duan</a></p></div></div><div class=\"teaser\"><div class=\"heading-container\"><h2 class=\"teaser-heading\" data-event=\"peoplelookedat-link-click\"><a data-event=\"peoplelookedat-link-click\" href=\"/articles/10.3389/fpls.2019.00931/full\">CrERF5, an AP2/ERF Transcription Factor, Positively Regulates the Biosynthesis of Bisindole Alkaloids and Their Precursors in Catharanthus roseus</a></h2><p class=\"teaser-authors\" data-event=\"peoplelookedat-link-click\"><a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/625039/overview\">Qifang Pan</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/700893/overview\">Chenyi Wang</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/701090/overview\">Zhiwei Xiong</a>, Hang Wang, Xueqing Fu, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/366528/overview\">Qian Shen</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/669885/overview\">Bowen Peng</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/771064/overview\">Yanan Ma</a>, Xiaofen Sun and <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/219192/overview\">Kexuan Tang</a></p></div></div><div class=\"teaser\"><div class=\"heading-container\"><h2 class=\"teaser-heading\" data-event=\"peoplelookedat-link-click\"><a data-event=\"peoplelookedat-link-click\" href=\"/articles/10.3389/fpls.2019.00935/full\">Marchantia polymorpha, a New Model Plant for Autophagy Studies</a></h2><p class=\"teaser-authors\" data-event=\"peoplelookedat-link-click\"><a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/740213/overview\">Takuya Norizuki</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/740248/overview\">Takehiko Kanazawa</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/740355/overview\">Naoki Minamino</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/26459/overview\">Hirokazu Tsukaya</a> and <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/49517/overview\">Takashi Ueda</a></p></div></div></article>\n",
       "</aside>\n",
       "<main class=\"\">\n",
       "<div class=\"article-section\">\n",
       "<div class=\"article-container\" data-html=\"True\">\n",
       "<div class=\"abstract-container\">\n",
       "<div class=\"article-header-container\">\n",
       "<!-- Start CrossMark Snippet v2.0 -->\n",
       "<!-- End CrossMark Snippet -->\n",
       "<div class=\"header-bar-one\">\n",
       "<h2>\n",
       "                                ORIGINAL RESEARCH article\n",
       "                            </h2>\n",
       "</div>\n",
       "<div class=\"header-bar-three-container\">\n",
       "<div class=\"header-bar-three\">\n",
       "                            Front. Plant Sci., 18 July 2019<br/>Sec. Plant Pathogen Interactions\n",
       "\n",
       "                                <br/>\n",
       "<a href=\"https://doi.org/10.3389/fpls.2019.00932\">https://doi.org/10.3389/fpls.2019.00932</a>\n",
       "</div>\n",
       "<div class=\"research-topic-container\" style=\"display: block;\">\n",
       "<a data-test-id=\"relatedRT-link\" href=\"/research-topics/8849#articles\">\n",
       "<div class=\"research-topic-data\">\n",
       "<h5 class=\"topic-title\">\n",
       "<span>This article is part of the Research Topic</span>\n",
       "</h5>\n",
       "<p style=\"margin-bottom:0;\">\n",
       "\n",
       "                    Importance of Root Symbiomes for Plant Nutrition: New Insights, Perspectives and Future Challenges\n",
       "                </p>\n",
       "<div class=\"topic-link-trim\" data-event=\"rt-link-click\">\n",
       "                        View all\n",
       "13                        Articles <span style=\"position: relative;top: 1px;\"></span>\n",
       "</div>\n",
       "</div>\n",
       "</a>\n",
       "</div></div>\n",
       "</div>\n",
       "<div class=\"JournalAbstract\">\n",
       "<a class=\"reset-hash-position\" id=\"h1\" name=\"h1\"></a><h1>Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter?</h1>\n",
       "<div class=\"authors\">\n",
       "<a class=\"user-id-639840\" href=\"https://www.frontiersin.org/people/u/639840\"><img alt=\"\" class=\"pr5\" onerror=\"this.src='https://f96a1a95aaa960e01625-a34624e694c43cdf8b40aa048a644ca4.ssl.cf2.rackcdn.com/Design/Images/newprofile_default_profileimage_new.jpg'\" src=\"https://loop.frontiersin.org/images/profile/639840/24\"/>Mélissande Nagati</a><sup>1,2*</sup>, <a class=\"user-id-399792\" href=\"https://www.frontiersin.org/people/u/399792\"><img alt=\"\" class=\"pr5\" onerror=\"this.src='https://f96a1a95aaa960e01625-a34624e694c43cdf8b40aa048a644ca4.ssl.cf2.rackcdn.com/Design/Images/newprofile_default_profileimage_new.jpg'\" src=\"https://loop.frontiersin.org/images/profile/399792/24\"/>Mélanie Roy</a><sup>2</sup>, <img alt=\"\" class=\"pr5\" src=\"https://f96a1a95aaa960e01625-a34624e694c43cdf8b40aa048a644ca4.ssl.cf2.rackcdn.com/Design/Images/newprofile_default_profileimage_new.jpg\"/>Annie Desrochers<sup>1</sup>, <img alt=\"\" class=\"pr5\" src=\"https://f96a1a95aaa960e01625-a34624e694c43cdf8b40aa048a644ca4.ssl.cf2.rackcdn.com/Design/Images/newprofile_default_profileimage_new.jpg\"/>Sophie Manzi<sup>2</sup>, <a class=\"user-id-112075\" href=\"https://www.frontiersin.org/people/u/112075\"><img alt=\"\" class=\"pr5\" onerror=\"this.src='https://f96a1a95aaa960e01625-a34624e694c43cdf8b40aa048a644ca4.ssl.cf2.rackcdn.com/Design/Images/newprofile_default_profileimage_new.jpg'\" src=\"https://loop.frontiersin.org/images/profile/112075/24\"/>Yves Bergeron</a><sup>1</sup> and <a class=\"user-id-766853\" href=\"https://www.frontiersin.org/people/u/766853\"><img alt=\"\" class=\"pr5\" onerror=\"this.src='https://f96a1a95aaa960e01625-a34624e694c43cdf8b40aa048a644ca4.ssl.cf2.rackcdn.com/Design/Images/newprofile_default_profileimage_new.jpg'\" src=\"https://loop.frontiersin.org/images/profile/766853/24\"/>Monique Gardes</a><sup>2</sup></div>\n",
       "<ul class=\"notes\">\n",
       "<li><span><sup>1</sup></span>UQAT-UQAM Industrial Chair in Sustainable Forest Management, Forest Research Institute, University of Québec in Abitibi-Témiscamingue, Rouyn-Noranda, QC, Canada</li>\n",
       "<li><span><sup>2</sup></span>UMR5174, Laboratory Evolution and Biological Diversity, Centre National de la Recherche Scientifique – IRD, Université Paul Sabatier, Toulouse, France</li>\n",
       "</ul>\n",
       "<p class=\"mb0\">Succession is generally well described above-ground in the boreal forest, and several studies have demonstrated the role of interspecific facilitation in tree species establishment. However, the role of mycorrhizal communities for tree establishment and interspecific facilitation, has been little explored. At the ecotone between the mixed boreal forest, dominated by balsam fir and hardwood species, and the boreal forest, dominated by black spruce, several stands of trembling aspen can be found, surrounded by black spruce forest. Regeneration of balsam fir seems to have increased in the recent decades within the boreal forest, and it seems better adapted to grow in trembling aspen stands than in black spruce stands, even when located in similar abiotic conditions. As black spruce stands are also covered by ericaceous shrubs, we investigated if differences in soil fungal communities and ericaceous shrubs abundance could explain the differences observed in balsam fir growth and nutrition. We conducted a study centered on individual saplings to link growth and foliar nutrient concentrations to local vegetation cover, mycorrhization rate, and mycorrhizal communities associated with balsam fir roots. We found that foliar nutrient concentrations and ramification indices (colonization by mycorrhiza per length of root) were greater in trembling aspen stands and were positively correlated to apical and lateral growth of balsam fir saplings. In black spruce stands, the presence of ericaceous shrubs near balsam fir saplings affected ectomycorrhizal communities associated with tree roots which in turn negatively correlated with N foliar concentrations. Our results reveal that fungal communities observed under aspen are drivers of balsam fir early growth and nutrition in boreal forest stands and may facilitate ecotone migration in a context of climate change.</p>\n",
       "<div class=\"clear\"></div>\n",
       "</div>\n",
       "<div class=\"JournalFullText\">\n",
       "<a class=\"reset-hash-position\" id=\"h2\" name=\"h2\"></a><h2>Introduction</h2>\n",
       "<p class=\"mb15\">Tree establishment, growth and survival in a new area are primarily dependent upon seed availability and local environmental conditions. For a given tree species, its capacity to establish under new biotic conditions is predominantly driven by either facilitation or competitive processes (<a href=\"#B10\">Callaway and Walker, 1997</a>). Within a multi-species environment, species can compete for resources such as light, nutrients or water. At the same time, survival of one species can be facilitated by another species providing protection against predators or extreme climate events (<a href=\"#B59\">Stachowicz, 2001</a>), or by the presence of their mycorrhizal symbionts which could result in mycorrhizal networks (<a href=\"#B56\">Simard et al., 2015</a>; <a href=\"#B47\">Pickles and Simard, 2017</a>), or exchanges of resources (<a href=\"#B8\">Brooker et al., 2008</a>; <a href=\"#B63\">Teste et al., 2009</a>). Among the consequences of climate change, changes in species distribution are already observed and could result in new or modified species interactions. Changes in distribution range of species are particularly visible at ecotones between forested ecosystems (<a href=\"#B37\">Messaoud et al., 2007</a>; <a href=\"#B2\">Barbeta and Peñuelas, 2017</a>) or at the northern tree line (<a href=\"#B19\">Harsch et al., 2009</a>; <a href=\"#B49\">Ratcliffe et al., 2017</a>).</p>\n",
       "<p class=\"mb15\">In Québec (Canada), the 49th parallel represents the ecotone between the southern balsam fir-paper birch (mixedwood boreal forest) and the northern black spruce-feather moss (boreal forest) bioclimatic domains. It has been suggested that this ecotone is likely to migrate northward with climate change (<a href=\"#B37\">Messaoud et al., 2007</a>). South of this ecotone, forest stands are mixed and dominated by <i>Abies balsamea</i> [L.] Mill. (balsam fir-BF), <i>Picea glauca</i> [Moench] Voss (white spruce), <i>Populus tremuloides</i> Michx.(trembling aspen-TA), and <i>Betula papyrifera</i> Marsh. (paper birch), whereas <i>Picea mariana</i> [Mill.] BSP (black spruce-BS), and <i>Pinus banksiana</i> Lamb.(Jack pine) dominate north of the ecotone (<a href=\"#B52\">Robitaille and Saucier, 1996</a>). In the southern part of the black spruce-feather moss domain, small stands of trembling aspen occur, in which trembling aspen can be dominant (more than 75% of the canopy cover) or mixed with black spruce (between 25 and 75% of the canopy cover; <a href=\"#B32\">Légaré et al., 2005</a>; <a href=\"#B11\">Cavard et al., 2011</a>). Following classical forest dynamics in these forests, TA would eventually be replaced by BS in the absence of disturbance (<a href=\"#B31\">Lecomte and Bergeron, 2005</a>; <a href=\"#B4\">Belleau et al., 2011</a>), thereby maintaining BS dominance in the area. In areas where balsam fir establishes under BS- and TA-dominated stands, a greater abundance of more vigorous saplings has been observed under TA (<a href=\"#B1\">Arbour and Bergeron, 2011</a>). As those two types of stands are very close geographically and grow under similar climate, slope and substrates (<a href=\"#B32\">Légaré et al., 2005</a>; <a href=\"#B11\">Cavard et al., 2011</a>), the idea of an abiotic determinism could be rejected. On the contrary, aspen-dominated and spruce-dominated stands have different understorey plant and fungal communities, and soil organic contents (<a href=\"#B32\">Légaré et al., 2005</a>; <a href=\"#B11\">Cavard et al., 2011</a>; <a href=\"#B38\">Nagati et al., 2018</a>), suggesting that biotic factors could explain differences in balsam fir growth. Indeed several mechanisms resulting from microbe–mediated interactions could enhance or reduce BF nutrition and growth, and may be involved in plant–plant interactions in the boreal forest.</p>\n",
       "<p class=\"mb15\">Among the fungi that interact with tree roots, including BF roots, ectomycorrhizal fungi (EMF) are particularly abundant in boreal soils (<a href=\"#B60\">Taylor et al., 2013</a>). These fungi improve tree growth and foliar nutrient status in ecosystems where nutrients are mainly found in organic forms, as in boreal forests (<a href=\"#B58\">Smith and Read, 2008</a>; <a href=\"#B21\">Inselsbacher and Näsholm, 2012</a>; <a href=\"#B16\">Franklin et al., 2014</a>). At a worldwide scale, closely related host species tend to share more similar EMF communities than more phylogenetically distant host species (<a href=\"#B62\">Tedersoo et al., 2013</a>), which may result in greater similarity between BF and BS fungal communities. However, at the local scale, cases of facilitation through mycorrhizal symbioses have rather been detected between phylogenetically distinct plants (<a href=\"#B40\">Nara and Hogetsu, 2004</a>; <a href=\"#B13\">Dickie et al., 2006</a>; <a href=\"#B65\">van der Heijden and Horton, 2009</a>). Mycorrhizal fungi could also shape plant–plant interactions through indirect interactions with plants (<a href=\"#B5\">Bever et al., 2010</a>). The understorey vegetation under TA is dominated by various ectomycorrhizal tree saplings, endomycorrhizal (AM) shrubs and herbaceous species (<a href=\"#B32\">Légaré et al., 2005</a>; <a href=\"#B11\">Cavard et al., 2011</a>) while BS stands understorey is covered by a thick layer of bryophytes and ericaceous shrubs. Limitations to tree establishment and growth for EMF tree species that are adjacent to ericaceous shrubs, which are associated with ericoid fungi, have been documented many times in various ecosystems (e.g., <a href=\"#B46\">Peterson, 1965</a>; <a href=\"#B72\">Zackrisson et al., 1997</a>; <a href=\"#B71\">Yamasaki et al., 1998</a>; <a href=\"#B68\">Walker et al., 1999</a>; <a href=\"#B35\">Mallik, 2003</a>). Although the mechanisms explaining such regeneration failures of EMF plant species near ericoid shrubs are not clearly understood (<a href=\"#B46\">Peterson, 1965</a>; <a href=\"#B17\">Gallet, 1994</a>; <a href=\"#B12\">Collier and Bidartondo, 2009</a>; <a href=\"#B51\">Richard et al., 2009</a>), potential alterations to BF mycorrhizal colonization and mycorrhizal networks cannot be excluded in our system that would explain BF growth differences between TA- and BS-dominated stands.</p>\n",
       "<p class=\"mb0\">Our aim was to disentangle the processes controlling BF early growth and nutrition at this boreal ecotone. To achieve this goal, we monitored growth parameters of individual BF saplings for two consecutive years and investigated their EMF communities and their foliar nutrient concentrations under BS and TA stands. We hypothesized that (1) local conditions under TA would lead to greater mycorrhization rates and different EMF communities than under BS and (2) the presence of ericaceous shrubs near BF saplings would reduce growth and mineral nutrition through their effects on EMF symbioses.</p>\n",
       "<a class=\"reset-hash-position\" id=\"h3\" name=\"h3\"></a><h2>Materials and Methods</h2>\n",
       "<h3 class=\"pt0\">Site Description and Fir Sapling Selection</h3>\n",
       "<p class=\"mb0\">The study was located across a 36-km<sup>2</sup> area within the Clay Belt of northern Québec and Ontario (Canada). Four paired and unmanaged sites within the black spruce-feather moss domain were selected for study (<a href=\"#T1\">Table 1</a>). Each pair of stands contained one that was dominated by TA, while the other one was dominated by BS (for more information on sites, see <a href=\"#B38\">Nagati et al., 2018</a>). At each site, 15 BF saplings that were 25–75 cm tall and separated from one another by at least 5 m were selected: five in TA stands; five in BS stands at least 4m away from an ericaceous shrub; and five in BS stands at a maximum distance of 3 m from an ericaceous shrub [BSE, Labrador tea (<i>Rhododendron groenlandicum</i> [Oeder] Kron and Judd) or sheep-laurel (<i>Kalmia angustifolia</i> L.)]. Our sampling design permitted to sample 15 saplings per site (10 in BS stands and 5 in TA stands), resulting in 60 saplings.</p>\n",
       "<div class=\"DottedLine\"></div>\n",
       "<div class=\"Imageheaders\">TABLE 1</div>\n",
       "<div class=\"FigureDesc\">\n",
       "<a href=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-t001.jpg\" name=\"table1\" target=\"_blank\">\n",
       "<img alt=\"www.frontiersin.org\" class=\"lazy\" data-src=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-t001.jpg\" id=\"T1\"/></a>\n",
       "<p><strong>Table 1.</strong> GPS coordinates of sites.</p>\n",
       "</div>\n",
       "<div class=\"clear\"></div>\n",
       "<div class=\"DottedLine\"></div>\n",
       "<h3>Balsam Fir Measurements</h3>\n",
       "<p class=\"mb0\">Annual apical growth (cm) was measured in August 2015 and 2016 with calipers and the two measures were summed. Annual lateral growth was measured on two randomly selected branches per sapling in August 2015 and 2016 with calipers, averaged by year, and then summed. In 2016, the basal diameter of stems (cm) was measured with calipers. Fir needles were harvested in August 2016, forced air-dried at 30°C for 12 h, and sent to the Laurentian Forestry Centre, Canadian Forest Service (Quebec) for chemical analyses. Percentage foliar C and N were measured with a Leco TruMac CNS mass spectrometer (LECO Corporation, St. Joseph, MI, United States). Minor and major cation concentrations (g/kg of needles) were obtained by ICP-OES with a Perkin-Elmer Optima 7300 DV (Waltham, MA, United States), following the method proposed by <a href=\"#B27\">Kalra (1998)</a>.</p>\n",
       "<h3>Characterization of Local Biotic Environment</h3>\n",
       "<p class=\"mb0\">Given that the main goal of our study was to determine whether biotic interactions affected BF growth and foliar nutritional status, we characterized their local biotic environment. Forest composition within a 3 m radius around each balsam fir sapling was evaluated by measuring the percentages of TA, BS and BF relative to their diameter at DBH (diameter at breast height, 1.3 m). Only trees &gt;5 cm DBH were taken into account. These data allowed to calculate the percent of each tree species (TA, BS, and BF) in the canopy around each BF sapling.</p>\n",
       "<h3>Mycorrhizal Root Tip Counts, Mycorrhizal DNA Extraction, Amplification, and Sequencing</h3>\n",
       "<p class=\"mb0\">Mycorrhizal communities were assessed for each sapling to test their effect on individual BF growth and foliar nutrient status. The root system of each sapling was gently excavated in August 2016 and washed to preserve mycorrhizal root tips. Mycorrhizal root tip counts were performed the same day that the roots were extracted. For each sapling, we randomly selected three 10-cm root fragments for live mycorrhizal root tip counts under a magnifying glass to calculate mycorrhization rate (number of EMF root tips/total number of root tips) and ramification index (number of EMF root tips per 10 cm root). For each fir sapling, 50 mycorrhizal root tips were sampled and stored in CTAB 2X at −20°C until DNA extraction. Root tips were pooled by tree sapling and manually ground with a pestle before DNA extraction. DNA was extracted with a PowerSoil DNA extraction kit (MoBio, Carlsbad, CA, United States) following the manufacturer’s instructions. A negative blank extraction (extraction without any material) was performed for every set of 23 extractions. In addition, a tool control was performed on tools that were used for field and laboratory work by extracting DNA from distilled water that was used to wash tools. DNA amplification was performed using the method described in <a href=\"#B38\">Nagati et al. (2018)</a>. Briefly, the fungal ITS1 region (Forward: ITS5 GGAAGTAAAAGTCGTAACAAGG, <a href=\"#B69\">White et al., 1990</a>; and a modified version of Reverse: 5.8S_Fungi CAAGAGATCCGTTGTTGAAAGTK, <a href=\"#B14\">Epp et al., 2012</a>) was amplified for 35 PCR cycles. PCR samples were sent to GENETOUL GetPlaGe (Toulouse, France) for sequencing on an Illumina MiSeq platform with the TruSeq Nano PCR-free kit. Sequencing was conducted using the paired-end sequencing technology (2 × 250pb) with the chemistry V2.</p>\n",
       "<h3>Bioinformatics and Sequence Analysis</h3>\n",
       "<p class=\"mb0\">An abundance matrix was constructed with <i>OBITool</i> packages (<a href=\"#B6\">Boyer et al., 2016</a>) and R script (<a href=\"#B48\">R Core Team, 2018</a>, version 3.2.3, 2017) based upon the occurrence of sequences among samples. We performed read-pairing assembly, read attribution to samples, read dereplication, and removal of low-quality sequences (shorter than expected, containing ambiguous nucleotides, displaying low score paired-end alignments, or corresponding to singletons). <i>OBITool Sumaclust</i> was used to cluster sequences as OTUs (Operational Taxonomic Units) at a 97% identity threshold (<a href=\"#B42\">Nilsson et al., 2008</a>). Taxonomic assignment was performed with the <i>OBITool Ecotag</i> function against the GenBank extracted database<sup id=\"footnotesuper1\"><a id=\"footnote1a\"></a><a class=\"footnoteanchor\" href=\"#footnote1\">1</a></sup>. As was the case for our previous dataset that was collected in the same geographic area (<a href=\"#B38\">Nagati et al., 2018</a>), taxonomic assignments were more accurate with Genbank than with the UNITE public database<sup id=\"footnotesuper2\"><a id=\"footnote2a\"></a><a class=\"footnoteanchor\" href=\"#footnote2\">2</a></sup>. Trophic guild assignment was based upon FUNguild software outputs (<a href=\"#B41\">Nguyen et al., 2016</a>). Sequences belonging to the same OTU were then summed by sample. Lastly, we removed OTUs that were dominant (with the highest read count) in negative or tool extraction and amplification controls, OTUs not belonging to fungi, or OTUs with coarse-resolution taxonomic assignments (i.e., assigned to Eukaryota).</p>\n",
       "<h3>Statistical Analyses</h3>\n",
       "<p class=\"mb15\">Statistical analyses were performed in R (<a href=\"#B48\">R Core Team, 2018</a>). Data are available at Dryad repository (doi: <a href=\"https://doi.org/10.5061/dryad.914j5m0\">10.5061/dryad.914j5m0</a>) Our main goal was to compare individual traits and fungal communities among sapling types (BS, BSE, and TA). Abundance of each OTU was used to avoid giving too much importance to rare OTUs, as recommended for fungal ITS (<a href=\"#B64\">Unterseher et al., 2011</a>; <a href=\"#B33\">Lindahl et al., 2013</a>). Ectomycorrhizal species richness, Shannon index (<i>H</i>’), ramification index and mycorrhization rate were each calculated and compared according to one-way ANOVA (three levels), followed by Tukey <i>post hoc</i> tests of the means. Foliar nutrient status among saplings growing under BS, BSE, and TA, were compared for each nutrient (N, P, K, Ca, and Mg) with non-parametric Kruskall–Wallis tests, given that the data were not normally distributed, followed by Dunn’s <i>post hoc</i> tests with Bonferroni corrections. As well, sapling growth between the three modalities was compared with Kruskall-Wallis tests and Dunn’s <i>post hoc</i> tests with Bonferonni corrections.</p>\n",
       "<p class=\"mb0\">Differences in abundance of each ectomycorrhizal family represented in root tip samples were tested between sapling types with Kruskall–Wallis tests, followed by Dunn <i>post hoc</i> tests with Bonferroni corrections. We only tested differences in abundance at the family level, given that about 30% of OTUs could not be assigned to a genus. Differences in ectomycorrhizal community structure between the roots of the three sapling types were visually described with Non-Metric Multidimensional Scaling (NMDS, <i>vegan</i> package in R; <a href=\"#B43\">Oksanen et al., 2017</a>) and coordinates of scores on the first two axes of NMDS were extracted for further analyses. Correlation between the NMDS space and individual measures of fir saplings (nutrient concentrations and growth) were tested with envfit (vegan package). Envfit vectors of individual measures were plotted on the NMDS space when <i>p</i>-Values where significant (<i>p</i> &lt; 0.05). Differences in ectomycorrhizal community structure between sites, dominant plant communities (BS, BSE, and TA) were tested with PERMANOVA (<i>Adonis</i> function, <i>vegan</i> package; <a href=\"#B43\">Oksanen et al., 2017</a>) with nested factors (site/plant community).</p>\n",
       "<h3>Path Analysis</h3>\n",
       "<p class=\"mb0\">To test direct and indirect effects of the local biotic environment on growth (for this analysis, growth measures from 2015 to 2016 were summed) and foliar N concentrations, we compared three hypotheses by fitting structural equation models (SEM) to the data (with lavaan package; <a href=\"#B53\">Rosseel et al., 2018</a>). We included only foliar N concentrations in our model, as it was the only nutrient concentration available for all saplings. We constructed three SEMs to represent three <i>a priori</i> hypotheses. Each of the hypotheses is rooted in current knowledge regarding the processes that have been described in the literature, and which are described below. All models were fitted to centered and reduced data.</p>\n",
       "<h4>Complete Model Mod1</h4>\n",
       "<p class=\"mb15\">Considering direct links, we formulated five hypotheses. The first was that foliar N concentration was positively correlated with growth (<a href=\"#B44\">Pallardy and Kozlowski, 2008</a>). The second one was that EMF abundance and communities were positively correlated with growth and foliar N concentrations (<a href=\"#B58\">Smith and Read, 2008</a>). The third hypothesis was that BS stands and the percent of BS near fir saplings were negatively correlated with BF growth and foliar N concentrations (<a href=\"#B1\">Arbour and Bergeron, 2011</a>). The fourth hypothesis was that the presence of ericaceous shrubs near saplings was negatively correlated with BF growth and nutrition (<a href=\"#B46\">Peterson, 1965</a>; <a href=\"#B71\">Yamasaki et al., 1998</a>; <a href=\"#B35\">Mallik, 2003</a>). Finally, we hypothesized that the percentage of conspecific mature trees near BF saplings was positively correlated with their growth and foliar N, given that local conditions have permitted their growth and survival. Lateral and apical growth were added as co-variables in the model and together co-varied with basal diameter of BF saplings.</p>\n",
       "<p class=\"mb0\">For this model, we also formulated indirect linkage hypotheses. The relative percentage of BF was correlated with stand type (<a href=\"#B1\">Arbour and Bergeron, 2011</a>). The relative percentage of BS was correlated with the stand (<a href=\"#B11\">Cavard et al., 2011</a>). The presence of ericaceous shrubs was correlated with the stand type (BF saplings in TA stands were never next to ericaceous shrubs). The ramification index was correlated with the stand (see <a href=\"#F3\">Figure 3</a>), and percent of BF (the presence of mature trees near BF saplings increased the probability of encountering EMF partners). The NMDS first axis was correlated with the stand and the second axis negatively with the presence of ericaceous shrubs (see <a href=\"#F5\">Figure 5</a> and PERMANOVA results). The complete model is presented in <a href=\"#F1\">Figure 1</a>.</p>\n",
       "<div class=\"DottedLine\"></div>\n",
       "<div class=\"Imageheaders\">FIGURE 1</div>\n",
       "<div class=\"FigureDesc\">\n",
       "<a href=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-g001.jpg\" name=\"figure1\" target=\"_blank\">\n",
       "<img alt=\"www.frontiersin.org\" class=\"lazy\" data-src=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-g001.jpg\" id=\"F1\"/></a>\n",
       "<p><strong>Figure 1.</strong> Direct acyclic graph of Mod1 (red, blue, and black arrows), Mod2 (blue and black arrows), and Mod3 (black arrows).</p>\n",
       "</div>\n",
       "<div class=\"clear\"></div>\n",
       "<div class=\"DottedLine\"></div>\n",
       "<h4>Nutrient Model Mod2</h4>\n",
       "<p class=\"mb0\">The indirect links are the same as for Mod1 but in this model we assumed that foliar N concentration was the only variable that was correlated to growth.</p>\n",
       "<h4>Fungi Centered Model Mod3</h4>\n",
       "<p class=\"mb15\">The links are the same as for Mod2, but in this model we assumed that only variables linked to EMF were correlated with foliar N concentrations. In this model, NMDS first and second axes and ramification index were the only variables directly linked to N concentrations.</p>\n",
       "<p class=\"mb0\">To ensure that our models respected independence between non-linked variables, we performed Fisher’s C test (<i>ggm</i> package, <a href=\"#B36\">Marchetti et al., 2015</a>). Models with <i>p</i>-Values greater than 0.05 are considered to have respected claims of independence (<a href=\"#B54\">Shipley, 2000</a>). A model was considered to be representative of the population if the <i>p</i>-Value of the Chi-square test was greater than 0.05. For each model, we calculated the Comparative Fit Index (CFI) and the Tucker-Lewis Index (TLI) to evaluate how each model fits to the data. Values greater than 0.95 were considered to be good fits (<a href=\"#B20\">Hu and Bentler, 1999</a>). Each structural equation model with Fisher’s C test and Chi-Square <i>p</i>-values &gt; 0.05, CFI &gt; 0.95, and TLI &gt; 0.95 could describe the data well, we used these values as a primary filter to ensure that models fit well the data and are representative of the population. After applying this filter our aim was to select the best model based on AIC criterion, however, only Mod1 passed the primary filter and was de facto selected.</p>\n",
       "<a class=\"reset-hash-position\" id=\"h4\" name=\"h4\"></a><h2>Results</h2>\n",
       "<h3 class=\"pt0\">Fir Sapling Traits and Differences Between Stands</h3>\n",
       "<p class=\"mb15\">Given that 6 of the 60 BF saplings were missing after the first year of fieldwork, the results presented here are based on 18 saplings for each sapling type (i.e., 54 saplings). For 11 saplings, the quantity of needles was insufficient to measure minor and major cations, thereby further reducing the sample size.</p>\n",
       "<p class=\"mb0\">Mean foliar N concentrations of BF saplings were highest in TA stands compared to BS and BSE stands (<i>p</i> &lt; 0.05, <a href=\"#F2\">Figure 2A</a>). Mean foliar K concentration was higher in TA than in BSE stands (<i>p</i> &lt; 0.05, <a href=\"#F2\">Figure 2C</a>). Foliar P (<a href=\"#F2\">Figure 2B</a>), Ca, and Mg did not vary between sapling types (<i>p</i> &gt; 0.05). No significant differences were found in root tip EMF richness and Shannon index among sapling types (<a href=\"#F3\">Figure 3A</a>, <i>p</i> &gt; 0.05). Mean lateral growth of fir saplings did not differ between stands in 2015 and 2016. Apical growth was greater in TA than in BSE stands in 2015, while no difference was detected for apical growth in 2016. Summed lateral and apical growth did not differ between stands. Ramification index of BF roots was higher in TA than in other stands (<a href=\"#F3\">Figure 3B</a>, <i>p</i> &lt; 0.05).</p>\n",
       "<div class=\"DottedLine\"></div>\n",
       "<div class=\"Imageheaders\">FIGURE 2</div>\n",
       "<div class=\"FigureDesc\">\n",
       "<a href=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-g002.jpg\" name=\"figure2\" target=\"_blank\">\n",
       "<img alt=\"www.frontiersin.org\" class=\"lazy\" data-src=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-g002.jpg\" id=\"F2\"/></a>\n",
       "<p><strong>Figure 2.</strong> Boxplot chart representing foliar <b>(A)</b> N, <b>(B)</b> P, and <b>(C)</b> K concentrations of BF needles sampled in different stands (BS, black spruce; BSE, black spruce plus ericaceous shrubs; TA, trembling aspen). Different letters indicate differences between modalities.</p>\n",
       "</div>\n",
       "<div class=\"clear\"></div>\n",
       "<div class=\"DottedLine mb15\"></div>\n",
       "<div class=\"Imageheaders\">FIGURE 3</div>\n",
       "<div class=\"FigureDesc\">\n",
       "<a href=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-g003.jpg\" name=\"figure3\" target=\"_blank\">\n",
       "<img alt=\"www.frontiersin.org\" class=\"lazy\" data-src=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-g003.jpg\" id=\"F3\"/></a>\n",
       "<p><strong>Figure 3.</strong> Boxplot charts showing <b>(A)</b> EMF richness and <b>(B)</b> number of EMF root tips per 10 cm roots of BF saplings sampled in different stands. BS, black spruce; BSE, black spruce plus ericaceous shrubs; TA, trembling aspen. Different letters indicate differences between modalities.</p>\n",
       "</div>\n",
       "<div class=\"clear\"></div>\n",
       "<div class=\"DottedLine\"></div>\n",
       "<h3>Ectomycorrhizal Community</h3>\n",
       "<p class=\"mb0\">A total of 400 EMF OTUs (655,495 reads) were found in EMF root tips from BF saplings, representing 19.5% of OTUs and 59.7% of reads, with an average of 35.4 OTUs and 12,056.8 reads per sample. Variation in EMF root tip communities was explained by plant community (<i>p</i>-Value = 0.035, <i>F</i> = 1.1852, <i>R</i><sup>2</sup> = 0.169), but no effect of site was detected (<a href=\"#T2\">Table 2</a>). Clavulinaceae were only found under TA. Further, the abundance of Cortinariaceae was higher under BS compared to TA stands, that of Gloniaceae was higher under TA than under BSE, and that of Helotiaceae was higher under BSE than under TA. Finally, abundance of Inocybaceae and Thelephoraceae were greater under TA than under BS and BSE (Kruskal-Wallis and Dunn <i>post hoc</i> tests, <i>p</i> &lt; 0.025, <a href=\"#F4\">Figure 4</a>). According to the NMDS, EMF communities on root tips from the TA stands were distinct from those under BS and BSE, while communities in BS and BSE overlapped (<a href=\"#F5\">Figure 5</a>). <i>Envfit</i> test demonstrated that N concentration were significantly associated with NMDS structure (<i>p</i> &lt; 0.05, <a href=\"#F5\">Figure 5</a>).</p>\n",
       "<div class=\"DottedLine\"></div>\n",
       "<div class=\"Imageheaders\">TABLE 2</div>\n",
       "<div class=\"FigureDesc\">\n",
       "<a href=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-t002.jpg\" name=\"table2\" target=\"_blank\">\n",
       "<img alt=\"www.frontiersin.org\" class=\"lazy\" data-src=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-t002.jpg\" id=\"T2\"/></a>\n",
       "<p><strong>Table 2.</strong> Results of PERMANOVA conducted on EMF community of balsam fir saplings.</p>\n",
       "</div>\n",
       "<div class=\"clear\"></div>\n",
       "<div class=\"DottedLine mb15\"></div>\n",
       "<div class=\"Imageheaders\">FIGURE 4</div>\n",
       "<div class=\"FigureDesc\">\n",
       "<a href=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-g004.jpg\" name=\"figure4\" target=\"_blank\">\n",
       "<img alt=\"www.frontiersin.org\" class=\"lazy\" data-src=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-g004.jpg\" id=\"F4\"/></a>\n",
       "<p><strong>Figure 4.</strong> Pie chart representing percent of EMF families per stand type for root tips samples of BF saplings (based on the abundance of reads). BS, black spruce; BSE, black spruce plus ericaceous shrub; TA, trembling aspen.</p>\n",
       "</div>\n",
       "<div class=\"clear\"></div>\n",
       "<div class=\"DottedLine mb15\"></div>\n",
       "<div class=\"Imageheaders\">FIGURE 5</div>\n",
       "<div class=\"FigureDesc\">\n",
       "<a href=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-g005.jpg\" name=\"figure5\" target=\"_blank\">\n",
       "<img alt=\"www.frontiersin.org\" class=\"lazy\" data-src=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-g005.jpg\" id=\"F5\"/></a>\n",
       "<p><strong>Figure 5.</strong> Non-metric multidimensional scaling (NMDS) plots representing similarity between EMF communities of BF root tips in black spruce (BS, red), black spruce + ericaceous shrubs (BSE, green) and trembling aspen (TA, blue) stands. Arrow represents the correlation between N concentration in fir needle and NMDS space.</p>\n",
       "</div>\n",
       "<div class=\"clear\"></div>\n",
       "<div class=\"DottedLine\"></div>\n",
       "<h3>Path Analysis</h3>\n",
       "<p class=\"mb0\">The <i>p</i>-Values of the Fisher’s <i>C</i>-tests were greater than 0.05 for the three models, indicating that all models could be accepted. The <i>p</i>-Values of the associated Chi-Square tests ranged between 0.019 and 0.401. CFI ranged between 0.930 and 0.997, while those for TLI ranged between 0.901 and 0.991 (<a href=\"#T3\">Table 3</a>). Given these results, the interactions model Mod1 was the only one passing the primary filter. Here, we present direct and indirect links for which <i>p</i>-Values were significant at <i>p</i> &lt; 0.05 (for a summary of the parameter estimates, see <a href=\"#S9\">Supplementary Table 1</a>). Apical and lateral growth was directly correlated with foliar N concentrations (path coefficients were 0.547 and 0.609, respectively) and the ramification index (0.253 and 0.254). Of interest, foliar N concentrations were negatively correlated with NMDS second axis (−0.226) and the percentage of BS (−0.469). The ramification index was negatively correlated with BS stand (−1.573). NMDS first axis was positively correlated with BS stands (1.499), while the second axis was positively correlated with the presence of ericaceous shrubs (0.773). The presence of ericaceous shrubs, in turn, was positively correlated with BS stands (0.500). Percentage of BS was positively correlated with BS stands (1.708), while the percentage of BF was negatively correlated with BS stands (−1.306). Significant direct and indirect links between variables are presented in <a href=\"#F6\">Figure 6</a>. All significant and non-significant parameter estimates are presented in <a href=\"#S9\">Supplementary Table 1</a>.</p>\n",
       "<div class=\"DottedLine\"></div>\n",
       "<div class=\"Imageheaders\">TABLE 3</div>\n",
       "<div class=\"FigureDesc\">\n",
       "<a href=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-t003.jpg\" name=\"table3\" target=\"_blank\">\n",
       "<img alt=\"www.frontiersin.org\" class=\"lazy\" data-src=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-t003.jpg\" id=\"T3\"/></a>\n",
       "<p><strong>Table 3.</strong> Statistics of each structural equation models.</p>\n",
       "</div>\n",
       "<div class=\"clear\"></div>\n",
       "<div class=\"DottedLine mb15\"></div>\n",
       "<div class=\"Imageheaders\">FIGURE 6</div>\n",
       "<div class=\"FigureDesc\">\n",
       "<a href=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-g006.jpg\" name=\"figure6\" target=\"_blank\">\n",
       "<img alt=\"www.frontiersin.org\" class=\"lazy\" data-src=\"https://www.frontiersin.org/files/Articles/464100/fpls-10-00932-HTML/image_m/fpls-10-00932-g006.jpg\" id=\"F6\"/></a>\n",
       "<p><strong>Figure 6.</strong> Direct acyclic graphs corresponding to Mod1, only significant links between variables are shown, path coefficients are indicated above each arrow.</p>\n",
       "</div>\n",
       "<div class=\"clear\"></div>\n",
       "<div class=\"DottedLine\"></div>\n",
       "<a class=\"reset-hash-position\" id=\"h5\" name=\"h5\"></a><h2>Discussion</h2>\n",
       "<p class=\"mb15\">The role of biotic interactions was investigated here to explain observed growth differences in balsam fir saplings growing at a northern ecotone of the boreal forest. We hypothesized that biotic interactions could explain the greater nutrition under aspen than under spruce. Our results suggest that facilitation through ectomycorrhizal fungi, together with competition with ericaceous shrubs under BS, are significant drivers of BF growth and nutrition.</p>\n",
       "<p class=\"mb15\">As soil EMF communities were strongly divergent between TA and BS-dominated stands (<a href=\"#B38\">Nagati et al., 2018</a>), we hypothesized that young BF would associate with distinct EMF in the two stands, and that BF would have greater mycorrhization rates under TA stands. Contrary to our hypothesis, observation of BF root tips revealed EM root tips in all conditions, with similar mycorrhization rate. Sequencing of EM root tips confirmed our observation as species richness was not different between stands. We rather detected differences of root architecture and community composition. Indeed, we observed a higher ramification index under TA than under BS stands, probably leading to differences in abilities to explore soil for resources. Our path analyses suggested that soil conditions in TA stands led to a higher ramification index, which in turn improved BF early growth, likely by enhancing foliar nutrient concentrations. Moreover we detected that percent BS around saplings was negatively correlated to sapling N content, this is probably linked to the greater N availability in TA than in BS stands (<a href=\"#B38\">Nagati et al., 2018</a>). As demonstrated by path analysis, N availability (reflected by the stand and so on, by the percent of BS) was not the only factor affecting sapling needle N content and EMF community structure was also correlated to N nutrition. There are therefore complex interactions leading to a better nutrition of young BF, and path analyses highlighted the importance of both stand dominance and EMF community.</p>\n",
       "<p class=\"mb15\">Between stands, communities of EMF on BF roots were not distinct in species richness but rather in composition, and for example Clavulinaceae were only detected under TA stands. Based on soil sequencing, our previous study also revealed a difference of composition but not species richness between stands (<a href=\"#B38\">Nagati et al., 2018</a>). More generally, changes in plant dominance rather shape EMF community structure than the species richness (<a href=\"#B61\">Tedersoo et al., 2012</a>). Such a change in EMF community can have strong functional consequences. Based on our results, changes in community composition were correlated with changes in N concentration. We detected a higher abundance of Inocybaceae and Thelephoraceae in BF roots tips in TA than in BSE and BS stands, and Helotiaceae were more abundant in BSE than TA stand. These differences of EMF abundance between stands could partly explain differences in N uptake by BF saplings. Depending on site conditions, different EMF species or assemblage could have different abilities to uptake nutrients in soil and to transfer these nutrients to plants (<a href=\"#B26\">Jonsson et al., 2001</a>; <a href=\"#B9\">Buée et al., 2007</a>). This result could suggest that some fungi are more efficient provider of N to young BF. <a href=\"#B39\">Nara (2006)</a> experiments illustrated how variable can be the result of EM interactions with distinct EMF fungi. By artificially connecting two different tree species through a common mycorrhizal network from different EMF species, <a href=\"#B39\">Nara (2006)</a> showed that N nutrition could be 6 times higher when trees were connected by <i>Hebeloma mesophaeum</i> as compared to trees connected by <i>Laccaria amethystina</i>. As BF needles were generally richer in N under TA, the question remain if this benefit under TA could be due to N transfer through a common mycorrhizal network (<a href=\"#B57\">Simard et al., 2012</a>) involving BF and TA, or only to better soil conditions under TA. We did not sequence TA roots, to detect possible shared fungi and experimental manipulations may be necessary to test whether TA could bring a direct benefit to young BF saplings. As suggested by the models of <a href=\"#B5\">Bever et al. (2010)</a>, the facilitation of BF by TA may not only be explained by shared fungi, but simply by changes in EMF abundance, which we detected between the two stands. The higher abundance of Helotiaceae associated with BF roots near ericaceous shrubs is also interesting. Several studies have demonstrated that some fungi in this family could have a dual-mode of root colonization and forms ectomycorrhiza as well as ericoid mycorrhiza (<a href=\"#B67\">Vrålstad et al., 2000</a>; <a href=\"#B66\">Villarreal-Ruiz et al., 2004</a>; <a href=\"#B18\">Grelet et al., 2010</a>). These species could be more efficient to provide nutrients to ericaceous shrubs than to BF saplings and could explain the lower N nutrition of BF near ericaceous shrubs.</p>\n",
       "<p class=\"mb15\">The lower nutrient concentrations found in fir needles (N and K) under BSE stands compared to TA stands (<a href=\"#F2\">Figure 2</a>), suggest a negative effect of black spruce and ericaceous shrubs association on BF seedlings. These results are consistent with previous studies where ericaceous shrubs that are associated with non-EMF fungi may compete with EMF-associated trees (<a href=\"#B71\">Yamasaki et al., 1998</a>; <a href=\"#B68\">Walker et al., 1999</a>; <a href=\"#B35\">Mallik, 2003</a>). Ericoid mycorrhizal fungi are particularly competitive for recalcitrant organic matter (ROM) decomposition and uptake of N and P (<a href=\"#B50\">Read, 1996</a>) and probably have the ability to take up nutrient from ROM found around ericaceous shrubs (<a href=\"#B25\">Joanisse et al., 2018</a>) contrarily to BF-associated EMF communities. In hardwood forests of the southern Appalachians, the presence of <i>Rhododendron maximum</i> L. reduced the ramification index of eastern hemlock (<i>Tsuga canadensis</i> [L.] Carrière) saplings four-fold (<a href=\"#B68\">Walker et al., 1999</a>). In our study, the presence of ericaceous shrubs (associated with ericoid fungi) close to fir saplings rather affected their root EMF communities than reduced the ramification index. Numerous studies have already shown how ericaceous shrubs affect EMF communities associated with trees, such as red oak (<i>Quercus rubra</i> L.), hemlock (<a href=\"#B68\">Walker et al., 1999</a>), pine (<i>Pinus strobus</i> L., <i>P. sylvestris</i> L.) (<a href=\"#B29\">Kohout et al., 2011</a>), and black spruce (<a href=\"#B71\">Yamasaki et al., 1998</a>; <a href=\"#B28\">Kennedy et al., 2018</a>). In the boreal forest, ericaceous shrubs not only compete with fir growth and nutrition, but also modify forest dynamics and lead to thick accumulation of soil organic matter and soil acidification. This phenomenon, is a recurrent problem within boreal forests that are situated in the Ontario-Quebec Clay Belt and is often associated with the presence of ericaceous shrubs and Sphagnum mosses (<a href=\"#B15\">Fenton et al., 2005</a>) and a loss of forest productivity (<a href=\"#B55\">Simard et al., 2007</a>). The thickness of the soil organic layer is correlated with a decrease in TA establishment and growth within the black spruce-feather moss domain (<a href=\"#B30\">Lafleur et al., 2015</a>), together with a decrease in black spruce establishment in sites that are dominated by <i>K. angustifolia</i> L. (<a href=\"#B34\">Mallik and Kravchenko, 2018</a>). Our results suggest that forest regeneration failure in areas with high ericaceous shrub abundance could be explained by their effect on EMF communities, and reciprocally, invite to consider below-ground interactions to avoid or limit regeneration failure.</p>\n",
       "<p class=\"mb15\">Whilst our study focused on fir growth, we revealed stronger differences in foliar N concentrations and ramification indices between stands. Indeed difference in annual growth between stands was detected only in 2015. Annual growth of trees is relatively slow in the boreal forest and is correlated with the length of the growing season, which can differ from 1 year to the next (<a href=\"#B24\">Jarvis and Linder, 2000</a>). Differences in early growth could thus be difficult to detect over a short period of time and would be probably more pronounced when studying several years of growth (see <a href=\"#B1\">Arbour and Bergeron, 2011</a>). Measuring fitness is always difficult for young trees over a short period of study and measures of foliar nutrient concentrations were more useful to detect differences between stands, and reflected the benefits of EMF symbiosis. Foliar nutrient concentrations were generally greater under TA compared to BS and correlated with growth, which confirmed our hypothesis. This result could be linked to the greater availability of nutrients in TA stands than in BS stands (<a href=\"#B11\">Cavard et al., 2011</a>; <a href=\"#B38\">Nagati et al., 2018</a>).</p>\n",
       "<p class=\"mb0\">A major goal of forest ecology today is to determine ecosystem trajectories in a context of climate change. In the case of the balsam fir-black spruce forest ecotone, it appears that trembling aspen stands would provide a favorable niche for fir establishment and growth. As demonstrated by <a href=\"#B1\">Arbour and Bergeron (2011)</a>, the higher abundance of balsam fir in TA than BS stands is more pronounced for saplings than seedlings. This result translates a lower mortality and better growth of balsam fir in TA stands which leads to a greater abundance of mature and reproductive trees in these stands. This in turn may result in an increase of mixed forests and deep changes in ecosystem functioning. The distributional ranges of numerous tree species are likely to change within the context of climate change (<a href=\"#B3\">Bellard et al., 2012</a>; <a href=\"#B23\">Iverson and McKenzie, 2013</a>; <a href=\"#B22\">IPCC Climate Change, 2014</a>). Migration has already begun for many tree species in North America (<a href=\"#B7\">Brandt, 2009</a>; <a href=\"#B70\">Woodall et al., 2009</a>). Their geographic ranges have been extending northward rapidly (up to 100 km/century; <a href=\"#B70\">Woodall et al., 2009</a>). Our results suggest that the climatic niche could not alone explain species abilities to establish and that the mutualistic niche (sensus <a href=\"#B45\">Peay, 2016</a>) have to be explored to ensure a better comprehension of tree migration processes.</p>\n",
       "<a class=\"reset-hash-position\" id=\"h6\" name=\"h6\"></a><h2>Data Availability</h2>\n",
       "<p class=\"mb0\">All datasets for this study are included in the manuscript and/or the <a href=\"#S9\">Supplementary Files</a>.</p>\n",
       "<a class=\"reset-hash-position\" id=\"h7\" name=\"h7\"></a><h2>Author Contributions</h2>\n",
       "<p class=\"mb0\">MG, YB, MN, MR, and AD designed the study and interpreted the results. MN and MR carried out the field and laboratory work. MN and SM carried out the molecular biology and bioinformatics. MN carried out the statistical analyses. MN, MG, and MR wrote the manuscript. All authors edited and approved the final version of the manuscript.</p>\n",
       "<a id=\"fun1\" name=\"fun1\"></a><h2>Funding</h2>\n",
       "<p class=\"mb0\">This work was supported by a MITACS Fellowship to MN in collaboration with the Norbord Inc. and the Ouranos Consortium on Climate Change (IT066831). Travel funds were provided by the Paul Sabatier University and sequencing costs were supported by the “Laboratoires d’Excellence (LABEX)” TULIP (ANR 10-LABX-0041) and CEBA (ANR 10-LABX-0025).</p>\n",
       "<a id=\"conf1\" name=\"conf1\"></a><h2>Conflict of Interest Statement</h2>\n",
       "<p class=\"mb0\">The authors declare that the research was conducted in the absence of any commercial or financial relationships that could be construed as a potential conflict of interest.</p>\n",
       "<a id=\"ack1\" name=\"ack1\"></a><h2>Acknowledgments</h2>\n",
       "<p class=\"mb0\">We thank Danielle Charron, Raynald Julien, and Elias Ganivet for their assistance in fieldwork, Francine Tremblay for laboratory access, Philippe Marchand and Benjamin Andrieux for statistical advice, and Genopole Toulouse for the computing and storage resources. We are grateful to W. F. J. Parsons for revising the manuscript.</p>\n",
       "<a id=\"S9\" name=\"S9\"></a><h2>Supplementary Material</h2>\n",
       "<p class=\"mb0\">The Supplementary Material for this article can be found online at: <a href=\"https://www.frontiersin.org/articles/10.3389/fpls.2019.00932/full#supplementary-material\">https://www.frontiersin.org/articles/10.3389/fpls.2019.00932/full#supplementary-material</a></p>\n",
       "<a id=\"FNT1\" name=\"FNT1\"></a><h2>Footnotes</h2>\n",
       "<ol>\n",
       "<li id=\"footnote1\"><strong><a class=\"footnotetextanchor\" href=\"#footnote1a\" style=\"color:gray;\" title=\"\">^</a></strong> <a href=\"ftp://ftp.ncbi.nih.gov/genbank/\" style=\"color:gray;\">ftp://ftp.ncbi.nih.gov/genbank/</a></li>\n",
       "<li id=\"footnote2\"><strong><a class=\"footnotetextanchor\" href=\"#footnote2a\" style=\"color:gray;\" title=\"\">^</a></strong> <a href=\"https://unite.ut.ee/\" style=\"color:gray;\">https://unite.ut.ee/</a></li>\n",
       "</ol>\n",
       "<a id=\"refer1\" name=\"refer1\"></a><h2>References</h2>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B1\" name=\"B1\"></a>Arbour, M. -L., and Bergeron, Y. (2011). Effect of increased <i>Populus</i> cover on <i>Abies</i> regeneration in the <i>Picea</i>-feathermoss boreal forest. <i>J. Veg. Sci.</i> 22, 1132–1142. doi: 10.1111/j.1654-1103.2011.01314.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1111/j.1654-1103.2011.01314.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Effect+of+increased+Populus+cover+on+Abies+regeneration+in+the+Picea-feathermoss+boreal+forest%2E&amp;journal=J%2E+Veg%2E+Sci%2E&amp;author=Arbour+M.+-L.&amp;author=Bergeron+Y.&amp;publication_year=2011&amp;volume=22&amp;pages=1132–1142\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B2\" name=\"B2\"></a>Barbeta, A., and Peñuelas, J. (2017). Increasing carbon discrimination rates and depth of water uptake favor the growth of mediterranean evergreen trees in the ecotone with temperate deciduous forests. <i>Glob. Change Biol.</i> 23, 5054–5068. doi: 10.1111/gcb.13770</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=28544424\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/gcb.13770\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Increasing+carbon+discrimination+rates+and+depth+of+water+uptake+favor+the+growth+of+mediterranean+evergreen+trees+in+the+ecotone+with+temperate+deciduous+forests%2E&amp;journal=Glob%2E+Change+Biol%2E&amp;author=Barbeta+A.&amp;author=Peñuelas+J.&amp;publication_year=2017&amp;volume=23&amp;pages=5054–5068\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B3\" name=\"B3\"></a>Bellard, C., Bertelsmeier, C., Leadley, P., Thuiller, W., and Courchamp, F. (2012). Impacts of climate change on the future of biodiversity. <i>Ecol. Lett.</i> 15, 365–377. doi: 10.1111/j.1461-0248.2011.01736.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=22257223\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/j.1461-0248.2011.01736.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Impacts+of+climate+change+on+the+future+of+biodiversity%2E&amp;journal=Ecol%2E+Lett%2E&amp;author=Bellard+C.&amp;author=Bertelsmeier+C.&amp;author=Leadley+P.&amp;author=Thuiller+W.&amp;author=Courchamp+F.&amp;publication_year=2012&amp;volume=15&amp;pages=365–377\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B4\" name=\"B4\"></a>Belleau, A., Leduc, A., Lecomte, N., and Bergeron, Y. (2011). Forest succession rate and pathways on different surface deposit types in the boreal forest of northwestern Quebec.<i>Écoscience</i> 18, 329–340. doi: 10.2980/18-4-3393</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.2980/18-4-3393\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Forest+succession+rate+and+pathways+on+different+surface+deposit+types+in+the+boreal+forest+of+northwestern+Quebec%2E&amp;journal=Écoscience&amp;author=Belleau+A.&amp;author=Leduc+A.&amp;author=Lecomte+N.&amp;author=Bergeron+Y.&amp;publication_year=2011&amp;volume=18&amp;pages=329–340\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B5\" name=\"B5\"></a>Bever, J. D., Dickie, I. A., Facelli, E., Facelli, J. M., Klironomos, J., Moora, M., et al. (2010). Rooting theories of plant community ecology in microbial interactions. <i>Trends Ecol. Evol.</i> 25, 468–478. doi: 10.1016/j.tree.2010.05.004</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=20557974\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1016/j.tree.2010.05.004\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Rooting+theories+of+plant+community+ecology+in+microbial+interactions%2E&amp;journal=Trends+Ecol%2E+Evol%2E&amp;author=Bever+J.+D.&amp;author=Dickie+I.+A.&amp;author=Facelli+E.&amp;author=Facelli+J.+M.&amp;author=Klironomos+J.&amp;author=Moora+M.&amp;publication_year=2010&amp;volume=25&amp;pages=468–478\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B6\" name=\"B6\"></a>Boyer, F., Mercier, C., Bonin, A., Le Bras, Y., Taberlet, P., and Coissac, E. (2016). obitools: a unix -inspired software package for DNA metabarcoding. <i>Mol. Ecol. Res.</i> 16, 176–182. doi: 10.1111/1755-0998.12428</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=25959493\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/1755-0998.12428\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=obitools%3A+a+unix+-inspired+software+package+for+DNA+metabarcoding%2E&amp;journal=Mol%2E+Ecol%2E+Res%2E&amp;author=Boyer+F.&amp;author=Mercier+C.&amp;author=Bonin+A.&amp;author=Le+Bras+Y.&amp;author=Taberlet+P.&amp;author=Coissac+E.&amp;publication_year=2016&amp;volume=16&amp;pages=176–182\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B7\" name=\"B7\"></a>Brandt, J. P. (2009). The extent of the north american boreal zone. <i>Environ. Rev.</i> 17, 101–161. doi: 10.1139/A09-004</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=23939809\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1139/A09-004\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=The+extent+of+the+north+american+boreal+zone%2E&amp;journal=Environ%2E+Rev%2E&amp;author=Brandt+J.+P.&amp;publication_year=2009&amp;volume=17&amp;pages=101–161\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B8\" name=\"B8\"></a>Brooker, R. W., Maestre, F. T., Callaway, R. M., Lortie, C. L., Cavieres, L. A., Kunstler, G., et al. (2008). Facilitation in plant communities: the past, the present, and the future. <i>J. Ecol.</i> 96, 18–34. doi: 10.1111/j.1365-2745.2007.01295.x.</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1111/j.1365-2745.2007.01295.x.\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Facilitation+in+plant+communities%3A+the+past%2C+the+present%2C+and+the+future%2E&amp;journal=J%2E+Ecol%2E&amp;author=Brooker+R.+W.&amp;author=Maestre+F.+T.&amp;author=Callaway+R.+M.&amp;author=Lortie+C.+L.&amp;author=Cavieres+L.+A.&amp;author=Kunstler+G.&amp;publication_year=2008&amp;volume=96&amp;pages=18–34\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B9\" name=\"B9\"></a>Buée, M., Courty, P. E., Mignot, D., and Garbaye, J. (2007). Soil niche effect on species diversity and catabolic activities in an ectomycorrhizal fungal community. <i>Soil Biol. Biochem.</i> 39, 1947–1955. doi: 10.1016/j.soilbio.2007.02.016</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1016/j.soilbio.2007.02.016\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Soil+niche+effect+on+species+diversity+and+catabolic+activities+in+an+ectomycorrhizal+fungal+community%2E&amp;journal=Soil+Biol%2E+Biochem%2E&amp;author=Buée+M.&amp;author=Courty+P.+E.&amp;author=Mignot+D.&amp;author=Garbaye+J.&amp;publication_year=2007&amp;volume=39&amp;pages=1947–1955\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B10\" name=\"B10\"></a>Callaway, R. M., and Walker, L. R. (1997). Competition and facilitation: a synthetic approach to interactions in plant communities. <i>Ecology</i> 78, 1958–1965. doi: 10.1890/0012-9658(1997)078%5B1958:cafasa%5D2.0.co;2</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1890/0012-9658(1997)078%5B1958:cafasa%5D2.0.co;2\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Competition+and+facilitation%3A+a+synthetic+approach+to+interactions+in+plant+communities%2E&amp;journal=Ecology&amp;author=Callaway+R.+M.&amp;author=Walker+L.+R.&amp;publication_year=1997&amp;volume=78&amp;pages=1958–1965\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B11\" name=\"B11\"></a>Cavard, X., Bergeron, Y., Chen, H. Y. H., and Paré, D. (2011). Effect of forest canopy composition on soil nutrients and dynamics of the understorey: mixed canopies serve neither vascular nor bryophyte strata. <i>J. Veg. Sci.</i> 22, 1105–1119. doi: 10.1111/j.1654-1103.2011.01311.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1111/j.1654-1103.2011.01311.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Effect+of+forest+canopy+composition+on+soil+nutrients+and+dynamics+of+the+understorey%3A+mixed+canopies+serve+neither+vascular+nor+bryophyte+strata%2E&amp;journal=J%2E+Veg%2E+Sci%2E&amp;author=Cavard+X.&amp;author=Bergeron+Y.&amp;author=Chen+H.+Y.+H.&amp;author=Paré+D.&amp;publication_year=2011&amp;volume=22&amp;pages=1105–1119\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B12\" name=\"B12\"></a>Collier, F. A., and Bidartondo, M. I. (2009). Waiting for fungi: the ectomycorrhizal invasion of lowland heathlands. <i>J. Ecol.</i> 97, 950–963. doi: 10.1111/j.1365-2745.2009.01544.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1111/j.1365-2745.2009.01544.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Waiting+for+fungi%3A+the+ectomycorrhizal+invasion+of+lowland+heathlands%2E&amp;journal=J%2E+Ecol%2E&amp;author=Collier+F.+A.&amp;author=Bidartondo+M.+I.&amp;publication_year=2009&amp;volume=97&amp;pages=950–963\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B13\" name=\"B13\"></a>Dickie, I. A., Oleksyn, J., Reich, P. B., Karolewski, P., Zytkowiak, R., Jagodzinski, A. M., et al. (2006). Soil modification by different tree species influences the extent of seedling ectomycorrhizal infection. <i>Mycorrhiza</i> 16, 73–79. doi: 10.1007/s00572-005-0013-x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=16322987\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1007/s00572-005-0013-x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Soil+modification+by+different+tree+species+influences+the+extent+of+seedling+ectomycorrhizal+infection%2E&amp;journal=Mycorrhiza&amp;author=Dickie+I.+A.&amp;author=Oleksyn+J.&amp;author=Reich+P.+B.&amp;author=Karolewski+P.&amp;author=Zytkowiak+R.&amp;author=Jagodzinski+A.+M.&amp;publication_year=2006&amp;volume=16&amp;pages=73–79\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B14\" name=\"B14\"></a>Epp, L. S., Boessenkool, S., Bellemain, E. P., Haile, J., Esposito, A., Riaz, T., et al. (2012). New environmental metabarcodes for analysing soil DNA: potential for studying past and present ecosystems. <i>Mol. Ecol.</i> 21, 1821–1833. doi: 10.1111/j.1365-294X.2012.05537.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=22486821\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/j.1365-294X.2012.05537.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=New+environmental+metabarcodes+for+analysing+soil+DNA%3A+potential+for+studying+past+and+present+ecosystems%2E&amp;journal=Mol%2E+Ecol%2E&amp;author=Epp+L.+S.&amp;author=Boessenkool+S.&amp;author=Bellemain+E.+P.&amp;author=Haile+J.&amp;author=Esposito+A.&amp;author=Riaz+T.&amp;publication_year=2012&amp;volume=21&amp;pages=1821–1833\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B15\" name=\"B15\"></a>Fenton, N., Lecomte, N., Légaré, S., and Bergeron, Y. (2005). Paludification in black spruce (<i>Picea mariana</i>) forests of eastern Canada: potential factors and management implications. <i>For. Ecol. Manag.</i> 213, 151–159. doi: 10.1016/j.foreco.2005.03.017</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1016/j.foreco.2005.03.017\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Paludification+in+black+spruce+%28Picea+mariana%29+forests+of+eastern+Canada%3A+potential+factors+and+management+implications%2E&amp;journal=For%2E+Ecol%2E+Manag%2E&amp;author=Fenton+N.&amp;author=Lecomte+N.&amp;author=Légaré+S.&amp;author=Bergeron+Y.&amp;publication_year=2005&amp;volume=213&amp;pages=151–159\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B16\" name=\"B16\"></a>Franklin, O., Näsholm, T., Högberg, P., and Högberg, M. N. (2014). Forests trapped in nitrogen limitation - an ecological market perspective on ectomycorrhizal symbiosis. <i>New Phytol.</i> 203, 657–666. doi: 10.1111/nph.12840</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=24824576\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/nph.12840\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Forests+trapped+in+nitrogen+limitation+-+an+ecological+market+perspective+on+ectomycorrhizal+symbiosis%2E&amp;journal=New+Phytol%2E&amp;author=Franklin+O.&amp;author=Näsholm+T.&amp;author=Högberg+P.&amp;author=Högberg+M.+N.&amp;publication_year=2014&amp;volume=203&amp;pages=657–666\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B17\" name=\"B17\"></a>Gallet, C. (1994). Allelopathic potential in bilberry-spruce forests: influence of phenolic compounds on spruce seedlings. <i>J. Chem. Ecol.</i> 20, 1009–1024. doi: 10.1007/BF02059738</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=24242299\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1007/BF02059738\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Allelopathic+potential+in+bilberry-spruce+forests%3A+influence+of+phenolic+compounds+on+spruce+seedlings%2E&amp;journal=J%2E+Chem%2E+Ecol%2E&amp;author=Gallet+C.&amp;publication_year=1994&amp;volume=20&amp;pages=1009–1024\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B18\" name=\"B18\"></a>Grelet, G. -A., Johnson, D., Vrålstad, T., Alexander, I. J., and Anderson, I. C. (2010). New insights into the mycorrhizal rhizoscyphus ericae aggregate: spatial structure and co-colonization of ectomycorrhizal and ericoid roots. <i>New Phytol.</i> 188, 210–222. doi: 10.1111/j.1469-8137.2010.03353.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=20618909\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/j.1469-8137.2010.03353.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=New+insights+into+the+mycorrhizal+rhizoscyphus+ericae+aggregate%3A+spatial+structure+and+co-colonization+of+ectomycorrhizal+and+ericoid+roots%2E&amp;journal=New+Phytol%2E&amp;author=Grelet+G.+-A.&amp;author=Johnson+D.&amp;author=Vrålstad+T.&amp;author=Alexander+I.+J.&amp;author=Anderson+I.+C.&amp;publication_year=2010&amp;volume=188&amp;pages=210–222\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B19\" name=\"B19\"></a>Harsch, M. A., Hulme, P. E., McGlone, M. S., and Duncan, R. P. (2009). Are treelines advancing? a global meta-analysis of treeline response to climate warming. <i>Ecol. Lett.</i> 12, 1040–1049. doi: 10.1111/j.1461-0248.2009.01355.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=19682007\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/j.1461-0248.2009.01355.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Are+treelines+advancing%B4+a+global+meta-analysis+of+treeline+response+to+climate+warming%2E&amp;journal=Ecol%2E+Lett%2E&amp;author=Harsch+M.+A.&amp;author=Hulme+P.+E.&amp;author=McGlone+M.+S.&amp;author=Duncan+R.+P.&amp;publication_year=2009&amp;volume=12&amp;pages=1040–1049\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B20\" name=\"B20\"></a>Hu, L., and Bentler, P. M. (1999). Cutoff criteria for fit indexes in covariance structure analysis: conventional criteria versus new alternatives. <i>Struct. Equ. Modeling</i> 6, 1–55. doi: 10.1080/10705519909540118.</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1080/10705519909540118.\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Cutoff+criteria+for+fit+indexes+in+covariance+structure+analysis%3A+conventional+criteria+versus+new+alternatives%2E&amp;journal=Struct%2E+Equ%2E+Modeling&amp;author=Hu+L.&amp;author=Bentler+P.+M.&amp;publication_year=1999&amp;volume=6&amp;pages=1–55\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B21\" name=\"B21\"></a>Inselsbacher, E., and Näsholm, T. (2012). The below-ground perspective of forest plants: soil provides mainly organic nitrogen for plants and mycorrhizal fungi. <i>New Phytol.</i> 195, 329–334. doi: 10.1111/j.1469-8137.2012.04169.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=22564239\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/j.1469-8137.2012.04169.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=The+below-ground+perspective+of+forest+plants%3A+soil+provides+mainly+organic+nitrogen+for+plants+and+mycorrhizal+fungi%2E&amp;journal=New+Phytol%2E&amp;author=Inselsbacher+E.&amp;author=Näsholm+T.&amp;publication_year=2012&amp;volume=195&amp;pages=329–334\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B22\" name=\"B22\"></a>IPCC Climate Change (2014). <i>Impacts, Adaptation, &amp; Vulnerability - Summary for Policymakers GlobalChange.gov.</i> Available at: <a href=\"http://www.globalchange.gov/browse/reports/ipcc-climate-change-2014-impacts-adaptation-vulnerability-summary-policymakers\">http://www.globalchange.gov/browse/reports/ipcc-climate-change-2014-impacts-adaptation-vulnerability-summary-policymakers</a> (accessed February 24, 2016).</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://scholar.google.com/scholar_lookup?&amp;journal=Impacts%2C+Adaptation%2C+&amp;+Vulnerability+-+Summary+for+Policymakers+GlobalChange%2Egov%2E&amp;publication_year=2014\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B23\" name=\"B23\"></a>Iverson, L. R., and McKenzie, D. (2013). Tree-species range shifts in a changing climate: detecting, modeling, assisting. <i>Lands. Ecol.</i> 28, 879–889. doi: 10.1007/s10980-013-9885-x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1007/s10980-013-9885-x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Tree-species+range+shifts+in+a+changing+climate%3A+detecting%2C+modeling%2C+assisting%2E&amp;journal=Lands%2E+Ecol%2E&amp;author=Iverson+L.+R.&amp;author=McKenzie+D.&amp;publication_year=2013&amp;volume=28&amp;pages=879–889\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B24\" name=\"B24\"></a>Jarvis, P., and Linder, S. (2000). Botany: constraints to growth of boreal forests. <i>Nature</i> 405, 904–905. doi: 10.1038/35016154.</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=10879523\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1038/35016154.\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Botany%3A+constraints+to+growth+of+boreal+forests%2E&amp;journal=Nature&amp;author=Jarvis+P.&amp;author=Linder+S.&amp;publication_year=2000&amp;volume=405&amp;pages=904–905\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B25\" name=\"B25\"></a>Joanisse, G. D., Bradley, R. L., and Preston, C. M. (2018). The spread of <i>almia angustifolia</i> on black spruce forest cutovers contributes to the spatial heterogeneity of soil resources. <i>PLoS One</i> 13:e0198860. doi: 10.1371/journal.pone.0198860</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=29927964\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1371/journal.pone.0198860\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=The+spread+of+almia+angustifolia+on+black+spruce+forest+cutovers+contributes+to+the+spatial+heterogeneity+of+soil+resources%2E&amp;journal=PLoS+One&amp;author=Joanisse+G.+D.&amp;author=Bradley+R.+L.&amp;author=Preston+C.+M.&amp;publication_year=2018&amp;volume=13&amp;issue=e0198860\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B26\" name=\"B26\"></a>Jonsson, L. M., Nilsson, M. -C., Wardle, D. A., and Zackrisson, O. (2001). Context dependent effects of ectomycorrhizal species richness on tree seedling productivity. <i>Oikos</i> 93, 353–364. doi: 10.1034/j.1600-0706.2001.930301.x.</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1034/j.1600-0706.2001.930301.x.\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Context+dependent+effects+of+ectomycorrhizal+species+richness+on+tree+seedling+productivity%2E&amp;journal=Oikos&amp;author=Jonsson+L.+M.&amp;author=Nilsson+M.+-C.&amp;author=Wardle+D.+A.&amp;author=Zackrisson+O.&amp;publication_year=2001&amp;volume=93&amp;pages=353–364\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B27\" name=\"B27\"></a>Kalra, Y. P. (1998). “Soil and plant analysis council,”in <i>Handbook of Reference Methods for Plant Analysis</i>, ed. Y. Kalra, (Boca Raton: CRC Press).</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Soil+and+plant+analysis+council&amp;journal=Handbook+of+Reference+Methods+for+Plant+Analysis&amp;author=Kalra+Y.+P.+(1998).+“Soil+and+plant+analysis+council”in+Handbook+of+Reference+Methods+for+Plant+Analysis+ed.+Kalra+Y.&amp;publication_year=1998\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B28\" name=\"B28\"></a>Kennedy, P. G., Mielke, L. A., and Nguyen, N. H. (2018). Ecological responses to forest age, habitat, and host vary by mycorrhizal type in boreal peatlands. <i>Mycorrhiza</i> 28, 315–328. doi: 10.1007/s00572-018-0821-4</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=29504037\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1007/s00572-018-0821-4\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Ecological+responses+to+forest+age%2C+habitat%2C+and+host+vary+by+mycorrhizal+type+in+boreal+peatlands%2E&amp;journal=Mycorrhiza&amp;author=Kennedy+P.+G.&amp;author=Mielke+L.+A.&amp;author=Nguyen+N.+H.&amp;publication_year=2018&amp;volume=28&amp;pages=315–328\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B29\" name=\"B29\"></a>Kohout, P., Sıkorová, Z., Bahram, M., Hadincová, V., Albrechtová, J., Tedersoo, L., et al. (2011). Ericaceous dwarf shrubs affect ectomycorrhizal fungal community of the invasive <i>Pinus strobus</i> and native <i>Pinus sylvestris</i> in a pot experiment. <i>Mycorrhiza</i> 21, 403–412. doi: 10.1007/s00572-010-0350-2</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=21161550\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1007/s00572-010-0350-2\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Ericaceous+dwarf+shrubs+affect+ectomycorrhizal+fungal+community+of+the+invasive+Pinus+strobus+and+native+Pinus+sylvestris+in+a+pot+experiment%2E&amp;journal=Mycorrhiza&amp;author=Kohout+P.&amp;author=Sıkorová+Z.&amp;author=Bahram+M.&amp;author=Hadincová+V.&amp;author=Albrechtová+J.&amp;author=Tedersoo+L.&amp;publication_year=2011&amp;volume=21&amp;pages=403–412\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B30\" name=\"B30\"></a>Lafleur, B., Cazal, A., Leduc, A., and Bergeron, Y. (2015). Soil organic layer thickness influences the establishment and growth of trembling aspen (<i>Populus tremuloides</i>) in boreal forests. <i>For. Ecol. Manag.</i> 347, 209–216. doi: 10.1016/j.foreco.2015.03.031</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1016/j.foreco.2015.03.031\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Soil+organic+layer+thickness+influences+the+establishment+and+growth+of+trembling+aspen+%28Populus+tremuloides%29+in+boreal+forests%2E&amp;journal=For%2E+Ecol%2E+Manag%2E&amp;author=Lafleur+B.&amp;author=Cazal+A.&amp;author=Leduc+A.&amp;author=Bergeron+Y.&amp;publication_year=2015&amp;volume=347&amp;pages=209–216\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B31\" name=\"B31\"></a>Lecomte, N., and Bergeron, Y. (2005). Successional pathways on different surficial deposits in the coniferous boreal forest of the Quebec Clay Belt. <i>Can. J. For. Res.</i> 35, 1984–1995. doi: 10.1139/x05-114</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1139/x05-114\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Successional+pathways+on+different+surficial+deposits+in+the+coniferous+boreal+forest+of+the+Quebec+Clay+Belt%2E&amp;journal=Can%2E+J%2E+For%2E+Res%2E&amp;author=Lecomte+N.&amp;author=Bergeron+Y.&amp;publication_year=2005&amp;volume=35&amp;pages=1984–1995\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B32\" name=\"B32\"></a>Légaré, S., Paré, D., and Bergeron, Y. (2005). Influence of aspen on forest floor properties in black spruce-dominated stands. <i>Plant Soil</i> 275, 207–220. doi: 10.1007/s11104-005-1482-6</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1007/s11104-005-1482-6\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Influence+of+aspen+on+forest+floor+properties+in+black+spruce-dominated+stands%2E&amp;journal=Plant+Soil&amp;author=Légaré+S.&amp;author=Paré+D.&amp;author=Bergeron+Y.&amp;publication_year=2005&amp;volume=275&amp;pages=207–220\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B33\" name=\"B33\"></a>Lindahl, B. D., Nilsson, R. H., Tedersoo, L., Abarenkov, K., Carlsen, T., Kjøller, R., et al. (2013). Fungal community analysis by high-throughput sequencing of amplified markers – a user’s guide. <i>New Phytol.</i> 199, 288–299. doi: 10.1111/nph.12243</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=23534863\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/nph.12243\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Fungal+community+analysis+by+high-throughput+sequencing+of+amplified+markers+–+a+user’s+guide%2E&amp;journal=New+Phytol%2E&amp;author=Lindahl+B.+D.&amp;author=Nilsson+R.+H.&amp;author=Tedersoo+L.&amp;author=Abarenkov+K.&amp;author=Carlsen+T.&amp;author=Kjøller+R.&amp;publication_year=2013&amp;volume=199&amp;pages=288–299\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B34\" name=\"B34\"></a>Mallik, A., and Kravchenko, D. (2018). Recruitment and ontogenic patterns of stunting and growth release of black spruce (<i>Picea mariana</i>) in post-fire <i>Kalmia</i> heaths. <i>For. Ecol. Manag.</i> 407, 135–144. doi: 10.1016/j.foreco.2017.09.068</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1016/j.foreco.2017.09.068\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Recruitment+and+ontogenic+patterns+of+stunting+and+growth+release+of+black+spruce+%28Picea+mariana%29+in+post-fire+Kalmia+heaths%2E&amp;journal=For%2E+Ecol%2E+Manag%2E&amp;author=Mallik+A.&amp;author=Kravchenko+D.&amp;publication_year=2018&amp;volume=407&amp;pages=135–144\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B35\" name=\"B35\"></a>Mallik, A. U. (2003). Conifer regeneration problems in boreal and temperate forests with ericaceous understory: role of disturbance, seedbed limitation, and keytsone species change. <i>Crit. Rev. Plant Sci.</i> 22, 341–366. doi: 10.1080/713610860</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1080/713610860\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Conifer+regeneration+problems+in+boreal+and+temperate+forests+with+ericaceous+understory%3A+role+of+disturbance%2C+seedbed+limitation%2C+and+keytsone+species+change%2E&amp;journal=Crit%2E+Rev%2E+Plant+Sci%2E&amp;author=Mallik+A.+U.&amp;publication_year=2003&amp;volume=22&amp;pages=341–366\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B36\" name=\"B36\"></a>Marchetti, G. M., Drton, M., and Sadeghi, K. (2015). <i>Ggm: Functions for Graphical Markov Models.</i> Available at: <a href=\"https://cran.r-project.org/web/packages/ggm/index.html\">https://cran.r-project.org/web/packages/ggm/index.html</a> (accessed December 13, 2018).</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://scholar.google.com/scholar_lookup?&amp;journal=Ggm%3A+Functions+for+Graphical+Markov+Models%2E&amp;author=Marchetti+G.+M.&amp;author=Drton+M.&amp;author=Sadeghi+K.&amp;publication_year=2015\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B37\" name=\"B37\"></a>Messaoud, Y., Bergeron, Y., and Leduc, A. (2007). Ecological factors explaining the location of the boundary between the mixedwood and coniferous bioclimatic zones in the boreal biome of eastern North America. <i>Glob. Ecol. Biogeogr.</i> 16, 90–102. doi: 10.1111/j.1466-8238.2006.00277.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1111/j.1466-8238.2006.00277.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Ecological+factors+explaining+the+location+of+the+boundary+between+the+mixedwood+and+coniferous+bioclimatic+zones+in+the+boreal+biome+of+eastern+North+America%2E&amp;journal=Glob%2E+Ecol%2E+Biogeogr%2E&amp;author=Messaoud+Y.&amp;author=Bergeron+Y.&amp;author=Leduc+A.&amp;publication_year=2007&amp;volume=16&amp;pages=90–102\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B38\" name=\"B38\"></a>Nagati, M., Roy, M., Manzi, S., Richard, F., Desrochers, A., Gardes, M., et al. (2018). Impact of local forest composition on soil fungal communities in a mixed boreal forest. <i>Plant Soil</i> 432, 345–357. doi: 10.1007/s11104-018-3806-3</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1007/s11104-018-3806-3\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Impact+of+local+forest+composition+on+soil+fungal+communities+in+a+mixed+boreal+forest%2E&amp;journal=Plant+Soil&amp;author=Nagati+M.&amp;author=Roy+M.&amp;author=Manzi+S.&amp;author=Richard+F.&amp;author=Desrochers+A.&amp;author=Gardes+M.&amp;publication_year=2018&amp;volume=432&amp;pages=345–357\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B39\" name=\"B39\"></a>Nara, K. (2006). Ectomycorrhizal networks and seedling establishment during early primary succession. <i>New Phytol.</i> 169, 169–178. doi: 10.1111/j.1469-8137.2005.01545.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=16390428\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/j.1469-8137.2005.01545.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Ectomycorrhizal+networks+and+seedling+establishment+during+early+primary+succession%2E&amp;journal=New+Phytol%2E&amp;author=Nara+K.&amp;publication_year=2006&amp;volume=169&amp;pages=169–178\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B40\" name=\"B40\"></a>Nara, K., and Hogetsu, T. (2004). Ectomycorrhizal fungi on established shrubs facilitate subsequent seedling establishment of successional plant species. <i>Ecology</i> 85, 1700–1707. doi: 10.1890/03-0373</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1890/03-0373\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Ectomycorrhizal+fungi+on+established+shrubs+facilitate+subsequent+seedling+establishment+of+successional+plant+species%2E&amp;journal=Ecology&amp;author=Nara+K.&amp;author=Hogetsu+T.&amp;publication_year=2004&amp;volume=85&amp;pages=1700–1707\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B41\" name=\"B41\"></a>Nguyen, N. H., Song, Z., Bates, S. T., Branco, S., Tedersoo, L., Menke, J., et al. (2016). FUNGuild: an open annotation tool for parsing fungal community datasets by ecological guild. <i>Fungal Ecol.</i> 20, 241–248. doi: 10.1016/j.funeco.2015.06.006</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1016/j.funeco.2015.06.006\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=FUNGuild%3A+an+open+annotation+tool+for+parsing+fungal+community+datasets+by+ecological+guild%2E&amp;journal=Fungal+Ecol%2E&amp;author=Nguyen+N.+H.&amp;author=Song+Z.&amp;author=Bates+S.+T.&amp;author=Branco+S.&amp;author=Tedersoo+L.&amp;author=Menke+J.&amp;publication_year=2016&amp;volume=20&amp;pages=241–248\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B42\" name=\"B42\"></a>Nilsson, R. H., Kristiansson, E., Ryberg, M., Hallenberg, N., and Larsson, K. -H. (2008). Intraspecific ITS variability in the kingdom fungi as expressed in the international sequence databases and its implications for molecular species identification. <i>Evol. Bioinform. Online</i> 4, 193–201.</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=19204817\" target=\"_blank\">PubMed Abstract</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Intraspecific+ITS+variability+in+the+kingdom+fungi+as+expressed+in+the+international+sequence+databases+and+its+implications+for+molecular+species+identification%2E&amp;journal=Evol%2E+Bioinform%2E+Online&amp;author=Nilsson+R.+H.&amp;author=Kristiansson+E.&amp;author=Ryberg+M.&amp;author=Hallenberg+N.&amp;author=Larsson+K.+-H.&amp;publication_year=2008&amp;volume=4&amp;pages=193–201\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B43\" name=\"B43\"></a>Oksanen, J., Blanchet, F. G., Friendly, M., Kindt, R., Legendre, P., McGlinn, D., et al. (2017). <i>Vegan: Community Ecology Package.</i> Available at: <a href=\"https://cran.r-project.org/web/packages/vegan/index.html\">https://cran.r-project.org/web/packages/vegan/index.html</a> (accessed January 18, 2018).</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://scholar.google.com/scholar_lookup?&amp;journal=Vegan%3A+Community+Ecology+Package%2E&amp;author=Oksanen+J.&amp;author=Blanchet+F.+G.&amp;author=Friendly+M.&amp;author=Kindt+R.&amp;author=Legendre+P.&amp;author=McGlinn+D.&amp;publication_year=2017\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B44\" name=\"B44\"></a>Pallardy, S. G., and Kozlowski, T. T. (2008). <i>Physiology of Woody Plants.</i> 3rd ed. Amsterdam?, NE: Elsevier.</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://scholar.google.com/scholar_lookup?&amp;journal=Physiology+of+Woody+Plants%2E&amp;author=Pallardy+S.+G.&amp;author=Kozlowski+T.+T.&amp;publication_year=2008\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B45\" name=\"B45\"></a>Peay, K. G. (2016). The mutualistic niche: mycorrhizal symbiosis and community dynamics. <i>Annu. Rev. Ecol., Evol. System.</i> 47, 143–164. doi: 10.1146/annurev-ecolsys-121415-032100</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1146/annurev-ecolsys-121415-032100\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=The+mutualistic+niche%3A+mycorrhizal+symbiosis+and+community+dynamics%2E&amp;journal=Annu%2E+Rev%2E+Ecol%2E%2C+Evol%2E+System%2E&amp;author=Peay+K.+G.&amp;publication_year=2016&amp;volume=47&amp;pages=143–164\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B46\" name=\"B46\"></a>Peterson, E. B. (1965). Inhibition of black spruce primary roots by a water-soluble substance in <i>Kalmia angustifolia</i>. <i>Forest Sci.</i> 11, 473–479.</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Inhibition+of+black+spruce+primary+roots+by+a+water-soluble+substance+in+Kalmia+angustifolia%2E&amp;journal=Forest+Sci%2E&amp;author=Peterson+E.+B.&amp;publication_year=1965&amp;volume=11&amp;pages=473–479\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B47\" name=\"B47\"></a>Pickles, B. J., and Simard, S. W. (2017). “Mycorrhizal networks and forest resilience to drought,” in <i>Mycorrhizal Mediation of Soil</i> (Amsterdam?, NE: Elsevier), 319–339. doi: 10.1016/b978-0-12-804312-7.00018-8</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1016/b978-0-12-804312-7.00018-8\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Mycorrhizal+networks+and+forest+resilience+to+drought&amp;journal=Mycorrhizal+Mediation+of+Soil&amp;author=Pickles+B.+J.&amp;author=Simard+S.+W.&amp;publication_year=2017&amp;pages=319–339\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B48\" name=\"B48\"></a>R Core Team (2018). <i>R: A Language and Environment for Statistical Computing.</i> Vienna: R Foundation for Statistical Computing</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://scholar.google.com/scholar_lookup?&amp;journal=R%3A+A+Language+and+Environment+for+Statistical+Computing%2E&amp;publication_year=2018\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B49\" name=\"B49\"></a>Ratcliffe, J. L., Creevy, A., Andersen, R., Zarov, E., Gaffney, P. P. J., Taggart, M. A., et al. (2017). Ecological and environmental transition across the forested-to-open bog ecotone in a west Siberian peatland. <i>Sci. Total Environ.</i> 607–608, 816–828. doi: 10.1016/j.scitotenv.2017.06.276</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=28711843\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1016/j.scitotenv.2017.06.276\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Ecological+and+environmental+transition+across+the+forested-to-open+bog+ecotone+in+a+west+Siberian+peatland%2E&amp;journal=Sci%2E+Total+Environ%2E&amp;author=Ratcliffe+J.+L.&amp;author=Creevy+A.&amp;author=Andersen+R.&amp;author=Zarov+E.&amp;author=Gaffney+P.+P.+J.&amp;author=Taggart+M.+A.&amp;publication_year=2017&amp;volume=607–608&amp;pages=816–828\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B50\" name=\"B50\"></a>Read, D. J. (1996). The structure and function of the ericoid mycorrhizal root. <i>Ann. Bot.</i> 77, 365–374. doi: 10.1006/anbo.1996.0044</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1006/anbo.1996.0044\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=The+structure+and+function+of+the+ericoid+mycorrhizal+root%2E&amp;journal=Ann%2E+Bot%2E&amp;author=Read+D.+J.&amp;publication_year=1996&amp;volume=77&amp;pages=365–374\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B51\" name=\"B51\"></a>Richard, F., Selosse, M. -A., and Gardes, M. (2009). Facilitated establishment of <i>Quercus ilex</i> in shrub- dominated communities within a mediterranean ecosystem: do mycorrhizal partners matter? <i>FEMS Microbiol. Ecol.</i> 68, 14–24. doi: 10.1111/j.1574-6941.2009.00646.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=19175678\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/j.1574-6941.2009.00646.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Facilitated+establishment+of+Quercus+ilex+in+shrub-+dominated+communities+within+a+mediterranean+ecosystem%3A+do+mycorrhizal+partners+matter%B4&amp;journal=FEMS+Microbiol%2E+Ecol%2E&amp;author=Richard+F.&amp;author=Selosse+M.+-A.&amp;author=Gardes+M.&amp;publication_year=2009&amp;volume=68&amp;pages=14–24\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B52\" name=\"B52\"></a>Robitaille, A., and Saucier, J. -P. (1996). “Land district, ecophysiographic units and areas: the landscape mapping of the ministère des ressources naturelles du Québec,” in <i>Global to Local: Ecological Land Classification</i>, eds R. A. Sims, I. G. W. Corns, and K. Klinka, (Springer: Dordrecht), 127–148 doi: 10.1007/bf00396141</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=24198002\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1007/bf00396141\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Land+district%2C+ecophysiographic+units+and+areas%3A+the+landscape+mapping+of+the+ministère+des+ressources+naturelles+du+Québec&amp;journal=Global+to+Local%3A+Ecological+Land+Classification&amp;author=Robitaille+A.&amp;author=Saucier+J.+-P.+(1996).+“Land+district+ecophysiographic+units+and+areas:+the+landscape+mapping+of+the+ministère+des+ressources+naturelles+du+Québec”+in+Global+to+Local:+Ecological+Land+Classification+eds+Sims+R.+A.&amp;author=Corns+I.+G.+W.&amp;author=Klinka+K.&amp;publication_year=1996&amp;pages=127–148\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B53\" name=\"B53\"></a>Rosseel, Y., Oberski, D., Byrnes, J., Vanbrabant, L., Savalei, V., Merkle, E., et al. (2018). <i>Lavaan: Latent Variable Analysi.</i> Available at: <a href=\"https://cran.r-project.org/package=lavaan\">https://cran.r-project.org/package=lavaan</a> (accessed November 21, 2018).</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://scholar.google.com/scholar_lookup?&amp;journal=Lavaan%3A+Latent+Variable+Analysi%2E&amp;author=Rosseel+Y.&amp;author=Oberski+D.&amp;author=Byrnes+J.&amp;author=Vanbrabant+L.&amp;author=Savalei+V.&amp;author=Merkle+E.&amp;publication_year=2018\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B54\" name=\"B54\"></a>Shipley, B. (2000). A new inferential test for path models based on directed acyclic graphs. <i>Struct. Equ. Modeling</i> 7, 206–218. doi: 10.1207/S15328007SEM0702_4</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1207/S15328007SEM0702_4\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=A+new+inferential+test+for+path+models+based+on+directed+acyclic+graphs%2E&amp;journal=Struct%2E+Equ%2E+Modeling&amp;author=Shipley+B.&amp;publication_year=2000&amp;volume=7&amp;pages=206–218\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B55\" name=\"B55\"></a>Simard, M., Lecomte, N., Bergeron, Y., Bernier, P. Y., and Paré, D. (2007). Forest productivity decline caused by successional paludification of boreal soils. <i>Ecol. Appl.</i> 17, 1619–1637. doi: 10.1890/06-1795.1</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=17913128\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1890/06-1795.1\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Forest+productivity+decline+caused+by+successional+paludification+of+boreal+soils%2E&amp;journal=Ecol%2E+Appl%2E&amp;author=Simard+M.&amp;author=Lecomte+N.&amp;author=Bergeron+Y.&amp;author=Bernier+P.+Y.&amp;author=Paré+D.&amp;publication_year=2007&amp;volume=17&amp;pages=1619–1637\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B56\" name=\"B56\"></a>Simard, S., Asay, A., Beiler, K., Bingham, M., Deslippe, J., He, X., et al. (2015). “Resource transfer between plants through ectomycorrhizal fungal networks,” in <i>Mycorrhizal Networks</i>, ed. T. R. Horton (Dordrecht: Springer), 133–176. doi: 10.1007/978-94-017-7395-9_5</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1007/978-94-017-7395-9_5\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Resource+transfer+between+plants+through+ectomycorrhizal+fungal+networks&amp;journal=Mycorrhizal+Networks&amp;author=Simard+S.&amp;author=Asay+A.&amp;author=Beiler+K.&amp;author=Bingham+M.&amp;author=Deslippe+J.&amp;author=He+X.+(2015).+“Resource+transfer+between+plants+through+ectomycorrhizal+fungal+networks”+in+Mycorrhizal+Networks+ed.+Horton+T.+R.&amp;publication_year=2015&amp;pages=133–176\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B57\" name=\"B57\"></a>Simard, S. W., Beiler, K. J., Bingham, M. A., Deslippe, J. R., Philip, L. J., and Teste, F. P. (2012). Mycorrhizal networks: mechanisms, ecology and modelling. <i>Fungal Biol. Rev.</i> 26, 39–60. doi: 10.1016/j.fbr.2012.01.001</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1016/j.fbr.2012.01.001\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Mycorrhizal+networks%3A+mechanisms%2C+ecology+and+modelling%2E&amp;journal=Fungal+Biol%2E+Rev%2E&amp;author=Simard+S.+W.&amp;author=Beiler+K.+J.&amp;author=Bingham+M.+A.&amp;author=Deslippe+J.+R.&amp;author=Philip+L.+J.&amp;author=Teste+F.+P.&amp;publication_year=2012&amp;volume=26&amp;pages=39–60\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B58\" name=\"B58\"></a>Smith, S. E., and Read, D. J. (2008). <i>Mycorrhizal Symbiosis.</i> Cambridge, MA: Academic Press.</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://scholar.google.com/scholar_lookup?&amp;journal=Mycorrhizal+Symbiosis%2E&amp;author=Smith+S.+E.&amp;author=Read+D.+J.&amp;publication_year=2008\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B59\" name=\"B59\"></a>Stachowicz, J. J. (2001). Mutualism, facilitation, and the structure of ecological communities. <i>BioScience</i> 51:235. doi: 10.1641/0006-3568(2001)051%5B0235:mfatso%5D2.0.co;2</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1641/0006-3568(2001)051%5B0235:mfatso%5D2.0.co;2\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Mutualism%2C+facilitation%2C+and+the+structure+of+ecological+communities%2E&amp;journal=BioScience&amp;author=Stachowicz+J.+J.&amp;publication_year=2001&amp;volume=51&amp;issue=235\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B60\" name=\"B60\"></a>Taylor, D. L., Hollingsworth, T. N., McFarland, J. W., Lennon, N. J., Nusbaum, C., and Ruess, R. W. (2013). A first comprehensive census of fungi in soil reveals both hyperdiversity and fine-scale niche partitioning. <i>Ecol. Monogr.</i> 84, 3–20. doi: 10.1890/12-1693.1</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1890/12-1693.1\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=A+first+comprehensive+census+of+fungi+in+soil+reveals+both+hyperdiversity+and+fine-scale+niche+partitioning%2E&amp;journal=Ecol%2E+Monogr%2E&amp;author=Taylor+D.+L.&amp;author=Hollingsworth+T.+N.&amp;author=McFarland+J.+W.&amp;author=Lennon+N.+J.&amp;author=Nusbaum+C.&amp;author=Ruess+R.+W.&amp;publication_year=2013&amp;volume=84&amp;pages=3–20\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B61\" name=\"B61\"></a>Tedersoo, L., Bahram, M., Toots, M., DiéDhiou, A. G., Henkel, T. W., KjøLler, R., et al. (2012). Towards global patterns in the diversity and community structure of ectomycorrhizal fungi: global metastudy of ectomycorrhizal fungi. <i>Mol. Ecol.</i> 21, 4160–4170. doi: 10.1111/j.1365-294X.2012.05602.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=22568722\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/j.1365-294X.2012.05602.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Towards+global+patterns+in+the+diversity+and+community+structure+of+ectomycorrhizal+fungi%3A+global+metastudy+of+ectomycorrhizal+fungi%2E&amp;journal=Mol%2E+Ecol%2E&amp;author=Tedersoo+L.&amp;author=Bahram+M.&amp;author=Toots+M.&amp;author=DiéDhiou+A.+G.&amp;author=Henkel+T.+W.&amp;author=KjøLler+R.&amp;publication_year=2012&amp;volume=21&amp;pages=4160–4170\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B62\" name=\"B62\"></a>Tedersoo, L., Mett, M., Ishida, T. A., and Bahram, M. (2013). Phylogenetic relationships among host plants explain differences in fungal species richness and community composition in ectomycorrhizal symbiosis. <i>New Phytol</i> 199, 822–831. doi: 10.1111/nph.12328</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=23692134\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/nph.12328\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Phylogenetic+relationships+among+host+plants+explain+differences+in+fungal+species+richness+and+community+composition+in+ectomycorrhizal+symbiosis%2E&amp;journal=New+Phytol&amp;author=Tedersoo+L.&amp;author=Mett+M.&amp;author=Ishida+T.+A.&amp;author=Bahram+M.&amp;publication_year=2013&amp;volume=199&amp;pages=822–831\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B63\" name=\"B63\"></a>Teste, F. P., Simard, S. W., Durall, D. M., Guy, R. D., Jones, M. D., and Schoonmaker, A. L. (2009). Access to mycorrhizal networks and roots of trees: importance for seedling survival and resource transfer. <i>Ecology</i> 90, 2808–2822. doi: 10.1890/08-1884.1</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=19886489\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1890/08-1884.1\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Access+to+mycorrhizal+networks+and+roots+of+trees%3A+importance+for+seedling+survival+and+resource+transfer%2E&amp;journal=Ecology&amp;author=Teste+F.+P.&amp;author=Simard+S.+W.&amp;author=Durall+D.+M.&amp;author=Guy+R.+D.&amp;author=Jones+M.+D.&amp;author=Schoonmaker+A.+L.&amp;publication_year=2009&amp;volume=90&amp;pages=2808–2822\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B64\" name=\"B64\"></a>Unterseher, M., Jumpponen, A., Öpik, M., Tedersoo, L., Moora, M., Dormann, C. F., et al. (2011). Species abundance distributions and richness estimations in fungal metagenomics – lessons learned from community ecology. <i>Mol. Ecol.</i> 20, 275–285. doi: 10.1111/j.1365-294X.2010.04948.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=21155911\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1111/j.1365-294X.2010.04948.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Species+abundance+distributions+and+richness+estimations+in+fungal+metagenomics+–+lessons+learned+from+community+ecology%2E&amp;journal=Mol%2E+Ecol%2E&amp;author=Unterseher+M.&amp;author=Jumpponen+A.&amp;author=Öpik+M.&amp;author=Tedersoo+L.&amp;author=Moora+M.&amp;author=Dormann+C.+F.&amp;publication_year=2011&amp;volume=20&amp;pages=275–285\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B65\" name=\"B65\"></a>van der Heijden, M. G. A., and Horton, T. R. (2009). Socialism in soil? The importance of mycorrhizal fungal networks for facilitation in natural ecosystems. <i>J. Ecol.</i> 97, 1139–1150. doi: 10.1111/j.1365-2745.2009.01570.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1111/j.1365-2745.2009.01570.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Socialism+in+soil%B4+The+importance+of+mycorrhizal+fungal+networks+for+facilitation+in+natural+ecosystems%2E&amp;journal=J%2E+Ecol%2E&amp;author=van+der+Heijden+M.+G.+A.&amp;author=Horton+T.+R.&amp;publication_year=2009&amp;volume=97&amp;pages=1139–1150\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B66\" name=\"B66\"></a>Villarreal-Ruiz, L., Anderson, I. C., and Alexander, I. J. (2004). Interaction between an isolate from the hymenoscyphus ericae aggregate and roots of pinus and vaccinium. <i>New Phytol.</i> 164, 183–192. doi: 10.1111/j.1469-8137.2004.01167.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1111/j.1469-8137.2004.01167.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Interaction+between+an+isolate+from+the+hymenoscyphus+ericae+aggregate+and+roots+of+pinus+and+vaccinium%2E&amp;journal=New+Phytol%2E&amp;author=Villarreal-Ruiz+L.&amp;author=Anderson+I.+C.&amp;author=Alexander+I.+J.&amp;publication_year=2004&amp;volume=164&amp;pages=183–192\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B67\" name=\"B67\"></a>Vrålstad, T., Fossheim, T. Schumacher, T. (2000). Piceirhiza bicolorata- the ectomycorrhizal expression of the hymenoscyphus ericae aggregate? <i>New Phytol.</i> 145, 549–563. doi: 10.1046/j.1469-8137.2000.00605.x</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1046/j.1469-8137.2000.00605.x\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Piceirhiza+bicolorata-+the+ectomycorrhizal+expression+of+the+hymenoscyphus+ericae+aggregate%B4&amp;journal=New+Phytol%2E&amp;author=Vrålstad+T.&amp;author=Fossheim+T.&amp;author=Schumacher+T.&amp;publication_year=2000&amp;volume=145&amp;pages=549–563\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B68\" name=\"B68\"></a>Walker, J. F. Jr., Miller, O. K., Lei, T., Semones, S., Nilsen, E., and Clinton, B. D. (1999). Suppression of ectomycorrhizae on canopy tree seedlings in <i>Rhododendron maximum</i> L. <i>(Ericaceae</i>) thickets in the southern Appalachians. <i>Mycorrhiza</i> 9, 49–56. doi: 10.1007/s005720050262</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1007/s005720050262\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Suppression+of+ectomycorrhizae+on+canopy+tree+seedlings+in+Rhododendron+maximum+L%2E+%28Ericaceae%29+thickets+in+the+southern+Appalachians%2E&amp;journal=Mycorrhiza&amp;author=Walker+J.+F.+Jr.&amp;author=Miller+O.+K.&amp;author=Lei+T.&amp;author=Semones+S.&amp;author=Nilsen+E.&amp;author=Clinton+B.+D.&amp;publication_year=1999&amp;volume=9&amp;pages=49–56\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B69\" name=\"B69\"></a>White, T., Bruns, T., Lee, S., Taylor, J., Innis, M., Gelfand, D., et al. (1990). Amplification and direct sequencing of fungal ribosomal RNA genes for phylogenetics. in: <i>PCR <u>Protocols</u> : a <u>Guide</u> to <u>Methods</u> and <u>Applications</u></i> ed. MA. Innis, (Cambridge: Academic Press), 315–322 doi: 10.1016/b978-0-12-372180-8.50042-1</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1016/b978-0-12-372180-8.50042-1\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Amplification+and+direct+sequencing+of+fungal+ribosomal+RNA+genes+for+phylogenetics%2E&amp;journal=PCR+Protocols+%3A+a+Guide+to+Methods+and+Applications&amp;author=White+T.&amp;author=Bruns+T.&amp;author=Lee+S.&amp;author=Taylor+J.&amp;author=Innis+M.&amp;author=Gelfand+D.+(1990).+Amplification+and+direct+sequencing+of+fungal+ribosomal+RNA+genes+for+phylogenetics.+in:+PCR+Protocols+:+a+Guide+to+Methods+and+Applications+ed.+Innis+MA.&amp;publication_year=1990&amp;pages=315–322\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B70\" name=\"B70\"></a>Woodall, C. W., Oswalt, C. M., Westfall, J. A., Perry, C. H., Nelson, M. D., and Finley, A. O. (2009). An indicator of tree migration in forests of the eastern United States. <i>For. Ecol. Manag.</i> 257, 1434–1444. doi: 10.1016/j.foreco.2008.12.013</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=30659144\" target=\"_blank\">PubMed Abstract</a> | <a href=\"https://doi.org/10.1016/j.foreco.2008.12.013\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=An+indicator+of+tree+migration+in+forests+of+the+eastern+United+States%2E&amp;journal=For%2E+Ecol%2E+Manag%2E&amp;author=Woodall+C.+W.&amp;author=Oswalt+C.+M.&amp;author=Westfall+J.+A.&amp;author=Perry+C.+H.&amp;author=Nelson+M.+D.&amp;author=Finley+A.+O.&amp;publication_year=2009&amp;volume=257&amp;pages=1434–1444\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B71\" name=\"B71\"></a>Yamasaki, S. H., Fyles, J. W., Egger, K. N., and Titus, B. D. (1998). The effect of <i>Kalmia angustifolia</i> on the growth, nutrition, and ectomycorrhizal symbiont community of black spruce. <i>For. Ecol. Manag.</i> 105, 197–207. doi: 10.1016/S0378-1127(97)00285</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.1016/S0378-1127(97)00285\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=The+effect+of+Kalmia+angustifolia+on+the+growth%2C+nutrition%2C+and+ectomycorrhizal+symbiont+community+of+black+spruce%2E&amp;journal=For%2E+Ecol%2E+Manag%2E&amp;author=Yamasaki+S.+H.&amp;author=Fyles+J.+W.&amp;author=Egger+K.+N.&amp;author=Titus+B.+D.&amp;publication_year=1998&amp;volume=105&amp;pages=197–207\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "<div class=\"References\" style=\"margin-bottom:0.5em;\">\n",
       "<p class=\"ReferencesCopy1\"><a id=\"B72\" name=\"B72\"></a>Zackrisson, O., Nilsson, M. -C., Dahlberg, A., and Jäderlund, A. (1997). Interference mechanisms in conifer-Ericaceae-feathermoss communities. <i>Oikos</i> 78, 209–220. doi: 10.2307/3546287</p>\n",
       "<p class=\"ReferencesCopy2\"><a href=\"https://doi.org/10.2307/3546287\" target=\"_blank\">CrossRef Full Text</a> | <a href=\"http://scholar.google.com/scholar_lookup?&amp;title=Interference+mechanisms+in+conifer-Ericaceae-feathermoss+communities%2E&amp;journal=Oikos&amp;author=Zackrisson+O.&amp;author=Nilsson+M.+-C.&amp;author=Dahlberg+A.&amp;author=Jäderlund+A.&amp;publication_year=1997&amp;volume=78&amp;pages=209–220\" target=\"_blank\">Google Scholar</a></p>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"thinLineM20\"></div>\n",
       "<div class=\"AbstractSummary\">\n",
       "<p><span>Keywords</span>: <i>Abies balsamea</i>, boreal forest, ectomycorrhiza, ericaceous shrubs, facilitation, <i>Picea mariana</i>, <i>Populus tremuloide</i>s</p>\n",
       "<p><span>Citation:</span> Nagati M, Roy M, Desrochers A, Manzi S, Bergeron Y and Gardes M (2019) Facilitation of Balsam Fir by Trembling Aspen in the Boreal Forest: Do Ectomycorrhizal Communities Matter? <i>Front. Plant Sci.</i> 10:932. doi: 10.3389/fpls.2019.00932</p>\n",
       "<p id=\"timestamps\">\n",
       "<span>Received:</span> 03 April 2019; <span>Accepted:</span> 03 July 2019;<br/><span>Published:</span> 18 July 2019.</p>\n",
       "<div>\n",
       "<p>Edited by:</p>\n",
       "<a href=\"https://loop.frontiersin.org/people/119073/overview\">Kevin Garcia</a>, North Carolina State University, United States</div>\n",
       "<div>\n",
       "<p>Reviewed by:</p>\n",
       "<a href=\"https://loop.frontiersin.org/people/113197/overview\">Jaqueline Hess</a>, University of Vienna, Austria<br/>\n",
       "<a href=\"https://loop.frontiersin.org/people/122827/overview\">Plassard Claude</a>, Ecologie Fonctionnelle et Biogéochimie des Sols et des Agro-Écosystèmes (Eco&amp;Sols) INRA, France</div>\n",
       "<p><span>Copyright</span> © 2019 Nagati, Roy, Desrochers, Manzi, Bergeron and Gardes. This is an open-access article distributed under the terms of the <a href=\"http://creativecommons.org/licenses/by/4.0/\" rel=\"license\" target=\"_blank\">Creative Commons Attribution License (CC BY)</a>. The use, distribution or reproduction in other forums is permitted, provided the original author(s) and the copyright owner(s) are credited and that the original publication in this journal is cited, in accordance with accepted academic practice. No use, distribution or reproduction is permitted which does not comply with these terms.</p>\n",
       "<p><span>*Correspondence:</span> Mélissande Nagati, <a href=\"mailto:melissande.nagati@uqat.ca\">melissande.nagati@uqat.ca</a></p>\n",
       "<div class=\"clear\"></div>\n",
       "<p data-test-id=\"article-disclaimer-text\"><span> Disclaimer: </span>All claims expressed in this article are solely those of the authors and do not necessarily represent those of their affiliated organizations, or those of the publisher, the editors and the reviewers. Any product that may be evaluated in this article or claim that may be made by its manufacturer is not guaranteed or endorsed by the publisher. </p></div>\n",
       "<!-- TODO:BEGIN IS PART OF RT-->\n",
       "<script type=\"text/javascript\">\n",
       "      \n",
       "            const isPartOfRTElements = document.getElementsByClassName('research-topic-container');\n",
       "\n",
       "            if (isPartOfRTElements.length > 0) {\n",
       "                const headerBar3Container = document.getElementsByClassName('header-bar-three-container');\n",
       "\n",
       "                if (headerBar3Container.length > 0) {\n",
       "                    const container = headerBar3Container[0];\n",
       "                    container.appendChild(isPartOfRTElements[0]);\n",
       "                    isPartOfRTElements[0].style.display = 'block';\n",
       "                }\n",
       "            }\n",
       "       \n",
       "</script>\n",
       "<!-- TODO:END IS PART OF RT-->\n",
       "<div class=\"thin-line-dark\"></div>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "</main>\n",
       "</div>\n",
       "<div class=\"side-article-subjects only-devices widget-listing people-also-looked-at\">\n",
       "<h5 class=\"like-h4\">People also looked at</h5>\n",
       "<div class=\"teaser\"><div class=\"heading-container\"><h2 class=\"teaser-heading\"><a href=\"/articles/10.3389/fpls.2022.759801/full\">A Closer Examination of the ‘Abundant-Center’ for Ectomycorrhizal Fungal Community Associated With Picea crassifolia in China</a></h2><p class=\"teaser-authors\"><a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/1673684/overview\">Xiaobing Wang</a> and <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/1444600/overview\">Qisheng Han</a></p></div></div><div class=\"teaser\"><div class=\"heading-container\"><h2 class=\"teaser-heading\"><a href=\"/articles/10.3389/fpls.2019.00895/full\">Aphids Influence Soil Fungal Communities in Conventional Agricultural Systems</a></h2><p class=\"teaser-authors\"><a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/639218/overview\">Thomas D. J. Wilkinson</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/736625/overview\">Jean-Pascal Miranda</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/699069/overview\">Julia Ferrari</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/122729/overview\">Sue E. Hartley</a> and <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/699023/overview\">Angela Hodge</a></p></div></div><div class=\"teaser\"><div class=\"heading-container\"><h2 class=\"teaser-heading\"><a href=\"/articles/10.3389/fmicb.2019.01888/full\">Perspectives on Endosymbiosis in Coralloid Roots: Association of Cycads and Cyanobacteria</a></h2><p class=\"teaser-authors\"><a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/691177/overview\">Aimee Caye G. Chang</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/786490/overview\">Tao Chen</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/690638/overview\">Nan Li</a> and <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/201185/overview\">Jun Duan</a></p></div></div><div class=\"teaser\"><div class=\"heading-container\"><h2 class=\"teaser-heading\"><a href=\"/articles/10.3389/fpls.2019.00931/full\">CrERF5, an AP2/ERF Transcription Factor, Positively Regulates the Biosynthesis of Bisindole Alkaloids and Their Precursors in Catharanthus roseus</a></h2><p class=\"teaser-authors\"><a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/625039/overview\">Qifang Pan</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/700893/overview\">Chenyi Wang</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/701090/overview\">Zhiwei Xiong</a>, Hang Wang, Xueqing Fu, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/366528/overview\">Qian Shen</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/669885/overview\">Bowen Peng</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/771064/overview\">Yanan Ma</a>, Xiaofen Sun and <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/219192/overview\">Kexuan Tang</a></p></div></div><div class=\"teaser\"><div class=\"heading-container\"><h2 class=\"teaser-heading\"><a href=\"/articles/10.3389/fpls.2019.00935/full\">Marchantia polymorpha, a New Model Plant for Autophagy Studies</a></h2><p class=\"teaser-authors\"><a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/740213/overview\">Takuya Norizuki</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/740248/overview\">Takehiko Kanazawa</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/740355/overview\">Naoki Minamino</a>, <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/26459/overview\">Hirokazu Tsukaya</a> and <a data-event=\"peoplelookedat-link-click\" href=\"https://loop.frontiersin.org/people/49517/overview\">Takashi Ueda</a></p></div></div></div>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div id=\"divTemplates\">\n",
       "<div aria-hidden=\"true\" aria-labelledby=\"myModalLabel\" class=\"modal fade modal-container impact-modal-container\" data-backdrop=\"static\" id=\"impactModal\" role=\"dialog\" tabindex=\"-1\">\n",
       "</div>\n",
       "<div aria-hidden=\"true\" aria-labelledby=\"mySupplementaryModalLabel\" class=\"modal fade modal-container supplementary-modal-container\" data-backdrop=\"static\" data-keyboard=\"false\" id=\"supplementaryFilesModal\" role=\"dialog\" style=\"display: none; padding-right: 17px;\" tabindex=\"-1\">\n",
       "</div>\n",
       "<script id=\"template-supplementary-files-modal\" type=\"text/template\">\n",
       "    <div class=\"supplementary-material-wrapper modal-dialog\" role=\"document\">\n",
       "        <div class=\"modal-content\">\n",
       "            <div class=\"modal-body\">\n",
       "                <a class=\"close\" data-dismiss=\"modal\"  aria-label=\"Close\"><span data-event=\"supplementaryclose-icon-click\" aria-hidden=\"true\">×</span></a>\n",
       "                <h4>Supplementary Material</h4><br>\n",
       "                <p class=\"supplementary-empty-message\" style=\"display: none;\">There is no supplementary material currently available for this article</p>\n",
       "                <div class=\"loading-wrapper\" id=\"loader\" style=\"display: none;\">\n",
       "                    Loading supplemental data...\n",
       "                    <img style=\"\" src=\"/Areas/Articles/Images/Icon/loading.gif\" alt=\"Loading..\">\n",
       "                </div>\n",
       "                <br>\n",
       "                <div class=\"table-responsive\" id=\"localFiles\">\n",
       "\n",
       "                </div>\n",
       "                <div id=\"figshare-widget-container\">\n",
       "\n",
       "                </div>\n",
       "            </div>\n",
       "            <br>\n",
       "            <div class=\"modal-footer bottom-links\">\n",
       "                <a class=\"btn btn-default\" data-dismiss=\"modal\">Close</a>\n",
       "            </div>\n",
       "        </div>\n",
       "\n",
       "    </div>\n",
       "</script>\n",
       "<script id=\"template-local-files-modal\" type=\"text/template\">\n",
       "    <table class=\"table table-striped supplementary-content\" id=\"localFilesTable\">\n",
       "        <thead>\n",
       "            <tr>\n",
       "                <th>&nbsp;</th>\n",
       "                <th>File Name</th>\n",
       "                <th>&nbsp;</th>\n",
       "            </tr>\n",
       "        </thead>\n",
       "        <tbody class=\"file-content\">\n",
       "            <% _.each(supplimentalFileDetails.FileDetails, function(fileViewModel){   %>\n",
       "            <tr>\n",
       "                <th scope=\"row\"><a data-test-id=\"<%= fileViewModel.TestId %>\" href=\"<%= fileViewModel.FileDownloadUrl %>\"><img class=\"img-figure\" src=\"<%= fileViewModel.ImageUrl %>\"></a></th>\n",
       "                <td>\n",
       "                    <a  href=\"<%= fileViewModel.FileDownloadUrl %>\" data-event=\"supplementarydownload-button-click\">\n",
       "                        <%= fileViewModel.FileName %>\n",
       "                    </a>\n",
       "                </td>\n",
       "                <td><a  href=\"<%= fileViewModel.FileDownloadUrl %>\" data-event=\"supplementarydownload-button-click\" class=\"btn-link\"><i class=\"fa fa-download\" aria-hidden=\"true\"></i></a></td>\n",
       "            </tr>\n",
       "            <% }); %>\n",
       "        </tbody>\n",
       "    </table>\n",
       "</script>\n",
       "<div aria-hidden=\"true\" aria-labelledby=\"Notify on publication\" class=\"modal fade modal-container notifyme-modal-container\" data-backdrop=\"static\" id=\"notifyModal\" role=\"dialog\" tabindex=\"-1\">\n",
       "</div>\n",
       "<script id=\"template-notify-me-modal\" type=\"text/template\">\n",
       "    <div class=\"modal-f page-container simple-modal\" role=\"dialog\" aria-labelledby=\"NotfifyMeModalLabel\" id=\"notify_me_popup\">\n",
       "            <div class=\"modal-dialog\" role=\"document\">\n",
       "                <div class=\"modal-content\">\n",
       "                    <form onsubmit=\"return false\" id=\"modalnotifyme\">\n",
       "                        <div class=\"modal-header\">\n",
       "                            <button type=\"button\" class=\"close\" data-dismiss=\"modal\" aria-label=\"Close\"><span aria-hidden=\"true\">×</span></button>\n",
       "                            <h3>Notify me on publication</h3>\n",
       "                        </div>\n",
       "                        <div class=\"modal-body\">\n",
       "\n",
       "                            <div class=\"form-group\">\n",
       "                                <label>Please enter your email address:</label>\n",
       "                            <input type=\"email\" required class=\"form-control\" id=\"txt_notification_email_id\" placeholder=\"Email\">\n",
       "                            </div>\n",
       "\n",
       "                        <div class=\"g-recaptcha\" id=\"recaptcha\"></div><br />\n",
       "\n",
       "                        <p class=\"small\" style=\"color: #555;\">If you already have an account, please <a href=\"<%=login.FrontiersLoginUrl%>?returnUrl=<%=currentPage%>\" class=\"text-blue\">login</a>.</p>\n",
       "                        <p class=\"small\" style=\"color: #555;\">You don't have a Frontiers account ? You can <a href=\"<%=login.FrontiersRegistrationUrl%>\" class=\"text-blue\">register here</a>.</p>\n",
       "                        </div>\n",
       "                        <div class=\"modal-footer\">\n",
       "                        <button type=\"button\" id=\"article_notify_non_registered_user\" class=\"btn btn-default btn-progress\" style=\"width: 118px; visibility:hidden;\">Notify me</button>\n",
       "                        </div>\n",
       "                    </form>\n",
       "                </div>\n",
       "            </div>\n",
       "        </div>\n",
       "</script>\n",
       "<script id=\"template-notifyme-mailselection-modal\" type=\"text/template\">\n",
       "    <div class=\"modal-f page-container simple-modal\" tabindex=\"-1\" role=\"dialog\" aria-labelledby=\"NotfifyMeModalLabel\">\n",
       "        <div class=\"modal-dialog\" role=\"document\">\n",
       "            <div class=\"modal-content\">\n",
       "                <div class=\"modal-header\">\n",
       "                    <button type=\"button\" class=\"close\" data-dismiss=\"modal\" aria-label=\"Close\"><span aria-hidden=\"true\">×</span></button>\n",
       "                    <h3>Select one of your emails</h3>\n",
       "                </div>\n",
       "                <div class=\"modal-body\">\n",
       "                    <form>\n",
       "                        <div class=\"form-group\">\n",
       "                            <label>You have multiple emails registered with Frontiers:</label>\n",
       "                        </div>\n",
       "                                           \n",
       "                        <% _.each(userEmailList, function(user){   %>\n",
       "                        <input type=\"radio\" value=\"<%=user%>\" name=\"notify_Emailcollection\"> <%=user%><br>\n",
       "                        <% }); %>\n",
       "\n",
       "                    </form>\n",
       "                </div>\n",
       "                <div class=\"modal-footer\">\n",
       "                    <button type=\"button\" id=\"article_notify_loggedin_user\" class=\"btn btn-default btn-progress\" style=\"width: 118px;\">Notify me</button>\n",
       "                </div>\n",
       "            </div>\n",
       "\n",
       "        </div>\n",
       "\n",
       "    </div>\n",
       "</script>\n",
       "<script id=\"template_notifyme_error_modal\" type=\"text/template\">\n",
       "    <div class=\"modal-f page-container simple-modal\" tabindex=\"-1\" role=\"dialog\" aria-labelledby=\"NotfifyMeModalLabel\">\n",
       "        <div class=\"modal-dialog\" role=\"document\">\n",
       "            <div class=\"modal-content\">\n",
       "                <div class=\"modal-header\">\n",
       "                    <button type=\"button\" class=\"close\" data-dismiss=\"modal\" aria-label=\"Close\"><span aria-hidden=\"true\">×</span></button>\n",
       "                    <h3>Notify me on publication</h3>\n",
       "                </div>\n",
       "                <div class=\"modal-body\">\n",
       "                    <form>\n",
       "                        <div class=\"form-group\">\n",
       "                            <p class=\"text-red\"><strong><%=message%></strong></p>\n",
       "                        </div>\n",
       "                    </form>\n",
       "                </div>\n",
       "                <div class=\"modal-footer\">\n",
       "                    <button type=\"button\" data-dismiss=\"modal\" class=\"btn btn-default btn-progress\" style=\"width: 118px;\">Cancel</button>\n",
       "                </div>\n",
       "            </div>\n",
       "        </div>\n",
       "    </div>\n",
       "</script>\n",
       "</div>\n",
       "</div>\n",
       "<a class=\"floating-button\" data-event=\"downloadfloating-button-click\" href=\"/articles/10.3389/fpls.2019.00932/pdf\" id=\"floating-download-pdf-btn\">\n",
       "<span class=\"floating-button-text\" data-event=\"downloadfloating-button-click\">Download</span>\n",
       "</a>\n",
       "<script type=\"text/javascript\">\n",
       "\n",
       "            $(document).ready(function () {\n",
       "                document.onload = observerDowloadFloatingBtn();\n",
       "            });\n",
       "            function downloadButtonVisibilityChange(entries) {\n",
       "                const btn = entries[0];\n",
       "                const floatingBtn = document.querySelector(\"#floating-download-pdf-btn\");\n",
       "                if (!btn.isIntersecting) {\n",
       "                    floatingBtn.classList.add(\"floating-button--show\");\n",
       "                } else {\n",
       "                    floatingBtn.classList.remove(\"floating-button--show\");\n",
       "                }\n",
       "            }\n",
       "\n",
       "            function observerDowloadFloatingBtn() {\n",
       "                const options = {\n",
       "                    root: document.getElementsByName('body')[0],\n",
       "                    rootMargin: '0px',\n",
       "                    threshold: 1.0\n",
       "                }\n",
       "\n",
       "                const observer = new IntersectionObserver(downloadButtonVisibilityChange, options);\n",
       "                const target = document.querySelector('[data-test-id=\"download-button\"]');\n",
       "                observer.observe(target);\n",
       "            }\n",
       "        </script>\n",
       "<frontiers-footer data='{\"blocks\":[{\"title\":\"Guidelines\",\"links\":[{\"text\":\"Author guidelines\",\"url\":\"/guidelines/author-guidelines\",\"target\":\"_self\",\"ariaLabel\":null},{\"text\":\"Editor guidelines\",\"url\":\"/guidelines/editor-guidelines\",\"target\":\"_self\",\"ariaLabel\":null},{\"text\":\"Policies and publication ethics\",\"url\":\"/guidelines/policies-and-publication-ethics\",\"target\":\"_self\",\"ariaLabel\":null},{\"text\":\"Fee policy\",\"url\":\"/about/fee-policy\",\"target\":\"_self\",\"ariaLabel\":null}]},{\"title\":\"Explore\",\"links\":[{\"text\":\"Articles\",\"url\":\"/articles\",\"target\":\"_self\",\"ariaLabel\":null},{\"text\":\"Research Topics \",\"url\":\"/research-topics\",\"target\":\"_self\",\"ariaLabel\":null},{\"text\":\"Journals\",\"url\":\"/journals\",\"target\":\"_self\",\"ariaLabel\":null}]},{\"title\":\"Outreach\",\"links\":[{\"text\":\"Frontiers Forum \",\"url\":\"https://forum.frontiersin.org/\",\"target\":\"_blank\",\"ariaLabel\":\"Frontiers Forum website\"},{\"text\":\"Frontiers Policy Labs \",\"url\":\"https://policylabs.frontiersin.org/\",\"target\":\"_blank\",\"ariaLabel\":null},{\"text\":\"Frontiers for Young Minds\",\"url\":\"https://kids.frontiersin.org/\",\"target\":\"_blank\",\"ariaLabel\":\"Contact Frontiers\"}]},{\"title\":\"Connect\",\"links\":[{\"text\":\"Help center\",\"url\":\"https://zendesk.frontiersin.org/hc/en-us\",\"target\":\"_blank\",\"ariaLabel\":\"Careers in Frontiers\"},{\"text\":\"Emails and alerts \",\"url\":\"https://loop.frontiersin.org/settings/email-preferences?a=publishers\",\"target\":\"_blank\",\"ariaLabel\":\"Subscribe to Frontiers emails\"},{\"text\":\"Contact us \",\"url\":\"/about/contact\",\"target\":\"_self\",\"ariaLabel\":\"Subscribe to newsletter\"},{\"text\":\"Submit your research\",\"url\":\"/submission/submit\",\"target\":\"_self\",\"ariaLabel\":null},{\"text\":\"Career opportunities\",\"url\":\"https://careers.frontiersin.org/\",\"target\":\"_blank\",\"ariaLabel\":null}]}],\"socialLinks\":[{\"link\":{\"text\":\"Frontiers Facebook\",\"url\":\"https://www.facebook.com/Frontiersin\",\"target\":\"_blank\",\"ariaLabel\":\"Frontiers Facebook\"},\"type\":\"Link\",\"color\":\"Grey\",\"icon\":\"Facebook\",\"size\":\"Medium\",\"hiddenText\":true},{\"link\":{\"text\":\"Frontiers Twitter\",\"url\":\"https://twitter.com/frontiersin\",\"target\":\"_blank\",\"ariaLabel\":null},\"type\":\"Link\",\"color\":\"Grey\",\"icon\":\"Twitter\",\"size\":\"Medium\",\"hiddenText\":true},{\"link\":{\"text\":\"Frontiers LinkedIn\",\"url\":\"https://www.linkedin.com/company/frontiers\",\"target\":\"_blank\",\"ariaLabel\":null},\"type\":\"Link\",\"color\":\"Grey\",\"icon\":\"LinkedIn\",\"size\":\"Medium\",\"hiddenText\":true},{\"link\":{\"text\":\"Frontiers Instagram\",\"url\":\"https://www.instagram.com/frontiersin_\",\"target\":\"_blank\",\"ariaLabel\":null},\"type\":\"Link\",\"color\":\"Grey\",\"icon\":\"Instagram\",\"size\":\"Medium\",\"hiddenText\":true}],\"copyright\":\"Frontiers Media S.A. All rights reserved\",\"termsAndConditionsUrl\":\"https://www.frontiersin.org/legal/terms-and-conditions\",\"privacyPolicyUrl\":\"https://www.frontiersin.org/legal/privacy-policy\"}'></frontiers-footer>\n",
       "<script src=\"https://crossmark.crossref.org/javascripts/v1.5/crossmark.min.js\"></script>\n",
       "<script type=\"text/javascript\">\n",
       "            var $ = jQuery.noConflict(); //==>>ToDo: Avoid this..!!\n",
       "        </script>\n",
       "<script async=\"\" src=\" https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML\"></script>\n",
       "<script async=\"\" src=\" https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js\"></script>\n",
       "<script src=\"https://static.frontiersin.org/areas/articles/js/frontiers?v=z1cL-Tf1-Qh97gw19mFq-3nre84X6laFLbhVr-YdIA81\"></script>\n",
       "<script type=\"text/javascript\">\n",
       "\n",
       "        var FRArticle = (function() {\n",
       "            return {\n",
       "                ArticleId: '464100',\n",
       "                DOI:'10.3389/fpls.2019.00932',\n",
       "                LoginUserId: '0',\n",
       "                DomainId: '1',\n",
       "                FieldId: '66',\n",
       "                SpecialityId: '1218',\n",
       "                IsPreview: FRSafe.boolean('False'),\n",
       "                IsViewImpactFromLoop: FRSafe.boolean('False'),\n",
       "                IsPublished: 'True',\n",
       "                FigShareApiUrl: 'https://api.figshare.com/v2/collections/search',\n",
       "                FigShareTimeOut: '3000',\n",
       "                ProvisionallyAcceptedText: 'The final, formatted version of the article will be published soon.',\n",
       "                NotifyMeButtonText: 'Notify me',\n",
       "                NotifyMeButtonSecondaryText: 'We&#39;ll notify you'\n",
       "            };\n",
       "        })();\n",
       "\n",
       "        var FRSocial = (function() {\n",
       "            return {\n",
       "                itemId: 14,\n",
       "                itemTypeId: 1,\n",
       "                entityId: '464100',\n",
       "                ownerId: '639840',\n",
       "                sanPath: 'https://www.frontiersin.org/files/',\n",
       "                subItemId: '8',\n",
       "                loginUserId: '0',\n",
       "                ownerNWDBId: '21',\n",
       "                pageType: 1,\n",
       "                loopUrl:'https://loop.frontiersin.org'\n",
       "            };\n",
       "        })();\n",
       "    </script>\n",
       "<script>\n",
       "        var FRAjaxSettings = (function () {\n",
       "\n",
       "            function urlLowercase() {\n",
       "                $.ajaxSetup({\n",
       "                    beforeSend: function (jqXHR, settings) {\n",
       "                        settings.url = settings.url.toLowerCase();\n",
       "                    }\n",
       "                });\n",
       "            }\n",
       "\n",
       "            return { urlLowercase: urlLowercase };\n",
       "        })();\n",
       "\n",
       "        FRAjaxSettings.urlLowercase();\n",
       "    </script>\n",
       "<script>\n",
       "        function setBodyOverflow(dropdownElement) {\n",
       "            if (!dropdownElement) return;\n",
       "            const windowWidth = window.innerWidth;\n",
       "            if (windowWidth <= 1024) {\n",
       "                const body = document.querySelector(\"body\");\n",
       "                if (dropdownElement.classList.contains(\"open\")) {\n",
       "                    body.style.overflow = \"hidden\";\n",
       "                } else {\n",
       "                    body.style.overflow = \"auto\";\n",
       "                }\n",
       "            }\n",
       "        }\n",
       "        $(document).ready(function () {\n",
       "            $(document).on('click', '.dropdown-menu-wrapper', function (e) {\n",
       "                e.stopPropagation();\n",
       "            });\n",
       "\n",
       "            $(document).on('click', '[data-toggle=\"dropdown\"]', function (e) {\n",
       "                const dropdownElement = e.target.closest(\".dropdown\")\n",
       "                if (dropdownElement) {\n",
       "                    setBodyOverflow(dropdownElement);\n",
       "                }\n",
       "            });\n",
       "        });\n",
       "\n",
       "        function supplementalDataToggle() {\n",
       "            const el = document.getElementById(\"supplementaryFilesModal\");\n",
       "            if (el) {\n",
       "                el.classList.toggle(\"in\");\n",
       "                if (el.classList.contains(\"in\")) {\n",
       "                    el.style.display = \"block\";\n",
       "                } else {\n",
       "                    el.style.display = \"none\";\n",
       "                }\n",
       "            }\n",
       "        }\n",
       "\n",
       "        function initToggleSupplementalData() {\n",
       "            const btn = document.getElementsByClassName(\"btn-open-supplemental\")[0];\n",
       "            const btn2 = document.getElementsByClassName(\"btn-open-supplemental\")[1];\n",
       "            const closeButtons = document.getElementById(\"supplementaryFilesModal\").querySelectorAll(\"[data-dismiss='modal']\");\n",
       "\n",
       "            if (btn) { btn.onclick = supplementalDataToggle; }\n",
       "            if (btn2) { btn2.onclick = supplementalDataToggle; }\n",
       "            if (closeButtons) { closeButtons.forEach(b => b.onclick = supplementalDataToggle); }\n",
       "        }\n",
       "\n",
       "\n",
       "        initToggleSupplementalData();\n",
       "    </script>\n",
       "<!-- BEGIN TODO UXD-1800: Inconsistent sizing of equations -->\n",
       "<script type=\"text/x-mathjax-config;executed=true\">\n",
       "        MathJax.Hub.Config({\n",
       "          CommonHTML: {\n",
       "            scale: 80\n",
       "          },\n",
       "          PreviewHTML: {\n",
       "            scale: 80\n",
       "          }\n",
       "        });\n",
       "    </script>\n",
       "<!-- END TODO UXD-1800: Inconsistent sizing of equations -->\n",
       "<script src=\"https://static.frontiersin.org/areas/articles/js/schema?v=13am1aL5JpCh_IIFOIFRdlFySxVv16JLaVykDEyHCu81\"></script>\n",
       "<script src=\"https://static.frontiersin.org/areas/articles/js/app?v=yc-RCjknzmUBNgZ4UvMYfxSO37HVPyGXugXlBXhhG9g1\"></script>\n",
       "<script src=\"https://static.frontiersin.org/areas/articles/js/webcomponents?v=bKOBZHIGOrdSOCedjsN1v9v4rZqWmv6pnhUmSmk47L01\"></script>\n",
       "<div class=\"crossmark-tooltip\" id=\"crossmark-tooltip-130\" style=\"display: none;\"><div class=\"cmtttop\"></div><div class=\"cmttmid\"><p>Click to get updates and verify authenticity.</p></div><div class=\"cmttbot\"></div></div><div style=\"display: none; visibility: hidden;\">\n",
       "<script async=\"null\" charset=\"UTF-8\" data-domain-script=\"1e9ad659-0bd5-4a8a-9508-487c73b460bd\" src=\"https://cdn.cookielaw.org/scripttemplates/otSDKStub.js\" type=\"text/javascript\"></script>\n",
       "<script type=\"text/javascript\">function OptanonWrapper(){};</script>\n",
       "</div><script src=\"https://api.altmetric.com/v1/doi/10.3389/fpls.2019.00932?callback=_altmetric.embed_callback&amp;domain=www.frontiersin.org&amp;key=3c130976ca2b8f2e88f8377633751ba1&amp;cache_until=17-14\" type=\"text/javascript\"></script><div class=\"altmetric-embed altmetric-popover altmetric-bottom\" data-uuid=\"02fd5df8-1e98-a739-d2f8-15d40a4df176\" id=\"_altmetric_popover_el_02fd5df8-1e98-a739-d2f8-15d40a4df176\" style=\"margin: 0px; display: none;\"><div class=\"altmetric_container\"><div class=\"altmetric_arrow altmetric-bottom\"></div><div class=\"altmetric-embed altmetric-popover-inner altmetric-floating altmetric-bottom\" data-uuid=\"02fd5df8-1e98-a739-d2f8-15d40a4df176\"><div class=\"altmetric-embed altmetric-popover-content altmetric-floating\"><div> <div style=\"padding-left: 10px; line-height:18px; border-left: 16px solid #FF0000;\">\n",
       "<a class=\"link-to-altmetric-details-tab\" href=\"https://www.altmetric.com/details.php?domain=www.frontiersin.org&amp;citation_id=63734010&amp;tab=news\" target=\"_self\">\n",
       "          News (1)\n",
       "      </a>\n",
       "</div>\n",
       "<div style=\"padding-left: 10px; line-height:18px; border-left: 16px solid #74CFED;\">\n",
       "<a class=\"link-to-altmetric-details-tab\" href=\"https://www.altmetric.com/details.php?domain=www.frontiersin.org&amp;citation_id=63734010&amp;tab=twitter\" target=\"_self\">\n",
       "          Twitter (16)\n",
       "      </a>\n",
       "</div>\n",
       "<div style=\"padding-left: 10px; line-height:18px; border-left: 16px solid #2445bd;\">\n",
       "<a class=\"link-to-altmetric-details-tab\" href=\"https://www.altmetric.com/details.php?domain=www.frontiersin.org&amp;citation_id=63734010&amp;tab=facebook\" target=\"_self\">\n",
       "          Facebook (1)\n",
       "      </a>\n",
       "</div>\n",
       "<div class=\"altmetric-embed readers\" style=\"margin-top: 10px;\">\n",
       "<div class=\"altmetric-embed tip_mendeley\" style=\"padding-left: 10px; line-height:18px; border-left: 16px solid #A60000;\">\n",
       "              Mendeley (32)\n",
       "          </div>\n",
       "</div>\n",
       "<div style=\"margin-top: 10px; text-align: center;\">\n",
       "<a class=\"altmetric_details\" href=\"https://www.altmetric.com/details.php?domain=www.frontiersin.org&amp;citation_id=63734010\" target=\"_self\">\n",
       "            See more details\n",
       "        </a>\n",
       "</div>\n",
       "</div></div></div></div></div><script id=\"\" src=\"https://s7.addthis.com/js/300/addthis_widget.js#pubid=ra-535f7e222fc3ca72\" type=\"text/javascript\"></script>\n",
       "<script id=\"\" type=\"text/javascript\">!function(b,e,f,g,a,c,d){b.fbq||(a=b.fbq=function(){a.callMethod?a.callMethod.apply(a,arguments):a.queue.push(arguments)},b._fbq||(b._fbq=a),a.push=a,a.loaded=!0,a.version=\"2.0\",a.queue=[],c=e.createElement(f),c.async=!0,c.src=g,d=e.getElementsByTagName(f)[0],d.parentNode.insertBefore(c,d))}(window,document,\"script\",\"https://connect.facebook.net/en_US/fbevents.js\");fbq(\"init\",\"2098247140231737\");fbq(\"track\",\"PageView\");</script>\n",
       "<noscript><img height=\"1\" src=\"https://www.facebook.com/tr?id=2098247140231737&amp;ev=PageView&amp;noscript=1\" style=\"display:none\" width=\"1\"/></noscript>\n",
       "<script id=\"\" src=\"https://widgets.figshare.com/static/figshare.js\" type=\"text/javascript\"></script>\n",
       "<script id=\"\" src=\"//7236711.collect.igodigital.com/collect.js\" type=\"text/javascript\"></script><script class=\"optanon-category-C0004\" src=\"https://widgets.figshare.com/static/figshare.js\" type=\"text/javascript\">\n",
       "</script><div id=\"onetrust-consent-sdk\"><div class=\"onetrust-pc-dark-filter ot-hide ot-fade-in\"></div><div aria-labelledby=\"ot-pc-title\" aria-modal=\"true\" class=\"otPcCenter ot-hide ot-fade-in\" id=\"onetrust-pc-sdk\" lang=\"en\" role=\"dialog\"><!-- Close Button --><div class=\"ot-pc-header\"><!-- Logo Tag --><div aria-label=\"Company Logo\" class=\"ot-pc-logo\" role=\"img\" style='background-image: url(\"https://cdn.cookielaw.org/logos/e7701c5d-5192-47e9-a4ce-1ea9fe52e7f1/9ada51fd-4243-488d-95e9-4c66173cb4f2/ae9be8b2-209f-4b2c-8812-1e46187618be/logo_frontiers.png\");\n",
       "                    background-position: left'></div><button aria-label=\"Close\" class=\"ot-close-icon\" id=\"close-pc-btn-handler\"></button></div><!-- Close Button --><div class=\"ot-pc-scrollbar\" id=\"ot-pc-content\"><h3 id=\"ot-pc-title\">Privacy Preference Center</h3><div id=\"ot-pc-desc\">Our website uses cookies that are necessary for its operation. Additional cookies are only used with your consent. These cookies are used to store and access information such as the characteristics of your device as well as certain personal data (IP address, navigation usage, geolocation data) and we process them to analyse the traffic on our website in order to provide you a better user experience, evaluate the efficiency of our communications and to personalise content to your interests. Some cookies are placed by third-party companies with which we work   to deliver relevant ads on social media and the internet. Click on the different categories' headings to change your cookie preferences. Click on \"More Information\" if you wish to learn more about how data is collected and shared.  \n",
       "                <br/><a aria-label=\"More information about your privacy, opens in a new tab\" class=\"privacy-notice-link\" href=\"https://www.frontiersin.org/legal/privacy-policy\" rel=\"noopener\" target=\"_blank\">More information</a></div><button id=\"accept-recommended-btn-handler\">Allow all</button><section class=\"ot-sdk-row ot-cat-grp\"><h3 id=\"ot-category-title\"> Manage Consent Preferences</h3><div class=\"ot-accordion-layout ot-cat-item\" data-optanongroupid=\"C0001\"><button aria-controls=\"ot-desc-id-C0001\" aria-expanded=\"false\" aria-labelledby=\"ot-header-id-C0001\" ot-accordion=\"true\"></button><!-- Accordion header --><div class=\"ot-acc-hdr ot-always-active-group\"><div class=\"ot-plus-minus\"><span></span><span></span></div><h4 class=\"ot-cat-header\" id=\"ot-header-id-C0001\">Strictly Necessary Cookies</h4><div class=\"ot-always-active\">Always Active</div></div><!-- accordion detail --><div class=\"ot-acc-grpcntr ot-acc-txt\"><p class=\"ot-acc-grpdesc ot-category-desc\" id=\"ot-desc-id-C0001\">These cookies are necessary for the website to function and cannot be switched off in our systems. They are usually only set in response to actions made by you which amount to a request for services, such as setting your privacy preferences, logging in or filling in forms. You can set your browser to block or alert you about these cookies, but some parts of the site will not then work. These cookies do not store any personally identifiable information.</p></div></div><div class=\"ot-accordion-layout ot-cat-item\" data-optanongroupid=\"C0002\"><button aria-controls=\"ot-desc-id-C0002\" aria-expanded=\"false\" aria-labelledby=\"ot-header-id-C0002\" ot-accordion=\"true\"></button><!-- Accordion header --><div class=\"ot-acc-hdr\"><div class=\"ot-plus-minus\"><span></span><span></span></div><h4 class=\"ot-cat-header\" id=\"ot-header-id-C0002\">Analytics Cookies</h4><div class=\"ot-tgl\"><input aria-checked=\"true\" aria-labelledby=\"ot-header-id-C0002\" checked=\"\" class=\"category-switch-handler\" data-optanongroupid=\"C0002\" id=\"ot-group-id-C0002\" name=\"ot-group-id-C0002\" role=\"switch\" type=\"checkbox\"/> <label class=\"ot-switch\" for=\"ot-group-id-C0002\"><span class=\"ot-switch-nob\"></span> <span class=\"ot-label-txt\">Analytics Cookies</span></label> </div></div><!-- accordion detail --><div class=\"ot-acc-grpcntr ot-acc-txt\"><p class=\"ot-acc-grpdesc ot-category-desc\" id=\"ot-desc-id-C0002\">These cookies allow us to count visits and traffic sources so we can measure and improve the performance of our site. They help us analyse which pages are the most and least popular and see how visitors move around the site.    All information these cookies collect is aggregated and therefore anonymous. </p></div></div><div class=\"ot-accordion-layout ot-cat-item\" data-optanongroupid=\"C0003\"><button aria-controls=\"ot-desc-id-C0003\" aria-expanded=\"false\" aria-labelledby=\"ot-header-id-C0003\" ot-accordion=\"true\"></button><!-- Accordion header --><div class=\"ot-acc-hdr\"><div class=\"ot-plus-minus\"><span></span><span></span></div><h4 class=\"ot-cat-header\" id=\"ot-header-id-C0003\">Functional Cookies</h4><div class=\"ot-tgl\"><input aria-checked=\"true\" aria-labelledby=\"ot-header-id-C0003\" checked=\"\" class=\"category-switch-handler\" data-optanongroupid=\"C0003\" id=\"ot-group-id-C0003\" name=\"ot-group-id-C0003\" role=\"switch\" type=\"checkbox\"/> <label class=\"ot-switch\" for=\"ot-group-id-C0003\"><span class=\"ot-switch-nob\"></span> <span class=\"ot-label-txt\">Functional Cookies</span></label> </div></div><!-- accordion detail --><div class=\"ot-acc-grpcntr ot-acc-txt\"><p class=\"ot-acc-grpdesc ot-category-desc\" id=\"ot-desc-id-C0003\">These cookies enable the website to provide enhanced functionality and personalisation. They may be set by us or by third party providers whose services we have added to our pages. If you do not allow these cookies then some or all of these services may not function properly.</p></div></div><div class=\"ot-accordion-layout ot-cat-item\" data-optanongroupid=\"C0004\"><button aria-controls=\"ot-desc-id-C0004\" aria-expanded=\"false\" aria-labelledby=\"ot-header-id-C0004\" ot-accordion=\"true\"></button><!-- Accordion header --><div class=\"ot-acc-hdr\"><div class=\"ot-plus-minus\"><span></span><span></span></div><h4 class=\"ot-cat-header\" id=\"ot-header-id-C0004\">Advertising Cookies</h4><div class=\"ot-tgl\"><input aria-checked=\"true\" aria-labelledby=\"ot-header-id-C0004\" checked=\"\" class=\"category-switch-handler\" data-optanongroupid=\"C0004\" id=\"ot-group-id-C0004\" name=\"ot-group-id-C0004\" role=\"switch\" type=\"checkbox\"/> <label class=\"ot-switch\" for=\"ot-group-id-C0004\"><span class=\"ot-switch-nob\"></span> <span class=\"ot-label-txt\">Advertising Cookies</span></label> </div></div><!-- accordion detail --><div class=\"ot-acc-grpcntr ot-acc-txt\"><p class=\"ot-acc-grpdesc ot-category-desc\" id=\"ot-desc-id-C0004\">These cookies may be set through our site by our advertising partners. They may be used by those companies to build a profile of your interests and show you relevant adverts on other sites.    They do not store directly personal information, but are based on uniquely identifying your browser and internet device. If you do not allow these cookies, you will experience less targeted advertising.</p></div></div><!-- Groups sections starts --><!-- Group section ends --><!-- Accordion Group section starts --><!-- Accordion Group section ends --></section></div><section class=\"ot-hide ot-hosts-ui ot-pc-scrollbar\" id=\"ot-pc-lst\"><div id=\"ot-pc-hdr\"><h3 id=\"ot-lst-title\"><button aria-label=\"Back\" class=\"ot-link-btn back-btn-handler\"><svg id=\"ot-back-arw\" viewbox=\"0 0 444.531 444.531\" x=\"0px\" xml:space=\"preserve\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" y=\"0px\"><title>Back Button</title><g><path d=\"M213.13,222.409L351.88,83.653c7.05-7.043,10.567-15.657,10.567-25.841c0-10.183-3.518-18.793-10.567-25.835\n",
       "                    l-21.409-21.416C323.432,3.521,314.817,0,304.637,0s-18.791,3.521-25.841,10.561L92.649,196.425\n",
       "                    c-7.044,7.043-10.566,15.656-10.566,25.841s3.521,18.791,10.566,25.837l186.146,185.864c7.05,7.043,15.66,10.564,25.841,10.564\n",
       "                    s18.795-3.521,25.834-10.564l21.409-21.412c7.05-7.039,10.567-15.604,10.567-25.697c0-10.085-3.518-18.746-10.567-25.978\n",
       "                    L213.13,222.409z\" fill=\"#656565\"></path></g></svg></button> <span>Performance Cookies</span></h3><div class=\"ot-lst-subhdr\"><div class=\"ot-search-cntr\"><p class=\"ot-scrn-rdr\" role=\"status\"></p><label class=\"ot-scrn-rdr\" for=\"vendor-search-handler\">Vendor Search</label> <input id=\"vendor-search-handler\" name=\"vendor-search-handler\" placeholder=\"Search...\" type=\"text\"/> <svg viewbox=\"0 -30 110 110\" x=\"0px\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" y=\"0px\"><title>Search Icon</title><path d=\"M55.146,51.887L41.588,37.786c3.486-4.144,5.396-9.358,5.396-14.786c0-12.682-10.318-23-23-23s-23,10.318-23,23\n",
       "            s10.318,23,23,23c4.761,0,9.298-1.436,13.177-4.162l13.661,14.208c0.571,0.593,1.339,0.92,2.162,0.92\n",
       "            c0.779,0,1.518-0.297,2.079-0.837C56.255,54.982,56.293,53.08,55.146,51.887z M23.984,6c9.374,0,17,7.626,17,17s-7.626,17-17,17\n",
       "            s-17-7.626-17-17S14.61,6,23.984,6z\" fill=\"#2e3644\"></path></svg></div><div class=\"ot-fltr-cntr\"><button aria-haspopup=\"true\" aria-label=\"Filter\" id=\"filter-btn-handler\"><svg aria-hidden=\"true\" role=\"presentation\" viewbox=\"0 0 402.577 402.577\" x=\"0px\" xml:space=\"preserve\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" y=\"0px\"><title>Filter Icon</title><g><path d=\"M400.858,11.427c-3.241-7.421-8.85-11.132-16.854-11.136H18.564c-7.993,0-13.61,3.715-16.846,11.136\n",
       "      c-3.234,7.801-1.903,14.467,3.999,19.985l140.757,140.753v138.755c0,4.955,1.809,9.232,5.424,12.854l73.085,73.083\n",
       "      c3.429,3.614,7.71,5.428,12.851,5.428c2.282,0,4.66-0.479,7.135-1.43c7.426-3.238,11.14-8.851,11.14-16.845V172.166L396.861,31.413\n",
       "      C402.765,25.895,404.093,19.231,400.858,11.427z\" fill=\"#fff\"></path></g></svg></button></div><div id=\"ot-anchor\"></div><section id=\"ot-fltr-modal\"><div id=\"ot-fltr-cnt\"><button id=\"clear-filters-handler\">Clear</button><div class=\"ot-fltr-scrlcnt ot-pc-scrollbar\"><div class=\"ot-fltr-opts\"><div class=\"ot-fltr-opt\"><div class=\"ot-chkbox\"><input aria-checked=\"false\" class=\"category-filter-handler\" id=\"chkbox-id\" type=\"checkbox\"/> <label for=\"chkbox-id\"><span class=\"ot-label-txt\">checkbox label</span></label> <span class=\"ot-label-status\">label</span></div></div></div><div class=\"ot-fltr-btns\"><button id=\"filter-apply-handler\">Apply</button> <button id=\"filter-cancel-handler\">Cancel</button></div></div></div></section></div></div><section class=\"ot-host-cnt ot-pc-scrollbar\" id=\"ot-lst-cnt\"><div id=\"ot-sel-blk\"><div class=\"ot-sel-all\"><div class=\"ot-sel-all-hdr\"><span class=\"ot-consent-hdr\">Consent</span> <span class=\"ot-li-hdr\">Leg.Interest</span></div><div class=\"ot-sel-all-chkbox\"><div class=\"ot-chkbox\" id=\"ot-selall-hostcntr\"><input aria-checked=\"false\" id=\"select-all-hosts-groups-handler\" type=\"checkbox\"/> <label for=\"select-all-hosts-groups-handler\"><span class=\"ot-label-txt\">checkbox label</span></label> <span class=\"ot-label-status\">label</span></div><div class=\"ot-chkbox\" id=\"ot-selall-vencntr\"><input aria-checked=\"false\" id=\"select-all-vendor-groups-handler\" type=\"checkbox\"/> <label for=\"select-all-vendor-groups-handler\"><span class=\"ot-label-txt\">checkbox label</span></label> <span class=\"ot-label-status\">label</span></div><div class=\"ot-chkbox\" id=\"ot-selall-licntr\"><input aria-checked=\"false\" id=\"select-all-vendor-leg-handler\" type=\"checkbox\"/> <label for=\"select-all-vendor-leg-handler\"><span class=\"ot-label-txt\">checkbox label</span></label> <span class=\"ot-label-status\">label</span></div></div></div></div><div class=\"ot-sdk-row\"><div class=\"ot-sdk-column\"></div></div></section></section><div class=\"ot-pc-footer\"><div class=\"ot-btn-container\"> <button class=\"save-preference-btn-handler onetrust-close-btn-handler\">Confirm My Choices</button></div><!-- Footer logo --><div class=\"ot-pc-footer-logo\"><a aria-label=\"Powered by OneTrust Opens in a new Tab\" href=\"https://www.onetrust.com/products/cookie-consent/\" rel=\"noopener noreferrer\" style='background-image: url(\"https://cdn.cookielaw.org/logos/static/poweredBy_ot_logo.svg\")' target=\"_blank\"></a></div></div><!-- Cookie subgroup container --><!-- Vendor list link --><!-- Cookie lost link --><!-- Toggle HTML element --><!-- Checkbox HTML --><!-- plus minus--><!-- Arrow SVG element --><!-- Accordion basic element --><span aria-atomic=\"true\" aria-live=\"polite\" class=\"ot-scrn-rdr\"></span><iframe aria-hidden=\"true\" class=\"ot-text-resize\" style=\"position:absolute;top:-50000px;width:100em;\" title=\"onetrust-text-resize\"></iframe></div><div class=\"otFlat bottom vertical-align-content\" id=\"onetrust-banner-sdk\" style=\"bottom: -99px; animation-name: slide-down-custom; animation-duration: 1000ms; animation-fill-mode: forwards; animation-timing-function: ease-out;\"><div aria-describedby=\"onetrust-policy-text\" aria-labelledby=\"onetrust-policy-title\" role=\"dialog\"><div class=\"ot-sdk-container\"><div class=\"ot-sdk-row\"><div class=\"ot-sdk-eight ot-sdk-columns\" id=\"onetrust-group-container\"><div class=\"banner_logo\"></div><div id=\"onetrust-policy\"><h3 id=\"onetrust-policy-title\">We use cookies</h3><p id=\"onetrust-policy-text\">Our website uses cookies that are necessary for its operation and to improve your experience. You can review and control your cookies by clicking on \"Accept Cookies\" or on \"Cookies Settings\".</p></div></div><div class=\"ot-sdk-three ot-sdk-columns\" id=\"onetrust-button-group-parent\"><div id=\"onetrust-button-group\"><button class=\"cookie-setting-link\" id=\"onetrust-pc-btn-handler\">Cookies Settings</button> <button id=\"onetrust-accept-btn-handler\">Accept Cookies</button></div></div></div></div><!-- Close Button --><div id=\"onetrust-close-btn-container\"><button aria-label=\"Close\" class=\"onetrust-close-btn-handler onetrust-close-btn-ui banner-close-button ot-close-icon\"></button></div><!-- Close Button END--></div></div></div><img height=\"1\" src=\"https://t.co/i/adsct?bci=3&amp;eci=2&amp;event_id=ba1bc512-78ef-41b5-bbbf-509a432455ae&amp;events=%5B%5B%22pageview%22%2C%7B%7D%5D%5D&amp;integration=advertiser&amp;p_id=Twitter&amp;p_user_id=0&amp;pl_id=e7e34631-c8de-41af-bd3a-afa69c5c59cc&amp;tw_document_href=https%3A%2F%2Fwww.frontiersin.org%2Farticles%2F10.3389%2Ffpls.2019.00932%2Ffull&amp;tw_iframe_status=0&amp;tw_order_quantity=0&amp;tw_sale_amount=0&amp;txn_id=o0l2c&amp;type=javascript&amp;version=2.3.29\" style=\"display: none;\" width=\"1\"/><img height=\"1\" src=\"https://analytics.twitter.com/i/adsct?bci=3&amp;eci=2&amp;event_id=ba1bc512-78ef-41b5-bbbf-509a432455ae&amp;events=%5B%5B%22pageview%22%2C%7B%7D%5D%5D&amp;integration=advertiser&amp;p_id=Twitter&amp;p_user_id=0&amp;pl_id=e7e34631-c8de-41af-bd3a-afa69c5c59cc&amp;tw_document_href=https%3A%2F%2Fwww.frontiersin.org%2Farticles%2F10.3389%2Ffpls.2019.00932%2Ffull&amp;tw_iframe_status=0&amp;tw_order_quantity=0&amp;tw_sale_amount=0&amp;txn_id=o0l2c&amp;type=javascript&amp;version=2.3.29\" style=\"display: none;\" width=\"1\"/><div id=\"_atssh\" style=\"visibility: hidden; height: 1px; width: 1px; position: absolute; top: -9999px; z-index: 100000;\"><iframe id=\"_atssh220\" src=\"https://s7.addthis.com/static/sh.f48a1a04fe8dbf021b4cda1d.html#rand=0.7940676074200392&amp;iit=1671056252853&amp;tmr=load%3D1671056252177%26core%3D1671056252214%26main%3D1671056252848%26ifr%3D1671056252857&amp;cb=0&amp;cdn=0&amp;md=0&amp;kw=Abies%20balsamea%2Cboreal%20forest%2CEctomycorrhiza%20(EcM)%2CEricaceous%20shrubs%2Cfacilitation%2CPicea%20mariana%2CPopulus%20tremuloides&amp;ab=-&amp;dh=www.frontiersin.org&amp;dr=&amp;du=https%3A%2F%2Fwww.frontiersin.org%2Farticles%2F10.3389%2Ffpls.2019.00932%2Ffull&amp;href=https%3A%2F%2Fwww.frontiersin.org%2Farticles%2F10.3389%2Ffpls.2019.00932%2Ffull&amp;dt=Facilitation%20of%20Balsam%20Fir%20by%20Trembling%20Aspen%20in%20the%20Boreal%20Forest%3A%20Do%20Ectomycorrhizal%20Communities%20Matter%3F&amp;dbg=0&amp;cap=tc%3D0%26ab%3D0&amp;inst=1&amp;jsl=8193&amp;prod=undefined&amp;lng=en&amp;ogt=image%2Curl%2Cdescription%2Ctitle%2Csite_name%2Ctype%3Darticle&amp;pc=men&amp;pub=ra-535f7e222fc3ca72&amp;ssl=1&amp;sid=639a4b7c7eb0dc80&amp;srf=0.01&amp;ver=300&amp;xck=0&amp;xtr=0&amp;og=type%3Darticle%26site_name%3DFrontiers%26title%3DFacilitation%2520of%2520Balsam%2520Fir%2520by%2520Trembling%2520Aspen%2520in%2520the%2520Boreal%2520Forest%253A%2520Do%2520Ectomycorrhizal%2520Communities%2520Matter%253F%26description%3DSuccession%2520is%2520generally%2520well%2520described%2520above-ground%2520in%2520the%2520boreal%2520forest%252C%2520and%2520several%2520studies%2520have%2520demonstrated%2520the%2520role%2520of%2520interspecific%2520facilitation%2520in%2520tree%2520species%2520establishment.%2520However%252C%2520the%2520role%2520of%2520mycorrhizal%2520communities%2520for%2520tree%2520establishment%2520and%2520interspecific%2520facilitation%252C%2520has%2520been%2520little%2520explored.%2520At%2520the%2520ecotone%2520between%2520the%2520mixed%2520boreal%2520forest%252C%2520dominated%2520by%2520balsam%2520fir%2520and%2520hardwood%2520species%252C%2520and%2520the%2520boreal%2520forest%252C%2520dominated%2520by%2520black%2520spruce%252C%2520several%2520stands%2520of%2520trembling%2520aspen%2520can%2520be%2520found%252C%2520surrounded%2520by%2520black%2520spruce%2520forest.%2520Regeneration%2520of%2520balsam%2520fir%2520seems%2520to%2520have%2520increased%2520in%2520the%2520recent%2520decades%2520within%2520the%2520boreal%2520forest%252C%2520and%2520it%2520seems%2520better%2520adapted%2520to%2520grow%2520in%2520trembling%2520aspen%2520stands%2520than%2520in%2520black%2520spruce%2520stands%252C%2520even%2520when%2520located%2520in%2520similar%2520abiotic%2520conditions.%2520As%2520black%2520spruce%2520stands%2520are%2520also%2520covered%2520by%2520ericaceous%2520shrubs%252C%2520we%2520investigated%2520if%2520differences%2520in%2520soil%2520fungal%2520communities%2520and%2520ericaceous%2520shrubs%2520abundance%2520could%2520explain%2520the%2520differences%2520observed%2520in%2520balsam%2520fir%2520growth%2520and%2520nutrition.%2520We%2520conducted%2520a%2520study%2520centered%2520on%2520individual%2520saplings%2520to%2520link%2520growth%2520and%2520foliar%2520nutrient%2520concentrations%2520to%2520local%2520vegetation%2520cover%252C%2520mycorrhization%2520rate%252C%2520and%2520mycorrhizal%2520communities%2520associated%2520with%2520balsam%2520fir%2520roots.%2520We%2520found%2520that%2520foliar%2520nutrient%2520concentrations%2520and%2520ramification%2520indices%2520(colonization%2520by%2520mycorrhiza%2520per%2520length%2520of%2520root)%2520were%2520greater%2520in%2520trembling%2520aspen%2520stands%2520and%2520were%2520positively%2520correlated%2520to%2520apical%2520and%2520lateral%2520growth%2520of%2520balsam%2520fir%2520saplings.%2520In%2520black%2520spruce%2520stands%252C%2520the%2520presence%2520of%2520er...%26url%3Dhttps%253A%252F%252Fwww.frontiersin.org%252Farticles%252F10.3389%252Ffpls.2019.00932%252Ffull%26image%3Dhttps%253A%252F%252Fwww.frontiersin.org%252Ffiles%252FMyHome%252520Article%252520Library%252F464100%252F464100_Thumb_400.jpg&amp;csi=undefined&amp;rev=v8.28.8-wp&amp;ct=1&amp;xld=1&amp;xd=1\" style=\"height: 1px; width: 1px; position: absolute; top: 0px; z-index: 100000; border: 0px; left: 0px;\" title=\"AddThis utility frame\"></iframe></div><style id=\"service-icons-0\"></style>\n",
       "<script id=\"\" type=\"text/javascript\">(function(){var a=a||{};a.data_track_addressbar=!1;a.data_track_clickback=!1;!$||$(\".page-container .side-article-share\").css(\"display\",\"flex\");!$||$(\".container-media\").css(\"display\",\"flex\")})();</script>\n",
       "<script id=\"\" type=\"text/javascript\">function getUrlParameter(a){a=a.replace(/[\\[]/,\"\\\\[\").replace(/[\\]]/,\"\\\\]\");a=new RegExp(\"[\\\\?\\x26]\"+a+\"\\x3d([^\\x26#]*)\");a=a.exec(location.search);return null===a?\"\":decodeURIComponent(a[1].replace(/\\+/g,\" \"))}var sfmc_sub=getUrlParameter(\"sfmc_sub\");null!=sfmc_sub&&\"\"!=sfmc_sub&&_etmc.push([\"setUserInfo\",{email:sfmc_sub}]);_etmc.push([\"setOrgId\",\"7236711\"]);_etmc.push([\"trackPageView\"]);</script><div style=\"display: none;\"><img alt=\"\" aria-hidden=\"true\" height=\"0\" src=\"https://nova.collect.igodigital.com/c2/7236711/track_page_view?payload=%7B%22title%22%3A%22Frontiers%20%7C%20Facilitation%20of%20Balsam%20Fir%20by%20Trembling%20Aspen%20in%20the%20Boreal%20Forest%3A%20Do%20Ectomycorrhizal%20Communities%20Matter%3F%22%2C%22url%22%3A%22https%3A%2F%2Fwww.frontiersin.org%2Farticles%2F10.3389%2Ffpls.2019.00932%2Ffull%22%2C%22referrer%22%3A%22%22%7D\" style=\"display: none;\" title=\"blank image\" width=\"0\"/></div>\n",
       "</body></html>"
      ]
     },
     "execution_count": 600,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 597,
   "id": "b7e79f9f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<section class=\"abstract\"><p class=\"chapter-para\">Mating system characteristics are of great importance as they may influence male and female reproductive success and reproductive isolation. The wood turtle (<em>Glyptemys insculpta</em>) is a terrestrial freshwater species listed as endangered by the International Union for Conservation of Nature. Considering its conservation status and the paucity of information currently available on parentage relationship for the species, we performed a microsatellite analysis to study the mating system of wood turtles in the Shawinigan River (Québec). We sampled 38 clutches over 2 years (14 in 2006 and 24 in 2007), for a total of 248 offspring genotyped with 7 microsatellite loci. The reconstructed genotypes of the fathers revealed that reproductive success in the sampled clutches varied greatly between males and are positively correlated with the number of mates and clutches sired. Frequency of multiple paternity was estimated at 37% through a consensus of 3 different estimation methods. Positive correlation was observed between the genetic diversity of clutches and the number of fathers. Repeat paternity, however, was observed in 88% of the clutches by the same female in successive years, which suggests either a frequent use of sperm storage, or remating with the same partner in successive years.</p></section>]"
      ]
     },
     "execution_count": 597,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bs.find_all(\"section\", class_=\"abstract\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 477,
   "id": "4a855b0c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:cliff.api:Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\ProgramData\\Anaconda3\\lib\\site-packages\\requests\\models.py\", line 971, in json\n",
      "    return complexjson.loads(self.text, **kwargs)\n",
      "  File \"C:\\ProgramData\\Anaconda3\\lib\\json\\__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"C:\\ProgramData\\Anaconda3\\lib\\json\\decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"C:\\ProgramData\\Anaconda3\\lib\\json\\decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\ProgramData\\Anaconda3\\lib\\site-packages\\cliff\\api.py\", line 65, in _query\n",
      "    return r.json()\n",
      "  File \"C:\\ProgramData\\Anaconda3\\lib\\site-packages\\requests\\models.py\", line 975, in json\n",
      "    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)\n",
      "requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "''"
      ]
     },
     "execution_count": 477,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import logging\n",
    "logging.basicConfig()\n",
    "\n",
    "from cliff.api import Cliff\n",
    "\n",
    "from pandas.io.json import json_normalize\n",
    "\n",
    "my_cliff = Cliff('http://localhost:8888')\n",
    "my_cliff.parse_text(\"This is about Einstien at the IIT in New Delhi.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 363,
   "id": "00bf3424",
   "metadata": {},
   "outputs": [],
   "source": [
    "URL = \"https://doi.org/10.1002/ece3.3947\"\n",
    "dr.get(URL)\n",
    "bs = BeautifulSoup(dr.page_source,\"html.parser\")\n",
    "title = bs.find('title')\n",
    "title = title.text\n",
    "\n",
    "section_abstract = [bs.find_all(\"div\", class_=\"article-section__content en main\"),\n",
    "                   bs.find_all(\"div\", class_=\"c-article-section__content\", attrs={'id' :\"Abs1-content\"})]\n",
    "for res in section_abstract:\n",
    "    if (len(res)>0) :\n",
    "        abstract = res[0].text\n",
    "\n",
    "if \"introduction\" in bs.text:\n",
    "    complete = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "id": "399428cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "from io import StringIO\n",
    "from html.parser import HTMLParser\n",
    "\n",
    "class MLStripper(HTMLParser):\n",
    "    def __init__(self):\n",
    "        super().__init__()\n",
    "        self.reset()\n",
    "        self.strict = False\n",
    "        self.convert_charrefs= True\n",
    "        self.text = StringIO()\n",
    "    def handle_data(self, d):\n",
    "        self.text.write(d)\n",
    "    def get_data(self):\n",
    "        return self.text.getvalue()\n",
    "\n",
    "def strip_tags(html):\n",
    "    s = MLStripper()\n",
    "    s.feed(html)\n",
    "    return s.get_data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "id": "4b44f5c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "for job_element in results:\n",
    "    results2 = soup.find_all(\"div\", class_=\"ArticleContentText\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "id": "0a79c442",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<div class=\"c-article-body\">\n",
      "<section aria-labelledby=\"Abs1\" data-title=\"Abstract\" lang=\"en\"><div class=\"c-article-section\" id=\"Abs1-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Abs1\">Abstract</h2><div class=\"c-article-section__content\" id=\"Abs1-content\"><p>Quantitative genetics approaches, and particularly animal models, are widely used to assess the genetic (co)variance of key fitness related traits and infer adaptive potential of wild populations. Despite the importance of precision and accuracy of genetic variance estimates and their potential sensitivity to various ecological and population specific factors, their reliability is rarely tested explicitly. Here, we used simulations and empirical data collected from an 11-year study on tree swallow (<i>Tachycineta bicolor</i>), a species showing a high rate of extra-pair paternity and a low recruitment rate, to assess the importance of identity errors, structure and size of the pedigree on quantitative genetic estimates in our dataset. Our simulations revealed an important lack of precision in heritability and genetic-correlation estimates for most traits, a low power to detect significant effects and important identifiability problems. We also observed a large bias in heritability estimates when using the social pedigree instead of the genetic one (deflated heritabilities) or when not accounting for an important cause of resemblance among individuals (for example, permanent environment or brood effect) in model parameterizations for some traits (inflated heritabilities). We discuss the causes underlying the low reliability observed here and why they are also likely to occur in other study systems. Altogether, our results re-emphasize the difficulties of generalizing quantitative genetic estimates reliably from one study system to another and the importance of reporting simulation analyses to evaluate these important issues.</p></div></div></section>\n",
      "<noscript>\n",
      "</noscript>\n",
      "<div class=\"main-content\">\n",
      "<section data-title=\"Introduction\"><div class=\"c-article-section\" id=\"Sec1-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec1\">Introduction</h2><div class=\"c-article-section__content\" id=\"Sec1-content\"><p>Understanding the genetic component underlying important fitness related traits is essential to infer the adaptive potential of wild populations (<a aria-label=\"Reference Lynch and Walsh, 1998\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR33\" id=\"ref-link-section-d28693445e365\" title=\"Lynch M, Walsh B . (1998) Genetics and Analysis of Quantitative Traits. Sinauer Associates: Sunderland, MA, USA.\">Lynch and Walsh, 1998</a>; <a aria-label=\"Reference Hendry et al., 2011\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR16\" id=\"ref-link-section-d28693445e368\" title=\"Hendry AP, Kinnison MT, Heino M, Day T, Smith TB, Fitt G et al. (2011). Evolutionary principles and their practical application. Evol Appl 4: 159–183.\">Hendry et al., 2011</a>). However, as most traits are likely under the control of numerous genes with small effects (<a aria-label=\"Reference Lande, 1981\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR28\" id=\"ref-link-section-d28693445e371\" title=\"Lande R . (1981). The minimum number of genes contributing to quantitative variation between and within populations. Genetics 99: 541–553.\">Lande, 1981</a>; <a aria-label=\"Reference Falconer and Mackay, 1996\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR8\" id=\"ref-link-section-d28693445e374\" title=\"Falconer DS, Mackay TFC . (1996) Introduction to Quantitative Genetics, 4th edn. Pearson: Essex, UK.\">Falconer and Mackay, 1996</a>; <a aria-label=\"Reference Husby et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR21\" id=\"ref-link-section-d28693445e377\" title=\"Husby A, Kawakami T, Rönnegård L, Smeds L, Ellegren H, Qvarnström A . (2015). Genome-wide association mapping in a wild avian population identifies a link between genetic and phenotypic variation in a life-history trait. Proc R Soc B 282: 20150156.\">Husby et al., 2015</a>), potentially interacting with the genetic components of other traits (<a aria-label=\"Reference Lande and Arnold, 1983\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR29\" id=\"ref-link-section-d28693445e381\" title=\"Lande R, Arnold SJ . (1983). The measurement of selection on correlated characters. Evolution 37: 1210–1226.\">Lande and Arnold, 1983</a>; <a aria-label=\"Reference Blows and Hoffmann, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR3\" id=\"ref-link-section-d28693445e384\" title=\"Blows MW, Hoffmann AA . (2005). A reassessment of genetic limits to evoutionary change. Ecology 86: 1371–1384.\">Blows and Hoffmann, 2005</a>) and/or with the environment in which they are expressed (<a aria-label=\"Reference Via and Lande, 1985\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR42\" id=\"ref-link-section-d28693445e387\" title=\"Via S, Lande R . (1985). Genotype-environment interaction and the evolution of phenotypic plasticity. Evolution 39: 505–522.\">Via and Lande, 1985</a>; <a aria-label=\"Reference Hoffmann and Merilä, 1999\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR17\" id=\"ref-link-section-d28693445e390\" title=\"Hoffmann AA, Merilä J . (1999). Heritable variation and evolution under favourable and unfavourable conditions. Trends Ecol Evol 14: 96–101.\">Hoffmann and Merilä, 1999</a>), this task remains challenging despite rapid advances in whole-genome analysis methods (<a aria-label=\"Reference Vinkhuyzen et al., 2013\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR44\" id=\"ref-link-section-d28693445e393\" title=\"Vinkhuyzen AAE, Wray NR, Yang J, Goddard ME, Visscher PM . (2013). Estimation and partition of heritability in human populations using whole-genome analysis methods. Annu Rev Genet 47: 75–95.\">Vinkhuyzen et al., 2013</a>). For these reasons, quantitative genetic approaches—statistical methods using known relationships between individuals to assess the genetic and environmental components of a population phenotypic variance—remain efficient ways to infer the underlying genetic variation of focal traits (<a aria-label=\"Reference Falconer and Mackay, 1996\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR8\" id=\"ref-link-section-d28693445e396\" title=\"Falconer DS, Mackay TFC . (1996) Introduction to Quantitative Genetics, 4th edn. Pearson: Essex, UK.\">Falconer and Mackay, 1996</a>; <a aria-label=\"Reference Kruuk et al., 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR26\" id=\"ref-link-section-d28693445e400\" title=\"Kruuk LEB, Slate J, Wilson AJ . (2008). New answers for old questions: the evolutionary quantitative genetics of wild animal populations. Annu Rev Ecol Evol Syst 39: 525–548.\">Kruuk et al., 2008</a>). For this purpose, animal models are particularly suitable for wild population datasets because they allow the use of all relationships between individuals and can account for missing data, unbalanced designs and other potential biases (for example, environmental causes of phenotypic similarity among individuals rather than genetic ones) (<a aria-label=\"Reference Henderson, 1984\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR15\" id=\"ref-link-section-d28693445e403\" title=\"Henderson CR . (1984) Applications of Linear Models in Animal Breeding. University of Guelph: Guelph, ON, Canada.\">Henderson, 1984</a>; <a aria-label=\"Reference Kruuk, 2004\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR24\" id=\"ref-link-section-d28693445e406\" title=\"Kruuk LEB . (2004). Estimating genetic parameters in natural populations using the ‘animal model’. Philos Trans R Soc B 359: 873–890.\">Kruuk, 2004</a>; <a aria-label=\"Reference Kruuk and Hadfield, 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d28693445e409\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890–1903.\">Kruuk and Hadfield, 2007</a>; <a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e412\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). Moreover, heritability estimates from wild populations obtained with animal models are generally more accurate than those estimated with traditional parent–offspring regressions (reviewed in <a aria-label=\"Reference Postma, 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR39\" id=\"ref-link-section-d28693445e415\" title=\"Postma E . (2014). Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates? In: Charmantier A, Garant D et al. (eds) Quantitative Genetics in the Wild. Oxford University Press, Oxford, UK, pp 16–33.\">Postma, 2014</a>).</p><p>Several factors will influence the validity of the genetic variance estimates obtained in the wild, which in turn will affect our capacity to infer population responses to selective pressures (for example, through breeder’s or Robertson–Price equation, see <a aria-label=\"Reference Morrissey et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR35\" id=\"ref-link-section-d28693445e421\" title=\"Morrissey MB, Kruuk LEB, Wilson AJ . (2010). The danger of applying the breeder’s equation in observational studies of natural populations. J Evol Biol 23: 2277–2288.\">Morrissey et al., 2010</a>). Precision—the repeatability of a measurement—is highly dependent on sample size both in terms of pedigree depth and completeness, but also on the complexity of the genetic architecture underlying the focal trait (<a aria-label=\"Reference Morrissey et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d28693445e424\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309–2321.\">Morrissey et al., 2007</a>; <a aria-label=\"Reference de Villemereuil et al., 2013\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR43\" id=\"ref-link-section-d28693445e427\" title=\"de Villemereuil P, Gimenez O, Doligez B . (2013). Comparing parent–offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits. Methods Ecol Evol 4: 260–275.\">de Villemereuil et al., 2013</a>). For example, given two pedigrees with identical sample sizes, a pedigree with a weak degree of connectivity between individuals, for example, in populations with strong natal dispersion and/or immigration resulting in few related individuals, will be less precise than a pedigree with known relationships between all individuals (<a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e430\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). However, precision is rarely tested <i>per se</i>, but rather deduced from the size of standard errors around estimates (for example, <a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e437\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>). Accuracy—the proximity between the estimated value and the true value—is affected mainly by pedigree errors (that is, erroneous links between individuals in the pedigree; <a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e440\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>; <a aria-label=\"Reference Morrissey et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d28693445e443\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309–2321.\">Morrissey et al., 2007</a>; <a aria-label=\"Reference Firth et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d28693445e446\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336–1344.\">Firth et al., 2015</a>) and by model parameterizations (<a aria-label=\"Reference Kruuk and Hadfield, 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d28693445e449\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890–1903.\">Kruuk and Hadfield, 2007</a>; <a aria-label=\"Reference Wilson, 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR46\" id=\"ref-link-section-d28693445e452\" title=\"Wilson AJ . (2008). Why h2 does not always equal VA/VP? J Evol Biol 21: 647–650.\">Wilson, 2008</a>; <a aria-label=\"Reference Wolak et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR51\" id=\"ref-link-section-d28693445e456\" title=\"Wolak ME, Roff DA, Fairbairn DJ . (2015). Are we underestimating the genetic variances of dimorphic traits? Ecol Evol 5: 590–597.\">Wolak et al., 2015</a>). Pedigree errors could be particularly common in study systems where extra-pair copulations are important and parental links in pedigrees constructed solely based on social pair observations (that is, social pedigree). Theoretically, the presence of extra-pair paternity (EPP), if not accounted for, can bias downward the heritability estimates, even though evidence so far suggests that this bias is generally small in wild populations (<a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e459\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>; <a aria-label=\"Reference Bérénos et al., 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR2\" id=\"ref-link-section-d28693445e462\" title=\"Bérénos C, Ellis PA, Pilkington JG, Pemberton JM . (2014). Estimating quantitative genetic parameters in wild populations: a comparison of pedigree and genomic approaches. Mol Ecol 23: 3434–3451.\">Bérénos et al., 2014</a>; <a aria-label=\"Reference Firth et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d28693445e465\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336–1344.\">Firth et al., 2015</a>; but see <a aria-label=\"Reference Lee and Pollak, 1997\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR30\" id=\"ref-link-section-d28693445e468\" title=\"Lee C, Pollak EJ . (1997). Influence of sire misidentification on sire x year interaction variance and direct-maternal genetic covariance for weaning weight in beef cattle. J Anim Sci 75: 2858–2863.\">Lee and Pollak, 1997</a> for reports of bias in the animal breeding litterature; see also <a aria-label=\"Reference Walther and Moore, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR45\" id=\"ref-link-section-d28693445e471\" title=\"Walther BA, Moore JL . (2005). The concepts of bias, precision and accuracy, and their use in testing the performance of species richness estimators, with a literature review of estimator performance. Ecography 28: 815–829.\">Walther and Moore, 2005</a> for a detailed discussion on the concepts of bias, precision and accuracy).</p><p>Despite previous knowledge of what can affect precision and accuracy of genetic variance estimates, extrapolating the reliability of these estimates from one study system to another is difficult given the large diversity of life history observed across species that inevitably modified dataset and pedigree structures. Therefore, simulation analyses are recommended for testing the limits of a particular dataset, pedigree and model to answer specific biological questions (<a aria-label=\"Reference Morrissey et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d28693445e477\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309–2321.\">Morrissey et al., 2007</a>; <a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e480\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). Simulation frameworks allow the assessment of power—the probability of detecting an effect given that this effect is true—of a particular dataset or model. In animal models, power is affected by the same factors as precision (<a aria-label=\"Reference Quinn et al., 2006\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR40\" id=\"ref-link-section-d28693445e483\" title=\"Quinn JL, Charmantier A, Garant D, Sheldon BC . (2006). Data depth, data completeness, and their influence on quantitative genetic estimation in two contrasting bird populations. J Evol Biol 19: 994–1002.\">Quinn et al., 2006</a>; <a aria-label=\"Reference Morrissey et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d28693445e486\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309–2321.\">Morrissey et al., 2007</a>; <a aria-label=\"Reference Bérénos et al., 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR2\" id=\"ref-link-section-d28693445e489\" title=\"Bérénos C, Ellis PA, Pilkington JG, Pemberton JM . (2014). Estimating quantitative genetic parameters in wild populations: a comparison of pedigree and genomic approaches. Mol Ecol 23: 3434–3451.\">Bérénos et al., 2014</a>). However, performing a power analysis alone could provide an incomplete picture of the validity of genetic variance estimates. In fact, a particular model applied on a dataset could have a high power to estimate a given effect, but the model itself might be missing a critical variance term, leading to precise but inaccurate estimates (<a aria-label=\"Reference Kruuk and Hadfield, 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d28693445e493\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890–1903.\">Kruuk and Hadfield, 2007</a>). Similarly, even if all important variance terms are included in a particular model, the model variance structure might not allow discrimination between two variance terms being included, a problem of statistical models related to their low ‘identifiability’ (<a aria-label=\"Reference Bolker, 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR4\" id=\"ref-link-section-d28693445e496\" title=\"Bolker B . (2008) Ecological Models and Data in R. Princeton, University Princeton, NJ, USA.\">Bolker, 2008</a>). Confounded parameters could be frequent when applying animal models to wild populations (<a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e499\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>), but the extent of identifiability problems is still unknown. Finally, reliability of quantitative genetics estimates is even less reported for multivariate models than univariate models, limiting our confidence on important parameter estimations such as genetic additive covariance/correlation among traits.</p><p>Here, we used an 11-year study on Tree swallow (<i>Tachycineta bicolor</i> (Vieillot, 1808)) to assess the effects of identity errors, structure and size of a pedigree on reliability of quantitative genetic estimates. Tree swallows are small migratory passerines and their breeding grounds are widely distributed across North America (<a aria-label=\"Reference Winkler et al., 2011\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR49\" id=\"ref-link-section-d28693445e508\" title=\"Winkler D, Hallinger K, Ardia D, Robertson R, Stutchbury B, Cohen R . (2011). Tree swallow (Tachycineta bicolor. Birds North Am Online. \n",
      "                  http://bna.birds.cornelledu/bna/\n",
      "                  \n",
      "                . Accessed on 1 Oct 2013.\">Winkler et al., 2011</a>). Similarly to other migrating species, tree swallow shows high natal dispersal, high mortality within the first year and low nestling recruitment rate (<a aria-label=\"Reference Hosner and Winkler, 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR18\" id=\"ref-link-section-d28693445e511\" title=\"Hosner PA, Winkler DW . (2007). Dispersal distances of tree swallows estimated from continent-wide and limited-area data. J Field Ornithol 78: 290–297.\">Hosner and Winkler, 2007</a>), which reduce kinship among individuals in monitored populations. This socially monogamous species also displays one of the highest rate of EPP documented in birds (<a aria-label=\"Reference Griffith et al., 2002\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR13\" id=\"ref-link-section-d28693445e514\" title=\"Griffith SC, Owens IPF, Thuman KA . (2002). Extra pair paternity in birds: a review of interspecific variation and adaptive function. Mol Ecol 11: 2195–2212.\">Griffith et al., 2002</a>), with more than 80% of nests containing at least one extra-pair offspring and overall around 50% of nestlings resulting from extra-pair copulations (<a aria-label=\"Reference Dunn et al., 2001\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR7\" id=\"ref-link-section-d28693445e517\" title=\"Dunn PO, Whittingham LA, Pitcher TE . (2001). Mating systems, sperm competition, and the evolution of sexual dimorphism in birds. Evolution 55: 161–175.\">Dunn et al., 2001</a>; <a aria-label=\"Reference Lessard et al., 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR32\" id=\"ref-link-section-d28693445e521\" title=\"Lessard A, Bourret A, Bélisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor. Behav Ecol Sociobiol 68: 733–742.\">Lessard et al., 2014</a>). In the context of building a pedigree for quantitative genetics analyses, this means that half of paternal links would be erroneous if using a social pedigree. This represents a higher proportion of EPP than those tested in previous studies to assess their impacts on heritability estimates (for example, in <a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e524\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>; <a aria-label=\"Reference Firth et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d28693445e527\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336–1344.\">Firth et al., 2015</a>), but it is a proportion that will be typically found in other similar study systems (see <a aria-label=\"Reference Griffith et al., 2002\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR13\" id=\"ref-link-section-d28693445e530\" title=\"Griffith SC, Owens IPF, Thuman KA . (2002). Extra pair paternity in birds: a review of interspecific variation and adaptive function. Mol Ecol 11: 2195–2212.\">Griffith et al., 2002</a> and references therein). Furthermore, reliability of genetic variance estimates for this and similar species is difficult to predict based on previous results obtained on other species, as the typical poorly connected pedigree available for migratory species could be compensated by the natural half-sibs design resulting from the high rate of EPP occurring in tree swallows.</p><p>In this study, we used a social and a genetic pedigree to estimate and compare genetic additive (co-)variances and heritability of traits differing in their completeness through the pedigree structure. More specifically, we first used empirical data to assess the bias resulting from using a social pedigree over a genetic one. Then, we used simulated data to assess precision and accuracy of quantitative genetic estimates obtained with both pedigrees, as well as power of datasets and models, and identifiability among variance terms.</p></div></div></section><section data-title=\"Materials and methods\"><div class=\"c-article-section\" id=\"Sec2-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec2\">Materials and methods</h2><div class=\"c-article-section__content\" id=\"Sec2-content\"><h3 class=\"c-article__sub-heading\" id=\"Sec3\">Study species, system and phenotypic data</h3><p>Tree swallows breed in tree cavities or nest boxes, they produce only one clutch per year, containing on average five eggs and both parents provide care to nestlings (<a aria-label=\"Reference Winkler et al., 2011\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR49\" id=\"ref-link-section-d28693445e549\" title=\"Winkler D, Hallinger K, Ardia D, Robertson R, Stutchbury B, Cohen R . (2011). Tree swallow (Tachycineta bicolor. Birds North Am Online. \n",
      "                  http://bna.birds.cornelledu/bna/\n",
      "                  \n",
      "                . Accessed on 1 Oct 2013.\">Winkler et al., 2011</a>). Since 2004, we have intensively followed tree swallow activities during the breeding season at 400 nest boxes equally distributed across 40 farms in an area covering 10 200 km<sup>2</sup> in southern Québec, Canada (detailed in <a aria-label=\"Reference Ghilain and Bélisle, 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR11\" id=\"ref-link-section-d28693445e554\" title=\"Ghilain A, Bélisle M . (2008). Breeding success of tree swallows along a gradient of agricultural intensification. Ecol Appl 18: 1140–1154.\">Ghilain and Bélisle, 2008</a>). Nest boxes were visited every 2 days from May to July to record nest box occupancy and important brood characteristics (for example, laying date, clutch size and hatching date). Adults and nestlings were individually marked with an aluminum band (US Fish and Wildlife Service). Adults were captured directly in their nest box by a trap system, during the incubation period and food provisioning period for females and males, respectively. Morphological measurements of body mass (±0.01 g), non-flattened wing length (±0.5 mm) and tarsus length (±0.02 mm) were taken on adult tree swallows during captures. Females were classified based on their plumage color as second-year (brown) or after-second-year (blue-green) (<a aria-label=\"Reference Hussell, 1983\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR22\" id=\"ref-link-section-d28693445e557\" title=\"Hussell DJT . (1983). Age and plumage color in female tree swallows. J Field Ornithol 54: 312–318.\">Hussell, 1983</a>) and a minimal age was determined for all adults based on the year they were first observed in the study system. Nestlings were captured at 16 days old before they fledged (fledging occurs around 18–22 days) to record body mass (±0.01 g), primary length (hereafter wing length; ±0.02 mm) and tarsus length (±0.02 mm). Since 2006, blood samples of adults and nestlings have been collected on a qualitative P8 grade filter paper (Thermo Fisher Scientific, Waltham, MA, USA) for molecular analysis (see below).</p><p>To reflect the large differences that can exist between traits in terms of sample size and amount of standing genetic variance, we focused on nine phenotypic traits grouped in three categories: (1) morphological traits, which included wing length, body mass and tarsus length of all adults; (2) reproductive traits, which were restricted to females, and included laying date (that is, date of the first egg laid), clutch size (that is, number of eggs laid within a nest box) and incubation duration (defined as hatching date−(laying date+clutch size−1)); and (3) nestling traits, which included wing length, body mass and tarsus length measured at the age of 16 days. Most traits were measured since the beginning of the research program (that is, in 2004), but some traits were first measured later (nestling body mass: 2005; nestling wing length: 2006; adult/nestling tarsus length: 2007) creating differences in data completeness among traits (sample size per traits can be found in <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table A1</a>).</p><h3 class=\"c-article__sub-heading\" id=\"Sec4\">Molecular analysis and pedigree construction</h3><p>DNA extraction, molecular sexing and microsatellite data analyses are detailed in <a aria-label=\"Reference Lessard et al. (2014)\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR32\" id=\"ref-link-section-d28693445e574\" title=\"Lessard A, Bourret A, Bélisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor. Behav Ecol Sociobiol 68: 733–742.\">Lessard et al. (2014)</a>. Briefly, DNA was extracted from blood samples following a standard salt-extraction method and DNA concentration was determined by migration on 2% agarose gels with a molecular weight standard. A molecular sexing technique was used to determine nestling sexes and to validate adult field observations. All DNA samples were characterized at six microsatellites loci using an AB3130xl automated DNA sequencer and allele lengths were determined using GeneMapper v4.1 (Applied Biosystems, Foster City, CA, USA).</p><p>We constructed a social pedigree using social male identities (that is, males caught in nest boxes while feeding the young) and a genetic pedigree using genetic father identities (that is, males assigned as fathers using genetic analyses—see below). In both pedigrees, dam identities were first determined by female captures during egg incubation and then verified genetically based on locus mismatches with nestlings (2.1% of broods with two females captured within the same nest box, from which only 3.6% were from mixed maternity; 11 nestling genotypes (0.15%) mismatched at more than two loci with their social mother genotype). Genetic fathers were determined by parental assignations with <span class=\"u-small-caps\">CERVUS</span> v3.0.7 (<a aria-label=\"Reference Kalinowski et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR23\" id=\"ref-link-section-d28693445e583\" title=\"Kalinowski ST, Taper ML, Marshall TC . (2007). Revising how the computer program cervus accommodates genotyping error increases success in paternity assignment. Mol Ecol 16: 1099–1106.\">Kalinowski et al., 2007</a>; <a aria-label=\"Reference Lemons et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR31\" id=\"ref-link-section-d28693445e586\" title=\"Lemons PR, Marshall TC, McCloskey SE, Sethi SA, Schmutz JA, Sedinger JS . (2015). A likelihood-based approach for assessment of extra-pair paternity and conspecific brood parasitism in natural populations. Mol Ecol Resour 7: 107–116.\">Lemons et al., 2015</a>) in a three step procedure. First, we proceeded to father assignments each year separately following a method slightly modified from <a aria-label=\"Reference Lessard et al. (2014)\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR32\" id=\"ref-link-section-d28693445e589\" title=\"Lessard A, Bourret A, Bélisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor. Behav Ecol Sociobiol 68: 733–742.\">Lessard et al. (2014)</a>. Candidate fathers considered in analyses included all males captured during a given breeding season within 15 km of the nest box of interest (see <a aria-label=\"Reference Lessard et al., 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR32\" id=\"ref-link-section-d28693445e592\" title=\"Lessard A, Bourret A, Bélisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (Tachycineta bicolor. Behav Ecol Sociobiol 68: 733–742.\">Lessard et al., 2014</a> for the rationale behind this approach), but also all males not captured in a given year but suspected of being present outside of our nest box system (that is, captured on the same farm on both previous and following years). These assignations were based on a 90% confidence level, assuming a 2% mistyping error rate and we used the percentage of social fathers captured as the percentage of candidate fathers known (variable among years, range: 64–88%). Second, social fathers, when known, were tested for being genetic fathers of offspring in their nest using the likelihood-based approach of <a aria-label=\"Reference Lemons et al. (2015)\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR31\" id=\"ref-link-section-d28693445e596\" title=\"Lemons PR, Marshall TC, McCloskey SE, Sethi SA, Schmutz JA, Sedinger JS . (2015). A likelihood-based approach for assessment of extra-pair paternity and conspecific brood parasitism in natural populations. Mol Ecol Resour 7: 107–116.\">Lemons et al. (2015)</a>. Briefly, we re-ran the parental analysis with the social father as the unique candidate father for a given nestling, and we defined the proportion of sampled fathers as the proportion of nestlings without any locus mismatch with its social father (that is, the probability of the social male being the true father). We then extracted, for each nestling, the critical logarithm of odds score associated with 95% confidence that its social father was not its true father and we compared these scores with those observed in regular parental analyses. Males significantly assigned to nestlings in the initial parental analysis (that is, step 1) were considered as their genetic fathers. For the remaining nestlings (that is, without a significant male assignment at step 1), if their social fathers could not be excluded (that is, step 2), they were considered as their genetic fathers but otherwise no genetic father were assigned to them (see <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure A1</a> for the exact number of fathers assigned to a nestling at each step).</p><p>Summary statistics for both pedigrees were obtained with the package <span class=\"u-small-caps\">PEDANTICS</span> (<a aria-label=\"Reference Morrissey et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d28693445e608\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309–2321.\">Morrissey et al., 2007</a>; <a aria-label=\"Reference Morrissey and Wilson, 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR37\" id=\"ref-link-section-d28693445e611\" title=\"Morrissey MB, Wilson AJ . (2010). PEDANTICS: an R package for pedigree-based genetic simulation and pedigree manipulation, characterization and viewing. Mol Ecol Resour 10: 711–719.\">Morrissey and Wilson, 2010</a>) in R v3.2.0 (<a aria-label=\"Reference R Core Team, 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR41\" id=\"ref-link-section-d28693445e614\" title=\"Core Team R . (2015). R: a language and environment for statistical computing \n",
      "                  http://www.R-project.org/\n",
      "                  \n",
      "                .\">R Core Team, 2015</a>). These statistics were computed for complete pedigrees, but also for pedigrees pruned to contain only informative individuals based on the availability of phenotypic data for each trait (hereafter pruned pedigrees) and are presented for each trait category in <a data-track=\"click\" data-track-action=\"table anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Tab1\">Table 1</a> (see also <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table A2</a> for more information on each trait).</p><div class=\"c-article-table\" data-container-section=\"table\" data-test=\"inline-table\" id=\"table-1\"><figure><figcaption class=\"c-article-table__figcaption\"><b data-test=\"table-caption\" id=\"Tab1\">Table 1 Summary statistics for social and genetic pedigrees, in complete pedigrees (total) or pruned pedigrees detailed for each trait category (means for all morphological, reproductive and nestling traits) based on data collected between 2004 and 2014 in our study system in southern Québec</b></figcaption><div class=\"u-text-right u-hide-print\"><a aria-label=\"Full size table 1\" class=\"c-article__pill-button\" data-test=\"table-link\" data-track=\"click\" data-track-action=\"view table\" data-track-label=\"button\" href=\"/articles/hdy201692/tables/1\" rel=\"nofollow\"><span>Full size table</span><svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-chevron-right\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></div></figure></div><h3 class=\"c-article__sub-heading\" id=\"Sec5\">Estimations of quantitative genetic parameters</h3><p>To estimate additive genetic (co)variances of our focal traits, we used both univariate and multivariate animal models (<a aria-label=\"Reference Kruuk, 2004\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR24\" id=\"ref-link-section-d28693445e1273\" title=\"Kruuk LEB . (2004). Estimating genetic parameters in natural populations using the ‘animal model’. Philos Trans R Soc B 359: 873–890.\">Kruuk, 2004</a>; <a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e1276\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). Fixed effects (for example, age, sex) were included for some traits based on mixed model analyses detailed in <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Information</a>. For adult morphological and reproductive traits, full univariate animal models were constructed as follows:</p><div class=\"c-article-equation\" id=\"Equ1\"><div class=\"c-article-equation__content\"><img alt=\"\" class=\"u-display-block\" src=\"//media.springernature.com/lw457/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Equ1_HTML.gif\"/></div></div><p>and for nestling traits:</p><div class=\"c-article-equation\" id=\"Equ2\"><div class=\"c-article-equation__content\"><img alt=\"\" class=\"u-display-block\" src=\"//media.springernature.com/lw457/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Equ2_HTML.gif\"/></div></div><p>where <i>V</i><sub>P</sub> is the phenotypic variance after accounting for fixed effects, <i>V</i><sub>A</sub> is the additive genetic variance, <i>V</i><sub>PE</sub> is the permanent environmental effect, <i>V</i><sub>Y</sub> and <i>V</i><sub>BY</sub> are the variance among years and among birth years, respectively, <i>V</i><sub>B</sub> is the variance among broods and <i>V</i><sub>R</sub> the residual variance. All traits followed a Gaussian distribution and animal models were resolved with a restricted maximum likelihood method (REML), using both the social and the genetic pedigree. Final animal models were constructed by sequential model-building from a model with only residual variance to more complex ones, with a comparison of models at each step using a likelihood ratio test (LRT; see <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Tables C1</a> for the increasing levels of complexity). Only <i>V</i><sub>A</sub> in the incubation duration models did not significantly improve the model likelihood (<i>χ</i><sup>2</sup>&lt;0.01, <i>P</i>&gt;0.99).</p><p>We also constructed three multivariate animal models, one for each trait category. For each of them, we first included the same variance terms as for univariate models. However, due to convergence problems when including <i>V</i><sub>PE</sub> for morphological and reproductive traits, we decided to use a model without the <i>V</i><sub>PE</sub> term on a reduced dataset comprising only one observation by individual (randomly chosen). Moreover, as we observed no <i>V</i><sub>A</sub> for incubation duration, this trait was not included in the multivariate analysis of reproductive traits. Covariances among traits for each variance components were estimated using unstructured variance models. Significance was tested by comparing a model including covariance estimation to a model where covariance was constrained to be equal to zero using LRT.</p><p>We estimated heritability (<i>h</i><sup>2</sup>=<i>V</i><sub>A</sub>/<i>V</i><sub>P</sub>, <a aria-label=\"Reference Falconer and Mackay, 1996\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR8\" id=\"ref-link-section-d28693445e1377\" title=\"Falconer DS, Mackay TFC . (1996) Introduction to Quantitative Genetics, 4th edn. Pearson: Essex, UK.\">Falconer and Mackay, 1996</a>) and coefficient of genetic variation (CV<sub>A</sub>=<img alt=\"\" src=\"//media.springernature.com/lw89/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq1_HTML.gif\" style=\"width:89px;max-width:none;\"/>, where <img alt=\"\" src=\"//media.springernature.com/lw11/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq2_HTML.gif\" style=\"width:11px;max-width:none;\"/> is the trait mean, <a aria-label=\"Reference Houle, 1992\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR19\" id=\"ref-link-section-d28693445e1399\" title=\"Houle D . (1992). Comparing evolvability and variability of quantitative traits. Genetics 130: 195–204.\">Houle, 1992</a>) for all traits within each analysis. For multivariate analyses we also estimated additive genetic correlations (<i>r</i><sub>A</sub>) between each pair of traits. All animal model analyses were conducted with ASRemL v3.0.5 (VSN International Ltd, Hemel Hempstead, UK). Standard errors (SEs) for variance components and <i>h</i><sup>2</sup> estimates were computed directly by ASRemL. For CV<sub>A</sub>, standard deviations were approximated following the equation provided by Lynch and Walsh <a aria-label=\"Reference (1998\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR33\" id=\"ref-link-section-d28693445e1413\" title=\"Lynch M, Walsh B . (1998) Genetics and Analysis of Quantitative Traits. Sinauer Associates: Sunderland, MA, USA.\">(1998</a>; equation A1.26c, Appendix 1).</p><h3 class=\"c-article__sub-heading\" id=\"Sec6\">Simulation analyses</h3><p>We simulated three different datasets of phenotypic data within both the social and genetic pedigree structures with different levels of complexity. In all cases, simulated traits were normally distributed among all individuals, with <i>V</i><sub>P</sub>=1, <img alt=\"\" src=\"//media.springernature.com/lw11/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq3_HTML.gif\" style=\"width:11px;max-width:none;\"/>=0 and <i>h</i><sup>2</sup> of 0.1, 0.3 and 0.5 for the focal trait. In dataset 1, we simulated one phenotype per individual with the parameters mentioned above to reflect the simplest scenario possible. In dataset 2, we simulated phenotypic traits with a more complex underlying structure based on equations (1) and (2) to better reflect our empirical dataset, with similar value of <i>h</i><sup><i>2</i></sup> as dataset 1. For nestling traits, we simulated phenotypes with <i>V</i><sub>Y</sub>=0.1 and <i>V</i><sub>B</sub>=0.3, whereas for morphological and reproductive traits we simulated phenotypic values with <i>V</i><sub>Y</sub>=0.1 and <i>V</i><sub>PE</sub>=0.1, thus implying multiple observations per individual. Finally, in dataset 3, different genetic and environmental correlations between traits were also implemented in simulations to assess the difference in power, precision and accuracy when using multivariate models. More specifically, we simulated phenotypes with <i>h</i><sup>2</sup> similar to dataset 1 for the focal trait, whereas fixing <i>h</i><sup>2</sup> of the two other traits at 0.3, for three different values of <i>r</i><sub>A</sub>, 0.1, 0.3 and 0.5, whereas fixing environmental correlation at 0.3. No other variance components were simulated. Simulations were performed in R, and breeding values were simulated with the package <span class=\"u-small-caps\">PEDANTICS</span>. Although simulations were performed within total social and genetic pedigree structures, only individuals with complete information in the empirical dataset were kept in these three simulated datasets.</p><p>Different analyses were performed on simulated datasets to answer two different questions. First, to assess if there was a bias created by not accounting for EPP, we used datasets simulated with the genetic pedigree in animal models using either the genetic pedigree (GG analysis) or the social pedigree (GS analysis), and then the results were compared. Second, to look at inherent differences in reliability caused by pedigree structures, datasets simulated with the social pedigree were analyzed using the social pedigree (SS analysis) and were compared with GG analysis. Animal models used for datasets 1 and 3 included only <i>V</i><sub>A</sub> and <i>V</i><sub>R</sub> as variance components (and covariance in dataset 3), whereas for dataset 2 they included all components described in equations (1) and (2). All these scenarios within each analysis were repeated 100 times for each trait, and animal model analyses were conducted with ASRemL.</p><p>Precision and accuracy for <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> estimates were checked visually with boxplots (that is, median for accuracy and distribution of estimates for precision) and also by computing the mean squared error (MSE) for each scenario. MSE is defined as <img alt=\"\" src=\"//media.springernature.com/lw62/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq4_HTML.gif\" style=\"width:62px;max-width:none;\"/>, where <i>d</i> is the true value (for example, the simulated parameter of <i>h</i><sup>2</sup> or <i>r</i><sub>A</sub>; these values are highly correlated (<i>r</i>&gt;0.99) with the realized value within our simulated population) and <img alt=\"\" src=\"//media.springernature.com/lw7/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq5_HTML.gif\" style=\"width:7px;max-width:none;\"/> is the estimated value; a small MSE indicates high precision and accuracy (<a aria-label=\"Reference Bolker, 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR4\" id=\"ref-link-section-d28693445e1532\" title=\"Bolker B . (2008) Ecological Models and Data in R. Princeton, University Princeton, NJ, USA.\">Bolker, 2008</a>). Root MSE (RMSE) was used to allow a comparison at the scale of the estimates (<a aria-label=\"Reference de Villemereuil et al., 2013\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR43\" id=\"ref-link-section-d28693445e1536\" title=\"de Villemereuil P, Gimenez O, Doligez B . (2013). Comparing parent–offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits. Methods Ecol Evol 4: 260–275.\">de Villemereuil et al., 2013</a>). Power to detect significant <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> within each scenario were estimated using a ‘rule of thumb’ approach by computing the proportion of estimates that were two times larger than their SE. Although LRTs could have provided a formal hypothesis testing, they were not possible to implement in our simulations procedures. Instead, the ‘rule of thumb’ approach is a practical indicator of statistical testing that could be easily integrated to simulation analyses and lead to results comparable to a LRT approach (<a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table A3</a>; see also <a aria-label=\"Reference Wilson et al., 2011\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR47\" id=\"ref-link-section-d28693445e1550\" title=\"Wilson AJ, Morrissey MB, Adams MJ, Walling CA, Guinness FE, Pemberton JM et al. (2011). Indirect genetics effects and evolutionary constraint: an analysis of social dominance in red deer, Cervus elaphus. J Evol Biol 24: 772–783.\">Wilson et al., 2011</a> for a similar approach). Finally, we also checked the identifiability of variance terms in dataset 2 using a Spearman’s rank correlation between estimates of phenotypic variance components. In this case, a low correlation between two components would indicate a high identifiability, whereas a high correlation would indicate a low identifiability.</p></div></div></section><section data-title=\"Results\"><div class=\"c-article-section\" id=\"Sec7-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec7\">Results</h2><div class=\"c-article-section__content\" id=\"Sec7-content\"><h3 class=\"c-article__sub-heading\" id=\"Sec8\">Pedigrees</h3><p>Our total dataset comprised 13 446 individuals, with 2839 individuals observed only as adults, 10 472 observed only as nestlings and 135 observed at both states (that is, recruits). From the field observations, we were able to determine the identity of the social father for 69.2% of nestlings (82.2% since 2006). Parental analyses allowed us to determine the genetic father for 53.3% of nestling (66.3% since 2006). On average, 1.74±0.82 (range: 1−5) genetic fathers were assigned per brood, and each adult male had on average 4.81±5.44 (range: 0−37) nestlings assigned over its entire lifespan. Overall, 49.3% of nestlings successfully genotyped with a known social father were extra-pair young (<i>N</i><sub>total</sub>=6382).</p><p>Pedigree size varied among the type of pedigree and traits (<a data-track=\"click\" data-track-action=\"table anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Tab1\">Table 1</a>). As expected, the social pedigree always had a higher number of links related to father identities than its genetic counterpart, a difference mainly caused by unsampled males contributing to the genetic pool of nestlings. Nestling traits had a much higher number of links than morphological and reproductive traits, which were both of similar order of magnitude, but with slightly fewer links for the latter (<a data-track=\"click\" data-track-action=\"table anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Tab1\">Table 1</a>). Pedigree sizes were similar for each trait within a given category (<a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table A2</a>).</p><h3 class=\"c-article__sub-heading\" id=\"Sec9\">Estimations of quantitative genetic parameters (empirical data)</h3><p>Variance components, and resulting heritability values, estimated from empirical data varied among traits and also often importantly between pedigrees (social versus genetic) and between models (univariate versus multivariate) (<a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig1\">Figure 1</a>; <a data-track=\"click\" data-track-action=\"table anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Tab2\">Table 2</a>). Biases in <i>h</i><sup>2</sup> and CV<sub>A</sub> resulting from using social instead of genetic pedigrees were moderate for morphological traits (<i>h</i><sup>2</sup>: −0.21 to −0.02; CV<sub>A</sub>: −0.008 to −0.001), almost null for reproductive traits (almost all around 0) and highly variable for nestling traits (<i>h</i><sup>2</sup>: −0.04–0.21; CV<sub>A</sub>: −0.001–0.051). Estimates of <i>r</i><sub>A</sub> obtained with the social and genetic pedigree were similar, except between nestling wing length and body mass where <i>r</i><sub>A</sub> was positive and significantly different from 0 with the social pedigree (<i>r</i><sub>A</sub>=0.56±0.13, <i>χ</i><sup>2</sup>=9.92, <i>P</i>=0.002), but not with the genetic pedigree (<i>r</i><sub>A</sub>=0.25±0.19, χ<sup>2</sup>=1.32, <i>P</i>=0.25; <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure C1</a>). Raw phenotypic variance for all traits can be found in <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table C4</a>.</p><div class=\"c-article-section__figure js-c-reading-companion-figures-item\" data-container-section=\"figure\" data-test=\"figure\" data-title=\"Figure 1\" id=\"figure-1\"><figure><figcaption><b class=\"c-article-section__figure-caption\" data-test=\"figure-caption-text\" id=\"Fig1\">Figure 1</b></figcaption><div class=\"c-article-section__figure-content\"><div class=\"c-article-section__figure-item\"><a class=\"c-article-section__figure-link\" data-test=\"img-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-label=\"image\" href=\"/articles/hdy201692/figures/1\" rel=\"nofollow\"><picture><source srcset=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig1_HTML.jpg?as=webp\" type=\"image/webp\"/><img alt=\"figure 1\" aria-describedby=\"Fig1\" height=\"705\" loading=\"lazy\" src=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig1_HTML.jpg\" width=\"560\"/></picture></a></div><div class=\"c-article-section__figure-description\" data-test=\"bottom-caption\" id=\"figure-1-desc\"><p>Proportion of phenotypic variance estimated from final animal models on empirical dataset, for (<b>a</b>) morphological, (<b>b</b>) reproductive and (<b>c</b>) nestling traits. Different models were assessed for each trait: univariate models using social (SU) and genetic (GU) pedigree, and multivariate models using social (SM) and genetic (GM) pedigree.</p></div></div><div class=\"u-text-right u-hide-print\"><a aria-label=\"Full size image figure 1\" class=\"c-article__pill-button\" data-test=\"article-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-dest=\"link:Figure1 Full size image\" data-track-label=\"button\" href=\"/articles/hdy201692/figures/1\" rel=\"nofollow\"><span>Full size image</span><svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-chevron-right\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></div></figure></div><div class=\"c-article-table\" data-container-section=\"table\" data-test=\"inline-table\" id=\"table-2\"><figure><figcaption class=\"c-article-table__figcaption\"><b data-test=\"table-caption\" id=\"Tab2\">Table 2 Summary of final A) univariate and B) multivariate animal models using empirical datasets and either the social pedigree or the genetic pedigree</b></figcaption><div class=\"u-text-right u-hide-print\"><a aria-label=\"Full size table 2\" class=\"c-article__pill-button\" data-test=\"table-link\" data-track=\"click\" data-track-action=\"view table\" data-track-label=\"button\" href=\"/articles/hdy201692/tables/2\" rel=\"nofollow\"><span>Full size table</span><svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-chevron-right\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></div></figure></div><h3 class=\"c-article__sub-heading\" id=\"Sec10\">Reliability of quantitative genetics estimates (simulated data)</h3><p>Visual inspection of boxplots (<a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig2\">Figures 2b, d, f</a> and <a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig3\">3b</a>) and comparison of RMSE values (<a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figures D1</a>) showed complementary information on precision and accuracy of <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> estimates. Precision of <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> estimates varied greatly depending on trait category (nestling&gt;morphological&gt;reproductive; mean±s.d. RMSE for <i>h</i><sup>2</sup>/<i>r</i><sub>A</sub>: <img alt=\"\" src=\"//media.springernature.com/lw46/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq6_HTML.gif\" style=\"width:46px;max-width:none;\"/>=0.055±0.045/0.081±0.025, <img alt=\"\" src=\"//media.springernature.com/lw45/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq7_HTML.gif\" style=\"width:45px;max-width:none;\"/>=0.122±0.026/0.342±0.061, <img alt=\"\" src=\"//media.springernature.com/lw33/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq8_HTML.gif\" style=\"width:33px;max-width:none;\"/>=0.200±0.045/0.430±0.025) but also among datasets used (dataset 2 (simulation of <i>h</i><sup>2</sup> and other source of resemblance among individuals)&gt;3 (simulation of <i>h</i><sup>2</sup> and <i>r</i><sub><i>A</i></sub>)&gt;1 (simulation of <i>h</i><sup>2</sup> only); mean±s.d. RMSE for <i>h</i><sup>2</sup>: <img alt=\"\" src=\"//media.springernature.com/lw49/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq9_HTML.gif\" style=\"width:49px;max-width:none;\"/>=0.144±0.086, <img alt=\"\" src=\"//media.springernature.com/lw45/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq10_HTML.gif\" style=\"width:45px;max-width:none;\"/>=0.100±0.047, <img alt=\"\" src=\"//media.springernature.com/lw45/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq11_HTML.gif\" style=\"width:45px;max-width:none;\"/>=0.127±0.072). Accuracy for <i>h</i><sup>2</sup> estimates was high for all GG and SS analyses (except for reproductive traits in dataset 3), but downwardly biased in GS analyses for nestling and reproductive traits, an effect more important as simulated <i>h</i><sup>2</sup> increased (mean±s.d. RMSE: <img alt=\"\" src=\"//media.springernature.com/lw27/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq12_HTML.gif\" style=\"width:27px;max-width:none;\"/>=0.117±0.076, <img alt=\"\" src=\"//media.springernature.com/lw24/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq13_HTML.gif\" style=\"width:24px;max-width:none;\"/>=0.143±0.061, <img alt=\"\" src=\"//media.springernature.com/lw31/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq14_HTML.gif\" style=\"width:31px;max-width:none;\"/>=0.115±0.075). Accuracy for <i>r</i><sub>A</sub> were similar for all type of analysis (mean±s.d. RMSE: <img alt=\"\" src=\"//media.springernature.com/lw31/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq15_HTML.gif\" style=\"width:31px;max-width:none;\"/>=0.283±0.160, <img alt=\"\" src=\"//media.springernature.com/lw29/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq16_HTML.gif\" style=\"width:29px;max-width:none;\"/>=0.294±0.151, <img alt=\"\" src=\"//media.springernature.com/lw21/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_IEq17_HTML.gif\" style=\"width:21px;max-width:none;\"/>=0.275±0.158).</p><div class=\"c-article-section__figure js-c-reading-companion-figures-item\" data-container-section=\"figure\" data-test=\"figure\" data-title=\"Figure 2\" id=\"figure-2\"><figure><figcaption><b class=\"c-article-section__figure-caption\" data-test=\"figure-caption-text\" id=\"Fig2\">Figure 2</b></figcaption><div class=\"c-article-section__figure-content\"><div class=\"c-article-section__figure-item\"><a class=\"c-article-section__figure-link\" data-test=\"img-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-label=\"image\" href=\"/articles/hdy201692/figures/2\" rel=\"nofollow\"><picture><source srcset=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig2_HTML.jpg?as=webp\" type=\"image/webp\"/><img alt=\"figure 2\" aria-describedby=\"Fig2\" height=\"871\" loading=\"lazy\" src=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig2_HTML.jpg\" width=\"685\"/></picture></a></div><div class=\"c-article-section__figure-description\" data-test=\"bottom-caption\" id=\"figure-2-desc\"><p>Power, precision and accuracy of heritability estimated from animal models for dataset 1 (<b>a</b> and <b>b</b>; simulation of <i>h</i><sup>2</sup> only), dataset 2 (<b>c</b> and <b>d</b>; simulation of <i>h</i><sup>2</sup> and other source of resemblance among individuals) and dataset 3 (<b>e</b> and <b>f</b>; simulation of <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub>) simulated on the pedigree structure of morphological traits (dark gray lines), reproductive traits (light gray lines) and nestling traits (black lines). Power of detecting significant heritability over 300 simulated trait values (100 simulations per trait) are presented in <b>a</b>, <b>c</b> and <b>e</b>. Distributions of these estimates are represented by boxplots (1st quartile, median, 3rd quartile) for 2 levels of heritability (dotted lines represent the <i>h</i><sup>2</sup> simulated value of 0.1 and 0.5) in <b>b</b>, <b>d</b> and <b>f</b>. Analysis types refer to datasets simulated with the genetic pedigree and analyzed with the genetic (GG analysis, solid lines) or the social pedigree (GS analysis, dashed lines), and datasets simulated with the social pedigree and analyzed with the social pedigree (SS analysis, dotted lines). For graphical representation, <i>r</i><sub>A</sub> was fixed at 0.3 in <b>e</b> and <b>f</b>.</p></div></div><div class=\"u-text-right u-hide-print\"><a aria-label=\"Full size image figure 2\" class=\"c-article__pill-button\" data-test=\"article-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-dest=\"link:Figure2 Full size image\" data-track-label=\"button\" href=\"/articles/hdy201692/figures/2\" rel=\"nofollow\"><span>Full size image</span><svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-chevron-right\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></div></figure></div><div class=\"c-article-section__figure js-c-reading-companion-figures-item\" data-container-section=\"figure\" data-test=\"figure\" data-title=\"Figure 3\" id=\"figure-3\"><figure><figcaption><b class=\"c-article-section__figure-caption\" data-test=\"figure-caption-text\" id=\"Fig3\">Figure 3</b></figcaption><div class=\"c-article-section__figure-content\"><div class=\"c-article-section__figure-item\"><a class=\"c-article-section__figure-link\" data-test=\"img-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-label=\"image\" href=\"/articles/hdy201692/figures/3\" rel=\"nofollow\"><picture><source srcset=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig3_HTML.jpg?as=webp\" type=\"image/webp\"/><img alt=\"figure 3\" aria-describedby=\"Fig3\" height=\"292\" loading=\"lazy\" src=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig3_HTML.jpg\" width=\"685\"/></picture></a></div><div class=\"c-article-section__figure-description\" data-test=\"bottom-caption\" id=\"figure-3-desc\"><p>Power, precision and accuracy of genetic additive correlation estimated from multivariate animal models, for dataset 3 (simulation of <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub>) simulated on the pedigree structure of morphological traits (dark gray lines), reproductive traits (light gray lines) and nestling traits (black lines). Power of detecting significant genetic correlation over 300 simulated trait values (100 simulations per trait), is presented in <b>a</b> and distributions of these estimates are represented by boxplots (1st quartile, median, 3rd quartile) in <b>b</b> (for 1 value of <i>r</i><sub>A</sub>—dotted lines represent the <i>r</i><sub>A</sub> true value of 0.5 simulated; 5 estimates&gt;1 are not presented). Analysis types refer to datasets simulated with the genetic pedigree and analyzed with the genetic (GG analysis, solid lines) or the social pedigree (GS analysis, dashed lines), and datasets simulated with the social pedigree and analyzed with the social pedigree (SS analysis, dotted lines). For graphical representation, <i>h</i><sup>2</sup> was fixed at 0.5 in <b>b</b>.</p></div></div><div class=\"u-text-right u-hide-print\"><a aria-label=\"Full size image figure 3\" class=\"c-article__pill-button\" data-test=\"article-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-dest=\"link:Figure3 Full size image\" data-track-label=\"button\" href=\"/articles/hdy201692/figures/3\" rel=\"nofollow\"><span>Full size image</span><svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-chevron-right\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></div></figure></div><p>Power to detect significant <i>h</i><sup>2</sup> or <i>r</i><sub>A</sub> was greater for traits with larger sample size (nestling&gt;morphological&gt;reproductive; <a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig2\">Figures 2a, c, e</a> and <a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig3\">3a</a>), but we observed no difference among traits within each category (data not shown). For multivariate animal models from dataset 3, power to detect <i>h</i><sup>2</sup> was similar for the different genetic correlations tested (data not shown), whereas power to detect genetic correlation increased with increasing heritability (<a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig3\">Figure 3a</a>). Within a trait category, power was similar between GG and SS analyses, except for nestling traits from dataset 2 were the simulated brood effect decreased power in SS analysis (<a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig2\">Figure 2c</a>). Finally, power comparison between GG and GS analyses revealed generally lower power in morphological and nestling trait for GS (<a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig2\">Figures 2a, c, e</a> and <a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig3\">3a</a>).</p><p>For adult traits, <i>V</i><sub>A</sub> and <i>V</i><sub>PE</sub> estimates from dataset 2 were highly negatively correlated, suggesting that these terms were almost completely confounded (morphological traits: <i>r</i><sub>s</sub>=−0.95, <i>P</i>&lt;0.001; reproductive traits: <i>r</i><sub>s</sub>=−0.94, <i>P</i>&lt;0.001, results from GG analysis with <i>h</i><sup>2</sup>=0.3; <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figures D3</a>). For nestling traits, <i>V</i><sub>A</sub> and <i>V</i><sub>B</sub> estimates were also correlated, but to a lesser extent (<i>r</i><sub>s</sub>=−0.42, <i>P</i>&lt;0.001; <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure D5</a>). All other variance terms showed correlations non-significantly different from 0 (all |<i>r</i><sub>s</sub>|&lt;0.09, <i>P</i>&gt;0.12).</p><p>Finally, given the large differences observed for some reproductive traits between SEs of <i>h</i><sup>2</sup> estimates from empirical and simulated dataset (<a data-track=\"click\" data-track-action=\"table anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Tab2\">Table 2</a>; <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure D6a</a>), we further inspected the impact of model parameters on SEs. More specifically, we assessed the impact of not accounting for an important cause of resemblance among individuals by running dataset 2 animal models and omitting <i>V</i><sub>PE</sub> or <i>V</i><sub>B</sub> components. By doing so, we observed a power of 1 at all traits, and all <i>h</i><sup>2</sup> estimates were very precise but showed a bias equivalent to the variance component not accounted for (that is, 0.1 for morphological and reproductive trait and 0.4 for nestling traits; <a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig4\">Figure 4</a>). As suspected, SEs for these biased <i>h</i><sup>2</sup> estimates were small and similar to those obtained in empirical analyses, where <i>V</i><sub>PE</sub> was almost completely confounded with <i>V</i><sub>A</sub> (<a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure D6b</a>).</p><div class=\"c-article-section__figure js-c-reading-companion-figures-item\" data-container-section=\"figure\" data-test=\"figure\" data-title=\"Figure 4\" id=\"figure-4\"><figure><figcaption><b class=\"c-article-section__figure-caption\" data-test=\"figure-caption-text\" id=\"Fig4\">Figure 4</b></figcaption><div class=\"c-article-section__figure-content\"><div class=\"c-article-section__figure-item\"><a class=\"c-article-section__figure-link\" data-test=\"img-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-label=\"image\" href=\"/articles/hdy201692/figures/4\" rel=\"nofollow\"><picture><source srcset=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig4_HTML.jpg?as=webp\" type=\"image/webp\"/><img alt=\"figure 4\" aria-describedby=\"Fig4\" height=\"340\" loading=\"lazy\" src=\"//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_Article_BFhdy201692_Fig4_HTML.jpg\" width=\"556\"/></picture></a></div><div class=\"c-article-section__figure-description\" data-test=\"bottom-caption\" id=\"figure-4-desc\"><p>Precision and accuracy of heritability estimated from animal models omitting an important cause of resemblance among individuals (<i>V</i><sub>PE</sub> or <i>V</i><sub>B</sub>), for morphological traits, reproductive traits and nestling traits simulated in dataset 2. Distribution of these estimates are represented by boxplots (1st quartile, median, 3rd quartile) for 2 levels of heritability (dotted lines represent the <i>h</i><sup>2</sup> true value of 0.1 and 0.5 simulated). Analysis types refer to datasets simulated with the genetic pedigree and analyzed with both the genetic and the social pedigree (GG and GS analyses, respectively), and datasets simulated with the social pedigree and analyzed with the social pedigree (SS analysis).</p></div></div><div class=\"u-text-right u-hide-print\"><a aria-label=\"Full size image figure 4\" class=\"c-article__pill-button\" data-test=\"article-link\" data-track=\"click\" data-track-action=\"view figure\" data-track-dest=\"link:Figure4 Full size image\" data-track-label=\"button\" href=\"/articles/hdy201692/figures/4\" rel=\"nofollow\"><span>Full size image</span><svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-chevron-right\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></div></figure></div></div></div></section><section data-title=\"Discussion\"><div class=\"c-article-section\" id=\"Sec11-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec11\">Discussion</h2><div class=\"c-article-section__content\" id=\"Sec11-content\"><p>Despite the importance of obtaining precise and accurate genetic (co)variance estimates when assessing a population’s adaptive potential, their reliability is rarely explicitly tested or reported in wild populations. Here, we formally assessed the reliability of genetic (co)variance estimates, in a species with a high rate of extra-pair reproduction and low recruitment, with a combination of empirical and simulated data. Altogether, our simulation analyses emphasized the limits of this particular dataset by revealing an important lack of precision in <i>h</i><sup>2</sup> and <i>r</i><sub>A</sub> estimates for all adult traits, a lack of power to detect significant effects and identifiability problems between <i>V</i><sub>A</sub> and V<sub>PE</sub>. Moreover, we observed a large bias in <i>h</i><sup><i>2</i></sup> when using the social pedigree instead of the genetic one, and also when non-genetic causes of resemblance among individuals (that is, repeated measurements or brood effects) were not accounted for in our analyses. We briefly discuss below (i) the factors contributing to estimate reliability, (ii) the caution needed with model parameterization, (iii) the impacts of high levels of EPP on genetic variance estimates and (iv) we finally conclude with remarks on the accessibility and utility of simulation analyses to address all these potential problems.</p><h3 class=\"c-article__sub-heading\" id=\"Sec12\">Estimate reliability and generalization among studies</h3><p>At first glance, our sample size seemed large enough to be powerful, with more than 10 years of sampling and a number of records within our pruned pedigrees that was larger than the median number of records typically reported for similar studies in the literature. Indeed, a recent compilation of quantitative genetics estimates obtained from wild populations by <a aria-label=\"Reference Postma (2014)\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR39\" id=\"ref-link-section-d28693445e3130\" title=\"Postma E . (2014). Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates? In: Charmantier A, Garant D et al. (eds) Quantitative Genetics in the Wild. Oxford University Press, Oxford, UK, pp 16–33.\">Postma (2014)</a> showed that for estimates obtained with animal models, the median number of records for life history traits was 377 (range 6–4992; from 39 studies, covering 19 species) and 363 for morphological traits (range 50–38 024; from 47 studies, covering 22 species). This suggests that sample size alone is not sufficient to infer pedigree quality, as it does not reflect the connectivity of pedigrees (<a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e3133\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). In our study system, the low recruitment of nestlings (1.3%) results in very few grandparent links within our pedigree, which greatly reduces its power. However, this problem is far less important for nestling traits as the high-EPP rate results in a genetic pedigree containing several half-sib families, which increase its power (see GG versus SS, <a data-track=\"click\" data-track-action=\"figure anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Fig2\">Figure 2c</a>) despite a smaller number of observations compared with the social pedigree for all categories (<a data-track=\"click\" data-track-action=\"table anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#Tab1\">Table 1</a>). Thus, given that pedigree structures are different from one study system to another, their impact on reliability of quantitative genetic estimates is best predicted from simulation analyses.</p><p>Various ecological and population specific factors could generate fluctuations in the precision and accuracy of quantitative genetic estimates among study systems. Here, our study system was characterized by a high rate of EPP and a low recruitment, two characteristics that affected the reliability of quantitative genetic estimates for some trait types. Although these characteristics are certainly common in other migratory species (for example, Common reed bunting (<i>Emberiza schoeniclus</i>) and Red-eyed vireo (<i>Vireo olivaceus</i>), <a aria-label=\"Reference Griffith et al., 2002\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR13\" id=\"ref-link-section-d28693445e3151\" title=\"Griffith SC, Owens IPF, Thuman KA . (2002). Extra pair paternity in birds: a review of interspecific variation and adaptive function. Mol Ecol 11: 2195–2212.\">Griffith et al., 2002</a>), other characteristics, such as low-recapture rate between years in open study systems, a sex bias in sampling or large natal dispersal may also have important consequences on the precision of estimates. Moreover, differential mortality or capture rate might bias the sampling of some traits, and consequently affect the accuracy of estimates that can be obtained (for example, <a aria-label=\"Reference Dingemanse et al., 2009\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR6\" id=\"ref-link-section-d28693445e3154\" title=\"Dingemanse NJ, Kazem AJN, Réale D, Wright J . (2009). Behavioural reaction norms: animal personality meets individual plasticity. Trends Ecol Evol 25: 81–89.\">Dingemanse et al., 2009</a>). It is thus important to appropriately identify the limits in a particular system and assess their impacts on the reliability of quantitative genetic estimates by using simulations.</p><h3 class=\"c-article__sub-heading\" id=\"Sec13\">Caution needed with model parameterization</h3><p>The choice of which variance terms to include in a model is a crucial step that can have substantial impacts on reliability of genetic variance estimates (<a aria-label=\"Reference Kruuk and Hadfield, 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d28693445e3165\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890–1903.\">Kruuk and Hadfield, 2007</a>; <a aria-label=\"Reference Wilson, 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR46\" id=\"ref-link-section-d28693445e3168\" title=\"Wilson AJ . (2008). Why h2 does not always equal VA/VP? J Evol Biol 21: 647–650.\">Wilson, 2008</a>). Here, the decision to include or not a particular variance term relied on LRTs, which is a widely used approach (<a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e3171\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). However, in some cases even if the inclusion of a particular component is not improving the model likelihood, it may still have to be included (<a aria-label=\"Reference Wilson et al., 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR48\" id=\"ref-link-section-d28693445e3174\" title=\"Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA et al. (2010). An ecologist’s guide to the animal model. J Anim Ecol 79: 13–26.\">Wilson et al., 2010</a>). This is the case for component of variance attributable to repeated measurement of a given individual (<i>V</i><sub>PE</sub>). In our models, we had to account for multiple observations by fitting a <i>V</i><sub>PE</sub> term in our model, but this component was almost completely confounded with <i>V</i><sub>A</sub>. This situation probably occurred because of the low number of observations per individual (mean of 1.5 (averaged across traits; range: 1–7) for morphological traits and 1.2 (range: 1–7) for reproductive traits), a situation that should be present in other short-lived species datasets. Not accounting for repeated measurements in an animal model is equivalent to considering each repeated measurement as a clone of the same individual, which is inappropriate. In such cases, using a dataset with only one observation per individual may lead to more accurate estimates and easier model convergence.</p><p>Standard errors are often viewed as predictors of an estimate’s accuracy or significance, but this can be misleading (<a aria-label=\"Reference Krzywinski and Altman, 2013\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR27\" id=\"ref-link-section-d28693445e3193\" title=\"Krzywinski M, Altman N . (2013). Points of significance: error bars. Nat Methods 10: 921–922.\">Krzywinski and Altman, 2013</a>). Estimates from animal models have generally smaller standard errors than those obtained with parent–offspring regressions (<a aria-label=\"Reference Kruuk, 2004\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR24\" id=\"ref-link-section-d28693445e3196\" title=\"Kruuk LEB . (2004). Estimating genetic parameters in natural populations using the ‘animal model’. Philos Trans R Soc B 359: 873–890.\">Kruuk, 2004</a>; <a aria-label=\"Reference Postma, 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR39\" id=\"ref-link-section-d28693445e3199\" title=\"Postma E . (2014). Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates? In: Charmantier A, Garant D et al. (eds) Quantitative Genetics in the Wild. Oxford University Press, Oxford, UK, pp 16–33.\">Postma, 2014</a>), partly due to their integration of multiple observations for a given individual (<a aria-label=\"Reference Åkesson et al., 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR1\" id=\"ref-link-section-d28693445e3202\" title=\"Åkesson M, Bensch S, Hasselquist D, Tarka M, Hansson B, Akesson M . (2008). Estimating heritabilities and genetic correlations: comparing the ‘animal model’ with parent–offspring regression using data from a natural population. PLoS One 3: e1739.\">Åkesson et al., 2008</a>). As previously stated, failing to account for multiple observations can create unpredictable bias in phenotypic variance component estimates (<a aria-label=\"Reference Kruuk and Hadfield, 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR25\" id=\"ref-link-section-d28693445e3205\" title=\"Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. J Evol Biol 20: 1890–1903.\">Kruuk and Hadfield, 2007</a>) and considerably reduce standard errors around these biased estimates (<a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Figure D6</a>). Moreover, it seems that problems of identifiability could also result in reduced standard errors. For example, in our empirical analyses, small standard errors were estimated around <i>h</i><sup>2</sup> estimates for reproductive traits from univariate animal models with repeated measurements per individual (SE range: 0.03–0.04), which could have led us to misleadingly conclude that our dataset was powerful and our estimates were precise for these traits.</p><p>This study was performed with a frequentist method (that is, REML) rather than a Bayesian method (that is, MCMC) because it generally gives more precise and accurate estimates for traits following a Gaussian distribution and it necessitates a shorter computation time (<a aria-label=\"Reference de Villemereuil et al., 2013\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR43\" id=\"ref-link-section-d28693445e3219\" title=\"de Villemereuil P, Gimenez O, Doligez B . (2013). Comparing parent–offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits. Methods Ecol Evol 4: 260–275.\">de Villemereuil et al., 2013</a>). Bayesian methods have other advantages, such as the use of posterior distributions for the calculation of derived parameters (for example, heritability) and associated standard errors and they are particularly useful for analyses of non-Gaussian traits (<a aria-label=\"Reference Morrissey et al., 2014\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR36\" id=\"ref-link-section-d28693445e3222\" title=\"Morrissey MB, de Villemereuil P, Doligez B, Gimenez O . (2014). Bayesian approaches to the quantitative genetic analysis of natural populations. In: Charmantier A, Garant D et al. (eds) Quantitative Genetics in the Wild. Oxford University Press, Oxford, UK, pp 228–253.\">Morrissey et al., 2014</a>). In the present context, problems related to missing of an important source of resemblance among individuals are not method specific and could impact any model regardless of the method used for its estimation. However, it is less clear if identifiability problems could be better uncovered with Bayesian approach, for example through an appropriate MCMC checking. Simulation studies such as this one could be applied to animal models resolved with a Bayesian approach to better assess its limits.</p><h3 class=\"c-article__sub-heading\" id=\"Sec14\">Impact of high level of EPP on estimates</h3><p>In theory, EPPs, if not accounted for, could downwardly bias additive genetic variance and resulting heritability estimates (<a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e3233\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>). Yet, previous simulation studies showed that even if biases increased with the importance of EPPs, with increasing heritability and when focal traits were directly related to the number of extra-pair young produced, underestimations were generally smaller than 15% (<a aria-label=\"Reference Firth et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d28693445e3236\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336–1344.\">Firth et al., 2015</a>). In our study, the rate of EPP was higher than those previously tested so far (up to 40% in <a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e3239\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>, 12.5% in <a aria-label=\"Reference Firth et al., 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR9\" id=\"ref-link-section-d28693445e3242\" title=\"Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. Evolution 69: 1336–1344.\">Firth et al., 2015</a>), but its effect on quantitative genetics estimates was complex. First, our simulation analyses showed that the bias in <i>h</i><sup>2</sup> estimates resulting from using the social instead of the genetic pedigree changed depending on the trait category (that is, higher for nestling traits). Also, our empirical analyses suggested that the impact of not accounting for EPP was noticeable for morphological and nestling traits, but sometimes resulted in higher <i>h</i><sup>2</sup> and CV<sub>A</sub> when using the social pedigree (that is, for nestling wing length and body mass). A similar unexpected finding was previously reported in blue tits (<i>Cyanistes caeruleus</i>), where <i>h</i><sup>2</sup> estimates from a social pedigree were sometimes higher than those from a genetic pedigree (<a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e3263\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>). These positive biases observed in empirical data could be due to the influence of the social father (for example through parental care), which could be captured in <i>V</i><sub>A</sub> components (<a aria-label=\"Reference Griffith et al., 1999\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR12\" id=\"ref-link-section-d28693445e3271\" title=\"Griffith SC, Owens IPF, Burke T . (1999). Environmental determination of a sexually selected trait. Science 400: 358–360.\">Griffith et al., 1999</a>; <a aria-label=\"Reference Charmantier and Réale, 2005\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR5\" id=\"ref-link-section-d28693445e3274\" title=\"Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? Mol Ecol 14: 2839–2850.\">Charmantier and Réale, 2005</a>) rather than due to pedigree errors. To assess this possible problem, we performed an additional animal model analysis including mother and social father identities as additional variance components. We found that new <i>h</i><sup>2</sup> and CV<sub>A</sub> values obtained using the social pedigree were smaller than values obtained with the genetic one (change in estimates for body mass <i>h</i><sup>2</sup>=−0.06, CV<sub>A</sub>=−0.011; wing length <i>h</i><sup>2</sup>=−0.04, CV<sub>A</sub>=−0.041; difference calculated with univariate animal models; see <a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Table C5</a> for details). This further emphasizes the importance of model parameterization on reliability of variance component estimates. Finally, even if the genetic pedigree better represents relationships among individuals than the social pedigree, it probably still contains a few incorrect links among individuals and thus the estimated heritabilities might also be slightly downwardly biased.</p></div></div></section><section data-title=\"Conclusion\"><div class=\"c-article-section\" id=\"Sec15-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec15\">Conclusion</h2><div class=\"c-article-section__content\" id=\"Sec15-content\"><p>Simulation analyses are now widely accessible to anyone with minimal programming skills in R, for instance by using the ‘<i>phensim</i>’ function within the package <span class=\"u-small-caps\">PEDANTICS</span>, the ‘<i>rbv</i>’ function within the package MCMCglmm (<a aria-label=\"Reference Hadfield, 2010\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR14\" id=\"ref-link-section-d28693445e3317\" title=\"Hadfield JD . (2010). MCMC methods for multi-response generalized linear mixed models: the MCMCglmm R package. J Stat Softw 33: 1–22.\">Hadfield, 2010</a>) or the ‘<i>drfx</i>’ and the ‘<i>grfx</i>’ function within the package <span class=\"u-small-caps\">NADIV</span> (<a aria-label=\"Reference Wolak, 2012\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR50\" id=\"ref-link-section-d28693445e3330\" title=\"Wolak ME . (2012). nadiv : an R package to create relatedness matrices for estimating non-additive genetic variances in animal models. Methods Ecol Evol 3: 792–796.\">Wolak, 2012</a>). Moreover, other methods such as profile likelihoods (<a aria-label=\"Reference Meyer, 2008\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR34\" id=\"ref-link-section-d28693445e3333\" title=\"Meyer K . (2008). Likelihood calculations to evaluate experimental designs to estimate genetic variances. Heredity 101: 212–221.\">Meyer, 2008</a>; implemented in the ‘proLik’ function within the package <span class=\"u-small-caps\">NADIV,</span> <a aria-label=\"Reference Wolak, 2012\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR50\" id=\"ref-link-section-d28693445e3339\" title=\"Wolak ME . (2012). nadiv : an R package to create relatedness matrices for estimating non-additive genetic variances in animal models. Methods Ecol Evol 3: 792–796.\">Wolak, 2012</a>) are more accessible and can also improve our comprehension of a particular dataset limits (<a aria-label=\"Reference Houle and Meyer, 2015\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR20\" id=\"ref-link-section-d28693445e3343\" title=\"Houle D, Meyer K . (2015). Estimating sampling error of evolutionary statistics based on genetic covariance matrices using maximum likelihood. J Evol Biol 28: 1542–1549.\">Houle and Meyer, 2015</a>). As each study system is unique, in terms of pedigree structure, number of repeated measures per individual or potential causes of pedigree errors, it is difficult to predict the reliability of quantitative genetics estimates without testing it formally. Detailed simulations should be routinely included when reporting quantitative genetics analyses performed within a new study system, or when applying more complex modeling to previously studied populations, to explicitly assess the precision and accuracy of genetic variance components (<a aria-label=\"Reference Morrissey et al., 2007\" data-test=\"citation-ref\" data-track=\"click\" data-track-action=\"reference anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#ref-CR38\" id=\"ref-link-section-d28693445e3346\" title=\"Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (Ovis aries. J Evol Biol 20: 2309–2321.\">Morrissey et al., 2007</a>).</p></div></div></section><section data-title=\"Data archiving\"><div class=\"c-article-section\" id=\"Sec16-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec16\">Data archiving</h2><div class=\"c-article-section__content\" id=\"Sec16-content\"><p>Data available from the Dryad Digital Repository: <a href=\"https://doi.org/10.5061/dryad.n8c57\">http://dx.doi.org/10.5061/dryad.n8c57</a>.</p></div></div></section>\n",
      "</div>\n",
      "<div>\n",
      "<div id=\"MagazineFulltextArticleBodySuffix\"><section aria-labelledby=\"Bib1\" data-title=\"References\"><div class=\"c-article-section\" id=\"Bib1-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Bib1\">References</h2><div class=\"c-article-section__content\" id=\"Bib1-content\"><div data-container-section=\"references\"><ul class=\"c-article-references\" data-track-component=\"outbound reference\"><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR1\">Åkesson M, Bensch S, Hasselquist D, Tarka M, Hansson B, Akesson M . (2008). Estimating heritabilities and genetic correlations: comparing the ‘animal model’ with parent–offspring regression using data from a natural population. <i>PLoS One</i> <b>3</b>: e1739.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 1\" data-doi=\"10.1371/journal.pone.0001739\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1371/journal.pone.0001739\" href=\"https://doi.org/10.1371%2Fjournal.pone.0001739\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 1\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18320057\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 1\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2254494\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 1\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimating%20heritabilities%20and%20genetic%20correlations%3A%20comparing%20the%20%E2%80%98animal%20model%E2%80%99%20with%20parent%E2%80%93offspring%20regression%20using%20data%20from%20a%20natural%20population&amp;journal=PLoS%20One&amp;doi=10.1371%2Fjournal.pone.0001739&amp;volume=3&amp;publication_year=2008&amp;author=%C3%85kesson%2CM&amp;author=Bensch%2CS&amp;author=Hasselquist%2CD&amp;author=Tarka%2CM&amp;author=Hansson%2CB&amp;author=Akesson%2CM\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR2\">Bérénos C, Ellis PA, Pilkington JG, Pemberton JM . (2014). Estimating quantitative genetic parameters in wild populations: a comparison of pedigree and genomic approaches. <i>Mol Ecol</i> <b>23</b>: 3434–3451.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 2\" data-doi=\"10.1111/mec.12827\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/mec.12827\" href=\"https://doi.org/10.1111%2Fmec.12827\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 2\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=24917482\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 2\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4149785\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 2\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimating%20quantitative%20genetic%20parameters%20in%20wild%20populations%3A%20a%20comparison%20of%20pedigree%20and%20genomic%20approaches&amp;journal=Mol%20Ecol&amp;doi=10.1111%2Fmec.12827&amp;volume=23&amp;pages=3434-3451&amp;publication_year=2014&amp;author=B%C3%A9r%C3%A9nos%2CC&amp;author=Ellis%2CPA&amp;author=Pilkington%2CJG&amp;author=Pemberton%2CJM\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR3\">Blows MW, Hoffmann AA . (2005). A reassessment of genetic limits to evoutionary change. <i>Ecology</i> <b>86</b>: 1371–1384.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 3\" data-doi=\"10.1890/04-1209\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1890/04-1209\" href=\"https://doi.org/10.1890%2F04-1209\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 3\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=A%20reassessment%20of%20genetic%20limits%20to%20evoutionary%20change&amp;journal=Ecology&amp;doi=10.1890%2F04-1209&amp;volume=86&amp;pages=1371-1384&amp;publication_year=2005&amp;author=Blows%2CMW&amp;author=Hoffmann%2CAA\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR4\">Bolker B . (2008) <i>Ecological Models and Data in R</i>. Princeton, University Princeton, NJ, USA.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Book reference 4\" data-doi=\"10.2307/j.ctvcm4g37\" data-track=\"click\" data-track-action=\"book reference\" data-track-label=\"10.2307/j.ctvcm4g37\" href=\"https://doi.org/10.2307%2Fj.ctvcm4g37\" rel=\"nofollow noopener\">Book</a> \n",
      "    <a aria-label=\"Google Scholar reference 4\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Ecological%20Models%20and%20Data%20in%20R&amp;doi=10.2307%2Fj.ctvcm4g37&amp;publication_year=2008&amp;author=Bolker%2CB\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR5\">Charmantier A, Réale D . (2005). How do misassigned paternities affect the estimation of heritability in the wild? <i>Mol Ecol</i> <b>14</b>: 2839–2850.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 5\" data-doi=\"10.1111/j.1365-294X.2005.02619.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1365-294X.2005.02619.x\" href=\"https://doi.org/10.1111%2Fj.1365-294X.2005.02619.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 5\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:CAS:528:DC%2BD2MXpt1Cju7g%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 5\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=16029482\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 5\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=How%20do%20misassigned%20paternities%20affect%20the%20estimation%20of%20heritability%20in%20the%20wild%3F&amp;journal=Mol%20Ecol&amp;doi=10.1111%2Fj.1365-294X.2005.02619.x&amp;volume=14&amp;pages=2839-2850&amp;publication_year=2005&amp;author=Charmantier%2CA&amp;author=R%C3%A9ale%2CD\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR6\">Dingemanse NJ, Kazem AJN, Réale D, Wright J . (2009). Behavioural reaction norms: animal personality meets individual plasticity. <i>Trends Ecol Evol</i> <b>25</b>: 81–89.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 6\" data-doi=\"10.1016/j.tree.2009.07.013\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1016/j.tree.2009.07.013\" href=\"https://doi.org/10.1016%2Fj.tree.2009.07.013\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 6\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=19748700\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 6\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Behavioural%20reaction%20norms%3A%20animal%20personality%20meets%20individual%20plasticity&amp;journal=Trends%20Ecol%20Evol&amp;doi=10.1016%2Fj.tree.2009.07.013&amp;volume=25&amp;pages=81-89&amp;publication_year=2009&amp;author=Dingemanse%2CNJ&amp;author=Kazem%2CAJN&amp;author=R%C3%A9ale%2CD&amp;author=Wright%2CJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR7\">Dunn PO, Whittingham LA, Pitcher TE . (2001). Mating systems, sperm competition, and the evolution of sexual dimorphism in birds. <i>Evolution</i> <b>55</b>: 161–175.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 7\" data-doi=\"10.1111/j.0014-3820.2001.tb01281.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.0014-3820.2001.tb01281.x\" href=\"https://doi.org/10.1111%2Fj.0014-3820.2001.tb01281.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 7\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD3M7mt1Gmsw%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 7\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=11263736\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 7\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Mating%20systems%2C%20sperm%20competition%2C%20and%20the%20evolution%20of%20sexual%20dimorphism%20in%20birds&amp;journal=Evolution&amp;doi=10.1111%2Fj.0014-3820.2001.tb01281.x&amp;volume=55&amp;pages=161-175&amp;publication_year=2001&amp;author=Dunn%2CPO&amp;author=Whittingham%2CLA&amp;author=Pitcher%2CTE\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR8\">Falconer DS, Mackay TFC . (1996) <i>Introduction to Quantitative Genetics</i>, 4th edn. Pearson: Essex, UK.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Google Scholar reference 8\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Introduction%20to%20Quantitative%20Genetics&amp;publication_year=1996&amp;author=Falconer%2CDS&amp;author=Mackay%2CTFC\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR9\">Firth JA, Hadfield JD, Santure AW, Slate J, Sheldon BC . (2015). The influence of nonrandom extra-pair paternity on heritability estimates derived from wild pedigrees. <i>Evolution</i> <b>69</b>: 1336–1344.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 9\" data-doi=\"10.1111/evo.12649\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/evo.12649\" href=\"https://doi.org/10.1111%2Fevo.12649\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 9\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=25800997\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 9\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4950017\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 9\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20influence%20of%20nonrandom%20extra-pair%20paternity%20on%20heritability%20estimates%20derived%20from%20wild%20pedigrees&amp;journal=Evolution&amp;doi=10.1111%2Fevo.12649&amp;volume=69&amp;pages=1336-1344&amp;publication_year=2015&amp;author=Firth%2CJA&amp;author=Hadfield%2CJD&amp;author=Santure%2CAW&amp;author=Slate%2CJ&amp;author=Sheldon%2CBC\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR10\">Garcia-Gonzalez F, Simmons LW, Tomkins JL, Kotiaho JS, Evans JP . (2012). Comparing evolvabilities: common errors surrounding the calculation and use of coefficients of additive genetic variation. <i>Evolution</i> <b>66</b>: 2341–2349.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 10\" data-doi=\"10.1111/j.1558-5646.2011.01565.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1558-5646.2011.01565.x\" href=\"https://doi.org/10.1111%2Fj.1558-5646.2011.01565.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 10\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=22834736\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 10\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Comparing%20evolvabilities%3A%20common%20errors%20surrounding%20the%20calculation%20and%20use%20of%20coefficients%20of%20additive%20genetic%20variation&amp;journal=Evolution&amp;doi=10.1111%2Fj.1558-5646.2011.01565.x&amp;volume=66&amp;pages=2341-2349&amp;publication_year=2012&amp;author=Garcia-Gonzalez%2CF&amp;author=Simmons%2CLW&amp;author=Tomkins%2CJL&amp;author=Kotiaho%2CJS&amp;author=Evans%2CJP\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR11\">Ghilain A, Bélisle M . (2008). Breeding success of tree swallows along a gradient of agricultural intensification. <i>Ecol Appl</i> <b>18</b>: 1140–1154.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 11\" data-doi=\"10.1890/07-1107.1\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1890/07-1107.1\" href=\"https://doi.org/10.1890%2F07-1107.1\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 11\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18686577\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 11\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Breeding%20success%20of%20tree%20swallows%20along%20a%20gradient%20of%20agricultural%20intensification&amp;journal=Ecol%20Appl&amp;doi=10.1890%2F07-1107.1&amp;volume=18&amp;pages=1140-1154&amp;publication_year=2008&amp;author=Ghilain%2CA&amp;author=B%C3%A9lisle%2CM\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR12\">Griffith SC, Owens IPF, Burke T . (1999). Environmental determination of a sexually selected trait. <i>Science</i> <b>400</b>: 358–360.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"CAS reference 12\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:CAS:528:DyaK1MXkvFOnsr4%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"Google Scholar reference 12\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Environmental%20determination%20of%20a%20sexually%20selected%20trait&amp;journal=Science&amp;volume=400&amp;pages=358-360&amp;publication_year=1999&amp;author=Griffith%2CSC&amp;author=Owens%2CIPF&amp;author=Burke%2CT\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR13\">Griffith SC, Owens IPF, Thuman KA . (2002). Extra pair paternity in birds: a review of interspecific variation and adaptive function. <i>Mol Ecol</i> <b>11</b>: 2195–2212.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 13\" data-doi=\"10.1046/j.1365-294X.2002.01613.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1046/j.1365-294X.2002.01613.x\" href=\"https://doi.org/10.1046%2Fj.1365-294X.2002.01613.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 13\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:CAS:528:DC%2BD38XptlSmurY%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 13\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=12406233\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 13\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Extra%20pair%20paternity%20in%20birds%3A%20a%20review%20of%20interspecific%20variation%20and%20adaptive%20function&amp;journal=Mol%20Ecol&amp;doi=10.1046%2Fj.1365-294X.2002.01613.x&amp;volume=11&amp;pages=2195-2212&amp;publication_year=2002&amp;author=Griffith%2CSC&amp;author=Owens%2CIPF&amp;author=Thuman%2CKA\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR14\">Hadfield JD . (2010). MCMC methods for multi-response generalized linear mixed models: the MCMCglmm R package. <i>J Stat Softw</i> <b>33</b>: 1–22.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 14\" data-doi=\"10.18637/jss.v033.i02\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.18637/jss.v033.i02\" href=\"https://doi.org/10.18637%2Fjss.v033.i02\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 14\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=MCMC%20methods%20for%20multi-response%20generalized%20linear%20mixed%20models%3A%20the%20MCMCglmm%20R%20package&amp;journal=J%20Stat%20Softw&amp;doi=10.18637%2Fjss.v033.i02&amp;volume=33&amp;pages=1-22&amp;publication_year=2010&amp;author=Hadfield%2CJD\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR15\">Henderson CR . (1984) <i>Applications of Linear Models in Animal Breeding</i>. University of Guelph: Guelph, ON, Canada.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Google Scholar reference 15\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Applications%20of%20Linear%20Models%20in%20Animal%20Breeding&amp;publication_year=1984&amp;author=Henderson%2CCR\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR16\">Hendry AP, Kinnison MT, Heino M, Day T, Smith TB, Fitt G <i>et al</i>. (2011). Evolutionary principles and their practical application. <i>Evol Appl</i> <b>4</b>: 159–183.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 16\" data-doi=\"10.1111/j.1752-4571.2010.00165.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1752-4571.2010.00165.x\" href=\"https://doi.org/10.1111%2Fj.1752-4571.2010.00165.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 16\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=25567966\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 16\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3352551\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 16\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Evolutionary%20principles%20and%20their%20practical%20application&amp;journal=Evol%20Appl&amp;doi=10.1111%2Fj.1752-4571.2010.00165.x&amp;volume=4&amp;pages=159-183&amp;publication_year=2011&amp;author=Hendry%2CAP&amp;author=Kinnison%2CMT&amp;author=Heino%2CM&amp;author=Day%2CT&amp;author=Smith%2CTB&amp;author=Fitt%2CG\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR17\">Hoffmann AA, Merilä J . (1999). Heritable variation and evolution under favourable and unfavourable conditions. <i>Trends Ecol Evol</i> <b>14</b>: 96–101.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 17\" data-doi=\"10.1016/S0169-5347(99)01595-5\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1016/S0169-5347(99)01595-5\" href=\"https://doi.org/10.1016%2FS0169-5347%2899%2901595-5\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 17\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BC2sbgsVGquw%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 17\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=10322508\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 17\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Heritable%20variation%20and%20evolution%20under%20favourable%20and%20unfavourable%20conditions&amp;journal=Trends%20Ecol%20Evol&amp;doi=10.1016%2FS0169-5347%2899%2901595-5&amp;volume=14&amp;pages=96-101&amp;publication_year=1999&amp;author=Hoffmann%2CAA&amp;author=Meril%C3%A4%2CJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR18\">Hosner PA, Winkler DW . (2007). Dispersal distances of tree swallows estimated from continent-wide and limited-area data. <i>J Field Ornithol</i> <b>78</b>: 290–297.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 18\" data-doi=\"10.1111/j.1557-9263.2007.00117.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1557-9263.2007.00117.x\" href=\"https://doi.org/10.1111%2Fj.1557-9263.2007.00117.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 18\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Dispersal%20distances%20of%20tree%20swallows%20estimated%20from%20continent-wide%20and%20limited-area%20data&amp;journal=J%20Field%20Ornithol&amp;doi=10.1111%2Fj.1557-9263.2007.00117.x&amp;volume=78&amp;pages=290-297&amp;publication_year=2007&amp;author=Hosner%2CPA&amp;author=Winkler%2CDW\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR19\">Houle D . (1992). Comparing evolvability and variability of quantitative traits. <i>Genetics</i> <b>130</b>: 195–204.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"CAS reference 19\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DyaK387itVGitQ%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 19\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=1732160\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 19\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1204793\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 19\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Comparing%20evolvability%20and%20variability%20of%20quantitative%20traits&amp;journal=Genetics&amp;volume=130&amp;pages=195-204&amp;publication_year=1992&amp;author=Houle%2CD\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR20\">Houle D, Meyer K . (2015). Estimating sampling error of evolutionary statistics based on genetic covariance matrices using maximum likelihood. <i>J Evol Biol</i> <b>28</b>: 1542–1549.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 20\" data-doi=\"10.1111/jeb.12674\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/jeb.12674\" href=\"https://doi.org/10.1111%2Fjeb.12674\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 20\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BC2MbjvVGguw%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 20\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=26079756\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 20\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimating%20sampling%20error%20of%20evolutionary%20statistics%20based%20on%20genetic%20covariance%20matrices%20using%20maximum%20likelihood&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fjeb.12674&amp;volume=28&amp;pages=1542-1549&amp;publication_year=2015&amp;author=Houle%2CD&amp;author=Meyer%2CK\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR21\">Husby A, Kawakami T, Rönnegård L, Smeds L, Ellegren H, Qvarnström A . (2015). Genome-wide association mapping in a wild avian population identifies a link between genetic and phenotypic variation in a life-history trait. <i>Proc R Soc B</i> <b>282</b>: 20150156.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 21\" data-doi=\"10.1098/rspb.2015.0156\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1098/rspb.2015.0156\" href=\"https://doi.org/10.1098%2Frspb.2015.0156\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 21\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=25833857\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 21\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4426624\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 21\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Genome-wide%20association%20mapping%20in%20a%20wild%20avian%20population%20identifies%20a%20link%20between%20genetic%20and%20phenotypic%20variation%20in%20a%20life-history%20trait&amp;journal=Proc%20R%20Soc%20B&amp;doi=10.1098%2Frspb.2015.0156&amp;volume=282&amp;publication_year=2015&amp;author=Husby%2CA&amp;author=Kawakami%2CT&amp;author=R%C3%B6nneg%C3%A5rd%2CL&amp;author=Smeds%2CL&amp;author=Ellegren%2CH&amp;author=Qvarnstr%C3%B6m%2CA\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR22\">Hussell DJT . (1983). Age and plumage color in female tree swallows. <i>J Field Ornithol</i> <b>54</b>: 312–318.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Google Scholar reference 22\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Age%20and%20plumage%20color%20in%20female%20tree%20swallows&amp;journal=J%20Field%20Ornithol&amp;volume=54&amp;pages=312-318&amp;publication_year=1983&amp;author=Hussell%2CDJT\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR23\">Kalinowski ST, Taper ML, Marshall TC . (2007). Revising how the computer program cervus accommodates genotyping error increases success in paternity assignment. <i>Mol Ecol</i> <b>16</b>: 1099–1106.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 23\" data-doi=\"10.1111/j.1365-294X.2007.03089.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1365-294X.2007.03089.x\" href=\"https://doi.org/10.1111%2Fj.1365-294X.2007.03089.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 23\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17305863\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 23\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Revising%20how%20the%20computer%20program%20cervus%20accommodates%20genotyping%20error%20increases%20success%20in%20paternity%20assignment&amp;journal=Mol%20Ecol&amp;doi=10.1111%2Fj.1365-294X.2007.03089.x&amp;volume=16&amp;pages=1099-1106&amp;publication_year=2007&amp;author=Kalinowski%2CST&amp;author=Taper%2CML&amp;author=Marshall%2CTC\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR24\">Kruuk LEB . (2004). Estimating genetic parameters in natural populations using the ‘animal model’. <i>Philos Trans R Soc B</i> <b>359</b>: 873–890.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 24\" data-doi=\"10.1098/rstb.2003.1437\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1098/rstb.2003.1437\" href=\"https://doi.org/10.1098%2Frstb.2003.1437\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 24\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimating%20genetic%20parameters%20in%20natural%20populations%20using%20the%20%E2%80%98animal%20model%E2%80%99&amp;journal=Philos%20Trans%20R%20Soc%20B&amp;doi=10.1098%2Frstb.2003.1437&amp;volume=359&amp;pages=873-890&amp;publication_year=2004&amp;author=Kruuk%2CLEB\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR25\">Kruuk LEB, Hadfield JD . (2007). How to separate genetic and environmental causes of similarity between relatives. <i>J Evol Biol</i> <b>20</b>: 1890–1903.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 25\" data-doi=\"10.1111/j.1420-9101.2007.01377.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1420-9101.2007.01377.x\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2007.01377.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 25\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD2svpsFGmuw%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 25\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17714306\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 25\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=How%20to%20separate%20genetic%20and%20environmental%20causes%20of%20similarity%20between%20relatives&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2007.01377.x&amp;volume=20&amp;pages=1890-1903&amp;publication_year=2007&amp;author=Kruuk%2CLEB&amp;author=Hadfield%2CJD\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR26\">Kruuk LEB, Slate J, Wilson AJ . (2008). New answers for old questions: the evolutionary quantitative genetics of wild animal populations. <i>Annu Rev Ecol Evol Syst</i> <b>39</b>: 525–548.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 26\" data-doi=\"10.1146/annurev.ecolsys.39.110707.173542\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1146/annurev.ecolsys.39.110707.173542\" href=\"https://doi.org/10.1146%2Fannurev.ecolsys.39.110707.173542\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 26\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=New%20answers%20for%20old%20questions%3A%20the%20evolutionary%20quantitative%20genetics%20of%20wild%20animal%20populations&amp;journal=Annu%20Rev%20Ecol%20Evol%20Syst&amp;doi=10.1146%2Fannurev.ecolsys.39.110707.173542&amp;volume=39&amp;pages=525-548&amp;publication_year=2008&amp;author=Kruuk%2CLEB&amp;author=Slate%2CJ&amp;author=Wilson%2CAJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR27\">Krzywinski M, Altman N . (2013). Points of significance: error bars. <i>Nat Methods</i> <b>10</b>: 921–922.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 27\" data-doi=\"10.1038/nmeth.2659\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1038/nmeth.2659\" href=\"https://doi.org/10.1038%2Fnmeth.2659\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 27\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:CAS:528:DC%2BC3sXhsFaksrzO\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 27\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=24161969\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 27\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Points%20of%20significance%3A%20error%20bars&amp;journal=Nat%20Methods&amp;doi=10.1038%2Fnmeth.2659&amp;volume=10&amp;pages=921-922&amp;publication_year=2013&amp;author=Krzywinski%2CM&amp;author=Altman%2CN\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR28\">Lande R . (1981). The minimum number of genes contributing to quantitative variation between and within populations. <i>Genetics</i> <b>99</b>: 541–553.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"CAS reference 28\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DyaL383htFOmsQ%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 28\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=7343418\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 28\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1214520\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 28\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20minimum%20number%20of%20genes%20contributing%20to%20quantitative%20variation%20between%20and%20within%20populations&amp;journal=Genetics&amp;volume=99&amp;pages=541-553&amp;publication_year=1981&amp;author=Lande%2CR\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR29\">Lande R, Arnold SJ . (1983). The measurement of selection on correlated characters. <i>Evolution</i> <b>37</b>: 1210–1226.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 29\" data-doi=\"10.1111/j.1558-5646.1983.tb00236.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1558-5646.1983.tb00236.x\" href=\"https://doi.org/10.1111%2Fj.1558-5646.1983.tb00236.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 29\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=28556011\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 29\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20measurement%20of%20selection%20on%20correlated%20characters&amp;journal=Evolution&amp;doi=10.1111%2Fj.1558-5646.1983.tb00236.x&amp;volume=37&amp;pages=1210-1226&amp;publication_year=1983&amp;author=Lande%2CR&amp;author=Arnold%2CSJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR30\">Lee C, Pollak EJ . (1997). Influence of sire misidentification on sire x year interaction variance and direct-maternal genetic covariance for weaning weight in beef cattle. <i>J Anim Sci</i> <b>75</b>: 2858–2863.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 30\" data-doi=\"10.2527/1997.75112858x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.2527/1997.75112858x\" href=\"https://doi.org/10.2527%2F1997.75112858x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 30\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:CAS:528:DyaK2sXntFemurw%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 30\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=9374297\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 30\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Influence%20of%20sire%20misidentification%20on%20sire%20x%20year%20interaction%20variance%20and%20direct-maternal%20genetic%20covariance%20for%20weaning%20weight%20in%20beef%20cattle&amp;journal=J%20Anim%20Sci&amp;doi=10.2527%2F1997.75112858x&amp;volume=75&amp;pages=2858-2863&amp;publication_year=1997&amp;author=Lee%2CC&amp;author=Pollak%2CEJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR31\">Lemons PR, Marshall TC, McCloskey SE, Sethi SA, Schmutz JA, Sedinger JS . (2015). A likelihood-based approach for assessment of extra-pair paternity and conspecific brood parasitism in natural populations. <i>Mol Ecol Resour</i> <b>7</b>: 107–116.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 31\" data-doi=\"10.1111/1755-0998.12287\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/1755-0998.12287\" href=\"https://doi.org/10.1111%2F1755-0998.12287\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 31\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=A%20likelihood-based%20approach%20for%20assessment%20of%20extra-pair%20paternity%20and%20conspecific%20brood%20parasitism%20in%20natural%20populations&amp;journal=Mol%20Ecol%20Resour&amp;doi=10.1111%2F1755-0998.12287&amp;volume=7&amp;pages=107-116&amp;publication_year=2015&amp;author=Lemons%2CPR&amp;author=Marshall%2CTC&amp;author=McCloskey%2CSE&amp;author=Sethi%2CSA&amp;author=Schmutz%2CJA&amp;author=Sedinger%2CJS\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR32\">Lessard A, Bourret A, Bélisle M, Pelletier F, Garant D . (2014). Individual and environmental determinants of reproductive success in male tree swallow (<i>Tachycineta bicolor</i>. <i>Behav Ecol Sociobiol</i> <b>68</b>: 733–742.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 32\" data-doi=\"10.1007/s00265-014-1686-y\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1007/s00265-014-1686-y\" href=\"https://doi.org/10.1007%2Fs00265-014-1686-y\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 32\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Individual%20and%20environmental%20determinants%20of%20reproductive%20success%20in%20male%20tree%20swallow%20%28Tachycineta%20bicolor&amp;journal=Behav%20Ecol%20Sociobiol&amp;doi=10.1007%2Fs00265-014-1686-y&amp;volume=68&amp;pages=733-742&amp;publication_year=2014&amp;author=Lessard%2CA&amp;author=Bourret%2CA&amp;author=B%C3%A9lisle%2CM&amp;author=Pelletier%2CF&amp;author=Garant%2CD\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR33\">Lynch M, Walsh B . (1998) <i>Genetics and Analysis of Quantitative Traits</i>. Sinauer Associates: Sunderland, MA, USA.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Google Scholar reference 33\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Genetics%20and%20Analysis%20of%20Quantitative%20Traits&amp;publication_year=1998&amp;author=Lynch%2CM&amp;author=Walsh%2CB\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR34\">Meyer K . (2008). Likelihood calculations to evaluate experimental designs to estimate genetic variances. <i>Heredity</i> <b>101</b>: 212–221.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 34\" data-doi=\"10.1038/hdy.2008.46\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1038/hdy.2008.46\" href=\"https://doi.org/10.1038%2Fhdy.2008.46\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 34\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD1crjtVSmug%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 34\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18523444\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 34\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Likelihood%20calculations%20to%20evaluate%20experimental%20designs%20to%20estimate%20genetic%20variances&amp;journal=Heredity&amp;doi=10.1038%2Fhdy.2008.46&amp;volume=101&amp;pages=212-221&amp;publication_year=2008&amp;author=Meyer%2CK\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR35\">Morrissey MB, Kruuk LEB, Wilson AJ . (2010). The danger of applying the breeder’s equation in observational studies of natural populations. <i>J Evol Biol</i> <b>23</b>: 2277–2288.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 35\" data-doi=\"10.1111/j.1420-9101.2010.02084.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1420-9101.2010.02084.x\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2010.02084.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 35\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BC3cbhtFCksQ%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 35\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=20831731\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 35\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20danger%20of%20applying%20the%20breeder%E2%80%99s%20equation%20in%20observational%20studies%20of%20natural%20populations&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2010.02084.x&amp;volume=23&amp;pages=2277-2288&amp;publication_year=2010&amp;author=Morrissey%2CMB&amp;author=Kruuk%2CLEB&amp;author=Wilson%2CAJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR36\">Morrissey MB, de Villemereuil P, Doligez B, Gimenez O . (2014). Bayesian approaches to the quantitative genetic analysis of natural populations. In: Charmantier A, Garant D <i>et al.</i> (eds) <i>Quantitative Genetics in the Wild</i>. Oxford University Press, Oxford, UK, pp 228–253.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Chapter reference 36\" data-doi=\"10.1093/acprof:oso/9780199674237.003.0014\" data-track=\"click\" data-track-action=\"chapter reference\" data-track-label=\"10.1093/acprof:oso/9780199674237.003.0014\" href=\"https://doi.org/10.1093%2Facprof%3Aoso%2F9780199674237.003.0014\" rel=\"nofollow noopener\">Chapter</a> \n",
      "    <a aria-label=\"Google Scholar reference 36\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Bayesian%20approaches%20to%20the%20quantitative%20genetic%20analysis%20of%20natural%20populations&amp;doi=10.1093%2Facprof%3Aoso%2F9780199674237.003.0014&amp;pages=228-253&amp;publication_year=2014&amp;author=Morrissey%2CMichael%20B.&amp;author=de%20Villemereuil%2CPierre&amp;author=Doligez%2CBlandine&amp;author=Gimenez%2COlivier\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR37\">Morrissey MB, Wilson AJ . (2010). PEDANTICS: an R package for pedigree-based genetic simulation and pedigree manipulation, characterization and viewing. <i>Mol Ecol Resour</i> <b>10</b>: 711–719.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 37\" data-doi=\"10.1111/j.1755-0998.2009.02817.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1755-0998.2009.02817.x\" href=\"https://doi.org/10.1111%2Fj.1755-0998.2009.02817.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 37\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=21565076\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 37\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=PEDANTICS%3A%20an%20R%20package%20for%20pedigree-based%20genetic%20simulation%20and%20pedigree%20manipulation%2C%20characterization%20and%20viewing&amp;journal=Mol%20Ecol%20Resour&amp;doi=10.1111%2Fj.1755-0998.2009.02817.x&amp;volume=10&amp;pages=711-719&amp;publication_year=2010&amp;author=Morrissey%2CMB&amp;author=Wilson%2CAJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR38\">Morrissey MB, Wilson AJ, Pemberton JM, Ferguson MM . (2007). A framework for power and sensitivity analyses for quantitative genetic studies of natural populations, and case studies in Soay sheep (<i>Ovis aries</i>. <i>J Evol Biol</i> <b>20</b>: 2309–2321.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 38\" data-doi=\"10.1111/j.1420-9101.2007.01412.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1420-9101.2007.01412.x\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2007.01412.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 38\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD2snjt1eksQ%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 38\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17956393\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 38\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=A%20framework%20for%20power%20and%20sensitivity%20analyses%20for%20quantitative%20genetic%20studies%20of%20natural%20populations%2C%20and%20case%20studies%20in%20Soay%20sheep%20%28Ovis%20aries&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2007.01412.x&amp;volume=20&amp;pages=2309-2321&amp;publication_year=2007&amp;author=Morrissey%2CMB&amp;author=Wilson%2CAJ&amp;author=Pemberton%2CJM&amp;author=Ferguson%2CMM\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR39\">Postma E . (2014). Four decades of estimating heritabilities in wild vertebrate populations: improved methods, more data, better estimates? In: Charmantier A, Garant D <i>et al.</i> (eds) <i>Quantitative Genetics in the Wild.</i> Oxford University Press, Oxford, UK, pp 16–33.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Chapter reference 39\" data-doi=\"10.1093/acprof:oso/9780199674237.003.0002\" data-track=\"click\" data-track-action=\"chapter reference\" data-track-label=\"10.1093/acprof:oso/9780199674237.003.0002\" href=\"https://doi.org/10.1093%2Facprof%3Aoso%2F9780199674237.003.0002\" rel=\"nofollow noopener\">Chapter</a> \n",
      "    <a aria-label=\"Google Scholar reference 39\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Four%20decades%20of%20estimating%20heritabilities%20in%20wild%20vertebrate%20populations%3A%20improved%20methods%2C%20more%20data%2C%20better%20estimates%3F&amp;doi=10.1093%2Facprof%3Aoso%2F9780199674237.003.0002&amp;pages=16-33&amp;publication_year=2014&amp;author=Postma%2CErik\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR40\">Quinn JL, Charmantier A, Garant D, Sheldon BC . (2006). Data depth, data completeness, and their influence on quantitative genetic estimation in two contrasting bird populations. <i>J Evol Biol</i> <b>19</b>: 994–1002.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 40\" data-doi=\"10.1111/j.1420-9101.2006.01081.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1420-9101.2006.01081.x\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2006.01081.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 40\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD283ltV2rtw%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 40\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=16674594\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 40\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Data%20depth%2C%20data%20completeness%2C%20and%20their%20influence%20on%20quantitative%20genetic%20estimation%20in%20two%20contrasting%20bird%20populations&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2006.01081.x&amp;volume=19&amp;pages=994-1002&amp;publication_year=2006&amp;author=Quinn%2CJL&amp;author=Charmantier%2CA&amp;author=Garant%2CD&amp;author=Sheldon%2CBC\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR41\">Core Team R . (2015). R: a language and environment for statistical computing <a href=\"http://www.R-project.org/\">http://www.R-project.org/</a>.</p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR42\">Via S, Lande R . (1985). Genotype-environment interaction and the evolution of phenotypic plasticity. <i>Evolution</i> <b>39</b>: 505–522.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 42\" data-doi=\"10.1111/j.1558-5646.1985.tb00391.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1558-5646.1985.tb00391.x\" href=\"https://doi.org/10.1111%2Fj.1558-5646.1985.tb00391.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 42\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=28561964\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 42\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Genotype-environment%20interaction%20and%20the%20evolution%20of%20phenotypic%20plasticity&amp;journal=Evolution&amp;doi=10.1111%2Fj.1558-5646.1985.tb00391.x&amp;volume=39&amp;pages=505-522&amp;publication_year=1985&amp;author=Via%2CS&amp;author=Lande%2CR\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR43\">de Villemereuil P, Gimenez O, Doligez B . (2013). Comparing parent–offspring regression with frequentist and Bayesian animal models to estimate heritability in wild populations: A simulation study for Gaussian and binary traits. <i>Methods Ecol Evol</i> <b>4</b>: 260–275.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 43\" data-doi=\"10.1111/2041-210X.12011\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/2041-210X.12011\" href=\"https://doi.org/10.1111%2F2041-210X.12011\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 43\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Comparing%20parent%E2%80%93offspring%20regression%20with%20frequentist%20and%20Bayesian%20animal%20models%20to%20estimate%20heritability%20in%20wild%20populations%3A%20A%20simulation%20study%20for%20Gaussian%20and%20binary%20traits&amp;journal=Methods%20Ecol%20Evol&amp;doi=10.1111%2F2041-210X.12011&amp;volume=4&amp;pages=260-275&amp;publication_year=2013&amp;author=de%20Villemereuil%2CP&amp;author=Gimenez%2CO&amp;author=Doligez%2CB\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR44\">Vinkhuyzen AAE, Wray NR, Yang J, Goddard ME, Visscher PM . (2013). Estimation and partition of heritability in human populations using whole-genome analysis methods. <i>Annu Rev Genet</i> <b>47</b>: 75–95.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 44\" data-doi=\"10.1146/annurev-genet-111212-133258\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1146/annurev-genet-111212-133258\" href=\"https://doi.org/10.1146%2Fannurev-genet-111212-133258\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 44\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:CAS:528:DC%2BC2cXhsFOjtw%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 44\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=23988118\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 44\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4037293\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 44\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Estimation%20and%20partition%20of%20heritability%20in%20human%20populations%20using%20whole-genome%20analysis%20methods&amp;journal=Annu%20Rev%20Genet&amp;doi=10.1146%2Fannurev-genet-111212-133258&amp;volume=47&amp;pages=75-95&amp;publication_year=2013&amp;author=Vinkhuyzen%2CAAE&amp;author=Wray%2CNR&amp;author=Yang%2CJ&amp;author=Goddard%2CME&amp;author=Visscher%2CPM\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR45\">Walther BA, Moore JL . (2005). The concepts of bias, precision and accuracy, and their use in testing the performance of species richness estimators, with a literature review of estimator performance. <i>Ecography</i> <b>28</b>: 815–829.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 45\" data-doi=\"10.1111/j.2005.0906-7590.04112.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.2005.0906-7590.04112.x\" href=\"https://doi.org/10.1111%2Fj.2005.0906-7590.04112.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 45\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=The%20concepts%20of%20bias%2C%20precision%20and%20accuracy%2C%20and%20their%20use%20in%20testing%20the%20performance%20of%20species%20richness%20estimators%2C%20with%20a%20literature%20review%20of%20estimator%20performance&amp;journal=Ecography&amp;doi=10.1111%2Fj.2005.0906-7590.04112.x&amp;volume=28&amp;pages=815-829&amp;publication_year=2005&amp;author=Walther%2CBA&amp;author=Moore%2CJL\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR46\">Wilson AJ . (2008). Why h<sup>2</sup> does not always equal VA/VP? <i>J Evol Biol</i> <b>21</b>: 647–650.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 46\" data-doi=\"10.1111/j.1420-9101.2008.01500.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1420-9101.2008.01500.x\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2008.01500.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 46\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BD1c3kvVKmtg%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 46\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18266683\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 46\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Why%20h2%20does%20not%20always%20equal%20VA%2FVP%3F&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2008.01500.x&amp;volume=21&amp;pages=647-650&amp;publication_year=2008&amp;author=Wilson%2CAJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR47\">Wilson AJ, Morrissey MB, Adams MJ, Walling CA, Guinness FE, Pemberton JM <i>et al</i>. (2011). Indirect genetics effects and evolutionary constraint: an analysis of social dominance in red deer, <i>Cervus elaphus</i>. <i>J Evol Biol</i> <b>24</b>: 772–783.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 47\" data-doi=\"10.1111/j.1420-9101.2010.02212.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1420-9101.2010.02212.x\" href=\"https://doi.org/10.1111%2Fj.1420-9101.2010.02212.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"CAS reference 47\" data-track=\"click\" data-track-action=\"cas reference\" data-track-label=\"link\" href=\"/articles/cas-redirect/1:STN:280:DC%2BC3M3msVCqtQ%3D%3D\" rel=\"nofollow noopener\">CAS</a> \n",
      "    <a aria-label=\"PubMed reference 47\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=21288272\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 47\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Indirect%20genetics%20effects%20and%20evolutionary%20constraint%3A%20an%20analysis%20of%20social%20dominance%20in%20red%20deer%2C%20Cervus%20elaphus&amp;journal=J%20Evol%20Biol&amp;doi=10.1111%2Fj.1420-9101.2010.02212.x&amp;volume=24&amp;pages=772-783&amp;publication_year=2011&amp;author=Wilson%2CAJ&amp;author=Morrissey%2CMB&amp;author=Adams%2CMJ&amp;author=Walling%2CCA&amp;author=Guinness%2CFE&amp;author=Pemberton%2CJM\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR48\">Wilson AJ, Réale D, Clements MN, Morrissey MM, Postma E, Walling CA <i>et al</i>. (2010). An ecologist’s guide to the animal model. <i>J Anim Ecol</i> <b>79</b>: 13–26.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 48\" data-doi=\"10.1111/j.1365-2656.2009.01639.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.1365-2656.2009.01639.x\" href=\"https://doi.org/10.1111%2Fj.1365-2656.2009.01639.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 48\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=20409158\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"Google Scholar reference 48\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=An%20ecologist%E2%80%99s%20guide%20to%20the%20animal%20model&amp;journal=J%20Anim%20Ecol&amp;doi=10.1111%2Fj.1365-2656.2009.01639.x&amp;volume=79&amp;pages=13-26&amp;publication_year=2010&amp;author=Wilson%2CAJ&amp;author=R%C3%A9ale%2CD&amp;author=Clements%2CMN&amp;author=Morrissey%2CMM&amp;author=Postma%2CE&amp;author=Walling%2CCA\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR49\">Winkler D, Hallinger K, Ardia D, Robertson R, Stutchbury B, Cohen R . (2011). Tree swallow (<i>Tachycineta bicolor</i>. Birds North Am Online. <a href=\"http://bna.birds.cornell.edu/bna/\">http://bna.birds.cornelledu/bna/</a>. Accessed on 1 Oct 2013.</p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR50\">Wolak ME . (2012). nadiv : an R package to create relatedness matrices for estimating non-additive genetic variances in animal models. <i>Methods Ecol Evol</i> <b>3</b>: 792–796.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 50\" data-doi=\"10.1111/j.2041-210X.2012.00213.x\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1111/j.2041-210X.2012.00213.x\" href=\"https://doi.org/10.1111%2Fj.2041-210X.2012.00213.x\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"Google Scholar reference 50\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=nadiv%20%3A%20an%20R%20package%20to%20create%20relatedness%20matrices%20for%20estimating%20non-additive%20genetic%20variances%20in%20animal%20models&amp;journal=Methods%20Ecol%20Evol&amp;doi=10.1111%2Fj.2041-210X.2012.00213.x&amp;volume=3&amp;pages=792-796&amp;publication_year=2012&amp;author=Wolak%2CME\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li><li class=\"c-article-references__item js-c-reading-companion-references-item\"><p class=\"c-article-references__text\" id=\"ref-CR51\">Wolak ME, Roff DA, Fairbairn DJ . (2015). Are we underestimating the genetic variances of dimorphic traits? <i>Ecol Evol</i> <b>5</b>: 590–597.</p><p class=\"c-article-references__links u-hide-print\"><a aria-label=\"Article reference 51\" data-doi=\"10.1002/ece3.1361\" data-track=\"click\" data-track-action=\"article reference\" data-track-label=\"10.1002/ece3.1361\" href=\"https://doi.org/10.1002%2Fece3.1361\" rel=\"nofollow noopener\">Article</a> \n",
      "    <a aria-label=\"PubMed reference 51\" data-track=\"click\" data-track-action=\"pubmed reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=25691983\" rel=\"nofollow noopener\">PubMed</a> \n",
      "    <a aria-label=\"PubMed Central reference 51\" data-track=\"click\" data-track-action=\"pubmed central reference\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4328764\" rel=\"nofollow noopener\">PubMed Central</a> \n",
      "    <a aria-label=\"Google Scholar reference 51\" data-track=\"click\" data-track-action=\"google scholar reference\" data-track-label=\"link\" href=\"http://scholar.google.com/scholar_lookup?&amp;title=Are%20we%20underestimating%20the%20genetic%20variances%20of%20dimorphic%20traits%3F&amp;journal=Ecol%20Evol&amp;doi=10.1002%2Fece3.1361&amp;volume=5&amp;pages=590-597&amp;publication_year=2015&amp;author=Wolak%2CME&amp;author=Roff%2CDA&amp;author=Fairbairn%2CDJ\" rel=\"nofollow noopener\">\n",
      "                    Google Scholar</a> \n",
      "                </p></li></ul><p class=\"c-article-references__download u-hide-print\"><a data-track=\"click\" data-track-action=\"download citation references\" data-track-label=\"link\" href=\"https://citation-needed.springer.com/v2/references/10.1038/hdy.2016.92?format=refman&amp;flavour=references\">Download references<svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-download\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></p></div></div></div></section></div><section data-title=\"Acknowledgements\"><div class=\"c-article-section\" id=\"Ack1-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Ack1\">Acknowledgements</h2><div class=\"c-article-section__content\" id=\"Ack1-content\"><p>We thank the 40 farm owners for providing access to their lands and all graduate students and field and laboratory assistants who have contributed in gathering data in our system over the years. We also thank Fanie Pelletier and Marc Bélisle for their long-term involvement in the southern Québec Tree swallow study. We also thank them as well as Denis Réale, Matthew Wolak, Michael Morrissey and two anonymous reviewers for constructive comments on a previous version of this manuscript. Animals were captured and handled in compliance with the Canadian Council on Animal Care, under the approval of the Université de Sherbrooke Animal Ethics Committee (Protocol Number: DG2010-01 and DG2014-01-Université de Sherbrooke). This work was supported by the Natural Sciences and Engineering Research Council of Canada (NSERC) discovery grants to DG, Fanie Pelletier and Marc Bélisle and by the Canada Research Chairs program to Fanie Pelletier and Marc Bélisle. AB was supported by a postgraduate NSERC scholarship.</p></div></div></section><section aria-labelledby=\"author-information\" data-title=\"Author information\"><div class=\"c-article-section\" id=\"author-information-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"author-information\">Author information</h2><div class=\"c-article-section__content\" id=\"author-information-content\"><h3 class=\"c-article__sub-heading\" id=\"affiliations\">Authors and Affiliations</h3><ol class=\"c-article-author-affiliation__list\"><li id=\"Aff1\"><p class=\"c-article-author-affiliation__address\">Département de Biologie, Université de Sherbrooke, Sherbrooke, Quebec, Canada</p><p class=\"c-article-author-affiliation__authors-list\">A Bourret &amp; D Garant</p></li></ol><div class=\"u-js-hide u-hide-print\" data-test=\"author-info\"><span class=\"c-article__sub-heading\">Authors</span><ol class=\"c-article-authors-search u-list-reset\"><li id=\"auth-A-Bourret\"><span class=\"c-article-authors-search__title u-h3 js-search-name\">A Bourret</span><div class=\"c-article-authors-search__list\"><div class=\"c-article-authors-search__item c-article-authors-search__list-item--left\"><a class=\"c-article-button\" data-track=\"click\" data-track-action=\"author link - publication\" data-track-label=\"link\" href=\"/search?author=A%20Bourret\" rel=\"nofollow\">View author publications</a></div><div class=\"c-article-authors-search__item c-article-authors-search__list-item--right\"><p class=\"search-in-title-js c-article-authors-search__text\">You can also search for this author in\n",
      "                        <span class=\"c-article-identifiers\"><a class=\"c-article-identifiers__item\" data-track=\"click\" data-track-action=\"author link - pubmed\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&amp;term=A%20Bourret\" rel=\"nofollow\">PubMed</a><span class=\"u-hide\"> </span><a class=\"c-article-identifiers__item\" data-track=\"click\" data-track-action=\"author link - scholar\" data-track-label=\"link\" href=\"http://scholar.google.co.uk/scholar?as_q=&amp;num=10&amp;btnG=Search+Scholar&amp;as_epq=&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=%22A%20Bourret%22&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;as_allsubj=all&amp;hl=en\" rel=\"nofollow\">Google Scholar</a></span></p></div></div></li><li id=\"auth-D-Garant\"><span class=\"c-article-authors-search__title u-h3 js-search-name\">D Garant</span><div class=\"c-article-authors-search__list\"><div class=\"c-article-authors-search__item c-article-authors-search__list-item--left\"><a class=\"c-article-button\" data-track=\"click\" data-track-action=\"author link - publication\" data-track-label=\"link\" href=\"/search?author=D%20Garant\" rel=\"nofollow\">View author publications</a></div><div class=\"c-article-authors-search__item c-article-authors-search__list-item--right\"><p class=\"search-in-title-js c-article-authors-search__text\">You can also search for this author in\n",
      "                        <span class=\"c-article-identifiers\"><a class=\"c-article-identifiers__item\" data-track=\"click\" data-track-action=\"author link - pubmed\" data-track-label=\"link\" href=\"http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&amp;term=D%20Garant\" rel=\"nofollow\">PubMed</a><span class=\"u-hide\"> </span><a class=\"c-article-identifiers__item\" data-track=\"click\" data-track-action=\"author link - scholar\" data-track-label=\"link\" href=\"http://scholar.google.co.uk/scholar?as_q=&amp;num=10&amp;btnG=Search+Scholar&amp;as_epq=&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=%22D%20Garant%22&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;as_allsubj=all&amp;hl=en\" rel=\"nofollow\">Google Scholar</a></span></p></div></div></li></ol></div><h3 class=\"c-article__sub-heading\" id=\"corresponding-author\">Corresponding author</h3><p id=\"corresponding-author-list\">Correspondence to\n",
      "                <a href=\"mailto:Audrey.Bourret@USherbrooke.ca\" id=\"corresp-c1\">A Bourret</a>.</p></div></div></section><section data-title=\"Ethics declarations\"><div class=\"c-article-section\" id=\"ethics-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"ethics\">Ethics declarations</h2><div class=\"c-article-section__content\" id=\"ethics-content\">\n",
      "<h3 class=\"c-article__sub-heading\">Competing interests</h3>\n",
      "<p>The authors declare no conflict of interest.</p>\n",
      "</div></div></section><section data-title=\"Additional information\"><div class=\"c-article-section\" id=\"additional-information-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"additional-information\">Additional information</h2><div class=\"c-article-section__content\" id=\"additional-information-content\"><p><a data-track=\"click\" data-track-action=\"supplementary material anchor\" data-track-label=\"link\" href=\"/articles/hdy201692#MOESM34\">Supplementary Information</a> accompanies this paper on Heredity website</p></div></div></section><section data-title=\"Supplementary information\"><div class=\"c-article-section\" id=\"Sec17-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"Sec17\">Supplementary information</h2><div class=\"c-article-section__content\" id=\"Sec17-content\"><div data-test=\"supplementary-info\"><div class=\"c-article-figshare-container\" data-test=\"figshare-container\" id=\"figshareContainer\"></div><div class=\"c-article-supplementary__item\" data-test=\"supp-item\" id=\"MOESM34\"><h3 class=\"c-article-supplementary__title u-h3\"><a class=\"print-link\" data-supp-info-image=\"\" data-test=\"supp-info-link\" data-track=\"click\" data-track-action=\"view supplementary info\" data-track-label=\"link\" href=\"https://static-content.springer.com/esm/art%3A10.1038%2Fhdy.2016.92/MediaObjects/41437_2017_BFhdy201692_MOESM34_ESM.doc\">Supplementary Information (DOC 711 kb)</a></h3></div></div></div></div></section><section data-title=\"Rights and permissions\"><div class=\"c-article-section\" id=\"rightslink-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"rightslink\">Rights and permissions</h2><div class=\"c-article-section__content\" id=\"rightslink-content\"><p class=\"c-article-rights\"><a data-track=\"click\" data-track-action=\"view rights and permissions\" data-track-label=\"link\" href=\"https://s100.copyright.com/AppDispatchServlet?title=An%20assessment%20of%20the%20reliability%20of%20quantitative%20genetics%20estimates%20in%20study%20systems%20with%20high%20rate%20of%20extra-pair%20reproduction%20and%20low%20recruitment&amp;author=A%20Bourret%20et%20al&amp;contentID=10.1038%2Fhdy.2016.92&amp;copyright=The%20Genetics%20Society%20Macmillan%20Publishers%20Limited%2C%20part%20of%20Springer%20Nature.&amp;publication=0018-067X&amp;publicationDate=2016-10-26&amp;publisherName=SpringerNature&amp;orderBeanReset=true\">Reprints and Permissions</a></p></div></div></section><section aria-labelledby=\"article-info\" data-title=\"About this article\"><div class=\"c-article-section\" id=\"article-info-section\"><h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"article-info\">About this article</h2><div class=\"c-article-section__content\" id=\"article-info-content\"><div class=\"c-bibliographic-information\"><div class=\"u-hide-print c-bibliographic-information__column c-bibliographic-information__column--border\"><a data-crossmark=\"10.1038/hdy.2016.92\" data-test=\"crossmark\" data-track=\"click\" data-track-action=\"Click Crossmark\" data-track-label=\"link\" href=\"https://crossmark.crossref.org/dialog/?doi=10.1038/hdy.2016.92\" rel=\"noopener\" target=\"_blank\"><img alt=\"Verify currency and authenticity via CrossMark\" height=\"81\" src=\"data:image/svg+xml;base64,<svg height="81" width="57" xmlns="http://www.w3.org/2000/svg"><g fill="none" fill-rule="evenodd"><path d="m17.35 35.45 21.3-14.2v-17.03h-21.3" fill="#989898"/><path d="m38.65 35.45-21.3-14.2v-17.03h21.3" fill="#747474"/><path d="m28 .5c-12.98 0-23.5 10.52-23.5 23.5s10.52 23.5 23.5 23.5 23.5-10.52 23.5-23.5c0-6.23-2.48-12.21-6.88-16.62-4.41-4.4-10.39-6.88-16.62-6.88zm0 41.25c-9.8 0-17.75-7.95-17.75-17.75s7.95-17.75 17.75-17.75 17.75 7.95 17.75 17.75c0 4.71-1.87 9.22-5.2 12.55s-7.84 5.2-12.55 5.2z" fill="#535353"/><path d="m41 36c-5.81 6.23-15.23 7.45-22.43 2.9-7.21-4.55-10.16-13.57-7.03-21.5l-4.92-3.11c-4.95 10.7-1.19 23.42 8.78 29.71 9.97 6.3 23.07 4.22 30.6-4.86z" fill="#9c9c9c"/><path d="m.2 58.45c0-.75.11-1.42.33-2.01s.52-1.09.91-1.5c.38-.41.83-.73 1.34-.94.51-.22 1.06-.32 1.65-.32.56 0 1.06.11 1.51.35.44.23.81.5 1.1.81l-.91 1.01c-.24-.24-.49-.42-.75-.56-.27-.13-.58-.2-.93-.2-.39 0-.73.08-1.05.23-.31.16-.58.37-.81.66-.23.28-.41.63-.53 1.04-.13.41-.19.88-.19 1.39 0 1.04.23 1.86.68 2.46.45.59 1.06.88 1.84.88.41 0 .77-.07 1.07-.23s.59-.39.85-.68l.91 1c-.38.43-.8.76-1.28.99-.47.22-1 .34-1.58.34-.59 0-1.13-.1-1.64-.31-.5-.2-.94-.51-1.31-.91-.38-.4-.67-.9-.88-1.48-.22-.59-.33-1.26-.33-2.02zm8.4-5.33h1.61v2.54l-.05 1.33c.29-.27.61-.51.96-.72s.76-.31 1.24-.31c.73 0 1.27.23 1.61.71.33.47.5 1.14.5 2.02v4.31h-1.61v-4.1c0-.57-.08-.97-.25-1.21-.17-.23-.45-.35-.83-.35-.3 0-.56.08-.79.22-.23.15-.49.36-.78.64v4.8h-1.61zm7.37 6.45c0-.56.09-1.06.26-1.51.18-.45.42-.83.71-1.14.29-.3.63-.54 1.01-.71.39-.17.78-.25 1.18-.25.47 0 .88.08 1.23.24.36.16.65.38.89.67s.42.63.54 1.03c.12.41.18.84.18 1.32 0 .32-.02.57-.07.76h-4.36c.07.62.29 1.1.65 1.44.36.33.82.5 1.38.5.29 0 .57-.04.83-.13s.51-.21.76-.37l.55 1.01c-.33.21-.69.39-1.09.53-.41.14-.83.21-1.26.21-.48 0-.92-.08-1.34-.25-.41-.16-.76-.4-1.07-.7-.31-.31-.55-.69-.72-1.13-.18-.44-.26-.95-.26-1.52zm4.6-.62c0-.55-.11-.98-.34-1.28-.23-.31-.58-.47-1.06-.47-.41 0-.77.15-1.07.45-.31.29-.5.73-.58 1.3zm2.5.62c0-.57.09-1.08.28-1.53.18-.44.43-.82.75-1.13s.69-.54 1.1-.71c.42-.16.85-.24 1.31-.24.45 0 .84.08 1.17.23s.61.34.85.57l-.77 1.02c-.19-.16-.38-.28-.56-.37-.19-.09-.39-.14-.61-.14-.56 0-1.01.21-1.35.63-.35.41-.52.97-.52 1.67 0 .69.17 1.24.51 1.66.34.41.78.62 1.32.62.28 0 .54-.06.78-.17.24-.12.45-.26.64-.42l.67 1.03c-.33.29-.69.51-1.08.65-.39.15-.78.23-1.18.23-.46 0-.9-.08-1.31-.24-.4-.16-.75-.39-1.05-.7s-.53-.69-.7-1.13c-.17-.45-.25-.96-.25-1.53zm6.91-6.45h1.58v6.17h.05l2.54-3.16h1.77l-2.35 2.8 2.59 4.07h-1.75l-1.77-2.98-1.08 1.23v1.75h-1.58zm13.69 1.27c-.25-.11-.5-.17-.75-.17-.58 0-.87.39-.87 1.16v.75h1.34v1.27h-1.34v5.6h-1.61v-5.6h-.92v-1.2l.92-.07v-.72c0-.35.04-.68.13-.98.08-.31.21-.57.4-.79s.42-.39.71-.51c.28-.12.63-.18 1.04-.18.24 0 .48.02.69.07.22.05.41.1.57.17zm.48 5.18c0-.57.09-1.08.27-1.53.17-.44.41-.82.72-1.13.3-.31.65-.54 1.04-.71.39-.16.8-.24 1.23-.24s.84.08 1.24.24c.4.17.74.4 1.04.71s.54.69.72 1.13c.19.45.28.96.28 1.53s-.09 1.08-.28 1.53c-.18.44-.42.82-.72 1.13s-.64.54-1.04.7-.81.24-1.24.24-.84-.08-1.23-.24-.74-.39-1.04-.7c-.31-.31-.55-.69-.72-1.13-.18-.45-.27-.96-.27-1.53zm1.65 0c0 .69.14 1.24.43 1.66.28.41.68.62 1.18.62.51 0 .9-.21 1.19-.62.29-.42.44-.97.44-1.66 0-.7-.15-1.26-.44-1.67-.29-.42-.68-.63-1.19-.63-.5 0-.9.21-1.18.63-.29.41-.43.97-.43 1.67zm6.48-3.44h1.33l.12 1.21h.05c.24-.44.54-.79.88-1.02.35-.24.7-.36 1.07-.36.32 0 .59.05.78.14l-.28 1.4-.33-.09c-.11-.01-.23-.02-.38-.02-.27 0-.56.1-.86.31s-.55.58-.77 1.1v4.2h-1.61zm-47.87 15h1.61v4.1c0 .57.08.97.25 1.2.17.24.44.35.81.35.3 0 .57-.07.8-.22.22-.15.47-.39.73-.73v-4.7h1.61v6.87h-1.32l-.12-1.01h-.04c-.3.36-.63.64-.98.86-.35.21-.76.32-1.24.32-.73 0-1.27-.24-1.61-.71-.33-.47-.5-1.14-.5-2.02zm9.46 7.43v2.16h-1.61v-9.59h1.33l.12.72h.05c.29-.24.61-.45.97-.63.35-.17.72-.26 1.1-.26.43 0 .81.08 1.15.24.33.17.61.4.84.71.24.31.41.68.53 1.11.13.42.19.91.19 1.44 0 .59-.09 1.11-.25 1.57-.16.47-.38.85-.65 1.16-.27.32-.58.56-.94.73-.35.16-.72.25-1.1.25-.3 0-.6-.07-.9-.2s-.59-.31-.87-.56zm0-2.3c.26.22.5.37.73.45.24.09.46.13.66.13.46 0 .84-.2 1.15-.6.31-.39.46-.98.46-1.77 0-.69-.12-1.22-.35-1.61-.23-.38-.61-.57-1.13-.57-.49 0-.99.26-1.52.77zm5.87-1.69c0-.56.08-1.06.25-1.51.16-.45.37-.83.65-1.14.27-.3.58-.54.93-.71s.71-.25 1.08-.25c.39 0 .73.07 1 .2.27.14.54.32.81.55l-.06-1.1v-2.49h1.61v9.88h-1.33l-.11-.74h-.06c-.25.25-.54.46-.88.64-.33.18-.69.27-1.06.27-.87 0-1.56-.32-2.07-.95s-.76-1.51-.76-2.65zm1.67-.01c0 .74.13 1.31.4 1.7.26.38.65.58 1.15.58.51 0 .99-.26 1.44-.77v-3.21c-.24-.21-.48-.36-.7-.45-.23-.08-.46-.12-.7-.12-.45 0-.82.19-1.13.59-.31.39-.46.95-.46 1.68zm6.35 1.59c0-.73.32-1.3.97-1.71.64-.4 1.67-.68 3.08-.84 0-.17-.02-.34-.07-.51-.05-.16-.12-.3-.22-.43s-.22-.22-.38-.3c-.15-.06-.34-.1-.58-.1-.34 0-.68.07-1 .2s-.63.29-.93.47l-.59-1.08c.39-.24.81-.45 1.28-.63.47-.17.99-.26 1.54-.26.86 0 1.51.25 1.93.76s.63 1.25.63 2.21v4.07h-1.32l-.12-.76h-.05c-.3.27-.63.48-.98.66s-.73.27-1.14.27c-.61 0-1.1-.19-1.48-.56-.38-.36-.57-.85-.57-1.46zm1.57-.12c0 .3.09.53.27.67.19.14.42.21.71.21.28 0 .54-.07.77-.2s.48-.31.73-.56v-1.54c-.47.06-.86.13-1.18.23-.31.09-.57.19-.76.31s-.33.25-.41.4c-.09.15-.13.31-.13.48zm6.29-3.63h-.98v-1.2l1.06-.07.2-1.88h1.34v1.88h1.75v1.27h-1.75v3.28c0 .8.32 1.2.97 1.2.12 0 .24-.01.37-.04.12-.03.24-.07.34-.11l.28 1.19c-.19.06-.4.12-.64.17-.23.05-.49.08-.76.08-.4 0-.74-.06-1.02-.18-.27-.13-.49-.3-.67-.52-.17-.21-.3-.48-.37-.78-.08-.3-.12-.64-.12-1.01zm4.36 2.17c0-.56.09-1.06.27-1.51s.41-.83.71-1.14c.29-.3.63-.54 1.01-.71.39-.17.78-.25 1.18-.25.47 0 .88.08 1.23.24.36.16.65.38.89.67s.42.63.54 1.03c.12.41.18.84.18 1.32 0 .32-.02.57-.07.76h-4.37c.08.62.29 1.1.65 1.44.36.33.82.5 1.38.5.3 0 .58-.04.84-.13.25-.09.51-.21.76-.37l.54 1.01c-.32.21-.69.39-1.09.53s-.82.21-1.26.21c-.47 0-.92-.08-1.33-.25-.41-.16-.77-.4-1.08-.7-.3-.31-.54-.69-.72-1.13-.17-.44-.26-.95-.26-1.52zm4.61-.62c0-.55-.11-.98-.34-1.28-.23-.31-.58-.47-1.06-.47-.41 0-.77.15-1.08.45-.31.29-.5.73-.57 1.3zm3.01 2.23c.31.24.61.43.92.57.3.13.63.2.98.2.38 0 .65-.08.83-.23s.27-.35.27-.6c0-.14-.05-.26-.13-.37-.08-.1-.2-.2-.34-.28-.14-.09-.29-.16-.47-.23l-.53-.22c-.23-.09-.46-.18-.69-.3-.23-.11-.44-.24-.62-.4s-.33-.35-.45-.55c-.12-.21-.18-.46-.18-.75 0-.61.23-1.1.68-1.49.44-.38 1.06-.57 1.83-.57.48 0 .91.08 1.29.25s.71.36.99.57l-.74.98c-.24-.17-.49-.32-.73-.42-.25-.11-.51-.16-.78-.16-.35 0-.6.07-.76.21-.17.15-.25.33-.25.54 0 .14.04.26.12.36s.18.18.31.26c.14.07.29.14.46.21l.54.19c.23.09.47.18.7.29s.44.24.64.4c.19.16.34.35.46.58.11.23.17.5.17.82 0 .3-.06.58-.17.83-.12.26-.29.48-.51.68-.23.19-.51.34-.84.45-.34.11-.72.17-1.15.17-.48 0-.95-.09-1.41-.27-.46-.19-.86-.41-1.2-.68z" fill="#535353"/></g></svg>\" width=\"57\"/></a></div><div class=\"c-bibliographic-information__column\"><h3 class=\"c-article__sub-heading\" id=\"citeas\">Cite this article</h3><p class=\"c-bibliographic-information__citation\">Bourret, A., Garant, D. An assessment of the reliability of quantitative genetics estimates in study systems with high rate of extra-pair reproduction and low recruitment.\n",
      "                    <i>Heredity</i> <b>118</b>, 229–238 (2017). https://doi.org/10.1038/hdy.2016.92</p><p class=\"c-bibliographic-information__download-citation u-hide-print\"><a data-test=\"citation-link\" data-track=\"click\" data-track-action=\"download article citation\" data-track-external=\"\" data-track-label=\"link\" href=\"https://citation-needed.springer.com/v2/references/10.1038/hdy.2016.92?format=refman&amp;flavour=citation\">Download citation<svg aria-hidden=\"true\" class=\"u-icon\" focusable=\"false\" height=\"16\" role=\"img\" width=\"16\"><use xlink:href=\"#global-icon-download\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"></use></svg></a></p><ul class=\"c-bibliographic-information__list\" data-test=\"publication-history\"><li class=\"c-bibliographic-information__list-item\"><p>Received<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2016-02-24\">24 February 2016</time></span></p></li><li class=\"c-bibliographic-information__list-item\"><p>Revised<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2016-06-26\">26 June 2016</time></span></p></li><li class=\"c-bibliographic-information__list-item\"><p>Accepted<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2016-08-19\">19 August 2016</time></span></p></li><li class=\"c-bibliographic-information__list-item\"><p>Published<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2016-10-26\">26 October 2016</time></span></p></li><li class=\"c-bibliographic-information__list-item\"><p>Issue Date<span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\"><time datetime=\"2017-03\">March 2017</time></span></p></li><li class=\"c-bibliographic-information__list-item c-bibliographic-information__list-item--doi\"><p><abbr title=\"Digital Object Identifier\">DOI</abbr><span class=\"u-hide\">: </span><span class=\"c-bibliographic-information__value\">https://doi.org/10.1038/hdy.2016.92</span></p></li></ul><div data-component=\"share-box\"><div class=\"c-article-share-box u-display-none\" hidden=\"\"><h3 class=\"c-article__sub-heading\">Share this article</h3><p class=\"c-article-share-box__description\">Anyone you share the following link with will be able to read this content:</p><button class=\"js-get-share-url c-article-share-box__button\" data-track=\"click\" data-track-action=\"get shareable link\" data-track-external=\"\" data-track-label=\"button\" id=\"get-share-url\">Get shareable link</button><div class=\"js-no-share-url-container u-display-none\" hidden=\"\"><p class=\"js-c-article-share-box__no-sharelink-info c-article-share-box__no-sharelink-info\">Sorry, a shareable link is not currently available for this article.</p></div><div class=\"js-share-url-container u-display-none\" hidden=\"\"><p class=\"js-share-url c-article-share-box__only-read-input\" data-track=\"click\" data-track-action=\"select share url\" data-track-label=\"button\" id=\"share-url\"></p><button class=\"js-copy-share-url c-article-share-box__button--link-like\" data-track=\"click\" data-track-action=\"copy share url\" data-track-external=\"\" data-track-label=\"button\" id=\"copy-share-url\">Copy to clipboard</button></div><p class=\"js-c-article-share-box__additional-info c-article-share-box__additional-info\">\n",
      "                            Provided by the Springer Nature SharedIt content-sharing initiative\n",
      "                        </p></div></div><div data-component=\"article-info-list\"></div></div></div></div></div></section>\n",
      "</div>\n",
      "<section>\n",
      "<div class=\"c-article-section js-article-section\" id=\"further-reading-section\">\n",
      "<h2 class=\"c-article-section__title js-section-title js-c-reading-companion-sections-item\" id=\"further-reading\">This article is cited by</h2>\n",
      "<div class=\"c-article-section__content js-collapsible-section\" id=\"further-reading-content\">\n",
      "<ul class=\"c-article-further-reading__list\" id=\"further-reading-list\">\n",
      "<li class=\"c-article-further-reading__item js-ref-item\">\n",
      "<h3 class=\"c-article-further-reading__title\">\n",
      "<a class=\"print-link\" data-track=\"click\" data-track-action=\"view further reading article\" data-track-label=\"link:Heritability of plumage colour morph variation in a wild population of promiscuous, long-lived Australian magpies\" href=\"https://doi.org/10.1038/s41437-019-0212-4\">\n",
      "                                        Heritability of plumage colour morph variation in a wild population of promiscuous, long-lived Australian magpies\n",
      "                                    </a>\n",
      "</h3>\n",
      "<ul class=\"c-author-list c-author-list--compact u-sans-serif u-mb-4 u-mt-auto\" data-test=\"author-list\">\n",
      "<li>Ana E. Dobson</li><li>Daniel J. Schmidt</li><li>Jane M. Hughes</li>\n",
      "</ul>\n",
      "<p class=\"c-article-further-reading__journal-title\"><i>Heredity</i> (2019)</p>\n",
      "</li>\n",
      "</ul>\n",
      "</div>\n",
      "</div>\n",
      "</section>\n",
      "<span data-recommended=\"jobs\"></span>\n",
      "</div>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for job_element in results2:\n",
    "    print(job_element, end=\"\\n\"*2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "5c33888d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import regex as re\n",
    "import string\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from matplotlib.gridspec import GridSpec\n",
    "np.random.seed(42)\n",
    "from nltk import wordpunct_tokenize, pos_tag, WordNetLemmatizer\n",
    "\n",
    "def get_wordnet_pos(tag):\n",
    "\n",
    "    # adjectifs\n",
    "    if tag.startswith('J'):\n",
    "        return \"a\"\n",
    "    #verbes\n",
    "    elif tag.startswith('V'):\n",
    "        return \"v\"\n",
    "    #noms\n",
    "    elif tag.startswith('N'):\n",
    "        return \"n\"\n",
    "    #adverbes\n",
    "    elif tag.startswith('R'):\n",
    "        return \"r\"\n",
    "    else:\n",
    "        return ''\n",
    "\n",
    "\n",
    "def TextProcessing(text , lemmatisation = False , selec_pos = False , stopwords = False, remove_punc = False, output_joined = False, lowercase = False):\n",
    "\n",
    "    TextProcessed = wordpunct_tokenize(text)\n",
    "    if lemmatisation == 'all':\n",
    "        lemmatisation = ['N' , 'V', 'J', 'R']\n",
    "    # Stopwords\n",
    "    if stopwords:\n",
    "        from stop_words import get_stop_words\n",
    "        stopWords = get_stop_words('english')  \n",
    "        TextFiltered = []\n",
    "        for word in TextProcessed:\n",
    "            if word not in stopWords:\n",
    "                TextFiltered.append(word)      \n",
    "        TextProcessed =  TextFiltered  \n",
    "    \n",
    "    \n",
    "    # POS-tag\n",
    "    if (lemmatisation is not False) or (selec_pos is not False):\n",
    "        TextTagged = pos_tag(TextProcessed)\n",
    "\n",
    "        CleanTextTagged = []\n",
    "        for word_tagged in enumerate(TextTagged):\n",
    "            word = word_tagged[1][0]\n",
    "            tag = word_tagged[1][1]\n",
    "            if re.search(r'^(\\W+)', tag) and word != tag:\n",
    "                pass\n",
    "            else:\n",
    "                CleanTextTagged.append((word , tag))  \n",
    "        TextProcessed = CleanTextTagged\n",
    "        \n",
    "    # Selection des mots selon leur POS\n",
    "    if selec_pos is not False:\n",
    "        CleanTextTaggedSelec = []\n",
    "        for word in TextProcessed:\n",
    "            tag = word[1]\n",
    "            if tag[0] in selec_pos:\n",
    "                CleanTextTaggedSelec.append(word)\n",
    "            else:\n",
    "                pass\n",
    "        if lemmatisation is False:\n",
    "            TextProcessed = [ word_tagged[0] for word_tagged in CleanTextTaggedSelec ]\n",
    "        else:\n",
    "            TextProcessed = CleanTextTaggedSelec\n",
    "  \n",
    "    #Lemmatisation\n",
    "    if lemmatisation is not False: \n",
    "        lemmatizer = WordNetLemmatizer()\n",
    "        TextLemm = []       \n",
    "        for word in TextProcessed:\n",
    "            tag = word[1]\n",
    "            if tag[0] in lemmatisation:\n",
    "                word_lemm = lemmatizer.lemmatize(word[0] , pos = get_wordnet_pos(tag))\n",
    "                TextLemm.append(word_lemm)\n",
    "            else:\n",
    "                TextLemm.append(word[0])\n",
    "        TextProcessed = TextLemm\n",
    "    \n",
    "   \n",
    "    # ponctuation\n",
    "    if remove_punc is True:\n",
    "        TextProcessed = [token for token in TextProcessed if token not in string.punctuation]\n",
    "    \n",
    "    # lowercase\n",
    "    if lowercase is True:\n",
    "        TextProcessed = [token.lower() for token in TextProcessed]\n",
    "               \n",
    "    if output_joined is True:\n",
    "        TextProcessed = ' '.join(TextProcessed)\n",
    "        \n",
    "    \n",
    "    return(TextProcessed)\n",
    "    \n",
    "\n",
    "from sklearn.model_selection import KFold, train_test_split, StratifiedKFold\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer, CountVectorizer, TfidfTransformer\n",
    "from sklearn.metrics import classification_report, confusion_matrix, accuracy_score, cohen_kappa_score, f1_score, recall_score, precision_score\n",
    "from sklearn.svm import LinearSVC\n",
    "from sklearn.preprocessing import LabelEncoder\n",
    "from sklearn.neural_network import MLPClassifier\n",
    "from sklearn.utils.multiclass import unique_labels\n",
    "from sklearn.naive_bayes import ComplementNB\n",
    "\n",
    "def identity(words):\n",
    "    return words\n",
    "\n",
    "\n",
    "\n",
    "def evaluate(all_tokens, all_idx, all_labels, n_folds, vec, clf,  multi = True, print_errors = True, formated_output = False):\n",
    "\n",
    "    kf = StratifiedKFold(n_splits=n_folds)\n",
    "    fold_scores = {}\n",
    "    errors = []\n",
    "    y_true_tot , y_pred_tot, y_confidence, ids = [] , [], [], []\n",
    "   \n",
    "    le = LabelEncoder()\n",
    "    le.fit(all_labels)\n",
    "    for fold_num, (train_idx, test_idx) in enumerate(kf.split(all_idx, all_labels)):\n",
    "        print('{:-^80}'.format(\" FOLD %d [%d/%d]\" % (fold_num + 1, len(train_idx), len(test_idx))))\n",
    "        train_tokens, test_tokens = [all_tokens[i] for i in train_idx], [all_tokens[i] for i in test_idx]\n",
    "        train_labels, test_labels = [all_labels[i] for i in train_idx], [all_labels[i] for i in test_idx]\n",
    "\n",
    "                   \n",
    "        if multi:\n",
    "            \n",
    "            train_labels = le.transform(train_labels)\n",
    "            test_labels = le.transform(test_labels)   \n",
    "            \n",
    "            \n",
    "        X_train = vec.fit_transform(train_tokens).toarray()\n",
    "        X_test = vec.transform(test_tokens).toarray()\n",
    "        \n",
    "      #  X_train = vec.fit_transform(train_tokens)\n",
    "      #  X_test = vec.transform(test_tokens)\n",
    "        \n",
    "        \n",
    "        clf.fit(X_train, train_labels)\n",
    "        y_pred = clf.predict(X_test)\n",
    "        \n",
    "        if hasattr(clf, 'predict_proba'):\n",
    "            y_pred_confidence = np.max(clf.predict_proba(X_test), axis=1)\n",
    "        elif hasattr(clf, 'decision_function'):\n",
    "            y_pred_confidence = clf.decision_function(X_test)\n",
    "        else:\n",
    "            y_pred_confidence = np.zeros(len(y_test))\n",
    "\n",
    "\n",
    "        fold_scores.setdefault('accuracy', []).append(accuracy_score(test_labels, y_pred))\n",
    "        fold_scores.setdefault('f1_score', []).append(f1_score(test_labels, y_pred, average='weighted', labels=np.unique(y_pred)))\n",
    "        fold_scores.setdefault('precision', []).append(recall_score(test_labels, y_pred, average='weighted', labels=np.unique(y_pred)))\n",
    "        fold_scores.setdefault('recall', []).append(precision_score(test_labels, y_pred, average='weighted', labels=np.unique(y_pred)))\n",
    "    \n",
    "        #errors.extend([(test_idx[i], le.inverse_transform(actual), le.inverse_transform(pred), confidence) for i, (actual, pred, confidence) in enumerate(zip(test_labels, y_pred, y_pred_confidence)) if actual != pred])\n",
    "        y_true_tot = y_true_tot + [le.inverse_transform([y])[0] for y in test_labels]\n",
    "        y_pred_tot = y_pred_tot + [le.inverse_transform([y])[0] for y in y_pred]\n",
    "        y_confidence = y_confidence + list(y_pred_confidence)\n",
    "        ids = ids + [i for i in test_idx]\n",
    "    fold_scores = {mname: (np.mean(vals), np.std(vals)) for mname, vals in fold_scores.items()} \n",
    "    print('-' * 80)\n",
    "    classes = unique_labels(y_true_tot, y_pred_tot)\n",
    "    for mname, (avg, std) in fold_scores.items():\n",
    "        print(\"%s: %.2f (+/-%.2f)\" % (mname, avg, std))  \n",
    "    \n",
    "    classif_report = classification_report(y_true_tot, y_pred_tot, target_names =  classes, digits=2)\n",
    "    print(classif_report)    \n",
    "    \n",
    "    if print_errors:\n",
    "        print('-' * 80)\n",
    "        print(\"Errors: %d\" % len(errors))\n",
    "        for doc_id, actual, pred, confidence in errors:\n",
    "            print(\"%20s %10s %10s \" % (doc_id, actual, pred))\n",
    "            #print(\"%20s %10s %10s (%.2f)\" % (doc_id, actual, pred, confidence))\n",
    "    if formated_output:\n",
    "        \n",
    "        cm = confusion_matrix(y_true_tot, y_pred_tot, labels=classes)\n",
    "        cm = pd.DataFrame(cm, columns = classes, index = classes)\n",
    "\n",
    "        return({'conf_matrix' : cm, 'scores' : classif_report, 'ids' : ids, 'pred_labels' : y_pred_tot, 'pred_conf': y_confidence\n",
    "                   })   \n",
    "    \n",
    "class TfidfEmbeddingVectorizer(object):\n",
    "    def __init__(self, vectors, oov=None):\n",
    "        self.vectors = vectors\n",
    "        self.term2weight = None\n",
    "        self.oov = oov\n",
    "        if self.oov is not None and self.oov not in self.vectors:\n",
    "            raise ValueError(\"OOV is not in vocabulary\")\n",
    "\n",
    "    @property\n",
    "    def dim(self):\n",
    "        return self.vectors.shape[1]\n",
    "\n",
    "    def _get_term_weight(self, t):\n",
    "        if t in self.term2weight:\n",
    "            return self.term2weight[t]\n",
    "        return max(self.tfidf.idf_)\n",
    "    \n",
    "    def get_token(self, t):\n",
    "        candidates = [t, t.lower(), t.title(), t.upper()]\n",
    "        for c in candidates:\n",
    "            if c in self.vectors:\n",
    "                return c\n",
    "        return self.oov\n",
    "        \n",
    "    \n",
    "    def preprocess(self, X):\n",
    "        Xt = [[self.get_token(t) for t in terms] for terms in X]\n",
    "        Xt = [[t for t in terms if t is not None] for terms in Xt]\n",
    "        return Xt\n",
    "    \n",
    "    def fit(self, X, y=None):\n",
    "        self.tfidf = TfidfVectorizer(analyzer=lambda x: x)\n",
    "        self.tfidf.fit(self.preprocess(X))\n",
    "\n",
    "        self.apply_tfidf()\n",
    "        return self\n",
    "\n",
    "    def apply_tfidf(self):\n",
    "        # if a term was never seen - it must be at least as infrequent\n",
    "        # as any of the known terms - so the default idf is the max of\n",
    "        # known idf's\n",
    "        self.term2weight = {t: self.tfidf.idf_[i]\n",
    "                            for t, i in self.tfidf.vocabulary_.items()}\n",
    "\n",
    "    def transform(self, X):\n",
    "        Xt = self.preprocess(X)\n",
    "        return np.array([\n",
    "            np.mean([self.vectors[t] * self._get_term_weight(t)\n",
    "                     for t in terms], axis=0)\n",
    "            for terms in Xt\n",
    "        ])\n",
    "\n",
    "    def fit_transform(self, X):\n",
    "        return self.fit(X).transform(X)    \n",
    "    \n",
    "    \n",
    "    \n",
    "def cleanhtml(raw_html, remove_punc = False, lower = False):\n",
    "    \"\"\"\n",
    "    Replace HTML tags in a text.\n",
    "\n",
    "    raw_html : str\n",
    "        html in its raw form\n",
    "    \"\"\"\n",
    "    \n",
    "    html_tag = re.compile('<.*?>')\n",
    "    html_tag_2 = re.compile('\\[\\d\\]')\n",
    "    url_pattern = re.compile('www.\\S+.com')\n",
    "    \n",
    "    cleantext = re.sub(html_tag, ' ', raw_html)\n",
    "    cleantext = re.sub(html_tag_2, '', cleantext)\n",
    "    cleantext = re.sub(url_pattern, '', cleantext)\n",
    "    \n",
    "    toRemove = ['\\n', '\\t' ,'âœ', 'ã', 'â' ,'€', \"\\'\", \"\\xa0\", '(' , ')' , '[', ']', ',']\n",
    "    for i in toRemove :\n",
    "        cleantext = cleantext.replace(i, \"\")\n",
    "        \n",
    "    toRemove = ['Â']\n",
    "    for i in toRemove :\n",
    "        cleantext = cleantext.replace(i, \" \")\n",
    "    \n",
    "    #add whitespace after a dot \n",
    "    rx = r\"\\.(?=\\S)\"\n",
    "    cleantext = re.sub(rx, \". \", cleantext)\n",
    "    \n",
    "    if remove_punc:\n",
    "        cleantext = re.sub('[^A-Za-z0-9]+', ' ', cleantext)\n",
    "        \n",
    "    if lower:\n",
    "        cleantext = cleantext.lower()\n",
    "    \n",
    "    return cleantext.strip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "f431cac3",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_species[\"full_text\"] = df_species['title'] + \" \"+ df_species['description'].fillna('') + \" \"+ df_species['method'].fillna('')+ \" \"+ df_species['notes'].fillna('') +\" \"+  df_species['keywords'].fillna('') +\" \"+  df_species['locations'].fillna('')\n",
    "df_species['text_clean'] = df_species['full_text'].loc[:,].apply(lambda x: cleanhtml(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 384,
   "id": "94390681",
   "metadata": {},
   "outputs": [],
   "source": [
    "def remove_words(text, words, replace = \"\"):\n",
    "    words = list(set(words))\n",
    "    new_text = text\n",
    "    for w in words:\n",
    "        new_text = new_text.replace(w, replace)\n",
    "    return(new_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 435,
   "id": "31363244",
   "metadata": {},
   "outputs": [],
   "source": [
    "allnames = allnames + ['caribou', 'bee', 'tick', 'fumiferanae', 'spruce', 'hageni', 'cedar']\n",
    "all_tokens = [remove_words(line, words = allnames, replace = \"\") for line in df_species2.text_clean.values]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 439,
   "id": "855ea7aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "#all_tokens = [TextProcessing(line , lemmatisation = \"all\", selec_pos = False , stopwords = True, remove_punc = True, lowercase = True, output_joined = False) for line in df_species.text_clean.values]\n",
    "all_tokens = [TextProcessing(line ,  lemmatisation = \"all\", selec_pos = False , stopwords = True, remove_punc = True, lowercase = True, output_joined = False) for line in df_species2.text_clean.values]\n",
    "tfidf_vect = TfidfVectorizer(tokenizer = identity, ngram_range=(1, 1),preprocessor = None, lowercase = False)\n",
    "#tfidf_vect = CountVectorizer(tokenizer = identity, ngram_range=(1, 1),preprocessor = None, lowercase = False)\n",
    "vec = tfidf_vect\n",
    "\n",
    "n_folds = 5\n",
    "\n",
    "\n",
    "all_idx = [i for i in range(len(df_species2.index))]\n",
    "\n",
    "all_labels = df_species2['relevant']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 571,
   "id": "a57ce09c",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df_species[df_species.cited_articles.isin(found)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 572,
   "id": "d3bca1b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21"
      ]
     },
     "execution_count": 572,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df.query(\"relevant == '1'\").index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 420,
   "id": "4ed40ee9",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_species2 = pd.concat([df_species.query(\"relevant == '1'\"), df_species.query(\"relevant == '0'\").iloc[0:50]], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 430,
   "id": "ca510989",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_species2 = df_species2.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 440,
   "id": "802a3406",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------- FOLD 1 [79/20]---------------------------------\n",
      "-------------------------------- FOLD 2 [79/20]---------------------------------\n",
      "-------------------------------- FOLD 3 [79/20]---------------------------------\n",
      "-------------------------------- FOLD 4 [79/20]---------------------------------\n",
      "-------------------------------- FOLD 5 [80/19]---------------------------------\n",
      "--------------------------------------------------------------------------------\n",
      "accuracy: 0.67 (+/-0.06)\n",
      "f1_score: 0.66 (+/-0.06)\n",
      "precision: 0.67 (+/-0.06)\n",
      "recall: 0.67 (+/-0.06)\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0       0.69      0.62      0.65        50\n",
      "           1       0.65      0.71      0.68        49\n",
      "\n",
      "    accuracy                           0.67        99\n",
      "   macro avg       0.67      0.67      0.67        99\n",
      "weighted avg       0.67      0.67      0.67        99\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>31</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    0   1\n",
       "0  31  19\n",
       "1  14  35"
      ]
     },
     "execution_count": 440,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "#tfidf_vect = TfidfVectorizer(tokenizer = identity, ngram_range=(1, 1),preprocessor = None, lowercase = False)\n",
    "clf = LinearSVC(class_weight=\"balanced\", dual = True,loss=\"hinge\")\n",
    "\n",
    "t = evaluate(all_tokens,all_idx, all_labels, n_folds, vec , clf, print_errors = False, formated_output = True)\n",
    "\n",
    "t['conf_matrix']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "06682af7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['conf_matrix', 'scores', 'ids', 'pred_labels', 'pred_conf'])"
      ]
     },
     "execution_count": 194,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "81c4e1d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_coefficients(classifier, feature_names, top_features=20):\n",
    "    coef = classifier.coef_.ravel()\n",
    "    top_positive_coefficients = np.argsort(coef)[-top_features:]\n",
    "    top_negative_coefficients = np.argsort(coef)[:top_features]\n",
    "    top_coefficients = np.hstack([top_negative_coefficients, top_positive_coefficients])\n",
    "    # create plot\n",
    "    plt.figure(figsize=(15, 5))\n",
    "    colors = [\"red\"if c < 0 else \"blue\"for c in coef[top_coefficients]]\n",
    "    plt.bar(np.arange(2 * top_features), coef[top_coefficients], color=colors)\n",
    "    feature_names = np.array(feature_names)\n",
    "    plt.xticks(np.arange(1, 1 + 2 * top_features), feature_names[top_coefficients], rotation=60, ha=\"right\")\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "dbc7bde3",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 441,
   "id": "b2a34e82",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\Anaconda3\\lib\\site-packages\\sklearn\\feature_extraction\\text.py:489: UserWarning: The parameter 'token_pattern' will not be used since 'tokenizer' is not None'\n",
      "  warnings.warn(\"The parameter 'token_pattern' will not be used\"\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA3kAAAFnCAYAAAAbs4AXAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAACUlUlEQVR4nO2dd7glRbW33zWJMMAMYYYwhAEkg6QBSZJzzhkJCkiSJFlyznEEhpyRKDkjiCQZEBBUlIv6iXgVvaZ71etV6/tjrbbr9OneJ86cM3t+7/Ps5+zTu7q7qrvCWqtWrbKUEkIIIYQQQggh2oMhA50BIYQQQgghhBD9h5Q8IYQQQgghhGgjpOQJIYQQQgghRBshJU8IIYQQQggh2ggpeUIIIYQQQgjRRkjJE0IIIYQQQog2YthAZ6A3zDXXXGn8+PEDnQ0hhBBCCCGEGBDefPPN36aUxtT9Nk0qeePHj2fy5MkDnQ0hhBBCCCGEGBDM7OdNv8ldUwghhBBCCCHaCCl5QgghhBBCCNFGSMkTQgghhBBCiDZCSp4QQgghhBBCtBFS8oQQQgghhBCijZCSJ4QQQgghhBBthJQ8IYQQQgghhGgjpOQJIYQQQgghRBshJU8IIYQQQggh2ggpeUIIIYQQQgjRRgwb6AwIIYQQQgghxNTGrHvpUpqy+ZgSaCZPCCGEEEIIIdoIKXlCCCGEEEII0UZIyRNCCCGEEEKINkJr8oQQQgghhBDTNO28vq43aCZPCCGEEEIIIdoIKXlCCCGEEEII0UZIyRNCCCGEEEKINkJKnhBCCCGEEEK0EVLyhBBCCCGEEKKN6Bclz8w2NbMPzOxDMzu+5vdjzOzt+LxnZv80sznit5+Z2ffjt8n9kR8hhBBCCCHEtIlZ9z6imT5voWBmQ4GJwEbAx8AbZvZwSukHRZqU0oXAhZF+K+DIlNJ/ZZdZL6X0277mRQghhBBCCCGmd/pjJm9V4MOU0kcppb8DdwPbtEi/G3BXP9xXCCGEEEIIIUSF/lDyxgG/yP7/OI51wsxmBjYF7s8OJ+BpM3vTzA7oh/wIIYQQQgghxHRLn901gTqP2Ka95LcCXq64aq6ZUvrEzMYCz5jZj1JK3+50E1cADwBYcMEF+5pnIYQQQgghhGhL+mMm72Nggez/+YFPGtLuSsVVM6X0Sfz9DfAg7v7ZiZTSpJTShJTShDFjxvQ500IIIYQQQgjRjvSHkvcGsJiZLWxmI3BF7uFqIjMbBawDPJQdG2lmsxbfgY2B9/ohT0IIIYQQQggxXdJnd82U0j/M7FDgKWAocGNK6X0z+3L8fk0k3Q54OqX0P9npcwMPmsdAHQbcmVJ6sq95EkIIIYQQQojpFUupafnc4GXChAlp8mRtqSeEEEIIIUS70d098HI1ZmqdM5gwszdTShPqfuuXzdCFEEIIIYQQQgwOpOQJIYQQQgghRBvRH1soCCGEEEIIIaYDpke3yGkRKXlCCCGEEEJMh0j5al/krimEEEIIIYQQbYSUPCGEEEIIIYRoI+SuKYQQQgghxDROd10vQe6X0wNS8oQQQgghhBhkaL2c6Aty1xRCCCGEEEKINkIzeUIIIYQQQkxBNCsnpjZS8oQQQgghhOgmUtjEtICUPCGEEEIIMV2iYCWiXdGaPCGEEEIIIYRoIzSTJ4QQQggh2gK5UgrhaCZPCCGEEEIIIdoIzeQJIYQQQogpSm9m2DQrJ0TvkZInhBBCCCF6hBQwIQY3ctcUQgghhBBCiDZCSp4QQgghhBBCtBFy1xRCCCGEmI6R66UQ7Ydm8oQQQgghhBCijdBMnhBCCCHEIKS7M2xQzrJpVk4IAVLyhBBCCCGmClLAhBBTCyl5QgghhBA9RAqbEGIwIyVPCCGEENM1UtiEEO1GvwReMbNNzewDM/vQzI6v+X1dM/ujmb0dn1O6e64QQgghRE8w695HCCHalT7P5JnZUGAisBHwMfCGmT2cUvpBJelLKaUte3muEEIIIQYRUysoiGbZhBCi5/THTN6qwIcppY9SSn8H7ga2mQrnCiGEEKKf0OyXEEK0D/2h5I0DfpH9/3Ecq7K6mb1jZk+Y2TI9PBczO8DMJpvZ5E8//bQfsi2EEEK0J1LYhBBi+qY/lLy6YaLqNPEWsFBKaXngSuCbPTjXD6Y0KaU0IaU0YcyYMb3NqxBCCDFNIYVNCCFET+kPJe9jYIHs//mBT/IEKaU/pZT+O74/Dgw3s7m6c64QQgjRLnRXYZPSJoQQoi/0h5L3BrCYmS1sZiOAXYGH8wRmNo+ZD1lmtmrc93fdOVcIIYQYrEhhE0IIMRjpc3TNlNI/zOxQ4ClgKHBjSul9M/ty/H4NsCNwkJn9A/grsGtKKQG15/Y1T0IIIYQQQggxvWJpGow5PGHChDR58uSBzoYQQojpnJ6G9x/M2w6oLNNuvnpzzmAty2DNV2/OUVl6f85gzddgw8zeTClNqPutXzZDF0IIIYQQQggxOJCSJ4QQQgghhBBthJQ8IYQQQgghhGgjpOQJIYQQQgghRBshJU8IIYQQQggh2ggpeUIIIYQQQgjRRkjJE0IIIYQQQog2QkqeEEIIIYQQQrQRUvKEEEIIIYQQoo2QkieEEEIIIYQQbYSUPCGEEEIIIYRoI6TkCSGEEEIIIUQbISVPCCGEEEIIIdoIKXlCCCGEEEII0UZIyRNCCCGEEEKINkJKnhBCCCGEEEK0EVLyhBBCCCGEEKKNkJInhBBCCCGEEG2ElDwhhBBCCCGEaCOk5AkhhBBCCCFEGyElTwghhBBCCCHaCCl5QgghhBBCCNFGSMkTQgghhBBCiDZCSp4QQgghhBBCtBH9ouSZ2aZm9oGZfWhmx9f8voeZvRufV8xs+ey3n5nZ983sbTOb3B/5EUIIIYQQQojplWF9vYCZDQUmAhsBHwNvmNnDKaUfZMl+CqyTUvq9mW0GTAI+l/2+Xkrpt33NixBCCCGEEEJM7/THTN6qwIcppY9SSn8H7ga2yROklF5JKf0+/n0NmL8f7iuEEEIIIYQQokJ/KHnjgF9k/38cx5r4IvBE9n8CnjazN83sgH7IjxBCCCGEEEJMt/TZXROwmmOpNqHZeriSt1Z2eM2U0idmNhZ4xsx+lFL6ds25BwAHACy44IJ9z7UQQgghhBBCtCH9MZP3MbBA9v/8wCfVRGb2WeB6YJuU0u+K4ymlT+Lvb4AHcffPTqSUJqWUJqSUJowZM6Yfsi2EEEIIIYQQ7Ud/KHlvAIuZ2cJmNgLYFXg4T2BmCwIPAHullH6cHR9pZrMW34GNgff6IU9CCCGEEEIIMV3SZ3fNlNI/zOxQ4ClgKHBjSul9M/ty/H4NcAowJ/B1MwP4R0ppAjA38GAcGwbcmVJ6sq95EkIIIYQQQojpFUupdvncoGbChAlp8mRtqSeEEGJgsbpV6TUUQ2130/fmnHw4nxrnTI9lGaz56s05g7UsgzVfvTlHZen9OYM1X4MNM3szJs460S+boQshhBBCCCGEGBxIyRNCCCGEEEKINkJKnhBCCCGEEEK0EVLyhBBCCCGEEKKNkJInhBBCCCGEEG2ElDwhhBBCCCGEaCOk5AkhhBBCCCFEGyElTwghhBBCCCHaCCl5QgghhBBCCNFGSMkTQgghhBBCiDZCSp4QQgghhBBCtBFS8oQQQgghhBCijZCSJ4QQQgghhBBthJQ8IYQQQgghhGgjpOQJIYQQQgghRBshJU8IIYQQQggh2ggpeUIIIYQQQgjRRkjJE0IIIYQQQog2QkqeEEIIIYQQQrQRUvKEEEIIIYQQoo2QkieEEEIIIYQQbYSUPCGEEEIIIYRoI6TkCSGEEEIIIUQbISVPCCGEEEIIIdoIKXlCCCGEEEII0Ub0i5JnZpua2Qdm9qGZHV/zu5nZFfH7u2a2UnfPFUIIIYQQQgjRffqs5JnZUGAisBmwNLCbmS1dSbYZsFh8DgCu7sG5QgghhBBCCCG6SX/M5K0KfJhS+iil9HfgbmCbSpptgFuT8xow2szm7ea5QgghhBBCCCG6ybB+uMY44BfZ/x8Dn+tGmnHdPBcAMzsAnwVkwQUX7FuOpxRm3UuXUu/P6W763pzTl3z15pzpsSyDNV+9OWewlmWw5qs356gsvT9nKuUrP72nt5vWzxms+erNOcrXlD9H+Zry5wzWfPXmnMGar2mJ/pjJqxsVq4+sKU13zvWDKU1KKU1IKU0YM2ZMD7MohBBCCCGEENMH/TGT9zGwQPb//MAn3UwzohvnCiGEEEIIIYToJv0xk/cGsJiZLWxmI4BdgYcraR4GvhBRNlcD/phS+lU3zxVCCCGEEEII0U36PJOXUvqHmR0KPAUMBW5MKb1vZl+O368BHgc2Bz4E/gLs2+rcvuZJCCGEEEIIIaZX+sNdk5TS47gilx+7JvuegEO6e64QQgghhBBCiN7RL5uhCyGEEEIIIYQYHEjJE0IIIYQQQog2QkqeEEIIIYQQQrQRUvKEEEIIIYQQoo2QkieEEEIIIYQQbYSUPCGEEEIIIYRoI6TkCSGEEEIIIUQbISVPCCGEEEIIIdoIKXlCCCGEEEII0UZIyRNCCCGEEEKINkJKnhBCCCGEEEK0EVLyhBBCCCGEEKKNkJInhBBCCCGEEG2ElDwhhBBCCCGEaCOk5AkhhBBCCCFEGyElTwghhBBCCCHaCCl5QgghhBBCCNFGSMkTQgghhBBCiDZCSp4QQgghhBBCtBFS8oQQQgghhBCijZCSJ4QQQgghhBBthJQ8IYQQQgghhGgjpOQJIYQQQgghRBshJU8IIYQQQggh2og+KXlmNoeZPWNmP4m/s9ekWcDMvmVmPzSz983s8Oy308zsl2b2dnw270t+hBBCCCGEEGJ6p68zeccDz6WUFgOei/+r/AM4OqW0FLAacIiZLZ39fmlKaYX4PN7H/AghhBBCCCHEdE1flbxtgFvi+y3AttUEKaVfpZTeiu9/Bn4IjOvjfYUQQgghhBBC1NBXJW/ulNKvwJU5YGyrxGY2HlgReD07fKiZvWtmN9a5ewohhBBCCCGE6D5dKnlm9qyZvVfz2aYnNzKzWYD7gSNSSn+Kw1cDiwIrAL8CLm5x/gFmNtnMJn/66ac9ubUQQgghhBBCTDcM6ypBSmnDpt/M7NdmNm9K6VdmNi/wm4Z0w3EF746U0gPZtX+dpbkOeLRFPiYBkwAmTJiQusq3EEIIIYQQQkyP9NVd82Fg7/i+N/BQNYGZGXAD8MOU0iWV3+bN/t0OeK+P+RFCCCGEEEKI6Zq+KnnnARuZ2U+AjeJ/zGw+MysiZa4J7AWsX7NVwgVm9n0zexdYDziyj/kRQgghhBBCiOmaLt01W5FS+h2wQc3xT4DN4/t3AGs4f6++3F8IIYQQQgghREf6OpMnhBBCCCGEEGIQISVPCCGEEEIIIdoIKXlCCCGEEEII0UZIyRNCCCGEEEKINkJKnhBCCCGEEEK0EVLyhBBCCCGEEKKNkJInhBBCCCGEEG2ElDwhhBBCCCGEaCOk5AkhhBBCCCFEGyElTwghhBBCCCHaCCl5QgghhBBCCNFGSMkTQgghhBBCiDZCSp4QQgghhBBCtBFS8oQQQgghhBCijRg20BkQQggh+p2Ups45QgghxCBESp4QQojeM7WUKSlgQgghRLeRu6YQQgghhBBCtBGayRNCCOFotkwIIYRoC6TkCSFEuyKlTQghhJgukbumEEIIIYQQQrQRmskTQoi+oMAjQgghhBhkSMkTQogcKVNCCCGEmMaRkieEaF+ksAkhhBBiOkRKnhCi70wN90MpbEIIIYQQ3UJKnhCiI1KmhBBCCCGmafoUXdPM5jCzZ8zsJ/F39oZ0PzOz75vZ22Y2uafnCyGEEEIIIYToHn3dQuF44LmU0mLAc/F/E+ullFZIKU3o5flCiJ6SUvc/QgghhBCiLeirkrcNcEt8vwXYdiqfL8T0hRQ2IYQQQgjRBX1V8uZOKf0KIP6ObUiXgKfN7E0zO6AX52NmB5jZZDOb/Omnn/Yx20IMAqSwCSGEEEKIKUCXgVfM7FlgnpqfTurBfdZMKX1iZmOBZ8zsRymlb/fgfFJKk4BJABMmTJDkKwYXUsaEEEIIIcQgoUslL6W0YdNvZvZrM5s3pfQrM5sX+E3DNT6Jv78xsweBVYFvA906XwghhBBCCCFE9+iru+bDwN7xfW/goWoCMxtpZrMW34GNgfe6e74QA4JcKYUQQgghxDRKX5W884CNzOwnwEbxP2Y2n5k9HmnmBr5jZu8A3wUeSyk92ep8IYQQQgghhBC9o0+boaeUfgdsUHP8E2Dz+P4RsHxPzhdCCCGEEEII0Tv6pOQJMU0gt0ohhBBCCDEd0Vd3TSGEEEIIIYQQgwjN5IlpC83KCSGEEEII0RLN5AkhhBBCCCFEGyElTwghhBBCCCHaCCl5QgghhBBCCNFGaE2eGDi0vk4IIYQQQoh+R0qe6D+ktAkhhBBCCDHgSMkT9UhhE0IIIYQQYppESt70gBQ2IYQQQgghphsUeEUIIYQQQggh2ggpeUIIIYQQQgjRRkjJE0IIIYQQQog2QkqeEEIIIYQQQrQRUvKEEEIIIYQQoo2QkieEEEIIIYQQbYSUPCGEEEIIIYRoI6TkCSGEEEIIIUQbISVPCCGEEEIIIdoIKXlCCCGEEEII0UZIyRNCCCGEEEKINkJKnhBCCCGEEEK0EVLyhBBCCCGEEKKNkJInhBBCCCGEEG2ElDwhhBBCCCGEaCP6pOSZ2Rxm9oyZ/ST+zl6TZgkzezv7/MnMjojfTjOzX2a/bd6X/AghhBBCCCHE9E5fZ/KOB55LKS0GPBf/dyCl9EFKaYWU0grAysBfgAezJJcWv6eUHu9jfoQQQgghhBBiuqavSt42wC3x/RZg2y7SbwD8R0rp5328rxBCCCGEEEKIGvqq5M2dUvoVQPwd20X6XYG7KscONbN3zezGOnfPAjM7wMwmm9nkTz/9tG+5FkIIIYQQQog2pUslz8yeNbP3aj7b9ORGZjYC2Bq4Nzt8NbAosALwK+DipvNTSpNSShNSShPGjBnTk1sLIYQQQgghxHTDsK4SpJQ2bPrNzH5tZvOmlH5lZvMCv2lxqc2At1JKv86u/e/vZnYd8Gj3sj0dk9JA50AIIYQQQggxiOmru+bDwN7xfW/goRZpd6PiqhmKYcF2wHt9zI8QQgghhBBCTNf0Vck7D9jIzH4CbBT/Y2bzmdm/I2Wa2czx+wOV8y8ws++b2bvAesCRfcyPEEIIIYQQQkzXdOmu2YqU0u/wiJnV458Am2f//wWYsybdXn25vxBCCCGEEEKIjvR1Jk8IIYQQQgghxCCiTzN5oh9QIBUhhBBCCCFEP6KZPCGEEEIIIYRoI6TkCSGEEEIIIUQbISVPCCGEEEIIIdoIKXlCCCGEEEII0UZIyRNCCCGEEEKINkJKnhBCCCGEEEK0EVLyhBBCCCGEEKKNkJInhBBCCCGEEG2ElDwhhBBCCCGEaCOk5AkhhBBCCCFEGyElTwghhBBCCCHaCCl5QgghhBBCCNFGSMkTQgghhBBCiDZCSp4QQgghhBBCtBFS8oQQQgghhBCijZCSJ4QQQgghhBBthJQ8IYQQQgghhGgjpOQJIYQQQgghRBshJU8IIYQQQggh2ggpeUIIIYQQQgjRRkjJE0IIIYQQQog2QkqeEEIIIYQQQrQRfVLyzGwnM3vfzP5lZhNapNvUzD4wsw/N7Pjs+Bxm9oyZ/ST+zt6X/AghhBBCCCHE9E5fZ/LeA7YHvt2UwMyGAhOBzYClgd3MbOn4+XjguZTSYsBz8b8QQgghhBBCiF7SJyUvpfTDlNIHXSRbFfgwpfRRSunvwN3ANvHbNsAt8f0WYNu+5GfASal7HyGEEEIIIYSYQkyNNXnjgF9k/38cxwDmTin9CiD+jm26iJkdYGaTzWzyp59+OsUyK4QQQgghhBDTMsO6SmBmzwLz1Px0UkrpoW7cw2qO9Xg6K6U0CZgEMGHCBE2HCSGEEEIIIUQNXSp5KaUN+3iPj4EFsv/nBz6J7782s3lTSr8ys3mB3/TxXkIIIYQQQggxXTM13DXfABYzs4XNbASwK/Bw/PYwsHd83xvozsygEEIIIYQQQogG+rqFwnZm9jGwOvCYmT0Vx+czs8cBUkr/AA4FngJ+CNyTUno/LnEesJGZ/QTYKP4XQgghhBBCCNFLLE2D0R4nTJiQJk+ePNDZEEIIIYQQQogBwczeTCnV7lU+Ndw1hRBCCCGEEEJMJaTkCSGEEEIIIUQbISVPCCGEEEIIIdoIKXlCCCGEEEII0UZIyRNCCCGEEEKINmKajK5pZp8CPx/ofExB5gJ+O4XPmRr36M05gzVfvTlnsOarN+coX9NvWQZrvnpzzmDNV2/OUb5UlnYoy2DNV2/OUb5Ulp6e0x8slFIaU/tLSkmfQfYBJk/pc6bGPdopXyqL8jW9lmWw5ktlUb5UlsF1j3bKVzuVZbDma3ovy9T4yF1TCCGEEEIIIdoIKXlCCCGEEEII0UZIyRucTJoK50yNe/TmnMGar96cM1jz1ZtzlK8pf47yNeXPGaz56s05yteUP2ew5qs35yhfU/4c5WvKnzNY89Xbc6Yo02TgFSGEEEIIIYQQ9WgmTwghhBBCCCHaCCl5QgghhBBCCNFGSMkTvcLMbKDzIIRof9TXTDn0bIXoGWozYlpCSt5UpF06BzObIbXZYs7evpvunlekM7NxvbnPYKU3z21KtQMzm7EP5w6Jv1Mqb72+bk/r2JQmq8tTbPzIr92TvqZd+ti+0t3nUDzbwfTcBlNephXMbNaBzsP0wmCVfcxshoHOQ47a8eBASt5UwsyGppSSmS1tZmv08hpT5H1lQtui3Ug7N/C+mX15SuSli3v3uPzdPae7HXe14+rOeWZm8e7nAy40s9m6ex8zW6HovLvqNJt+705n290Boirg92bA68053XyPXzezo8xsWE+vmVL6V2/z1sU9VjOzCT0Vpnuj5GT3OKQn98ru2VX9KvI0Lu73r55cvyeklP5lZiPN7Dozm6OLfFl2Xp/eX38LJma2kZlt1NPr99EoMLSb/dKFZrYT9EzZM7NlepmvcV2146KvjO8b9+Y+xXV6kHa53t5nStKDurI6cHCMFT0ydPWk7L3tT8xsRE/Oq7nOoJFTzWw/Mxs10PmoYma7A1t3I12vjbJmNmvIMN06J2SecWa2WE/vOTUoymJmQwc6L1OSQdN42hkzmwk4wMw2A84H/hXHW1auGoWipVBl5WzEkma2bqtrVa6bzGwscI+ZfaHVPVJKvwaOAPYysxfNbL3u3CNLM6eZLdAdQSHrXIbGvf+VH+/i3NH5OS3SLRh/Dzezc7q6biaArGlmXzGzzbt7DrAt8E5K6U/dOcfcOnsysEvlOnXlyIWjrc1sx+LddCXwmdluwA5dKUdmNiTyNRNwlJnd1EMhYR0z+1p3nlmkH2Yx89mN9zgz8CSwFHCXmW3V1fWz+nRKKIdXmdn83czbOma2jZkd38UgtjrwpJldamZzdkeYNrPZI8nCZnZPV+8lzina/j5AYUTqrpC4aHcUg+wdHGNmj0U+uyrL0Pi7vplt0538ZPwL+Cf+DFvlK5nZcDP7opkdYmYH9uSZxffZsmvVjotZWVY2s4O7KVAvDNxqZjea2UJdvf/s3iPMbEzUgVm6KEeRr+XN7CjgYTPbr4tzRgI/B3Y0V6RXgtblj/PmxtvXoa2un6VfJD67AYeH8t6qXq5lZpuZ2YnAqnn5urjPfOYK9RFmNqIbfd7i5krRgsDXrAsjV9a+ZjCnWzNnlTrW5TnZfYYAMxTjWBfMBCwDfAnvxxft5hi5GHC2mS1QzWslnWXfR5jZeOuG4hZ1aQbgoFZ1qnqf6PeXinY8qlXfX3m+Q+uON51jZrPE+xzdVd4i/dbAuimlP3YjbXGP0Wa2lplta2bjW+UvK/8yZrasma3Rzfc4DJgAfJBfp4EiX/ub2brduX7WlvbHjQnrWAtDddSNrc1sCeBCYEwc704dmN/MNojyz9ON9EV51jGzXcxlnwW7cd5mwCFRvn92lT7O2djMvtSdtIMJKXlTgZTSX4Hv43toTAB+Hcf/CaVCkmOlQD2vmR1jZo+a2ZatGkrWGV6CCxe5spNaNeiU0m+AY4HNzOyzRR5q8mUppUdTSmsCDwCTzOwOM1u4GwPrXMDdwEH4rMsSLdIW5R8PnGVmz5rZdllZOg38mbCzG3CzmX3fXBBvGrxmArYzszsjT48VZWzKU/zdCzgBGAncFB15VxbqxYGNgdXM7DOt0haklP4MXAt8ycy2yMtYk7YQHo8BzsMF/T3N7AJroYiZC6qnAN9PKf3DzFYxsy3MlaZOt4m/lwHLAn8G7jCzi60L5cjMdgGOA/4B3B91oStuA042sz+Y2YTsWp3eT0rpL3h9PAd4BtjPzK41s+Ub8lO8y0OBJYG/AGsCfzKzUVE3msoyG3ABMD+wW5xfa7FOKV0KfBaYF3jFzA6L463aytrA5cD1wEcppX+0SFvc51/mysAewLezY13Vy0OBm4Efmdmh5gaYrgb+o4D3gZZGhOgr/hl5OA/4XVflqJTpr8AjwPlmtkfDPYrynQWsjL+LraIuN1rco3/5l7l1+gp8hn1i9GOF8t/h2WXCwPnA31NKf8vL2lCGScBiwN+Bb5nZaWY2Y4v3Xxw/F68DFwLHmSswTW2/yNeFwP8DPgXWiXyNbjjnf/D6dRau7J1pZmeY2dhWQnUY+fYDVjWzFVuV3cyGR9kPxdvlR3GNoq+qK89vgFOBE4Ef5eWz1krS3cB44Azgay3SFQzDn/E7wH+klP43y3eH957XFfyZXQYca2ZbNfST1fPmMrPL8Gd8gzUo7dFeijZ7Mz7G3GhdzISklJ7HBfC3gU2BL+Pj+Nwtn4DLIe/jSu7QFu+9eL/HAVfh9b9LYTfqxYp4fzbEglanxN8L4/q7AHe1un48r+Ex7h0ffdio1GAQzt7J/MCVwI3AGd14VgDLE4pUV2TPchLexjYCTjezg8xs1uqzttLTaxW8fh0KnA50OfsNnAnsgCv7reSjoj8eAxwJ/KZqdKp7P9n93wGG4+9lT/PJhDpj2oy4bvFN4HPAu/kzsdYGlZuBHfGx/LiQRWo9ObJ3uSReJ5cHrgDmiN9bGYd+DmxkZueb2czdGPMAZgY27Ea6wUVKSZ8p+AFWAuaL71filfB7wJVxbAHg8hbn3493eDcA/wM8AazRIv2uwCPx/XPAa8DrwII1aS37Piz+HgLcC8zZlB6YNTs2E3Ap8JP4O7RF3m6PsuwLPBPHxgMLtzjnYXww3hv4BfAisFSLvI3AO5VV4j6vAXcCq7a4x51x7dOAFYDhcXz7hvRvAAvhnfBZcWwzYPUW9xgNHBDv81RgXWCOFuUYnr2T7YCJdenj902jzMsDZ0edmhkX9L4W9/xyw7nHRt0agQsH7+ID35oN+ZoX+GZ2fKFI/1Nghxblfx4X+L4CXBLHNgc2bEi/P3ATMEtcey5gVmCemrRDao4tGc/5IeDiuI5V0gwDnsWFiwuAU+P4xsCOLcpyPm4UWBp4tWgT+OA0Y/7M8rzhws5LwCvAZk3Xj7RnAP+FC1UrEG0On9WaoeGchXCjwMfx3mfr4h4zAZOBscDngXvwtr8bMKqun8iOrRl15aBqObM0Q+PvAcCNlWPDgSVa5G2p7Ptn8b5zsbr3DcwHTI7vjwK7xfcdgeUbrl/U54l4u98v6tlP4/8RDeetC7xY5IOyjS7X8Jzy978IrrS+BezRIk+r4f3WjMB/AFfjQs+pwGca8rUx3o8Nw/unReL4acBnu9FePh/P+EF8xq3pvcwQf7+Ej0WdxpVqerz9/QcujO5L9PfATsCYmnMOwceKl6NM47J3NV9N+p2A2+P7a8D4+P4FYJYWedsYr//fi7KPjeP7AKvUvJfrgGNwBfSHeFs7BZ/daVXHbgCOB74KPJnV2WEN6S/Ax7x9gVfx9jU3MFer+8T3McDR8fxOjvea/15tO7NEOc4h6nwlfZGnpXAhf1yUfdM4vhiVMR/YCzg4+/85YPHs/yVpGPPjeq/j9fibxJiCe8HMU0k7JP5eGHXjVFxpfY7oA1q8+2/iysrpwGNxbCzN7X5V3Hj4ErAFNfU2S1v0CQsDV2Xl2hFX4O4DPtdw7vO4seoo4I44tjhRpxvOWSjy9mdg37p6UUl/OnBhfJ8x/o6oPt+Gc4dGOR7GPWd2rbsPMCrKchfedx0Wx+fEFcy6c/ajHCc+jPf6HnAR0Z815GkSsAku8xXta0Fg/Yb0Rb2ZAzekTeiq3JF+Jnx8XB6YO46t1+pZD4bPgGeg3T+4ULsEPoM3Oo4thXf8H+EzfF9oOHct4KX4/kJc4zrcjalWCI2OaxIusE8CtorG8pUWeTwzGt4huBD/Ei5YzBa//1uIwwWVy4Fv4YPJ/HF8GUJ4b7jHzMDV8f0RYJP4fhwVoSK712LAbTV5/Rewf+V4MRhtS3SO8f9QvFP7FbBy9R7xfXFgHlywfi6e3Zm4a2W1HMPxDniVeE4jsjLtUElbCLRzxHMbhw8k58b7P4kaoSXO+TIuHB0Snco7cY/xeXmz9JfgVvAPyQYQfADYiYqgl/2+Ytzn+7jA8hncgnxZi3x9D1ci5smOb0CNABp1Zzgu5GyAKziFwvIg2aBUOe9aXIm6DDgje7eT6t57fD8EnzH6cnZs3bjGzA33ORkX8F7Ojr0E7N6iLu+BC2CPAhtl7fwbNe9+QbwNLp39tjeuwNUq3pFmdnywPw43bByPWxH/QLS5Sr3PjQMb4Mrh/cCXWtxj1njnM2XH9sUF3xVr0h9LaXBZFxcgH6CFwBPn7YQLVXNkx7anwbiFGynexwXVq3BB/zu40DBnJa3hgurFuAD+QHb8PWCFFvlagFJJfxqf/d4B+D1wc5Zutuz7nMA36Fj3l8MVq2HZseL9rx/v7lTCcIKvnfll0/vH+6uN8b787jh2fzyPubJ0s2bvfU5cobsBOCKOLYkr4k1GgS3xfr8Q2IcAO+OC2VyVtIa3o5OiziyEj0k3EgYBapSDrE6OA3bHhfEz4u871fS4cD8MF6jmwdvzR3gbuL2hHFvhfcMlwDlxbHmy/rkhXytHOcbi/c0PcEPlO1QMJHg7fji+P44Lt1/EZ3YOa1HH5geeiu/PZc/6K8X3SvqZgHvj+03EOAfsSaUtZ3Vsvahjh+FGIMNljmup6cdwg+MkvA2ugNfHa4BtWpRjnyjvBEoj8mjc26LaJjfD2+8LkY/D8THlPLztPAks2XCfFXDD0dbAg9nxFwkjT00bfjO+PxrP6UTgZ8D5Ld5JoUA9BXw+vp8D7NLiGYyMZ3wrLvusRdZ31qT/DvBs5d2uGOWrM6LPjtfhxXADR2GouYHmvmKG7PsmeB1+tyhTwzmbE8pUduw44IaG9NviXk7rZMfG42P5Bg11cjG8f5oVN1LfhMsv36dmTIo6ey4+5p8HHBPHrwaub1GWEfi4dCDwXULOwdvx17N0hUw5Lz4e7R7v4+x4Xuvm6Sr3mDvy9gw+Nv4dH89eAW5pyttg+Qx4BqaHDz54Pou7n+yXHV+O1h3r/LgFaSNKa9P8+GA/d5auGCBH49bfe3Dha4k4fhedlaKi0g/DB/rd4pxrgTtwIeSomjw9jFtyLoo0T0YDG9mN53A0PmP2ePw/czT6pRvSXxiN6gg6CrZjyGYasuMjKQfp48gGBWDehnt8EXeNOC3e0zq4In0hMQuQPd/5Iu1luNvheXF8b+DblesW58yAd/anA38lZijwgflC6i1am+Od/da41e8M4JbojI6opM0Fy1Hx7v8nT0fD7Gq8e4v7bJwdf51QFLN6UpRnOVxAuRYXLNanQYis3OsoXEE5N/7fC3itIe2iuAB6J/B8dvwZKgaRLH8n4q4eBwO/jWe1VSVNrtjPh7etJXBr+X244n4o2cBckzfDLbTfxQWZBfF29zahGFXu8zqugPwBF2yL5zqMbNYvf0+48LlO5G8ornhfF8/8sDxt5flOwuvx4lG+3XHloG72c2W8nbyHC1A7ZL/lSt/n8ZmFxePdn4YLOk/iwu7HuGCRK7Ez4IpzUWdmwgWVL0a5xsXz2qByzsisPc2EK7XL4PXtJLwfu7hSjsXxgf5I3JBzahw/lxBaaLZofxYXKpYl6hmuLJ1D6X0xBDcejCEEM9zI9SHe7y0dz6KTEBbnTsbr+veBA+ved01dHo/346cQM36Rp4Mq6b+KC9BLxv8X4QawQ/B6/XTTPSPvj+AC0qeRthCQRlbyY3h9PQjvmybhSsH1eH9+Rl07yfqyHeIzFG8vh+J92jo1503EFce7CAs73h/slz3/6mzUWFyh+02Rf3wG/9Ca9liU6Uzcq2Zo9tvq8Qy3rcnXWLw9Lkooe3H8YbKxqea8YVFf7iIMVLhR4sfUG8WG4oanb9Ox73uLju2leL5DcKPbl/B2eB/eRpfM01XOWRtvUw/Fe3gx3uOPCONxvPP83PnxmdtPsvdyNpkxpOZZH4K3kz9TKsarAQu1eF6z4MacfxEGGrzu1wrTuMF8A1x5+nZWX66kNIhW68sQfOx6lXIGuJBFOuUtnu3JuMFpKK7cXY4rrJ3kCso2ti3u4fQe2WwRrRXDg+Nd3JzK/u0taoyUeH09B5fX8vH7a4RBveEes+MG1nvwcXYZfLws5MXq8/oM3k88gI9l8+Ieag+26L9+QTkBMCKusTlwdIu+ojj/DMp+/G5gy6ayRJoV8D74Xnxs+QzeFuapqZPn4m3/AVxxPTHq5/U0y0mr4H34hnj9/RZutJ+Dymz8YPwMeAba+UPn2ZbN8UH/eWDt7pwTx5aKSrlvNMyjqunxTvjKojPIjh8FfKfhXrPjwuMi2bF58Y5sZbwTXCP7bQ1cGLD4bdlo/P8N7NriOSyEW2VnxJWdu6NB3glcWpM+7ywOjXTHxf3nbHpOcXwUbqm5HB+EdqB0wymeSdEJbxXvYnvcR3t8NQ+Va18I7Bnfv4S7ID0T72ad/NrZORfhne5ylK438xMuSJXn/iW8w/omMetbvKfs+X+PUF4q5x9I6UKwZqT7kIYOEnflvQS3xO6MDyYz4gp1MeNaHejnwoX+objAcwYu7B1S97yycxaJa22Gu8M9gtfj9arPDO+gn42/N+MK22G4kvBQwz3mwAfSGfEBb09c6P8rlZm/7JytI+0suIBwAq54nAIs23DOupSK4wR84H83ynJK9ZnhgsSF+EzGD3Ah4Te4u+foyrVzo8AH8YyuJ9ZXVut85T5fxIW77eP6RX0ZTsye0bkvuhR3iZ0bt9Lfi1tbV6/kZ1N8YJwEbEM2M4LX4+FRZw6unLcJbs29Kf5fAhfgv4EPridX8rMGLsjfRMx8VH4fFu/5QVxI3QAX1l6mbHt7xfk/AL5O2e7rhPwtceV1CK4UXI/P0N5BuNJHuplxoW54vItt4vvn8b7jNuD0hne5b/Eb8CalG9fReD/VYRa2pswbAv+HKwg/pKPyPRMuEF6E16etcGFqDVwIuQU4qeG6I3ADxdh4J8fEs/gzYbiqpB+JK8RrUrqyF14ei+AzVGtn6Yv+dTV8vDs48v84Ne7Z2XP4Sry/NfCZmNF4m56hmja+zwecGd8Xx5WVyfFObq4re9ZfvJnVj3+7EFfS1Y0BM+IuoQ9Enek0+1F9n1FX3sTb2Brxbi4o7lGTfkyU4T58/L6EmNGtuddReD88K95/bY33/Y8By2XpFo33vSkxe52XGe8Dd4t7dTBAxbExePu6F+87zsP7vk7G5vi+U/ydJfL3H/F+h9eVo3gPeJ8yMz6u/AyfNXuOzFug+rzi2Dy4XLEM3h6+Xvm9+oyXxvvYM6Ms91Jf99fDjbSfA/5IRzfylZruQccx7SB8/eMjdJ4h7tQH4DLSVbgh8SEy19fKuW/jfdNbeHu+noobaNSvon4vG+l3iud0AGWsiKL/rip4I/D2Pxted0/Dx/BXaZYt7gUOie87R7n3afHeZ4/6ty4uBy0XZX8VeLSunmTteKn4eygu/z6Ot80DsvKvFGVYl6xPxPuMsfh48iBuXJqhcq9hVJRyXDG8qKk8g+0z4Blo109WEefBB+uFst8OA/5EZytrIYCMiIZ4OuE2gysAl+Mdf93g803gyPi+FG7hLSrwKpW0RceyMy4QvY9Pi1fdLq6rNIqR0QDXonTZmDUa9ewtnsUt+OC9Hi4srBr5W4mOnWFR/uHx3JbAhdmlcCHkG1TcT+joHrAd3gmNxWdbjsQHy60b8vUMrkgeQwwK+IDcyc8cVzpuIJtJig5kFWqsOZHvofgAOXs8o73jtwOouLbiA9sRUS/eaFGfLqCcDSuO7YoPRMPoKNAeTPje11zvTVyoPgm3aF8a73IholOj4wBxLD7gvE4IE3F8T5rXpBTrtm7EFaG14/gYmpXCo4ATsv+3wwXZzYEFGs4Zj88oLgp8N47NiStYnWbX4v958Tp/HS0Ej8o57xF+/lG/1seFmFE1dXgYPuO7cNxnnzh+LRUX5Mp5X6B0VVk76uaVuCDStCbnblzYPpJydnl7YqCtSb86rswUrqYz4IP+OXXn4ILhHfjM30R8Jj8XSjbH+5HqGqNR+ED9c7L1rVHP8hnomShn4f8PD1SwNKWCOpZSGH0bVx6uxGdHX8/bUHG96nOtKdNVlG5ahivKN+AC4qjquZSzmJfH31Xqrlu5x9q4svki5YzcHlSMbtm7PwI3ol1MqUwugs9s1q4bwQW3w6Nend7NfC2MG7/mBd6utIfCjT5/v/fhBrO/Eu6QlevdBJxYc/w5vA/4Mm6oOxA3YNxN51nsIfGsFseNJ4XRZDua3e7urN4XN06OouzDOlnn413eQ2fh7WIqBsH4fh7eNrbG+67RuOHjdCpr/siEdnwM3wd3u10Zb59Px3OYoeY+O+GK1DjciLQDruTsSr173xB8tn4BfFwoXDuPIZTfLO06+EzcL2jwBohjP8fb+/q4wHwT5TqpOeJ9fBGXR2rXeuLGrVvI+lV83H+DEL6z48X4cgjetzxD6XK7LC78j254/3vGs9w2u+934t3Omz2jqlx1Cq60jIvzT41nXac8Phzv7svZc1gLN5BVZYTiPl/GFY3b8f5kIbyPe5RMfsnSzx1140FKRWcZ3JDVtHbxQFw5G4kbNpbDjU4f0XFZSj7j+xJuzPwezfEGjI4eJTfiCvpZeP0fistmneIixDkz4P3I6riifiHeTh6iZglAnHMr3va/TUxgxLtZjsyIUCnPqKgr7+Fy4d64LDYnmSwa5d4i0v2e0i2zKGPex79FZVYe76sm4XLZqlne3qEb3muD4TPgGWj3T1TcifiMxMWULoCz0DxAXIMrbbfjgsyJ1TSV8xYn1hThwt03o1F9vSZtcY818UF4LVwYuRoXoL6Ypd2P0mKbC2Vz45bCK/Bp8mO7eAZz4MrEI/gAvnRDOfKF1LfgitEkSsF6c5qFnddxi+HdeKd0eBzfKG+MZG4PuEKxPy6IFo3+fmoCD8S9n4lnuwGVzqdF2b+EW7Gfi/9nwBWsz2dpitmTUfhAcC8+IO4XxzfDFTbDB6MxlXs8SjkrVggPC1bS5J3ZRsBd2f/zRX05o3JOUVdmxweGBeL/TfFZwsa1KJHugqhDc+PCSDFbtE5dXcaVtG/gne3mtLb6zp7lfRt8sJsDF3bXwAeW2jUG2TVmwtvaxTQoUFnaQ4E74/tOuKXxDroQqnHB4jJKofU+KkFtsrRjcCvp5dmx2eKdH1CTPp/JuyTqVXHscZrXcZyHt5ebcSGqqDOzULbBXPh8Cbfy744LF3fiQu9q8ft6hNs5pVFqR8qgCZtFfXmVmoXqRP8W738XXDC6ER+4l45nXQywy2Tn3YUP3k9kdWod4Pgu3slW+GzXZXRcW5fPGA2hXEO7LuFmFN9Pxfu+k+noEj4i6sTy2bGLIv87RlneIgsIkD3vefE2v1Fc/9q4/rp0Vp7rhPNF47yJ1XxV7rMwLgzPjredSbgx7YtU1ulE+jUogxm8lL3zzfH+aIZ4Z1UL+Ly4sDsDLoQuHsdvAnZueC87433Z29mxpwnDWqXOLE7m7k2p4C9HQyCgyru9jwialbXvJ2ue17H4WHd+PKtz8f6/VsCjbH+n48r9JbgA+w18lqo6G194X3wZn5UulJS9aFiv3XDfA/HxaZGob/n4UrSdnfFZyIdw5bNQhI6O80bixpOi7AfhXjqvE8pvHJ+/Wv6s3CPj/oWyXAT2aOWiODveLhbDZYSD4vgiNfUq91D6Pl6X745nXPWOqS41KOSqO3GB/6st8lSccwDeD71C6cZ8C52NC0X6JfC+bhXcXfOguG+n8mfn3Bvv/xbckHIvLQLFxTkr4m35MMKFHe+Hm9xaT8eV28XwsXwIPrZsRMXgkp3zbbxtfwZXQh+mxZrF7Lxd8fHnEsqx5S3qXVs3oVyz+kPCi4bmMbJ4ZqcBp8X3rXFF8QY6z8YPz9K/j/ex+Zr9Q3GldUkqa8RpDv42khYux4PtM+AZaMdPpSMqomj+OCrML6PCzVo5p1AyxuCKRnGNZfDB9c+ElbXmfiMp14ncjrsWzB3n1UYXw90UCuFqeNz39jjnYiJgRnb94+P3lePYWnGsVbCVWekYVWuZaPy/ALZreGYr4ELBbLi15CJ8kL2SzlaWohPfnZhZjP8LwXCzSvoVcQvlkCjzGrhC+BI+0OwMvFKTp8/iwt5suCXs9ni3a9V0KnPjszibE8ovruD+Gheq7wCuyd87LtCNxC2XRT3YIfL1Gj5LskJeTyr5u5DSilscu6WuvkS5L41yH0YojPgM5sPV9Fn5HyrKFseWww0DtT7puAB5JKXwPRwXzE6hIRhIpFkLV0IuwxXb5WvSDcMF9WOjrhQulDPh9fp+XIgr1j8OoRygN8SVgt3xGaGt4pyuorGti9fFb+EC34Zx//OzNMW7mzPqVhGUZ0Lk6SUiqELDPYbgg/1PcGEkd7fqFPmu8n6ewYWJzaK+Nblobxzv7Yh4Rk/hQl5TIIRlCQNF1j+dhwdV2KamvV+BCwPv0Dka3rH4QF6NBrgyPoBOolzvsxVeh28mM0hUrlcIEEfi9XkS3l46KQWV8z6Dr2d7EJ8FW5XO/fEwXAA5C599OjL7bSSlMjqquFe8v4vwtU1n433KGHxm5daoAwdX7xN/TyObfcEt4efifc24hnJch48pkynd4zbB+4NRDedcQjmrODPerz6AKxdFW80NQlvhCsTJwK1xbKF4v2Mr185dedeKss+MC+DL4H3GizQYb/D2e3WU6dSoS881pJ0XF0I3rhx7gRYGm3hP8+IGoRdxwe+aKP/ilbSj4p0VCuRn4zncSb3RpRiPxuD1dniWrzOpGB9wheB2fPb+IUrD1Q64ond3Xr5KH7M8rpifgrfp4VHHXiQ8PSLdwvgsmeHGsNlwo+6jce9j8bEpN+IeiyvnC+MGrVPxmYxr4/fnaI5cOA7vu9am41h1M82zP1vi4+kiZP1WlOcz2f/5rNRRlIaDpfDlHC9QMTjTtVz1PzSPk2PiOf8/3Jg0G658v06DEQHvFy6pPI97qLizV9p5YaB/Hh83bsLXJO7UVI+z89fEjQMH4u1hi2objv/3xWe0bqFccrIXDeNRvIun6bhsZLuok1UFf2hW1nVwg1Mxk264wa6TK2z8vg2uSB9OKSsvh7fHJtl1nniX+YTEiMjvStmxLfB2Z3h7N1z+ezqe1TmUxvd/y7vZ+XXB37YDruvqvQymz4BnoF0/lNHIFscHq8JH+CYa/Ovj9+OjAu5Kx6hue5BFiss6q2IgWTI6jMKSPgk4u8V91sWtGstnx86ktDwtnN3jOtxKchE+uF9MNywZ0YncjQ9GxQBmuEDaFO3xKFyw2JzSJXRSXGehLF2+ZuDz8VxHZ89jHyquinHtuXGL197R8JeLxv7TeF/VQdXwgfR+XDEYigu+19LZ8jMi3tvpuJUxD7KzAC5ML0P9Iuo58AAIj1Te8w6UrpFNA8u2+MC7AT7gfQF4q65Oxt/l4ll8HRdUT8Sth8V9hkU58vUH1+AC74z4IHsQmWJd88wew5XTF3FlqlBSZsrykQv7Y+MZFeu8tsUF0klkszdFnY938AweCOAAOs7IrEUWPIJSuR8adeULuLJ9XuTvb8D/66IuD8Hr89GU1ulnKAfVvCyP4LNW78ZzGo8PFp+h4tZMKRjOEp+ReB09GZ+Zu6l45jXv8bP4oDMWV6ovifd0ZvHM6ChozY0Ldy/hAtSRuAD3LeCKhnKPwYWa8yndJ5chmyWls+AyGbdIV4XaxpmJeEan4O3qJFxgGEGN6x3eN52A19vC5XIULszWbuORPbO5oj7MHM9vYtzzy3QWXmaP5/nreE47Uc5+TKDjWub8/V+EB4H5LT6TX9T9piiXo/D+9fdU3GWpWPQpleDto/4tigszP6R0JxvdcJ9l4r1Xg3AtSxn5OS/HjJQRFP+DMhjNtcTYQkfl4HORlxspt0kZiveHP8LbRafZf7yvPxk38A3F29kJ+LhRBG+om73cH++z98ENBTdTurLn7aWoN3vgSu1blELbepHvTq7g+Dj0W1zxnKmS31qFJX4/DTfKrld59q+RzWbg/UGxrvI1MsUR7+NOzK9RucczuGD8HOElUNQxOs/8zoIrD7fihr3ime5HGCuzc2fCDQVz4P3DQvHbvLhx7D0qHhJ4HV8x+/+YKNOieH9+ABVlvfJ+5ohr/4DSpf9gKmtzKfvKTXEF9ThKQ89IXMEoggdV+6Tj6IZclR3fCV8TuCButLsF788vIvr8LO3CWR1bBnd53Z+y3Z9KTXCi+G1BXG7bkDIg3TjckFA387Uo3k+eSmx1gytwV1ATrZiOY8X38XZcjI2vUXpJ1S0BugQ3NBXK9Fjc22t4zfVnwtvVybiCWrzHOfGYBHXXXxVve4/i4/hycfwOsiUbNedtihsoX4hnsGAc/4DKenq8Tp8aZS2Cng0ltvWhnLSo8yrqVvC3wf4Z8Ay02yer9MXahpnwgaWY1bmTyhomXHnYhzJy00RcuNsFt/Y1Cfcz4kLwObjAPgYX0D+LB69odFmJ8w/AZxhvxgWkH8fx71B2/PPRcV+0cfgC35/ROtjKENziuzc+y3gJ3nGeSOwdk6fNvo/GZwS+QumueBoRKS1LdzHewRUd/z24QLZ0Vv5di3eCK7VXRpm/HmU+CR/QZqZGAMuuvRTubjUxyl4IWuNq8j8nrjD9J+4KsCalcLwC2UJ7XDG/iY4+9CfjfvWXUxPGvPJ3JG5xK/zOn497307ZyeZC/hy4YLFS/L8qbvUvrFpFyOahlGuCjsKtmXNShrK/AXdfaZr9OZCY4cI72BdwxWO5hvSjcGvkmZGXk3Cr6ZyE1bHhvC/gg/c58e63jTLdUCl3odx/Hq+PxeA7Y9SNTWjYsw1XPr5Atn4Wb2OHEeH6K+nXIgLE4IPRN3EhuaWbC25IuQ2fVVklji0dZcm3Hije/cZ4270en/k7DupnrrJzJ1GG15+AK24vxPtaIStbdcBbEhcsbsYV45cIKyodIzDmQtjyuCv3L/G6uTMRITh/jlm7GBXve1N8Fuw6fLZt1ppyPI8rz9+NenNuXKMpQEchgK2OW3En4YLYEMp1TUc3nFss7j8Ib//Hx7N4t5KuKPvB8c5G4a5Q7+LC1ZaVMsyPu+IXM/BDKAO/3EF9eP0FcEPQ3VHmbbPfZsKF+E5h5rM06+OzrN/EZ3xr17hm9eAGXAhdGm/3d+PC1YP5e8/OWTny8AcqEZfjt4Vq6vG2uIJ6Hq7k3kSL/R1xxf82XGmZI97LxXidPKMuX9nzfQ2vy9dTKqnzV9Nm58xGOaN8H92YWcmuOQn3WDk73tvexExoJe2MuNHhVLzvPoUW237EOdtTBjR6l9LwcAAdt9go6v2suMFiE1xJuYyGdX7ZudviY+jjuLJdGFrmobMxZGNcmD4Wd1McHve5DxfgH6QSQTvL2964oP4Ebhi5Ae9b3yYzBGfvdTl89vU63Jh0Eg11Pq5bKCir0025Kjv/y2TKGTVxB+J5FOv5i+UMm+DyzkP42PQjMgNXXX3DlcMHcAPHnTS4kuJGyX3x8feyIg+VZzSE0thdGA/nwhXJ0+L53Uvn4DTFOynGx+VxmekMXKZ4muZ13ufjY9B8hBs1Xu9r6zLeX72Ey6yfxw0p90c9eKwmfbU9j8b72q/jRshvEO6b8XvV0HECruBeSxdLbehF8LfB/BnwDLTTJ2tkS+ADV9FY1sA7uxerlSQq+6FRWc/DLQiGC0W34q51m9FxMM19zE+OCvxiHBsbHU9VQSjytj9uaXsQD3yyIG4FOhTvQHcAns7OOwj34T6FjlbIdWi9QeXNuEI4Iy54HxSN+Elq9vmK7ydQzkRuhM+w3IgrPfli2pF4pzgLsX8LPrCcjQuAN9B5lm0cMbuHd7xfxQfir+OzM6tTERDovOn78vHM36SidFbutQjecZ2HCwh7xzP/fiXdrHgAiGfi+RZK47zx7v9I5wheRUf8ObxDvAIfHDaM443bGeBW3/twi/rNxMCLCzIX4B3tsln6UZGvR3Gr8XBcSFqWhplcvPN9mWxdI94e7iTcfWrOuRFX6nfAZ8CuwWeXvliTtnhHI7O6Mk/Ur+twxacInV5V7q+Jcp8a9avLdZW4InE5Pqj+gnKj7fHZ/XOFcjEqEWOjbI/QvPnxV6MerorX9VnwdjOqWu7s/8sp18EV0Xd/QiijdB4UCzfd6lqS28kiy2X1az5cgDoXV2wOxIW5C2i97cvyeHtcDhc2NsYtrE9Rv8H0kHi262S/zYgLaOfWPLNNgfvi+xu4sPotvE3Wznpk5z6JKzr7U4ZOX46O+83lgRo+jxu/Do53shJulZ5IjcAf9e02stmquN4LxF5elfQbxvt+gTJA0NxRX16mJnodLkyfh7uQPUfHfvF7VIJMZeVaDZ9xmIEyYvGJ+Ox/07qnC+LdjI//t8TrfRHhON/Yft/4vk3UqSIwyUa4Ee0iatob3n9tkJXtZuB/ifVy1I97F+GzBmdSWvGHZ++tbtZgHXx8mA9XDoqx+Wbq94QsookujLfpL+B95+00rGGK8+aI8s4SdeubeDChxyk9AIZmZTkSF8DnwMf50yg3XR9WU/YhuIHmLLzfLAJVrB3vv06BuAiv+7NG+b8cz/2SSv0p3ufulNE/N8UF3suIwBuVaxfK3+fx9vg2Xj9nwhXLsXQO6DY6/m4b7/E83H38LrxOH0PDmjRcmdguvq9AROPF22jeD8+YpTsCNzK0lKtq7jUP3jc+QYuNwuNdrxxl3w8f/1fFDcNfoeMsZ/Eeh+FeGF/LnseB+Fj1zepzjt9Xotzf71XK8ftomhXd2/D+cVK8801x4/uC2fvOjdTzR335Nq5M7o33xbsBm9f1E1kelsYV273i2EE0rxE8kY5rYlfAZbpVqLiaV+6zF94nbx7/FzObV0ceVqmkP4py3etc+Dj7/2g9U3gUPQz+Npg/A56BdvpkDfgMMh9kSgVhBSrrGLI0B0Zn8kxUsrnwTvJUaoQq3Mp4T3y/g4jahAswtbMG+AD9JmXUw20qv8+Cd4iLZmVZGu9Ar42GuRENG0tn1zkAeKJyrBgMqsFmioH5VMI6mf22ZDTqpqhMS0WDvx+35s5D6b7x7w6McsZgM9ySPRkftI7GrU+1VtqsQ8j3uFkOH1zHV9558Xd53DJb7DmzQVznuvx54x1aoSQsiyuDT+KDRCGArFxX7vjt+SjPOVFvnsA79E6BLeL/lfABa3i832Oirl2FW8eXxTv1PDDNYVFn1sIHn4eICKEt8rVAvIsf40LiAtlvhctu1VWncO16AhckF4i8XdBQV8bjMwuP4IaTYtZyJJ0DTlSV+6Pxunw1rriuVX1W2bmfI4KtxP/b4YrUj6nfp/ICfPD9YdSp/H2PaLjH0HgHS+KC8XFxfDcqe8Jl52wVZd+FjoLg9lQi9lbOWxlvK/vE+x6Gz258FhcsxmdpH8cHtlOiHlwKLFq5XlHufEuSb8e5V8YzLtZ8ztdw7vnZ+18LFxRPwgWyUTX1ZW18IN+Jco3QalGuVsEdxuBCzqy44l60z2vIwntndWxJXDn9Ci54vk1ljW/DfTbFhcg5s2MXEQFL8ucV34+LMv8Zt0YXZa7OfHQIJoK3gW/gVvkr4jqXNuRpSDzT+/BxZiZcebmMmA1oUZ7DaHA1y9Isis9ejKOcmVwKV1iuxJWETq7deP/1LD5e5u1pBWpcoujoJrYA3md+jBs661zgi3c5Oz6OFS53RVTZbem4nUBRj/ePZ/sGbuAq3BiXo7L5c+W8XaMevo4rYdvh7Xs7fNx5KcpWpN8CV3Lyci2C98N7NLSXAyj3cv0Fsa0JPh7s1eIdnUtE7Y3/16BmK5747XVKt8ct47m9go9P+Wz5LLgB6Kt4f7wQPht/I65wHllz7fH40ogiavDycXwuXLk8h+Z13ovis+JPVOrLztQYXfF6PwdubLgEbyNz44bIU6iXq9bH2+9W8X0Y3n4Pb1G/inezebznW/H2UKeoFWm/Hmnvxme+j6let+bcXfCx9UnKPnMpfLzJDdKHRT3aMq4/Dje2HUFNpFq8jhZum7fi7XZnfDy7ipp18ZXzd8bltHeICQLKYDpr1KQfgiv0/6Jhm5eG53xCnHdmlPlBSq+XzSKvh1C2r4Mpg0bNghtrZsAV8Ka4AIvSzeBv08pnwDPQLh9KJWaL6Ih+j3dYnab4s3Nyt7tXifDp+KB3K24Jaurw5sc7rxfoGBXsXSJaY80550SDXI3Y8BnvXE+mjKKZC/lzU3Z0K8f9rsUtwU0d0RB8dq2IwjZr/F2azoEYivLPgluuCyVgxqyMVdeQz8Tfz+FW0JlwIfUUXPC9EFeA66y5bxFKAK6s3h3nHE5na/ZC8feEeJd3x7M6i2wfrUiTL4Z/PfLwAt5JLtzwnI7DBc9VKd1ttohndyeZFZ/OCtvKRPAWXPhciDJyWu0MI64AFntKDcEtnStFXvN3vgYueH+DmA2N9z8fLqzcR2XdYsM7XTOe1TejrLUR73AhekI8izsJI0Dcp9Z6iiscp+PCzf74Wo5v0DlwQq+U++z8q3AXlX+vKczqRFFHc0HyPnwWdhy+TuVe3ABT69aaXW973Br9VnbsecJYE88ut05uiitSj+BGhFE1daTOom/4YF8ETnkRFwiWoGNEw8WBl7L/x+PK0DlN147jr1EOukvF+69VVCPNiEizc+RjIj7L8CzhLl6pU4dRhsBeDxcqNsEH/tp9nir3uxxf+3F61o7eoz7y3bWEwh3/74gbyGqDP8X3wkXqoqhrZ+Ltril4yES8jxiG94F34e6Yp7YowyqR72JdzbrxHj+hfl1ZUT8XxQXXS/CxpXiOnTYYx/uF1fB+ZRZcqLydzG24oc2vTxl2v4jouCI+y1Pn6rYs3o4fwJXPlWmIOB3fd8dnFvJIj8fjbn6L1uUt0lyNz0LtiCuFd+H9xhtUZn/xNvJy1M2zKev8ijQEs4nfZ8SVl6Xi2e0X9ymew2y4op0bDJ8i+vnsfc4S965zN56T2IAa728Oi3fzFNnWFtS3/UVww1mnGYzKvQqX5vXw+vsw3iYPpt51+jO44es/yZah4PX0l9SE6o96chu+p2c1ovObdNxzsdqvrY+PE9+kIeIwZZ2fOcpxbuT/MNwg8iWag/+MK957vL8f4Yarv1XLn93ncDpuVL8P3ofdSsfAJUX6uejo8rwC3nf+nhojQqRZI/K+R7zHPaNsD1FunTUUN2KdG+/tObLlDlEHnqXzbP8muCJ/PN4WcwP513BDx6iGsq9LWScPBH6Hy0r307CXHG6gmRMfi36K9+Pr1qXNzpkTl6uH4mP9EVHOnwFfy65bLI0ZgRsalsZll+vxMfW8ynXrPF66DP42LX0GPAPt8KG0aB2DDxArxefaaMz7dXH+AcSmzXgntWI05CfoGBjl3+v98E5oYVywvBzvaK6iYXo8zlsPFypeo/RV/xodZyuKxntidCBFAInt4/huNGzknl3jbDoHPXkJ2LEh/egox750tGp+k46uDsNxy+VD+AzbunF8KC7orosPTHVW3blxASTvjOfDO8LVs+tsQLl+owjqMBIX1t/CO68OPvDZ9Z6M8w/GB8pzcGH6VDpHbrK47n3x3rYk9rnDFZDavaHi3LnwDm0Nyoh3S+PWwaKDzgfulaI+/RSvo/lMQiGYFgrRKFzp/zMurC2UpV2chnV18fsReJ09I8qzCK7A3Eb9Pk874wPI8Cj3dXH+Nwh3upq6X5Rzjey32fC21mnLkPi928p99m6G4YP1HbhgvAUVIZqOgvEx8S6LGaIZcOPDudVnRmcL8FC8DV+Pt8/riEX42TnzR77Op1wvezjuinMJbixoucYku1YR3GURXDh9hMwVB3exqgaCWCLeVVO47RlxQWHT7NiceDtoiqq3Ey64/ySe8/g4/grlIvniGc0W9WjB7PxDIk+d9h2snPs5XMmZGR+4P4469DTl2sKqG9qBeLs1yrZxI837Sx2Ju0Tdj1uMV4j3szU1hh68vl9NFigm6swtdHbpL9r0tngf9i1csD+Q0oi2dvX6RV+Rt3W8/UzEBaC6ICjr4YajIsjWJbgh5p91Za/WObxPPQTvo8+js+Hl367weH8yNN7PpMjXftRE1cM9H87F28lJlLNxXyJzz64p/3iyQGe4cnQevn5sg2recCPNpfHevpv99lLTu4/fJ1CZrcTH0CvoaCDKA7icTcUlHR/z9mi4x0S87yrWfhWK4eiG9HvjBtndcOVzPG4Q2JuaWabsvCOI9W7x/+ZkM5416Q+O+3wHVxLG4fW/05Yclbr+JbztP4T3y+vTed1uUV8OoVyjX7hdFuvExjfc4y68vVyA1/fbo+xnUR+1eSzedvOInkvgbtWrVdLmCtoPKY2Ta8Q5o4itIGruczQuF+xMxxm4fWnw3sHdK5/Ex/GLcePsA9Ss3cNnL9fGx+//ouN68mfp6GFSBFLbIZ7TZHwsXTJL826LOnYqWYC/KPeOeD/QyXsFV1Ifwg2zN+Ay2wn4rF7tlj9Z+98hnm2xF+7SkdcVatIPxeWvK/D+cj9KOahp/X23g79NS58Bz0C7fCgtWr+iVA5mwq2nP6RmQMm+r4BbjPK9bQ6qa8Dx2xGUPskb40riDfhMYO2idVxIHRkdw89xZWld3DpSrGsoOq458E6xiKa2Ae6idngXz6BYi7Ao3rnejVucjiZb5xdpT6ajxW4HXOjZDLekHEXF5TPSzYgLez/H11PkQuU21ARDyX7fN87dDleU1iATpin3CfoD2SbL2e8jsmdUFW4WwjveYbg1shhUn6TB9SZ71l+MZ3Um5QzokLq/xTnZ9xfxjuhVyr0B87Tz4YaGGfCO8g18tnCLavmyc1bDBZRj4z2eSClkNwVbOQwXPNfBB9HrKUO1j254Zq8Swnx27Ax8pqopEuFBeHu6FTdy5ALUsJryd6nc19xjGN7Jz4cPFl/GBYSz6BiiuXg3i+MDwbfwQWd5SsG8lQvhfni73SDqwe64sLQL5T5TQ/F+5JC49kd03JZkLN5urmq6TxdtdihuNKqzmF+OC/vbF/WzUu416DgzsTsuRByIK3grks0QVs5dhix6HqXAejGx7qSmv/gFHY1exfsuQtwPrbnPnFFv78MF91kpXZMXb3gm80eaF3FXtKXiGb1LRzex4h5bEJGKcQXkp2TrTRruMRoXvr9NaXAbhvdptVFIcUVjgXgWz+IKwVO0Vj6uxGcINsmOLYG3z8KDIO+PbsH73xkij4fggvFNNC81mAEf586Jd14I+ZfS0YBYCOwr4cacG/Exs/As+BI1m6rHOTfHeevgfcTEKP9PKF2Ci+vnLoXn4WPZMTSswc3e43yUSwDeo5wd3gr4dotnXNS/b+KK25Lx/9bA/Vm6VQgviHjX61GuSVsCH786KVOUfcnRuHB8H5UZ5SztmpRbiKyJt+FT8bp7Gy7Ev9NF3ZyB0ngwJM7pjqvyXLhHzUeRz07Bg2rOmR3vV/+Gyxh5+87XX9+MGxpepxwj56WyfCCrAzPFc8039r4J70MOrsnHKNxAfwMeoONZGrxwaurlwfh4cTrenh+iZhuPrDzb4J4RN+KGrsWomX3NzssN38cRBo7KNf+9ljj7bSSlMly8/2rQuxMpDaBD8b7+63i/PxEfW05vyNfSce3v4W13obp3UTn2Cu61cxben30X76M+T2WszN7dqPg7Iy5n3UAZ/fX6pvvFOz+4qLu40tZJpqy8/zPpQfC3aeEz4Blopw+lS8DLeEc8d1Tep1ucsz1uyXwMF6SuiIb/IfXWpm1wq8cF2bEml84l8YHkBOCFODYLrni9hisHRQTKXDBeLjqqeSgFqWVxIbP2XpHmqqxBLYALMS8Qi3Jr8jYCt7YVbmn7RsfyTHQutYFd4txlcJebh3Dr/Fdo2OctO69w77sEt1I/Q30Uu7vwQf5xSoV9XTL3rYbrG7HfYPw/W9wjt94W6+1WwIWAQ/DBai58MH6MLAhF/m5w4fNW3A3we/Fux+KzyE0zDDdQbuS8PG4N/zEeMSqfJS061KF0XOe1YVzjXZr3Kyt87NfPjm2JK7tNe3zNHGWpRh2cSOdtLKqD15hI90TUmXG0HiRbKvcN9fg23ADyMN6Ox+MDcycrIB7ZdUG8bV2CCwgn0mBhjnN2jXf4lXgfd9I6LPt4XMCZjCuHa1LOwi5LJpS1qqNN7y/+zohbYfeOz564Uv0YHQPpFILUdvHst8UVg5kpF6n/POrpdg33PJ9yg/ghUe9GxHObv1oWXAi+J657eA/KdjUutNyHCxPn4m2vaUZye8p9q1bABfeHcaVo/4ZzbqZjsJW547xFKukWwtvwNpSBRb6GW9vvwQ1wtzfcY6V4rnMB78axpfBZt6b9U4vADPvj7nRPxXO8gpr1e7hi37i9T807KTwarop3fRPe352BKzEzU+Nyjc++bo7PlN+bvd8ZqTdqfZ4sMiWuGKwXx//thZH9fgguOA/Bx7Cj8P7pSFzR6uR2GOdNoty+41a833sE72fWqaQdSmlQnBh5WQFXdK+NOvMy5az7UFxBHRvvf58ox2dxxfDncZ2WyhTeRs7B5YPT6bzMYDm8f9wDNx7mCvzGkc9Vqs+s4V7D8L6yy3VTlfNmJwuy1M1zlqHi8UTZz0zCjV8rxbv8A97mRlfTZv/vGml2p1yKYnj/1qlfjjqcx1G4ADfQju4i37tFXXkv8rgQPk40htmPujA03tFtuGFr02oZIu2CeB9xLO55cSY+Lu1aV+44tkek2x+ffFiY0jBUGJSKPrdY/nInbtAagdf/w3Al7NxKHao+55FRz67A5ZdtaYjZgBszL4nrvx3HVsIV71o31UhzFyGnUXqMvIq7XxZ9XN5fbIHLvMtnxxbBjbBr1NV9ehD8bVr7DHgG2vFDKbD/FBeON638nkexKoIUfIDPxJ2Iu2t1cm3EB80h0Yh/G5WwVcjsZfCB/X8I1wGaA0DsTqaI4cLRuXHPYXgH1qhERcN6Kb6PwmcXm9xOCuF0BC5UfRzPIQ8zPGPlnKLDXzg6i2L93rK4VehblNtWdBUaeTQubNVGSqLzJsvXRsOv7bjj2b1HGYb+NnxtzfNkAyQdlaf34rpX4sETCpejjWlW2ovZpHlwC91PqQi8dJ4lfAYXKo7EBcnj4tNklb8r8vUDOoZoX5CaWeLsvRyFu9zls0zPU5mpq3lud0X+5o069FrD9WfA28b5xCJ1XJF8lnB1bnGfbin3kXYd3No4FrfOnoW34XGVdIUwuio+IOTh4ifgAsJedWWJ79sREVHj/7NxL4DLGsq/Ej6DvwGutFyLCzLfpIWLdnc+2T2uwYX1m/H+62JKQd7yclfOPyee6f54nzMaV3Tq1mENwYXQV3HBdvdWecrq8Szx+TwuQL9KZW1JTXmWJ4tqibswvRjPbNe6e8X/V9JxhrIuKmSevy0IN/GsXjxCFgCLcnP1C3AB/fDst1lwxX0J6tcHzowLUzPh1vPHcAFtaWI9TJZ2drxPnZ0IspX9dgVu3LuZyr54lNFXf4Qr9qOrdb1yn03xmZLN6ThbtWDkr9ZVD7eMXxVl+S5lgI8ryQLgZOlniPy+jrf3kXXXrdSv1eL5XBLPwvD2cwXeh+6Slz2+L4uPIfNn19kIV8Lq1jouiCt/bwKvZMdXxg0eW1JGTK3WrwNxQfWEqBMjqTE6UMoJO+MK9M1ElFLcW+ZN6iMQj8RnH2/DDcfbtnpmXfUNdKEMTskPPta9RccZ9E1wb44P6OjJU7S9Yv3sE7gCtg9uGG5loD6ZkLmy69xCxdujUmfmx43ZmxBru3Fl4q1K+yne4z5R9yfihogxeN92RtM7wseSA/Gx9Tq8r/0UX9KwYE3Zt8f74oPw9XFF/z2CSrC47NzV8T7sGtzrp1gnuhKViQZK2W1zfDw+He9rlsANJDfTWiadCTccPEZpRLiprt7F332p7JcYx8fQcS/covzb0nFLlvsoZcVVGso/B90M/jYtfgY8A+38icpfGwo4fn8J75BPobTszkcmYMSxYn3KVXTc+PRC3KrVanP11XBL2JNx/or4APgo5QA7FB8UhuFuLStH3m/FrZE3xt9a96a4xh3RsSyHd2IPRINfopIuDyH8bvb/BdEob6Rz2PSik1wzntnVuPX71Op1e/mecgXyBLqxyTKdZ5cOxKNSFa6v69HZFfE7xHoSOu7psiwuXC/RVB5ccH6OjorwsvFuW22bsB0ucN4fndnsuJCZb2NRPN8D4j2OxRWO3+DCdG3nWLnPKFxQOyWew3nE7HGLc4bhbh6X4MLlnWSKT/6cceHm4qifr1TSfCZP2+J+o2mh3GfP4KLKczmFcAui4yA/JN7bB3jdn62LZ1SUZRd89vklOgrh46nZXD3+n0gZRXQEPtN2YVxnTH79Xtb9+YH3K3Xr3yHa695d/N056thduAJ9Ex5prRo2vc7ivAlu7LifyrqK7FntgPdDjxJtBrfmHkTs+deibMtF/c3d4FfEB/7X6Dw7sw2u5D+P903H4kaF2+nY7vLADpviLnHv4vV/R1zhe6UmP6Pieh/hRpF9iYAhuHFndCV98b7/3e/jQveVuDX/+1QCB+FK0Zn4ZvSTu3g+ufXbog4chRsRjseVlVoBH1d+Lot38yJuzR8Vvw3H+6pRDe/kiqgrRQCcZXCj0qi6/OF96RW4kLs7PjvRFPynWKs3BDdsXY33Lytl9bWw5uebYm8Uz/NNWgSWqtxrKN72/5uOxry5aXYF3w/34DgOb19XRT7/vYVPJf3s8Wx2wpcyHI63tTkq6YZRCvRfxQ0cC1EGgLqOilwxrXzwfi7fx3I23Ph2KFlUyuz3yyhdEPeM5zyRZqPQvnh/MpnSFXRU1NGm/f3OIFsDHs+/WFudR8os+tfReB+xJm6wuBdX1HanhfJZuXexFcBS+Fh4N50DFd0dv+9FGZxtfSp7Gmdt6zO4YjMKn2HbAx+H7yFbm1g5d6Wo93vhffcHlEsz1uxGOYbGO3qMiqJekzbfMqPwglqVZqX4CjpuyXIj7gpc6wYe6Xoc/G1a+gx4BqbXT3RU5+Cdfr7A+zkqIb1xC/5v8Ahq1WiTY4HjK8eKjiVfrzQLPvi/EBX/OzV5Go0rIY/gA8UMuMVwCZrXABT32gC3HP4UtwoNi0ayZ0P6vYBzK7/NjQu++zTc6zlcCNsbt8a/gSsjnSyzvXwnz9PNTZazvwfia7LWiw7pNGoCB0Taz+GzRP8NfKPy21XEmqeGc5fHZ3vyRc5z44LJmBbnDcWF0SL66/VEUBw6CnlD8c5+PC4UHRDHf4AHuWkSqjbHraCfxa10+8YzOJFSgO3KLWjWKMuoht/HEmtV8FmcwlXlS2QCfD+1y8/Eu88txDfRsK9O1POd8XZ1Pq4cFWt0Orm54Ir2j/B2dikuUH+VhgE1ztkJtyoeQkfX3xkphY4eK3iVe6yNW3FXzY6tFMdmq6QdHX+3pdwX6yv4AHknLsTsXzmnECoOiPJeQ7gz4rOG/00l6ACuePwAN3w9TfRztFj7gQuE82T/HxrPeR/KtWi74+32a9l7mRnvry6Kuvu/eB+zPQ3bUkRZH4/6/ia+r+V18b5Wr5Y9vq8aac6iDNZ1OZ330ByO9/uf4v1+h3WH+MzECpVzlqLjzNjLuGFpqzi2BTXbIeBW6xeydKtG/m6ixZ6IkXYjvA7fjguIW8bzqwvmtWk8s2PxPvYDfCbgGWKTZTq6XY7GDW+FUrY5PrtyNTXG0yj/U/j4tlwcWwZXqK6PMuWBaK7HvUgmZPc7Gu9jLqMh2ALZ7Bbep88T9/0ENxTcRk1IfNwA8AFe90/H+80L8TbWFJRse2LdEd5Hj43n3SHYDD6DdA5eZ/NtIWbDFevTadjIerB96Gzg+iyuIL2Gezs9ifch6wHPV9Juia/dOzY7NhZvo53cmvG2+lh8PwCXJx7DFbDLGvI3FlcA54q6cEZ81qdZMToEN6AsHOVYCF9q8iI9UL7p2Nd9SOl+OTTy8hW8X3uH0oX/PhqWmkR9z/eEmwlvRyfR0Stn7iJd1N8jst/WJ7xfelCOOXDX6fW7SLcnLufla+EfoH6LlU1p3pJl74brdzv427T6GfAMTM8ffPbsR7iVaVj830n5irRn4wLVS4QVg3IWsGk7g3Nwgf0CYuNPXIhdk9IqlEd3Ojw6nzXwQe5h3ELbKiz5kMjHeLwzLhadb0BN8JL4bWbckvVTuvDdpxzEFsAHxBnwtUzFmo9XqCiLvXwXm9LDTZZxoeBnkYd98M70L/iAVBVYc0FvZ1x4ewW3IM0Xz3r9SpkL4WAfXLApXHTexQeMRwmhrdU7Kq6JDzCnU1rEjGzNVPw+Oq5bzCKcQ8MgRGfXiBtpWO/Sx3czJ95xn0W2LhB3c12xn+81FB+0/h8ufF6CC6TVQDjzRflPwNf/zBdtZjIVq2nl+rsR67fiee8Qz+8OWgdTuAsfwIp1hX1S6irX/wxuAb4bF8KPwK2655FF/Y2/42m9z9UZ+Exj7ppcPLOlcQVw66g3+RrOuuir6+AW67nxGblCsL6Pmohq8duG8fdu3DAyEg+ccxmu9FyP91kv0TEYyY7xLvaO/3eifu+tpsAOy+NW5P9Hg4sybh1/HDcefS/q13G4ItrkPlzX7w/H22V1Xcl58f72pHQV/EI867vj2e+QlyO+j8KNjQ/hM3NL4u1gRyI4S4u6MyKe5y64ovd9XFEeVXlec1PuFXcrruhdHuVrCoBTBO/6blx3FVxpOZ2atoKPDQtHnXkZF1SLcW59XNmvrpM8Bl/OMInSeLEclaAxNXV5Jlyh2Btfrz0EV6JfBq6u1pf4fill9N11cAH7LFwpnqF6j/i+YLy/LbJjhwMTa97DWriL3svxPoplB3PiRpxOkZcH2yerL4b3lbtTzsLug/fH+8fvL9J5r98RuAHnPVx5Xr6L+71KZ5fMA4mgWw3nbIS7y28R+bkXl01OqKTLgz9tgc8UfY3SgHooDcH1unpGuGHhZDqPS6vggUMm4n3S3nTcmievj4tS7gm3CR377GoQlGuj3s2Gt9szibWF8fttwM798P7zYDsj8b7u+LjfzXgf91rDucvSgy1Zsvffo+Bv09pnwDMwPX8ow7RfjCt7t9BR8Ckq/J6UG//uhq8TuyoqfVOo+e1w6/cmuHXqxuiI1qGjtXRNfFC+n7BcRb7mxa2S95BFc6opwwn4AvW7i0aOC3zn0hwMZEh0QJfgA/nJ1OzBRMfBbmVcAVkE71SXxF2vJkLnqIq9eBdr07tNllfCLT97xzM7M577uGo5yAILxP8X4G5Vv6V5/eIwXADJo0PujVsmt6Ays9iNchY+9UNwAeJwfJY3n7k6ArcWX0O2X1rNtaquEbfgrhFn9SRPTXUk/m4Q+fkSHjziWnywvIyKAtLHtnhCPOe3KKOU7hjtpdOMZNTBYtb7uuz46lSsuZQW1flwN+PvUAZtMNzSWA02U+feuEPUretxK3Zf6nt1sDuEUoA6GTckfJ0yaEFeb9eneZ+r79HgshPX3wzvcx6PY/Pgwm5duO3Z45wPKWd6tgZe7KpMuDDwY7w9FnsnzRj1dA+8P8zXRj8S9fl13FK/UpSlycDTFNhhN7I922hWCleKuvzz6j2yNHX9/kS8nXXaNgIXunaJ36/CjYaG9yG7V+tY5dxhuAJ/DG58Oo/W68S+GPXj6/jswTy4UHYc9ZtMTyKs/7hh63zKoFyd3I3x/v6e7P8v4MpuHtq9bi+5TaL8N+D912Pxvofm5aE0dG2NGyvewmdiT87SdFJws/d5NT423Iz345dl1xxZU54tgX+Qed3gyvWuVAydWVm+EHn/Ft4eb45jH9AxaFbRV47BhdsN8fZ5F674XUdlucFg/WRlOTPKcD0+8zWRjlsbLEm2dKXmOnNQzmjdGO+/2uc1Bf+6jtaBQGaKev84sak23m92igocv11LqahujyvtO+HtuV+MlLjR5yZcKVo+nt2d0QaKpSdVo9BwOu8J12mLpEh3aqT5dtTZwli1Ct7nvE2LfSR7UZ77sme7SNzzy7i8vFBRHsr22KMtWbp4/52Cv03LnwHPgD4JXMgeS8OiclzwKNxPNsMHse9Eh16rgEQjXzYa7iQ8YMGH+OCUu3zNhgurf4rrVveIabUv2krRuMfgEZKKSGLzU3EhpBy45sSF2oVwoWJdXJD7Dp2FnXVx16ETKDf8HIJbPyfFvb8Sx3ss6GcdRI83WcYF1Yn4jMfmuFVx+/itU/jkOH5qdF7nEJEUcYX4NsoF/9WBaBN8kH+abIPounL0of59Dp+duRW3hM+KrxM5tOn909o1ojGyWC/ydgXlrOJauPD2Bq7kdtrIuZf3WCDe+eqR/6dxQbdpM/YNKJWUNyhns/ah87qqOQnLc3avW3A3xKbNfIu0S0YdOwdXcufBLdWnUBMdsYdlzhfqbxzv/Cp8dnQNXBlqdAWlh/tcxfet4x5vEeHJcQH5hhb52wpXqI/HhaWXKWfrcqV7ruz53hTfF4xz36Gj+9xoMoEEn12aJZ5rsR6zcFmasSZPvQns0KQU7kpzFOG839+UFv1+VmeKbWwOwfuvM3FjVSeX8/h+GllwnXg2RTTSmRvuMTuudKyGz7LdE5+9qVfWi6AuJ1aO3xr3OafmnLvxGYOhlMrT8TS4TWfnvUEp1M6D92M/iOtV9ypdHB8TR2Xt7QPg1zT0tZFuPJl7LT5DfRMtZmXwdnsAPsP0AOFWW5NudPzdFjcynIcrwo/jfdJZdAzYVNTJeXHlZBSusC+Oj/1PAs/1pa+Y2h9cfngnvj+Cz3Rei3tJHB7Hh9KNtWy4u+7hLX6vBv/anIaZoixvS0bdmic79gM6Bq7L3aovr1zjmKgv+/TjM/sMPiv3Q7KATzXpinyNpQd7wkWa7+OGkI1w+e1K3NAxEditH8pQ5G19wpCH9zUn4Ibe2SvpC+N+YWjv9pYsLd5/p+Bv0/pnwDOgTxcvyAXPZ3Dl40zc5eXA6CiqAUqKDn/B6KxmxQeUQhi+iXLdRT7QrxYd6Vdwy+fJuCXsTho2jozzDsCtOOsTEdZwJeEWmsNUP4Z32L+hjOI0X1wn3/9tOK7gPY27oFQ3Wd6Rhj3buvlcC4Gl2P+t25ssR5rlcWv0ZNz1aCIeUKEaOGRods3bcWHtb/FOv0bmx569v/zdzIwrYZfgM65f7Eu58+cbf1eIOjYvbpn7Pi4YtRxA6YVrRC/yuGO8/0PouFfQLP11j7jGV+i8rvV8Gqzf+MB9JD74FOsbF8SFslGVtAvGM14cVxrGx/H1cIH9hzSv43gOF5wfxi2o18YzmYH+mb2eAVfQ/gsXoDfH3Ve+R0PgiJprzE7DPleVdFviFvDCbXAr3PD0A7I9AbO6dRqubGyJC6v74saYplnC3fCB/jtUNvmOZ/069cJLvjb69ez44zSEz6abgR3om1LY7X4/O2dlfOZja8qZuTOiLGs3pH813sHRlH3CtZQz9HUK/tG4B8Dicf4C+FqwF2jezHllXCDcJ97vMHyG5rNRB8dnaVfC2/3Pol6OirpzLxEIiPqZ7qHxHvLIqTPgsxnr1qRfgUrQMtyo9kBTOSLN3Pj67dUreX6Qhn1qs3T5DNMNdJyNGE+9K/Qc+Ize2dV3n517DdkWAHGsWMc5Om9fg/GDGxiKgBer4W1mZWJ/QtyoeivZLHk/3bc7wb/y2evb8HH+G1G3DFcQOu29F+e8jCv23epPe1mG3Ni1Pi6TPEvz8o/R9GJPODyOws54X/ooLgM1ejn1oTwn4TLVLriX22P4WNA0Hj+Gj1370o0tWXr6/qf1z4BnQJ9uvCS3YkwmInhFhX65Ie1s+PqemaMDmohbWw8h9leKdEXHtQ4u4MwY564X57xLi4Wn0bE9Eh3vzygtzheTRZ2KY0Unsw9htccFwrGRx9r9wXCLzMX4gH8vbvUv1licRBd72HTz2XZ7k2VKIWhOSuvyLPE+dsMtrV+quccIXBgahytqR+Izsn8lZg4a8jYhOqAVKDfLvio6pE7h6XtQ5plxIW23eLa5608RfXAyzXuJ9dg1opf5zNeibR/PoF8FFVy5/SXewa+WHZ8IXNFQj2eLuv9/lNH8vkHnmYoi/VB8Nu92XIDek9Kl64vU+P7jLld3xffv4bNtN+OzYI2bX/ei/EvjwvfTeJveGxc0V+vhdZahss9V9tsiuCC/HN7mD8Jnvq+gXCOWCyqv44rEJZHmOCozRHQ0hBTumKfhCuudZLNjdLSw1ykH1bXRW9Pcv/YosEP8fhk9iPaXnXcE3ez34/fl4rldhisExbq81eisHMyHKzPD47zXcWHpCmo2y6aj62GxTcaRlOuLDqP1TJZFHT4PVwZfjHOWIPbMyvL1IC5AL4PPzL2K9wP5XoRNgaC2whXqc3EDy9ot3uWMUS9vp+zbT6N+s+xCWCzWuR2OGyy+EnnutH61G+3l8JrjrVyh36ReWR8bz7MYGwtlaUNaBOUaTB/cFf0MMtfVeEb3ZG3h61Pw/l0F/xqCK0az4AbAs+P4sk3p4+8iuPfUj2joH3uZ3+L6K+NLc+akHG9Wjbye3XDujVF/d6CXe8JR7u/6MG6A6atB90xKj51F8ImClynH10lk/QuuYM6Gu4jXbclyBT2YKe3q/U/LnwHPgD7deEk+4P17s2N84N+skiZ3vzqu8tu5NPhmx+Cwe36dqPDz0EJgj46usJ48jgtiF+IWylEN5xyHW2euIrY/wN0EvlVXluz/obigcjYexexbtJhl6+GzXYIebLKMz1R+BxdUTsFdCBuVoez7urhb6/OUSuOtlK5nVWvbNvis2pW45bBwxVmCbBF+L8s8BB8cfo6vc1u28vs8VCIYZr/12jWim3mbomvRGp5FEcjiI3xwOREXwosZpkLpH0ZpFZ8DN5x8L55Hp73qsnd6HOU+SoU1+CIyIb9abtz9Znl8RqdYl/XZOLfPC8JxAfhYYsF9vL/aPev64V4z4QrBE4SbZl7XK89qPToG11k82sxpde8dn0UqZvzWxwXDM3BF9Qy83/g1NYFdsmu0XBtdSdvTwA49VgqzdN3u97P/h8czPA2fCTyNGq8KfBbp6Pi+PK6kvIorYgvF8XycKPqs/fFZpe1xd+638FmXn3f1LOL8kbgwtQjebzwCbN5Fvt6J5/0wzVvyLIkrWzPghqoz8ciZd1GGYO+whg8XEmfABciPcSPKm1T2RaycdyHuqjYCX1d1Pm4QnUjN+tVetpduuUJXzplELGmI/2fGXVEX7kteptYnyncB7qJ7AD7OWby/n+PjbeHi3W/9fw/ytzLeZ6+FGx6Kcfrblfqbb6+yOO76PhR38X812kvjtlq9yNfFuHJzXuRxGB4I7HJKQ/SQrJ3MQbkn3JP0YU+4KGNt5OEeXmc4Pua+Ffkugq4U+9utHe2/eLbjor4X231dgfcRRRC6ZXAPmVEDXa8Hw2fAM6BPD15WuXnkSQ2/zwP8C3i2m9dbr0gbDa2wZq4ef5uspVvisxhfjf+3wS1Dh9NiITE+EH8jOqXRcewpmjcZPwOfxTkLt64vgitLx1JaLfvirjYH3dhkOZ7NcdFZX4ALrJ/Dlc5rcavWmnQUBrbE1+Dla2EKd6K74zqPtMjbZWTuFriC80wlTV+tZxfhBoDJuCvvDPjANLHFOY/SS9eIbuQnX8cwRdaiVe53EB2F7wVwY8gndW0MF/BexxWtvfFZsOHUR4bMXWPOIHMVxNdBnEln96rinE1w15gT8RnrH+AzQI9QrknpqyC5LN5e78EV1Vtxxeiw/rh+9Z3G96Oj7i/flBYXWj6M9z4qjq8BPNBwzuzRzm7CXftWjuPjo2zfICzUdDELTBdroytpuwzsEOl6pBQ23Kurfn8MLmxunB0bjbtQfq0h78/g/faR8ZyOwwXFTuuJ6bhO7Ee4Ul1sNfE8Loj3OLBHPLNNupGvr+GK4cW48FbMphXtZUXKEPvn4kah0bhxITckFOkPxw0ON8X1Z8Nn9BdpKH9x3smEd0ucsxVu9Jq5mqd+ajuNrtDVuoYbgN7ABffd4p1c0N95mhKfqFfzxPvaBRf0L6BczjEn5Wz9VCtLzTM+LepZEWBuC8KdtKauXIWPYU+RRf/G+8B1+5iv4h4bxT2Owr1vHsPHzHcoZbNqGQ6n3BPuDgbBnnBZeXaL5/su2f6s+HYZW8f3odHeNsfH46ujDXe5Jcv0+im0ezGNYGaGN4p/1vw2J27luAUfIA5JKT3W4lqz48rDsSml/4hjn8MH8QmpoXKY2QhcyD8KVxBOKc6vpBuSUvqXmQ3BrZx/MLNdcKHnh/is2O9TSnvUnPMlfLA+G1/HNjfeod6WP4umPLYoc3H9HXBhYg580+DTzGzGKNcMKaXLsnMWxQXJWXDhYdeU0m+iXFtFPr+VUrqpcq9huCC0F/4unjazIlLUorg74DtFnrLzdqZ0B7oxpfQ/cfxJ3Mr9fk/KXMmTpZSSmQ1LKf0jjo3H3WG2xV1IT00p3VNz7hy4wnIMPgP8pZTSu2Z2BR6m+ebe5qtyn+dwpWNHfPbyh3jH/Qjwz5TSP6rPrBf3GIULhRumlH4Zz7xoD+vjAu//4TOp/4UrWrfhsxhb4C5uf8Ktjy+klD5tuM/5eFSws1JKF1Z+myWl9N95WcxsLC6wP4nPQC2MK9F/w93ajuttmZuINr8aMbubUjq5j9cryjUf3pe8js++j8QF8plSSmc2nLsqruTOh5f/17gl95GU0p15va2ctxJeXxbG3+sDKaVfmNnMKaW/RJoe9xfdKOsyeB26vIt0c+BrC3fCZ4v2B/7V3fx00e/PiNexXXAF+eSU0s/M7Ak8UMnbNX3Mdrh79d8jLwk3cG2UUvpFlm48/u6+jitCk1JKPzKzdfD6MhKPGnlBXd56SkO+vovvlfepmc2WUvpT5ZxrcGH6Vdz4sghuqHkJb5v/yvr9mXDPk4vi2p/D++Lv4grcf2XX/Xd9if7i2/gzngUXLGcC3p8SbbJSvmXwMO835vmKOvFlfGz8Be6mfCTeX3wL3wLi71Oi3vcX0d+dghtDnsNdiMcQa+7xPvbhlNIrA5jHvXBj0efw8e8/8W03FsPHyhcq6T+HBxpaB++LjkopvWhma6aUXu5jXop6PA8+Vo3GPVC+h7fF3wOvpJS+F+n/3W+E7LUX7t75T9xAPT/+jP8vpbRnX/LWx/IshT+zG3DD6Q74eHxBSunJLP3QKMuseH2fDa/77+L9019xmenDqVyUwctAa5n69O2DW6zH4ArLM9nxffA9m96L35tm5c7E3SFOwgW9bwF7dfPehRX9HbzDabJmX4grLI/gbmJj8A5wJeq3ThiGW8fzbQM2xxty7fq9Hj4zowebLEf6+XAXpW/jFtONsnRzkFn/Kd37VozzDsKtTHfSDfcZ3EXvtngX2+JhijcmW7vSy3IXFrPl8HVYH+FCQbE+bBFab5exHFPYNQJ3aynWor1FuRbte/TvWrRD4hnPHXX/eXwWb5cszWbx7h7FB/ncIvsZPPDKrXSOJFu1nu5HGdm2KWJnUdeOBL4c30fjM8SXAgtU32N/f6Dz3na9uMZs+NrRIfha0q3wGZOL4hn+Dvc2KDbxzdc5Lkbsf4fP2E/EhaTzgflaPLMRRB+Hz3hNwgWYLzIFAgP08Rm3jPbXg+sMza63FmUfdR5ulHiWzO217nzcIj5T/H89ZRChqgtosU7sV3TcOHk0brCq3Rewt+Vqka/C06QaNOxdOs6UrxFtrZP7MT4bd0b2/8J4H3sDNXub0nHz9JPx9Xsv4YaIhXHFsnafvylYh4p+/PQo5/HAdxvSDto98bI8Lo1vDTERH1+KZSXLR7/Rci/dKZSnon2th49Ds2W/7YmPU8tmx4r+y6JuHIDLC7fF73Ph43m/rJHEl28UEceLLUlejjq6bl2/xyDeEw6fgT8vvg/Dlc/HcQWuU/ThKH8RhCnfkuUcugh+NL19BjwD+vTxBboQ/B/AH4j9iciCFOCzbcW6jjx87jq4ZWoW3HXr/rzh9DAPjYJLdJI/wGcYz8KtRldQKhZ1SuEmMZA+RxZND18Lt04/PLN16OYmy5QKWxGifSHcPfPBKMdyeTnoKIA8Qsd95C7Bg3x02rw9O38YpevPhviMzrv4YL5GHO+rq9438UForbj+83ThbhUDwZ24q+wUc42g41q0IpBBv61Fy+4zH648fYzPms6KB4K4IUtTvIdDop78Dp9ZzqPA1oa/j99WxYWBRfAZyRtxg8pxDenH4TNXT1SO39V0zmD7UG7JMCae23xxvGjvq1EKcbkr3dO4EvjtKO9n4vguuKB3JR2FqnwT8gdxBf3pODYjPgvU5zWig/2De1JsVzk2Fu/bi8BRjf0FLpQuHG25WMNT1ycX68R+QIstAPqxXI35omMkziIK6dtUAi1U+yNcabsLNzIclx2fgRrjG76NxgmUroLz4W74xZ5nXwa+MUDvfS7gjfh+HxHUA1dA1hnoetndd1x5B8vg8sqN+OzeknV1cSrn8VuUroKtZJYHiueO9/Uv47JOUVcm0k9LDfCZrkuouGLjBogLcFnhsEr/Oqj3hMMNSU+QLX3AZY26aPBN5b+Xmm2JpvfPgGdAn354iR7W9sUYgL+aDYpnUSoneYN/KDqhW3EBavN+zk/eIBfNB1981uthXGCeueH8WXAXrUtxZex63B/+3n7K3+x0Y5NlSqFiZjwISjH7MAPuqnQ27hJTd489cB/56n5WSwM7Vo4Vltnlcbeju+N9boMLO4fEYHMWvdw+ISvLsriSlweF2RtXPmoXUePK8MO4AH5rdL6XR/n7bMWm40bGezIF16LFNUbhsxCrEIpD1Lm3KaNzzYZbyI+ItrISbjG8Jp5fpyiqlbJ8iVL5yGfYV6VFlDVcKXwZny3eBxd0Hi/eO4PYMo8rqRfj7ryn4krbBbiLYp0QXdTJBSnXd86BC3hv4S6CQ+M9HdJwzyfwmZiTiM2IKQM0NO7zNy1/sv5iJyIYBxFcATcSNQaZaXHN4fm1W6TL1yJ22AJgCpW1U77wWZ/V6BhBdXe837yNGM+q+cLX3M5JaRh9h4bN7ivn7Y27j+6YHVsGN5AtXs3fVKoDM+DjwRFk61XxPqzfZlanUlnGRxsv9p1bA/cwuptsi6EByFcRnXz9yvGbiUBa8f880V/vjkfYHoF7MUzCDdWF8arfvAoot+zYB/ewGYp7PSyNz6532ueRQbwnXDzrQiY7E3cp/Tml/Fpty03lX6Au/fT8GfAM6NPLF9dRQHoK98deFp9leQIXet7P0hcCz8qUMyQr4xHSLsOF9n6xzlIKIQfiVtb3ojEulKXp5H4Vx8dHvsbgytWuuJXslrhGp812e5m3LjdZzp7x5ZSuBKvgys0WuLI4b8N9dsP945+nYW+fmo7ratySOR63bD1L6bq3ROR3196UO7vHUVFHinWXxfHhNMyU4YPVEfE9d404mnA/offufcUznhsP6nBpvJNJuKL7JHB+P7abzXH3wbtwZXIYruB9gdiUOcvTZ/BB51eUs08zRd35IZk7ceUeM0adnxUXOo+N4+vSUblucqEeiguWv8Jdrns8uz4QH7wP2iLqx4X4DOnluDJwEi5cdDLs4Irt+XSMvLlknDe6xf0WxNdfQMeNwy8Eth3o5zGFnnEeyGYjKkGScOXn2qmQj35xO+1jHh7BvRyK2ZJ5ok1fXX1euLHoIdxwdEO0xRPxWb2DunGvOXGj3VvZ/cbH3wExIuCzNb/Hg3wMwcelWwfynfQg78V72QJXhC6N/v6Y6D9nosVehVMwX9UxeV9cNtgDl0dWIYuumaVbL/r8dyi3SxmDKyMbUrMspa/5xI0V51J6PZ3exTmDZk+47P2vj8tKm0QbWxZX7k8nlmdQ4yHUUP7T4re2Mur19aPAK9M4ZnYD8P2U0mVmtjyuFG2NK0WPpZTeLxarRvrH8Uawafw/AndfnIAHAvmfPuanWEg7Mz4IX4ULy2OB/8ZnxJ5JKf05O6dYTLsFroS8i3eO38QtabPiQu8SwI9SSpf2Mk/L4gu6x+KzIwviC3VnwQOHdFoUHUENbsNnJjaNPIzDXTGOS1mQgppz58AX6G8W9zs9pfTXhrRbUAZo+V0cWxfvlA9JKf2xJ2WuXDsPIDAD/mx3xAWc93Fr3o/qFuib2XB8NubTlNI52fFb8UHvxymlE/uaNzM7EvhrSukaMxuNC5E74uGefxFp+xRsJa7xQ7x9XI3P3J5pZvPjgU1+H/UwD4RyMK507YorXYfgEUhPTilt3HCPcZHuHuDylNI6cfxZPADLC93M6+y4YWNvXIk5NKX0f70q+BSmsiD+LrzN/hJfS/kvfPD+eUrptEhftMkjcAvzbLjF+zl8P88/ZNeuDbYSv92Iu9W9kFI6wcyWxo1Cq6ZKgI52INrJpJTS/5jZSNyI9gruVjgWV2DuTSndMJgDbvQXMebdjxuuDkop/bEI/FNJ9wplFOh18Jmw1/E2OrmpX66532dxw9MjKaUD+68k3cfMlgR+iwf/WBVf87sC7u1xYfJAUn3uK6cGZvZd3LV6H2L/WbweX5lSenAA8lP0S/sAH6aUvhNBudbG688rwEMppTsj/Yr4jNNw3Fj8Y7wv+wh35f35FM7vSHwcHon3r6mrdx999MzA3/oiV/SWbMwfhQehegP4Iy4fPou73f+rmr7hWnXlb/t+r0cMtJapT+8/NIecPoOaUNBxzna4NfPbZK4qNLhO9iFvpwJnZv+vg7v53EhYumrO+S7usngp7hZ4Nz4TthVuqdyCXoQgz67fo02Ws2OH4bOlD1DuIfU6HTdZLixT8+LuYwdRuhoti1uRJ7XI20G4MPxQdmwxfK+kLkO6t7huHsp/Z2JtE+66dCDdWIeJz6zejw/Ey+IWwdfwtXIPka2R6WUep8paNHygPgdXQF6nDO5wKzEL1OLcuaJO/xQ3QmzaIm2xZuAvlBtZ7w081ct8LwN8pb+ew5T80HFB/Mr4DN138IAAhYW7qJMjcWPUXLiidkH0D0flbaty/aKdbYrPzlwZdecx3HXtCcptINoqfDY+U7UerqBciK/1nCP6pfdxN7LrBjqfA/BcDJ8N+DtZ4KTs9w2iPY4jglfhhq5f0DAb3437FWtGp4pbWNZm9sdni76FK/Yb4mPjzHn+BvqddLNME/AZm2I5xGzRrp9kCu3Z2c1nvBTuwTFn/D8i8rZg5TkPwWWD2fAAWePj+DbRl03CI+pqZqnjcy4mlw4kZtGjLhyGy2WXMoBuuu32GfAM6NPHF+hK2yO4ED4H7kL4AVk0vkg3MxHEJDqtQ3Gl6n5gXD/nqWmB+6yE21vNOavivth1Hf4efchLjzdZphQkZ4yOfWj8LQLYnEnDZuy4y8nhwG9w1579st+KPWmGVP6Ow90VF8XdIP9E6bKwSfHO+lj+2/H1mo+SbT6P+/A3Bg8proG7RpwX+XsxOuQl6GPEz+weU3QtGm4dfh6fGf6AUJqijr3Sg+vMTovNbKP+jsCDRhyGu8S8EWVZLa9f7faheUH8fdQsiMfd5T6iY+TQrfAZveXr6mH8nRt3l70i2tokXLm7hob+pZ0+uKvh2fha5X2jfyr2kvz3BsgDnc8BeC4z0GzcnAlf6/UY5b6DNw10nrtZrjyK7GPRt6wadeDruIvj8gOdz16Ua1bcwLMycH0cW4PMUNHXfr+X+bqQ0jg3Y/ydg5r18HF8Ntw4dTlhOI8+6iAa1m5P7x/cGP4JcEt2bATu9dRreU+fmmc90BnQp48vsItQ2NkAsSm+v8uVhFCPKxfX0s9+2XRc4P4hbnlcvybdSngQj7Hx/xxTosOnF5ssx++P4m59fyPCOOOhfa+nY2jtQlnbDrgvvr+KR+H8NT7rtXDl2rmC9xIupL6Ez1bujSu6vwDW7of38VncnY0QEjaK74dW89XFdUbG4LUIrgA/Qj8G7WEKrkXDLapH42sqnscFo4twxbLYdLdXylf2LjfAlbkH8HWux+DrK9fJ6tw0YWXvw3NuWhC/YLX8+Kz949EuD8iON60NrdtmYnbcin4l2drXdnvOlIanDfHZ6AUJd2Zc0N8bF5KmO+WuJ88w2uVjuKGncTZ+MH7wWbyXK8e2xWfDp/r6tV6WoWjDM0TbLaJXfwv3RHqbmMWb2nU5a2P7RD0p9pgD9476asN5w3F32SNwd+nTiS09yLal0adjPcA9i97BXfTXz37rEBNBn759tCavTYhNL8fjHdTZqWYT1Nh49Ew87PT9+HqnP9dcrjf3L9bk7IlHfVsMV3Ruw5Wps/C1ZVdn52yDBw8AV6iewWf/ngc+xWfbLki+CXJfN7/u1ibLmb/4EZR7ob2JK59DcSvwH1PN2igzuxyfMVsTd007xMx2w/fr2ibVrHc0s+vxmbEfAXfgMxv/wgeZJePvYSmliX0o+6z4uqdReDS4L5nZUHymcduU0k97cc2huHHgqd7mq8W1+3UtWqyRexP4Tkppx6gLK+NCxusppef6nmuIzacn4TOxo/HZ47tTSt/pj+tPC0Q/tBFe9rVwF7qXUkqnVdY6zhK/DY90X43vZ6eUnmlaVxHv8i18De1m2fE7gXdSSudP2RIOLGZ2FW5IesHMhuF9xJq4S+8JdX2MKIl10oviXhnPD3R+ukv0tzvgbs+/BC5OKT0Tv3XaIH4wkq13WxGfhf8hbkC9DXeB3wT4dUrp9QHMJmY2Lz6OPx2fRXHFbYOU0n9GmkLemR83aP8xpfQzM1sdL8fyeMCw1wakEIOQ7P0PwQ15f42YEAfg8Qg+xQOg/aEvsp7oiJS8NsPMhqeU/s/M5gH2TSmda2ZrAR+llD6JNOvgSt77wBapski9j/fv0QJ3M9sIX1uzOD6TdTu+eHlz4Je97fAzZW02fPZpZErp7QhmshMu5L+IB8b4pOb8yymjXf40nuMuuPV335r0qwE7p5SOMrNt8Vm9U/DB4eGU0gN5AJw4Z258BmI/fFbsoOQBUN7FBbkzokMcnlL63x6Wv+hQd8KjfK2G7zm2CR4M42Lgv1NKR/TkulMTM1sGH1iv6IdrbYAbOP4JnFoV8Pq6WNvMdsBnozaJ/2fD3XWWwwexv/bl+tMarQICmAcZ2hF3c3ozpXROKH3FOtZzmq/873d5Bj5zVWyYfgk+81sbQKgdMLPN8Nnot3Hj12/i+Cx4FNKP27XswgkD2N64EeUXeCTR9wc2Vz3DzK4GfoYbgZfHXU9/ghvEfhxppno9DqPzQbgHzp/wGfPR+NKLb6cItpKlnxmfiXoB92B4Fu+X/hf3lnliauV9sJO/TzO7CPcu+hDvx34WyvIheGAyGar6ESl5bYqZXYsL8x/ga1VuwzfOfSWl9CczOwb4v5TSZf14zw1wd8OL8cieK5jZSnhwjn3yGZNMCXsYd9UYik/hr4R3mk+llH6Wp+1lnp7Gw0zPi1tAT04pfRgK2yq4EnptSum9bDZvBXxW9Eu4T/6Gca0XgatSSvfW3GcBXOAsZuG2jXvOm1Jau0X+Zsaje16HhwN+HXd3OyyVUSX7Uv6H8P3v/tPMTsMV6lfwWZRDeqo8TsuENXwPXOn6Jx4x89d9tRrG7NX++GL7x4HjU0r/LyzWV6aU1upbztsLM/sRvvbuatyN+Cwzmz+l9HGWpqsIcUPxunwe8H/AZSmlS6Zw1qc6FeFoQdwrYDnc6+E1PNpwtyJDimmLbLZoFXx2fAHgxJTS781sOXwz9o9SShcPaEZ7QBh4tsfHnr+FMXopPOLxuymlmwYwbwvh48OGuDfRRcD/No29ZrYlsG5K6atmtgQeyG15fBuLy6dStqcJMgPfKZR7JT+FK/t34TPTf8zTDlhm24xhA50B0f+Yh73/BHfl2RNXPobh6/JWNbO/4LM66/TnfVNKz8VM3orAL8OdaEbg2apLXCh4y+H7rF0a+V4QX++1Hx5O92dF2p7kI1MgF8TdKHYJN51DgXvM7DU8KMZHeBCN9+I+/4g8XYwrAOsBXwjlaDHgkwYFb1goZFtHJ7YEruj9HndB+/eAXfPM/gL8xXxrizNxl8pvpZR+UZSjDwreJvhsyWeB/0zuMncZPlvyaW+uOS0Tz/9WM3sUF5Z/25fBJBPAF8HXxbyJ15s7zew/8RDnx0fa4WmQbn8wNcgG+TXwrVF+hQdeKATUc83sopTSOwBdvZd4l7eEkWgfYO8QtAbtNhN9wcz2wLcxOTVcwvbG2/UbZvYNWb/bj2y8uBnf23MW4KdmdnFK6UzgkBjrB2Tmq5fMTRksaJ/k7o//aWY/xWcmp2pZ8nullH5uZpfgS0cOxtfJPwxcmskURT82O+6p8CczG5VS+gDYz8w2x71lBOXzjWc2F278Xh/3YNoe+ANu1B+Dz6J22feLnjFkoDMg+p8Qcs7BLVJLAf+BLwh+HPgHrrBcNiUEg7Aqv467XzyED07faEj+C2CYmR0VitL/w8Pnfh/f+LuYJelpHooBYjHgIzNbOKX0XymlM/C1aTPgazLeSClNNLO5zeyEOGd9fB+/T1NKh+Oup//Egxt8pXqvsP4daWaHmrvovYO7om0Wlqn/ijx1UvAq3IxbAo/How72CXM3z7kJP3cz28XMxqWU/jA9Kng5URcuTO7W3Os+MAb9cXggh6VSSm/i9fdC3LgxB+7uU7TJ6ZIwcPwrjD7r4i6ck4E7kq/L2AQPmvJOT6+dUvp9GIn2AN5vp+cczy1FH7MFcLaZnYu7kh2M97PzSMFrP4p+yXxpxasppYtSSnvh6y83NrO/h1HjH9BzQ+jUJCvL2vh62lOANc3st+b7kJJS+lkxRk7NskT7WtLMjg4F7m/4FlNX4EbaefI8ZQrI1/FAK9sDu5rZCmY2Y0rp8ZAbhGMxW0tK6be4F82swEL4fngv4V5MN0Ri6ST9jNw12xgz2wv3Dz8Sn1X4WppKC4GtYYF7ZglbgFIQ3gt31XwTXy/3bErp4t5M21svNlm20rX1x/iMzO9w3/qnuhKgzIN4fB4PLjMaX+f3xbjv4r1VqPrDmhkubcXC5oVxhfNN4PHBLBRMK2R17Rh8pvS27LfFgc/hs9IXpZQeG6h8DjRmth0+u784brV9Bw/E9AQuRK2KRwR+pGnGe3rGzB7B3cfAPQNWxPc5vSNFwA25OLUfZjYjHshpYby9vJIiUJqZbZlSenQg89cdshmwcfgM/mP4TNfueITY+3EldqsBzOPu+Nr5T/HAT4+ZBwS5A99q51eRrnCfnYCvn/+imW0FbIm7jH8PD+T2m4EpyeDDzJbCFfsrU0rHZsevwA0Uo/AgLHsOUBbbHil50wlmdig+E/VjXJH629QW9DOheBzuh/3f+Nq32/Bol1sCT6aUJvXxPiNxS9vRuBvAvvh+PO/F9X+QpR2Oz5zNjm/IeQvlxqc/BV7rrmIcgn2xZ9W4lNLtU1P4ygahdXC3iM3w/XsuwpXQbfE1PNdPjfy0K5ngsioudO+Luz4/hNfj1XA3xDfw7Uo+GLDMDgKiPd6BbzFxSkrpUvNgUPPjM3v3pGko0uHUJNzHr00prRH/L4RH/J0fD8By+fQ+M9+OZGPl+ni/PSNuIHkT+Ekqo9ROE26a5oHMXsVnya5OKa0Zx1fF3eY/GqCxcg98u4MTzSNhn44/53lx9+j9as59AiBFdF/zdfX74Ea9gzWz3pGQsU7A3TFPTCndZB5UbXt8/9pLUko/laFqyiAlbzoiOqO9UkrXDnA+8m0DbsTdNmfFNxN9JKX0v31p8GZ2Ih40ZZ1UBi/ZCl+feE7VLSw6oe/hHfs+uJV8dXxN3px4J/QDesFADMJm9i3c9XNPPOreF+Ld/y++b890E2xlSmJmX8MDBT2FD2J/wYPnzAbcrMG+xMw+jwtBq+MGlxNTSi+bb31waErpv6YVgXVqEmt/ngLuBCaGi/HieCj9kcD9KaU7BjKPov/IlI9Z8X5k/pTS6+YBS3bGPXJuSQO8zUBPCBe8A/GgZLviHkUvm9lZwAIppb0HMG+v41GR3zGzTYHd8LWPFwLfqxsrwzPhbNzj59TCQGW+Nu+PUy/30xZmNiduRB+DR37Pje3q+6cQUvLEVMU6bxtwcErph2b2FvBQSun0frjH8riwXeyzNymOz9Ck4NS4tp6IB35ZI6X0zb7maUpjZrMnj7q2Nj6zdDS+x882KaVfmtkNwA0ppVcGNKPTOJmFfT3c1Xh23B14LLB/Sun4atoByuqgJNbl7Q4ci68JfyildIIG+ZIuXNrfwJ/fZXidW1FrgNoPM7sHH48WxwN87It74RwNXJeySLTTAma2GL72Crwu/w2Pqr1dSuknA9FXmgcwOh2XFdbHI2M+BsyVUjq7Jv3MeHt72cr93b4AfIy7dU5T72SgCPnsSXw5zD4DnJ22R0qemOpY520Dvgs8gIdV/mUfZ/F6tcly5RqFa+tk3OI0qGe+wsq7JnASHlTmWHyd0wMppRvN7LPArcBKUjp6T2V9yf24QLArvgD/O3hUtidTShcOXC6nDczMcPfoN0OhkZJHS5f223FX4K3wenYb7v62f/KAP6JNiJmir6SU1ov/98T3a905pfRhHJvm2kvM5m8JrI1vZ/Rq6uXa+37M0xG4x8uDKaWzzaNjnlS4k0aaot/fFA+Qdi9wabiYjsMDydybUnp26pdg2iT6/0VSSv8xLdblaQkpeWLAMLMD8UiUxbYBx/WmwVs/bbJcuebMwN4ppat7kpeBwHwvvFNwN7i/42vCvobPmL6P76d0fcoCg4jeU1lfcmVKaZ2YoToO+GZK6X0NXKIvNLi0z4bP4D2CR9tcJnkofdFGhKKxbkrpWDMbkVL6u3n05/9JKV0x0PnrCzGuDsdlzz/EsQHrK81sBmBESunP4Vb6XXwv3dqNzM1sLL7V0eq4oe+SFMFwhBiMaJ88MZDcjLsfjcYFml6RWQEvptxkuQjFOzqfWemu1TD5/nXTgoI3BHgJ3xx5SEppLuClWBz+Ndzl59qU0p0DmM22IZ73j3DDxOWU212cCHwmpfQ+DO6Q5mJwEy7tswEP4grdzplL+zIppfuAB8K4I9qAzFC5HR68az0zOygzMq6MR4iepolxtXpswPrK8NL53zDSrYrP6D0B/26H+6WUzjUPFPVRSukT4EDz4Gb3A+ub2RYppf8eqDII0QrN5IlBQx9n8dYAtsZDTT+HW0L/ama34SHse7wH12AngjKclFL6qpk9BawAfAAcnVJ6I9IMSyn9YwCz2XY0rC95Adh2oNaXiPaiK5f2gcyb6F+yMWw+fPPtLfF9xG7AjXTfxbfj2WAAs9n2hAvhkBRbuFi5tdIHwDW4i/RkfCuLP5lvnfN/KaXLBijLQnSJZvLEoKEXCl4Riay6yfLXUx83WZ5GOB6Yz8w+g2+NsImZnQo8G1b+g2Vh7H9CkTsZF8buwdeX3CgFT/QXMePxFzN7HHcPK1zafylX4PYi6y+OAx5NKf0n8J/Asma2PfD/cEXj32PewOS0vYk2VSh4w4FPgCXxNXsTcXl5U2BVM/sLsAu+dY4QgxYpeWJaZmszyzdZ/hawNDDCzC7C3S/OhbYdHF/GF7HfBVwAkFI6PdaM3QT83MzmTyn9dQDz2JaklF4yszeprC8BJHyL/uRm+sGlXUwT/ByPpJozBlg6pTQZoA3HsEFJ8q1KzqHcWuk/8OBan8MDRi0GXJa0TY4Y5MhdU0yz2HS8yXKU/W94+P618H173gZ+kFL6XaSZL9YQCCHaAM3itS9mtixwDu6W+z18Bu81YIuU0od691Mf67y10tdSSq8NbK6E6D5S8sQ0jU2HmyzH7OUcKaXnzGwH4L+AjXGL48vAW8D7KaW/DWA2hRBC9AAz2xDfs21DfP+111JKF8gNfODJtlb6MbAT8Ld2kitEeyIlT7QFNh1tsmxmR+EzmIsDi+B79PzFzDYDdsBdCE9UgAYhhJi2iKA7I4BhKaXfxrG2G8emReLd7JVSunag8yJEd5CSJ9qKiJDVtpssm9m6uCL3fWAlYEbgI3wG70VcOFgvpfTYAGVRCCGEEEIMMFLyhJiGMLNxwEbAMsD/Ab/DZ/QS8BPg+ZTS9wYuh0IIIYQQYqAZ0nUSIcRgIPa8+yXwa2AJfB3eAvgagZ/i6xIXG7gcCiGEEEKIwYBm8oSYxjCzt4BdYl+2jYAv4nsEPgtck1L63wHNoBBCCCGEGFA0kyfENISZzY1vlLsgQErpGeAoYGbgu1LwhBBCCCGENkMXYhoipfRrM7sX2MfMZgNeAsYD/5tSenVAMyeEEEIIIQYFUvKEmPa4A2+7nwdOA34DXDyQGRJCCCGEEIMHrckTYhrFzEYDs+Dt+BcDnB0hhBBCCDFIkJInhBBCCCGEEG2EAq8IIYQQQgghRBshJU8IIYQQQggh2ggpeUIIIYQQQgjRRkjJE0IIIYQQQog2QkqeEEIIIYQQQrQRUvKEEEIIIYQQoo2QkieEEEIIIYQQbcT/BwTaVa3UlJMBAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 1080x360 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "vec.fit(all_tokens)\n",
    "X_train = vec.transform(all_tokens)\n",
    "clf.fit(X_train, all_labels)\n",
    "plot_coefficients(clf, vec.get_feature_names(), top_features=30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "id": "1dbcdeba",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "d = {'url':df_species[\"url\"].tolist(),'label':df_species[\"relevant\"].tolist(), 'pred':t[\"pred_conf\"], \"pred_lab\":t[\"pred_labels\"]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "21d73c9a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>url</th>\n",
       "      <th>label</th>\n",
       "      <th>pred</th>\n",
       "      <th>pred_lab</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://doi.org/10.5061/dryad.m0cfxpp47</td>\n",
       "      <td>0</td>\n",
       "      <td>-1.036231</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>https://doi.org/10.5061/dryad.q573n5tdx</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.453066</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>https://doi.org/10.5061/dryad.r6s1c</td>\n",
       "      <td>1</td>\n",
       "      <td>0.346018</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>https://doi.org/10.5061/dryad.jk680</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.422488</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>https://doi.org/10.5061/dryad.1c59zw3zf</td>\n",
       "      <td>1</td>\n",
       "      <td>0.100888</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>https://doi.org/10.5061/dryad.ft94b</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.859239</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>https://doi.org/10.5061/dryad.3275s</td>\n",
       "      <td>0</td>\n",
       "      <td>-1.015995</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>https://doi.org/10.5061/dryad.2f050</td>\n",
       "      <td>0</td>\n",
       "      <td>-1.187807</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>https://doi.org/10.5061/dryad.931zcrjg3</td>\n",
       "      <td>0</td>\n",
       "      <td>-1.034201</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>https://doi.org/10.5061/dryad.77r3p</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.516454</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>252 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         url label      pred pred_lab\n",
       "0    https://doi.org/10.5061/dryad.m0cfxpp47     0 -1.036231        0\n",
       "1    https://doi.org/10.5061/dryad.q573n5tdx     0 -0.453066        0\n",
       "2        https://doi.org/10.5061/dryad.r6s1c     1  0.346018        1\n",
       "3        https://doi.org/10.5061/dryad.jk680     0 -0.422488        0\n",
       "4    https://doi.org/10.5061/dryad.1c59zw3zf     1  0.100888        1\n",
       "..                                       ...   ...       ...      ...\n",
       "247      https://doi.org/10.5061/dryad.ft94b     0 -0.859239        0\n",
       "248      https://doi.org/10.5061/dryad.3275s     0 -1.015995        0\n",
       "249      https://doi.org/10.5061/dryad.2f050     0 -1.187807        0\n",
       "250  https://doi.org/10.5061/dryad.931zcrjg3     0 -1.034201        0\n",
       "251      https://doi.org/10.5061/dryad.77r3p     0 -0.516454        0\n",
       "\n",
       "[252 rows x 4 columns]"
      ]
     },
     "execution_count": 201,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(d)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "26665b2d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>url</th>\n",
       "      <th>label</th>\n",
       "      <th>pred</th>\n",
       "      <th>pred_lab</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>https://doi.org/10.5061/dryad.jwstqjqbr</td>\n",
       "      <td>0</td>\n",
       "      <td>0.159819</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>https://doi.org/10.25338/B83627</td>\n",
       "      <td>0</td>\n",
       "      <td>0.156061</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>https://doi.org/10.5061/dryad.vdncjsxvd</td>\n",
       "      <td>0</td>\n",
       "      <td>0.457681</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>https://doi.org/10.5061/dryad.5p6kr</td>\n",
       "      <td>0</td>\n",
       "      <td>0.228101</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>https://doi.org/10.5061/dryad.fxpnvx0ns</td>\n",
       "      <td>0</td>\n",
       "      <td>0.172458</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>https://doi.org/10.5061/dryad.72602</td>\n",
       "      <td>0</td>\n",
       "      <td>0.032584</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>https://doi.org/10.5061/dryad.2bvq83brd</td>\n",
       "      <td>0</td>\n",
       "      <td>0.656191</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>https://doi.org/10.5061/dryad.4k275</td>\n",
       "      <td>0</td>\n",
       "      <td>0.219829</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>https://doi.org/10.5061/dryad.tqjq2bvwz</td>\n",
       "      <td>0</td>\n",
       "      <td>0.503782</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>https://doi.org/10.5061/dryad.sj3tx967s</td>\n",
       "      <td>0</td>\n",
       "      <td>0.247829</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>https://doi.org/10.5061/dryad.np5hqbzvd</td>\n",
       "      <td>0</td>\n",
       "      <td>0.223405</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>https://doi.org/10.5061/dryad.008b59c</td>\n",
       "      <td>0</td>\n",
       "      <td>0.287187</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>https://doi.org/10.5061/dryad.xsj3tx9bb</td>\n",
       "      <td>0</td>\n",
       "      <td>0.437857</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>https://doi.org/10.5061/dryad.kwh70rz74</td>\n",
       "      <td>0</td>\n",
       "      <td>0.195563</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>https://doi.org/10.5061/dryad.bd124</td>\n",
       "      <td>0</td>\n",
       "      <td>0.250817</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>https://doi.org/10.5061/dryad.4v700</td>\n",
       "      <td>0</td>\n",
       "      <td>0.397456</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>https://doi.org/10.5061/dryad.tc45g</td>\n",
       "      <td>0</td>\n",
       "      <td>0.395403</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         url label      pred pred_lab\n",
       "51   https://doi.org/10.5061/dryad.jwstqjqbr     0  0.159819        1\n",
       "54           https://doi.org/10.25338/B83627     0  0.156061        1\n",
       "61   https://doi.org/10.5061/dryad.vdncjsxvd     0  0.457681        1\n",
       "63       https://doi.org/10.5061/dryad.5p6kr     0  0.228101        1\n",
       "65   https://doi.org/10.5061/dryad.fxpnvx0ns     0  0.172458        1\n",
       "68       https://doi.org/10.5061/dryad.72602     0  0.032584        1\n",
       "75   https://doi.org/10.5061/dryad.2bvq83brd     0  0.656191        1\n",
       "83       https://doi.org/10.5061/dryad.4k275     0  0.219829        1\n",
       "96   https://doi.org/10.5061/dryad.tqjq2bvwz     0  0.503782        1\n",
       "149  https://doi.org/10.5061/dryad.sj3tx967s     0  0.247829        1\n",
       "157  https://doi.org/10.5061/dryad.np5hqbzvd     0  0.223405        1\n",
       "188    https://doi.org/10.5061/dryad.008b59c     0  0.287187        1\n",
       "206  https://doi.org/10.5061/dryad.xsj3tx9bb     0  0.437857        1\n",
       "210  https://doi.org/10.5061/dryad.kwh70rz74     0  0.195563        1\n",
       "232      https://doi.org/10.5061/dryad.bd124     0  0.250817        1\n",
       "235      https://doi.org/10.5061/dryad.4v700     0  0.397456        1\n",
       "244      https://doi.org/10.5061/dryad.tc45g     0  0.395403        1"
      ]
     },
     "execution_count": 209,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = \"0\"\n",
    "b = \"1\"\n",
    "df.query((\"label == @a\")).query(\"pred_lab == @b\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "54071b9e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>url</th>\n",
       "      <th>label</th>\n",
       "      <th>pred</th>\n",
       "      <th>pred_lab</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>https://doi.org/10.5061/dryad.dncjsxm24</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.348457</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         url label      pred pred_lab\n",
       "127  https://doi.org/10.5061/dryad.dncjsxm24     0 -0.348457        0"
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c= \"https://doi.org/10.5061/dryad.dncjsxm24\"\n",
    "df.query((\"url == @c\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "01d1f80c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '1',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0',\n",
       " '0']"
      ]
     },
     "execution_count": 206,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.pred_lab.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "id": "6ed1bf1f",
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'cliff.api'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-276-38fd4fed2803>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[1;31m# import required modules\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 5\u001b[1;33m \u001b[1;32mfrom\u001b[0m \u001b[0mcliff\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mapi\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mCliff\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      6\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mjson\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      7\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mnumpy\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mnp\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mModuleNotFoundError\u001b[0m: No module named 'cliff.api'"
     ]
    }
   ],
   "source": [
    "import logging\n",
    "logging.basicConfig()\n",
    "\n",
    "# import required modules\n",
    "from cliff.api import Cliff\n",
    "import json\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from pandas.io.json import json_normalize\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "45df1a03",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
